<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Katharina Mikulik" />


<title>Discussion</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Abstract</a>
</li>
<li>
  <a href="Introduction.html">Introduction</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="HIV_CSF_dataset_analysis.html">1. CSF/Blood samples</a>
    </li>
    <li>
      <a href="TF_activity_analysis_of_myeloid2_cluster.html">2. TF activity Myeloid2</a>
    </li>
    <li>
      <a href="Triculture_csf_integration.html">3. Integration CSF/Blood &amp; triculture</a>
    </li>
    <li>
      <a href="Triculture_dataset_GRN_with_Scenic.html">4. TF activity iMg</a>
    </li>
  </ul>
</li>
<li>
  <a href="Discussion.html">Discussion</a>
</li>
<li>
  <a href="Methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Katharina782/report">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Discussion</h1>
<h4 class="author">Katharina Mikulik</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttree270aca65f4da298bf474a0f567f84d05abd61650targetblank270aca6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Katharina782/report/tree/270aca65f4da298bf474a0f567f84d05abd61650" target="_blank">270aca6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttree270aca65f4da298bf474a0f567f84d05abd61650targetblank270aca6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Katharina782/report/tree/270aca65f4da298bf474a0f567f84d05abd61650" target="_blank">270aca6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/Independent_cell_type_identification_Farhadian_dataset_cache/
    Ignored:    analysis/Triculture_csf_integration_cache/

Untracked files:
    Untracked:  README.html
    Untracked:  analysis/Triculture_csf_integration/
    Untracked:  data/triculture_seurat_object
    Untracked:  rebuild file.Rmd
    Untracked:  rebuild-file.Rmd
    Untracked:  rebuild-file.html
    Untracked:  rebuild-file_cache/
    Untracked:  rebuild-file_files/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Discussion.Rmd</code>) and HTML (<code>docs/Discussion.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/analysis/Discussion.Rmd" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
<td>
wflow_publish(all = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/f35ba45536a8b7546c4144a569fe908ce206aa4d/docs/Discussion.html" target="_blank">f35ba45</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/75af25c571d2f255bb408c63df6ee70c423b892a/docs/Discussion.html" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/3d3c73318704913e8bd8fd6101547ea25899045f/docs/Discussion.html" target="_blank">3d3c733</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/552f3e790fdab3c4f5386c882cd6d94f4d50d84f/analysis/Discussion.Rmd" target="_blank">552f3e7</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
wflow_publish(c(“analysis/Triculture_csf_integration.Rmd,” “analysis/Discussion.Rmd,”
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/Discussion.html" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/6f10f69a4503424be5a0ca9a1aff78b920c45951/docs/Discussion.html" target="_blank">6f10f69</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/c38fcb6fc45ce5f318ef7bc20c4cfdfbedda8697/analysis/Discussion.Rmd" target="_blank">c38fcb6</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
<td>
wflow_publish(“analysis/Discussion.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/6c5f3b175db2cc63caa1ba67e59c08e1a22f412b/docs/Discussion.html" target="_blank">6c5f3b1</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/d2f59b2d133fb7c2394b97182e5052ef1e4aadd5/analysis/Discussion.Rmd" target="_blank">d2f59b2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-27
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/d85a63137b84a09c19647281ffe67edbcce2e1eb/docs/Discussion.html" target="_blank">d85a631</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/9597c36bc5a6df5f071249d97949566b2c7db870/docs/Discussion.html" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/ea7e899746b03ca612d93ad003c875fc101d2329/analysis/Discussion.Rmd" target="_blank">ea7e899</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/40651d96647f10219a4aec953c8d618d57bc13f5/docs/Discussion.html" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/Discussion.html" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/Discussion.html" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/6dbe91df3b41c4456f433e2a7c27090249d67a72/docs/Discussion.html" target="_blank">6dbe91d</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/e01fe473a5abf3bc519ccfc9e701bd6306aa5077/docs/Discussion.html" target="_blank">e01fe47</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/e87fa963caeed78821ac7f482359072ce58b8529/analysis/Discussion.Rmd" target="_blank">e87fa96</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
new files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>GRN-inference on gene expression data from the Myeloid2 cells identified eleven TFs with different activity between three cell clusters in all three patients, which were also shown to have different binding scores in the C20 cell line. In all three patients there was one cluster which was different from the other two clusters. In this cluster of interest ETS1, REST and YY1 had an increased activity, while the other TFs had a decreased activity. Since YY1 represses HIV-1 transcription it promotes latent infection <span class="citation">(<a href="#ref-He" role="doc-biblioref">He G 2002</a>; <a href="#ref-Romerio" role="doc-biblioref">Romerio F 1997</a>; <a href="#ref-Lee" role="doc-biblioref">Lee Yu K 2020</a>)</span>. ETS1 binds to HIV-1 promoter to decrease HIV-1 transcription <span class="citation">(<a href="#ref-Zhang" role="doc-biblioref">Zhang Q 2021</a>)</span>. REST acts as a repressor of cell migration in microglia <span class="citation">(<a href="#ref-Yu" role="doc-biblioref">Yu T 2020</a>)</span>. Conversely, transcription factors with decreased activity, like for example MEF2C and MEF2A suppress the inflammatory response, meaning that if they are less active in this cluster, the cell might be in a more reactive state [<span class="citation"><a href="#ref-Deczkowska" role="doc-biblioref">Deczkowska A</a> (<a href="#ref-Deczkowska" role="doc-biblioref">2017</a>)</span>; Butovsky]. MEF2A is downregulated in disease and loosing MEF2C might prime microglia to increased sensitivity to immune stimuli <span class="citation">(<a href="#ref-Butovsky" role="doc-biblioref">Butovsky O 2018</a>; <a href="#ref-Deczkowska" role="doc-biblioref">Deczkowska A 2017</a>; <a href="#ref-Li" role="doc-biblioref">Li H 2021</a>)</span>. The MEF2 Family is<br />
expressed primarily in microglia <span class="citation">(<a href="#ref-Li" role="doc-biblioref">Li H 2021</a>)</span>. Lower activity of members of the MEF family does not necessarily mean that the cells are not microglia, but could be associated with the disease state.</p>
<p>CEBPB also had decreased TF activity in the cluster of interest. CEBPB regulates pro-inflammatory gene expression and has been shown to be upregulated in microglia activation <span class="citation">(<a href="#ref-Ejarque-Ortiz" role="doc-biblioref">Ejarque-Ortiz A 2007</a>; <a href="#ref-Ndoja" role="doc-biblioref">Ndoja A 2020</a>)</span>. Dysregulated CEBPB is observed during HAND, with an overall decrease of CEBPB in the brain. While neurons slightly decrease CEBPB expression, astrocytes and likely other brain cells increase their expression of CEBPB <span class="citation">(<a href="#ref-Canchi" role="doc-biblioref">Canchi S 2020</a>)</span>. However, there is no data from microglia cells. An increased expression in the brain might be caused by an increased response to inflammation, because CEBPB is involved in the inflammatory response. The TF KLF4 is also proposed to be involved in neuroinflammation <span class="citation">(<a href="#ref-Rickert" role="doc-biblioref">Rickert U 2018</a>)</span>. Additionally, JUNB, CEBPB, MEF2A, MEF2C, ETV6 and KLF4 are used as specific TFs characterizing microglia cells <span class="citation">(<a href="#ref-Gosselin" role="doc-biblioref">David Gosselin 2020</a>)</span>.</p>
<p>Because of the functions of TFs listed above the hypothesis is that the cells in the first cluster/cluster of interest of each patient are latently infected microglia-like cells. TFs which show decreased activity are involved in inflammation. Conversely, they show increased activity of two TFs which have been related to latent HIV-1 infection, YY1 and ETS1. These microglia-like cells could be latently infected cells which entered the brain as macrophages and acquired microglia markers and functions which they kept when leaving the brain and migrating into CSF <span class="citation">(<a href="#ref-Grassivaro" role="doc-biblioref">Francesca Grassivaro and Martino 2018</a>)</span>.</p>
<p>Analysing the TF activity of iMg and comparing it to the footprinting binding scores of the C20 cell was interesting, because in this case the infection status of each cell is known, as well as the treatment condition. It was shown that the treatment has a very large effect on the TF activity and, therefore, on gene expression. The effect of the treatment seems to overshadow the effect of latent or active infection in iMg. The two TFs RUNX3 and JUNB were more active in treated cells, while NFAT5, TCF7L and CEBPB were more active in not treated cells, independently of the infection status. An important insight to keep in mind for any further analysis is this huge effect that the treatment has on the cells. Comparing latently infected cells which are affected by a drug to non-infected cells which are affected by a drug might yield very different results than comparing them with non-infected cells which are not affected by a drug.</p>
<p>Finding a pattern of activity which corresponds to the infection status or resembles the footprinting binding scores was more difficult and even though some TFs followed the footprinting pattern it was usually only the case for either non-treated cells or treated cells, but not for both treatment conditions. The reason for this remains unclear.</p>
<p>Comparing the activities in iMg to the activities in Myeloid2 cells from CSF, there are only five transcription factors which are found to have differential TF activities in both model systems. Transferring the information from the iMg, where the infection status is known, to the Myeloid2 cells could support the hypothesis that the group of Myeloid2 cells described above are latently infected. YY1 is more active in the cluster of interest in Myeloid2 and more active in infected + EFZ than in uninfected + EFZ iMg. CEBPB is less active in the cluster of interest and more active in uninfected + EFZ than infected + EFZ iMg. Similarly, JUNB is less active in the cluster of interest and more active uninfected + EFZ than infected + EFZ iMg. This would suggest that the cluster of interest are indeed latently infected microglia-like cells. However, MEF2A and MEF2C are suggesting the opposite. No clear conclusion can be drawn from these five TFs alone.</p>
<p>Integrating the triculture dataset and the CSF/Blood dataset it became evident that iMg and all Myeloid clusters, not only the Myeloid2 share some similarities, because they are from the same cell lineage. Using the up- and downregulated gene signature of infected and treated iMg, 51 Myeloid2 cells were identified with high scores for the signatures. These cells probably represent latently infected cells, however, they did not correspond to the cells in the cluster of interest (see above) hypothesised to be latently infected cells. Consequently, two different approaches yielded two different groups of cells which are potentially latently infected. Since the infection status of the iMg cells is known and, therefore, the identified gene signatures most likely related to the infection, this result is more reliable. Interpretation of the cluster of interest from k-means clustering of Myeloid2 cells was based on TFs with different activities between the clusters. Literature search of the functions of these TFs indicated that these cells might be latently infected. Yet, the role of these TFs was not entirely clear, the number of them probably too small and only relying on TF activity alone cannot result in a reliable conclusion. Since the infection status of the iMg cells is known and, therefore, the identified gene signatures most likely related to the infection, this result is more reliable.</p>
<p>It could be possible to transfer infection signatures from cell culture to microglia-like myeloid cells from CSF. This would offer a new diagnostic tool for HAND, since it would be possible to identify infected cells. Nevertheless, too little is known about the exact functions and characteristics of these microglia-like cells, as well as, what role they play in HAND. Additionally, it remains to be elucidated in more detail how they are related to microglia cells and whether it is really possible to infer infection status of microglia, number of infected microglia and details on HAND pathogenesis from cells of the CSF.</p>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Butovsky" class="csl-entry">
Butovsky O, Weiner HL. 2018. <em>Microglial Signatures and Their Role in Health and Disease</em>. <em>Nat Rev Neurosci</em>. Vol. 19(10). <a href="https://doi: 10.1038/s41583-018-0057-5">doi: 10.1038/s41583-018-0057-5</a>.
</div>
<div id="ref-Canchi" class="csl-entry">
Canchi S, Rissman RA, Swinton MK. 2020. <em>Transcriptomic Analysis of Brain Tissues Identifies a Role for CCAAT Enhancer Binding Protein β in HIV-Associated Neurocognitive Disorder</em>. <em>J Neuroinflammation</em>. Vol. 17(1). <a href="https://doi: 10.1186/s12974-020-01781-w">doi: 10.1186/s12974-020-01781-w</a>.
</div>
<div id="ref-Gosselin" class="csl-entry">
David Gosselin, Nicole G. Coufal, Dylan Skola. 2020. <em>Single-Cell RNA Sequencing Reveals Microglia-Like Cells in Cerebrospinal Fluid During Virologically Suppressed HIV</em>. <em>JNEUROSCI</em>. Vol. 356, 1248. <a href="https://DOI: 10.1126/science.aal3222">DOI: 10.1126/science.aal3222</a>.
</div>
<div id="ref-Deczkowska" class="csl-entry">
Deczkowska A, Tsitsou-Kampeli A, Matcovitch-Natan O. 2017. <em>Mef2C Restrains Microglial Inflammatory Response and Is Lost in Brain Ageing in an IFN-i-Dependent Manner</em>. <em>Nat Commun</em>. Vol. 8(1). <a href="https://doi: 10.1038/s41467-017-00769-0">doi: 10.1038/s41467-017-00769-0</a>.
</div>
<div id="ref-Ejarque-Ortiz" class="csl-entry">
Ejarque-Ortiz A, Tusell JM, Medina MG. 2007. <em>Upregulation of CCAAT/Enhancer Binding Protein Beta in Activated Astrocytes and Microglia</em>. <em>Glia</em>. Vol. 55(2). <a href="https://doi: 10.1002/glia.20446">doi: 10.1002/glia.20446</a>.
</div>
<div id="ref-Grassivaro" class="csl-entry">
Francesca Grassivaro, Massimo Acquaviva, Ramesh Menon, and Gianvito Martino. 2018. <em>Convergence Between Microglia and Peripheral Macrophages Phenotype During Development and Neuroinflammation</em>. <em>JCI Insight</em>. Vol. 40 (4). <a href="https://doi.org/10.1523/JNEUROSCI.1523-19.2019">https://doi.org/10.1523/JNEUROSCI.1523-19.2019</a>.
</div>
<div id="ref-He" class="csl-entry">
He G, Margolis DM. 2002. <em>Counterregulation of Chromatin Deacetylation and Histone Deacetylase Occupancy at the Integrated Promoter of Human Immunodeficiency Virus Type 1 (HIV-1) by the HIV-1 Repressor Yy1 and HIV-1 Activator Tat</em>. <em>Mol Cell Biol</em>. Vol. 22(9). <a href="https://doi: 10.1128/MCB.22.9.2965-2973.2002">doi: 10.1128/MCB.22.9.2965-2973.2002</a>.
</div>
<div id="ref-Lee" class="csl-entry">
Lee Yu K, Park SH, Jung YM. 2020. <em>Human Transcription Factor Yy1 Could Upregulate the HIV-1 Gene Expression</em>. <em>BMB Rep</em>. Vol. 53(5). <a href="https://doi: 10.5483/BMBRep.2020.53.5.222">doi: 10.5483/BMBRep.2020.53.5.222</a>.
</div>
<div id="ref-Li" class="csl-entry">
Li H, Guo X, Wang F. 2021. <em>Decreased Mef2a Expression Regulated by Its Enhancer Methylation Inhibits Autophagy and May Play an Important Role in the Progression of Alzheimer’s Disease</em>. <em>Front Neurosci</em>. Vol. 15:682247. <a href="https://doi: 10.3389/fnins.2021.682247">doi: 10.3389/fnins.2021.682247</a>.
</div>
<div id="ref-Ndoja" class="csl-entry">
Ndoja A, Lee SH, Reja R. 2020. <em>Ubiquitin Ligase Cop1 Suppresses Neuroinflammation by Degrading c/EBPβ in Microglia</em>. <em>Cell</em>. Vol. 182(5). <a href="https://doi: 10.1016/j.cell.2020.07.011">doi: 10.1016/j.cell.2020.07.011</a>.
</div>
<div id="ref-Rickert" class="csl-entry">
Rickert U, Heimke M, Cossais F. 2018. <em>Anti-Inflammatory Properties of Honokiol in Activated Primary Microglia and Astrocytes</em>. <em>J Neuroimmunol</em>. Vol. 323. <a href="https://doi: 10.1016/j.jneuroim.2018.07.013">doi: 10.1016/j.jneuroim.2018.07.013</a>.
</div>
<div id="ref-Romerio" class="csl-entry">
Romerio F, Margolis DM, Gabriel MN. 1997. <em>Repression of Human Immunodeficiency Virus Type 1 Through the Novel Cooperation of Human Factors Yy1 and LSF</em>. <em>J Virol</em>. Vol. 71(12). <a href="https://doi: 10.1128/JVI.71.12.9375-9382.1997">doi: 10.1128/JVI.71.12.9375-9382.1997</a>.
</div>
<div id="ref-Yu" class="csl-entry">
Yu T, Xu Y, Lin Y. 2020. <em>Repressor Element 1 Silencing Transcription Factor (REST) Governs Microglia-Like Bv2 Cell Migration via Progranulin (PGRN)</em>. <em>Neural Plast</em>. Vol. 2020:8855822. <a href="https://doi: 10.1155/2020/8855822">doi: 10.1155/2020/8855822</a>.
</div>
<div id="ref-Zhang" class="csl-entry">
Zhang Q, Wang S, Kang Y. 2021. <em>HIV Reprograms Host m6Am RNA Methylome by Viral Vpr Protein-Mediated Degradation of Pcif1</em>. <em>Nat Commun</em>. Vol. 12(1). <a href="https://doi: 10.1038/s41467-021-25683-4">doi: 10.1038/s41467-021-25683-4</a>.
</div>
</div>
</div>

<div id="rmd-source-code">---
title: "Discussion"
author: "Katharina Mikulik"
output: html_document
bibliography: references.bib
link-citations: yes
---


GRN-inference on gene expression data from the Myeloid2 cells identified eleven
TFs with different activity
between three cell clusters in all three patients, which were also shown to 
have different binding scores in the C20 cell line. In all three patients there was 
one cluster which was different from the other two clusters. In this cluster of
interest ETS1, REST and YY1 had an increased activity, while the other TFs had a
decreased activity. Since YY1 represses HIV-1 transcription it promotes latent infection 
[@He; @Romerio; @Lee]. ETS1 binds to HIV-1 promoter to decrease HIV-1 transcription
[@Zhang]. REST acts as a repressor of cell migration in microglia [@Yu].
Conversely, transcription factors with decreased activity, like for example
MEF2C and MEF2A suppress the inflammatory response, meaning that if they are 
less active in this cluster, the cell might be in a more reactive state
[@Deczkowska; Butovsky]. MEF2A is downregulated in disease and loosing MEF2C 
might prime microglia to increased sensitivity to immune stimuli [@Butovsky; 
@Deczkowska; @Li]. The MEF2 Family is  
expressed primarily in microglia [@Li]. Lower activity of members of the MEF 
family does not necessarily mean that the cells are not microglia, but could
be associated with the disease state.

CEBPB also had decreased TF activity in the cluster of interest. CEBPB regulates 
pro-inflammatory gene expression and has been shown to be upregulated in microglia
activation [@Ejarque-Ortiz; @Ndoja]. Dysregulated CEBPB is 
observed during HAND, with an overall decrease of CEBPB in the brain. While
neurons slightly decrease CEBPB expression, astrocytes and likely other brain 
cells increase their expression of CEBPB [@Canchi]. However, there is no data from 
microglia cells. An increased expression in the brain might be caused by an
increased response to inflammation, because CEBPB is involved in the
inflammatory response. The TF KLF4 is also proposed to be involved in
neuroinflammation [@Rickert]. Additionally, JUNB, CEBPB, MEF2A, MEF2C, ETV6 
and KLF4 are used as specific 
TFs characterizing microglia cells [@Gosselin].

Because of the functions of TFs listed above the hypothesis is that the cells 
in the first cluster/cluster of interest of each patient are latently infected
microglia-like cells. TFs which show decreased activity 
are involved in inflammation. Conversely, they show increased activity 
of two TFs which have been related to latent HIV-1 infection, YY1 and ETS1. These
microglia-like cells could be latently infected cells which entered the brain 
as macrophages and
acquired microglia markers and functions which they kept when leaving the brain 
and migrating into CSF [@Grassivaro].

Analysing the TF activity of iMg and comparing it to the footprinting binding
scores of the C20 cell was interesting, because in this case the infection status 
of each cell is known, as well as the treatment condition. It was shown that the 
treatment has a very large effect on the TF activity and, therefore, on gene 
expression. The effect of the treatment seems to overshadow the effect of latent 
or active infection in iMg. The two 
TFs RUNX3 and JUNB were more active in treated cells, while NFAT5, TCF7L and CEBPB 
were more active in not treated cells, independently of the infection status. An 
important insight to keep in mind for any further analysis is this huge effect that
the treatment has on the cells. Comparing latently infected cells which are affected
by a drug to non-infected cells which are affected by a drug might yield very 
different results than comparing them with non-infected cells which are not 
affected by a drug. 

Finding a pattern of activity which corresponds to the infection status or 
resembles the footprinting binding scores was more difficult and even though 
some TFs followed
the footprinting pattern it was usually only the case for either non-treated cells
or treated cells, but not for both treatment conditions. The reason for this 
remains unclear.

Comparing the activities in iMg to the activities in Myeloid2 cells from CSF,
there are only five transcription factors which are found to have differential
TF activities in both model systems. Transferring the information from the iMg, where 
the infection status is known, to the Myeloid2 cells could support the hypothesis
that the group of Myeloid2 cells described above are latently infected. YY1 is 
more active in the cluster of interest in Myeloid2 and more active in infected +
EFZ than in uninfected + EFZ iMg. CEBPB is less active in the cluster of interest
and more active in uninfected + EFZ than infected + EFZ iMg. Similarly, JUNB is 
less active in the cluster of interest and more active uninfected + EFZ than 
infected + EFZ iMg. This would suggest that the cluster of interest are indeed 
latently infected microglia-like cells. However, MEF2A and MEF2C are suggesting
the opposite. No clear conclusion can be drawn from these five TFs alone.

Integrating the triculture dataset and the CSF/Blood dataset it became evident 
that iMg and all Myeloid clusters, not only the Myeloid2 share some similarities,
because they are from the same cell lineage. Using the up- and downregulated
gene signature of infected and treated iMg, 51 Myeloid2 cells were identified 
with high scores for the signatures. These cells probably represent latently infected cells,
however, they did not correspond to the cells in the cluster of interest (see 
above) hypothesised to be latently infected cells. Consequently, two 
different approaches yielded two different groups of cells which are potentially
latently infected. Since the infection status of the iMg cells is known and, 
therefore, the identified gene signatures most likely related to the infection, 
this result is more reliable. Interpretation of the cluster of interest from
k-means clustering of Myeloid2 cells was based on TFs with different activities 
between the clusters. Literature search of the functions of these TFs indicated that
these cells might be latently infected. Yet, the role of these TFs was not 
entirely clear, the number of them probably too small and only relying on TF 
activity alone cannot result in a reliable conclusion. Since the infection status of
the iMg cells is known and, therefore, the identified gene signatures most 
likely related to the infection, this result is more reliable. 

It could be possible to transfer infection signatures from cell
culture to microglia-like myeloid cells from CSF. This would offer a new 
diagnostic tool for HAND, since it would be possible to identify infected 
cells. Nevertheless, too little is known about the exact functions and 
characteristics of these microglia-like cells, as well as, what role they play in
HAND. Additionally, it remains to be elucidated in more detail how they are
related to microglia cells and whether it is really possible to infer infection
status of microglia, number of infected microglia and details on HAND pathogenesis
from cells of the CSF. 

# References </div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("Discussion.Rmd");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
