<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="kmikulik" />


<title>Analysis of HIV CSF/Blood dataset</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Introduction.html">Introduction</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="HIV_CSF_dataset_analysis.html">1. CSF</a>
    </li>
    <li>
      <a href="TF_activity_analysis_of_myeloid2_cluster.html">2. TF activity Myeloid2</a>
    </li>
    <li>
      <a href="Triculture_datset_analysis_and_integration_Farhadian_dataset.html">3. Integration CSF &amp; triculture</a>
    </li>
    <li>
      <a href="Triculture_dataset_GRN_with_Scenic.html">4. TF activity iMg</a>
    </li>
  </ul>
</li>
<li>
  <a href="Discussion.html">Discussion</a>
</li>
<li>
  <a href="Methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Katharina782/report">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis of HIV CSF/Blood dataset</h1>
<h4 class="author">kmikulik</h4>
<h4 class="date">3 12 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-20
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 4 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 3
</p>
<p>
<strong>Knit directory:</strong> <code>report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
unnamed-chunk-1
</li>
<li>
unnamed-chunk-10
</li>
<li>
unnamed-chunk-11
</li>
<li>
unnamed-chunk-12
</li>
<li>
unnamed-chunk-13
</li>
<li>
unnamed-chunk-14
</li>
<li>
unnamed-chunk-15
</li>
<li>
unnamed-chunk-16
</li>
<li>
unnamed-chunk-17
</li>
<li>
unnamed-chunk-18
</li>
<li>
unnamed-chunk-19
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-20
</li>
<li>
unnamed-chunk-21
</li>
<li>
unnamed-chunk-22
</li>
<li>
unnamed-chunk-23
</li>
<li>
unnamed-chunk-24
</li>
<li>
unnamed-chunk-25
</li>
<li>
unnamed-chunk-26
</li>
<li>
unnamed-chunk-27
</li>
<li>
unnamed-chunk-28
</li>
<li>
unnamed-chunk-29
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-30
</li>
<li>
unnamed-chunk-31
</li>
<li>
unnamed-chunk-32
</li>
<li>
unnamed-chunk-33
</li>
<li>
unnamed-chunk-34
</li>
<li>
unnamed-chunk-35
</li>
<li>
unnamed-chunk-36
</li>
<li>
unnamed-chunk-37
</li>
<li>
unnamed-chunk-38
</li>
<li>
unnamed-chunk-39
</li>
<li>
unnamed-chunk-4
</li>
<li>
unnamed-chunk-40
</li>
<li>
unnamed-chunk-41
</li>
<li>
unnamed-chunk-5
</li>
<li>
unnamed-chunk-6
</li>
<li>
unnamed-chunk-7
</li>
<li>
unnamed-chunk-8
</li>
<li>
unnamed-chunk-9
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>HIV_CSF_dataset_analysis_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/media/ag-cherrmann/kmikulik/HIV_microglia/src/report/
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttreef7843b5a7865b5270e721642ea06e9cb5261d374targetblankf7843b5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Katharina782/report/tree/f7843b5a7865b5270e721642ea06e9cb5261d374" target="_blank">f7843b5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttreef7843b5a7865b5270e721642ea06e9cb5261d374targetblankf7843b5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Katharina782/report/tree/f7843b5a7865b5270e721642ea06e9cb5261d374" target="_blank">f7843b5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/HIV_CSF_dataset_analysis_cache/
    Ignored:    analysis/Independent_cell_type_identification_Farhadian_dataset_cache/
    Ignored:    analysis/TF_activity_analysis_of_myeloid2_cluster_cache/
    Ignored:    analysis/Triculture_dataset_GRN_with_Scenic_cache/
    Ignored:    analysis/Triculture_dataset_analysis_and_integration_Farhadian_dataset_cache/
    Ignored:    analysis/figure/

Unstaged changes:
    Modified:   analysis/HIV_CSF_dataset_analysis.Rmd
    Modified:   analysis/Independent_cell_type_identification_Farhadian_dataset.Rmd
    Modified:   analysis/Introduction.Rmd
    Modified:   analysis/Triculture_dataset_GRN_with_Scenic.Rmd
    Modified:   analysis/Triculture_dataset_analysis_and_integration_Farhadian_dataset.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/html_test.Rmd
    Modified:   analysis/references.bib

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/HIV_CSF_dataset_analysis.Rmd</code>) and HTML (<code>docs/HIV_CSF_dataset_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/HIV_CSF_dataset_analysis.html" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/HIV_CSF_dataset_analysis.html" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/9591a5f95e1c4ab7b8385a71d4d720ee51590f26/analysis/HIV_CSF_dataset_analysis.Rmd" target="_blank">9591a5f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
wflow_publish(“analysis/HIV_CSF_dataset_analysis.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/HIV_CSF_dataset_analysis.html" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/ca459b400362711f3ba362318af3d2a456f569b4/docs/HIV_CSF_dataset_analysis.html" target="_blank">ca459b4</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/04f675329bc4ccf704fe3326b6e562264c9dbbd3/analysis/HIV_CSF_dataset_analysis.Rmd" target="_blank">04f6753</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-17
</td>
<td>
wflow_publish(files = rmd)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/HIV_CSF_dataset_analysis.html" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/3ba29868e10097faef3f8ff84c501c88bf93da86/analysis/HIV_CSF_dataset_analysis.Rmd" target="_blank">3ba2986</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
wflow_publish(“analysis/HIV_CSF_dataset_analysis.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/analysis/HIV_CSF_dataset_analysis.Rmd" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
<td>
recent updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/dde10972b37ffd997474921968223514c8cb1b1f/docs/HIV_CSF_dataset_analysis.html" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
<td>
recent updates
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style>
body {
text-align: justify}
</style>
<pre class="r"><code>setwd(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/report/&quot;)</code></pre>
<pre class="r"><code>library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggplot2)
library(dplyr)
library(ggrepel)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(corrplot)
library(rstatix)
library(knitr)
library(metap)
library(multtest)
library(readxl)
library(harmony)
library(SeuratDisk)
library(patchwork)</code></pre>
<p>First, the analysis done by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> was reproduced and then more extended analysis of the dataset was performed. The microglia-like cell cluster identified by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> was of particular interest.</p>
<p>The dataset included three CSF samples from HIV infected patients, two blood samples from HIV infected patients, as well as, two CSF samples from uninfected patients. <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> first used two CSF and two bloods samples from infected patients for clustering and cell type annotation. After having identified a novel subcluster of myeloid cells, which they propose are microglia-like, they then confirm their results using the third CSF samples from an infected patient. Additionally, they compare the number of Myeloid2-specific transcripts to the CSF samples from uninfected patients. They show that the percentage of Myeloid2 transcripts was higher in CSF from infected patients than in CSF from uninfected patients and higher in CSF than in blood in general.</p>
<p>Blood and CSF were processed separately for scRNAseq using SeqWell. The seven samples were sequenced separately, one SeqWell array per CSF or PBMC sample <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span>.</p>
<p>For more details see <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span>.</p>
<p>First, the following analysis steps were performed similar as in the paper:</p>
<ul>
<li>Import files</li>
<li>Create Seurat objects</li>
<li>Quality Control</li>
<li>Normalization &amp; Scaling</li>
<li>Clustering</li>
<li>Dimensionality reduction</li>
</ul>
</details>
<details>
<summary>
Importing files &amp; creating Seurat objects
</summary>
<p><strong>Read in data(count matrices)</strong> The data is available under <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE117397" class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE117397</a>. There is one count matrix for each of the 7 samples.</p>
<pre><code>path &lt;- &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/GSE117397_RAW (1)/&quot;
all_files &lt;- list.files(path)

#Function to create a sparse matrix from each of the dataframes
sparse_matrix_from_df &lt;- function(df) {
  #print(df)
  #class(df)
  genes &lt;- df$GENE
  df_minus &lt;- df[,-1]
  #matrix &lt;- as.matrix(df_minus)
  sparse_matrix &lt;- as.sparse(df_minus)
  rownames(sparse_matrix) &lt;- genes
  return(sparse_matrix)
}

hiv_object_list &lt;- map(seq.int(2,6), function(n) {
  file &lt;- all_files[grepl(paste0(&quot;GSM329382&quot;, n), all_files)]
  name &lt;- str_extract(file, &quot;HIV[1-3]_\\w{3}&quot;)
  df &lt;- read_delim(paste0(path, file), delim = &quot;\t&quot;)
  sparse_matrix &lt;- sparse_matrix_from_df(df)
  seurat_ob &lt;- CreateSeuratObject(sparse_matrix, project = name, 
                     assay = &quot;RNA&quot;, min.cells = 1,
                     min.features = 1)
  list(df=df, sparse_matrix = sparse_matrix, seurat_ob = seurat_ob, name=name)
})

uninfected_list &lt;- map(seq.int(7,8), function(n) {
  file &lt;- all_files[grepl(paste0(&quot;GSM329382&quot;, n), all_files)]
  name &lt;- str_extract(file, &quot;Uninfected[12]_CSF&quot;)
  df &lt;- read_delim(paste0(path, file), delim = &quot;\t&quot;)
  sparse_matrix &lt;- sparse_matrix_from_df(df)
  seurat_ob &lt;- CreateSeuratObject(sparse_matrix, project = name, 
                     assay = &quot;RNA&quot;, min.cells = 1,
                     min.features = 1, verbose = FALSE)
  list(df=df, sparse_matrix = sparse_matrix, seurat_ob = seurat_ob, name=name)
})

#write.table(uninfected_list[[1]]$sparse_matrix,
 #           &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/job_test.tsv&quot;, 
  #          sep = &quot;\t&quot;)</code></pre>
<p><strong>Create one Seurat object with all samples</strong></p>
<pre><code>
hiv &lt;- merge(hiv_object_list[[1]]$seurat_ob, 
                   c(hiv_object_list[[2]]$seurat_ob,
                     hiv_object_list[[3]]$seurat_ob,
                     hiv_object_list[[4]]$seurat_ob,
                     hiv_object_list[[5]]$seurat_ob, 
                     uninfected_list[[1]]$seurat_ob,
                     uninfected_list[[2]]$seurat_ob), 
                   project = &quot;hiv_total&quot;,
                   add.cell.ids = c(&quot;1B&quot;, &quot;1C&quot;, &quot;2B&quot;, &quot;2C&quot;, &quot;3C&quot;, &quot;U1&quot;, &quot;U2&quot;))

# add percentage of mitochondrial genes to the metadata
hiv[[&quot;percent_mt&quot;]] &lt;- PercentageFeatureSet(hiv, pattern = &quot;^MT-&quot;)
#saveRDS(hiv, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv_unprocessed&quot;)</code></pre>
<hr />
</details>
<p><strong>Quality control was performed on every individual object.</strong></p>
<p>Quality control was performed as described in <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span>. The 10% of cells with the highest percentage of mitochondrial genes was removed, as well as features which are expressed in less than 3 cells. Also, cells with number of features below 500 or above 2500 were removed. Quality control was done for each sample separately and then the samples were combined again before normalization.</p>
<pre class="r"><code># read in unprocessed seurat object containing all 7 samples
hiv &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv_unprocessed&quot;)

# FUNCTION:
# to remove features that are expressed in less than three cells
# returns a list of feature names to keep

remove_features &lt;- function(seurat_ob) {
  count_matrix &lt;- seurat_ob@assays$RNA@counts 
  ncells &lt;- rowSums(count_matrix != 0) # how many cells are expressing a 
  #certain gene? 
  names(ncells[ncells &gt; 3]) #extract gene names of genes expressed in more 
  #than 3 cells
}


# FUNCTION:
# to remove 10% of cell with highest mitochondrial gene expression:
#returns a list of cell names to keep
remove_mt_genes &lt;- function(seurat_ob){
  thr &lt;- quantile(seurat_ob@meta.data$percent_mt, .9)
  seurat_ob@meta.data %&gt;% 
    filter(percent_mt &lt; thr) %&gt;%
    rownames()
}



# SOME THOUGHTS:

# When doing the quality control how important is the order of the operations?
# If you first remove nFeatures &gt; 500 and then the features exppressed in more than
# 3 cells, then the nFeatures might be below 500.

separate_objects &lt;- SplitObject(hiv, split.by = &quot;orig.ident&quot;)#[-c(5,6,7)] 
# only keep two blood samples and two csf samples

qc &lt;- lapply(X = separate_objects, FUN = function(x) {
    #x[[&quot;percent_mt&quot;]] &lt;- PercentageFeatureSet(x, pattern = &quot;^MT-&quot;) 
    # add percentage of mitochondrial genes to the metadata
    x &lt;- subset(x, features = remove_features(x)) 
    #create a list of cells excluding the 10% of cells with the highest mt gene expression
    x &lt;- subset(x, nFeature_RNA &lt; 2500 &amp; nFeature_RNA &gt; 500)
    x &lt;- subset(x, cells = remove_mt_genes(x)) 
    # remove 10% of cells with highest mitochondrial gene expression
    #x &lt;- NormalizeData(x)
    #x &lt;- ScaleData(x)
})

hiv_filt &lt;- merge(qc[[1]],
                        c(qc[[2]],
                        qc[[3]],
                        qc[[4]],
                        qc[[5]],
                        qc[[6]],
                        qc[[7]]),
                        project = &quot;hiv_filt&quot;)#,
                        #add.cell.ids = c(&quot;1B&quot;, &quot;1C&quot;, &quot;2B&quot;, &quot;2C&quot;, &quot;3C&quot;, &quot;U1&quot;, &quot;U2&quot;))</code></pre>
<div id="quality-of-different-samples-before-quality-control" class="section level2 tabset">
<h2 class="tabset">Quality of different samples before quality control</h2>
<div id="number-of-features" class="section level3">
<h3>Number of Features</h3>
<pre class="r"><code>
#have a look at the number of detected features
p1 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_boxplot(aes(x = orig.ident, y = nFeature_RNA, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p2 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_histogram(aes(x = nFeature_RNA, fill = orig.ident), bins = 100) +
  facet_wrap(~ orig.ident, ncol = 4) +
  NoLegend()

p3 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = orig.ident, y = nFeature_RNA, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p4 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = orig.ident, y = nFeature_RNA, fill = orig.ident)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

figure &lt;- ggarrange(p1, p3, p4, p2, ncol = 2, nrow = 2)
annotate_figure(figure, top = text_grob(&quot;Number of Features&quot;,
                                        size = 25,  face = &quot;bold&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-4-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>

hiv@meta.data %&gt;%
  rename(sample = orig.ident) %&gt;% 
  group_by(sample) %&gt;% 
  summarise(n = n(), q25 = quantile(nFeature_RNA, .25),
            q50 = quantile(nFeature_RNA, .5), 
            q75 = quantile(nFeature_RNA, .75), 
            mean = mean(nFeature_RNA)) %&gt;%
    kable(caption = &quot;Number of Features per sample&quot;)</code></pre>
<table>
<caption>Number of Features per sample</caption>
<thead>
<tr class="header">
<th align="left">sample</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">4998</td>
<td align="right">128</td>
<td align="right">458.0</td>
<td align="right">1021</td>
<td align="right">654.7989</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">4995</td>
<td align="right">16</td>
<td align="right">32.0</td>
<td align="right">341</td>
<td align="right">315.3642</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">5000</td>
<td align="right">360</td>
<td align="right">581.0</td>
<td align="right">987</td>
<td align="right">738.4262</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">4996</td>
<td align="right">126</td>
<td align="right">584.5</td>
<td align="right">1087</td>
<td align="right">697.9960</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">5000</td>
<td align="right">175</td>
<td align="right">309.0</td>
<td align="right">666</td>
<td align="right">542.9580</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">4994</td>
<td align="right">22</td>
<td align="right">44.0</td>
<td align="right">114</td>
<td align="right">151.0993</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">4990</td>
<td align="right">31</td>
<td align="right">79.0</td>
<td align="right">206</td>
<td align="right">205.2401</td>
</tr>
</tbody>
</table>
</div>
<div id="library-size" class="section level3">
<h3>Library Size</h3>
<pre class="r"><code># have a look at the library size
p1 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_boxplot(aes(x = orig.ident, y = nCount_RNA, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p2 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_histogram(aes(x = nCount_RNA, fill = orig.ident), bins = 100) +
  facet_wrap(~ orig.ident, ncol = 4) +
  NoLegend()

p3 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = orig.ident, y = nCount_RNA, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p4 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = orig.ident, y = nCount_RNA, fill = orig.ident)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()
  
figure &lt;- ggarrange(p1, p3, p4, p2, ncol = 2, nrow = 2)
annotate_figure(figure, top = text_grob(&quot;Library size (number of counts)&quot;,
                                        size = 25,  face = &quot;bold&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-5-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hiv@meta.data %&gt;%
  rename(sample = orig.ident) %&gt;% 
  group_by(sample) %&gt;% 
  summarise(n = n(), q25 = quantile( nCount_RNA, .25),
            q50 = quantile(nCount_RNA, .5), 
            q75 = quantile(nCount_RNA, .75), 
            mean = mean(nCount_RNA)) %&gt;% 
    kable(caption = &quot;Library size per sample&quot;)</code></pre>
<table>
<caption>Library size per sample</caption>
<thead>
<tr class="header">
<th align="left">sample</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">4998</td>
<td align="right">196</td>
<td align="right">772.0</td>
<td align="right">2098.00</td>
<td align="right">1424.6823</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">4995</td>
<td align="right">19</td>
<td align="right">39.0</td>
<td align="right">517.00</td>
<td align="right">648.8895</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">5000</td>
<td align="right">577</td>
<td align="right">1006.0</td>
<td align="right">1970.50</td>
<td align="right">1523.0344</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">4996</td>
<td align="right">173</td>
<td align="right">929.0</td>
<td align="right">2053.25</td>
<td align="right">1369.0498</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">5000</td>
<td align="right">238</td>
<td align="right">473.5</td>
<td align="right">1280.50</td>
<td align="right">1216.9344</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">4994</td>
<td align="right">26</td>
<td align="right">53.0</td>
<td align="right">146.00</td>
<td align="right">269.5036</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">4990</td>
<td align="right">37</td>
<td align="right">96.5</td>
<td align="right">272.00</td>
<td align="right">356.2377</td>
</tr>
</tbody>
</table>
</div>
<div id="percentage-of-mitochondrial-genes" class="section level3">
<h3>Percentage of Mitochondrial genes</h3>
<pre class="r"><code># have a look at the percentage of mitochondrial genes
p1 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_boxplot(aes(x = orig.ident, y = percent_mt, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p2 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_histogram(aes(x = percent_mt, fill = orig.ident), bins = 100) +
  facet_wrap(~ orig.ident, ncol = 4) +
  NoLegend()

p3 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = orig.ident, y = percent_mt, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p4 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = orig.ident, y = percent_mt, fill = orig.ident)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()
  
figure &lt;- ggarrange(p1, p3, p4, p2, ncol = 2, nrow = 2)
Warning: Transformation introduced infinite values in continuous y-axis
Warning: Removed 2014 rows containing non-finite values (stat_ydensity).
annotate_figure(figure, top = text_grob(&quot;Percentage of Mitochondrial gene&quot;,
                                        size = 25,  face = &quot;bold&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-6-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>
hiv@meta.data %&gt;%
  rename(sample = orig.ident) %&gt;% 
  group_by(sample) %&gt;% 
  summarise(n = n(), q25 = quantile( percent_mt, .25),
            q50 = quantile(percent_mt, .5), 
            q75 = quantile(percent_mt, .75), 
            mean = mean(percent_mt)) %&gt;% 
    kable(caption = &quot;Percentage of mitochondrial genes per sample&quot;)</code></pre>
<table>
<caption>Percentage of mitochondrial genes per sample</caption>
<thead>
<tr class="header">
<th align="left">sample</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">4998</td>
<td align="right">6.140696</td>
<td align="right">10.314638</td>
<td align="right">16.83490</td>
<td align="right">13.746867</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">4995</td>
<td align="right">4.000000</td>
<td align="right">8.522727</td>
<td align="right">15.38462</td>
<td align="right">12.092172</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">5000</td>
<td align="right">9.558731</td>
<td align="right">14.370956</td>
<td align="right">20.53664</td>
<td align="right">16.562072</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">4996</td>
<td align="right">7.387619</td>
<td align="right">11.188337</td>
<td align="right">16.68489</td>
<td align="right">13.814837</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">5000</td>
<td align="right">2.590861</td>
<td align="right">4.413543</td>
<td align="right">7.26931</td>
<td align="right">5.785039</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">4994</td>
<td align="right">4.165372</td>
<td align="right">8.333333</td>
<td align="right">14.70588</td>
<td align="right">10.832587</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">4990</td>
<td align="right">3.675874</td>
<td align="right">7.462687</td>
<td align="right">12.73541</td>
<td align="right">9.618704</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="section" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
</div>
<div id="quality-of-the-different-samples-after-quality-control" class="section level2 tabset">
<h2 class="tabset">Quality of the different samples after quality control</h2>
<p>It seems that the CSF1 samples, as well as the two uninfected samples have very few cells with nFeature_RNA &gt; 500, so we are left with less cells from these samples</p>
<div id="number-of-features-1" class="section level3">
<h3>Number of features</h3>
<pre class="r"><code>
# Have a look at the number of Feature RNA
p1 &lt;- hiv_filt@meta.data %&gt;%
  ggplot() +
  geom_boxplot(aes(x = orig.ident, y = nFeature_RNA, fill = orig.ident)) +
  NoLegend()

p2 &lt;- hiv_filt@meta.data %&gt;% 
  ggplot()+
  geom_violin(aes(x = orig.ident, y = nFeature_RNA, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p3 &lt;- hiv_filt@meta.data %&gt;% 
  ggplot() +
  geom_histogram(aes(x = nFeature_RNA, fill = orig.ident)) +
  facet_wrap(~orig.ident, ncol = 4) +
  NoLegend()

figure &lt;- ggarrange(p1, p2, p3, ncol = 2, nrow = 2)
annotate_figure(figure, top = text_grob(&quot;Number of features after quality control&quot;,
                                        size = 25,  face = &quot;bold&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-7-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
hiv_filt@meta.data %&gt;% 
  rename(sample = orig.ident) %&gt;% 
  group_by(sample) %&gt;%
  summarise(n = n(), q25 = quantile(nFeature_RNA, .25),
            q50 = quantile(nFeature_RNA, .5),
            q75 = quantile(nFeature_RNA, .75),
            mean = mean(nFeature_RNA)) %&gt;%
    kable(caption = &quot;Number of Features per sample after quality control&quot;)</code></pre>
<table>
<caption>Number of Features per sample after quality control</caption>
<thead>
<tr class="header">
<th align="left">sample</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">2076</td>
<td align="right">786.75</td>
<td align="right">1053.0</td>
<td align="right">1413.25</td>
<td align="right">1140.157</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">932</td>
<td align="right">800.00</td>
<td align="right">1077.0</td>
<td align="right">1395.00</td>
<td align="right">1142.855</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">2520</td>
<td align="right">672.00</td>
<td align="right">914.0</td>
<td align="right">1277.00</td>
<td align="right">1023.094</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">2309</td>
<td align="right">809.00</td>
<td align="right">1063.0</td>
<td align="right">1359.00</td>
<td align="right">1129.844</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">1384</td>
<td align="right">703.00</td>
<td align="right">1091.0</td>
<td align="right">1492.25</td>
<td align="right">1153.576</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">216</td>
<td align="right">672.50</td>
<td align="right">887.5</td>
<td align="right">1526.00</td>
<td align="right">1133.796</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">340</td>
<td align="right">670.50</td>
<td align="right">1066.5</td>
<td align="right">1606.00</td>
<td align="right">1155.576</td>
</tr>
</tbody>
</table>
</div>
<div id="library-size-1" class="section level3">
<h3>Library size</h3>
<pre class="r"><code>p1 &lt;- hiv_filt@meta.data %&gt;%
  ggplot() +
  geom_boxplot(aes(x = orig.ident, y = nCount_RNA, fill = orig.ident)) +
  NoLegend()

p2 &lt;- hiv_filt@meta.data %&gt;% 
  ggplot()+
  geom_violin(aes(x = orig.ident, y = nCount_RNA, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p3 &lt;- hiv_filt@meta.data %&gt;% 
  ggplot() +
  geom_histogram(aes(x = nCount_RNA, fill = orig.ident)) +
  facet_wrap(~orig.ident, ncol = 4) +
  NoLegend()

figure &lt;- ggarrange(p1, p2, p3, ncol = 2, nrow = 2)
annotate_figure(figure, top = text_grob(&quot;Library size (number of counts) after quality control&quot;,
                                        size = 25,  face = &quot;bold&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-8-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
hiv_filt@meta.data %&gt;% 
  rename(sample = orig.ident) %&gt;% 
  group_by(sample) %&gt;%
  summarise(n = n(), q25 = quantile(nCount_RNA, .25),
            q50 = quantile(nCount_RNA, .5),
            q75 = quantile(nCount_RNA, .75),
            mean = mean(nCount_RNA)) %&gt;%
  kable(caption = &quot;Library size per sample after quality control&quot;)</code></pre>
<table>
<caption>Library size per sample after quality control</caption>
<thead>
<tr class="header">
<th align="left">sample</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">2076</td>
<td align="right">1458.75</td>
<td align="right">2179.5</td>
<td align="right">3264.25</td>
<td align="right">2522.768</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">932</td>
<td align="right">1410.75</td>
<td align="right">2102.5</td>
<td align="right">3066.75</td>
<td align="right">2378.643</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">2520</td>
<td align="right">1187.50</td>
<td align="right">1779.5</td>
<td align="right">2822.25</td>
<td align="right">2158.089</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">2309</td>
<td align="right">1374.00</td>
<td align="right">1984.0</td>
<td align="right">2745.00</td>
<td align="right">2222.453</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">1384</td>
<td align="right">1377.25</td>
<td align="right">2381.5</td>
<td align="right">3845.25</td>
<td align="right">2842.006</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">216</td>
<td align="right">1063.25</td>
<td align="right">1491.5</td>
<td align="right">2968.75</td>
<td align="right">2177.278</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">340</td>
<td align="right">1063.50</td>
<td align="right">1876.0</td>
<td align="right">3399.50</td>
<td align="right">2328.871</td>
</tr>
</tbody>
</table>
</div>
<div id="percentage-of-mitochondrial-genes-1" class="section level3">
<h3>Percentage of Mitochondrial genes</h3>
<pre class="r"><code>p1 &lt;- hiv_filt@meta.data %&gt;%
  ggplot() +
  geom_boxplot(aes(x = orig.ident, y = percent_mt, fill = orig.ident)) +
  NoLegend()

p2 &lt;- hiv_filt@meta.data %&gt;% 
  ggplot()+
  geom_violin(aes(x = orig.ident, y = percent_mt, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p3 &lt;- hiv_filt@meta.data %&gt;% 
  ggplot() +
  geom_histogram(aes(x = percent_mt, fill = orig.ident)) +
  facet_wrap(~orig.ident, ncol = 4) +
  NoLegend()

figure &lt;- ggarrange(p1, p2, p3, ncol = 2, nrow = 2)
annotate_figure(figure, top = text_grob(&quot;Percentage of Mitochondrial genes after quality control&quot;,
                                        size = 25,  face = &quot;bold&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-9-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
hiv_filt@meta.data %&gt;% 
  rename(sample = orig.ident) %&gt;% 
  group_by(sample) %&gt;%
  summarise(n = n(), q25 = quantile(percent_mt, .25),
            q50 = quantile(percent_mt, .5),
            q75 = quantile(percent_mt, .75),
            mean = mean(percent_mt)) %&gt;% 
    kable(caption = &quot;Percentage of mitochondrial genes per sample after quality control&quot;)</code></pre>
<table>
<caption>Percentage of mitochondrial genes per sample after quality control</caption>
<thead>
<tr class="header">
<th align="left">sample</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">2076</td>
<td align="right">5.659778</td>
<td align="right">8.616470</td>
<td align="right">12.057771</td>
<td align="right">9.006083</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">932</td>
<td align="right">5.067679</td>
<td align="right">7.526483</td>
<td align="right">10.327807</td>
<td align="right">7.820102</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">2520</td>
<td align="right">8.428856</td>
<td align="right">12.017682</td>
<td align="right">15.653130</td>
<td align="right">12.183780</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">2309</td>
<td align="right">7.420091</td>
<td align="right">10.051282</td>
<td align="right">13.119900</td>
<td align="right">10.296168</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">1384</td>
<td align="right">2.045621</td>
<td align="right">3.169997</td>
<td align="right">4.804080</td>
<td align="right">3.582988</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">216</td>
<td align="right">4.339334</td>
<td align="right">5.927984</td>
<td align="right">8.122936</td>
<td align="right">6.373704</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">340</td>
<td align="right">3.683668</td>
<td align="right">5.999618</td>
<td align="right">8.409969</td>
<td align="right">6.088557</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="section-1" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
</details>
<details>
**
<summary>
Code for Dimensionality reduction &amp; Clustering
</summary>
<p>**</p>
<p>Use blood and CSF samples from two patients (HIV1_CSF/Bld, HIV2_CSF/Bld) for further analysis like in the paper.</p>
<ul>
<li><strong>Subset Seurat object</strong></li>
</ul>
<pre class="r"><code>hiv4 &lt;- subset(hiv_filt, orig.ident %in% c(&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV1_Bld&quot;, &quot;HIV2_Bld&quot;))</code></pre>
<ul>
<li><strong>Normalize, Scale and find variable features</strong></li>
</ul>
<pre class="r"><code>hiv4 &lt;- NormalizeData(hiv4, verbose = FALSE)
hiv4 &lt;- ScaleData(hiv4, verbose = FALSE)
hiv4 &lt;- FindVariableFeatures(hiv4, verbose = FALSE)</code></pre>
<ul>
<li><strong>Dimensionality reduction</strong></li>
</ul>
<p><strong>PCA</strong></p>
<p>Compute PCA for highly variable genes: The first 10 dimensions seem to account for most of the variance (see elbow plot). The plots seem to indicate that the blood samples and csf samples are different.</p>
<pre class="r"><code>hiv4 &lt;- RunPCA(hiv4, features = VariableFeatures(hiv4), verbose = FALSE)

# elbow plot to determine which PCs account for most of the variance
ElbowPlot(hiv4)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-12-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-12-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-12-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#pca plots
pca_plots &lt;- comprehenr::to_list(for (i in 1:13)
  DimPlot(hiv4, reduction = &quot;pca&quot;, dims = i : (i+1)) +
    theme())
do.call(gridExtra::grid.arrange, c(pca_plots, ncol=3, nrow = 5))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-13-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p><strong>Clustering &amp; tSNE embedding</strong></p>
<ul>
<li>compute KNN-graph</li>
<li>find clusters based on KNN-graph with Louvian algorithm</li>
</ul>
<pre class="r"><code>hiv4 &lt;- FindNeighbors(hiv4, verbose = FALSE)
hiv4 &lt;- FindClusters(hiv4, verbose = FALSE)

table(hiv4$seurat_clusters)

   0    1    2    3    4    5    6    7    8    9   10   11   12   13   14 
1301 1119  948  868  742  687  677  415  308  301  132  106  105   72   56 </code></pre>
<pre class="r"><code>hiv4 &lt;- RunTSNE(hiv4, dims = 1:10, verbose = FALSE)</code></pre>
<hr />
</details>
</div>
<div id="visualization-of-quality-in-tsne-plot" class="section level1">
<h1>Visualization of Quality in tSNE plot</h1>
<p>It does not seem that there is any batch effect, but rather the samples separate according to their origing (CSF or Blood).</p>
<pre class="r"><code>tsne_coord &lt;- Embeddings(hiv4[[&quot;tsne&quot;]]) # tsne coordinates
meta &lt;- hiv4@meta.data # metadata
cm &lt;- hiv4@assays$RNA@counts # count matrix

# UMAP with number of features detected per cell
p1 &lt;- cbind(meta, tsne_coord) %&gt;%
  ggplot() +
  geom_point(aes(x=tSNE_1, y=tSNE_2, col=nFeature_RNA), size=.5) +
  scale_color_viridis_c()

# percentage of mitochondrial genes per cell
p2 &lt;- cbind(meta, tsne_coord) %&gt;% 
  ggplot() +
  geom_point(aes(x = tSNE_1, y = tSNE_2, col = percent_mt ), size = .5) +
  scale_color_viridis_c()

# UMAP indicating the sample where each cell came from
p3 &lt;- cbind(meta, tsne_coord) %&gt;% 
  ggplot() +
  geom_point(aes(x = tSNE_1, y = tSNE_2, col = orig.ident ), size = .5) 

# UMAP labeled according to clusters
p4 &lt;- cbind(meta, tsne_coord) %&gt;% 
  ggplot() +
  geom_point(aes(x = tSNE_1, y = tSNE_2, col = seurat_clusters ), size = .5)

ggarrange(p1, p2, p3, p4, ncol = 2, nrow = 2)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-16-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="marker-genes-cell-type-annotation" class="section level1">
<h1>Marker genes &amp; Cell type annotation</h1>
<pre><code>marker_genes &lt;- FindAllMarkers(hiv4, logfc.threshold = .01, min.pct = .01)</code></pre>
<div id="top-marker-genes" class="section level2">
<h2>Top marker genes</h2>
<p>The function <code>FindAllMarkers()</code> identifies differentially expressed genes between two groups of cells, in this case I compared one group of cells at a time against the rest of cells using a Wilcoxon Rank Sum test. The table below lists the two most highly differentially expressed genes for each clusters.</p>
<pre class="r"><code># filter for significant marker genes
#sign_marker_genes &lt;- marker_genes %&gt;% filter(p_val_adj &lt;= .1)

sign_marker_genes &lt;- read.table(file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/sign_marker_genes.tsv&quot;)

#show the top 10 differentially expressed genes for each cluster
#sign_marker_genes %&gt;% group_by(cluster) %&gt;% top_n(n = 2, wt = avg_logFC ) #%&gt;%
  #kable(caption = &quot;Top two differentially expressed genes for each cluster&quot;)#, digits = c(3, format.args = list(scientific = TRUE)) #= TRUE))</code></pre>
</div>
<div id="cell-type-annotation-with-canonical-marker-genes" class="section level2 tabset">
<h2 class="tabset">Cell type annotation with canonical marker genes</h2>
<div id="b-cell-markers" class="section level3">
<h3>B cell markers</h3>
<p>Are found primarily in cluster 7.</p>
<pre class="r"><code>p1 &lt;- VlnPlot(hiv4, features = c(&quot;MS4A1&quot;, &quot;CD79A&quot;, &quot;TCL1A&quot;), pt.size = .01)
p2 &lt;- RidgePlot(hiv4, features =  c(&quot;MS4A1&quot;, &quot;CD79A&quot;, &quot;TCL1A&quot;), sort = &quot;decreasing&quot; )
ggarrange(p1,p2, nrow = 2, ncol = 1)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-18-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-18-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-18-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nk-cells" class="section level3">
<h3>NK cells</h3>
<p>NK cells are primarily found in cluster 5 and 9. GZMB is also found in cluster 14.</p>
<pre class="r"><code>p1 &lt;- VlnPlot(hiv4, features = c(&quot;NKG7&quot;, &quot;GNLY&quot;, &quot;GZMB&quot;),
        pt.size = .01, ncol = 3)#, stack = TRUE)
p2 &lt;- RidgePlot(hiv4, features = c(&quot;NKG7&quot;, &quot;GNLY&quot;, &quot;GZMB&quot;), sort = &quot;decreasing&quot;)

ggarrange(p1,p2, nrow = 2, ncol = 1)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-19-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="cd4-cd8-t-cells" class="section level3">
<h3>CD4+ &amp; CD8+ T cells</h3>
<p><strong>CD4+ T cells:</strong></p>
<ul>
<li>A general T cell marker CD3E is expressed in cluster 0,1,3,4,5,6.</li>
<li>IL7R in clusters 0-6</li>
<li>TRAC in clusters 0-6, excluding 2</li>
<li>CCR7 in 0,6,7</li>
<li>LDHB not very clear, but 0,1,3,4,6</li>
<li>LTB in 0-7</li>
<li>CD2 in 1,3,4,5,9</li>
</ul>
<p>Tcells1/2/3 are assigned arbitrarily, because they are not used for further analysis.</p>
<p><strong>CD8+ T cells:</strong></p>
<ul>
<li>CD8A is expressed in decreasing order in 1,5,3</li>
</ul>
<pre class="r"><code># general T cell marker for CD4+ and CD8+ Tcells
p1 &lt;- VlnPlot(hiv4, features = c(&quot;CD8A&quot;,&quot;CD3E&quot;), pt.size = .01, ncol = 2)
p2 &lt;- RidgePlot(hiv4, features = c(&quot;CD8A&quot;,&quot;CD3E&quot;), sort = &quot;decreasing&quot;)


p3 &lt;- VlnPlot(hiv4, features = c(&quot;IL7R&quot;, &quot;TRAC&quot;), pt.size = 0.01)
p4 &lt;- RidgePlot(hiv4, features = c(&quot;IL7R&quot;, &quot;TRAC&quot;), sort = &quot;decreasing&quot;)

# helper T cells
p5 &lt;- VlnPlot(hiv4, features = c(&quot;LTB&quot;, &quot;CD2&quot;), pt.size = .01, ncol = 2)
p6 &lt;- RidgePlot(hiv4, features = c(&quot;LTB&quot;, &quot;CD2&quot;), sort = &quot;decreasing&quot;)

# CD4+ naive T cells
p7 &lt;- VlnPlot(hiv4, features = c(&quot;LDHB&quot;, &quot;CCR7&quot;), pt.size = .01, ncol = 2)
p8 &lt;- RidgePlot(hiv4, features = c(&quot;LDHB&quot;, &quot;CCR7&quot;), sort = &quot;decreasing&quot;)


ggarrange(p1, p2,p3, p4, p5, p6, p7,p8, ncol = 2, nrow = 4)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-20-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-20-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-20-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="plasma-cells" class="section level3">
<h3>plasma cells</h3>
<p>Cluster 14 seems to contain plasma cells.</p>
<pre class="r"><code>p1 &lt;- VlnPlot(hiv4, features = c(&quot;IGJ&quot;), pt.size = .01, ncol = 2)
p2 &lt;- RidgePlot(hiv4, features = c(&quot;IGJ&quot;), sort = &quot;decreasing&quot;)
ggarrange(p1,p2, nrow = 1, ncol = 2)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-21-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-21-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-21-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="myeloid-cells" class="section level3">
<h3>Myeloid cells</h3>
<ul>
<li>granulocytes</li>
<li>monocytes</li>
<li>macrophages</li>
<li>dendritic cells</li>
</ul>
<p>The clusters 8, 10, 11, 12, 13 are myeloid cells</p>
<p>Myeloid1, 3, 4 are assigned arbitrarily, because they are not of interest.</p>
<pre class="r"><code>p1 &lt;- VlnPlot(hiv4, features = c(&quot;LYZ&quot;, &quot;FCER1A&quot;, &quot;CD14&quot;, &quot;FCGR3A&quot;), # FCG3A is also expressed in cluster 9 (NK cells)
        pt.size = .01, ncol = 2)
p2 &lt;- RidgePlot(hiv4, features = c(&quot;LYZ&quot;, &quot;FCER1A&quot;),
          sort = &quot;decreasing&quot;, ncol = 2) # DC markers
p3 &lt;- RidgePlot(hiv4, features = c(&quot;CD14&quot;, &quot;FCGR3A&quot;), 
          sort = &quot;decreasing&quot;, ncol = 2) # monocytes (LYC also in monocytes)
ggarrange(p1,p2,p3, nrow = 3, ncol = 1)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-22-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-22-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-22-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="myeloid-5" class="section level3">
<h3>Myeloid 5</h3>
<ul>
<li>cluster 12</li>
<li>was characterized by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> with the markers FCER1A, CD1C</li>
</ul>
<pre class="r"><code># Myeloid 5 in the paper
p1 &lt;- VlnPlot(hiv4, features = c( &quot;FCER1A&quot;, &quot;CD1C&quot;), pt.size = .01, ncol = 2)
p2 &lt;- RidgePlot(hiv4, features = c(&quot;FCER1A&quot;, &quot;CD1C&quot;), sort = &quot;decreasing&quot;, ncol = 2)
ggarrange(p1, p2, nrow = 2, ncol = 1)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-23-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-23-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-23-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="myeloid-2" class="section level3">
<h3>Myeloid 2</h3>
<ul>
<li>cluster 11</li>
<li>AXL, APOE, TREM2 (these genes are enriched in neurodegenerative disease-associated microglia <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span>)</li>
<li>CTSB, APOC1, MSR1, C1QA-C are expressed more highly in myeloid 2 compared to other myeloid clusters</li>
</ul>
<pre class="r"><code># Myeloid 2
p1 &lt;- VlnPlot(hiv4, features = c(&quot;AXL&quot;, &quot;APOE&quot;, &quot;TREM2&quot;), pt.size = .01, ncol = 3)
p2 &lt;- RidgePlot(hiv4, features = c(&quot;AXL&quot;, &quot;APOE&quot;, &quot;TREM2&quot;), sort = &quot;decreasing&quot;)
p3 &lt;- VlnPlot(hiv4, features = c(&quot;CTSB&quot;, &quot;APOC1&quot;, &quot;MSR1&quot;, &quot;C1QA&quot;), 
              pt.size = .01, ncol = 4)
p4 &lt;- RidgePlot(hiv4, features = c(&quot;CTSB&quot;, &quot;APOC1&quot;, &quot;MSR1&quot;, &quot;C1QA&quot;),
                sort = &quot;decreasing&quot;, ncol = 4)

ggarrange(p1, p2, p3, p4, ncol = 1, nrow = 4)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-24-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-24-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-24-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="section-2" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<ul>
<li>Why is the mixed population primarily composed of cells from HIV1 patient blood sample?</li>
<li>Why are T cells primarily from CSF samples and CD8 T cells primarily from blood?</li>
</ul>
</details>
<details>
**
<summary>
Table of annotation of cell types to clusters
</summary>
<p>**</p>
<pre class="r"><code>Idents(hiv4) &lt;- &quot;seurat_clusters&quot;
cell_types &lt;- c(&quot;Tcell2&quot;, &quot;CD8&quot;, &quot;mixed&quot;, &quot;CD81&quot;, &quot;Tcell3&quot;, &quot;NKcell1&quot;,
                &quot;Tcell1&quot;, &quot;Bcell&quot;, &quot;Myeloid1&quot;, &quot;NKcell2&quot;, &quot;Myeloid3&quot;, &quot;Myeloid2&quot;
                , &quot;Myeloid5&quot;, &quot;Myeloid4&quot;, &quot;plasma cell&quot; )

names(cell_types) &lt;- levels(hiv4)
hiv4 &lt;- RenameIdents(hiv4, cell_types)
as.data.frame(cell_types) %&gt;%  rownames_to_column(&quot;cluster&quot;) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cluster</th>
<th align="left">cell_types</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">0</td>
<td align="left">Tcell2</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">CD8</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">mixed</td>
</tr>
<tr class="even">
<td align="left">3</td>
<td align="left">CD81</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="left">Tcell3</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="left">NKcell1</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="left">Tcell1</td>
</tr>
<tr class="even">
<td align="left">7</td>
<td align="left">Bcell</td>
</tr>
<tr class="odd">
<td align="left">8</td>
<td align="left">Myeloid1</td>
</tr>
<tr class="even">
<td align="left">9</td>
<td align="left">NKcell2</td>
</tr>
<tr class="odd">
<td align="left">10</td>
<td align="left">Myeloid3</td>
</tr>
<tr class="even">
<td align="left">11</td>
<td align="left">Myeloid2</td>
</tr>
<tr class="odd">
<td align="left">12</td>
<td align="left">Myeloid5</td>
</tr>
<tr class="even">
<td align="left">13</td>
<td align="left">Myeloid4</td>
</tr>
<tr class="odd">
<td align="left">14</td>
<td align="left">plasma cell</td>
</tr>
</tbody>
</table>
<pre class="r"><code>
# add cell types to metadata
hiv4 &lt;- AddMetaData(hiv4, metadata = Idents(hiv4), col.name = &quot;cell_type&quot;)


# add condition blood or csf to metadata
Idents(hiv4) &lt;- &quot;orig.ident&quot;
conditions &lt;- c(&quot;Blood&quot;, &quot;Blood&quot;, &quot;CSF&quot;, &quot;CSF&quot;)
names(conditions) &lt;- c(&quot;HIV1_Bld&quot;, &quot;HIV2_Bld&quot;, &quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;)
hiv4 &lt;- RenameIdents(hiv4, conditions)

hiv4 &lt;- AddMetaData(hiv4, col.name = &quot;Condition&quot;, metadata = Idents(hiv4))
</code></pre>
</details>
<p><strong>Overview of some of the most important marker genes:</strong></p>
<pre class="r"><code>
DotPlot(object = hiv4, features = c(&quot;MS4A1&quot;, # B cells
                                    &quot;TRAC&quot;, &quot;IL7R&quot;, 
                                    &quot;CD8A&quot;, # CD8 T cells
                                    &quot;NKG7&quot;, &quot;GNLY&quot;, # NK cells
                                    &quot;CD3E&quot;, # T cells
                                    &quot;CD14&quot;,&quot;FCGR3A&quot;, &quot;LYZ&quot;, # myeloid cells
                                    &quot;IGJ&quot;, # plasma cells
                                    &quot;AXL&quot;, &quot;APOE&quot;, &quot;TREM2&quot;, # myeloid 2
                                     &quot;FCER1A&quot;, &quot;CD1C&quot;) , # myeloid 5
        group.by = &quot;cell_type&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-26-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-26-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-26-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-26-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="tsne-visualizations-of-cell-types-conditions-samples-and-sequencing-quality" class="section level1">
<h1>tSNE visualizations of cell types, conditions, samples and sequencing quality</h1>
<pre class="r"><code>p1 &lt;- DimPlot(hiv4, reduction = &quot;tsne&quot;, label = TRUE, 
              group.by = &quot;cell_type&quot;, pt.size = 0.1) +
  NoLegend() +
  labs(title = &quot;Cell types&quot;)
p2 &lt;- DimPlot(hiv4, reduction = &quot;tsne&quot;, group.by =  &quot;Condition&quot;, pt.size = 0.1) +
  labs(title = &quot;CSF or Blood origin&quot;)
p3 &lt;- DimPlot(hiv4, reduction = &quot;tsne&quot;, group.by = &quot;orig.ident&quot;, pt.size = 0.1) +
  labs(title = &quot;Sample origin - patient&quot;)
p4 &lt;- FeaturePlot(hiv4, features= &quot;nCount_RNA&quot;, pt.size = 0.1) + 
    scale_color_viridis_c() +
  labs(title= &quot;number of counts&quot;)
ggarrange(p1,p2,p3,p4, ncol = 2, nrow = 2)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-27-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#p1 + p2 + p3 + p4</code></pre>
</div>
<div id="relative-contributions-of-csf-and-blood-cells-to-each-cluster" class="section level1">
<h1>Relative contributions of CSF and blood cells to each cluster</h1>
<p>The histogram below shows that Myeloid2 and Myeloid5 are predominantly composed of CSF cells, while Myeloid1, Myeloid3 and Myeloid4 are predominantly composed of Blood cells. This reflects the findings of <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> very nicely.</p>
<pre class="r"><code>
custom_color &lt;- c(&quot;#C3D7A4&quot;, &quot;#4E84C4&quot;, &quot;#52854C&quot;, &quot;#293352&quot;)

#plot a barplot of the proportions of samples and cell types
ggplot(hiv4@meta.data) +
  geom_bar(aes(x = cell_type, fill = orig.ident)) +
  scale_fill_manual( values = custom_color) +
  ylab(&quot;Number of cells in each cluster&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
  labs(title = &quot;Relative contribution of CSF and blood samples to each cluster&quot;)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-28-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-28-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-28-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-28-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Question</strong>: Myeloid1, Myeloid2 are mainly found in Blood cells, Myeloid4 is only found in Blood cells from HIV1 patient. What does that mean?</p>
<div id="how-big-are-myeloid2-and-myeloid5-clusters-and-where-do-cells-come-from" class="section level3">
<h3>How big are Myeloid2 and Myeloid5 clusters and where do cells come from?</h3>
<p>Myeloid2 cells originate from the CSF samples of HIV infected patients, while a few cells (thirteen) classified as Myeloid5 cells originate from Blood samples.</p>
<pre class="r"><code>hiv4@meta.data %&gt;% 
  filter(cell_type %in% c(&quot;Myeloid2&quot;, &quot;Myeloid5&quot;)) %&gt;% 
  group_by(cell_type, orig.ident) %&gt;% 
  summarize(n = n()) %&gt;%
  kable(caption = &quot;Number of Myeloid2 &amp; Myeloid5 cells from different patients/samples&quot;)</code></pre>
<table>
<caption>Number of Myeloid2 &amp; Myeloid5 cells from different patients/samples</caption>
<thead>
<tr class="header">
<th align="left">cell_type</th>
<th align="left">orig.ident</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Myeloid2</td>
<td align="left">HIV1_CSF</td>
<td align="right">71</td>
</tr>
<tr class="even">
<td align="left">Myeloid2</td>
<td align="left">HIV2_CSF</td>
<td align="right">35</td>
</tr>
<tr class="odd">
<td align="left">Myeloid5</td>
<td align="left">HIV1_Bld</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">Myeloid5</td>
<td align="left">HIV1_CSF</td>
<td align="right">24</td>
</tr>
<tr class="odd">
<td align="left">Myeloid5</td>
<td align="left">HIV2_Bld</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">Myeloid5</td>
<td align="left">HIV2_CSF</td>
<td align="right">68</td>
</tr>
</tbody>
</table>
</div>
<div id="which-genes-are-upregulated-in-myeloid-2-compared-to-the-four-other-myeloid-subsets" class="section level2">
<h2>Which genes are upregulated in Myeloid-2 compared to the four other myeloid subsets?</h2>
<p>In the paper they find 60 genes that are upregulated compared to the other 4 myeloid subsets. (FDR &lt; 0.01)</p>
<p>Compare the genes upregulated in myeloid-2 to the four other myeloid subsets. <code>FindAllMarkers()</code> will compare one cluster to all other clusters. <code>FindMarkers()</code> will compare an indicated cluster with a second indicated cluster. Therefore, the difference will not be diluted, which might be the case if we compare one cluster to many other clusters.</p>
<pre class="r"><code>#import the 60 genes that were upregulated in Myeloid-2 in the paper compared to the four other myeloid subsets
paper &lt;- read.table(file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/Myeloid-2_genes_paper.csv&quot;, sep = &quot;\t&quot;,
                    header = FALSE, skipNul = TRUE)#, n_max = 60)
paper[1,] &lt;- &quot;APOC1&quot;
colnames(paper) &lt;- &quot;&quot;
paper &lt;- as.vector(paper[,1])

Idents(hiv4) &lt;- &quot;cell_type&quot;
all.comp &lt;- purrr::map(c(1,3,4,5), function(n){
  markers &lt;- FindMarkers(hiv4,
              ident.1 = &quot;Myeloid2&quot;,
              ident.2 = paste0(&quot;Myeloid&quot;, n),
              only.pos = TRUE, 
              verbose = FALSE)
  markers &lt;- rownames_to_column(markers, var = &quot;gene&quot;)
  name &lt;- paste0(&quot;Comparison Myeloid 2 vs Myeloid &quot;, n)
  list(df = markers, name = name)
})

my2_genes &lt;- map(seq.int(1,4), function(n) {
  name &lt;- all.comp[[]]$name
  my2_genes &lt;- all.comp[[n]]$df %&gt;% 
  filter(p_val_adj &lt;= .2) %&gt;%
  filter(gene %in% paper)
  my2_genes &lt;- my2_genes[,&quot;gene&quot;]
})</code></pre>
<div id="expression-of-the-60-genes-identified-in-the-paper-as-myeloid-2-markers" class="section level3">
<h3>Expression of the 60 genes identified in the paper as Myeloid 2 markers</h3>
<p>The genes show the highest expression in Myeloid2 and for some genes like CD74 and genes of the HLA family are also highly expressed in Myeloid5 cells. However, there are a few genes, like FAU, TPT1, TMSB10 and MIR4426 which do not show expression specific to the Myeloid2 cluster. It is unclear why <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> chose these genes as signature genes.</p>
<pre class="r"><code>DotPlot(hiv4, features = c(paper)) + 
  coord_flip() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1))
Warning in FetchData(object = object, vars = features, cells = cells): The
following requested variables were not found: SELENOP</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-31-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="percentage-of-myeloid2-transcripts-per-cell-in-blood-csf-and-uninfected-samples" class="section level3">
<h3>Percentage of Myeloid2 transcripts per Cell in Blood, CSF and Uninfected samples</h3>
<p>First, it is important to mention, that after quality control the number of cells is significantly lower in the two uninfected CSF samples, 216 and 340 respectively, compared to all other samples which contain between 932 and 2520 cells. However, also the CSF sample from patient1 (HIV1_CSF) with 932 cells contains much fewer cells compared to all other samples.</p>
<p>The results shown by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> are controversial, because when they compare the percentage of Myeloid2 transcripts between the three conditions, they pool 1770 cells for the uninfected condition, which is a lot more than we find here. Yet, they do not mention that they used different thresholds for quality control of the uninfected samples. Therefore, I was not able to reproduce their findings. Below, I will describe the percentage of Myeloid2 transcripts when comparing the conditions with the number of cells obtained when using the same thresholds for quality control as described in the paper.</p>
<p>The interpretability of the comparison probably suffers from the fact that the number of cells in the uninfected conditions is so much smaller.</p>
<pre class="r"><code>hiv7 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_filtered_seurat_object.rds&quot;)

hiv7@meta.data %&gt;% group_by(orig.ident) %&gt;% 
  summarise(number_of_cells = n()) %&gt;% 
  kable(caption = &quot;Number of cells per sample&quot;)</code></pre>
<table>
<caption>Number of cells per sample</caption>
<thead>
<tr class="header">
<th align="left">orig.ident</th>
<th align="right">number_of_cells</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">2076</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">932</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">2520</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">2309</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">1384</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">216</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">340</td>
</tr>
</tbody>
</table>
<pre class="r"><code>hiv7 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_filtered_seurat_object.rds&quot;)

# create a list for the 3 different conditons
subset.list &lt;- list(infected_blood = c(&quot;HIV1_Bld&quot;, &quot;HIV2_Bld&quot;), 
                    infected_csf = c(&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;),
                    uninfected_csf = c(&quot;Uninfected1_CSF&quot;, &quot;Uninfected2_CSF&quot;))

# for each cell of each condition calculate a percentage of Myeloid2 transcripts
# and combine to a dataframe
percentage_my2_transcripts &lt;- map2_dfr(subset.list, names(subset.list), function(n, name) {
  count_m &lt;- subset(hiv7, orig.ident %in% n)@assays$RNA@counts
  total_transcripts &lt;- colSums(count_m) # number of transcripts per cell
  my2_matrix &lt;- count_m[rownames(count_m) %in% paper, ] # get only myeloid2 genes 
  my2_transcripts &lt;- colSums(my2_matrix) # sum of myeloid2 transcripts
  tibble(percentage =  my2_transcripts/total_transcripts, condition = name)
})</code></pre>
<p><strong>Is there a significant difference in percentage of Myeloid2 transcripts between the three conditions?</strong></p>
<pre class="r"><code>percentage_my2_transcripts %&gt;% 
  group_by(condition) %&gt;% 
  summarise(n = n(), q25 = quantile(percentage, .25),
            q50 = quantile(percentage, .5), 
            q75 = quantile(percentage, .75), 
            mean = mean(percentage),
            median = median(percentage)) %&gt;% 
  kable(digits = 4, 
        caption = &quot;Summary statistics of percentage of Myeloid2 transcripts for different conditions&quot;)</code></pre>
<table>
<caption>Summary statistics of percentage of Myeloid2 transcripts for different conditions</caption>
<thead>
<tr class="header">
<th align="left">condition</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
<th align="right">median</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">infected_blood</td>
<td align="right">4596</td>
<td align="right">0.0155</td>
<td align="right">0.0189</td>
<td align="right">0.0224</td>
<td align="right">0.0192</td>
<td align="right">0.0189</td>
</tr>
<tr class="even">
<td align="left">infected_csf</td>
<td align="right">4625</td>
<td align="right">0.0151</td>
<td align="right">0.0183</td>
<td align="right">0.0218</td>
<td align="right">0.0203</td>
<td align="right">0.0183</td>
</tr>
<tr class="odd">
<td align="left">uninfected_csf</td>
<td align="right">556</td>
<td align="right">0.0120</td>
<td align="right">0.0151</td>
<td align="right">0.0287</td>
<td align="right">0.0216</td>
<td align="right">0.0151</td>
</tr>
</tbody>
</table>
<p>To see if there is a difference between the three conditions a Kruskal-Wallis-Test was performed. We get a p-value of 4.88 * 10^-9. Therefore, we will perform a pairwise Wilcox test to see which condtions are different from each other.</p>
<pre class="r"><code>percentage_my2_transcripts %&gt;% 
  rstatix::kruskal_test(percentage ~ condition) %&gt;%
  kable(caption = &quot;Kruskal Wallis Test&quot;)</code></pre>
<table>
<caption>Kruskal Wallis Test</caption>
<thead>
<tr class="header">
<th align="left">.y.</th>
<th align="right">n</th>
<th align="right">statistic</th>
<th align="right">df</th>
<th align="right">p</th>
<th align="left">method</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">percentage</td>
<td align="right">9777</td>
<td align="right">38.27649</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="left">Kruskal-Wallis</td>
</tr>
</tbody>
</table>
<p><strong>Pairwise Wilcox test:</strong></p>
<p>In the table and boxplot below we can see that all three conditions are significantly different from each other. As mentioned above, the sample size is much smaller for the uninfected condition. P-values are listed in the table below.</p>
<p>Only the p-value for the comparison between infected CSF vs uninfected CSF is the same as in the paper, where they state that the Myeloid2 transcripts were found at lower levels in the uninfected samples compared to the infected samples. Otherwise, the results contradict the results found by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span>. Contradictory to what is claimed in the paper, it is not evident that both infected and uninfected CSF samples have higher rates of Myeloid2 transcripts, rather the Blood samples seem to have the highest rate of Myeloid2 transcripts.</p>
<p>The comparison drawn is questionable in general, because what the boxplots below show is that there is some level of Myeloid2 transcript expression in all conditions. However, there are more outliers with very high percentage of Myeloid2 transcripts in both CSF samples. This might be a hint that in the CSF samples there is a outlier population representing the Myeloid2 cluster, which cannot be seen in the blood sample. Since the median is rather robust to outliers, even if in CSF samples there are some cells which have much higher frequencies of transcripts, it might be that the median is still lower than in blood samples.</p>
<p>Therefore, comparing the overall percentage of Myeloid2 transcripts might not provide any insights on the number of Myeloid2 cells in a particular sample.</p>
<pre class="r"><code>p1 &lt;- percentage_my2_transcripts %&gt;% ggplot() +
  geom_boxplot(aes(x = condition, y = percentage, fill = condition)) +
  xlab(&quot;Sample origin&quot;) +
  ylab(&quot;Frequency of myeloid2 transcripts per cell&quot;) +
  labs(title = &quot;All cells&quot;) +
  stat_pvalue_manual(percentage_my2_transcripts %&gt;% 
    rstatix::pairwise_wilcox_test(percentage ~ condition) %&gt;% 
      rstatix::add_xy_position()) +
  NoLegend()

p2 &lt;- percentage_my2_transcripts %&gt;% ggplot() +
  geom_boxplot(aes(x = condition, y = percentage, fill = condition), 
               outlier.shape = NA) +
  xlab(&quot;Sample origin&quot;) +
  ylab(&quot;Frequency of myeloid2 transcripts per cell&quot;) +
  labs(title = &quot;Remove outliers to see difference&quot;) +
  NoLegend()

ggarrange(p1, p2, ncol = 2, nrow = 1)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-36-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>percentage_my2_transcripts %&gt;% 
  rstatix::wilcox_test(percentage ~ condition) %&gt;% kable(caption = &quot;Pairwise Wilcox Test&quot;)</code></pre>
<table>
<caption>Pairwise Wilcox Test</caption>
<thead>
<tr class="header">
<th align="left">.y.</th>
<th align="left">group1</th>
<th align="left">group2</th>
<th align="right">n1</th>
<th align="right">n2</th>
<th align="right">statistic</th>
<th align="right">p</th>
<th align="right">p.adj</th>
<th align="left">p.adj.signif</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">percentage</td>
<td align="left">infected_blood</td>
<td align="left">infected_csf</td>
<td align="right">4596</td>
<td align="right">4625</td>
<td align="right">11090858</td>
<td align="right">2.95e-04</td>
<td align="right">2.95e-04</td>
<td align="left">***</td>
</tr>
<tr class="even">
<td align="left">percentage</td>
<td align="left">infected_blood</td>
<td align="left">uninfected_csf</td>
<td align="right">4596</td>
<td align="right">556</td>
<td align="right">1441190</td>
<td align="right">8.00e-07</td>
<td align="right">1.60e-06</td>
<td align="left">****</td>
</tr>
<tr class="odd">
<td align="left">percentage</td>
<td align="left">infected_csf</td>
<td align="left">uninfected_csf</td>
<td align="right">4625</td>
<td align="right">556</td>
<td align="right">1456760</td>
<td align="right">3.00e-07</td>
<td align="right">9.00e-07</td>
<td align="left">****</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="label-transfer" class="section level1">
<h1>Label Transfer</h1>
<p>After reproducing the results from <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> the next step was to transfer the labels from the four HIV+ Blood and CSF samples to the remaining samples:</p>
<ul>
<li>1 CSF smaple from HIV-1 infected patient</li>
<li>2 CSF samples from uninfected patients</li>
</ul>
<p>For reference see: <a href="https://satijalab.org/seurat/articles/integration_mapping.html" class="uri">https://satijalab.org/seurat/articles/integration_mapping.html</a></p>
<pre class="r"><code># Seurat objects
hiv4 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/non_integrated_HIV1_HIV2_4samples_seurat_object.rds&quot;)
#hiv4 &lt;- readRDS(&quot;seurat_objects/Seurat_objects/non_integrated_HIV1_HIV2_4samples_seurat_object.rds&quot;)</code></pre>
<div id="find-integration-anchors" class="section level2">
<h2>Find Integration anchors</h2>
<p>We perform Normalization separately for each dataset and try to find variable Features.</p>
<pre class="r"><code>#hiv_list &lt;- SplitObject(hiv7, split.by = &quot;orig.ident&quot;)

#for (i in 1:length(hiv_list)) {
 # hiv_list[[i]] &lt;- NormalizeData(hiv_list[[i]], verbose = FALSE)
  #hiv_list[[i]] &lt;- FindVariableFeatures(hiv_list[[1]],
   #                                     selection.method = &quot;vst&quot;,
    #                                    nfeatures = 2000,
     #                                   verbose = FALSE)
#}

#saveRDS(hiv_list, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv_list&quot;)
#hiv_list &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv_list&quot;)
#reference.list &lt;- SplitObject(hiv4, split.by = &quot;orig.ident&quot;)</code></pre>
<pre class="r"><code>hiv_filt &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_filtered_seurat_object.rds&quot;)
hiv_list &lt;- SplitObject(hiv_filt, split.by = &quot;orig.ident&quot;)</code></pre>
</div>
<div id="cell-type-classification-using-an-integrated-reference" class="section level2">
<h2>Cell type classification using an integrated reference</h2>
<ul>
<li>in data transfer we begin with identifying anchors</li>
<li>we do not correct or modify the query expression data</li>
<li>it is suggested to use the default method tsne (which project the PCA structure of a reference onto a query dataset (instead of learning a joint structure)</li>
<li>we classify the query cells based on the reference data (a vector of reference cell type labels)</li>
</ul>
<p>We will use the four samples we have analyzed above as a reference. The query dataset will be the HIV+ CSF3 sample and the two uninfected sample:</p>
<p>Myeloid 1, 3, 4 are not/hardly present in the CSF samples which is expected, since they originated exclusively from the Blood samples in the reference dataset.</p>
<pre class="r"><code># merge the 3 query Seurat objects
query &lt;- merge(hiv_list[[&quot;Uninfected1_CSF&quot;]], 
               c(hiv_list[[&quot;Uninfected2_CSF&quot;]], hiv_list[[&quot;HIV3_CSF&quot;]]))

# find transfer anchors between the 3 query samples and the 4 samples we have 
# analyzed previously
hiv4_transfer_anchors &lt;- FindTransferAnchors(reference = hiv4,
                                             query = query, 
                                             dims = 1:15,
                                             normalization.method = &quot;LogNormalize&quot;,
                                             reference.reduction = &quot;pca&quot;,
                                             reduction = &quot;pcaproject&quot;,
                                             verbose = FALSE)

predictions &lt;- TransferData(anchorset = hiv4_transfer_anchors, 
                            refdata = hiv4$cell_type, 
                            dims = 1:15,
                            verbose = FALSE)
#predictions %&gt;% head %&gt;% kable()

query &lt;- AddMetaData(query, metadata = predictions)
#saveRDS(query, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/query_seurat_object_label_transfer&quot;)
table(query$predicted.id) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Bcell</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">CD8</td>
<td align="right">137</td>
</tr>
<tr class="odd">
<td align="left">CD81</td>
<td align="right">271</td>
</tr>
<tr class="even">
<td align="left">mixed</td>
<td align="right">57</td>
</tr>
<tr class="odd">
<td align="left">Myeloid1</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">Myeloid2</td>
<td align="right">132</td>
</tr>
<tr class="odd">
<td align="left">Myeloid4</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">Myeloid5</td>
<td align="right">43</td>
</tr>
<tr class="odd">
<td align="left">NKcell1</td>
<td align="right">55</td>
</tr>
<tr class="even">
<td align="left">plasma cell</td>
<td align="right">12</td>
</tr>
<tr class="odd">
<td align="left">Tcell1</td>
<td align="right">153</td>
</tr>
<tr class="even">
<td align="left">Tcell2</td>
<td align="right">732</td>
</tr>
<tr class="odd">
<td align="left">Tcell4</td>
<td align="right">341</td>
</tr>
</tbody>
</table>
<div id="check-the-gene-expression-levels-in-the-predicted-cell-types" class="section level3 tabset">
<h3 class="tabset">Check the gene expression levels in the predicted cell types</h3>
<p>The Label Transfer seems to work nicely, judging by the correct expression of common cell type markers for the individual cell types.</p>
<div id="myeloid2-marker-genes" class="section level4">
<h4>Myeloid2 marker genes</h4>
<pre class="r"><code># ceck the Myeloid2 markers
p1 &lt;- VlnPlot(query,
              features =  c(&quot;AXL&quot;, &quot;APOE&quot;, &quot;TREM2&quot;), 
              group.by = &quot;predicted.id&quot;, 
              pt.size = .01, ncol = 3)
p2 &lt;- VlnPlot(query, 
              features =  c(&quot;CTSB&quot;, &quot;MSR1&quot;, &quot;C1QA&quot;), 
              group.by = &quot;predicted.id&quot;,
              pt.size = .01, ncol = 4)

figure &lt;- ggarrange(p1, p2, ncol = 1, nrow = 2)
annotate_figure(figure, top = text_grob(&quot;Myeloid2 marker genes&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-42-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="myeloid5-marker-genes" class="section level4">
<h4>Myeloid5 marker genes</h4>
<pre class="r"><code>
# Myeloid 5 markers
p1 &lt;- VlnPlot(query, c(&quot;FCER1A&quot;, &quot;CD1C&quot;),
        group.by = &quot;predicted.id&quot;, pt.size = .01, ncol = 2) 
p1</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-43-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="plasma-cell-marker-genes" class="section level4">
<h4>Plasma cell marker genes</h4>
<pre class="r"><code># plasma cell markers
p2 &lt;- VlnPlot(query, features = c(&quot;IGJ&quot;), 
        group.by = &quot;predicted.id&quot;, pt.size = .01, ncol = 2) 
p2</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-44-1">
Past versions of unnamed-chunk-44-1.png
</button>
</p>
<div id="fig-unnamed-chunk-44-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-44-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-44-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-44-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-44-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cd8-t-cell-marker-genes" class="section level4">
<h4>CD8 T cell marker genes</h4>
<pre class="r"><code># CD8 T Cells
p3 &lt;- VlnPlot(query, features = c(&quot;CD8A&quot;),
        group.by = &quot;predicted.id&quot;, pt.size = .01, ncol = 2) + NoLegend()
p3</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-45-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-45-1">
Past versions of unnamed-chunk-45-1.png
</button>
</p>
<div id="fig-unnamed-chunk-45-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-45-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-45-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-45-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-45-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="t-cell-marker-genes" class="section level4">
<h4>T cell marker genes</h4>
<pre class="r"><code># T cells 
VlnPlot(query, features = c(&quot;IL7R&quot;, &quot;TRAC&quot;),
        group.by = &quot;predicted.id&quot;, pt.size = .01, ncol = 2)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-46-1">
Past versions of unnamed-chunk-46-1.png
</button>
</p>
<div id="fig-unnamed-chunk-46-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-46-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-46-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-46-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-46-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nk-cell-marker-genes" class="section level4">
<h4>NK cell marker genes</h4>
<pre class="r"><code># NK cells
VlnPlot(query, features = c(&quot;NKG7&quot;, &quot;GNLY&quot;, &quot;GZMB&quot;), 
        group.by = &quot;predicted.id&quot;, pt.size = .01, ncol = 3)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-47-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-47-1">
Past versions of unnamed-chunk-47-1.png
</button>
</p>
<div id="fig-unnamed-chunk-47-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-47-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-47-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-47-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-47-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="b-cell-marker-genes" class="section level4">
<h4>B cell marker genes</h4>
<pre class="r"><code># B cells
VlnPlot(query, features = c(&quot;MS4A1&quot;, &quot;CD79A&quot;, &quot;TCL1A&quot;), 
        group.by = &quot;predicted.id&quot;, pt.size = .01, ncol = 3)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-48-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-48-1">
Past versions of unnamed-chunk-48-1.png
</button>
</p>
<div id="fig-unnamed-chunk-48-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-48-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-48-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-48-1.png" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-48-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
</div>
<div id="section-3" class="section level3 unnumbered">
<h3 class="unnumbered"></h3>
</div>
</div>
<div id="umap-projection-of-additional-samples" class="section level2">
<h2>UMAP projection of additional samples</h2>
<ul>
<li>Calculate UMAP on the four reference samples</li>
<li>Integrate the embeddings of the reference samples with the embeddings of the query samples</li>
</ul>
<pre class="r"><code>hiv4_test &lt;- RunUMAP(hiv4, dims = 1:15, 
                     reduction = &quot;pca&quot;, 
                     return.model = TRUE, 
                     verbose = FALSE)
Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to &#39;umap-learn&#39; and metric to &#39;correlation&#39;
This message will be shown once per session

hiv3csf &lt;- MapQuery(anchorset = hiv4_transfer_anchors,
                    reference = hiv4_test, query = query,
                    refdata = list(celltype = &quot;cell_type&quot;),
                    reference.reduction = &quot;pca&quot;,
                    reduction.model = &quot;umap&quot;, 
                    verbose = FALSE)
Warning: Keys should be one or more alphanumeric characters followed by an
underscore, setting key from ref.pca_ to refpca_
Warning: UNRELIABLE VALUE: One of the &#39;future.apply&#39; iterations
(&#39;future_lapply-1&#39;) unexpectedly generated random numbers without declaring so.
There is a risk that those random numbers are not statistically sound and the
overall results might be invalid. To fix this, specify &#39;future.seed=TRUE&#39;. This
ensures that proper, parallel-safe random numbers are produced via the L&#39;Ecuyer-
CMRG method. To disable this check, use &#39;future.seed = NULL&#39;, or set option
&#39;future.rng.onMisuse&#39; to &quot;ignore&quot;.
Warning: Keys should be one or more alphanumeric characters followed by an
underscore, setting key from ref.pca_ to refpca_
Warning: All keys should be one or more alphanumeric characters followed by an
underscore &#39;_&#39;, setting key to refpca_
Warning: No assay specified, setting assay as RNA by default.




query &lt;- TransferData(anchorset = hiv4_transfer_anchors,
                               reference = hiv4_test,
                               query = query,
                               refdata = list(celltype = &quot;cell_type&quot;), 
                      verbose = FALSE)

query &lt;- IntegrateEmbeddings(anchorset = hiv4_transfer_anchors,
                                      reference = hiv4_test,
                                      query = query,
                               new.reduction.name = &quot;ref.pca&quot;,
                             verbose = FALSE)
Warning: Keys should be one or more alphanumeric characters followed by an
underscore, setting key from ref.pca_ to refpca_
Warning: UNRELIABLE VALUE: One of the &#39;future.apply&#39; iterations
(&#39;future_lapply-1&#39;) unexpectedly generated random numbers without declaring so.
There is a risk that those random numbers are not statistically sound and the
overall results might be invalid. To fix this, specify &#39;future.seed=TRUE&#39;. This
ensures that proper, parallel-safe random numbers are produced via the L&#39;Ecuyer-
CMRG method. To disable this check, use &#39;future.seed = NULL&#39;, or set option
&#39;future.rng.onMisuse&#39; to &quot;ignore&quot;.
Warning: Keys should be one or more alphanumeric characters followed by an
underscore, setting key from ref.pca_ to refpca_
Warning: All keys should be one or more alphanumeric characters followed by an
underscore &#39;_&#39;, setting key to refpca_

query &lt;- ProjectUMAP(query = query,
                       query.reduction = &quot;ref.pca&quot;,
                       reference = hiv4_test,
                       reference.reduction = &quot;pca&quot;,
                       reduction.model = &quot;umap&quot;, verbose = FALSE)
Warning: No assay specified, setting assay as RNA by default.</code></pre>
<pre class="r"><code># Visualize the query cells alongside the reference
p1 &lt;- DimPlot(hiv4_test,
              reduction = &quot;umap&quot;,
              group.by = &quot;cell_type&quot;,
              label = TRUE,
              label.size = 5,
              repel = TRUE, 
              pt.size = .1) + NoLegend() +
  ggtitle(&quot;Reference anntoations&quot;)

p2 &lt;- DimPlot(query, 
              reduction = &quot;ref.umap&quot;,
              group.by = &quot;predicted.celltype&quot;, 
              label = TRUE,
              label.size = 5,
              pt.size = .1,
              repel = TRUE) + NoLegend() +
   ggtitle(&quot;Transferred Labels&quot;)

p1 + p2</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-50-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#save this new Seurat obejct including the SCTransformatin

#saveRDS(hiv_label_transfer, file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds&quot;)</code></pre>
</div>
<div id="frequency-of-cell-types-in-each-sample" class="section level2">
<h2>Frequency of cell types in each sample</h2>
<p>How are cell types distributed across samples, considering also the additional HIV+ CSF sample and the two uninfected samples which were annotated using label transfer.</p>
<pre class="r"><code>hiv7 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds&quot;)

custom_color &lt;- c(&quot;#C3D7A4&quot;, &quot;#4E84C4&quot;, &quot;#52854C&quot;, &quot;#293352&quot;, &quot;lightskyblue&quot;, &quot;orangered2&quot;, &quot;orange2&quot;)

#plot a barplot of the proportions of samples and cell types
p2 &lt;- ggplot(hiv7@meta.data) +
  geom_bar(aes(x = cell_type, fill = orig.ident)) +
  scale_fill_manual( values = custom_color) +
  ylab(&quot;Number of cells in each cluster&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
  labs(title = &quot;Frequency of cell types from each sample after label transfer&quot;)

custom_color &lt;- c(&quot;#C3D7A4&quot;, &quot;#4E84C4&quot;, &quot;#52854C&quot;, &quot;#293352&quot;)

#plot a barplot of the proportions of samples and cell types
p1 &lt;- ggplot(hiv4@meta.data) +
  geom_bar(aes(x = cell_type, fill = orig.ident)) +
  scale_fill_manual( values = custom_color) +
  ylab(&quot;Number of cells in each cluster&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
  labs(title = &quot;Frequecy of cell types from each sample after label transfer&quot;)

ggarrange(p1, p2, ncol = 2, nrow = 1)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-52-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="independent-annotation-of-myeloid2-and-myeloid5-clusters-as-microglia-like-cells" class="section level1">
<h1>Independent annotation of Myeloid2 and Myeloid5 clusters as Microglia-like cells</h1>
<p>In the following part we tried to confirm that the Myeloid2 (and maybe the Myeloid5) cluster are microglia-like cells. <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> used 60 genes to characterize Myeloid2 cells, some of which are found in disease-associated microglia. Searching the literature for additional microglia signatures we might be able to independently confirm this microglia-like signature.</p>
<p>Below you can click on the tabs to see the results of using different gene signatures to confirm the microglia-like signature of Myeloid2 and Myeloid5. The function used to calculate the gene signature score in different cells is <code>AddModuleScore()</code>.</p>
<div id="function-add-module-score" class="section level4">
<h4>Function Add Module Score</h4>
<p>This Function calculates the average expression levels of each cluster on single cell level, subtracted by the aggregated expression of control feature sets. The analyzed features are binned based on averaged expression and the control features are randomly selected from each bin.</p>
<pre class="r"><code># read in Seurat object
hiv4 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/non_integrated_HIV1_HIV2_4samples_seurat_object.rds&quot;)</code></pre>
</div>
<div id="different-gene-signatures-found-in-the-literature" class="section level2 tabset">
<h2 class="tabset">Different gene signatures found in the literature</h2>
<div id="farhadian-et-al." class="section level3">
<h3>Farhadian et al.</h3>
<p>60 genes were used by Farhadian et al. to distinguish the Myeloid2 cell type. 28 of the 60 genes used as microglia signature in the <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> Paper are also found in the 881 microglia signature genes defined by <span class="citation">(<a href="#ref-Gosselin" role="doc-biblioref">David Gosselin 2020</a>)</span>.</p>
<pre class="r"><code>## Read in list of microglia signature genes found in the Gosselin paper
## which are also found in the HIV1 and HIV2 samples:
microglia_genes &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/Microglia_gene_signatur_found_in_HIV1_HIV2_samples&quot;)</code></pre>
<pre class="r"><code># Are there any genes overlapping between Farhadian and Glass paper?
#import the 60 genes that were upregulated in Myeloid-2 in the paper compared to the four other myeloid subsets
paper &lt;- read.table(file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/Myeloid-2_genes_paper.csv&quot;, sep = &quot;\t&quot;, header = FALSE, skipNul = TRUE)#, n_max = 60)
paper[1,] &lt;- &quot;APOC1&quot;
colnames(paper) &lt;- &quot;&quot;
paper &lt;- as.vector(paper[,1])

# Are there any genes overlapping between Farhadian and Glass paper?
#paper %in% as.vector(microglia_genes[[&quot;gene_name&quot;]])

overlap &lt;- as.vector(microglia_genes[[&quot;gene_name&quot;]]) [as.vector(microglia_genes[[&quot;gene_name&quot;]]) %in% paper]
#length(overlap)
#overlap %&gt;% kable()</code></pre>
<pre class="r"><code># get only genes present in the HIV1 and HIV2 samples!
paper &lt;- paper[paper %in% rownames(hiv4@assays$RNA@counts)]


hiv_farhadian &lt;- AddModuleScore(hiv4, 
                           features = paper)

Idents(hiv_farhadian) &lt;- &quot;cell_type&quot;
FeaturePlot(hiv_farhadian, features = &quot;Cluster1&quot; , pt.size = .5, label = TRUE, repel = TRUE) +
    scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Farhadian et al., 2018: Microglia signature genes&quot;)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-56-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gosselin-et-al." class="section level3">
<h3>Gosselin et al. </h3>
<p>Here, I will first use the 881 microglia marker genes and then also try the list of TFs provided by <span class="citation">(<a href="#ref-Gosselin" role="doc-biblioref">David Gosselin 2020</a>)</span>.</p>
<p>The Transcription Factors mentioned in the paper do not seem to be specific for the Myeloid2 cluster. Interestingly, there is a very low score<br />
of these microglia genes in Myeloid5, while there is a higher score in Myeloid1 and Myeloid4 cells.</p>
<pre class="r"><code>gosselin_tfs &lt;- read.table( &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/TFs_microglia_Glass_paper.txt&quot;)


hiv4_glass &lt;- AddModuleScore(hiv4, 
                           features = as.vector(gosselin_tfs))
Warning: The following features are not present in the object: MEF2B, SALL1, not
searching for symbol synonyms

Idents(hiv4_glass) &lt;- &quot;cell_type&quot;
FeaturePlot(hiv4_glass, features = &quot;Cluster1&quot;, label = TRUE, repel = TRUE, pt.size = .7) +
    scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Gosselin D. et al., 2017: Microglia Transcription Factors&quot;) </code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-57-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-57-1">
Past versions of unnamed-chunk-57-1.png
</button>
</p>
<div id="fig-unnamed-chunk-57-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-57-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-57-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="hand-signature" class="section level3">
<h3>HAND signature</h3>
<p>(HAND = HIV-associated neurodegenerative disease)</p>
<p><span class="citation">(<a href="#ref-Gosselin" role="doc-biblioref">David Gosselin 2020</a>)</span> defined a group of genes upregulated in microglia from patients with HAND.</p>
<p>This gene signature can, however not be detected in any of the Myeloid clusters.</p>
<pre class="r"><code>hand_genes &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/HAND_upregulated_microglia_signature_genes&quot;)
hand_genes &lt;- as.vector(hand_genes[[&quot;genes&quot;]])

hiv_hand &lt;- AddModuleScore(hiv4,
                           features = hand_genes)

Idents(hiv_hand) &lt;- &quot;cell_type&quot;
FeaturePlot(hiv_hand, features = &quot;Cluster1&quot;, pt.size = .01) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;microglia signature genes upregulated in HAND&quot;)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-58-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-58-1">
Past versions of unnamed-chunk-58-1.png
</button>
</p>
<div id="fig-unnamed-chunk-58-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-58-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-58-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="grassivaro-et-al." class="section level3">
<h3>Grassivaro et al.</h3>
<p><span class="citation">(<a href="#ref-Grassivaro" role="doc-biblioref">Francesca Grassivaro and Martino 2018</a>)</span> identified features related to convergence of microglia and macrophages. They identified 14 genes that are conserved in microglia throughout development, but not expressed in myeloid-derived monocytes (MDM). Only in the context of neuroinflammation are the 14 genes expressed in macrophages as well.</p>
<pre class="r"><code>tfs &lt;- c(&quot;CRYBB1&quot;, &quot;GARNL3&quot;, &quot;GPR34&quot;, &quot;LAG3&quot;, &quot;NUAK1&quot;, &quot;OLFML3&quot;, &quot;RTN1&quot;,  &quot;SALL3&quot;,  &quot;SLC1A3&quot;, &quot;SPARC&quot;, &quot;TNFRSF17&quot;)

tfs %in% rownames(hiv4@assays$RNA@counts)
 [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE


hivt &lt;- AddModuleScore(hiv4,
                           features = tfs)

Idents(hivt) &lt;- &quot;cell_type&quot;
FeaturePlot(hivt, features = &quot;Cluster2&quot;,
            label = TRUE, 
            repel = TRUE, pt.size = .1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;14 genes expressed in microglia &amp; macrophages in neuroinfl.&quot;)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-59-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-59-1">
Past versions of unnamed-chunk-59-1.png
</button>
</p>
<div id="fig-unnamed-chunk-59-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-59-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-59-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="keren-shaul-et-al.-kraseman-et-al." class="section level3">
<h3>Keren-Shaul et al. &amp; Kraseman et al.</h3>
<p>The marker genes mentioned by <span class="citation">(<a href="#ref-Keren-Shaul" role="doc-biblioref">Hadas Keren-Shaul 2017</a>)</span> and <span class="citation">(<a href="#ref-Kraseman" role="doc-biblioref">Susanne Krasemann 2017</a>)</span> were referred to by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> and used as marker genes for microglia associated with neurodegenerative disease.</p>
<p>Neither of these signatures show a high score for the Myeloid2/5 clusters.</p>
<ul>
<li>The signature genes defined by <span class="citation">(<a href="#ref-Keren-Shaul" role="doc-biblioref">Hadas Keren-Shaul 2017</a>)</span> are expressed to some level in all myeloid clusters</li>
<li>The signature defined by <span class="citation">(<a href="#ref-Kraseman" role="doc-biblioref">Susanne Krasemann 2017</a>)</span> is also mainly expressed in the Myeloid clusters</li>
<li>CD9 and SPP1 might be added to the list of microglia marker genes expressed in Myeloid2 cluster</li>
</ul>
<p>It is not clear why <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> used these genes as marker genes.</p>
<pre class="r"><code># upregulated in disease associated microglia (Keren-Shaul)
disease &lt;- c(&quot;SPP1&quot;, &quot;ITGAX&quot;, &quot;AXL&quot;, &quot;LILRB4&quot;, &quot;CLEC7A&quot;, &quot;CCL2&quot;, &quot;CSF1&quot;) 

#28 inflammatory molecules upregulated in disease associated microglia (Krasemann)
mgnd &lt;- c(&quot;APOE&quot;, &quot;AXL&quot;, &quot;CCL2&quot;, &quot;TLR2&quot;, &quot;SPP1&quot;, &quot;CYBB&quot;, &quot;MSR1&quot;, &quot;ITGAX&quot;,
          &quot;CLEC7A&quot;, &quot;CHI3L3&quot;, &quot;ARG1&quot;, &quot;SIGLEC1&quot;, &quot;CFP&quot;, &quot;CXCL10&quot;, &quot;ALCAM&quot;, 
          &quot;FER1L3&quot;, &quot;LILRB4&quot;, &quot;GPX3&quot;, &quot;GAS7&quot;, &quot;CCRL2&quot;,&quot;CXCL16&quot;, &quot;CXCR4&quot;,
          &quot;GPNMB&quot;, &quot;LGALS3&quot;, &quot;IFI202B&quot;, &quot;CSF1&quot;, &quot;LIRB4&quot;, &quot;LAG3&quot;)
# only APOE, AXL and MSR1 overlap, but these are the most important genes

# remove the genes that are also used by the Farhadian Paper
mgnd &lt;- c(&quot;CCL2&quot;, &quot;TLR2&quot;, &quot;SPP1&quot;, &quot;CYBB&quot;,  &quot;ITGAX&quot;,
          &quot;CLEC7A&quot;, &quot;CHI3L3&quot;, &quot;ARG1&quot;, &quot;SIGLEC1&quot;, &quot;CFP&quot;, &quot;CXCL10&quot;, &quot;ALCAM&quot;, 
          &quot;FER1L3&quot;, &quot;LILRB4&quot;, &quot;GPX3&quot;, &quot;GAS7&quot;, &quot;CCRL2&quot;,&quot;CXCL16&quot;, &quot;CXCR4&quot;,
          &quot;GPNMB&quot;, &quot;LGALS3&quot;, &quot;IFI202B&quot;, &quot;CSF1&quot;, &quot;LIRB4&quot;, &quot;LAG3&quot;)

#hiv4 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/non_integrated_HIV1_HIV2_4samples_seurat_object.rds&quot;)

#get genes also present in the HIV samples
mgnd &lt;- mgnd[mgnd %in% rownames(hiv4@assays$RNA@counts)]


hiv_disease &lt;- AddModuleScore(hiv4,
                           features = disease)

keren &lt;- FeaturePlot(hiv_disease, features = &quot;Cluster1&quot;, pt.size = .01) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Keren-Shaul et al., Genes upregulated in disease-associated microglia&quot;)



hiv_mgnd &lt;- AddModuleScore(hiv4,
                           features = mgnd)

kraseman &lt;- FeaturePlot(hiv_mgnd, features = &quot;Cluster1&quot;, pt.size = .01) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Kraseman et al., 28 inflammatory genes upregulated in 
       disease associated microglia&quot;)

ggarrange(keren, kraseman, ncol = 2, nrow = 1)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-60-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-60-1">
Past versions of unnamed-chunk-60-1.png
</button>
</p>
<div id="fig-unnamed-chunk-60-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-60-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-60-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Keren Shaul et al. signature
p1 &lt;- DotPlot(hiv4, features = disease, group.by = &quot;cell_type&quot;) + 
  coord_flip() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
    labs(title = &quot;Keren-Shaul et al., Genes upregulated in disease-associated microglia&quot;)

# Kraseman et al. signature
p2 &lt;- DotPlot(hiv4, features = mgnd, group.by = &quot;cell_type&quot;) + 
  coord_flip() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
    labs(title = &quot;Kraseman et al., 28 inflammatory genes upregulated in 
       disease associated microglia&quot;)


# some genes that might be markers for Myeloid2
p3 &lt;- DotPlot(hiv4, features = c(&quot;CD9&quot;, &quot;APOE&quot;, &quot;AXL&quot;, &quot;CTSL&quot;, &quot;SPP1&quot;), group.by = &quot;cell_type&quot;) + 
  coord_flip() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
  labs(title = &quot;Genes that might be used as markers for Myeloid2&quot;)

ggarrange(p1, p2, p3, ncol = 3, nrow = 1)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-61-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-61-1">
Past versions of unnamed-chunk-61-1.png
</button>
</p>
<div id="fig-unnamed-chunk-61-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-61-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-61-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="esaulova-et-al." class="section level3">
<h3>Esaulova et al.</h3>
<p>Gene signature defined by <span class="citation">(<a href="#ref-Esaulova" role="doc-biblioref">Ekaterina Esaulova 2020</a>)</span></p>
<div id="homeostatic-genes" class="section level5">
<h5>Homeostatic genes:</h5>
<ul>
<li>the homeostatic microglia gene signature is found in all myeloid clusters, not only in the Myeloid 2 (except for SLC2A5 which was also included in the Farhadian paper)</li>
<li>NK cell cluster also exhibits high scores for this gene signature</li>
</ul>
<pre class="r"><code>#homeostatic microglia gene signature
microglia &lt;- c(&quot;CX3CR1&quot;,&quot;CSF1R&quot;, &quot;SLC2A5&quot;, &quot;MARCKS&quot;, &quot;P2RY13&quot;)
FeaturePlot(hiv4, features = microglia, ncol = 3)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-62-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-62-1">
Past versions of unnamed-chunk-62-1.png
</button>
</p>
<div id="fig-unnamed-chunk-62-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-62-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-62-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dotplot &lt;- DotPlot(hiv4, features = microglia, group.by = &quot;cell_type&quot;) + 
  coord_flip() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1))

hivt &lt;- AddModuleScore(hiv4, features = microglia)
Idents(hivt) &lt;- &quot;cell_type&quot;
tsne &lt;- FeaturePlot(hivt, features = &quot;Cluster1&quot;, label =TRUE, repel = TRUE, pt.size = .7) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
    labs(title = &quot;Homeostatic microglia signature from Esaulova paper&quot;)

ggarrange(dotplot, tsne, ncol =2, nrow = 1)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-63-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-63-1">
Past versions of unnamed-chunk-63-1.png
</button>
</p>
<div id="fig-unnamed-chunk-63-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-63-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-63-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="microglia-specific-genes" class="section level5">
<h5>Microglia-specific genes:</h5>
<ul>
<li><strong>GPR34</strong>, <strong>TMEM119</strong> supports the hypothesis that myeloid2 are microglia-like, since they are expressed at higher levels in the Myeloid2 cluster</li>
<li>they use 10 genes to discriminate microglia, four of which have not been inlcuded by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span>: <strong>CLEC9A</strong>, <strong>TIMD4</strong>, <strong>TNFSF18</strong>, <strong>CXCL12</strong></li>
<li>the genes <strong>CLEC9A</strong>, <strong>TIMD4</strong>, <strong>TNFSF18</strong>, <strong>CXCL12</strong> are not included in the 60 genes used by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span>, but they do not show specific expression for Myeloid2 cells</li>
</ul>
<pre class="r"><code># 10 genes that discriminate microglia
microglia &lt;- c(&quot;C1QC&quot;, &quot;C1QB&quot;, &quot;C1QA&quot;, &quot;TREM2&quot;, &quot;TIMD4&quot;, &quot;APOC1&quot;, 
               &quot;APOE&quot;, &quot;TNFSF18&quot;, &quot;CLEC9A&quot;, &quot;CXCL12&quot;)
#FeaturePlot(hiv4, features = microglia, ncol = 3)

#microglia %in% paper

Idents(hiv4) &lt;- &quot;cell_type&quot;

#  two genes whcih might support that myeloid2 cluster contains microglia-like cells
#FeaturePlot(hiv4, features = c(&quot;GPR34&quot;, &quot;TMEM119&quot;), label = TRUE, pt.size = .7,
       #     repel = TRUE)
#DotPlot(hiv4, features = c(&quot;TMEM119&quot;, &quot;GPR34&quot;), group.by = &quot;cell_type&quot;) + 
  #   coord_flip() +
   #  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1))

# these are not inlcuded in the Farhadian gene signature
FeaturePlot(hiv4, features = c(&quot;CLEC9A&quot;, &quot;TIMD4&quot;, &quot;TNFSF18&quot;, &quot;CXCL12&quot;, 
                               &quot;GPR34&quot;, &quot;TMEM119&quot;),
            pt.size = .7, label = TRUE, repel = TRUE, ncol= 2)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-64-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-64-1">
Past versions of unnamed-chunk-64-1.png
</button>
</p>
<div id="fig-unnamed-chunk-64-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-64-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-64-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="sankowski-et-al." class="section level3">
<h3>Sankowski et al.</h3>
<p><span class="citation">(<a href="#ref-Sankowski" role="doc-biblioref">Roman Sankowski 2019</a>)</span> describe different states of microglia cells from the human brain.</p>
<p>In the paper microglia cells were obtained through brain surgery. Microglia were defined by a core signature called “microglia signature” below. <span class="citation">(<a href="#ref-Sankowski" role="doc-biblioref">Roman Sankowski 2019</a>)</span> performed unsupervised clustering using Race ID3 which resulted in 9 major subclusters of the microglia cells. The authors suggested that these subclusters represent the wide spectrum of transcriptional states in human microglia.</p>
<p><strong>RaceID:</strong> RaceID is a clustering algorithm for the identification of cell types from scRNA-seq data. It was designed to work particularly well for the detection of rare cells. See for more details: <a href="https://github.com/dgrun/RaceID3_StemID2_package" class="uri">https://github.com/dgrun/RaceID3_StemID2_package</a></p>
<ul>
<li>The microglia signature is most pronounced in Myeloid2 cluster which supports the hypothesis that Myeoid2 are microglia-like cells.</li>
<li>The monocyte signature is less pronounced in Myeloid2 cluster and more pronounced in the other Myeloid clusters which is in line with the assumption that the cells in blood and CSF are monocytes or monocyte-derived macrophages rather than microglia-like cells.</li>
<li>Homeostatic microglia gene signature (core signature genes expressed by all clusters in the paper) is found in all Myeloid cluster as well as NK cells, with high expression also in Myeloid2. It is not clear why NK cells would express a homeostatic microglia signature. The reason why all Myeloid cluster express this signature might be that homeostatic microlglia have some overlap in gene expression with peripheral macrophages.</li>
</ul>
<p>Two potential markers for Myeloid2 are <strong>CSF1R</strong> and <strong>MARCKS</strong> which are most highly expressed in Myeloid2 cells.</p>
<pre class="r"><code># microglia gene signature from supplementary table 3a
microglia &lt;- c(&quot;P2RY12&quot;, &quot;CX3CR1&quot;, &quot;CSF1R&quot;, &quot;TMEM119&quot;, &quot;SLC2A5&quot; )

# monocyte gene signature from supplementary table 3a 
monocyte &lt;- c(&quot;CCR2&quot;, &quot;CLEC12A&quot;, &quot;PLAC8&quot;, &quot;FCN1&quot;, &quot;S100A9&quot;)

# homeostatic microlgia gene siganture
homeostatic &lt;- c(&quot;CX3CR1&quot;, &quot;TMEM119&quot;, &quot;CSF1R&quot;, &quot;P2RY12&quot;,
                 &quot;P2RY13&quot;,&quot;SELPLG&quot;, &quot;MARCKS&quot;)


tsne_list &lt;- map(seq.int(1:3), function(n){
  names &lt;- c(&quot;microglia&quot;, &quot;monocyte&quot;, &quot;homeostatic&quot;)
  feature_list &lt;- list(microglia, monocyte, homeostatic)
  features = feature_list[[n]]
  hiv &lt;- AddModuleScore(hiv4, features = features)
  # feature plot
  Idents(hiv) &lt;- &quot;cell_type&quot;
  FeaturePlot(hiv, features = &quot;Cluster1&quot;, pt.size = .5, label = TRUE, repel=TRUE)+
    scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
    labs(title = paste0(names[n],&quot; microglia gene signature&quot;))
})
  
dotplot_list &lt;- map(seq.int(1:3), function(n){
  names &lt;- c(&quot;microglia&quot;, &quot;monocyte&quot;, &quot;homeostatic&quot;)
  feature_list &lt;- list(microglia, monocyte, homeostatic)
  features = feature_list[[n]] 
  # dot plot
  DotPlot(hiv4, features = features, group.by = &quot;cell_type&quot;) +
    coord_flip() +
        theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
  labs(title = paste0(names[n],&quot; gene signature&quot;))
})
  
do.call(grid.arrange, c(append(tsne_list, dotplot_list), ncol = 3, nrow = 2))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-65-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-65-1">
Past versions of unnamed-chunk-65-1.png
</button>
</p>
<div id="fig-unnamed-chunk-65-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-65-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-65-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="clusters-across-the-transcriptional-spectrum-of-microglia" class="section level4">
<h4>Clusters across the transcriptional spectrum of microglia</h4>
<p><span class="citation">(<a href="#ref-Sankowski" role="doc-biblioref">Roman Sankowski 2019</a>)</span> defined 9 subclusters within the microglia cell population. Can we find these gene signatures in the CSF-derived Myeloid cells? This could further confirm that Myeloid2 and Myeloid5 are microglia-like cells.</p>
<ul>
<li><strong>homeostatic clusters</strong>
<ul>
<li>C2: strong expresssion of MCH-II and antiviral immunity genes (HLA-DRA, CD74, IFI44L)</li>
<li>C3: high expression of microglial core genes CX3CR1, TMEM119<br />
</li>
</ul></li>
<li><strong>antigen processing and peptide antigen presentation</strong> C6 and C7 were characterized by low expression of CX3CR1 and high expression of integrin-receptor binding protein and metabolism genes (SPP1, APOE, LPL).</li>
<li>** proinflammatory clusters:** C1, C5, C8, C9 were characterized by expression of chemokine and cytokine genes (CCL2, IL1B).</li>
</ul>
<p>Since Cluster C2 was characterized as a homeostati microglia cluster and showed expression of MHC-II and antivrial immunity genes like IFI44L, HLA-DRA, CD74, this supports the microglia-like identity of Myeloid2 and Myeloid5. However, there is also some expression of this signature in Meloid1 and Myeloid3 which mainly originate from Blood samples and also in B cells.</p>
<p>There is also a high score for C5 and C8 signature genes, which are both proinflammatory gene signatures.</p>
<pre class="r"><code># get differentially expressed genes per cluster across the transcriptional 
# spectrum of microglia 
excel_sheets(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/2019_Sankowski_Mapping_microgliastates_in_the_human_brain_Supplementary_tables.xlsx&quot;)
 [1] &quot;Suppl. table 1&quot;  &quot;Suppl. table 2&quot;  &quot;Suppl. table 3&quot;  &quot;Suppl. table 4&quot; 
 [5] &quot;Suppl. table 5&quot;  &quot;Suppl. table 6&quot;  &quot;Suppl. table 7&quot;  &quot;Suppl. table 8&quot; 
 [9] &quot;Suppl. table 9&quot;  &quot;Suppl. table 10&quot; &quot;Suppl. table 11&quot; &quot;Suppl. table 12&quot;
cluster_genes &lt;- read_excel(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/2019_Sankowski_Mapping_microgliastates_in_the_human_brain_Supplementary_tables.xlsx&quot;, sheet = &quot;Suppl. table 3&quot;)
#names(microglia_sign_df) [1] &lt;- &quot;gene_name&quot;
#microglia_sign &lt;- microglia_sign_df[,&quot;gene_name&quot;]

# C3

cluster_list &lt;- map(seq.int(1:8), function(n) {
  names &lt;- c(&quot;c1&quot;, &quot;c2&quot;, &quot;c3&quot;, &quot;c5&quot;, &quot;c6&quot;, &quot;c7&quot;, &quot;c8&quot;, &quot;c9&quot;)
  name &lt;- names[n]
  cluster &lt;- c(1,2,3,5,6,7,8,9)
  cluster_number &lt;- cluster[n]
  c &lt;- pull(cluster_genes %&gt;% filter(Cluster == cluster_number), GENEID)
  c &lt;- c[c %in% rownames(hiv4@assays$RNA@counts)]
  cluster_list &lt;- list(name = name, cluster = c)
})


cluster_plots &lt;- map(seq.int(1:8), function(n){
  c &lt;- cluster_list[[n]]$cluster
  name &lt;- cluster_list[[n]]$name
  hivc &lt;- AddModuleScore(hiv4, features = c)
  Idents(hivc) &lt;- &quot;cell_type&quot;
  FeaturePlot(hivc, features = &quot;Cluster1&quot;, pt.size = .5, label = TRUE,
              repel = TRUE) +
      scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
    labs(title = paste0(&quot;Differentially expressed genes of cluster &quot;, name))
})

do.call(grid.arrange, c(cluster_plots, ncol = 4, nrow = 2))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-66-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-66-1">
Past versions of unnamed-chunk-66-1.png
</button>
</p>
<div id="fig-unnamed-chunk-66-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-66-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-66-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
#map(seq.int(1:8), function(n){
 # c &lt;- cluster_list[[n]]$cluster
  #name &lt;- cluster_list[[n]]$name
  #DotPlot(hiv4, features = c, group.by = &quot;cell_type&quot;) +
  #coord_flip() +
  #      theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
  #labs(title = paste0(&quot;Differentially expressed genes of cluster &quot;, name))
#})</code></pre>
</div>
</div>
</div>
<div id="section-4" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<hr />
</div>
<div id="genes-which-are-upregulated-in-myeloid2-compared-to-all-other-myeloid-clusters" class="section level2">
<h2>Genes which are upregulated in Myeloid2 compared to all other Myeloid clusters</h2>
<ul>
<li><p>overlap of 5 genes with Kraseman microglia -&gt; inflammatory molecules upregulated in disease associated microglia (Krasemann)</p></li>
<li><p>overlap of 3 genes with disease-associated microglia (Keren-Shaul)</p></li>
<li><p>Sankowski signature</p>
<ul>
<li>4 genes overlap with microglia signature</li>
<li><strong>zero overlap with monocyte gene signature</strong></li>
<li>4 genes overlap with homeostatic microglia signature</li>
</ul></li>
<li><p>Esaulova paper -&gt; 8 of 10 genes that discriminate microglia are found upregulated in Myeloid2 cluster</p></li>
<li><p>Glass paper -&gt; 132 of the microglia signature are found in the genes upregulated in Myeloid2, 28 of these are also included in the Farhadian Paper signature, 104 are new marekr genes</p></li>
</ul>
<pre class="r"><code>Idents(hiv4) &lt;- &quot;cell_type&quot;
# Find Markers that distinguish My2 from the other Myeloid clusters
my2_markers &lt;- FindMarkers(hiv4, ident.1  = &quot;Myeloid2&quot;,
            ident.2  = c(&quot;Myeloid1&quot;, &quot;Myeloid3&quot;, &quot;Myeloid4&quot;, &quot;Myeloid5&quot;))
my2_markers &lt;- my2_markers %&gt;% rownames_to_column(var = &quot;gene&quot;)
# Can we find mgnd genes? (Krasemann)
my2_markers %&gt;% filter(gene %in% mgnd &amp; avg_log2FC &gt; 0)
    gene        p_val avg_log2FC pct.1 pct.2    p_val_adj
1   SPP1 3.626442e-14  2.2989596 0.103 0.002 9.018961e-10
2 CXCL16 8.438072e-11  1.0379235 0.439 0.172 2.098549e-06
3   GPX3 3.358479e-10  0.4856146 0.112 0.010 8.352537e-06
4   CYBB 1.267019e-09  0.9216583 0.860 0.728 3.151077e-05
5 LILRB4 4.108097e-08  0.6611217 0.271 0.089 1.021684e-03
# genes in disease-assocaited microglia (Keren-Shaul)
my2_markers %&gt;% filter(gene %in% disease &amp; avg_log2FC &gt; 0)
    gene        p_val avg_log2FC pct.1 pct.2    p_val_adj
1    AXL 7.682308e-19  1.0183192 0.252 0.029 1.910590e-14
2   SPP1 3.626442e-14  2.2989596 0.103 0.002 9.018961e-10
3 LILRB4 4.108097e-08  0.6611217 0.271 0.089 1.021684e-03

# Sankowski
# microglia gene signature
microglia &lt;- c(&quot;P2RY12&quot;, &quot;CX3CR1&quot;, &quot;CSF1R&quot;, &quot;TMEM119&quot;, &quot;SLC2A5&quot; )

# monocyte gene signature
monocyte &lt;- c(&quot;CCR2&quot;, &quot;CLEC12A&quot;, &quot;PLAC8&quot;, &quot;FCN1&quot;, &quot;S100A9&quot;)

# homeostatic microlgia gene siganture
homeostatic &lt;- c(&quot;CX3CR1&quot;, &quot;TMEM119&quot;, &quot;CSF1R&quot;, &quot;P2RY12&quot;,
                 &quot;P2RY13&quot;,&quot;SELPLG&quot;, &quot;MARCKS&quot;)


my2_markers %&gt;% filter(gene %in% microglia &amp; avg_log2FC &gt; 0)
    gene        p_val avg_log2FC pct.1 pct.2    p_val_adj
1 SLC2A5 1.038249e-50  1.3177001 0.346 0.000 2.582126e-46
2 P2RY12 1.971868e-11  0.5465335 0.140 0.015 4.904036e-07
3  CSF1R 2.060436e-11  1.2036520 0.664 0.407 5.124303e-07
4 CX3CR1 1.359952e-04  0.5885714 0.439 0.271 1.000000e+00
my2_markers %&gt;% filter(gene %in% monocyte &amp; avg_log2FC &gt; 0)
[1] gene       p_val      avg_log2FC pct.1      pct.2      p_val_adj 
&lt;0 rows&gt; (or 0-length row.names)
my2_markers %&gt;% filter(gene %in% homeostatic &amp; avg_log2FC &gt; 0)
    gene        p_val avg_log2FC pct.1 pct.2    p_val_adj
1 MARCKS 3.672024e-12  1.1346931 0.664 0.363 9.132325e-08
2 P2RY12 1.971868e-11  0.5465335 0.140 0.015 4.904036e-07
3  CSF1R 2.060436e-11  1.2036520 0.664 0.407 5.124303e-07
4 CX3CR1 1.359952e-04  0.5885714 0.439 0.271 1.000000e+00


# Esaulova
# 10 genes that discriminate microglia
microglia &lt;- c(&quot;C1QC&quot;, &quot;C1QB&quot;, &quot;C1QA&quot;, &quot;TREM2&quot;, &quot;TIMD4&quot;, &quot;APOC1&quot;, 
               &quot;APOE&quot;, &quot;TNFSF18&quot;, &quot;CLEC9A&quot;, &quot;CXCL12&quot;)
# Can we find these 10 genes in the Myeloid2 upregulated genes?
my2_markers %&gt;% filter(gene %in% microglia &amp; avg_log2FC &gt; 0)
    gene         p_val avg_log2FC pct.1 pct.2     p_val_adj
1   C1QC 1.891156e-120  4.1160497 0.794 0.002 4.703305e-116
2   C1QB 8.023728e-114  4.1322588 0.860 0.026 1.995501e-109
3   APOE 4.641643e-108  3.5767082 0.720 0.002 1.154377e-103
4   C1QA  3.455807e-92  3.1390482 0.785 0.039  8.594593e-88
5  APOC1  2.142095e-53  2.0205009 0.402 0.006  5.327391e-49
6  TREM2  1.158412e-51  1.5152244 0.374 0.003  2.880971e-47
7  TIMD4  5.003322e-18  0.5471981 0.131 0.002  1.244326e-13
8 CLEC9A  1.921500e-06  0.4152332 0.112 0.021  4.778770e-02


# Glass
# 84 of the microglia signature genes are more highly expressed in the 
microglia_genes &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/Microglia_gene_signatur_found_in_HIV1_HIV2_samples&quot;)
m1 &lt;- my2_markers %&gt;% filter(gene %in% microglia_genes [[&quot;gene_name&quot;]] &amp; avg_log2FC &gt;0)
# remove the genes that were already used by the Farhadian paper
m1 &lt;- m1[[&quot;gene&quot;]][m1[[&quot;gene&quot;]] %in% paper == FALSE]

hiv_new &lt;- AddModuleScore(hiv4, features = m1)
FeaturePlot(hiv_new, features = &quot;Cluster1&quot;) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
    labs(title = &quot;Selected signature genes from the Glass paper&quot;)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-67-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-67-1">
Past versions of unnamed-chunk-67-1.png
</button>
</p>
<div id="fig-unnamed-chunk-67-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2be0932ddfed52b8d9f4a4086763fe9eeb4602b0/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-67-1.png" target="_blank">2be0932</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/b315085a0df3faa48f8d2e2a07393671e6e80b97/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-67-1.png" target="_blank">b315085</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="conclusions-on-myloid2-cell-type" class="section level2">
<h2>Conclusions on Myloid2 cell type</h2>
<p>Taken together the results above do not provide a clear picture on whether the Myeloid2 cluster consists of microglia-like cells. There are several genes which are associated with microglia and disease-associated microglia, several of which referred to by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span>. Nevertheless, there are also many genes which are described to be microglia markers in the literature, but do not show a significantly higher expression in Myeloid2 than in the other Myeloid clusters. Therefore, the cluster will be considered microglia-like for now and the following analysis will provide additional insights on the identity and characteristics of this cluster.</p>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Gosselin" class="csl-entry">
David Gosselin, Nicole G. Coufal, Dylan Skola. 2020. <em>Single-Cell RNA Sequencing Reveals Microglia-Like Cells in Cerebrospinal Fluid During Virologically Suppressed HIV</em>. <em>JNEUROSCI</em>. Vol. 356, 1248. <a href="https://DOI: 10.1126/science.aal3222">DOI: 10.1126/science.aal3222</a>.
</div>
<div id="ref-Esaulova" class="csl-entry">
Ekaterina Esaulova, Irina Shchukina, Claudia Cantoni. 2020. <em>Single-Cell RNA-Seq Analysis of Human CSF Microglia and Myeloid Cells in Neuroinflammation</em>. <em>Nat Neurosci</em>. Vol. 7:e732. <a href="https://doi:10.1212/NXI.0000000000000732">doi:10.1212/NXI.0000000000000732</a>.
</div>
<div id="ref-Grassivaro" class="csl-entry">
Francesca Grassivaro, Massimo Acquaviva, Ramesh Menon, and Gianvito Martino. 2018. <em>Convergence Between Microglia and Peripheral Macrophages Phenotype During Development and Neuroinflammation</em>. <em>JCI Insight</em>. Vol. 40 (4). <a href="https://doi.org/10.1523/JNEUROSCI.1523-19.2019">https://doi.org/10.1523/JNEUROSCI.1523-19.2019</a>.
</div>
<div id="ref-Keren-Shaul" class="csl-entry">
Hadas Keren-Shaul, Assaf Weiner, Amit Spinrad. 2017. <em>A Unique Microglia Type Associated with Restricting Development of Alzheimer’s Disease</em>. <em>Cell</em>. Vol. 169. <a href="http://dx.doi.org/10.1016/j.cell.2017.05.018">http://dx.doi.org/10.1016/j.cell.2017.05.018</a>.
</div>
<div id="ref-Sankowski" class="csl-entry">
Roman Sankowski, Takahiro Masuda, Chotima Böttcher. 2019. <em>Mapping Microglia States in the Human Brain Through the Integration of High-Dimensional Techniques.</em> <em>Nat. Neurosci.</em> Vol. 22. <a href="https://doi.org/10.1038/s41593-019-0532-y">https://doi.org/10.1038/s41593-019-0532-y</a>.
</div>
<div id="ref-Farhadian" class="csl-entry">
Shelli F. Farhadian, Chrysoula Zografou, Sameet S. Mehta, and Serena S. Spudich. 2018. <em>Single-Cell RNA Sequencing Reveals Microglia-Like Cells in Cerebrospinal Fluid During Virologically Suppressed HIV</em>. <em>JCI Insight</em>. Vol. 3(18):e121718. <a href="https://doi.org/10.1172/jci.insight.121718">https://doi.org/10.1172/jci.insight.121718</a>.
</div>
<div id="ref-Kraseman" class="csl-entry">
Susanne Krasemann, Ron Cialic, Charlotte Madore. 2017. <em>The Trem2-APOE Pathway Drives the Transcriptional Phenotype of Dysfunctional Microglia in Neurodegenerative Diseases</em>. <em>Immunity</em>. Vol. 47. <a href="http://dx.doi.org/10.1016/j.immuni.2017.08.008">http://dx.doi.org/10.1016/j.immuni.2017.08.008</a>.
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] patchwork_1.1.1       SeuratDisk_0.0.0.9019 harmony_0.1.0        
 [4] Rcpp_1.0.7            readxl_1.3.1          multtest_2.46.0      
 [7] Biobase_2.50.0        BiocGenerics_0.36.1   metap_1.6            
[10] knitr_1.30            rstatix_0.6.0         corrplot_0.92        
[13] gridExtra_2.3         ggpubr_0.4.0          circlize_0.4.13      
[16] ComplexHeatmap_2.9.4  ggcorrplot_0.1.3      pheatmap_1.0.12      
[19] RColorBrewer_1.1-2    ggrepel_0.9.1         data.table_1.14.2    
[22] Matrix_1.3-4          edgeR_3.32.0          limma_3.46.0         
[25] SeuratObject_4.0.3    Seurat_4.0.5          forcats_0.5.0        
[28] stringr_1.4.0         dplyr_1.0.7           purrr_0.3.4          
[31] readr_1.4.0           tidyr_1.1.2           tibble_3.1.6         
[34] ggplot2_3.3.5         tidyverse_1.3.0      

loaded via a namespace (and not attached):
  [1] utf8_1.2.2            reticulate_1.22       tidyselect_1.1.1     
  [4] htmlwidgets_1.5.4     Rtsne_0.15            munsell_0.5.0        
  [7] mutoss_0.1-12         codetools_0.2-18      ica_1.0-2            
 [10] future_1.23.0         miniUI_0.1.1.1        withr_2.4.3          
 [13] colorspace_2.0-2      highr_0.8             rstudioapi_0.13      
 [16] stats4_4.0.2          ROCR_1.0-11           ggsignif_0.6.0       
 [19] tensor_1.5            listenv_0.8.0         labeling_0.4.2       
 [22] Rdpack_2.1.2          git2r_0.27.1          mnormt_2.0.2         
 [25] polyclip_1.10-0       farver_2.1.0          bit64_4.0.5          
 [28] rprojroot_2.0.2       TH.data_1.0-10        parallelly_1.29.0    
 [31] vctrs_0.3.8           generics_0.1.1        xfun_0.29            
 [34] R6_2.5.1              doParallel_1.0.16     clue_0.3-60          
 [37] locfit_1.5-9.4        hdf5r_1.3.3           spatstat.utils_2.2-0 
 [40] assertthat_0.2.1      promises_1.2.0.1      scales_1.1.1         
 [43] multcomp_1.4-15       gtable_0.3.0          globals_0.14.0       
 [46] goftest_1.2-3         sandwich_3.0-0        workflowr_1.6.2      
 [49] rlang_0.4.12          GlobalOptions_0.1.2   splines_4.0.2        
 [52] lazyeval_0.2.2        spatstat.geom_2.3-0   broom_0.7.2          
 [55] yaml_2.2.1            reshape2_1.4.4        abind_1.4-5          
 [58] modelr_0.1.8          backports_1.2.0       httpuv_1.6.3         
 [61] tools_4.0.2           ellipsis_0.3.2        spatstat.core_2.3-1  
 [64] ggridges_0.5.2        TFisher_0.2.0         plyr_1.8.6           
 [67] comprehenr_0.6.10     rpart_4.1-15          deldir_1.0-6         
 [70] pbapply_1.5-0         GetoptLong_1.0.5      cowplot_1.1.1        
 [73] S4Vectors_0.28.1      zoo_1.8-9             haven_2.3.1          
 [76] cluster_2.1.0         fs_1.5.1              magrittr_2.0.1       
 [79] RSpectra_0.16-0       scattermore_0.7       openxlsx_4.2.3       
 [82] lmtest_0.9-39         reprex_0.3.0          RANN_2.6.1           
 [85] tmvnsim_1.0-2         mvtnorm_1.1-1         whisker_0.4          
 [88] fitdistrplus_1.1-6    matrixStats_0.61.0    hms_1.1.1            
 [91] mime_0.12             evaluate_0.14         xtable_1.8-4         
 [94] rio_0.5.16            IRanges_2.24.1        shape_1.4.6          
 [97] compiler_4.0.2        KernSmooth_2.23-18    crayon_1.4.2         
[100] htmltools_0.5.2       mgcv_1.8-33           later_1.3.0          
[103] lubridate_1.7.9.2     DBI_1.1.1             dbplyr_2.0.0         
[106] MASS_7.3-53           car_3.0-10            cli_3.1.0            
[109] rbibutils_2.2.4       igraph_1.2.9          pkgconfig_2.0.3      
[112] sn_2.0.1              numDeriv_2016.8-1.1   foreign_0.8-80       
[115] plotly_4.9.2.1        spatstat.sparse_2.0-0 xml2_1.3.2           
[118] foreach_1.5.1         rvest_0.3.6           digest_0.6.29        
[121] sctransform_0.3.2     RcppAnnoy_0.0.19      spatstat.data_2.1-0  
[124] rmarkdown_2.6         cellranger_1.1.0      leiden_0.3.5         
[127] uwot_0.1.10           curl_4.3.2            shiny_1.7.1          
[130] rjson_0.2.20          lifecycle_1.0.1       nlme_3.1-150         
[133] jsonlite_1.7.2        carData_3.0-4         viridisLite_0.4.0    
[136] fansi_0.5.0           pillar_1.6.4          lattice_0.20-41      
[139] plotrix_3.8-2         fastmap_1.1.0         httr_1.4.2           
[142] survival_3.2-7        glue_1.5.1            zip_2.1.1            
[145] png_0.1-7             iterators_1.0.13      bit_4.0.4            
[148] stringi_1.7.6         mathjaxr_1.4-0        irlba_2.3.3          
[151] future.apply_1.8.1   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
