<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="kmikulik" />


<title>Analysis of HIV-1 CSF dataset</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Introduction.html">Introduction</a>
</li>
<li>
  <a href="HIV_CSF_dataset_analysis.html">CSF</a>
</li>
<li>
  <a href="TF_activity_analysis_of_myeloid2_cluster.html">TF activity Myeloid2</a>
</li>
<li>
  <a href="Independent_cell_type_identification_Farhadian_dataset.html">Myeloid2 are a microglia-like</a>
</li>
<li>
  <a href="Triculture_datset_analysis_and_integration_Farhadian_dataset.html">Integration CSF &amp; triculture</a>
</li>
<li>
  <a href="Triculture_dataset_GRN_with_Scenic.html">TF activity iMg</a>
</li>
<li>
  <a href="Discussion.html">Discussion</a>
</li>
<li>
  <a href="Methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Katharina782/report">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis of HIV-1 CSF dataset</h1>
<h4 class="author">kmikulik</h4>
<h4 class="date">3 12 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
unnamed-chunk-1
</li>
<li>
unnamed-chunk-10
</li>
<li>
unnamed-chunk-11
</li>
<li>
unnamed-chunk-12
</li>
<li>
unnamed-chunk-13
</li>
<li>
unnamed-chunk-14
</li>
<li>
unnamed-chunk-15
</li>
<li>
unnamed-chunk-16
</li>
<li>
unnamed-chunk-17
</li>
<li>
unnamed-chunk-18
</li>
<li>
unnamed-chunk-19
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-20
</li>
<li>
unnamed-chunk-21
</li>
<li>
unnamed-chunk-22
</li>
<li>
unnamed-chunk-23
</li>
<li>
unnamed-chunk-24
</li>
<li>
unnamed-chunk-25
</li>
<li>
unnamed-chunk-26
</li>
<li>
unnamed-chunk-27
</li>
<li>
unnamed-chunk-28
</li>
<li>
unnamed-chunk-29
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-30
</li>
<li>
unnamed-chunk-31
</li>
<li>
unnamed-chunk-32
</li>
<li>
unnamed-chunk-33
</li>
<li>
unnamed-chunk-34
</li>
<li>
unnamed-chunk-35
</li>
<li>
unnamed-chunk-36
</li>
<li>
unnamed-chunk-37
</li>
<li>
unnamed-chunk-38
</li>
<li>
unnamed-chunk-39
</li>
<li>
unnamed-chunk-4
</li>
<li>
unnamed-chunk-40
</li>
<li>
unnamed-chunk-41
</li>
<li>
unnamed-chunk-42
</li>
<li>
unnamed-chunk-43
</li>
<li>
unnamed-chunk-44
</li>
<li>
unnamed-chunk-45
</li>
<li>
unnamed-chunk-46
</li>
<li>
unnamed-chunk-47
</li>
<li>
unnamed-chunk-48
</li>
<li>
unnamed-chunk-49
</li>
<li>
unnamed-chunk-5
</li>
<li>
unnamed-chunk-50
</li>
<li>
unnamed-chunk-51
</li>
<li>
unnamed-chunk-52
</li>
<li>
unnamed-chunk-53
</li>
<li>
unnamed-chunk-54
</li>
<li>
unnamed-chunk-55
</li>
<li>
unnamed-chunk-56
</li>
<li>
unnamed-chunk-57
</li>
<li>
unnamed-chunk-58
</li>
<li>
unnamed-chunk-59
</li>
<li>
unnamed-chunk-6
</li>
<li>
unnamed-chunk-60
</li>
<li>
unnamed-chunk-61
</li>
<li>
unnamed-chunk-62
</li>
<li>
unnamed-chunk-63
</li>
<li>
unnamed-chunk-64
</li>
<li>
unnamed-chunk-65
</li>
<li>
unnamed-chunk-66
</li>
<li>
unnamed-chunk-67
</li>
<li>
unnamed-chunk-68
</li>
<li>
unnamed-chunk-69
</li>
<li>
unnamed-chunk-7
</li>
<li>
unnamed-chunk-8
</li>
<li>
unnamed-chunk-9
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>HIV_CSF_dataset_analysis_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttree04f675329bc4ccf704fe3326b6e562264c9dbbd3targetblank04f6753a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Katharina782/report/tree/04f675329bc4ccf704fe3326b6e562264c9dbbd3" target="_blank">04f6753</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttree04f675329bc4ccf704fe3326b6e562264c9dbbd3targetblank04f6753a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Katharina782/report/tree/04f675329bc4ccf704fe3326b6e562264c9dbbd3" target="_blank">04f6753</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/HIV_CSF_dataset_analysis_cache/
    Ignored:    analysis/Independent_cell_type_identification_Farhadian_dataset_cache/
    Ignored:    analysis/TF_activity_analysis_of_myeloid2_cluster_cache/
    Ignored:    analysis/Triculture_dataset_analysis_and_integration_Farhadian_dataset_cache/

Untracked files:
    Untracked:  analysis/html_test.Rmd

Unstaged changes:
    Modified:   analysis/Independent_cell_type_identification_Farhadian_dataset.Rmd
    Modified:   analysis/Introduction.Rmd
    Modified:   analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd
    Modified:   analysis/Triculture_dataset_analysis_and_integration_Farhadian_dataset.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/references.bib

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/HIV_CSF_dataset_analysis.Rmd</code>) and HTML (<code>docs/HIV_CSF_dataset_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/04f675329bc4ccf704fe3326b6e562264c9dbbd3/analysis/HIV_CSF_dataset_analysis.Rmd" target="_blank">04f6753</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-17
</td>
<td>
wflow_publish(files = rmd)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/5830d4f67d499d645db8566d2cbfe280298bfdae/docs/HIV_CSF_dataset_analysis.html" target="_blank">5830d4f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/3ba29868e10097faef3f8ff84c501c88bf93da86/analysis/HIV_CSF_dataset_analysis.Rmd" target="_blank">3ba2986</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
wflow_publish(“analysis/HIV_CSF_dataset_analysis.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/analysis/HIV_CSF_dataset_analysis.Rmd" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
<td>
recent updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/dde10972b37ffd997474921968223514c8cb1b1f/docs/HIV_CSF_dataset_analysis.html" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
<td>
recent updates
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div style="text-align: justify">
<pre><code>knitr::opts_chunk$set(echo = TRUE, collapse = TRUE, message = FALSE,
                      cache = TRUE, cache.lazy = FALSE, autodep = TRUE)
set.seed(42)</code></pre>
<p>The aim was to reproduce the analysis done by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span>. Some additional analysis was performed on the Microlglia-like clusters identified in this dataset.</p>
<p>The dataset contains:</p>
<ul>
<li>3 CSF samples from HIV1-infected patients</li>
<li>2 Blood samples from HIV1-infected patients</li>
<li>2 CSF samples from uninfected patients</li>
</ul>
<p>For more details see <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span>.</p>
<ul>
<li>Import libraries</li>
</ul>
<pre class="r"><code>library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggplot2)
library(dplyr)
library(ggrepel)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(corrplot)
library(rstatix)
library(knitr)
library(metap)
library(multtest)
library(readxl)
library(harmony)
library(SeuratDisk)
library(patchwork)</code></pre>
<ul>
<li>Read in count matrices</li>
</ul>
<p>The data is available under <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE117397" class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE117397</a> There is one count matrix for each of the 7 samples.</p>
<pre class="r"><code>path &lt;- &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/GSE117397_RAW (1)/&quot;
all_files &lt;- list.files(path)

#Function to create a sparse matrix from each of the dataframes
sparse_matrix_from_df &lt;- function(df) {
  #print(df)
  #class(df)
  genes &lt;- df$GENE
  df_minus &lt;- df[,-1]
  #matrix &lt;- as.matrix(df_minus)
  sparse_matrix &lt;- as.sparse(df_minus)
  rownames(sparse_matrix) &lt;- genes
  return(sparse_matrix)
}

hiv_object_list &lt;- map(seq.int(2,6), function(n) {
  file &lt;- all_files[grepl(paste0(&quot;GSM329382&quot;, n), all_files)]
  name &lt;- str_extract(file, &quot;HIV[1-3]_\\w{3}&quot;)
  df &lt;- read_delim(paste0(path, file), delim = &quot;\t&quot;)
  sparse_matrix &lt;- sparse_matrix_from_df(df)
  seurat_ob &lt;- CreateSeuratObject(sparse_matrix, project = name, 
                     assay = &quot;RNA&quot;, min.cells = 1,
                     min.features = 1)
  list(df=df, sparse_matrix = sparse_matrix, seurat_ob = seurat_ob, name=name)
})

── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  GENE = col_character()
)
ℹ Use `spec()` for the full column specifications.
Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)

── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  GENE = col_character()
)
ℹ Use `spec()` for the full column specifications.
Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)

── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  GENE = col_character()
)
ℹ Use `spec()` for the full column specifications.
Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)

── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  GENE = col_character()
)
ℹ Use `spec()` for the full column specifications.
Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)

── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  GENE = col_character()
)
ℹ Use `spec()` for the full column specifications.
Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)

uninfected_list &lt;- map(seq.int(7,8), function(n) {
  file &lt;- all_files[grepl(paste0(&quot;GSM329382&quot;, n), all_files)]
  name &lt;- str_extract(file, &quot;Uninfected[12]_CSF&quot;)
  df &lt;- read_delim(paste0(path, file), delim = &quot;\t&quot;)
  sparse_matrix &lt;- sparse_matrix_from_df(df)
  seurat_ob &lt;- CreateSeuratObject(sparse_matrix, project = name, 
                     assay = &quot;RNA&quot;, min.cells = 1,
                     min.features = 1, verbose = FALSE)
  list(df=df, sparse_matrix = sparse_matrix, seurat_ob = seurat_ob, name=name)
})

── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  GENE = col_character()
)
ℹ Use `spec()` for the full column specifications.
Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)

── Column specification ────────────────────────────────────────────────────────
cols(
  .default = col_double(),
  GENE = col_character()
)
ℹ Use `spec()` for the full column specifications.
Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)

#write.table(uninfected_list[[1]]$sparse_matrix,
 #           &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/job_test.tsv&quot;, 
  #          sep = &quot;\t&quot;)</code></pre>
<ul>
<li>Create a Seurat object with all samples</li>
</ul>
<pre class="r"><code>hiv &lt;- merge(hiv_object_list[[1]]$seurat_ob, 
                   c(hiv_object_list[[2]]$seurat_ob,
                     hiv_object_list[[3]]$seurat_ob,
                     hiv_object_list[[4]]$seurat_ob,
                     hiv_object_list[[5]]$seurat_ob, 
                     uninfected_list[[1]]$seurat_ob,
                     uninfected_list[[2]]$seurat_ob), 
                   project = &quot;hiv_total&quot;,
                   add.cell.ids = c(&quot;1B&quot;, &quot;1C&quot;, &quot;2B&quot;, &quot;2C&quot;, &quot;3C&quot;, &quot;U1&quot;, &quot;U2&quot;))

# add percentage of mitochondrial genes to the metadata
hiv[[&quot;percent_mt&quot;]] &lt;- PercentageFeatureSet(hiv, pattern = &quot;^MT-&quot;)</code></pre>
<div id="quality-control" class="section level1">
<h1>Quality Control</h1>
<p>(on every individual object)</p>
<ul>
<li>Function to remove features that are expressed in less than three cells:</li>
</ul>
<pre class="r"><code>#returns a list of feature names to keep
remove_features &lt;- function(seurat_ob) {
  count_matrix &lt;- seurat_ob@assays$RNA@counts 
  ncells &lt;- rowSums(count_matrix != 0) # how many cells are expressing a 
  #certain gene? 
  names(ncells[ncells &gt; 3]) #extract gene names of genes expressed in more 
  #than 3 cells
}</code></pre>
<ul>
<li>Function to remove 10% of cell with highest mitochondrial gene expression</li>
</ul>
<pre class="r"><code># remove 10% of cells with the highest mitochondrial percentage
#returns a list of cell names to keep
remove_mt_genes &lt;- function(seurat_ob){
  thr &lt;- quantile(seurat_ob@meta.data$percent_mt, .9)
  seurat_ob@meta.data %&gt;% 
    filter(percent_mt &lt; thr) %&gt;%
    rownames()
}</code></pre>
<ul>
<li><p>Perform Quality Control</p></li>
<li><p>I do the quality control for each sample separately and then combine them again.</p></li>
<li><p>Question: Does it make sense to scale the data separately? Not really, because when you merge them you have to scale them anyways.</p></li>
<li><p>When doing the quality control how important is the order of the operations? If you first remove nFeatures &gt; 500 and then features exppressed in more than 3 cells -&gt; then the nFeatures might be below 500.</p></li>
</ul>
<pre class="r"><code>separate_objects &lt;- SplitObject(hiv, split.by = &quot;orig.ident&quot;)#[-c(5,6,7)] 
# only keep two blood samples and two csf samples

qc &lt;- lapply(X = separate_objects, FUN = function(x) {
    #x[[&quot;percent_mt&quot;]] &lt;- PercentageFeatureSet(x, pattern = &quot;^MT-&quot;) 
    # add percentage of mitochondrial genes to the metadata
    x &lt;- subset(x, features = remove_features(x)) 
    #create a list of cells excluding the 10% of cells with the highest mt gene expression
    x &lt;- subset(x, nFeature_RNA &lt; 2500 &amp; nFeature_RNA &gt; 500)
    x &lt;- subset(x, cells = remove_mt_genes(x)) 
    # remove 10% of cells with highest mitochondrial gene expression
    #x &lt;- NormalizeData(x)
    #x &lt;- ScaleData(x)
})

hiv_filt &lt;- merge(qc[[1]],
                        c(qc[[2]],
                        qc[[3]],
                        qc[[4]],
                        qc[[5]],
                        qc[[6]],
                        qc[[7]]),
                        project = &quot;hiv_filt&quot;)#,
                        #add.cell.ids = c(&quot;1B&quot;, &quot;1C&quot;, &quot;2B&quot;, &quot;2C&quot;, &quot;3C&quot;, &quot;U1&quot;, &quot;U2&quot;))</code></pre>
</div>
<div id="have-a-look-at-the-quality-of-the-different-samples." class="section level1">
<h1>Have a look at the quality of the different samples.</h1>
<div id="number-of-features" class="section level4">
<h4>Number of Features</h4>
<pre class="r"><code>
#have a look at the number of detected features
p1 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_boxplot(aes(x = orig.ident, y = nFeature_RNA, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p2 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_histogram(aes(x = nFeature_RNA, fill = orig.ident), bins = 100) +
  facet_wrap(~ orig.ident, ncol = 4) +
  NoLegend()

p3 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = orig.ident, y = nFeature_RNA, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p4 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = orig.ident, y = nFeature_RNA, fill = orig.ident)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

figure &lt;- ggarrange(p1, p3, p4, p2, ncol = 2, nrow = 2)
annotate_figure(figure, top = text_grob(&quot;Number of Features&quot;,
                                        size = 25,  face = &quot;bold&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-7-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>

hiv@meta.data %&gt;%
  group_by(orig.ident) %&gt;% 
  summarise(n = n(), q25 = quantile(nFeature_RNA, .25),
            q50 = quantile(nFeature_RNA, .5), 
            q75 = quantile(nFeature_RNA, .75), 
            mean = mean(nFeature_RNA)) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">orig.ident</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">4998</td>
<td align="right">128</td>
<td align="right">458.0</td>
<td align="right">1021</td>
<td align="right">654.7989</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">4995</td>
<td align="right">16</td>
<td align="right">32.0</td>
<td align="right">341</td>
<td align="right">315.3642</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">5000</td>
<td align="right">360</td>
<td align="right">581.0</td>
<td align="right">987</td>
<td align="right">738.4262</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">4996</td>
<td align="right">126</td>
<td align="right">584.5</td>
<td align="right">1087</td>
<td align="right">697.9960</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">5000</td>
<td align="right">175</td>
<td align="right">309.0</td>
<td align="right">666</td>
<td align="right">542.9580</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">4994</td>
<td align="right">22</td>
<td align="right">44.0</td>
<td align="right">114</td>
<td align="right">151.0993</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">4990</td>
<td align="right">31</td>
<td align="right">79.0</td>
<td align="right">206</td>
<td align="right">205.2401</td>
</tr>
</tbody>
</table>
</div>
<div id="library-size" class="section level4">
<h4>Library Size</h4>
<pre class="r"><code># have a look at the library size
p1 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_boxplot(aes(x = orig.ident, y = nCount_RNA, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p2 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_histogram(aes(x = nCount_RNA, fill = orig.ident), bins = 100) +
  facet_wrap(~ orig.ident, ncol = 4) +
  NoLegend()

p3 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = orig.ident, y = nCount_RNA, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p4 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = orig.ident, y = nCount_RNA, fill = orig.ident)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()
  
figure &lt;- ggarrange(p1, p3, p4, p2, ncol = 2, nrow = 2)
annotate_figure(figure, top = text_grob(&quot;Library size (number of counts)&quot;,
                                        size = 25,  face = &quot;bold&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-8-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>hiv@meta.data %&gt;%
  group_by(orig.ident) %&gt;% 
  summarise(n = n(), q25 = quantile( nCount_RNA, .25),
            q50 = quantile(nCount_RNA, .5), 
            q75 = quantile(nCount_RNA, .75), 
            mean = mean(nCount_RNA)) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">orig.ident</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">4998</td>
<td align="right">196</td>
<td align="right">772.0</td>
<td align="right">2098.00</td>
<td align="right">1424.6823</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">4995</td>
<td align="right">19</td>
<td align="right">39.0</td>
<td align="right">517.00</td>
<td align="right">648.8895</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">5000</td>
<td align="right">577</td>
<td align="right">1006.0</td>
<td align="right">1970.50</td>
<td align="right">1523.0344</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">4996</td>
<td align="right">173</td>
<td align="right">929.0</td>
<td align="right">2053.25</td>
<td align="right">1369.0498</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">5000</td>
<td align="right">238</td>
<td align="right">473.5</td>
<td align="right">1280.50</td>
<td align="right">1216.9344</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">4994</td>
<td align="right">26</td>
<td align="right">53.0</td>
<td align="right">146.00</td>
<td align="right">269.5036</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">4990</td>
<td align="right">37</td>
<td align="right">96.5</td>
<td align="right">272.00</td>
<td align="right">356.2377</td>
</tr>
</tbody>
</table>
</div>
<div id="percentage-of-mitochondrial-genes" class="section level4">
<h4>Percentage of Mitochondrial genes</h4>
<pre class="r"><code># have a look at the percentage of mitochondrial genes
p1 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_boxplot(aes(x = orig.ident, y = percent_mt, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p2 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_histogram(aes(x = percent_mt, fill = orig.ident), bins = 100) +
  facet_wrap(~ orig.ident, ncol = 4) +
  NoLegend()

p3 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = orig.ident, y = percent_mt, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p4 &lt;- hiv@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = orig.ident, y = percent_mt, fill = orig.ident)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()
  
figure &lt;- ggarrange(p1, p3, p4, p2, ncol = 2, nrow = 2)
Warning: Transformation introduced infinite values in continuous y-axis
Warning: Removed 2014 rows containing non-finite values (stat_ydensity).
annotate_figure(figure, top = text_grob(&quot;Percentage of Mitochondrial gene&quot;,
                                        size = 25,  face = &quot;bold&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-9-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
hiv@meta.data %&gt;%
  group_by(orig.ident) %&gt;% 
  summarise(n = n(), q25 = quantile( percent_mt, .25),
            q50 = quantile(percent_mt, .5), 
            q75 = quantile(percent_mt, .75), 
            mean = mean(percent_mt)) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">orig.ident</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">4998</td>
<td align="right">6.140696</td>
<td align="right">10.314638</td>
<td align="right">16.83490</td>
<td align="right">13.746867</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">4995</td>
<td align="right">4.000000</td>
<td align="right">8.522727</td>
<td align="right">15.38462</td>
<td align="right">12.092172</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">5000</td>
<td align="right">9.558731</td>
<td align="right">14.370956</td>
<td align="right">20.53664</td>
<td align="right">16.562072</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">4996</td>
<td align="right">7.387619</td>
<td align="right">11.188337</td>
<td align="right">16.68489</td>
<td align="right">13.814837</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">5000</td>
<td align="right">2.590861</td>
<td align="right">4.413543</td>
<td align="right">7.26931</td>
<td align="right">5.785039</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">4994</td>
<td align="right">4.165372</td>
<td align="right">8.333333</td>
<td align="right">14.70588</td>
<td align="right">10.832587</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">4990</td>
<td align="right">3.675874</td>
<td align="right">7.462687</td>
<td align="right">12.73541</td>
<td align="right">9.618704</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="quality-of-the-different-samples-after-quality-control" class="section level1">
<h1>Quality of the different samples after quality control</h1>
<ul>
<li>removing the 10% of cells with the highest percentage of mitochdonrial genes</li>
<li>remove features that are expressed in less than 3 cells</li>
<li>remove cells with 500 &lt; nFeature_RNA &gt; 2500</li>
</ul>
<p>It seems that the CSF1 samples, as well as the two uninfected samples have very few cells with nFeature_RNA &gt; 500, so we are left with less cells from these samples</p>
<div id="number-of-features-after-quality-control" class="section level4">
<h4>Number of features after Quality Control</h4>
<pre class="r"><code>
# Have a look at the number of Feature RNA
p1 &lt;- hiv_filt@meta.data %&gt;%
  ggplot() +
  geom_boxplot(aes(x = orig.ident, y = nFeature_RNA, fill = orig.ident)) +
  NoLegend()

p2 &lt;- hiv_filt@meta.data %&gt;% 
  ggplot()+
  geom_violin(aes(x = orig.ident, y = nFeature_RNA, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p3 &lt;- hiv_filt@meta.data %&gt;% 
  ggplot() +
  geom_histogram(aes(x = nFeature_RNA, fill = orig.ident)) +
  facet_wrap(~orig.ident, ncol = 4) +
  NoLegend()

figure &lt;- ggarrange(p1, p2, p3, ncol = 2, nrow = 2)
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
annotate_figure(figure, top = text_grob(&quot;Number of features after quality control&quot;,
                                        size = 25,  face = &quot;bold&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-10-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
hiv_filt@meta.data %&gt;% group_by(orig.ident) %&gt;%
  summarise(n = n(), q25 = quantile(nFeature_RNA, .25),
            q50 = quantile(nFeature_RNA, .5),
            q75 = quantile(nFeature_RNA, .75),
            mean = mean(nFeature_RNA)) %&gt;% kable</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">orig.ident</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">2076</td>
<td align="right">786.75</td>
<td align="right">1053.0</td>
<td align="right">1413.25</td>
<td align="right">1140.157</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">932</td>
<td align="right">800.00</td>
<td align="right">1077.0</td>
<td align="right">1395.00</td>
<td align="right">1142.855</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">2520</td>
<td align="right">672.00</td>
<td align="right">914.0</td>
<td align="right">1277.00</td>
<td align="right">1023.094</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">2309</td>
<td align="right">809.00</td>
<td align="right">1063.0</td>
<td align="right">1359.00</td>
<td align="right">1129.844</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">1384</td>
<td align="right">703.00</td>
<td align="right">1091.0</td>
<td align="right">1492.25</td>
<td align="right">1153.576</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">216</td>
<td align="right">672.50</td>
<td align="right">887.5</td>
<td align="right">1526.00</td>
<td align="right">1133.796</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">340</td>
<td align="right">670.50</td>
<td align="right">1066.5</td>
<td align="right">1606.00</td>
<td align="right">1155.576</td>
</tr>
</tbody>
</table>
</div>
<div id="library-size-after-quality-control" class="section level4">
<h4>Library size after Quality Control</h4>
<pre class="r"><code>p1 &lt;- hiv_filt@meta.data %&gt;%
  ggplot() +
  geom_boxplot(aes(x = orig.ident, y = nCount_RNA, fill = orig.ident)) +
  NoLegend()

p2 &lt;- hiv_filt@meta.data %&gt;% 
  ggplot()+
  geom_violin(aes(x = orig.ident, y = nCount_RNA, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p3 &lt;- hiv_filt@meta.data %&gt;% 
  ggplot() +
  geom_histogram(aes(x = nCount_RNA, fill = orig.ident)) +
  facet_wrap(~orig.ident, ncol = 4) +
  NoLegend()

figure &lt;- ggarrange(p1, p2, p3, ncol = 2, nrow = 2)
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
annotate_figure(figure, top = text_grob(&quot;Library size (number of counts) after quality control&quot;,
                                        size = 25,  face = &quot;bold&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-11-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-11-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
hiv_filt@meta.data %&gt;% group_by(orig.ident) %&gt;%
  summarise(n = n(), q25 = quantile(nCount_RNA, .25),
            q50 = quantile(nCount_RNA, .5),
            q75 = quantile(nCount_RNA, .75),
            mean = mean(nCount_RNA)) %&gt;% kable</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">orig.ident</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">2076</td>
<td align="right">1458.75</td>
<td align="right">2179.5</td>
<td align="right">3264.25</td>
<td align="right">2522.768</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">932</td>
<td align="right">1410.75</td>
<td align="right">2102.5</td>
<td align="right">3066.75</td>
<td align="right">2378.643</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">2520</td>
<td align="right">1187.50</td>
<td align="right">1779.5</td>
<td align="right">2822.25</td>
<td align="right">2158.089</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">2309</td>
<td align="right">1374.00</td>
<td align="right">1984.0</td>
<td align="right">2745.00</td>
<td align="right">2222.453</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">1384</td>
<td align="right">1377.25</td>
<td align="right">2381.5</td>
<td align="right">3845.25</td>
<td align="right">2842.006</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">216</td>
<td align="right">1063.25</td>
<td align="right">1491.5</td>
<td align="right">2968.75</td>
<td align="right">2177.278</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">340</td>
<td align="right">1063.50</td>
<td align="right">1876.0</td>
<td align="right">3399.50</td>
<td align="right">2328.871</td>
</tr>
</tbody>
</table>
</div>
<div id="percentage-of-mitochondrial-genes-after-quality-conrol" class="section level4">
<h4>Percentage of Mitochondrial genes after Quality Conrol</h4>
<pre class="r"><code>p1 &lt;- hiv_filt@meta.data %&gt;%
  ggplot() +
  geom_boxplot(aes(x = orig.ident, y = percent_mt, fill = orig.ident)) +
  NoLegend()

p2 &lt;- hiv_filt@meta.data %&gt;% 
  ggplot()+
  geom_violin(aes(x = orig.ident, y = percent_mt, fill = orig.ident)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1)) +
  NoLegend()

p3 &lt;- hiv_filt@meta.data %&gt;% 
  ggplot() +
  geom_histogram(aes(x = percent_mt, fill = orig.ident)) +
  facet_wrap(~orig.ident, ncol = 4) +
  NoLegend()

figure &lt;- ggarrange(p1, p2, p3, ncol = 2, nrow = 2)
`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
annotate_figure(figure, top = text_grob(&quot;Percentage of Mitochondrial genes after quality control&quot;,
                                        size = 25,  face = &quot;bold&quot;))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-12-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-12-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
hiv_filt@meta.data %&gt;% group_by(orig.ident) %&gt;%
  summarise(n = n(), q25 = quantile(percent_mt, .25),
            q50 = quantile(percent_mt, .5),
            q75 = quantile(percent_mt, .75),
            mean = mean(percent_mt)) %&gt;% kable</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">orig.ident</th>
<th align="right">n</th>
<th align="right">q25</th>
<th align="right">q50</th>
<th align="right">q75</th>
<th align="right">mean</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HIV1_Bld</td>
<td align="right">2076</td>
<td align="right">5.659778</td>
<td align="right">8.616470</td>
<td align="right">12.057771</td>
<td align="right">9.006083</td>
</tr>
<tr class="even">
<td align="left">HIV1_CSF</td>
<td align="right">932</td>
<td align="right">5.067679</td>
<td align="right">7.526483</td>
<td align="right">10.327807</td>
<td align="right">7.820102</td>
</tr>
<tr class="odd">
<td align="left">HIV2_Bld</td>
<td align="right">2520</td>
<td align="right">8.428856</td>
<td align="right">12.017682</td>
<td align="right">15.653130</td>
<td align="right">12.183780</td>
</tr>
<tr class="even">
<td align="left">HIV2_CSF</td>
<td align="right">2309</td>
<td align="right">7.420091</td>
<td align="right">10.051282</td>
<td align="right">13.119900</td>
<td align="right">10.296168</td>
</tr>
<tr class="odd">
<td align="left">HIV3_CSF</td>
<td align="right">1384</td>
<td align="right">2.045621</td>
<td align="right">3.169997</td>
<td align="right">4.804080</td>
<td align="right">3.582988</td>
</tr>
<tr class="even">
<td align="left">Uninfected1_CSF</td>
<td align="right">216</td>
<td align="right">4.339334</td>
<td align="right">5.927984</td>
<td align="right">8.122936</td>
<td align="right">6.373704</td>
</tr>
<tr class="odd">
<td align="left">Uninfected2_CSF</td>
<td align="right">340</td>
<td align="right">3.683668</td>
<td align="right">5.999618</td>
<td align="right">8.409969</td>
<td align="right">6.088557</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="subset-seurat-object" class="section level1">
<h1>Subset Seurat object</h1>
<p>Use 4 samples (2 blood, 2 csf) from two patients for further analysis like in the paper:</p>
<pre class="r"><code>hiv4 &lt;- subset(hiv_filt, orig.ident %in% c(&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV1_Bld&quot;, &quot;HIV2_Bld&quot;))</code></pre>
<ul>
<li>Normalize, Scale and find variable features</li>
</ul>
<pre class="r"><code>hiv4 &lt;- NormalizeData(hiv4, verbose = FALSE)
hiv4 &lt;- ScaleData(hiv4, verbose = FALSE)
hiv4 &lt;- FindVariableFeatures(hiv4, verbose = FALSE)</code></pre>
<ul>
<li>Save the log-normalized count matrix as a tsv file for Scenic Workflow</li>
</ul>
<pre><code># use only the infected samples, but not the uninfected samples
scenic_samples &lt;- subset(hiv_filt, orig.ident %in% c(&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;,
                                                     &quot;HIV1_Bld&quot;, &quot;HIV2_Bld&quot;, &quot;HIV3_CSF&quot;))

# because the snakemake pipeline of Carlos pySCENIC_pipeline: https://github.com/hdsu-bioquant/pySCENIC_pipeline/blob/master/input/gene_exp_mtx_example.tsv
# requires an input matrix which has genes as columns and cells as rows
# I will transpose the matrix

## Try with SCTTransform 
# select only the 3000 most highly variable features
# this might solve the problem with the snakemake file, because then you do not have genes with very low expression...
scenic_samples &lt;- SCTransform(scenic_samples, verbose = FALSE)
#hvg &lt;- scenic_samples@assays$SCT@var.features
sct_test_seurat &lt;- subset(scenic_samples, 
                          features = scenic_samples@assays$SCT@var.features)

sct_test_mtx &lt;- sct_test_seurat@assays$SCT@scale.data

# full matrix for Scenic workflow
sct_hiv_mtx &lt;- t(sct_test_mtx)
#write.table(x = sct_hiv_mtx,
#           file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/pySCENIC_pipeline/input/sct_hi#v_mtx.tsv&quot;, sep = &quot;\t&quot;)
</code></pre>
<div id="dimensionality-reduction" class="section level3">
<h3>Dimensionality reduction</h3>
<div id="pca" class="section level4">
<h4>PCA</h4>
<p>Compute PCA for highly variable genes: The first 10 dimensions seem to account for most of the variance (see elbow plot). The plots seem to indicate that the blood samples and csf samples are different.</p>
<pre class="r"><code>hiv4 &lt;- RunPCA(hiv4, features = VariableFeatures(hiv4), verbose = FALSE)

# elbow plot to determine which PCs account for most of the variance
ElbowPlot(hiv4)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-15-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>
#pca plots
pca_plots &lt;- comprehenr::to_list(for (i in 1:13)
  DimPlot(hiv4, reduction = &quot;pca&quot;, dims = i : (i+1)) +
    theme())
do.call(gridExtra::grid.arrange, c(pca_plots, ncol=3, nrow = 5))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-15-2.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="clustering-tsne-embedding" class="section level3">
<h3>Clustering &amp; tSNE embedding</h3>
<ul>
<li>compute KNN-graph</li>
<li>find clusters based on KNN-graph with Louvian algorithm</li>
</ul>
<pre class="r"><code>hiv4 &lt;- FindNeighbors(hiv4, verbose = FALSE)
hiv4 &lt;- FindClusters(hiv4, verbose = FALSE)

table(hiv4$seurat_clusters)

   0    1    2    3    4    5    6    7    8    9   10   11   12   13   14 
1301 1119  948  868  742  687  677  415  308  301  132  106  105   72   56 </code></pre>
<p>Questions:</p>
<p>When looking at the nFeatureRNA, percent_mt should some clustes be removed?</p>
<p>Is there a batch effect?</p>
<pre class="r"><code>hiv4 &lt;- RunTSNE(hiv4, dims = 1:10, verbose = FALSE)</code></pre>
<pre class="r"><code>tsne_coord &lt;- Embeddings(hiv4[[&quot;tsne&quot;]]) # tsne coordinates
meta &lt;- hiv4@meta.data # metadata
cm &lt;- hiv4@assays$RNA@counts # count matrix

# UMAP with number of features detected per cell
p1 &lt;- cbind(meta, tsne_coord) %&gt;%
  ggplot() +
  geom_point(aes(x=tSNE_1, y=tSNE_2, col=nFeature_RNA), size=.5) +
  scale_color_viridis_c()

# percentage of mitochondrial genes per cell
p2 &lt;- cbind(meta, tsne_coord) %&gt;% 
  ggplot() +
  geom_point(aes(x = tSNE_1, y = tSNE_2, col = percent_mt ), size = .5) +
  scale_color_viridis_c()

# UMAP indicating the sample where each cell came from
p3 &lt;- cbind(meta, tsne_coord) %&gt;% 
  ggplot() +
  geom_point(aes(x = tSNE_1, y = tSNE_2, col = orig.ident ), size = .5) 

# UMAP labeled according to clusters
p4 &lt;- cbind(meta, tsne_coord) %&gt;% 
  ggplot() +
  geom_point(aes(x = tSNE_1, y = tSNE_2, col = seurat_clusters ), size = .5)

ggarrange(p1, p2, p3, p4, ncol = 2, nrow = 2)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-18-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-18-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="find-marker-genes" class="section level3">
<h3>Find Marker genes</h3>
<pre><code>marker_genes &lt;- FindAllMarkers(hiv4, logfc.threshold = .01, min.pct = .01)</code></pre>
<pre class="r"><code># filter for significant marker genes
#sign_marker_genes &lt;- marker_genes %&gt;% filter(p_val_adj &lt;= .1)

sign_marker_genes &lt;- read.table(file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/sign_marker_genes.tsv&quot;)

#show the top 10 differentially expressed genes for each cluster
sign_marker_genes %&gt;% group_by(cluster) %&gt;% top_n(n = 2, wt = avg_logFC ) %&gt;% kable</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">p_val</th>
<th align="right">avg_logFC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
<th align="right">cluster</th>
<th align="left">gene</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0</td>
<td align="right">0.9583899</td>
<td align="right">0.545</td>
<td align="right">0.144</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">CCR7</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0.7910095</td>
<td align="right">0.463</td>
<td align="right">0.149</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">LEF1</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">1.3378085</td>
<td align="right">0.955</td>
<td align="right">0.401</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">CCL5</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">1.3312006</td>
<td align="right">0.627</td>
<td align="right">0.117</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="left">GZMK</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0.9648694</td>
<td align="right">0.818</td>
<td align="right">0.141</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="left">LYZ</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0.9325701</td>
<td align="right">0.039</td>
<td align="right">0.002</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="left">CLC</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0.5249648</td>
<td align="right">0.814</td>
<td align="right">0.617</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="left">IL7R</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0.5271194</td>
<td align="right">0.358</td>
<td align="right">0.172</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="left">GZMK</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0.5918064</td>
<td align="right">0.681</td>
<td align="right">0.468</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="left">LTB</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0.4685953</td>
<td align="right">0.831</td>
<td align="right">0.618</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="left">IL7R</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">1.3880325</td>
<td align="right">0.713</td>
<td align="right">0.183</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="left">GNLY</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">1.1492939</td>
<td align="right">0.677</td>
<td align="right">0.177</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="left">NKG7</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">0.7306204</td>
<td align="right">0.875</td>
<td align="right">0.457</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="left">LTB</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">0.6855607</td>
<td align="right">0.481</td>
<td align="right">0.172</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="left">MAF</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">2.7480043</td>
<td align="right">0.983</td>
<td align="right">0.093</td>
<td align="right">0</td>
<td align="right">7</td>
<td align="left">MS4A1</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">2.2013054</td>
<td align="right">0.670</td>
<td align="right">0.041</td>
<td align="right">0</td>
<td align="right">7</td>
<td align="left">IGHM</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">3.4870174</td>
<td align="right">1.000</td>
<td align="right">0.191</td>
<td align="right">0</td>
<td align="right">8</td>
<td align="left">LYZ</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">3.1688361</td>
<td align="right">0.977</td>
<td align="right">0.123</td>
<td align="right">0</td>
<td align="right">8</td>
<td align="left">S100A9</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">2.1622754</td>
<td align="right">0.804</td>
<td align="right">0.063</td>
<td align="right">0</td>
<td align="right">9</td>
<td align="left">FGFBP2</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">2.2885232</td>
<td align="right">0.957</td>
<td align="right">0.201</td>
<td align="right">0</td>
<td align="right">9</td>
<td align="left">GNLY</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">2.3732654</td>
<td align="right">0.955</td>
<td align="right">0.098</td>
<td align="right">0</td>
<td align="right">10</td>
<td align="left">FCGR3A</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">2.4715316</td>
<td align="right">0.977</td>
<td align="right">0.137</td>
<td align="right">0</td>
<td align="right">10</td>
<td align="left">AIF1</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">3.0007641</td>
<td align="right">0.860</td>
<td align="right">0.004</td>
<td align="right">0</td>
<td align="right">11</td>
<td align="left">C1QB</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">2.8593634</td>
<td align="right">0.794</td>
<td align="right">0.002</td>
<td align="right">0</td>
<td align="right">11</td>
<td align="left">C1QC</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">2.8500142</td>
<td align="right">1.000</td>
<td align="right">0.235</td>
<td align="right">0</td>
<td align="right">12</td>
<td align="left">HLA-DPA1</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">2.8524059</td>
<td align="right">1.000</td>
<td align="right">0.239</td>
<td align="right">0</td>
<td align="right">12</td>
<td align="left">HLA-DRA</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">3.7525676</td>
<td align="right">0.986</td>
<td align="right">0.014</td>
<td align="right">0</td>
<td align="right">13</td>
<td align="left">FCGR3B</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">3.6378226</td>
<td align="right">0.653</td>
<td align="right">0.053</td>
<td align="right">0</td>
<td align="right">13</td>
<td align="left">IL8</td>
</tr>
<tr class="odd">
<td align="right">0</td>
<td align="right">4.1426093</td>
<td align="right">0.911</td>
<td align="right">0.039</td>
<td align="right">0</td>
<td align="right">14</td>
<td align="left">IGJ</td>
</tr>
<tr class="even">
<td align="right">0</td>
<td align="right">4.0013955</td>
<td align="right">0.321</td>
<td align="right">0.026</td>
<td align="right">0</td>
<td align="right">14</td>
<td align="left">IGHG1</td>
</tr>
</tbody>
</table>
</div>
<div id="cell-type-annotation-with-canonical-marker-genes" class="section level2">
<h2>Cell type annotation with canonical marker genes:</h2>
<div id="b-cell-markers" class="section level4">
<h4>B cell markers</h4>
<p>Are found primarily in cluster 7.</p>
<pre class="r"><code>p1 &lt;- VlnPlot(hiv4, features = c(&quot;MS4A1&quot;, &quot;CD79A&quot;, &quot;TCL1A&quot;), pt.size = .01)
p2 &lt;- RidgePlot(hiv4, features =  c(&quot;MS4A1&quot;, &quot;CD79A&quot;, &quot;TCL1A&quot;), sort = &quot;decreasing&quot; )
ggarrange(p1,p2, nrow = 2, ncol = 1)
Picking joint bandwidth of 0.0311
Picking joint bandwidth of 0.0413
Picking joint bandwidth of 2.99e-06</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-20-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-20-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nk-cells" class="section level4">
<h4>NK cells</h4>
<p>In cluster 5 and 9. GZMB is also foudn in cluster 14.</p>
<pre class="r"><code>p1 &lt;- VlnPlot(hiv4, features = c(&quot;NKG7&quot;, &quot;GNLY&quot;, &quot;GZMB&quot;),
        pt.size = .01, ncol = 3)#, stack = TRUE)
p2 &lt;- RidgePlot(hiv4, features = c(&quot;NKG7&quot;, &quot;GNLY&quot;, &quot;GZMB&quot;), sort = &quot;decreasing&quot;)

ggarrange(p1,p2, nrow = 2, ncol = 1)
Picking joint bandwidth of 0.0482
Picking joint bandwidth of 0.0544
Picking joint bandwidth of 0.0471</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-21-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-21-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tcells" class="section level4">
<h4>Tcells</h4>
<ul>
<li>A general T cell marker CD3E is expressed in cluster 0,1,3,4,5,6.</li>
<li>IL7R in clusters 0-6</li>
<li>TRAC in clusters 0-6, excluding 2</li>
<li>CCR7 in 0,6,7</li>
<li>LDHB not very clear, but 0,1,3,4,6</li>
<li>LTB in 0-7</li>
<li>CD2 in 1,3,4,5,9</li>
</ul>
<p>Tcells 1, 2, 3 are assigned arbitrarily, because they are not used for further analysis.</p>
<p>CD8+ T cells * CD8A is expressed in decreasing order in 1,5,3 (could cluster 1 and 5 be CD8 T cells?)</p>
<pre class="r"><code># general T cell marker for CD4+ and CD8+ Tcells
p1 &lt;- VlnPlot(hiv4, features = c(&quot;CD8A&quot;,&quot;CD3E&quot;), pt.size = .01, ncol = 2)
p2 &lt;- RidgePlot(hiv4, features = c(&quot;CD8A&quot;,&quot;CD3E&quot;), sort = &quot;decreasing&quot;)


p3 &lt;- VlnPlot(hiv4, features = c(&quot;IL7R&quot;, &quot;TRAC&quot;), pt.size = 0.01)
p4 &lt;- RidgePlot(hiv4, features = c(&quot;IL7R&quot;, &quot;TRAC&quot;), sort = &quot;decreasing&quot;)

# helper T cells
p5 &lt;- VlnPlot(hiv4, features = c(&quot;LTB&quot;, &quot;CD2&quot;), pt.size = .01, ncol = 2)
p6 &lt;- RidgePlot(hiv4, features = c(&quot;LTB&quot;, &quot;CD2&quot;), sort = &quot;decreasing&quot;)

# CD4+ naive T cells
p7 &lt;- VlnPlot(hiv4, features = c(&quot;LDHB&quot;, &quot;CCR7&quot;), pt.size = .01, ncol = 2)
p8 &lt;- RidgePlot(hiv4, features = c(&quot;LDHB&quot;, &quot;CCR7&quot;), sort = &quot;decreasing&quot;)


ggarrange(p1, p2,p3, p4, p5, p6, p7,p8, ncol = 2, nrow = 4)
Picking joint bandwidth of 0.0515
Picking joint bandwidth of 0.0941
Picking joint bandwidth of 0.169
Picking joint bandwidth of 0.1
Picking joint bandwidth of 0.17
Picking joint bandwidth of 0.161
Picking joint bandwidth of 0.248
Picking joint bandwidth of 0.0395</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-22-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-22-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="plasma-cells" class="section level4">
<h4>plasma cells</h4>
<p>in cluster 14</p>
<pre class="r"><code>p1 &lt;- VlnPlot(hiv4, features = c(&quot;IGJ&quot;), pt.size = .01, ncol = 2)
p2 &lt;- RidgePlot(hiv4, features = c(&quot;IGJ&quot;), sort = &quot;decreasing&quot;)
ggarrange(p1,p2, nrow = 1, ncol = 2)
Picking joint bandwidth of 0.0585</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-23-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-23-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="myeloid-cells" class="section level4">
<h4>Myeloid cells</h4>
<ul>
<li>granulocytes</li>
<li>monocytes</li>
<li>macrophages</li>
<li>dendritic cells</li>
</ul>
<p>-&gt; clusters 8, 10, 11, 12, 13 are myeloid cells</p>
<p>The 3 genes (AXL, APOE, TREM2) associated with neurodegenrative-disaes-associated microglia are expressed in myeloid2, however, only at very low concentrations</p>
<p>Myeloid1, 3, 4 are assigned arbitrarily, because they are not of interest.</p>
<pre class="r"><code>p1 &lt;- VlnPlot(hiv4, features = c(&quot;LYZ&quot;, &quot;FCER1A&quot;, &quot;CD14&quot;, &quot;FCGR3A&quot;), # FCG3A is also expressed in cluster 9 (NK cells)
        pt.size = .01, ncol = 2)
p2 &lt;- RidgePlot(hiv4, features = c(&quot;LYZ&quot;, &quot;FCER1A&quot;),
          sort = &quot;decreasing&quot;, ncol = 2) # DC markers
p3 &lt;- RidgePlot(hiv4, features = c(&quot;CD14&quot;, &quot;FCGR3A&quot;), 
          sort = &quot;decreasing&quot;, ncol = 2) # monocytes (LYC also in monocytes)
ggarrange(p1,p2,p3, nrow = 3, ncol = 1)
Picking joint bandwidth of 0.14
Picking joint bandwidth of 0.0228
Picking joint bandwidth of 0.0393
Picking joint bandwidth of 0.0896</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-24-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-24-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="myeloid-5" class="section level5">
<h5>Myeloid 5</h5>
<ul>
<li>(from paper)</li>
<li>was characterized by: FCER1A, CD1C</li>
<li>cluster 12</li>
</ul>
<pre class="r"><code># Myeloid 5 in the paper
p1 &lt;- VlnPlot(hiv4, features = c( &quot;FCER1A&quot;, &quot;CD1C&quot;), pt.size = .01, ncol = 2)
p2 &lt;- RidgePlot(hiv4, features = c(&quot;FCER1A&quot;, &quot;CD1C&quot;), sort = &quot;decreasing&quot;, ncol = 2)
ggarrange(p1, p2, nrow = 2, ncol = 1)
Picking joint bandwidth of 0.0228
Picking joint bandwidth of 0.0264</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-25-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-25-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="myeloid-2" class="section level5">
<h5>Myeloid 2</h5>
<ul>
<li>(from paper) -&gt; cluster 11</li>
<li>AXL, APOE, TREM2 (these genes are enriched in neurodegenerative disease-associated microglia)</li>
<li>“CTSB,” “APOC1,” “MSR1,” “C1QA-C” are expressed more highly in myeloid 2 compared to other myeloid clusters</li>
</ul>
<pre class="r"><code># Myeloid 2
p1 &lt;- VlnPlot(hiv4, features = c(&quot;AXL&quot;, &quot;APOE&quot;, &quot;TREM2&quot;), pt.size = .01, ncol = 3)
p2 &lt;- RidgePlot(hiv4, features = c(&quot;AXL&quot;, &quot;APOE&quot;, &quot;TREM2&quot;), sort = &quot;decreasing&quot;)
p3 &lt;- VlnPlot(hiv4, features = c(&quot;CTSB&quot;, &quot;APOC1&quot;, &quot;MSR1&quot;, &quot;C1QA&quot;), 
              pt.size = .01, ncol = 4)
p4 &lt;- RidgePlot(hiv4, features = c(&quot;CTSB&quot;, &quot;APOC1&quot;, &quot;MSR1&quot;, &quot;C1QA&quot;),
                sort = &quot;decreasing&quot;, ncol = 4)

ggarrange(p1, p2, p3, p4, ncol = 1, nrow = 4)
Picking joint bandwidth of 0.0119
Picking joint bandwidth of 0.0305
Picking joint bandwidth of 0.0205
Picking joint bandwidth of 0.0836
Picking joint bandwidth of 0.0249
Picking joint bandwidth of 0.0291
Picking joint bandwidth of 0.0248</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-26-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-26-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="annotate-cells" class="section level2">
<h2>Annotate cells</h2>
<ul>
<li>Why is the mixed population primarily composed of cells from HIV1 patient blood sample?</li>
<li>Why are T cells primarily from CSF samples and CD8 T cells primarily from blood?</li>
</ul>
<pre class="r"><code>Idents(hiv4) &lt;- &quot;seurat_clusters&quot;
cell_types &lt;- c(&quot;Tcell2&quot;, &quot;CD8&quot;, &quot;mixed&quot;, &quot;CD81&quot;, &quot;Tcell3&quot;, &quot;NKcell1&quot;,
                &quot;Tcell1&quot;, &quot;Bcell&quot;, &quot;Myeloid1&quot;, &quot;NKcell2&quot;, &quot;Myeloid3&quot;, &quot;Myeloid2&quot;
                , &quot;Myeloid5&quot;, &quot;Myeloid4&quot;, &quot;plasma cell&quot; )

names(cell_types) &lt;- levels(hiv4)
hiv4 &lt;- RenameIdents(hiv4, cell_types)
as.data.frame(cell_types) %&gt;%  rownames_to_column(&quot;cluster&quot;) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cluster</th>
<th align="left">cell_types</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">0</td>
<td align="left">Tcell2</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">CD8</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">mixed</td>
</tr>
<tr class="even">
<td align="left">3</td>
<td align="left">CD81</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="left">Tcell3</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="left">NKcell1</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="left">Tcell1</td>
</tr>
<tr class="even">
<td align="left">7</td>
<td align="left">Bcell</td>
</tr>
<tr class="odd">
<td align="left">8</td>
<td align="left">Myeloid1</td>
</tr>
<tr class="even">
<td align="left">9</td>
<td align="left">NKcell2</td>
</tr>
<tr class="odd">
<td align="left">10</td>
<td align="left">Myeloid3</td>
</tr>
<tr class="even">
<td align="left">11</td>
<td align="left">Myeloid2</td>
</tr>
<tr class="odd">
<td align="left">12</td>
<td align="left">Myeloid5</td>
</tr>
<tr class="even">
<td align="left">13</td>
<td align="left">Myeloid4</td>
</tr>
<tr class="odd">
<td align="left">14</td>
<td align="left">plasma cell</td>
</tr>
</tbody>
</table>
<pre class="r"><code>
# add cell types to metadata
hiv4 &lt;- AddMetaData(hiv4, metadata = Idents(hiv4), col.name = &quot;cell_type&quot;)


# add condition blood or csf to metadata
Idents(hiv4) &lt;- &quot;orig.ident&quot;
conditions &lt;- c(&quot;Blood&quot;, &quot;Blood&quot;, &quot;CSF&quot;, &quot;CSF&quot;)
names(conditions) &lt;- c(&quot;HIV1_Bld&quot;, &quot;HIV2_Bld&quot;, &quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;)
hiv4 &lt;- RenameIdents(hiv4, conditions)

hiv4 &lt;- AddMetaData(hiv4, col.name = &quot;Condition&quot;, metadata = Idents(hiv4))
</code></pre>
<p>Overview of some of the most important marker genes:</p>
<pre class="r"><code>
DotPlot(object = hiv4, features = c(&quot;MS4A1&quot;, # B cells
                                    &quot;TRAC&quot;, &quot;IL7R&quot;, 
                                    &quot;CD8A&quot;, # CD8 T cells
                                    &quot;NKG7&quot;, &quot;GNLY&quot;, # NK cells
                                    &quot;CD3E&quot;, # T cells
                                    &quot;CD14&quot;,&quot;FCGR3A&quot;, &quot;LYZ&quot;, # myeloid cells
                                    &quot;IGJ&quot;, # plasma cells
                                    &quot;AXL&quot;, &quot;APOE&quot;, &quot;TREM2&quot;, # myeloid 2
                                     &quot;FCER1A&quot;, &quot;CD1C&quot; # myeloid 5
)) + theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-28-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Visualize the cell types, conditions, samples and nCountRNA:</p>
<pre class="r"><code>p1 &lt;- DimPlot(hiv4, reduction = &quot;tsne&quot;, label = TRUE, 
              group.by = &quot;cell_type&quot;, pt.size = 0.1) +
  NoLegend()
p2 &lt;- DimPlot(hiv4, reduction = &quot;tsne&quot;, group.by =  &quot;Condition&quot;, pt.size = 0.1)
p3 &lt;- DimPlot(hiv4, reduction = &quot;tsne&quot;, group.by = &quot;orig.ident&quot;, pt.size = 0.1)
p4 &lt;- FeaturePlot(hiv4, features= &quot;nCount_RNA&quot;, pt.size = 0.1) + 
    scale_color_viridis_c()
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
ggarrange(p1,p2,p3,p4, ncol = 2, nrow = 2)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-29-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-29-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#p1 + p2 + p3 + p4</code></pre>
<ul>
<li>Blood and CSF were processed separately for scRNAseq using SeqWell.</li>
<li>The 4 samples were sequenced separately.</li>
<li>One SeqWell array per CSF or PBMC samples. <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span></li>
</ul>
</div>
</div>
<div id="relative-contributions-of-csf-and-blood-cells-to-each-cluster" class="section level1">
<h1>Relative contributions of CSF and blood cells to each cluster</h1>
<p>Plot a histogram in order to identify clusters that are predominantly composed of CSF cells.</p>
<pre class="r"><code>
custom_color &lt;- c(&quot;#C3D7A4&quot;, &quot;#4E84C4&quot;, &quot;#52854C&quot;, &quot;#293352&quot;)

#plot a barplot of the proportions of samples and cell types
ggplot(hiv4@meta.data) +
  geom_bar(aes(x = cell_type, fill = orig.ident)) +
  scale_fill_manual( values = custom_color) +
  ylab(&quot;Number of cells in each cluster&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-30-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Question</strong>: Myeloid-1, Myeloid-2 are mainly found in Blood cells, Myeloid-4 is only found in Blood cells from HIV1 patient. What does that mean?</p>
<div id="how-big-are-myeloid2-and-myeloid5-clusters-and-where-do-cells-come-from" class="section level2">
<h2>How big are Myeloid2 and Myeloid5 clusters and where do cells come from?</h2>
<p>Myeloid2 cells originate from the CSF samples of HIV-1-infected patients, while a few (13) cells classified as Myeloid5 cells originate from Blood samples.</p>
<pre class="r"><code>hiv4@meta.data %&gt;% 
  filter(cell_type %in% c(&quot;Myeloid2&quot;, &quot;Myeloid5&quot;)) %&gt;% 
  group_by(cell_type, orig.ident) %&gt;% 
  summarize(n = n()) %&gt;% kable()
`summarise()` has grouped output by &#39;cell_type&#39;. You can override using the `.groups` argument.</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">cell_type</th>
<th align="left">orig.ident</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Myeloid2</td>
<td align="left">HIV1_CSF</td>
<td align="right">71</td>
</tr>
<tr class="even">
<td align="left">Myeloid2</td>
<td align="left">HIV2_CSF</td>
<td align="right">35</td>
</tr>
<tr class="odd">
<td align="left">Myeloid5</td>
<td align="left">HIV1_Bld</td>
<td align="right">4</td>
</tr>
<tr class="even">
<td align="left">Myeloid5</td>
<td align="left">HIV1_CSF</td>
<td align="right">24</td>
</tr>
<tr class="odd">
<td align="left">Myeloid5</td>
<td align="left">HIV2_Bld</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">Myeloid5</td>
<td align="left">HIV2_CSF</td>
<td align="right">68</td>
</tr>
</tbody>
</table>
</div>
<div id="which-genes-are-upregulated-in-myeloid-2-compared-to-the-four-other-myeloid-subsets" class="section level2">
<h2>Which genes are upregulated in Myeloid-2 compared to the four other myeloid subsets?</h2>
<p>In the paper they find 60 genes that are upregulated compared to the other 4 myeloid subsets. (FDR &lt; 0.01)</p>
<p>Compare the genes upregulated in myeloid-2 to the four other myeloid subsets. <code>FindAllMarkers()</code> will compare one cluster to all other clusters. <code>FindMarkers()</code> will compare an indicated cluster with a second indicated cluster. Therefore, the difference will not be diluted, which might be the case if we compare one cluster to many other clusters.</p>
<pre class="r"><code>#import the 60 genes that were upregulated in Myeloid-2 in the paper compared to the four other myeloid subsets
paper &lt;- read.table(file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/Myeloid-2_genes_paper.csv&quot;, sep = &quot;\t&quot;,
                    header = FALSE, skipNul = TRUE)#, n_max = 60)
paper[1,] &lt;- &quot;APOC1&quot;
colnames(paper) &lt;- &quot;&quot;
paper &lt;- as.vector(paper[,1])

Idents(hiv4) &lt;- &quot;cell_type&quot;
all.comp &lt;- purrr::map(c(1,3,4,5), function(n){
  markers &lt;- FindMarkers(hiv4,
              ident.1 = &quot;Myeloid2&quot;,
              ident.2 = paste0(&quot;Myeloid&quot;, n), only.pos = TRUE)
  markers &lt;- rownames_to_column(markers, var = &quot;gene&quot;)
  name &lt;- paste0(&quot;Comparison Myeloid 2 vs Myeloid &quot;, n)
  list(df = markers, name = name)
})


map(seq.int(1,4), function(n) {
  all.comp[[n]]$df %&gt;% 
  filter(p_val_adj &lt;= .2) %&gt;%
  filter(gene %in% paper) %&gt;% head() %&gt;% kable()
})</code></pre>
<p>[[1]]</p>
<table>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="right">p_val</th>
<th align="right">avg_log2FC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C1QB</td>
<td align="right">0</td>
<td align="right">4.322413</td>
<td align="right">0.858</td>
<td align="right">0.006</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">C1QC</td>
<td align="right">0</td>
<td align="right">4.154311</td>
<td align="right">0.792</td>
<td align="right">0.000</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">C1QA</td>
<td align="right">0</td>
<td align="right">3.320555</td>
<td align="right">0.783</td>
<td align="right">0.016</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">APOE</td>
<td align="right">0</td>
<td align="right">3.594296</td>
<td align="right">0.726</td>
<td align="right">0.000</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">LYVE1</td>
<td align="right">0</td>
<td align="right">3.889012</td>
<td align="right">0.651</td>
<td align="right">0.000</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">MSR1</td>
<td align="right">0</td>
<td align="right">2.784011</td>
<td align="right">0.679</td>
<td align="right">0.058</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>[[2]]</p>
<table>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="right">p_val</th>
<th align="right">avg_log2FC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C1QB</td>
<td align="right">0</td>
<td align="right">3.830956</td>
<td align="right">0.858</td>
<td align="right">0.068</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">C1QC</td>
<td align="right">0</td>
<td align="right">4.022519</td>
<td align="right">0.792</td>
<td align="right">0.008</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">CD14</td>
<td align="right">0</td>
<td align="right">3.605687</td>
<td align="right">0.830</td>
<td align="right">0.053</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">HLA-DPB1</td>
<td align="right">0</td>
<td align="right">2.540395</td>
<td align="right">0.962</td>
<td align="right">0.470</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">HLA-DRA</td>
<td align="right">0</td>
<td align="right">2.381817</td>
<td align="right">0.981</td>
<td align="right">0.720</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">APOE</td>
<td align="right">0</td>
<td align="right">3.594296</td>
<td align="right">0.726</td>
<td align="right">0.000</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>[[3]]</p>
<table>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="right">p_val</th>
<th align="right">avg_log2FC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HLA-DRA</td>
<td align="right">0</td>
<td align="right">4.314835</td>
<td align="right">0.981</td>
<td align="right">0.069</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">HLA-DPB1</td>
<td align="right">0</td>
<td align="right">3.571806</td>
<td align="right">0.962</td>
<td align="right">0.097</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">CD74</td>
<td align="right">0</td>
<td align="right">3.850998</td>
<td align="right">0.972</td>
<td align="right">0.236</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">HLA-DPA1</td>
<td align="right">0</td>
<td align="right">3.969028</td>
<td align="right">0.915</td>
<td align="right">0.083</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">C1QB</td>
<td align="right">0</td>
<td align="right">4.376544</td>
<td align="right">0.858</td>
<td align="right">0.000</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">HLA-DRB1</td>
<td align="right">0</td>
<td align="right">3.970074</td>
<td align="right">0.887</td>
<td align="right">0.069</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>[[4]]</p>
<table>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="right">p_val</th>
<th align="right">avg_log2FC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">C1QB</td>
<td align="right">0</td>
<td align="right">3.957066</td>
<td align="right">0.858</td>
<td align="right">0.048</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">C1QC</td>
<td align="right">0</td>
<td align="right">4.154311</td>
<td align="right">0.792</td>
<td align="right">0.000</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">CD14</td>
<td align="right">0</td>
<td align="right">3.253946</td>
<td align="right">0.830</td>
<td align="right">0.133</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">C1QA</td>
<td align="right">0</td>
<td align="right">3.136295</td>
<td align="right">0.783</td>
<td align="right">0.048</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">APOE</td>
<td align="right">0</td>
<td align="right">3.564194</td>
<td align="right">0.726</td>
<td align="right">0.010</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">LYVE1</td>
<td align="right">0</td>
<td align="right">3.889012</td>
<td align="right">0.651</td>
<td align="right">0.000</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>
my2_genes &lt;- map(seq.int(1,4), function(n) {
  name &lt;- all.comp[[]]$name
  my2_genes &lt;- all.comp[[n]]$df %&gt;% 
  filter(p_val_adj &lt;= .2) %&gt;%
  filter(gene %in% paper)
  my2_genes &lt;- my2_genes[,&quot;gene&quot;]
})</code></pre>
</div>
<div id="show-expression-of-the-60-genes-identified-in-the-paper-as-myeloid-2-markers" class="section level2">
<h2>Show expression of the 60 genes identified in the paper as Myeloid 2 markers</h2>
<p>The genes show the highest expression in Myeloid2 and Myeloid5 cells.</p>
<pre class="r"><code>DotPlot(hiv4, features = c(paper)) + 
  coord_flip() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1))
Warning in FetchData(object = object, vars = features, cells = cells): The
following requested variables were not found: SELENOP</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-33-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-33-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="percentage-of-myeloid2-transcripts-per-cell-in-blood-csf-and-uninfected-samples" class="section level2">
<h2>Percentage of Myeloid2 transcripts per Cell in Blood, CSF and Uninfected samples</h2>
<pre class="r"><code>Idents(hiv_filt) = &quot;orig.ident&quot;
table(Idents(hiv_filt))

       HIV1_Bld        HIV1_CSF        HIV2_Bld        HIV2_CSF        HIV3_CSF 
           2076             932            2520            2309            1384 
Uninfected1_CSF Uninfected2_CSF 
            216             340 
bld &lt;- subset(hiv_filt, 
              orig.ident %in% c(&quot;HIV1_Bld&quot;, 
                                &quot;HIV2_Bld&quot;))@assays$RNA@counts
csf &lt;- subset(hiv_filt, 
              orig.ident %in% c(&quot;HIV1_CSF&quot;, 
                                &quot;HIV2_CSF&quot;, 
                                &quot;HIV3_CSF&quot;))@assays$RNA@counts
healthy &lt;- subset(hiv_filt,
                  orig.ident %in% c(&quot;Uninfected1_CSF&quot;,
                                    &quot;Uninfected2_CSF&quot;))@assays$RNA@counts

matrix_list &lt;- list(&quot;blood&quot; = bld, &quot;csf&quot; = csf, &quot;uninfected&quot; = healthy)

percentage_my2_transcripts &lt;- map(seq(1,3), function(n) {
  count_m &lt;- matrix_list[[n]]
  total_transcripts &lt;- colSums(count_m) # number of transcripts per cell
  my2_matrix &lt;- count_m[rownames(count_m) %in% paper, ] # get only myeloid2 genes 
  my2_transcripts &lt;- colSums(my2_matrix) # sum of myeloid2 transcripts
  percentage_my2_transcripts &lt;- my2_transcripts/total_transcripts # fraction
})


ggplot() +
  geom_boxplot(aes(x = &quot;HIV+ CSF&quot;, y= percentage_my2_transcripts[[2]], fill = &quot;CSF&quot;)) +
  geom_boxplot(aes(x = &quot;Uninfected_CSF&quot;, y = percentage_my2_transcripts[[3]], fill = &quot;Uninfeced_CSF&quot;)) +
  geom_boxplot(aes(x = &quot;HIV+ Blood&quot;,y = percentage_my2_transcripts[[1]], fill = &quot;Blood&quot;)) +
  xlab(&quot;Sample origin&quot;) +
  ylab(&quot;Frequency of myeloid2 transcripts per cell&quot;) </code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-34-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="summary-statistics" class="section level3">
<h3>Summary statistics</h3>
<div id="csf-from-hiv-1-infected-patients" class="section level4">
<h4>CSF from HIV-1 infected patients</h4>
<pre class="r"><code>#CSF
summary(percentage_my2_transcripts[[2]])#%&gt;% kable()</code></pre>
<pre><code>Min.  1st Qu.   Median     Mean  3rd Qu.     Max. </code></pre>
<p>0.003974 0.015144 0.018277 0.020325 0.021825 0.135734</p>
</div>
<div id="blood-from-hiv-1-infected-patients" class="section level4">
<h4>Blood from HIV-1 infected patients</h4>
<pre class="r"><code>#Blood
summary(percentage_my2_transcripts[[1]]) </code></pre>
<pre><code> Min.   1st Qu.    Median      Mean   3rd Qu.      Max. </code></pre>
<p>0.0009302 0.0154600 0.0189051 0.0191886 0.0223501 0.0556745</p>
</div>
<div id="csf-from-uninfected-patients" class="section level4">
<h4>CSF from uninfected patients</h4>
<pre class="r"><code># Uninfected
summary(percentage_my2_transcripts[[3]])</code></pre>
<pre><code>Min.  1st Qu.   Median     Mean  3rd Qu.     Max. </code></pre>
<p>0.005771 0.012050 0.015132 0.021576 0.028695 0.096354</p>
</div>
</div>
<div id="what-are-the-differences-in-myeloid2-transcripts-between-samples" class="section level3">
<h3>What are the differences in Myeloid2 transcripts between samples?</h3>
<ul>
<li><strong>HIV+ Blood</strong> vs. <strong>HIV+ CSF</strong>: p-value = 2.95 * 10^-4</li>
<li><strong>HIV+ CSF</strong> vs. <strong>HIV- CSF</strong>: p-value = 2.87 * 10^-7 (same result as in the paper)</li>
<li><strong>HIV+ Blood</strong> vs. <strong>HIV- CSF</strong>: p-value = 7.98 * 10^-7</li>
</ul>
<p>In the paper they found the largest difference between infected blood and csf samples, but I find larger differences between HIV+ CSF/Blood and HIV- CSF samples.</p>
<ul>
<li>My results suggest that Myeloid2 transcripts are detected at lower levels in HIV- CSF samples than in HIV+ CSF samples.</li>
<li>Myeloid2 transcripts are found at lower levels in HIV- CSF samples than in HIV+ Blood samples.</li>
<li>Myeloid2 transcripts are detected at similar levels in HIV+ CSF samples and in HIV+ Blood samples, however there are far more outliers with very high frequency of Myeloid2 transcripts in HIV+ CSF samples compared to HIV+ Blood.</li>
</ul>
<p>How much sense does this interpretation make? The median is rather robust to outliers, so even if in CSF there are some cells which have much higher frequencies of transcripts, it might be that the median is still lower than in blood.</p>
<p>Wilcoxon rank sum test of the null hypothesis that the distribution of x and y differs by a location shift of mu and the alternative is that they differ by some other location shif. (If I set mu = 0, the null hypothesis is that the distributiosn differ by a location shift of 0).</p>
<div id="wilcoxon-test-between-hiv-blood-an-hiv-csf" class="section level5">
<h5>Wilcoxon test between HIV+ Blood an HIV+ CSF</h5>
<pre class="r"><code>
# bld vs csf
wilcox.test(percentage_my2_transcripts[[1]], # blood samples
            percentage_my2_transcripts[[2]], # csf samples
            alternative = &quot;two.sided&quot;, 
            paired = FALSE, 
            conf.int = TRUE)# %&gt;% kable()</code></pre>
<pre><code>Wilcoxon rank sum test with continuity correction</code></pre>
<p>data: percentage_my2_transcripts[[1]] and percentage_my2_transcripts[[2]] W = 11090858, p-value = 0.0002952 alternative hypothesis: true location shift is not equal to 0 95 percent confidence interval: 0.0001932858 0.0006133828 sample estimates: difference in location 0.0004188108</p>
</div>
<div id="wilcoxon-test-between-hiv-csf-and-hiv--csf" class="section level5">
<h5>Wilcoxon test between HIV+ CSF and HIV- CSF</h5>
<pre class="r"><code># csf vs uninfected
wilcox.test(percentage_my2_transcripts[[2]], # csf samples
            percentage_my2_transcripts[[3]],
            alternative = &quot;two.sided&quot;,
            conf.int = TRUE) # uninfected samples

    Wilcoxon rank sum test with continuity correction

data:  percentage_my2_transcripts[[2]] and percentage_my2_transcripts[[3]]
W = 1456760, p-value = 2.87e-07
alternative hypothesis: true location shift is not equal to 0
95 percent confidence interval:
 0.001120779 0.002379251
sample estimates:
difference in location 
           0.001757612 </code></pre>
</div>
<div id="wilcoxon-test-between-hiv-blood-and-hiv--csf" class="section level5">
<h5>Wilcoxon test between HIV+ Blood and HIV- CSF</h5>
<pre class="r"><code># blood vs uninfected
wilcox.test(percentage_my2_transcripts[[1]], # blood samples
            percentage_my2_transcripts[[3]], # uninfected samples
            alternative = &quot;two.sided&quot;,
            conf.int = TRUE)

    Wilcoxon rank sum test with continuity correction

data:  percentage_my2_transcripts[[1]] and percentage_my2_transcripts[[3]]
W = 1441190, p-value = 7.984e-07
alternative hypothesis: true location shift is not equal to 0
95 percent confidence interval:
 0.001101844 0.002441881
sample estimates:
difference in location 
           0.001778961 </code></pre>
</div>
</div>
</div>
</div>
<div id="label-transfer" class="section level1">
<h1>Label Transfer</h1>
<p>After reproducing the results from <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> the next step was to transfer the labels from the four HIV+ Blood and CSF samples to the remaining samples:</p>
<ul>
<li>1 CSF smaple from HIV-1 infected patient</li>
<li>2 CSF samples from uninfected patients</li>
</ul>
<p>For reference see: <a href="https://satijalab.org/seurat/articles/integration_mapping.html" class="uri">https://satijalab.org/seurat/articles/integration_mapping.html</a> <a href="https://satijalab.org/seurat/articles/multimodal_reference_mapping.html" class="uri">https://satijalab.org/seurat/articles/multimodal_reference_mapping.html</a></p>
<pre class="r"><code># Seurat objects
hiv4 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/non_integrated_HIV1_HIV2_4samples_seurat_object.rds&quot;)

hiv7 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_filtered_seurat_object.rds&quot;)</code></pre>
<div id="find-integration-anchors" class="section level2">
<h2>Find Integration anchors</h2>
<p>We perform Normalization separately for each dataset and try to find Variable Features.</p>
<pre class="r"><code>#hiv_list &lt;- SplitObject(hiv7, split.by = &quot;orig.ident&quot;)

#for (i in 1:length(hiv_list)) {
 # hiv_list[[i]] &lt;- NormalizeData(hiv_list[[i]], verbose = FALSE)
  #hiv_list[[i]] &lt;- FindVariableFeatures(hiv_list[[1]],
   #                                     selection.method = &quot;vst&quot;,
    #                                    nfeatures = 2000,
     #                                   verbose = FALSE)
#}

#saveRDS(hiv_list, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv_list&quot;)
hiv_list &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv_list&quot;)
#reference.list &lt;- SplitObject(hiv4, split.by = &quot;orig.ident&quot;)</code></pre>
</div>
<div id="cell-type-classification-using-an-integrated-reference" class="section level2">
<h2>Cell type classification using an integrated reference</h2>
<ul>
<li>in data transfer we begin with identifying anchors</li>
<li>we do not correct or modify the query expression data</li>
<li>it is suggested to use the default method tsne (which project the PCA structure of a reference onto a query dataset (instead of learning a joint structure)</li>
<li>we classify the query cells based on the reference data (a vector of reference cell type labels)</li>
</ul>
<p>We will use the four samples we have analyzed above as a reference. The query dataset will be the HIV+ CSF3 sample and the two uninfected sample:</p>
<p>The label transfer identifies hardly any B cells and Myeloid 1, 3, 4 are also not/hardly present in the CSF samples which is expected, since they originated exclusively from the Blood samples.</p>
<pre class="r"><code># get Seurat object for CSF3 sample
#hiv3csf &lt;- hiv_list[[&quot;HIV3_CSF&quot;]]
# merge the 3 query Seurat objects
query &lt;- merge(hiv_list[[&quot;Uninfected1_CSF&quot;]], 
               c(hiv_list[[&quot;Uninfected2_CSF&quot;]], hiv_list[[&quot;HIV3_CSF&quot;]]))
Warning in CheckDuplicateCellNames(object.list = objects): Some cell names are
duplicated across objects provided. Renaming to enforce unique cell names.

# find transfer anchors between the 3 query samples and the 4 samples we have 
# analyzed previously
hiv4.transfer_anchors &lt;- FindTransferAnchors(reference = hiv4,
                                             query = query, 
                                             dims = 1:15,
                                             normalization.method = &quot;LogNormalize&quot;,
                                             reference.reduction = &quot;pca&quot;,
                                             reduction = &quot;pcaproject&quot;,
                                             verbose = FALSE)

predictions &lt;- TransferData(anchorset = hiv4.transfer_anchors, 
                            refdata = hiv4$cell_type, 
                            dims = 1:15,
                            verbose = FALSE)
#predictions %&gt;% head %&gt;% kable()

query &lt;- AddMetaData(query, metadata = predictions)
table(query$predicted.id) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Var1</th>
<th align="right">Freq</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Bcell</td>
<td align="right">501</td>
</tr>
<tr class="even">
<td align="left">CD8</td>
<td align="right">180</td>
</tr>
<tr class="odd">
<td align="left">CD81</td>
<td align="right">141</td>
</tr>
<tr class="even">
<td align="left">mixed</td>
<td align="right">144</td>
</tr>
<tr class="odd">
<td align="left">Myeloid1</td>
<td align="right">381</td>
</tr>
<tr class="even">
<td align="left">Myeloid3</td>
<td align="right">237</td>
</tr>
<tr class="odd">
<td align="left">Myeloid4</td>
<td align="right">207</td>
</tr>
<tr class="even">
<td align="left">Myeloid5</td>
<td align="right">12</td>
</tr>
<tr class="odd">
<td align="left">NKcell1</td>
<td align="right">738</td>
</tr>
<tr class="even">
<td align="left">NKcell2</td>
<td align="right">450</td>
</tr>
<tr class="odd">
<td align="left">plasma cell</td>
<td align="right">72</td>
</tr>
<tr class="even">
<td align="left">Tcell1</td>
<td align="right">258</td>
</tr>
<tr class="odd">
<td align="left">Tcell2</td>
<td align="right">2541</td>
</tr>
<tr class="even">
<td align="left">Tcell4</td>
<td align="right">366</td>
</tr>
</tbody>
</table>
</div>
<div id="check-the-gene-expression-levels-in-the-predicted-cell-types" class="section level2">
<h2>Check the gene expression levels in the predicted cell types</h2>
<p>The Label Transfer seems to work nicely, judging by the correct expression of common cell type markers for the individual cell types.</p>
<div id="myeloid2-marker-genes" class="section level4">
<h4>Myeloid2 marker genes</h4>
<p>```#{r, fig.width=12, fig.height = } # ceck the Myeloid2 markers p1 &lt;- VlnPlot(query, features = c(“AXL,” “APOE,” “TREM2”), group.by = “predicted.id,” pt.size = .01, ncol = 3) p2 &lt;- VlnPlot(query, features = c(“CTSB,” “MSR1,” “C1QA”), group.by = “predicted.id,” pt.size = .01, ncol = 4)</p>
<p>figure &lt;- ggarrange(p1, p2, ncol = 1, nrow = 2) annotate_figure(figure, top = text_grob(“Myeloid2 marker genes”))</p>
<pre><code>
#### Myeloid5 marker genes

```r

# Myeloid 5 markers
p1 &lt;- VlnPlot(query, c(&quot;FCER1A&quot;, &quot;CD1C&quot;),
        group.by = &quot;predicted.id&quot;, pt.size = .01, ncol = 2) 
p1</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-44-1">
Past versions of unnamed-chunk-44-1.png
</button>
</p>
<div id="fig-unnamed-chunk-44-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-44-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="plasma-cell-marker-genes" class="section level4">
<h4>Plasma cell marker genes</h4>
<pre class="r"><code># plasma cell markers
p2 &lt;- VlnPlot(query, features = c(&quot;IGJ&quot;), 
        group.by = &quot;predicted.id&quot;, pt.size = .01, ncol = 2) 
p2</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-45-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-45-1">
Past versions of unnamed-chunk-45-1.png
</button>
</p>
<div id="fig-unnamed-chunk-45-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-45-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cd8-t-cell-marker-genes" class="section level4">
<h4>CD8 T cell marker genes</h4>
<pre class="r"><code># CD8 T Cells
p3 &lt;- VlnPlot(query, features = c(&quot;CD8A&quot;),
        group.by = &quot;predicted.id&quot;, pt.size = .01, ncol = 2) + NoLegend()
p3</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-46-1">
Past versions of unnamed-chunk-46-1.png
</button>
</p>
<div id="fig-unnamed-chunk-46-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-46-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="t-cell-marker-genes" class="section level4">
<h4>T cell marker genes</h4>
<pre class="r"><code># T cells 
VlnPlot(query, features = c(&quot;IL7R&quot;, &quot;TRAC&quot;),
        group.by = &quot;predicted.id&quot;, pt.size = .01, ncol = 2)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-47-1">
Past versions of unnamed-chunk-47-1.png
</button>
</p>
<div id="fig-unnamed-chunk-47-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-47-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nk-cell-marker-genes" class="section level4">
<h4>NK cell marker genes</h4>
<pre class="r"><code># NK cells
VlnPlot(query, features = c(&quot;NKG7&quot;, &quot;GNLY&quot;, &quot;GZMB&quot;), 
        group.by = &quot;predicted.id&quot;, pt.size = .01, ncol = 3)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-48-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-48-1">
Past versions of unnamed-chunk-48-1.png
</button>
</p>
<div id="fig-unnamed-chunk-48-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-48-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="b-cell-marker-genes" class="section level4">
<h4>B cell marker genes</h4>
<pre class="r"><code># B cells
VlnPlot(query, features = c(&quot;MS4A1&quot;, &quot;CD79A&quot;, &quot;TCL1A&quot;), 
        group.by = &quot;predicted.id&quot;, pt.size = .01, ncol = 3)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-49-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-49-1">
Past versions of unnamed-chunk-49-1.png
</button>
</p>
<div id="fig-unnamed-chunk-49-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-49-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="umap-projection-of-additional-samples" class="section level2">
<h2>UMAP projection of additional samples</h2>
<ul>
<li>Calculate UMAP on the four reference samples</li>
<li>Integrate the embeddings of the reference samples with the embeddings of the query samples</li>
</ul>
<pre class="r"><code>hiv4_test &lt;- RunUMAP(hiv4, dims = 1:15, 
                     reduction = &quot;pca&quot;, 
                     return.model = TRUE, 
                     verbose = FALSE)
Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to &#39;umap-learn&#39; and metric to &#39;correlation&#39;
This message will be shown once per session

hiv3csf &lt;- MapQuery(anchorset = hiv4.transfer_anchors,
                    reference = hiv4_test, query = query,
                    refdata = list(celltype = &quot;cell_type&quot;),
                    reference.reduction = &quot;pca&quot;,
                    reduction.model = &quot;umap&quot;, 
                    verbose = FALSE)
Warning: Keys should be one or more alphanumeric characters followed by an
underscore, setting key from ref.pca_ to refpca_
Warning: UNRELIABLE VALUE: One of the &#39;future.apply&#39; iterations
(&#39;future_lapply-1&#39;) unexpectedly generated random numbers without declaring so.
There is a risk that those random numbers are not statistically sound and the
overall results might be invalid. To fix this, specify &#39;future.seed=TRUE&#39;. This
ensures that proper, parallel-safe random numbers are produced via the L&#39;Ecuyer-
CMRG method. To disable this check, use &#39;future.seed = NULL&#39;, or set option
&#39;future.rng.onMisuse&#39; to &quot;ignore&quot;.
Warning: Keys should be one or more alphanumeric characters followed by an
underscore, setting key from ref.pca_ to refpca_
Warning: All keys should be one or more alphanumeric characters followed by an
underscore &#39;_&#39;, setting key to refpca_
Computing nearest neighbors
Running UMAP projection
10:43:01 Read 6228 rows and found  numeric columns
10:43:01 Processing block 1 of 1
10:43:01 Commencing smooth kNN distance calibration using 1 thread
10:43:01 Initializing by weighted average of neighbor coordinates using 1 thread
10:43:01 Commencing optimization for 167 epochs, with 124560 positive edges
10:43:05 Finished
Warning: No assay specified, setting assay as RNA by default.




query &lt;- TransferData(anchorset = hiv4.transfer_anchors,
                               reference = hiv4_test,
                               query = query,
                               refdata = list(celltype = &quot;cell_type&quot;), 
                      verbose = FALSE)

query &lt;- IntegrateEmbeddings(anchorset = hiv4.transfer_anchors,
                                      reference = hiv4_test,
                                      query = query,
                               new.reduction.name = &quot;ref.pca&quot;,
                             verbose = FALSE)
Warning: Keys should be one or more alphanumeric characters followed by an
underscore, setting key from ref.pca_ to refpca_
Warning: UNRELIABLE VALUE: One of the &#39;future.apply&#39; iterations
(&#39;future_lapply-1&#39;) unexpectedly generated random numbers without declaring so.
There is a risk that those random numbers are not statistically sound and the
overall results might be invalid. To fix this, specify &#39;future.seed=TRUE&#39;. This
ensures that proper, parallel-safe random numbers are produced via the L&#39;Ecuyer-
CMRG method. To disable this check, use &#39;future.seed = NULL&#39;, or set option
&#39;future.rng.onMisuse&#39; to &quot;ignore&quot;.
Warning: Keys should be one or more alphanumeric characters followed by an
underscore, setting key from ref.pca_ to refpca_
Warning: All keys should be one or more alphanumeric characters followed by an
underscore &#39;_&#39;, setting key to refpca_

query &lt;- ProjectUMAP(query = query,
                       query.reduction = &quot;ref.pca&quot;,
                       reference = hiv4_test,
                       reference.reduction = &quot;pca&quot;,
                       reduction.model = &quot;umap&quot;, verbose = FALSE)
Computing nearest neighbors
Warning: No assay specified, setting assay as RNA by default.</code></pre>
<pre class="r"><code># Visualize the query cells alongside the reference
p1 &lt;- DimPlot(hiv4_test,
              reduction = &quot;umap&quot;,
              group.by = &quot;cell_type&quot;,
              label = TRUE,
              label.size = 5,
              repel = TRUE, 
              pt.size = .1) + NoLegend() +
  ggtitle(&quot;Reference anntoations&quot;)

p2 &lt;- DimPlot(query, 
              reduction = &quot;ref.umap&quot;,
              group.by = &quot;predicted.celltype&quot;, 
              label = TRUE,
              label.size = 5,
              pt.size = .1,
              repel = TRUE) + NoLegend() +
   ggtitle(&quot;Transferred Labels&quot;)

p1 + p2</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-51-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-51-1">
Past versions of unnamed-chunk-51-1.png
</button>
</p>
<div id="fig-unnamed-chunk-51-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-51-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#save this new Seurat obejct including the SCTransformatin

#saveRDS(hiv_label_transfer, file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds&quot;)</code></pre>
</div>
<div id="frequency-of-cell-types-in-each-sample" class="section level2">
<h2>Frequency of cell types in each sample</h2>
<p>How are cell types distributed across samples, considering also the additional HIV+ CSF sample and the two uninfected samples which were annotated using label transfer.</p>
<pre class="r"><code>hiv7 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds&quot;)

custom_color &lt;- c(&quot;#C3D7A4&quot;, &quot;#4E84C4&quot;, &quot;#52854C&quot;, &quot;#293352&quot;, &quot;lightskyblue&quot;, &quot;orangered2&quot;, &quot;orange2&quot;)

#plot a barplot of the proportions of samples and cell types
p2 &lt;- ggplot(hiv7@meta.data) +
  geom_bar(aes(x = cell_type, fill = orig.ident)) +
  scale_fill_manual( values = custom_color) +
  ylab(&quot;Number of cells in each cluster&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
  labs(title = &quot;Frequency of cell types from each sample after label transfer&quot;)

custom_color &lt;- c(&quot;#C3D7A4&quot;, &quot;#4E84C4&quot;, &quot;#52854C&quot;, &quot;#293352&quot;)

#plot a barplot of the proportions of samples and cell types
p1 &lt;- ggplot(hiv4@meta.data) +
  geom_bar(aes(x = cell_type, fill = orig.ident)) +
  scale_fill_manual( values = custom_color) +
  ylab(&quot;Number of cells in each cluster&quot;) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
  labs(title = &quot;Frequecy of cell types from each sample after label transfer&quot;)

ggarrange(p1, p2, ncol = 2, nrow = 1)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-53-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="independent-annotation-of-myeloid2-and-myeloid5-clusters-as-microglia-like-cells" class="section level1">
<h1>Independent annotation of Myeloid2 and Myeloid5 clusters as Microglia-like cells</h1>
<p>Can we identify the microglia-like myeloid2/myeloid5 cluster they found in the paper by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> when using independent genes for annotations?</p>
<p>Here, I will first use the 881 Microglia marker genes and then also try the list of TFs provided by <span class="citation">(<a href="#ref-Gosselin" role="doc-biblioref">David Gosselin 2020</a>)</span>.</p>
<div id="function-add-module-score" class="section level4">
<h4>Function Add Module Score</h4>
<p>This Function Caluclate the average expression levels of each cluster on single cell level, subtracted by the aggregated expression of control feature sets. The analyzed features are binned based on averaged expression and the control features are randomly selected from each bin.</p>
</div>
<div id="read-in-seurat-object" class="section level4">
<h4>read in Seurat object</h4>
<pre class="r"><code>hiv4 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/non_integrated_HIV1_HIV2_4samples_seurat_object.rds&quot;)</code></pre>
</div>
<div id="gene-signature-defined-by-farhadian" class="section level2">
<h2>Gene signature defined by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span></h2>
<p>Do we find the 60 signature genes from <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> for myeloid2 among the signature defined by @[Gossselin]?</p>
<div id="read-in-list-of-microglia-signature-genes-found-in-the-gosselin-paper" class="section level5">
<h5>Read in list of microglia signature genes found in the Gosselin paper</h5>
<p>which are also found in the HIV1 and HIV2 samples:</p>
<pre class="r"><code>microglia_genes &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/Microglia_gene_signatur_found_in_HIV1_HIV2_samples&quot;)</code></pre>
<p>28 of the 60 genes used as microglia signature in the <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> Paper are also found in the 881 microglia signature genes defined by <span class="citation">(<a href="#ref-Gosselin" role="doc-biblioref">David Gosselin 2020</a>)</span>.</p>
<pre class="r"><code># Are there any genes overlapping between Farhadian and Glass paper?
#import the 60 genes that were upregulated in Myeloid-2 in the paper compared to the four other myeloid subsets
paper &lt;- read.table(file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/Myeloid-2_genes_paper.csv&quot;, sep = &quot;\t&quot;, header = FALSE, skipNul = TRUE)#, n_max = 60)
paper[1,] &lt;- &quot;APOC1&quot;
colnames(paper) &lt;- &quot;&quot;
paper &lt;- as.vector(paper[,1])

# Are there any genes overlapping between Farhadian and Glass paper?
#paper %in% as.vector(microglia_genes[[&quot;gene_name&quot;]])

overlap &lt;- as.vector(microglia_genes[[&quot;gene_name&quot;]]) [as.vector(microglia_genes[[&quot;gene_name&quot;]]) %in% paper]
length(overlap)</code></pre>
<p>[1] 28</p>
<pre class="r"><code>overlap %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">x</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">SCIN</td>
</tr>
<tr class="even">
<td align="left">CCR5</td>
</tr>
<tr class="odd">
<td align="left">HLA-DMB</td>
</tr>
<tr class="even">
<td align="left">MSR1</td>
</tr>
<tr class="odd">
<td align="left">HAVCR2</td>
</tr>
<tr class="even">
<td align="left">VSIG4</td>
</tr>
<tr class="odd">
<td align="left">IL18</td>
</tr>
<tr class="even">
<td align="left">C3</td>
</tr>
<tr class="odd">
<td align="left">SLC2A5</td>
</tr>
<tr class="even">
<td align="left">FCGBP</td>
</tr>
<tr class="odd">
<td align="left">CD14</td>
</tr>
<tr class="even">
<td align="left">OLR1</td>
</tr>
<tr class="odd">
<td align="left">TREM2</td>
</tr>
<tr class="even">
<td align="left">FOLR2</td>
</tr>
<tr class="odd">
<td align="left">HLA-DMA</td>
</tr>
<tr class="even">
<td align="left">ZFHX3</td>
</tr>
<tr class="odd">
<td align="left">HLA-DQA1</td>
</tr>
<tr class="even">
<td align="left">HLA-DPA1</td>
</tr>
<tr class="odd">
<td align="left">DAB2</td>
</tr>
<tr class="even">
<td align="left">C1QC</td>
</tr>
<tr class="odd">
<td align="left">C1QB</td>
</tr>
<tr class="even">
<td align="left">LYVE1</td>
</tr>
<tr class="odd">
<td align="left">CYFIP1</td>
</tr>
<tr class="even">
<td align="left">HLA-DRB1</td>
</tr>
<tr class="odd">
<td align="left">BHLHE41</td>
</tr>
<tr class="even">
<td align="left">MAF</td>
</tr>
<tr class="odd">
<td align="left">C1QA</td>
</tr>
<tr class="even">
<td align="left">HLA-DRA</td>
</tr>
</tbody>
</table>
</div>
<div id="genes-described-by-farhadian-et-al.-to-distinguish-the-myeloid2-cell-type" class="section level4">
<h4>60 genes described by Farhadian et al. to distinguish the Myeloid2 cell type</h4>
<pre class="r"><code># get only genes present in the HIV1 and HIV2 samples!
paper &lt;- paper[paper %in% rownames(hiv4@assays$RNA@counts)]


hiv_farhadian &lt;- AddModuleScore(hiv4, 
                           features = paper)

Idents(hiv_farhadian) &lt;- &quot;cell_type&quot;
FeaturePlot(hiv_farhadian, features = &quot;Cluster1&quot; , pt.size = .5, label = TRUE, repel = TRUE) +
    scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Farhadian et al., 2018: Microglia signature genes&quot;)
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-57-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="gene-signature-defined-by-gosselin" class="section level2">
<h2>Gene signature defined by @[Gosselin]</h2>
<p>The Transcription Factors mentioned in the paper do not seem to be specific for the Myeloid2 cluster. Interestingly, there is a very low score<br />
of these microglia genes in Myeloid5, while there is a higher score in Myeloid1 and Myeloid4 cells.</p>
<pre class="r"><code>gosselin_tfs &lt;- read.table( &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/TFs_microglia_Glass_paper.txt&quot;)


hiv4_glass &lt;- AddModuleScore(hiv4, 
                           features = as.vector(gosselin_tfs))
Warning: The following features are not present in the object: MEF2B, SALL1, not
searching for symbol synonyms

Idents(hiv4_glass) &lt;- &quot;cell_type&quot;
FeaturePlot(hiv4_glass, features = &quot;Cluster1&quot;, label = TRUE, repel = TRUE, pt.size = .7) +
    scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Gosselin D. et al., 2017: Microglia Transcription Factors&quot;) 
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-58-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="signature-of-hand-hiv-associated-neurodegenerative-disease" class="section level3">
<h3>signature of HAND (HIV-associated neurodegenerative disease)</h3>
<p><span class="citation">(<a href="#ref-Gosselin" role="doc-biblioref">David Gosselin 2020</a>)</span> defined a group of genes upregulated in microglia from patients with HAND.</p>
<p>This gene signature can, however not be detected in any of the Myeloid clusters.</p>
<pre class="r"><code>hand_genes &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/HAND_upregulated_microglia_signature_genes&quot;)
hand_genes &lt;- as.vector(hand_genes[[&quot;genes&quot;]])

hiv_hand &lt;- AddModuleScore(hiv4,
                           features = hand_genes)

Idents(hiv_hand) &lt;- &quot;cell_type&quot;
FeaturePlot(hiv_hand, features = &quot;Cluster1&quot;, pt.size = .01) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;microglia signature genes upregulated in HAND&quot;)
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-59-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="tfs-downregulated-in-in-vitro-microglia" class="section level5">
<h5>TFs downregulated in in vitro microglia</h5>
<p>The list of TFs is derived from <span class="citation">(<a href="#ref-Gosselin" role="doc-biblioref">David Gosselin 2020</a>)</span>:</p>
<p><strong>Comparison between microglia and primitive yolk sac macrphages:</strong></p>
<p>In this paper they found:</p>
<ul>
<li>a subset of mRNAs was expressed in primitive yolk sac macrophages and maintained expression levels in brain microglia</li>
<li>a second subset were preferentially expressed in adult mouse/human microglia -&gt; increase in expression from ebryonic to adult stage of brain development</li>
<li>a majority of the TFs that showed reduced expression after transfer to in <em>in vitro</em> environment were induced following migration of primitive macrophages into the developing brain -&gt; induction by local <strong>environment factors</strong></li>
</ul>
<pre class="r"><code># TFs downregulated in in vitro microglia
tfs_down &lt;- c( &quot;IRF1&quot;, &quot;IRF2&quot;, &quot;IRF3&quot;, &quot;IRF8&quot;, &quot;IRF9&quot;, &quot;CTCF&quot;, &quot;MEF2A&quot;, &quot;MEF2C&quot;, &quot;MEF2D&quot;,&quot;MAF&quot;, &quot;MAF1&quot;, &quot;MAFB&quot;, &quot;MAFF&quot;,&quot;MAFG&quot;, &quot;MAFK&quot;, &quot;RUNX1&quot;, &quot;RUNX2&quot;, &quot;SMAD3&quot;, &quot;CEBPB&quot;, &quot;CEBPA&quot;, &quot;CEBPG&quot;, &quot;JUND&quot;, &quot;JUNB&quot;, &quot;JUN&quot;, &quot;FOS&quot;, &quot;FOSB&quot;, &quot;ARID3A&quot;, &quot;ARID5A&quot;, &quot;BHLHE41&quot;, &quot;BTG2&quot;, &quot;DBP&quot;, &quot;EGR1&quot;, &quot;EGR2&quot;, &quot;EGR3&quot;, &quot;ERF&quot;, &quot;ELF1&quot;, &quot;ELMSAN1&quot;, &quot;ETV6&quot;, &quot;FLI1&quot;, &quot;KLF2&quot;, &quot;KLF4&quot;, &quot;KLF6&quot;, &quot;MLXIPL&quot;, &quot;MNT&quot;, &quot;MYCL&quot;, &quot;NFATC2&quot;, &quot;NFE2L2&quot;,&quot;NR4A1&quot;, &quot;RELA&quot;, &quot;RREB1&quot;, &quot;STAT3&quot;, &quot;TAL1&quot;, &quot;TCF4&quot;, &quot;TFEB&quot;, &quot;USF2&quot;, &quot;ZEB2&quot;, &quot;ZFHX3&quot;, &quot;ZNF217&quot;, &quot;ZNF691&quot;)

#map(seq.int(1:59), function(n){
 # VlnPlot(hiv4, features = tfs_down[n], group.by = &quot;cell_type&quot;)
#})

tfs_down %in% rownames(hiv4@assays$RNA@counts)
 [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[16] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[31] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[46] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE

hiv_down &lt;- AddModuleScore(hiv4,
                           features = tfs_down)
hiv_hand@meta.data %&gt;% head
                orig.ident nCount_RNA nFeature_RNA percent_mt RNA_snn_res.0.8
1B_CCTTTCCTGTCT   HIV1_Bld       4403         1697  13.203267               6
1B_GTCGAGTCATGC   HIV1_Bld       5530         2489  10.604148               8
1B_GCTCGATAGATC   HIV1_Bld       6311         2329   6.147541              14
1B_GAGCGGAATGTC   HIV1_Bld       5008         2256   9.982068               8
1B_AGCACTAAATGA   HIV1_Bld       5477         2272   9.848623               8
1B_AAATACCTCCTT   HIV1_Bld       4856         2067  11.826194               8
                seurat_clusters   cell_type Condition    Cluster1    Cluster2
1B_CCTTTCCTGTCT               6      Tcell1     Blood -0.01185150 -0.01185150
1B_GTCGAGTCATGC               8    Myeloid1     Blood -0.05162929 -0.01032586
1B_GCTCGATAGATC              14 plasma cell     Blood -0.03326784  0.00000000
1B_GAGCGGAATGTC               8    Myeloid1     Blood -0.07683537 -0.02195094
1B_AGCACTAAATGA               8    Myeloid1     Blood -0.01038798  0.00000000
1B_AAATACCTCCTT               8    Myeloid1     Blood -0.03354566  0.00000000
                  Cluster3    Cluster4    Cluster5   Cluster6    Cluster7
1B_CCTTTCCTGTCT -0.1711938 -0.09481203 -0.05267871  0.8129650 -0.04082721
1B_GTCGAGTCATGC  0.8367329 -0.08757767 -0.19580706  0.5080454 -0.06692596
1B_GCTCGATAGATC -0.1760869 -0.09495456 -0.05697274 -0.2900837 -0.03798182
1B_GAGCGGAATGTC -0.1528933 -0.12920008  1.02071849  0.5238480 -0.08704052
1B_AGCACTAAATGA  0.8389540 -0.07770004 -0.14459255 -0.4540580 -0.06190830
1B_AAATACCTCCTT  1.0063699 -0.08945510 -0.12887531 -0.4381535 -0.04472755
                    Cluster8   Cluster9   Cluster10  Cluster11  Cluster12
1B_CCTTTCCTGTCT  0.109149738 -0.2556897 -0.01185150 -0.2344179 -0.2594270
1B_GTCGAGTCATGC  1.083696302 -0.5916853 -0.05660010 -0.2198099  1.7947622
1B_GCTCGATAGATC -0.031303009 -0.4116142 -0.04747728 -0.3022513 -0.2660752
1B_GAGCGGAATGTC  0.501772959  0.5163614 -0.04390187 -0.2097841  1.4557755
1B_AGCACTAAATGA -1.144407281  0.5573299 -0.08599533  1.5305484  1.5297456
1B_AAATACCTCCTT -0.006147703  1.2222966 -0.02236378 -0.2630477 -0.2209738
                  Cluster13  Cluster14  Cluster15   Cluster16   Cluster17
1B_CCTTTCCTGTCT -0.10535742 -0.1917523 -0.2238725 -0.04740602 -0.06796459
1B_GTCGAGTCATGC  0.89834966  0.9024493  1.1869001 -0.10617957 -0.05660010
1B_GCTCGATAGATC -0.14572067 -0.1362252 -0.2283970 -0.07124966 -0.09824339
1B_GAGCGGAATGTC -0.13681173  0.9497430  0.8364252 -0.08193440 -0.02195094
1B_AGCACTAAATGA -0.09638331  1.3664463  0.7920274 -0.07770004 -0.10886396
1B_AAATACCTCCTT -0.09460206  0.9978417 -0.2390786 -0.11607785 -0.03354566
                  Cluster18   Cluster19  Cluster20  Cluster21   Cluster22
1B_CCTTTCCTGTCT -0.09481203 -0.07638172 -0.1172089 -0.3433302 -0.05925752
1B_GTCGAGTCATGC -0.06692596 -0.09585371  1.6250830  1.2799337  1.37185756
1B_GCTCGATAGATC -0.04747728 -0.01899091 -0.1077388 -0.4447810 -0.06026156
1B_GAGCGGAATGTC -0.08780374 -0.09877921  0.8864123  0.6806042 -0.07095893
1B_AGCACTAAATGA -0.10346020 -0.05692409  2.3814315 -0.4915839  1.75405936
1B_AAATACCTCCTT -0.06709133 -0.04472755  0.9566267  0.7409743 -0.10578394
                 Cluster23   Cluster24   Cluster25   Cluster26  Cluster27
1B_CCTTTCCTGTCT -0.2509493 -0.11851504 -1.15216148 -0.08508879 -0.2172937
1B_GTCGAGTCATGC  1.2982862 -0.17590541  0.38108650 -0.23299993  0.7424081
1B_GCTCGATAGATC -0.2140687 -0.11723430 -0.34307750 -0.12357591  0.7500735
1B_GAGCGGAATGTC  2.6422368 -0.09877921 -0.08850152  1.42995107 -0.3944558
1B_AGCACTAAATGA  2.4130680 -0.08808801 -0.23350686  1.68938237 -0.2253309
1B_AAATACCTCCTT  2.4387579 -0.04987450 -1.23274033 -0.13667593 -0.2579007
                  Cluster28  Cluster29  Cluster30  Cluster31  Cluster32
1B_CCTTTCCTGTCT -0.02370301 -0.1727292 -0.1012842 -0.1422180 -0.1119362
1B_GTCGAGTCATGC  0.00000000  1.8499190 -0.2276558  1.6614615 -0.2065630
1B_GCTCGATAGATC -0.01899091 -0.2021407 -0.1053900 -0.1742070 -0.2075424
1B_GAGCGGAATGTC -0.01097547  1.6176990 -0.2052651  0.9587215  0.9032500
1B_AGCACTAAATGA  0.00000000  1.7506792  0.7532265  0.8701180 -0.2597700
1B_AAATACCTCCTT -0.01118189  1.9353344 -0.1721028 -0.2493567 -0.1977323
                 Cluster33 Cluster34   Cluster35  Cluster36   Cluster37
1B_CCTTTCCTGTCT -0.3721854 -1.125850 -0.02897570 -0.2870966 -0.03555451
1B_GTCGAGTCATGC -0.5647451  2.737049 -0.15985868  2.7042712 -0.05162929
1B_GCTCGATAGATC  0.4933810 -1.116219 -0.05704023 -0.5267935 -0.11723430
1B_GAGCGGAATGTC -0.6183499  2.822841 -0.13094239  2.9561573 -0.05487734
1B_AGCACTAAATGA -0.4939252  3.451155 -0.11163267  1.9075026 -0.07271583
1B_AAATACCTCCTT -0.3572089  2.484923 -0.06105639  1.5958590 -0.03354566
                   Cluster38   Cluster39   Cluster40  Cluster41  Cluster42
1B_CCTTTCCTGTCT  0.000000000  0.13493767 -0.01185150 -1.1829772 -0.3213450
1B_GTCGAGTCATGC -0.010325858 -0.09277292 -0.05660010  0.4476676 -0.5277103
1B_GCTCGATAGATC -0.009495456  0.67305550 -0.10396532 -0.2774846 -0.4287091
1B_GAGCGGAATGTC  0.000000000  0.18601703 -0.08704052 -1.1488277 -0.4549405
1B_AGCACTAAATGA  0.000000000 -1.01032267 -0.08101112 -1.0408202  0.5870177
1B_AAATACCTCCTT  0.000000000  0.46748856 -0.03089199 -1.3687689 -0.3249921
                 Cluster43  Cluster44  Cluster45  Cluster46   Cluster47
1B_CCTTTCCTGTCT -0.2238725 -0.9900413 -0.4285220 -0.1982705 -0.12935038
1B_GTCGAGTCATGC -0.2982971  1.2641662 -0.4845284  1.3185883 -0.08222262
1B_GCTCGATAGATC  0.6000566  0.3007009 -0.3806861 -0.3542705 -0.04747728
1B_GAGCGGAATGTC -0.2728674  1.4044580 -0.4439579  0.4901317 -0.10388533
1B_AGCACTAAATGA -0.2231027  1.3671608  0.5616674 -0.3738313 -0.06271839
1B_AAATACCTCCTT  0.8852764  0.7487823 -0.4555017 -0.4358931 -0.06092504
                  Cluster48   Cluster49   Cluster50    Cluster51  Cluster52
1B_CCTTTCCTGTCT -0.02897570 -0.02370301 -0.13776750 -0.093505919 -0.4540209
1B_GTCGAGTCATGC -0.04130343 -0.06654171  0.84834007 -0.082606866  1.0113769
1B_GCTCGATAGATC -0.01899091 -0.01899091 -0.14734829 -0.009495456 -0.3771029
1B_GAGCGGAATGTC -0.04900799 -0.01097547 -0.13170561 -0.098779208  1.9882014
1B_AGCACTAAATGA -0.07229627 -0.09139910  2.21153126  0.971485482  2.7827850
1B_AAATACCTCCTT  1.03138734 -0.03869262 -0.07827321 -0.027510729  2.5344562
                 Cluster53  Cluster54  Cluster55  Cluster56   Cluster57
1B_CCTTTCCTGTCT -0.1409119  0.6037291 -0.3555921 -0.3491957 -0.05267871
1B_GTCGAGTCATGC  1.8946643  0.3738071 -0.5557669 -0.5430518 -0.03097757
1B_GCTCGATAGATC -0.1201293 -1.1744456 -0.3359002 -0.4263089 -0.08545910
1B_GAGCGGAATGTC -0.1266945  1.1070367 -0.5678757 -0.5849845 -0.08780374
1B_AGCACTAAATGA -0.1479036  1.1981006 -0.4946537 -0.5641671 -0.06232785
1B_AAATACCTCCTT  0.9136752 -0.1423067 -0.4513638  0.7052750 -0.04472755
                 Cluster58   Cluster59  Cluster60  Cluster61   Cluster62
1B_CCTTTCCTGTCT -1.1373113  0.00000000 -0.1785947 -0.2054421 -0.12378773
1B_GTCGAGTCATGC  0.7903234 -0.03594838 -0.2507875  1.7872109 -0.18304713
1B_GCTCGATAGATC  0.2474767 -0.03138141 -0.1879988 -0.3491627 -0.15960381
1B_GAGCGGAATGTC  0.4903356  1.07559582 -0.3143516  0.7502687 -0.09801599
1B_AGCACTAAATGA -1.2166026 -0.05193988 -0.2991006 -0.3409670 -0.15163427
1B_AAATACCTCCTT -1.0861159 -0.02236378 -0.1718559 -0.2390786 -0.14271087
                  Cluster63  Cluster64  Cluster65   Cluster66   Cluster67
1B_CCTTTCCTGTCT  0.81672117 -0.1817391 -0.2621110  0.18039554 -1.25034485
1B_GTCGAGTCATGC  1.43524033  1.1033488  1.4264031  2.21258660  1.96318860
1B_GCTCGATAGATC -1.08782855 -0.2090350 -0.3318330 -0.91974930 -0.28829958
1B_GAGCGGAATGTC -0.03249219 -0.4184202 -0.2221099  0.13565848 -0.98699721
1B_AGCACTAAATGA -0.11431323 -0.3430326 -0.2957895  0.82143600 -0.02969245
1B_AAATACCTCCTT -1.30370109 -0.2896807 -0.4075242 -0.03488195 -1.35790912
                 Cluster68   Cluster69   Cluster70 Cluster71  Cluster72
1B_CCTTTCCTGTCT -0.2681340 -0.03555451 -0.05267871 -1.215753 -0.1303665
1B_GTCGAGTCATGC  0.6744683 -0.04130343  0.88305300 -1.313267  2.2069444
1B_GCTCGATAGATC  0.6485413 -0.08874793 -0.15199479  0.823277 -0.3329982
1B_GAGCGGAATGTC -0.2453111 -0.05487734  1.48482841 -1.078429  1.7231933
1B_AGCACTAAATGA -0.2908231  1.50107044  1.79062705 -1.059206  2.2326691
1B_AAATACCTCCTT -0.3205915 -0.03354566 -0.12034709 -1.349252  1.7878411
                 Cluster73   Cluster74   Cluster75   Cluster76   Cluster77
1B_CCTTTCCTGTCT -0.2523878  1.33009093 -1.21787218 -0.11193623 -0.10535742
1B_GTCGAGTCATGC  1.5404264  0.48496090  0.48755674 -0.17018454  1.75693101
1B_GCTCGATAGATC -0.3734878  1.08436515 -0.02464803 -0.10773884 -0.14422802
1B_GAGCGGAATGTC -0.2731314 -0.01376322 -0.99466757 -0.10474360 -0.14528167
1B_AGCACTAAATGA -0.2508056 -1.17079332  0.04049850 -0.05193988 -0.16327582
1B_AAATACCTCCTT -0.3317546 -0.13631696 -1.16056259 -0.08945510 -0.04472755
                  Cluster78   Cluster79  Cluster80   Cluster81   Cluster82
1B_CCTTTCCTGTCT -0.02370301 -0.01185150 -0.1119362 -0.10535742 -0.05925752
1B_GTCGAGTCATGC  0.98095653 -0.05162929 -0.1908821  0.87031150  0.98095653
1B_GCTCGATAGATC -0.04747728  0.88967780 -0.1884165 -0.09980352 -0.09980352
1B_GAGCGGAATGTC -0.03292640 -0.05998346 -0.1562571 -0.16473620  0.97672156
1B_AGCACTAAATGA -0.05193988 -0.02077595 -0.2259942 -0.18225581 -0.11426773
1B_AAATACCTCCTT -0.07561954  0.00000000  0.9676773 -0.05590944 -0.06709133
                 Cluster83   Cluster84  Cluster85   Cluster86  Cluster87
1B_CCTTTCCTGTCT -1.1084673 -0.12906043  0.6270388 -0.07110902  0.6124116
1B_GTCGAGTCATGC  0.6994148 -0.21569028  1.7336853 -0.12352605  1.0639570
1B_GCTCGATAGATC -0.3055906 -0.09675074 -1.0324675 -0.09495456 -1.1153019
1B_GAGCGGAATGTC  0.7977828 -0.10975468  1.4463889 -0.08780374  1.6497459
1B_AGCACTAAATGA  1.0123225 -0.21909806  1.4454089 -0.08101112  1.6101716
1B_AAATACCTCCTT -1.2415499 -0.13329467  0.8322270 -0.10063699  1.3653924
                 Cluster88    Cluster89  Cluster90  Cluster91  Cluster92
1B_CCTTTCCTGTCT -1.0273151 -0.011851504 -0.1409119 -0.1066635 -0.1799008
1B_GTCGAGTCATGC  0.8146881 -0.046274239  0.8698062 -0.1135844 -0.1776662
1B_GCTCGATAGATC -1.0785523 -0.009495456 -0.1386578 -0.1168405 -0.1299511
1B_GAGCGGAATGTC  1.1966899 -0.021950935  0.9717105 -0.1419179 -0.1587627
1B_AGCACTAAATGA -1.0735243 -0.018683271 -0.2151749 -0.1017871 -0.1657590
1B_AAATACCTCCTT -1.0946856 -0.027510729 -0.1952390 -0.1565464 -0.1505115
                  Cluster93  Cluster94   Cluster95  Cluster96  Cluster97
1B_CCTTTCCTGTCT -0.02370301 -0.2602491 -0.07638172 -0.3161928 -0.2768411
1B_GTCGAGTCATGC -0.06195515  1.7903794  0.84834007 -0.5200312  0.6603221
1B_GCTCGATAGATC -0.06653568 -0.2598011 -0.18588331 -0.3497250 -0.1837700
1B_GAGCGGAATGTC -0.03803963  1.3665967 -0.24916700  1.8467934 -0.3982116
1B_AGCACTAAATGA -0.03116393  1.8657579 -0.22467074  1.6666701 -0.3214288
1B_AAATACCTCCTT -0.02236378  0.8842571  2.03757985  1.2962593 -0.1916974
                  Cluster98   Cluster99 Cluster100  Cluster101  Cluster102
1B_CCTTTCCTGTCT -0.01185150 -0.08823322          0 -0.08296053 -0.02360777
1B_GTCGAGTCATGC  1.00696330  0.82310180          0 -0.16482949 -0.09946586
1B_GCTCGATAGATC -0.09446986  0.76099413          0 -0.10773884 -1.17803126
1B_GAGCGGAATGTC -0.04900799 -0.16723260          0  0.95052278 -1.12727850
1B_AGCACTAAATGA -0.06232785  0.84685879          0 -0.09760171 -1.15480994
1B_AAATACCTCCTT -0.05325577 -0.20025465          0 -0.10063699  0.52139808
                Cluster103 Cluster104 Cluster105  Cluster106 Cluster107
1B_CCTTTCCTGTCT -0.3352159  1.0034112 -0.2048494 -0.01185150 -1.1166494
1B_GTCGAGTCATGC -0.5544440  0.7054361 -0.2251649 -0.05162929  0.4630843
1B_GCTCGATAGATC -0.3462002 -0.2588965 -0.2028284 -0.01899091  1.5239962
1B_GAGCGGAATGTC -0.4657253  1.2053406  0.8230650  1.08657129  0.9802060
1B_AGCACTAAATGA  1.0372588 -0.2993294  0.8111012 -0.01038798  2.1795529
1B_AAATACCTCCTT  1.1307912 -0.3049902  0.8654350 -0.02236378  1.1580833
                Cluster108  Cluster109 Cluster110  Cluster111 Cluster112
1B_CCTTTCCTGTCT -0.1356392 -0.02370301  1.1265450 -0.01185150 -0.1422180
1B_GTCGAGTCATGC -0.1747711 -0.01032586  1.5753559 -0.05162929 -0.1793317
1B_GCTCGATAGATC -0.1252371  0.00000000 -0.3764991 -0.03798182 -0.1140130
1B_GAGCGGAATGTC -0.2458032  0.00000000  1.3614005  1.07559582 -0.2078657
1B_AGCACTAAATGA  0.9353373  0.00000000  1.8433200  1.50605465 -0.1911287
1B_AAATACCTCCTT -0.0894551 -0.02236378  1.3860165 -0.02236378 -0.1675679
                 Cluster113 Cluster114 Cluster115 Cluster116  Cluster117
1B_CCTTTCCTGTCT -0.14091193  0.9235952 -0.2238725 -0.2168332 -0.08296053
1B_GTCGAGTCATGC  1.71475641  1.8398197  1.3284584  0.5823203  0.84915444
1B_GCTCGATAGATC -0.08842164  1.0385270 -0.1519948 -0.2090350 -0.09973603
1B_GAGCGGAATGTC  0.93291474  0.5923622 -0.2669909 -0.2618919 -0.11822461
1B_AGCACTAAATGA -0.20650084 -0.5321882 -0.2729386 -0.3493561 -0.09097954
1B_AAATACCTCCTT -0.19949802  0.6372349  0.9298726 -0.2344989 -0.12300076
                Cluster118  Cluster119 Cluster120  Cluster121 Cluster122
1B_CCTTTCCTGTCT -1.2076899 -0.07079162 -0.2232797 -0.01185150 -0.1172089
1B_GTCGAGTCATGC -1.0517050 -1.16952251 -0.6566440 -0.04627424  2.0832844
1B_GCTCGATAGATC  1.1457948 -0.32503174 -0.4552643 -0.02848637 -0.2099750
1B_GAGCGGAATGTC  0.1729526 -1.15319221  0.5379019 -0.03042087  1.2803265
1B_AGCACTAAATGA  0.4877517 -1.07727198 -0.4140552 -0.08310380  1.6063836
1B_AAATACCTCCTT -0.1686116 -0.30483206 -0.5359407 -0.01118189 -0.2055330
                Cluster123  Cluster124  Cluster125  Cluster126 Cluster127
1B_CCTTTCCTGTCT -0.4069647 -0.06453022 -0.11851504 -0.07110902 -0.2067483
1B_GTCGAGTCATGC  0.5184371 -0.06692596 -0.12888110  1.48339218 -0.3250211
1B_GCTCGATAGATC -0.3263140 -0.08074512 -0.07736605 -0.09024058 -0.3581957
1B_GAGCGGAATGTC -0.5266163 -0.10984973 -0.04900799  0.99876755  0.7964394
1B_AGCACTAAATGA -0.5333348 -0.09349178 -0.06232785  2.01817321 -0.1935984
1B_AAATACCTCCTT  0.6766841 -0.03354566 -0.13152898 -0.02751073 -0.2158269
                 Cluster128 Cluster129  Cluster130 Cluster131   Cluster132
1B_CCTTTCCTGTCT -0.01185150 -1.1065425  0.13435332 -0.2285524 -0.011851504
1B_GTCGAGTCATGC -0.02065172  0.4514533 -0.90143293 -0.5326558 -0.030977575
1B_GCTCGATAGATC -0.01899091  0.6326517 -1.02530694 -0.3266565 -0.009495456
1B_GAGCGGAATGTC -0.03803253 -1.2470763  0.08632263 -0.4966426 -0.010975468
1B_AGCACTAAATGA -0.01038798 -0.1481001 -0.96904150 -0.5291638 -0.010387975
1B_AAATACCTCCTT -0.03354566 -0.1361671  1.12277748 -0.4133988  0.000000000
                Cluster133 Cluster134 Cluster135 Cluster136  Cluster137
1B_CCTTTCCTGTCT -0.1422180 -0.3066390  0.9731294 -0.1514573 -0.05795141
1B_GTCGAGTCATGC -0.2879912 -0.3358558  0.6525514  1.2649020 -0.09336284
1B_GCTCGATAGATC  1.5349016 -0.3265071 -0.3725806 -0.2375214 -0.07124966
1B_GAGCGGAATGTC -0.2703618  0.7824319  1.3784375  0.7452576 -0.04900799
1B_AGCACTAAATGA -0.3679807 -0.2778290 -0.2600787  0.8414663 -0.09307222
1B_AAATACCTCCTT -0.2330437 -0.2605544 -0.2303900 -0.2717363 -0.11093090
                Cluster138  Cluster139  Cluster140  Cluster141  Cluster142
1B_CCTTTCCTGTCT -0.1000847 -0.09481203 -0.06453022 -0.08296053  0.00000000
1B_GTCGAGTCATGC -0.1677505 -0.19337299  0.86700627  1.77261192 -0.06195515
1B_GCTCGATAGATC -0.1757672  1.48281140 -0.05225875 -0.11401296 -0.01899091
1B_GAGCGGAATGTC -0.1621265 -0.20786572 -0.13170561 -0.09290986 -0.02195094
1B_AGCACTAAATGA -0.1894555 -0.23724522 -0.10136752  0.97646969  0.98685767
1B_AAATACCTCCTT -0.1546391 -0.16507464 -0.15903971 -0.04472755 -0.08680143
                 Cluster143 Cluster144  Cluster145 Cluster146  Cluster147
1B_CCTTTCCTGTCT  1.14959588 -0.1290604  0.00000000  0.9213464  0.00000000
1B_GTCGAGTCATGC -0.06692596  1.3925093 -0.01032586  0.9972166 -0.01032586
1B_GCTCGATAGATC -0.06497555 -0.1548898 -0.01899091 -0.3829489 -0.01899091
1B_GAGCGGAATGTC -0.09877921  1.4248378 -0.01097547  0.7730779 -0.02195094
1B_AGCACTAAATGA -0.12423615 -0.2279818 -0.01038798 -0.5018886 -0.02077595
1B_AAATACCTCCTT -0.03354566 -0.1771342 -0.02236378 -0.4220874 -0.04472755
                Cluster148  Cluster149 Cluster150  Cluster151  Cluster152
1B_CCTTTCCTGTCT          0 -0.09481203  0.9430272 -0.08823322 -0.12064330
1B_GTCGAGTCATGC          0 -0.16270444  0.5495315 -0.17481697 -0.09546946
1B_GCTCGATAGATC          0 -0.06175421  0.1512467 -0.22504063 -0.05076611
1B_GAGCGGAATGTC          0 -0.17484426 -0.1629187  0.90585769 -0.08193440
1B_AGCACTAAATGA          0 -0.15080641  0.4550047 -0.25598639 -0.05193988
1B_AAATACCTCCTT          0 -0.09371407  0.5476277 -0.18389676  1.09067804

FeaturePlot(hiv_down, features = &quot;Cluster1&quot;,
            label = TRUE, 
            repel = TRUE, pt.size = .1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;TFs downregulated in in vitro microglia&quot;)
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-60-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="features-identified-in-a-paper-on-convergence-of-microglia-and-macrophages-grassivaro" class="section level2">
<h2>Features identified in a paper on convergence of microglia and macrophages <span class="citation">(<a href="#ref-Grassivaro" role="doc-biblioref">Francesca Grassivaro and Martino 2018</a>)</span></h2>
<p>They identified 14 genes that are conserved in microglia throughout development, but not expressed in myeloid-derived monocytes MDM. Only in the context of neuroinflammation are the 14 genes expressed in macrophages.</p>
<pre class="r"><code>tfs &lt;- c(&quot;CRYBB1&quot;, &quot;GARNL3&quot;, &quot;GPR34&quot;, &quot;LAG3&quot;, &quot;NUAK1&quot;, &quot;OLFML3&quot;, &quot;RTN1&quot;,  &quot;SALL3&quot;,  &quot;SLC1A3&quot;, &quot;SPARC&quot;, &quot;TNFRSF17&quot;)

tfs %in% rownames(hiv4@assays$RNA@counts)
 [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE


hivt &lt;- AddModuleScore(hiv4,
                           features = tfs)

Idents(hivt) &lt;- &quot;cell_type&quot;
FeaturePlot(hivt, features = &quot;Cluster2&quot;,
            label = TRUE, 
            repel = TRUE, pt.size = .1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;14 genes expressed in microglia &amp; macrophages in neuroinfl.&quot;)
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-61-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="marker-genes-referred-to-by-farhadian" class="section level2">
<h2>Marker genes referred to by <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span></h2>
<p>References:</p>
<ul>
<li><span class="citation">(<a href="#ref-Keren-Shaul" role="doc-biblioref">Hadas Keren-Shaul 2017</a>)</span></li>
<li><span class="citation">(<a href="#ref-Kraseman" role="doc-biblioref">Susanne Krasemann 2017</a>)</span></li>
</ul>
<p>Neither of these signatures shows a high score for the Myeloid2 clusters.</p>
<pre class="r"><code># upregulated in disease associated microglia (Keren-Shaul)
disease &lt;- c(&quot;SPP1&quot;, &quot;ITGAX&quot;, &quot;AXL&quot;, &quot;LILRB4&quot;, &quot;CLEC7A&quot;, &quot;CCL2&quot;, &quot;CSF1&quot;) 

#28 inflammatory molecules upregulated in disease associated microglia (Krasemann)
mgnd &lt;- c(&quot;APOE&quot;, &quot;AXL&quot;, &quot;CCL2&quot;, &quot;TLR2&quot;, &quot;SPP1&quot;, &quot;CYBB&quot;, &quot;MSR1&quot;, &quot;ITGAX&quot;,
          &quot;CLEC7A&quot;, &quot;CHI3L3&quot;, &quot;ARG1&quot;, &quot;SIGLEC1&quot;, &quot;CFP&quot;, &quot;CXCL10&quot;, &quot;ALCAM&quot;, 
          &quot;FER1L3&quot;, &quot;LILRB4&quot;, &quot;GPX3&quot;, &quot;GAS7&quot;, &quot;CCRL2&quot;,&quot;CXCL16&quot;, &quot;CXCR4&quot;,
          &quot;GPNMB&quot;, &quot;LGALS3&quot;, &quot;IFI202B&quot;, &quot;CSF1&quot;, &quot;LIRB4&quot;, &quot;LAG3&quot;)
# only APOE, AXL and MSR1 overlap, but these are the most important genes

# remove the genes that are also used by the Farhadian Paper
mgnd &lt;- c(&quot;CCL2&quot;, &quot;TLR2&quot;, &quot;SPP1&quot;, &quot;CYBB&quot;,  &quot;ITGAX&quot;,
          &quot;CLEC7A&quot;, &quot;CHI3L3&quot;, &quot;ARG1&quot;, &quot;SIGLEC1&quot;, &quot;CFP&quot;, &quot;CXCL10&quot;, &quot;ALCAM&quot;, 
          &quot;FER1L3&quot;, &quot;LILRB4&quot;, &quot;GPX3&quot;, &quot;GAS7&quot;, &quot;CCRL2&quot;,&quot;CXCL16&quot;, &quot;CXCR4&quot;,
          &quot;GPNMB&quot;, &quot;LGALS3&quot;, &quot;IFI202B&quot;, &quot;CSF1&quot;, &quot;LIRB4&quot;, &quot;LAG3&quot;)

#hiv4 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/non_integrated_HIV1_HIV2_4samples_seurat_object.rds&quot;)

#get genes also present in the HIV samples
mgnd &lt;- mgnd[mgnd %in% rownames(hiv4@assays$RNA@counts)]


hiv_disease &lt;- AddModuleScore(hiv4,
                           features = disease)

FeaturePlot(hiv_disease, features = &quot;Cluster1&quot;, pt.size = .01) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Keren-Shaul et al., Genes upregulated in disease-associated microglia&quot;)
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-62-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>


hiv_mgnd &lt;- AddModuleScore(hiv4,
                           features = mgnd)

FeaturePlot(hiv_mgnd, features = &quot;Cluster1&quot;, pt.size = .01) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Kraseman et al., 28 inflammatory genes upregulated in 
       disease associated microglia&quot;)
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-62-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Neither of the signature genes are very specific for the Myeloid2/5 cluster.</p>
<ul>
<li>The signature genes defined by <span class="citation">(<a href="#ref-Keren-Shaul" role="doc-biblioref">Hadas Keren-Shaul 2017</a>)</span> are expressed to some level in all myeloid clusters</li>
<li>The signature defined by <span class="citation">(<a href="#ref-Kraseman" role="doc-biblioref">Susanne Krasemann 2017</a>)</span> is also mainly expressed in the Myeloid clusters</li>
<li>CD9 and SPP1 might be added to the list of microglia marker genes expressed in Myeloid2 cluster</li>
</ul>
<pre class="r"><code># Keren Shaul et al. signature
DotPlot(hiv4, features = disease, group.by = &quot;cell_type&quot;) + 
  coord_flip() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
    labs(title = &quot;Keren-Shaul et al., Genes upregulated in disease-associated microglia&quot;)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-63-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>
# Kraseman et al. signature
DotPlot(hiv4, features = mgnd, group.by = &quot;cell_type&quot;) + 
  coord_flip() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
    labs(title = &quot;Kraseman et al., 28 inflammatory genes upregulated in 
       disease associated microglia&quot;)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-63-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>

# some genes that might be markers for Myeloid2
DotPlot(hiv4, features = c(&quot;CD9&quot;, &quot;APOE&quot;, &quot;AXL&quot;, &quot;CTSL&quot;, &quot;SPP1&quot;), group.by = &quot;cell_type&quot;) + 
  coord_flip() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
  labs(title = &quot;Genes that might be used as markers for Myeloid2&quot;)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-63-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-signature-defined-by-esaulova" class="section level2">
<h2>Gene signature defined by <span class="citation">(<a href="#ref-Esaulova" role="doc-biblioref">Ekaterina Esaulova 2020</a>)</span></h2>
<div id="homeostatic-genes" class="section level5">
<h5>Homeostatic genes</h5>
<ul>
<li>the homeostatic microglia gene signature is found in all myeloid clusters, not only in the Myeloid 2 (except for SLC2A5 which was also included in the Farhadian paper)</li>
<li>NK cell cluster also exhibits high scores for this gene signature</li>
</ul>
<pre class="r"><code>#homeostatic microglia gene signature
microglia &lt;- c(&quot;CX3CR1&quot;,&quot;CSF1R&quot;, &quot;SLC2A5&quot;, &quot;MARCKS&quot;, &quot;P2RY13&quot;)
FeaturePlot(hiv4, features = microglia, ncol = 3)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-64-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DotPlot(hiv4, features = microglia, group.by = &quot;cell_type&quot;) + 
  coord_flip() +
    theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-65-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>
hivt &lt;- AddModuleScore(hiv4, features = microglia)
Idents(hivt) &lt;- &quot;cell_type&quot;
FeaturePlot(hivt, features = &quot;Cluster1&quot;, label =TRUE, repel = TRUE, pt.size = .7) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
    labs(title = &quot;Homeostatic microglia signature from Esaulova paper&quot;)
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-65-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="microglia-specific-genes" class="section level5">
<h5>Microglia-specific genes</h5>
<ul>
<li><strong>GPR34</strong>, <strong>TMEM119</strong> supports the hypothesis that myeloid2 are microglia-like, since they are expressed at higher levels in the myeloid2 cluster</li>
<li>they use 10 genes to discriminate microglia, four of which have not been inlcude in the Farhadian Paper: <strong>CLEC9A</strong>, <strong>TIMD4</strong>, <strong>TNFSF18</strong>, <strong>CXCL12</strong></li>
<li>the genes <strong>CLEC9A</strong>, <strong>TIMD4</strong>, <strong>TNFSF18</strong>, <strong>CXCL12</strong> are not included in the 60 genes used by Farhadian et al., but they do not show specific expression for myeloid2 cells</li>
</ul>
<pre class="r"><code># 10 genes that discriminate microglia
microglia &lt;- c(&quot;C1QC&quot;, &quot;C1QB&quot;, &quot;C1QA&quot;, &quot;TREM2&quot;, &quot;TIMD4&quot;, &quot;APOC1&quot;, 
               &quot;APOE&quot;, &quot;TNFSF18&quot;, &quot;CLEC9A&quot;, &quot;CXCL12&quot;)
#FeaturePlot(hiv4, features = microglia, ncol = 3)

#microglia %in% paper

Idents(hiv4) &lt;- &quot;cell_type&quot;

#  two genes whcih might support that myeloid2 cluster contains microglia-like cells
FeaturePlot(hiv4, features = c(&quot;GPR34&quot;, &quot;TMEM119&quot;), label = TRUE, pt.size = .7,
            repel = TRUE)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-66-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DotPlot(hiv4, features = c(&quot;TMEM119&quot;, &quot;GPR34&quot;), group.by = &quot;cell_type&quot;) + 
     coord_flip() +
     theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1))</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-66-2.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>
# these are not inlcuded in the Farhadian gene signature
FeaturePlot(hiv4, features = c(&quot;CLEC9A&quot;, &quot;TIMD4&quot;, &quot;TNFSF18&quot;, &quot;CXCL12&quot;),
            pt.size = .7, label = TRUE, repel = TRUE, ncol= 2)</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-66-3.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="microglia-in-human-brain" class="section level2">
<h2>Microglia in human brain</h2>
<p>Reference: Sankowski R. et al. 2019. Mapping microglia states in the human brain through the integration of high-dimensional techniques. Nature Neuroscience. Vol 22 <a href="https://www.nature.com/articles/s41593-019-0532-y#Sec29" class="uri">https://www.nature.com/articles/s41593-019-0532-y#Sec29</a></p>
<p>In the paper microglia cells were obtained through brain surgery. Microglia were defined by the core signature called “microglia signature” below. Sankowski et al. performed unsupervised clustering using Race ID3 which resulted in 9 major sublcusters of the microglia cells. The authors suggested that these subclusters represent the wide spectrum of transcriptional states in human microglia.</p>
<p><strong>RaceID:</strong> RaceID is a clustering algorithm for the identification of cell types from scRNA-seq data. It was designed to work particularly well for the detection of rare cells. <a href="https://github.com/dgrun/RaceID3_StemID2_package" class="uri">https://github.com/dgrun/RaceID3_StemID2_package</a></p>
<ul>
<li>The microglia signature is most pronounced in Myeloid2 cluster</li>
<li>The monocyte signature is less pronounced in Myeloid2 cluster and more pronounced in the other Myeloid clsuters</li>
<li>Homeostatic microglia (core signature genes expressed by all clusters in the paper) gene signature is found in all Myeloid cluster as well as NK cells, with high expression also in Myeloid2
<ul>
<li><strong>CSF1R</strong> and <strong>MARCKS</strong> are most highly expressed in Myeloid2</li>
</ul></li>
</ul>
<pre class="r"><code># microglia gene signature from supplementary table 3a
microglia &lt;- c(&quot;P2RY12&quot;, &quot;CX3CR1&quot;, &quot;CSF1R&quot;, &quot;TMEM119&quot;, &quot;SLC2A5&quot; )

# monocyte gene signature from supplementary table 3a 
monocyte &lt;- c(&quot;CCR2&quot;, &quot;CLEC12A&quot;, &quot;PLAC8&quot;, &quot;FCN1&quot;, &quot;S100A9&quot;)

# homeostatic microlgia gene siganture
homeostatic &lt;- c(&quot;CX3CR1&quot;, &quot;TMEM119&quot;, &quot;CSF1R&quot;, &quot;P2RY12&quot;,
                 &quot;P2RY13&quot;,&quot;SELPLG&quot;, &quot;MARCKS&quot;)


map(seq.int(1:3), function(n){
  names &lt;- c(&quot;microglia&quot;, &quot;monocyte&quot;, &quot;homeostatic&quot;)
  feature_list &lt;- list(microglia, monocyte, homeostatic)
  features = feature_list[[n]]
  hiv &lt;- AddModuleScore(hiv4, features = features)
  # feature plot
  Idents(hiv) &lt;- &quot;cell_type&quot;
  FeaturePlot(hiv, features = &quot;Cluster1&quot;, pt.size = .5, label = TRUE, repel=TRUE)+
    scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
    labs(title = paste0(names[n],&quot; microglia gene signature&quot;))
})
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
[[1]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-67-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-67-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-67-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  
map(seq.int(1:3), function(n){
  names &lt;- c(&quot;microglia&quot;, &quot;monocyte&quot;, &quot;homeostatic&quot;)
  feature_list &lt;- list(microglia, monocyte, homeostatic)
  features = feature_list[[n]] 
  # dot plot
  DotPlot(hiv4, features = features, group.by = &quot;cell_type&quot;) +
    coord_flip() +
        theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
  labs(title = paste0(names[n],&quot; gene signature&quot;))
})
[[1]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-67-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-67-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-67-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>Check the clusters across the transcriptional spectrum of microglia:</p>
<p>Sankowski R. et al., Nature Neuroscience, 2019 defined 9 subclusters within the microglia cell population. Can we find these gene signatures in the CSF-derived Myeloid cells? This could further confirm that Myeloid2 and Myeloid5 are microglia-like cells.</p>
<ul>
<li><strong>homeostatic clusters</strong>
<ul>
<li>C2: strong expresssion of MCH-II and antiviral immunity genes (HLA-DRA, CD74, IFI44L)</li>
<li>C3: high expression of microglial core genes CX3CR1, TMEM119<br />
</li>
</ul></li>
<li><strong>antigen processing and peptide antigen presentation</strong> C6 and C7 were characterized by low expression of CX3CR1 and high expression of integrin-receptor binding protein and metabolism genes (SPP1, APOE, LPL)</li>
<li>** proinflammatory clusters:** C1, C5, C8, C9 were characterized by expression of chemokine and cytokine genes (CCL2, IL1B)</li>
</ul>
<p>Since Cluster C2 was characterized as a homeostati microglia cluster and showed expression of MHC-II and antivrial immunity genes like IFI44L, HLA-DRA, CD74, this supports the microglia-like identity of Myeloid2 and Myeloid5. However, there is also some expression of this signature in Meloid1 and Myeloid3 which mainly originate from Blood samples and also in B cells.</p>
<p>There is also a high score for C5 and C8 signature genes, which are both proinflammatory gene signatures.</p>
<pre class="r"><code># get differentially expressed genes per cluster across the transcriptional 
# spectrum of microglia 
excel_sheets(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/2019_Sankowski_Mapping_microgliastates_in_the_human_brain_Supplementary_tables.xlsx&quot;)
 [1] &quot;Suppl. table 1&quot;  &quot;Suppl. table 2&quot;  &quot;Suppl. table 3&quot;  &quot;Suppl. table 4&quot; 
 [5] &quot;Suppl. table 5&quot;  &quot;Suppl. table 6&quot;  &quot;Suppl. table 7&quot;  &quot;Suppl. table 8&quot; 
 [9] &quot;Suppl. table 9&quot;  &quot;Suppl. table 10&quot; &quot;Suppl. table 11&quot; &quot;Suppl. table 12&quot;
cluster_genes &lt;- read_excel(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/2019_Sankowski_Mapping_microgliastates_in_the_human_brain_Supplementary_tables.xlsx&quot;, sheet = &quot;Suppl. table 3&quot;)
#names(microglia_sign_df) [1] &lt;- &quot;gene_name&quot;
#microglia_sign &lt;- microglia_sign_df[,&quot;gene_name&quot;]

# C3

cluster_list &lt;- map(seq.int(1:8), function(n) {
  names &lt;- c(&quot;c1&quot;, &quot;c2&quot;, &quot;c3&quot;, &quot;c5&quot;, &quot;c6&quot;, &quot;c7&quot;, &quot;c8&quot;, &quot;c9&quot;)
  name &lt;- names[n]
  cluster &lt;- c(1,2,3,5,6,7,8,9)
  cluster_number &lt;- cluster[n]
  c &lt;- pull(cluster_genes %&gt;% filter(Cluster == cluster_number), GENEID)
  c &lt;- c[c %in% rownames(hiv4@assays$RNA@counts)]
  cluster_list &lt;- list(name = name, cluster = c)
})


map(seq.int(1:8), function(n){
  c &lt;- cluster_list[[n]]$cluster
  name &lt;- cluster_list[[n]]$name
  hivc &lt;- AddModuleScore(hiv4, features = c)
  Idents(hivc) &lt;- &quot;cell_type&quot;
  FeaturePlot(hivc, features = &quot;Cluster1&quot;, pt.size = .5, label = TRUE,
              repel = TRUE) +
      scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
    labs(title = paste0(&quot;Differentially expressed genes of cluster &quot;, name))
})
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
[[1]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-68-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-68-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[3]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-68-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[4]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-68-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[5]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-68-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[6]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-68-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[7]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-68-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[8]]</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-68-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>


#map(seq.int(1:8), function(n){
 # c &lt;- cluster_list[[n]]$cluster
  #name &lt;- cluster_list[[n]]$name
  #DotPlot(hiv4, features = c, group.by = &quot;cell_type&quot;) +
  #coord_flip() +
  #      theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust=1)) +
  #labs(title = paste0(&quot;Differentially expressed genes of cluster &quot;, name))
#})</code></pre>
</div>
<div id="look-for-genes-which-are-upregulated-in-myeloid2-compared-to-all-other-myeloid" class="section level2">
<h2>Look for genes which are upregulated in Myeloid2 compared to all other Myeloid</h2>
<p>clusters</p>
<ul>
<li><p>overlap of 5 genes with Kraseman microglia -&gt; inflammatory molecules upregulated in disease associated microglia (Krasemann)</p></li>
<li><p>overlap of 3 genes with disease-associated microglia (Keren-Shaul)</p></li>
<li><p>Sankowski signature</p>
<ul>
<li>4 genes overlap with microglia signature</li>
<li><strong>zero overlap with monocyte gene signature</strong></li>
<li>4 genes overlap with homeostatic microglia signature</li>
</ul></li>
<li><p>Esaulova paper -&gt; 8 of 10 genes that discriminate microglia are found upregulated in Myeloid2 cluster</p></li>
<li><p>Glass paper -&gt; 132 of the microglia signature are found in the genes upregulated in Myeloid2, 28 of these are also included in the Farhadian Paper signature, 104 are new marekr genes</p></li>
</ul>
<pre class="r"><code>Idents(hiv4) &lt;- &quot;cell_type&quot;
# Find Markers that distinguish My2 from the other Myeloid clusters
my2_markers &lt;- FindMarkers(hiv4, ident.1  = &quot;Myeloid2&quot;,
            ident.2  = c(&quot;Myeloid1&quot;, &quot;Myeloid3&quot;, &quot;Myeloid4&quot;, &quot;Myeloid5&quot;))
my2_markers &lt;- my2_markers %&gt;% rownames_to_column(var = &quot;gene&quot;)
# Can we find mgnd genes? (Krasemann)
my2_markers %&gt;% filter(gene %in% mgnd &amp; avg_log2FC &gt; 0)
    gene        p_val avg_log2FC pct.1 pct.2    p_val_adj
1   SPP1 3.626442e-14  2.2989596 0.103 0.002 9.018961e-10
2 CXCL16 8.438072e-11  1.0379235 0.439 0.172 2.098549e-06
3   GPX3 3.358479e-10  0.4856146 0.112 0.010 8.352537e-06
4   CYBB 1.267019e-09  0.9216583 0.860 0.728 3.151077e-05
5 LILRB4 4.108097e-08  0.6611217 0.271 0.089 1.021684e-03
# genes in disease-assocaited microglia (Keren-Shaul)
my2_markers %&gt;% filter(gene %in% disease &amp; avg_log2FC &gt; 0)
    gene        p_val avg_log2FC pct.1 pct.2    p_val_adj
1    AXL 7.682308e-19  1.0183192 0.252 0.029 1.910590e-14
2   SPP1 3.626442e-14  2.2989596 0.103 0.002 9.018961e-10
3 LILRB4 4.108097e-08  0.6611217 0.271 0.089 1.021684e-03

# Sankowski
# microglia gene signature
microglia &lt;- c(&quot;P2RY12&quot;, &quot;CX3CR1&quot;, &quot;CSF1R&quot;, &quot;TMEM119&quot;, &quot;SLC2A5&quot; )

# monocyte gene signature
monocyte &lt;- c(&quot;CCR2&quot;, &quot;CLEC12A&quot;, &quot;PLAC8&quot;, &quot;FCN1&quot;, &quot;S100A9&quot;)

# homeostatic microlgia gene siganture
homeostatic &lt;- c(&quot;CX3CR1&quot;, &quot;TMEM119&quot;, &quot;CSF1R&quot;, &quot;P2RY12&quot;,
                 &quot;P2RY13&quot;,&quot;SELPLG&quot;, &quot;MARCKS&quot;)


my2_markers %&gt;% filter(gene %in% microglia &amp; avg_log2FC &gt; 0)
    gene        p_val avg_log2FC pct.1 pct.2    p_val_adj
1 SLC2A5 1.038249e-50  1.3177001 0.346 0.000 2.582126e-46
2 P2RY12 1.971868e-11  0.5465335 0.140 0.015 4.904036e-07
3  CSF1R 2.060436e-11  1.2036520 0.664 0.407 5.124303e-07
4 CX3CR1 1.359952e-04  0.5885714 0.439 0.271 1.000000e+00
my2_markers %&gt;% filter(gene %in% monocyte &amp; avg_log2FC &gt; 0)
[1] gene       p_val      avg_log2FC pct.1      pct.2      p_val_adj 
&lt;0 rows&gt; (or 0-length row.names)
my2_markers %&gt;% filter(gene %in% homeostatic &amp; avg_log2FC &gt; 0)
    gene        p_val avg_log2FC pct.1 pct.2    p_val_adj
1 MARCKS 3.672024e-12  1.1346931 0.664 0.363 9.132325e-08
2 P2RY12 1.971868e-11  0.5465335 0.140 0.015 4.904036e-07
3  CSF1R 2.060436e-11  1.2036520 0.664 0.407 5.124303e-07
4 CX3CR1 1.359952e-04  0.5885714 0.439 0.271 1.000000e+00


# Esaulova
# 10 genes that discriminate microglia
microglia &lt;- c(&quot;C1QC&quot;, &quot;C1QB&quot;, &quot;C1QA&quot;, &quot;TREM2&quot;, &quot;TIMD4&quot;, &quot;APOC1&quot;, 
               &quot;APOE&quot;, &quot;TNFSF18&quot;, &quot;CLEC9A&quot;, &quot;CXCL12&quot;)
# Can we find these 10 genes in the Myeloid2 upregulated genes?
my2_markers %&gt;% filter(gene %in% microglia &amp; avg_log2FC &gt; 0)
    gene         p_val avg_log2FC pct.1 pct.2     p_val_adj
1   C1QC 1.891156e-120  4.1160497 0.794 0.002 4.703305e-116
2   C1QB 8.023728e-114  4.1322588 0.860 0.026 1.995501e-109
3   APOE 4.641643e-108  3.5767082 0.720 0.002 1.154377e-103
4   C1QA  3.455807e-92  3.1390482 0.785 0.039  8.594593e-88
5  APOC1  2.142095e-53  2.0205009 0.402 0.006  5.327391e-49
6  TREM2  1.158412e-51  1.5152244 0.374 0.003  2.880971e-47
7  TIMD4  5.003322e-18  0.5471981 0.131 0.002  1.244326e-13
8 CLEC9A  1.921500e-06  0.4152332 0.112 0.021  4.778770e-02


# Glass
# 84 of the microglia signature genes are more highly expressed in the 
microglia_genes &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/Microglia_gene_signatur_found_in_HIV1_HIV2_samples&quot;)
m1 &lt;- my2_markers %&gt;% filter(gene %in% microglia_genes [[&quot;gene_name&quot;]] &amp; avg_log2FC &gt;0)
# remove the genes that were already used by the Farhadian paper
m1 &lt;- m1[[&quot;gene&quot;]][m1[[&quot;gene&quot;]] %in% paper == FALSE]

hiv_new &lt;- AddModuleScore(hiv4, features = m1)
FeaturePlot(hiv_new, features = &quot;Cluster1&quot;) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
    labs(title = &quot;Selected signature genes from the Glass paper&quot;)
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<p><img src="figure/HIV_CSF_dataset_analysis.Rmd/unnamed-chunk-69-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
</div>
</div>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Gosselin" class="csl-entry">
David Gosselin, Nicole G. Coufal, Dylan Skola. 2020. <em>Single-Cell RNA Sequencing Reveals Microglia-Like Cells in Cerebrospinal Fluid During Virologically Suppressed HIV</em>. <em>JNEUROSCI</em>. Vol. 356, 1248. <a href="https://DOI: 10.1126/science.aal3222">DOI: 10.1126/science.aal3222</a>.
</div>
<div id="ref-Esaulova" class="csl-entry">
Ekaterina Esaulova, Irina Shchukina, Claudia Cantoni. 2020. <em>Single-Cell RNA-Seq Analysis of Human CSF Microglia and Myeloid Cells in Neuroinflammation</em>. <em>Nat Neurosci</em>. Vol. 7:e732. <a href="https://doi:10.1212/NXI.0000000000000732">doi:10.1212/NXI.0000000000000732</a>.
</div>
<div id="ref-Grassivaro" class="csl-entry">
Francesca Grassivaro, Massimo Acquaviva, Ramesh Menon, and Gianvito Martino. 2018. <em>Convergence Between Microglia and Peripheral Macrophages Phenotype During Development and Neuroinflammation</em>. <em>JCI Insight</em>. Vol. 40 (4). <a href="https://doi.org/10.1523/JNEUROSCI.1523-19.2019">https://doi.org/10.1523/JNEUROSCI.1523-19.2019</a>.
</div>
<div id="ref-Keren-Shaul" class="csl-entry">
Hadas Keren-Shaul, Assaf Weiner, Amit Spinrad. 2017. <em>A Unique Microglia Type Associated with Restricting Development of Alzheimer’s Disease</em>. <em>Cell</em>. Vol. 169. <a href="http://dx.doi.org/10.1016/j.cell.2017.05.018">http://dx.doi.org/10.1016/j.cell.2017.05.018</a>.
</div>
<div id="ref-Farhadian" class="csl-entry">
Shelli F. Farhadian, Chrysoula Zografou, Sameet S. Mehta, and Serena S. Spudich. 2018. <em>Single-Cell RNA Sequencing Reveals Microglia-Like Cells in Cerebrospinal Fluid During Virologically Suppressed HIV</em>. <em>JCI Insight</em>. Vol. 3(18):e121718. <a href="https://doi.org/10.1172/jci.insight.121718">https://doi.org/10.1172/jci.insight.121718</a>.
</div>
<div id="ref-Kraseman" class="csl-entry">
Susanne Krasemann, Ron Cialic, Charlotte Madore. 2017. <em>The Trem2-APOE Pathway Drives the Transcriptional Phenotype of Dysfunctional Microglia in Neurodegenerative Diseases</em>. <em>Immunity</em>. Vol. 47. <a href="http://dx.doi.org/10.1016/j.immuni.2017.08.008">http://dx.doi.org/10.1016/j.immuni.2017.08.008</a>.
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] patchwork_1.1.1       SeuratDisk_0.0.0.9019 harmony_0.1.0        
 [4] Rcpp_1.0.7            readxl_1.3.1          multtest_2.46.0      
 [7] Biobase_2.50.0        BiocGenerics_0.36.1   metap_1.6            
[10] knitr_1.30            rstatix_0.6.0         corrplot_0.92        
[13] gridExtra_2.3         ggpubr_0.4.0          circlize_0.4.13      
[16] ComplexHeatmap_2.9.4  ggcorrplot_0.1.3      pheatmap_1.0.12      
[19] RColorBrewer_1.1-2    ggrepel_0.9.1         data.table_1.14.2    
[22] Matrix_1.3-4          edgeR_3.32.0          limma_3.46.0         
[25] SeuratObject_4.0.3    Seurat_4.0.5          forcats_0.5.0        
[28] stringr_1.4.0         dplyr_1.0.7           purrr_0.3.4          
[31] readr_1.4.0           tidyr_1.1.2           tibble_3.1.6         
[34] ggplot2_3.3.5         tidyverse_1.3.0       workflowr_1.6.2      

loaded via a namespace (and not attached):
  [1] utf8_1.2.2            reticulate_1.22       tidyselect_1.1.1     
  [4] htmlwidgets_1.5.4     Rtsne_0.15            munsell_0.5.0        
  [7] mutoss_0.1-12         codetools_0.2-18      ica_1.0-2            
 [10] future_1.23.0         miniUI_0.1.1.1        withr_2.4.3          
 [13] colorspace_2.0-2      highr_0.8             rstudioapi_0.13      
 [16] stats4_4.0.2          ROCR_1.0-11           ggsignif_0.6.0       
 [19] tensor_1.5            listenv_0.8.0         labeling_0.4.2       
 [22] Rdpack_2.1.2          git2r_0.27.1          mnormt_2.0.2         
 [25] polyclip_1.10-0       farver_2.1.0          bit64_4.0.5          
 [28] rprojroot_2.0.2       TH.data_1.0-10        parallelly_1.29.0    
 [31] vctrs_0.3.8           generics_0.1.1        xfun_0.19            
 [34] R6_2.5.1              doParallel_1.0.16     clue_0.3-60          
 [37] locfit_1.5-9.4        hdf5r_1.3.3           spatstat.utils_2.2-0 
 [40] assertthat_0.2.1      promises_1.2.0.1      scales_1.1.1         
 [43] multcomp_1.4-15       gtable_0.3.0          globals_0.14.0       
 [46] goftest_1.2-3         sandwich_3.0-0        rlang_0.4.12         
 [49] GlobalOptions_0.1.2   splines_4.0.2         lazyeval_0.2.2       
 [52] spatstat.geom_2.3-0   broom_0.7.2           yaml_2.2.1           
 [55] reshape2_1.4.4        abind_1.4-5           modelr_0.1.8         
 [58] backports_1.2.0       httpuv_1.6.3          tools_4.0.2          
 [61] ellipsis_0.3.2        spatstat.core_2.3-1   ggridges_0.5.2       
 [64] TFisher_0.2.0         plyr_1.8.6            rpart_4.1-15         
 [67] deldir_1.0-6          pbapply_1.5-0         GetoptLong_1.0.5     
 [70] cowplot_1.1.1         S4Vectors_0.28.1      zoo_1.8-9            
 [73] haven_2.3.1           cluster_2.1.0         fs_1.5.1             
 [76] magrittr_2.0.1        RSpectra_0.16-0       scattermore_0.7      
 [79] openxlsx_4.2.3        lmtest_0.9-39         reprex_0.3.0         
 [82] RANN_2.6.1            tmvnsim_1.0-2         mvtnorm_1.1-1        
 [85] whisker_0.4           fitdistrplus_1.1-6    matrixStats_0.61.0   
 [88] hms_1.1.1             mime_0.12             evaluate_0.14        
 [91] xtable_1.8-4          rio_0.5.16            IRanges_2.24.1       
 [94] shape_1.4.6           compiler_4.0.2        KernSmooth_2.23-18   
 [97] crayon_1.4.2          htmltools_0.5.2       mgcv_1.8-33          
[100] later_1.3.0           lubridate_1.7.9.2     DBI_1.1.1            
[103] dbplyr_2.0.0          MASS_7.3-53           car_3.0-10           
[106] cli_3.1.0             rbibutils_2.2.4       igraph_1.2.9         
[109] pkgconfig_2.0.3       sn_2.0.1              numDeriv_2016.8-1.1  
[112] foreign_0.8-80        plotly_4.9.2.1        spatstat.sparse_2.0-0
[115] xml2_1.3.2            foreach_1.5.1         rvest_0.3.6          
[118] digest_0.6.29         sctransform_0.3.2     RcppAnnoy_0.0.19     
[121] spatstat.data_2.1-0   rmarkdown_2.6         cellranger_1.1.0     
[124] leiden_0.3.5          uwot_0.1.10           curl_4.3.2           
[127] shiny_1.7.1           rjson_0.2.20          lifecycle_1.0.1      
[130] nlme_3.1-150          jsonlite_1.7.2        carData_3.0-4        
[133] viridisLite_0.4.0     fansi_0.5.0           pillar_1.6.4         
[136] lattice_0.20-41       plotrix_3.8-2         fastmap_1.1.0        
[139] httr_1.4.2            survival_3.2-7        glue_1.5.1           
[142] zip_2.1.1             png_0.1-7             iterators_1.0.13     
[145] bit_4.0.4             stringi_1.7.6         mathjaxr_1.4-0       
[148] irlba_2.3.3           future.apply_1.8.1   </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
