<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Katharina Mikulik" />


<title>Methods</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Abstract</a>
</li>
<li>
  <a href="Introduction.html">Introduction</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="HIV_CSF_dataset_analysis.html">1. CSF/Blood samples</a>
    </li>
    <li>
      <a href="TF_activity_analysis_of_myeloid2_cluster.html">2. TF activity Myeloid2</a>
    </li>
    <li>
      <a href="Triculture_csf_integration.html">3. Integration CSF/Blood &amp; triculture</a>
    </li>
    <li>
      <a href="Triculture_dataset_GRN_with_Scenic.html">4. TF activity iMg</a>
    </li>
  </ul>
</li>
<li>
  <a href="Discussion.html">Discussion</a>
</li>
<li>
  <a href="Methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Katharina782/report">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Methods</h1>
<h4 class="author">Katharina Mikulik</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-31
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 1 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttree3d3c73318704913e8bd8fd6101547ea25899045ftargetblank3d3c733a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Katharina782/report/tree/3d3c73318704913e8bd8fd6101547ea25899045f" target="_blank">3d3c733</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttree3d3c73318704913e8bd8fd6101547ea25899045ftargetblank3d3c733a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Katharina782/report/tree/3d3c73318704913e8bd8fd6101547ea25899045f" target="_blank">3d3c733</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/Independent_cell_type_identification_Farhadian_dataset_cache/
    Ignored:    analysis/Triculture_csf_integration_cache/

Untracked files:
    Untracked:  README.html
    Untracked:  analysis/Triculture_csf_integration/
    Untracked:  data/triculture_seurat_object
    Untracked:  rebuild file.Rmd
    Untracked:  rebuild-file.Rmd
    Untracked:  rebuild-file.html
    Untracked:  rebuild-file_cache/
    Untracked:  rebuild-file_files/

Unstaged changes:
    Modified:   README.md
    Modified:   analysis/Methods.Rmd
    Modified:   analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Methods.Rmd</code>) and HTML (<code>docs/Methods.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/Methods.html" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/823b10f4420e0f95577043b0080d3c2f7c15f7b5/docs/Methods.html" target="_blank">823b10f</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/ed143813aba0b4b6f39ff555599d585c28ac9b6d/analysis/Methods.Rmd" target="_blank">ed14381</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
<td>
wflow_publish(“analysis/Methods.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/6c5f3b175db2cc63caa1ba67e59c08e1a22f412b/docs/Methods.html" target="_blank">6c5f3b1</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/d2f59b2d133fb7c2394b97182e5052ef1e4aadd5/analysis/Methods.Rmd" target="_blank">d2f59b2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-27
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/d85a63137b84a09c19647281ffe67edbcce2e1eb/docs/Methods.html" target="_blank">d85a631</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/fdf25ce5decf9dff024fb0470f591b4a70f6aee6/analysis/Methods.Rmd" target="_blank">fdf25ce</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-22
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/9597c36bc5a6df5f071249d97949566b2c7db870/docs/Methods.html" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/ea7e899746b03ca612d93ad003c875fc101d2329/analysis/Methods.Rmd" target="_blank">ea7e899</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/Methods.html" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/Methods.html" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/00b53d3e1f3372da3c0f2981b43c28382b08aed2/docs/Methods.html" target="_blank">00b53d3</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/19c74e54e16a026fa484d9bb51a97275ae4b12d8/docs/Methods.html" target="_blank">19c74e5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/e87fa963caeed78821ac7f482359072ce58b8529/analysis/Methods.Rmd" target="_blank">e87fa96</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
new files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="quality-control" class="section level2">
<h2>Quality Control</h2>
<p>For Quality Control I used the publications from which the datasets were obtained as an orientation, in order to be able to reproduce the results and findings. For the dataset from <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> genes expressed in less than 3 cells were removed. Additionally, the ten percent of cells with the highest percentage of mitochondrial genes and cells with a number of features below 500 or above 2500 was removed. For the dataset from <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span> cells with a percentage of mitochondrial genes above 20% were removed, as well as, cells with a number of features below 200 and above 5000.</p>
</div>
<div id="normalization" class="section level2">
<h2>Normalization</h2>
<p>Log normalization was used to normalize count matrices, using the function <code>NormalizeData()</code>from the Seurat package. The counts of a particular feature for a particular cell are divided by the total counts for that cell and multiplied by a scaling factor of 10,000. Then the result is natural-log transformed. For GRN inference with SCENIC the function <code>SCTransform()</code> from the Seurat package was used, so that the data structure would work with the SCENIC pipeline. This method uses regularized negative binomial regression to normalize the count matrix.</p>
</div>
<div id="clustering" class="section level2">
<h2>Clustering</h2>
<p>After calculating a PCA the cells were clustered based on their nearest neighbors. For clustering the functions <code>FindNeigbors()</code> and <code>FindClusters()</code> from the Seurat package were used. First, a k nearest neighbor (KNN) graph is built based on the PCA space. Then, the Louvain algorithm is used, which iteratively optimizes the standard modularity function. Modularity measures the strength of division of a network into modules. High modularity is equivalent to dense connections within modules and sparse connections between nodes in different modules.</p>
</div>
<div id="integration-label-transfer" class="section level2">
<h2>Integration &amp; Label Transfer</h2>
<p>See here for more details on the method and implementation: <span class="citation">(<a href="#ref-Stuart" role="doc-biblioref">Stuart T 2019</a>)</span></p>
<p>See here for a vignette: <a href="https://satijalab.org/seurat/articles/integration_mapping.html" class="uri">https://satijalab.org/seurat/articles/integration_mapping.html</a></p>
<div id="integration" class="section level3">
<h3>Integration</h3>
<p>First, canonical correlation analysis is performed to reduce dimensionality of both datasets. This method can find shared biological markers, even if batch effects are very large. Second, expression data is L2 normalized. After the projection of distinct datasets into a shared subspace, pairs of MNN in this shared subspace are identified. These MNN correspond to cells which are biologically similar. The MNN serve as integration anchors for the integration, but beforehand the anchors are scored. The score is a measure of how similar the two anchor cells are. Using anchors and their corresponding scores, correction vectors are computed for each cell. The correction vectors contain information on the difference between two anchor cells. Using the correction vectors the expression values of each cell are transformed, so that they are part of an integrated dataset <span class="citation">(<a href="#ref-Stuart" role="doc-biblioref">Stuart T 2019</a>)</span>. This integrated dataset can then be used for further analysis. Finding anchors between datasets is not only required for integration of two distinct datasets, but also for label transfer.</p>
</div>
<div id="label-transfer" class="section level3">
<h3>Label Transfer</h3>
<p>After finding anchors between distinct datasets as described above, it is also possible to transfer information, for example cell type annotations. This means that instead of de-nove analysis of cell type markers the cell labels are learned from a reference dataset. Instead of using canonical correlation analysis, the PCA computed for the reference dataset is projected onto the query dataset. A binary classification matrix is build with rows corresponding to all possible classes and columns corresponding to reference anchors of the anchor pairs. If a reference anchor belongs to a certain class the entry will be one, otherwise it will be zero. The classification matrix is then multiplied with a weight matrix. This way we get prediction scores for each class and cell in the query dataset <span class="citation">(<a href="#ref-Stuart" role="doc-biblioref">Stuart T 2019</a>)</span>.</p>
</div>
</div>
<div id="correlation" class="section level2">
<h2>Correlation</h2>
<p>Since the TF activity is not normally distributed, instead of using the Pearson correlation coefficient which assumes normally distributed data, the Spearman correlation coefficient was used, which is rank-based and non-parametric. The Spearman correlation is a measure of strength and direction of a monotonic association. The null hypothesis of the Spearman correlation is that there is no association between the two variables. Statistical significance does not give any information on the strength of correlation.</p>
<p>Spearman’s rank order correlation:</p>
<p><span class="math inline">\(\rho = \frac{\sum_{i} (x_{i} - \overline{x}) (y_{i} - \overline{y})} {\sqrt{\sum_{i}(x_{i} - \overline{x})^2\sum_{i}(y_{i} - \overline{y})^2}}\)</span></p>
</div>
<div id="gene-signature-scores" class="section level2">
<h2>Gene signature scores</h2>
<p>The function <code>AddModuleScore()</code> from the Seurat package calculates gene expression scores for a particular set of signature genes. For example, I calculated scores for the genes defining the infected iMg from the triculture dataset in Myeloid2. Calculating the gene signature score for each cell provides information on whether a particular cell shows a higher/unchanged/lower expression of a particular set of genes (signature for a particular cell tpye).</p>
<p>First, the average expression of each gene across all cells is calculated and the genes are ordered according to their average expression values. Next, genes are sorted into bins of equal size. For each gene of the signature gene set, 100 genes are drawn from the same expression bin as a control group. This way, it is ensured that the control genes have a similar distribution of expression levels as the signature gene. Finally, the average expression of control genes and signature genes is computed in each cell. The resulting average control gene expression is subtracted from the average signature gene expression in each cell. This yields a score in each cell, which, if close to zero, can be interpreted as no enrichment or depletion of the signature genes. If the score in a particular cell is bigger than zero this means that there is an enrichment for the signature genes in this particular cell. Conversely, if the score is smaller than zero this means that there is a depletion. <span class="citation">(<a href="#ref-Tirosh" role="doc-biblioref">Tirosh I 2016</a>; <a href="#ref-Muskovic" role="doc-biblioref">Muskovic, n.d.</a>)</span></p>
</div>
<div id="k-means-clustering" class="section level2">
<h2>K-means clustering</h2>
<p>The algorithm starts with k randomly selected centroids. All data points are assigned to one of the datapoints depending on the distance between them. The process of defining centroids, assigning each point to one centroid and updating the centroids is repeated until the clusters stabilize or a maximum number of iterations has been reached.</p>
<div id="determine-k-with-the-elbow-method" class="section level3">
<h3>Determine k with the elbow method</h3>
<p>To determine the optimal k for k-means clustering I computed the within sum of square (WSS) distance. More precisely, I computed the sum of all pairwise distances between points in the same cluster. If a cluster would only contain one point, then WSS would be zero. With increasing k the WSS decreases. The elbow/kink in the plot shows at which k adding additional clusters does not improve the WSS anymore.</p>
</div>
</div>
<div id="test-for-differential-tf-activity" class="section level2">
<h2>Test for differential TF activity</h2>
<p>To determine whether there is a significant difference between the clusters computed with k-means clustering, a Kruskal Wallis Test was performed for all TFs between the different clusters. In contrast to ANOVA, the Kruskal Wallis Test is non-paramteric and, therefore, better suited for the data at hand, which is not normally distributed. Using the FDR method I corrected for multiple hypothesis testing. For TFs with different activity between clusters I then did a posthoc analysis to determine which clusters are different exactly. A pairwise wilcox test was used, which again, is a non-parametric alternative to the t-test.</p>
</div>
<div id="differential-gene-expression" class="section level2">
<h2>Differential gene expression</h2>
<p>The Seurat functions <code>FindAllMarkers()</code> and <code>FindMarkers()</code> uses a Wilcoxon Rank Sum test to identify differentially expressed genes between two groups. For the dataset from <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> to find also genes with small log-fold changes, I set the threshold to 0.01 and the minimum fraction of cells in which the gene is detected to 0.01. This increased the computation time, but made it possible to also include very infrequently expressed genes, which might for example be the case for Myeloid2-specific genes. In contrast, for the datset from <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span> I used the default settings with a minimum log-fold change of 0.25 and a minimum fraction of cells which express the gene of 0.1. For the minimum number of cells expressing the feature in at least one of the two compared groups I used the default of 3. Similarly, for the minimum number of cells in a group I used the default of 3.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Muskovic" class="csl-entry">
Muskovic, Walter. n.d. <em>Seurat’s AddModuleScore Function</em>. <a href="https://www.waltermuskovic.com/2021/04/15/seurat-s-addmodulescore-function/">https://www.waltermuskovic.com/2021/04/15/seurat-s-addmodulescore-function/</a>.
</div>
<div id="ref-Ryan" class="csl-entry">
Ryan SK, Garifallou JP, Gonzalez MV. 2020. <em>Neuroinflammation and Eif2 Signaling Persist Despite Antiretroviral Treatment in an hiPSC Tri-Culture Model of HIV Infection</em>. <em>#SStem Cell Reports</em>. Vol. 14(4). <a href="https://doi: 10.1016/j.stemcr.2020.02.010">doi: 10.1016/j.stemcr.2020.02.010</a>.
</div>
<div id="ref-Farhadian" class="csl-entry">
Shelli F. Farhadian, Chrysoula Zografou, Sameet S. Mehta, and Serena S. Spudich. 2018. <em>Single-Cell RNA Sequencing Reveals Microglia-Like Cells in Cerebrospinal Fluid During Virologically Suppressed HIV</em>. <em>JCI Insight</em>. Vol. 3(18):e121718. <a href="https://doi.org/10.1172/jci.insight.121718">https://doi.org/10.1172/jci.insight.121718</a>.
</div>
<div id="ref-Stuart" class="csl-entry">
Stuart T, Hoffman P, Butler A. 2019. <em>Comprehensive Integration of Single-Cell Data</em>. <em>Cell</em>. Vol. 177(7). <a href="https://doi: 10.1016/j.cell.2019.05.031">doi: 10.1016/j.cell.2019.05.031</a>.
</div>
<div id="ref-Tirosh" class="csl-entry">
Tirosh I, Prakadan SM, Izar B. 2016. <em>Dissecting the Multicellular Ecosystem of Metastatic Melanoma by Single-Cell RNA-Seq</em>. <em>Science</em>. Vol. 352(6282). <a href="https://doi: 10.1126/science.aad0501">doi: 10.1126/science.aad0501</a>.
</div>
</div>
</div>

<div id="rmd-source-code">---
title: "Methods"
author: "Katharina Mikulik"
output: html_document
bibliography: references.bib
link-citations: yes
---



## Quality Control

For Quality Control I used the publications from which the datasets were obtained
as an orientation, in order to be able to reproduce the results and findings. For
the dataset from [@Farhadian] genes expressed in less than 3 cells were 
removed. Additionally, the ten percent of cells with the highest percentage
of mitochondrial genes and cells with a number of features below 500 or above 
2500 was removed. For the dataset from [@Ryan] cells with a percentage of mitochondrial genes above 20% were removed, as well as, cells with a number of features below 200 
and above 5000. 

## Normalization

Log normalization was used to normalize count matrices, using the function `NormalizeData()`from the Seurat package. The counts of a particular
feature for a particular cell are divided by the total counts for that cell and
multiplied by a scaling factor of 10,000. Then the result is natural-log 
transformed. For GRN inference with SCENIC the function `SCTransform()` from the 
Seurat package was used, so that the data structure would work with the SCENIC pipeline. This method uses regularized negative binomial regression to normalize 
the count matrix. 

## Clustering

After calculating a PCA the cells were clustered based on their nearest neighbors.
For clustering the functions `FindNeigbors()` and `FindClusters()` from the 
Seurat package were used. First, a k nearest neighbor (KNN) graph is built
based on the PCA space. Then, the Louvain algorithm is used, which iteratively 
optimizes the standard modularity function. Modularity measures the strength of division of a network into modules. High modularity is equivalent to dense 
connections within modules and sparse connections between nodes in different
modules. 

  
## Integration & Label Transfer

See here for more details on the method and implementation:
[@Stuart]

See here for a vignette:
https://satijalab.org/seurat/articles/integration_mapping.html

### Integration

First, canonical correlation analysis is performed to reduce dimensionality of both
datasets. This method can find shared biological markers, even if batch effects 
are very large. Second, expression data is L2 normalized. After the 
projection of distinct datasets into a shared subspace, pairs of MNN in this shared subspace are identified. These MNN correspond to cells which are biologically
similar. The MNN serve as integration anchors for the integration, but 
beforehand the anchors are scored. The score is a measure of how similar the two 
anchor cells are. Using anchors and their corresponding scores, 
correction vectors are computed for each cell. The correction vectors contain 
information on the difference between two anchor cells. Using the correction vectors 
the expression values of each cell are transformed, so that they are part of an
integrated dataset [@Stuart]. This integrated dataset can then be used for 
further analysis. Finding anchors between datasets is not only required for 
integration of two distinct datasets, but also for label transfer. 

### Label Transfer

After finding anchors between distinct datasets as described above, it 
is also possible to transfer information, for example cell type annotations. This
means that instead of de-nove analysis of cell type markers the cell labels are 
learned from a reference dataset. Instead of using canonical correlation analysis, the
PCA computed for the reference dataset is projected onto the query dataset. 
A binary classification matrix is build with rows corresponding to all possible 
classes and columns corresponding to reference anchors of the anchor pairs. 
If a reference anchor belongs to a certain class the entry will be one, otherwise
it will be zero. The classification matrix is then multiplied with a weight matrix.
This way we get prediction scores for each class and cell in the query dataset
[@Stuart].

## Correlation  

Since the TF activity is not normally distributed, instead of using
the Pearson correlation coefficient which assumes normally distributed data,
the Spearman correlation coefficient was used, which is rank-based and 
non-parametric. The Spearman correlation is a measure of strength and direction 
of a monotonic association.
The null hypothesis of the Spearman correlation is that there is no association 
between the two variables. Statistical significance does not give any information
on the strength of correlation. 

Spearman's rank order correlation:


$\rho = \frac{\sum_{i} (x_{i} - \overline{x}) (y_{i} - \overline{y})}
{\sqrt{\sum_{i}(x_{i} - \overline{x})^2\sum_{i}(y_{i} - \overline{y})^2}}$
  
## Gene signature scores

The function `AddModuleScore()` from the Seurat package calculates gene expression
scores for a particular set of signature genes. For example, I calculated 
scores for the genes defining the infected iMg from the triculture dataset in 
Myeloid2. Calculating the gene signature score for each cell provides information 
on whether a particular cell shows a higher/unchanged/lower expression of a
particular set of genes (signature for a particular cell tpye).

First, the average expression of each gene across all cells is calculated and 
the genes are ordered according to their average expression values. Next, genes
are sorted into bins of equal size. For each gene of the signature gene set, 
100 genes are drawn from the same expression bin as a control group. This way, 
it is ensured that the control genes have a similar distribution of expression
levels as the signature gene. Finally, the average expression of control genes
and signature genes is computed in each cell. The resulting average control 
gene expression is subtracted from the average signature gene expression in 
each cell. This yields a score in each cell, which, if close to zero, can be 
interpreted as no enrichment or depletion of the signature genes. If the score 
in a particular cell is bigger than zero this means that there is an 
enrichment for the signature genes in this particular cell. Conversely, if the
score is smaller than zero this means that there is a depletion. [@Tirosh; @Muskovic]

## K-means clustering

The algorithm starts with k randomly selected centroids. All data points are 
assigned to one of the datapoints depending on the distance between them. 
The process of defining centroids, assigning each point to one centroid and 
updating the centroids is repeated until the clusters stabilize or a maximum 
number of iterations has been reached.

### Determine k with the elbow method

To determine the optimal k for k-means clustering I computed the within sum of
square (WSS) distance. More precisely, I computed the sum of all pairwise 
distances between points in the same cluster. If a cluster would only contain one 
point, then WSS would be zero. With increasing k the WSS decreases. The 
elbow/kink in the plot shows at which k adding additional clusters does not 
improve the WSS anymore. 
  
  
## Test for differential TF activity

To determine whether there is a significant difference between the clusters 
computed with k-means clustering, a Kruskal Wallis Test was performed for all
TFs between the different clusters. In contrast to ANOVA, the Kruskal Wallis Test
is non-paramteric and, therefore, better suited for the data at hand, which is 
not normally distributed. Using the FDR method I corrected for 
multiple hypothesis testing. For TFs with different activity between clusters I 
then did a posthoc analysis to determine which clusters are different exactly. 
A pairwise wilcox test was used, which again, is a non-parametric alternative to 
the t-test. 

## Differential gene expression

The Seurat functions `FindAllMarkers()` and `FindMarkers()` uses a Wilcoxon Rank
Sum test to identify differentially expressed genes between two groups. For the
dataset from [@Farhadian] to find also genes with small log-fold changes,
I set the threshold to 0.01 and 
the minimum fraction of cells in which the gene is detected to 0.01. This 
increased the computation time, but made it possible to also include
very infrequently expressed genes, which might for example be the case for 
Myeloid2-specific genes. 
In contrast, for the datset from [@Ryan] I used the default settings with a 
minimum log-fold change of 0.25 and a minimum fraction of cells which express
the gene of 0.1. For the minimum number of cells expressing the feature in at least one of the two compared groups I used the default of 3. Similarly, for the minimum number 
of cells in a group I used the default of 3. 


# References</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("Methods.Rmd");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
