<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Katharina Mikulik" />


<title>TF activity in Myeloid2 cells</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Abstract</a>
</li>
<li>
  <a href="Introduction.html">Introduction</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="HIV_CSF_dataset_analysis.html">1. CSF/Blood samples</a>
    </li>
    <li>
      <a href="TF_activity_analysis_of_myeloid2_cluster.html">2. TF activity Myeloid2</a>
    </li>
    <li>
      <a href="Triculture_csf_integration.html">3. Integration CSF/Blood &amp; triculture</a>
    </li>
    <li>
      <a href="Triculture_dataset_GRN_with_Scenic.html">4. TF activity iMg</a>
    </li>
  </ul>
</li>
<li>
  <a href="Discussion.html">Discussion</a>
</li>
<li>
  <a href="Methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Katharina782/report">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">TF activity in Myeloid2 cells</h1>
<h4 class="author">Katharina Mikulik</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttree12b0e376866060301ab81b01f5d7eac64029a77btargetblank12b0e37a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Katharina782/report/tree/12b0e376866060301ab81b01f5d7eac64029a77b" target="_blank">12b0e37</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttree12b0e376866060301ab81b01f5d7eac64029a77btargetblank12b0e37a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Katharina782/report/tree/12b0e376866060301ab81b01f5d7eac64029a77b" target="_blank">12b0e37</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/Independent_cell_type_identification_Farhadian_dataset_cache/
    Ignored:    analysis/Triculture_csf_integration_cache/

Untracked files:
    Untracked:  README.html
    Untracked:  analysis/Triculture_csf_integration/
    Untracked:  data/triculture_seurat_object
    Untracked:  rebuild file.Rmd
    Untracked:  rebuild-file.Rmd
    Untracked:  rebuild-file.html
    Untracked:  rebuild-file_cache/
    Untracked:  rebuild-file_files/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd</code>) and HTML (<code>docs/TF_activity_analysis_of_myeloid2_cluster.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/12b0e376866060301ab81b01f5d7eac64029a77b/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">12b0e37</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-09
</td>
<td>
wflow_publish(all = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/8481440d0c52c2f99c1682672636c891d233d870/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
<td>
wflow_publish(all = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/270aca65f4da298bf474a0f567f84d05abd61650/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
<td>
wflow_publish(all = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/f35ba45536a8b7546c4144a569fe908ce206aa4d/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">f35ba45</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/75af25c571d2f255bb408c63df6ee70c423b892a/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/dfcd540d3a574f3ad19196570abd4c6c51bd0aa3/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">dfcd540</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/dfcd540d3a574f3ad19196570abd4c6c51bd0aa3/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">dfcd540</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/3d3c73318704913e8bd8fd6101547ea25899045f/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">3d3c733</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/552f3e790fdab3c4f5386c882cd6d94f4d50d84f/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">552f3e7</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
wflow_publish(c(“analysis/Triculture_csf_integration.Rmd,” “analysis/Discussion.Rmd,”
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/cae354b1737a3561e0416a6c1fa88bb7a0498e9b/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">cae354b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/cae354b1737a3561e0416a6c1fa88bb7a0498e9b/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">cae354b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/f52af640acecbae9b1bf642ba2bbdf59693c0815/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">f52af64</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
<td>
wflow_publish(“analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/dacae7572064c5e5508b606f45daf0cf6258e003/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">dacae75</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
<td>
wflow_publish(“analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/6c5f3b175db2cc63caa1ba67e59c08e1a22f412b/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">6c5f3b1</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/d2f59b2d133fb7c2394b97182e5052ef1e4aadd5/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">d2f59b2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-27
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/d2f59b2d133fb7c2394b97182e5052ef1e4aadd5/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">d2f59b2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-27
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/d85a63137b84a09c19647281ffe67edbcce2e1eb/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">d85a631</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/fdf25ce5decf9dff024fb0470f591b4a70f6aee6/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">fdf25ce</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-22
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/bdbb921b65681909caf7252afe838d8b807681c3/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">bdbb921</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/efc0f94be22d7f357aeb93d42cfe3cec476f700a/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">efc0f94</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
<td>
wflow_publish(c(“analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd,”
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/9597c36bc5a6df5f071249d97949566b2c7db870/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/ea7e899746b03ca612d93ad003c875fc101d2329/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">ea7e899</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/40651d96647f10219a4aec953c8d618d57bc13f5/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/f94eb8c2238df11d56188f301f5c0d8bec07a0cb/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">f94eb8c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
wflow_publish(“analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd,”
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/bb965a8fe71e28af44fba8f3042204001cddaa0c/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">bb965a8</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
<td>
wflow_publish(“analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/91e34e269b1382b79f47740d549c4f68abc4cd46/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">91e34e2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
<td>
wflow_publish(“analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/2a95e927f953c093f2e48cb2b6551d152f735267/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/fec6c6e9430260f415f472edf227cbf6434c20ae/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">fec6c6e</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/e5e16d2fafe64661ad4400d6f14ff63f9bb0efce/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">e5e16d2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-17
</td>
<td>
wflow_publish(files = rmd)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/5bb41240b1a4dff64e2b5fded39e9d88dc15805b/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">5bb4124</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/5d4419ed1127a959edfae38d9632cfbafc52671d/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">5d4419e</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
wflow_publish(c("analysis/_site.yml“,”analysis/index.Rmd“,”analysis/references.bib",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/550997780789f62be907ec2095aa2ff2a1fd9087/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">5509977</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
wflow_publish(“analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
<td>
recent updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/dde10972b37ffd997474921968223514c8cb1b1f/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
<td>
recent updates
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style>
body {
text-align: justify}
</style>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, collapse = TRUE, message = FALSE,
                      cache = FALSE, autodep = TRUE)</code></pre>
<pre class="r"><code>library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggplot2)
library(dplyr)
library(ggrepel)
#library(harmony)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(knitr)
library(ggVennDiagram)
library(corrplot)
library(rstatix)
# &quot;Multitasking/Multiprocessing&quot;
library(future)
#plan(&quot;multisession&quot;, workers = 24)</code></pre>
<div id="grn-inference-with-scenic" class="section level1">
<h1>GRN inference with Scenic</h1>
<p>Using Scenic we tried to infer Gene regulatory networks from the gene expression data. The output is a matrix of TF (Transcription Factor) activity per cell. The input matrix for SCENIC included the 10,000 most highly variable genes of the Myeloid2 cell cluster. The resulting activity matrix contains activity scores of 125 different transcription factors.</p>
<p>First, we wanted to see if we can identify patterns in the TF activity in the Myeloid2 cells. We compared the TF activity to TFs shown to have differential binding scores between three infection states (uninfected, latent infection, active infection) in primary microglia (bulk ATACseq). For the comparison with the TF activity in Myeloid2 cells we focused on the TF with differential binding scores between <strong>uninfected</strong> and <strong>latently infected</strong> states.</p>
<p>The question we wanted to answer is: Can we see patterns of increased/decreased TF activity in Myeloid2 cells that correlate with the binding scores between infection states in primary microglia?</p>
</details>
<details>
<summary>
TF activity matrix and HIV CSF/Blood dataset
</summary>
<ul>
<li>read in Seurat object</li>
</ul>
<pre class="r"><code>hiv7 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds&quot;)
hiv4 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/non_integrated_HIV1_HIV2_4samples_seurat_object.rds&quot;)</code></pre>
<ul>
<li>read in AuCell matrix (output from Scenic)</li>
</ul>
<pre class="r"><code>auc_mtx &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/aucell_matrix_10k_hvg.tsv&quot;, sep = &quot;\t&quot;)

auc_mtx %&gt;% head %&gt;%  kable(caption = &quot;TF activity matrix (output from SCENIC)&quot;)</code></pre>
<table>
<caption>TF activity matrix (output from SCENIC)</caption>
<colgroup>
<col width="1%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">AHR</th>
<th align="right">ATF3</th>
<th align="right">BACH1</th>
<th align="right">BCL11A</th>
<th align="right">BCL6</th>
<th align="right">BHLHE40</th>
<th align="right">BPTF</th>
<th align="right">CEBPB</th>
<th align="right">CEBPD</th>
<th align="right">CEBPE</th>
<th align="right">CEBPG</th>
<th align="right">CLOCK</th>
<th align="right">CREB1</th>
<th align="right">CREB5</th>
<th align="right">CTCF</th>
<th align="right">CUX1</th>
<th align="right">DDIT3</th>
<th align="right">E2F2</th>
<th align="right">E2F3</th>
<th align="right">EGR1</th>
<th align="right">EGR3</th>
<th align="right">ELF1</th>
<th align="right">ELF2</th>
<th align="right">EOMES</th>
<th align="right">ETS1</th>
<th align="right">ETS2</th>
<th align="right">ETV2</th>
<th align="right">ETV3</th>
<th align="right">ETV5</th>
<th align="right">ETV6</th>
<th align="right">FLI1</th>
<th align="right">FOS</th>
<th align="right">FOSB</th>
<th align="right">FOSL2</th>
<th align="right">FOXN3</th>
<th align="right">FOXO3</th>
<th align="right">FOXO4</th>
<th align="right">FOXP1</th>
<th align="right">FOXP3</th>
<th align="right">GABPA</th>
<th align="right">GATA2</th>
<th align="right">GATA3</th>
<th align="right">GMEB1</th>
<th align="right">GMEB2</th>
<th align="right">HDX</th>
<th align="right">HIVEP1</th>
<th align="right">HMGA2</th>
<th align="right">HOXA9</th>
<th align="right">IKZF3</th>
<th align="right">IRF1</th>
<th align="right">IRF2</th>
<th align="right">IRF4</th>
<th align="right">IRF7</th>
<th align="right">IRF8</th>
<th align="right">JUN</th>
<th align="right">JUNB</th>
<th align="right">KLF10</th>
<th align="right">KLF12</th>
<th align="right">KLF4</th>
<th align="right">KLF5</th>
<th align="right">KLF6</th>
<th align="right">KLF9</th>
<th align="right">LEF1</th>
<th align="right">MAF</th>
<th align="right">MAFB</th>
<th align="right">MAFF</th>
<th align="right">MAFG</th>
<th align="right">MBD1</th>
<th align="right">MBD4</th>
<th align="right">MEF2A</th>
<th align="right">MEF2C</th>
<th align="right">MEF2D</th>
<th align="right">MGA</th>
<th align="right">MITF</th>
<th align="right">MXI1</th>
<th align="right">MYB</th>
<th align="right">MYBL1</th>
<th align="right">MYC</th>
<th align="right">NFATC2</th>
<th align="right">NFE2</th>
<th align="right">NFE2L2</th>
<th align="right">NFIL3</th>
<th align="right">NR1H3</th>
<th align="right">PAX5</th>
<th align="right">PBX3</th>
<th align="right">PRDM1</th>
<th align="right">REL</th>
<th align="right">REST</th>
<th align="right">RFX2</th>
<th align="right">RORA</th>
<th align="right">RUNX1</th>
<th align="right">RXRA</th>
<th align="right">SETDB1</th>
<th align="right">SOX13</th>
<th align="right">SOX4</th>
<th align="right">SOX5</th>
<th align="right">SP3</th>
<th align="right">SP4</th>
<th align="right">SPI1</th>
<th align="right">SPIB</th>
<th align="right">STAT1</th>
<th align="right">STAT3</th>
<th align="right">STAT4</th>
<th align="right">STAT6</th>
<th align="right">TBX19</th>
<th align="right">TBX21</th>
<th align="right">TCF12</th>
<th align="right">TCF7</th>
<th align="right">TEF</th>
<th align="right">TFAP4</th>
<th align="right">VEZF1</th>
<th align="right">XBP1</th>
<th align="right">YY1</th>
<th align="right">ZBTB33</th>
<th align="right">ZIK1</th>
<th align="right">ZKSCAN7</th>
<th align="right">ZNF133</th>
<th align="right">ZNF143</th>
<th align="right">ZNF234</th>
<th align="right">ZNF263</th>
<th align="right">ZNF354B</th>
<th align="right">ZNF420</th>
<th align="right">ZNF502</th>
<th align="right">ZNF721</th>
<th align="right">ZSCAN30</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">U1_AACGGTCTACAA</td>
<td align="right">0.0435556</td>
<td align="right">0.0523006</td>
<td align="right">0.0502414</td>
<td align="right">0.0652398</td>
<td align="right">0.0000000</td>
<td align="right">0.0074783</td>
<td align="right">0.0129231</td>
<td align="right">0.0490091</td>
<td align="right">0.0505306</td>
<td align="right">0.0000000</td>
<td align="right">0.0686000</td>
<td align="right">0.0000</td>
<td align="right">0.0270270</td>
<td align="right">0.1120000</td>
<td align="right">0.0121304</td>
<td align="right">0.0617534</td>
<td align="right">0.0343587</td>
<td align="right">0.0195325</td>
<td align="right">0.0462857</td>
<td align="right">0.0342381</td>
<td align="right">0.0386753</td>
<td align="right">0.0413333</td>
<td align="right">0.0897273</td>
<td align="right">0.0105820</td>
<td align="right">0.0096548</td>
<td align="right">0.0586475</td>
<td align="right">0.0220</td>
<td align="right">0.0251111</td>
<td align="right">0.0753957</td>
<td align="right">0.0793378</td>
<td align="right">0.0116645</td>
<td align="right">0.0559574</td>
<td align="right">0.0447588</td>
<td align="right">0.0292139</td>
<td align="right">0.0000000</td>
<td align="right">0.0355259</td>
<td align="right">0.0652308</td>
<td align="right">0.0046154</td>
<td align="right">0.0000000</td>
<td align="right">0.0080769</td>
<td align="right">0.0525283</td>
<td align="right">0.02280</td>
<td align="right">0.0000000</td>
<td align="right">0.0000</td>
<td align="right">0.0598</td>
<td align="right">0.0464</td>
<td align="right">0.0000000</td>
<td align="right">0.0307059</td>
<td align="right">0.01304</td>
<td align="right">0.0291937</td>
<td align="right">0.0328387</td>
<td align="right">0.0648571</td>
<td align="right">0.0638710</td>
<td align="right">0.0669279</td>
<td align="right">0.0536981</td>
<td align="right">0.0451377</td>
<td align="right">0.0057468</td>
<td align="right">0.02936</td>
<td align="right">0.0642699</td>
<td align="right">0.0000000</td>
<td align="right">0.0427604</td>
<td align="right">0.00000</td>
<td align="right">0.0000000</td>
<td align="right">0.00000</td>
<td align="right">0.0552796</td>
<td align="right">0.0276386</td>
<td align="right">0.0355833</td>
<td align="right">0.0000000</td>
<td align="right">0.0760000</td>
<td align="right">0.0913770</td>
<td align="right">0.0526324</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0532381</td>
<td align="right">0.00000</td>
<td align="right">0.0123636</td>
<td align="right">0.0136848</td>
<td align="right">0.0000000</td>
<td align="right">0.0311915</td>
<td align="right">0.0352090</td>
<td align="right">0.0053898</td>
<td align="right">0.0503357</td>
<td align="right">0.0686667</td>
<td align="right">0.0423636</td>
<td align="right">0.1161538</td>
<td align="right">0.0162290</td>
<td align="right">0.0726425</td>
<td align="right">0.0185217</td>
<td align="right">0</td>
<td align="right">0.0000</td>
<td align="right">0.0443077</td>
<td align="right">0.0439688</td>
<td align="right">0.1188571</td>
<td align="right">0.0000</td>
<td align="right">0.0682963</td>
<td align="right">0.0000000</td>
<td align="right">0.0502623</td>
<td align="right">0.0013671</td>
<td align="right">0.0510571</td>
<td align="right">0.0573205</td>
<td align="right">0.0407797</td>
<td align="right">0.0506569</td>
<td align="right">0.0000</td>
<td align="right">0.0450769</td>
<td align="right">0.1448333</td>
<td align="right">0.0101270</td>
<td align="right">0.0494615</td>
<td align="right">0.0092346</td>
<td align="right">0.0492222</td>
<td align="right">0.000</td>
<td align="right">0.0922857</td>
<td align="right">0.0191852</td>
<td align="right">0.0174412</td>
<td align="right">0.0847778</td>
<td align="right">0.0000000</td>
<td align="right">0.0000</td>
<td align="right">0</td>
<td align="right">0.0463636</td>
<td align="right">0</td>
<td align="right">0.0000</td>
<td align="right">0.0000000</td>
<td align="right">0.000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">U1_GGTGAATCTAGC</td>
<td align="right">0.1788889</td>
<td align="right">0.0203043</td>
<td align="right">0.0187299</td>
<td align="right">0.0189825</td>
<td align="right">0.0256667</td>
<td align="right">0.0072754</td>
<td align="right">0.0268462</td>
<td align="right">0.0198003</td>
<td align="right">0.0201004</td>
<td align="right">0.0203556</td>
<td align="right">0.0155636</td>
<td align="right">0.0000</td>
<td align="right">0.0150270</td>
<td align="right">0.0000000</td>
<td align="right">0.0117391</td>
<td align="right">0.0396438</td>
<td align="right">0.0092735</td>
<td align="right">0.0290130</td>
<td align="right">0.0181905</td>
<td align="right">0.0000000</td>
<td align="right">0.0209351</td>
<td align="right">0.0263590</td>
<td align="right">0.0490000</td>
<td align="right">0.0368042</td>
<td align="right">0.0253690</td>
<td align="right">0.0221786</td>
<td align="right">0.0000</td>
<td align="right">0.0241111</td>
<td align="right">0.0188426</td>
<td align="right">0.0188829</td>
<td align="right">0.0217290</td>
<td align="right">0.0204197</td>
<td align="right">0.0150161</td>
<td align="right">0.0254451</td>
<td align="right">0.0000000</td>
<td align="right">0.0361333</td>
<td align="right">0.0298462</td>
<td align="right">0.0484615</td>
<td align="right">0.1257778</td>
<td align="right">0.0169487</td>
<td align="right">0.0153962</td>
<td align="right">0.02680</td>
<td align="right">0.0816364</td>
<td align="right">0.2125</td>
<td align="right">0.0000</td>
<td align="right">0.0000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.04440</td>
<td align="right">0.0156634</td>
<td align="right">0.0185054</td>
<td align="right">0.0000000</td>
<td align="right">0.0123623</td>
<td align="right">0.0188840</td>
<td align="right">0.0215547</td>
<td align="right">0.0202077</td>
<td align="right">0.0201013</td>
<td align="right">0.01744</td>
<td align="right">0.0293006</td>
<td align="right">0.0000000</td>
<td align="right">0.0197812</td>
<td align="right">0.00000</td>
<td align="right">0.0179167</td>
<td align="right">0.00000</td>
<td align="right">0.0278817</td>
<td align="right">0.0081928</td>
<td align="right">0.0142292</td>
<td align="right">0.0000000</td>
<td align="right">0.0437391</td>
<td align="right">0.0297705</td>
<td align="right">0.0332794</td>
<td align="right">0.0300000</td>
<td align="right">0.0173939</td>
<td align="right">0.0332143</td>
<td align="right">0.00000</td>
<td align="right">0.0384545</td>
<td align="right">0.0299758</td>
<td align="right">0.0018462</td>
<td align="right">0.0179574</td>
<td align="right">0.0080226</td>
<td align="right">0.0354915</td>
<td align="right">0.0252494</td>
<td align="right">0.0353333</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0247634</td>
<td align="right">0.0278964</td>
<td align="right">0.0397971</td>
<td align="right">0</td>
<td align="right">0.0308</td>
<td align="right">0.0056923</td>
<td align="right">0.0126719</td>
<td align="right">0.0000000</td>
<td align="right">0.0000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0440328</td>
<td align="right">0.0201266</td>
<td align="right">0.0196625</td>
<td align="right">0.0227436</td>
<td align="right">0.0088405</td>
<td align="right">0.0178175</td>
<td align="right">0.0000</td>
<td align="right">0.0000000</td>
<td align="right">0.0296667</td>
<td align="right">0.0179365</td>
<td align="right">0.0000000</td>
<td align="right">0.0280741</td>
<td align="right">0.0731111</td>
<td align="right">0.036</td>
<td align="right">0.0000000</td>
<td align="right">0.0115000</td>
<td align="right">0.0203088</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000</td>
<td align="right">0</td>
<td align="right">0.0160909</td>
<td align="right">0</td>
<td align="right">0.0000</td>
<td align="right">0.0960000</td>
<td align="right">0.048</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">U1_TTCAGATACCTA</td>
<td align="right">0.0573333</td>
<td align="right">0.0138701</td>
<td align="right">0.0158276</td>
<td align="right">0.0121696</td>
<td align="right">0.0000000</td>
<td align="right">0.0334783</td>
<td align="right">0.0559615</td>
<td align="right">0.0119481</td>
<td align="right">0.0141201</td>
<td align="right">0.0000000</td>
<td align="right">0.0238000</td>
<td align="right">0.0930</td>
<td align="right">0.0261622</td>
<td align="right">0.0000000</td>
<td align="right">0.0358261</td>
<td align="right">0.0132329</td>
<td align="right">0.0151390</td>
<td align="right">0.0073506</td>
<td align="right">0.0000000</td>
<td align="right">0.0186190</td>
<td align="right">0.0214026</td>
<td align="right">0.0410256</td>
<td align="right">0.0061818</td>
<td align="right">0.0535344</td>
<td align="right">0.0255714</td>
<td align="right">0.0135694</td>
<td align="right">0.0000</td>
<td align="right">0.0240667</td>
<td align="right">0.0183745</td>
<td align="right">0.0116388</td>
<td align="right">0.0141548</td>
<td align="right">0.0166328</td>
<td align="right">0.0164309</td>
<td align="right">0.0270751</td>
<td align="right">0.0445714</td>
<td align="right">0.0169333</td>
<td align="right">0.0397692</td>
<td align="right">0.0020000</td>
<td align="right">0.1806667</td>
<td align="right">0.0304359</td>
<td align="right">0.0322642</td>
<td align="right">0.00000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000</td>
<td align="right">0.0000</td>
<td align="right">0.0000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.04236</td>
<td align="right">0.0189472</td>
<td align="right">0.0147312</td>
<td align="right">0.0000000</td>
<td align="right">0.0100099</td>
<td align="right">0.0148056</td>
<td align="right">0.0178491</td>
<td align="right">0.0095801</td>
<td align="right">0.0039747</td>
<td align="right">0.07088</td>
<td align="right">0.0155092</td>
<td align="right">0.0000000</td>
<td align="right">0.0051563</td>
<td align="right">0.00650</td>
<td align="right">0.0273333</td>
<td align="right">0.10600</td>
<td align="right">0.0089032</td>
<td align="right">0.0136546</td>
<td align="right">0.0263125</td>
<td align="right">0.0130000</td>
<td align="right">0.0492174</td>
<td align="right">0.0169180</td>
<td align="right">0.0167059</td>
<td align="right">0.0684286</td>
<td align="right">0.0362424</td>
<td align="right">0.0278095</td>
<td align="right">0.00000</td>
<td align="right">0.0000000</td>
<td align="right">0.0367879</td>
<td align="right">0.0107179</td>
<td align="right">0.0468936</td>
<td align="right">0.0132542</td>
<td align="right">0.0587797</td>
<td align="right">0.0143077</td>
<td align="right">0.0000000</td>
<td align="right">0.0174545</td>
<td align="right">0.0000000</td>
<td align="right">0.0299542</td>
<td align="right">0.0114197</td>
<td align="right">0.0090435</td>
<td align="right">0</td>
<td align="right">0.0000</td>
<td align="right">0.0612308</td>
<td align="right">0.0077344</td>
<td align="right">0.0000000</td>
<td align="right">0.0000</td>
<td align="right">0.0019259</td>
<td align="right">0.1536667</td>
<td align="right">0.0446885</td>
<td align="right">0.0077975</td>
<td align="right">0.0147593</td>
<td align="right">0.0084872</td>
<td align="right">0.0097975</td>
<td align="right">0.0115693</td>
<td align="right">0.0892</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0607302</td>
<td align="right">0.0072308</td>
<td align="right">0.0133580</td>
<td align="right">0.0000000</td>
<td align="right">0.013</td>
<td align="right">0.0000000</td>
<td align="right">0.0271296</td>
<td align="right">0.0281618</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.1228</td>
<td align="right">0</td>
<td align="right">0.0174545</td>
<td align="right">0</td>
<td align="right">0.1128</td>
<td align="right">0.0000000</td>
<td align="right">0.000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0796667</td>
</tr>
<tr class="even">
<td align="left">U1_GGCATAGGATTN</td>
<td align="right">0.1244444</td>
<td align="right">0.1293581</td>
<td align="right">0.0819425</td>
<td align="right">0.0657368</td>
<td align="right">0.0400000</td>
<td align="right">0.0114493</td>
<td align="right">0.0077308</td>
<td align="right">0.0926018</td>
<td align="right">0.1091048</td>
<td align="right">0.0180000</td>
<td align="right">0.1254909</td>
<td align="right">0.0000</td>
<td align="right">0.0475135</td>
<td align="right">0.4828889</td>
<td align="right">0.0000000</td>
<td align="right">0.0292055</td>
<td align="right">0.1213901</td>
<td align="right">0.0327273</td>
<td align="right">0.0000000</td>
<td align="right">0.0739524</td>
<td align="right">0.1063896</td>
<td align="right">0.1273333</td>
<td align="right">0.0647273</td>
<td align="right">0.0040952</td>
<td align="right">0.0071310</td>
<td align="right">0.1326794</td>
<td align="right">0.0000</td>
<td align="right">0.0471556</td>
<td align="right">0.1554809</td>
<td align="right">0.0968227</td>
<td align="right">0.0690581</td>
<td align="right">0.0880918</td>
<td align="right">0.0726302</td>
<td align="right">0.1370058</td>
<td align="right">0.1240000</td>
<td align="right">0.1585630</td>
<td align="right">0.0671538</td>
<td align="right">0.0064615</td>
<td align="right">0.0313333</td>
<td align="right">0.0155897</td>
<td align="right">0.0352075</td>
<td align="right">0.01520</td>
<td align="right">0.0061818</td>
<td align="right">0.0685</td>
<td align="right">0.0840</td>
<td align="right">0.0586</td>
<td align="right">0.0000000</td>
<td align="right">0.0742353</td>
<td align="right">0.02352</td>
<td align="right">0.0468180</td>
<td align="right">0.0541613</td>
<td align="right">0.2425714</td>
<td align="right">0.0707246</td>
<td align="right">0.0723668</td>
<td align="right">0.1212604</td>
<td align="right">0.0766998</td>
<td align="right">0.0190886</td>
<td align="right">0.00000</td>
<td align="right">0.0873497</td>
<td align="right">0.0022222</td>
<td align="right">0.0876042</td>
<td align="right">0.00000</td>
<td align="right">0.0044167</td>
<td align="right">0.36125</td>
<td align="right">0.0980753</td>
<td align="right">0.0853976</td>
<td align="right">0.1486042</td>
<td align="right">0.0130000</td>
<td align="right">0.0552174</td>
<td align="right">0.2110984</td>
<td align="right">0.1282794</td>
<td align="right">0.1105714</td>
<td align="right">0.0000000</td>
<td align="right">0.1315000</td>
<td align="right">0.27325</td>
<td align="right">0.0651818</td>
<td align="right">0.0159758</td>
<td align="right">0.0048718</td>
<td align="right">0.0324255</td>
<td align="right">0.0679774</td>
<td align="right">0.0818644</td>
<td align="right">0.0885408</td>
<td align="right">0.0624444</td>
<td align="right">0.1314545</td>
<td align="right">0.0038462</td>
<td align="right">0.1265191</td>
<td align="right">0.1317409</td>
<td align="right">0.0135652</td>
<td align="right">0</td>
<td align="right">0.0000</td>
<td align="right">0.0283077</td>
<td align="right">0.0450938</td>
<td align="right">0.0000000</td>
<td align="right">0.0000</td>
<td align="right">0.1118519</td>
<td align="right">0.0000000</td>
<td align="right">0.0568852</td>
<td align="right">0.0008608</td>
<td align="right">0.0829198</td>
<td align="right">0.0460000</td>
<td align="right">0.0639190</td>
<td align="right">0.0423285</td>
<td align="right">0.0934</td>
<td align="right">0.0390769</td>
<td align="right">0.0423333</td>
<td align="right">0.0082063</td>
<td align="right">0.0000000</td>
<td align="right">0.0157037</td>
<td align="right">0.0717778</td>
<td align="right">0.000</td>
<td align="right">0.0000000</td>
<td align="right">0.0190556</td>
<td align="right">0.0344706</td>
<td align="right">0.0838889</td>
<td align="right">0.1400000</td>
<td align="right">0.0000</td>
<td align="right">0</td>
<td align="right">0.0663636</td>
<td align="right">0</td>
<td align="right">0.0000</td>
<td align="right">0.1056667</td>
<td align="right">0.000</td>
<td align="right">0.0744444</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">U1_TTCTAAAGTCTA</td>
<td align="right">0.0262222</td>
<td align="right">0.0120705</td>
<td align="right">0.0176839</td>
<td align="right">0.0272749</td>
<td align="right">0.0000000</td>
<td align="right">0.0042029</td>
<td align="right">0.0332692</td>
<td align="right">0.0213333</td>
<td align="right">0.0225808</td>
<td align="right">0.0135111</td>
<td align="right">0.0295273</td>
<td align="right">0.1440</td>
<td align="right">0.0129189</td>
<td align="right">0.0000000</td>
<td align="right">0.0985652</td>
<td align="right">0.0113699</td>
<td align="right">0.0183318</td>
<td align="right">0.0044935</td>
<td align="right">0.0000000</td>
<td align="right">0.0233333</td>
<td align="right">0.0223117</td>
<td align="right">0.0271282</td>
<td align="right">0.0120909</td>
<td align="right">0.0509418</td>
<td align="right">0.0434762</td>
<td align="right">0.0259394</td>
<td align="right">0.0000</td>
<td align="right">0.0376222</td>
<td align="right">0.0179830</td>
<td align="right">0.0207090</td>
<td align="right">0.0259226</td>
<td align="right">0.0181836</td>
<td align="right">0.0114791</td>
<td align="right">0.0159191</td>
<td align="right">0.0000000</td>
<td align="right">0.0143407</td>
<td align="right">0.0000000</td>
<td align="right">0.0556923</td>
<td align="right">0.0417778</td>
<td align="right">0.0407179</td>
<td align="right">0.0106415</td>
<td align="right">0.05352</td>
<td align="right">0.0000000</td>
<td align="right">0.0000</td>
<td align="right">0.1724</td>
<td align="right">0.0000</td>
<td align="right">0.0007273</td>
<td align="right">0.0000000</td>
<td align="right">0.02088</td>
<td align="right">0.0189746</td>
<td align="right">0.0115484</td>
<td align="right">0.0000000</td>
<td align="right">0.0119702</td>
<td align="right">0.0170439</td>
<td align="right">0.0200453</td>
<td align="right">0.0157336</td>
<td align="right">0.0041772</td>
<td align="right">0.05496</td>
<td align="right">0.0173865</td>
<td align="right">0.0000000</td>
<td align="right">0.0122188</td>
<td align="right">0.11475</td>
<td align="right">0.0830000</td>
<td align="right">0.01300</td>
<td align="right">0.0150645</td>
<td align="right">0.0128996</td>
<td align="right">0.0041875</td>
<td align="right">0.0391667</td>
<td align="right">0.0513913</td>
<td align="right">0.0299672</td>
<td align="right">0.0221471</td>
<td align="right">0.0330000</td>
<td align="right">0.0547273</td>
<td align="right">0.0411429</td>
<td align="right">0.10400</td>
<td align="right">0.0000000</td>
<td align="right">0.0525455</td>
<td align="right">0.0000000</td>
<td align="right">0.0289362</td>
<td align="right">0.0145198</td>
<td align="right">0.0231864</td>
<td align="right">0.0221725</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0481221</td>
<td align="right">0.0161347</td>
<td align="right">0.0489855</td>
<td align="right">0</td>
<td align="right">0.0000</td>
<td align="right">0.0355385</td>
<td align="right">0.0190781</td>
<td align="right">0.0000000</td>
<td align="right">0.0000</td>
<td align="right">0.0545926</td>
<td align="right">0.1063333</td>
<td align="right">0.0186230</td>
<td align="right">0.0428354</td>
<td align="right">0.0194425</td>
<td align="right">0.0194744</td>
<td align="right">0.0168101</td>
<td align="right">0.0198175</td>
<td align="right">0.0000</td>
<td align="right">0.0000000</td>
<td align="right">0.0303333</td>
<td align="right">0.0217778</td>
<td align="right">0.0678462</td>
<td align="right">0.0388642</td>
<td align="right">0.0364444</td>
<td align="right">0.115</td>
<td align="right">0.0385714</td>
<td align="right">0.0248148</td>
<td align="right">0.0214559</td>
<td align="right">0.0822222</td>
<td align="right">0.0000000</td>
<td align="right">0.0000</td>
<td align="right">0</td>
<td align="right">0.0333636</td>
<td align="right">0</td>
<td align="right">0.0000</td>
<td align="right">0.1436667</td>
<td align="right">0.084</td>
<td align="right">0.0000000</td>
<td align="right">0.0893333</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">U1_CTGCCCTGAGTC</td>
<td align="right">0.0000000</td>
<td align="right">0.0148609</td>
<td align="right">0.0131954</td>
<td align="right">0.0070117</td>
<td align="right">0.0000000</td>
<td align="right">0.0393043</td>
<td align="right">0.0351154</td>
<td align="right">0.0138054</td>
<td align="right">0.0143690</td>
<td align="right">0.0000000</td>
<td align="right">0.0150364</td>
<td align="right">0.0804</td>
<td align="right">0.0340000</td>
<td align="right">0.0095556</td>
<td align="right">0.0586522</td>
<td align="right">0.0250685</td>
<td align="right">0.0064036</td>
<td align="right">0.0308052</td>
<td align="right">0.0184762</td>
<td align="right">0.0250952</td>
<td align="right">0.0087532</td>
<td align="right">0.0353333</td>
<td align="right">0.0000000</td>
<td align="right">0.0500952</td>
<td align="right">0.0140714</td>
<td align="right">0.0180128</td>
<td align="right">0.0232</td>
<td align="right">0.0530889</td>
<td align="right">0.0171149</td>
<td align="right">0.0125619</td>
<td align="right">0.0270581</td>
<td align="right">0.0180000</td>
<td align="right">0.0187588</td>
<td align="right">0.0188439</td>
<td align="right">0.0000000</td>
<td align="right">0.0017037</td>
<td align="right">0.0000000</td>
<td align="right">0.0100000</td>
<td align="right">0.0000000</td>
<td align="right">0.0307692</td>
<td align="right">0.0092830</td>
<td align="right">0.06296</td>
<td align="right">0.0020000</td>
<td align="right">0.0000</td>
<td align="right">0.0000</td>
<td align="right">0.0000</td>
<td align="right">0.0000000</td>
<td align="right">0.0180000</td>
<td align="right">0.05020</td>
<td align="right">0.0185871</td>
<td align="right">0.0163118</td>
<td align="right">0.0000000</td>
<td align="right">0.0130074</td>
<td align="right">0.0095110</td>
<td align="right">0.0145057</td>
<td align="right">0.0148623</td>
<td align="right">0.0080000</td>
<td align="right">0.04872</td>
<td align="right">0.0194969</td>
<td align="right">0.0373333</td>
<td align="right">0.0230104</td>
<td align="right">0.00000</td>
<td align="right">0.0295833</td>
<td align="right">0.05550</td>
<td align="right">0.0157527</td>
<td align="right">0.0111004</td>
<td align="right">0.0061458</td>
<td align="right">0.0466667</td>
<td align="right">0.0000000</td>
<td align="right">0.0100492</td>
<td align="right">0.0020588</td>
<td align="right">0.0000000</td>
<td align="right">0.0218788</td>
<td align="right">0.0015714</td>
<td align="right">0.05550</td>
<td align="right">0.0412727</td>
<td align="right">0.0340970</td>
<td align="right">0.0119744</td>
<td align="right">0.0338723</td>
<td align="right">0.0055367</td>
<td align="right">0.0148136</td>
<td align="right">0.0114452</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0552308</td>
<td align="right">0.0456489</td>
<td align="right">0.0018238</td>
<td align="right">0.0165217</td>
<td align="right">0</td>
<td align="right">0.1132</td>
<td align="right">0.0356923</td>
<td align="right">0.0044687</td>
<td align="right">0.0000000</td>
<td align="right">0.0765</td>
<td align="right">0.0324444</td>
<td align="right">0.0000000</td>
<td align="right">0.0351148</td>
<td align="right">0.0371392</td>
<td align="right">0.0120281</td>
<td align="right">0.0013718</td>
<td align="right">0.0142076</td>
<td align="right">0.0101606</td>
<td align="right">0.0000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0205397</td>
<td align="right">0.0065385</td>
<td align="right">0.0208148</td>
<td align="right">0.0000000</td>
<td align="right">0.000</td>
<td align="right">0.0000000</td>
<td align="right">0.0094074</td>
<td align="right">0.0346471</td>
<td align="right">0.0000000</td>
<td align="right">0.0073333</td>
<td align="right">0.0000</td>
<td align="right">0</td>
<td align="right">0.0000000</td>
<td align="right">0</td>
<td align="right">0.0756</td>
<td align="right">0.0000000</td>
<td align="right">0.062</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.1450000</td>
</tr>
</tbody>
</table>
<ul>
<li>Combine AuCell Matrix and Seurat object</li>
</ul>
<pre class="r"><code># add auc matrix to metadata of hiv4 seurat object
hiv7_scenic &lt;- AddMetaData(hiv7, 
                    metadata = auc_mtx, 
                    col.name = colnames(auc_mtx))</code></pre>
<hr />
</details>
</div>
<div id="bulk-atacseq-footprinting" class="section level1">
<h1>Bulk ATACseq footprinting</h1>
<p>Using bulk ATACseq data from the C20 cell line (immortalized microglia) with three different conditions, active infection, latent infection, uninfected, differential binding scores were obtained using TOBIAS. Here, I will use the TFs which were previously shown to have differential binding scores in these primary microglia. Since I expect the CSF Myeloid cells to be latently infected, because they have been obtained from patients receiving ART, I will use the TFs with differential binding scores between latent infection and uninfected. We would like to know if the TF activity inferred from RNAseq data of Myeloid2 cells correlates with these differential binding scores. In other words, we want to know if the patterns of differential TF activity are similar to the patterns of differential TF binding scores. For example, if a TF was shown to be more bound in latently infected primary microglia compared to uninfected primary microglia, we would expect the TF to be more active in a group of Myeloid2 cells which are latently infected compared to a group of Myeloid2 cells which are uninfected.</p>
<p>Comparing the TFs from the TF activity matrix with the differentially bound TFs from the footprinting, 24 of the TFs overlap.</p>
<p>Using less than 10,000 most highly variable genes as an input matrix for SCENIC resulted in a lower number of TFs which overlap. Increasing the number of input features for SCENIC increased the number of TFs included in the TF activity matrix. Therefore, 10,000 most highly variable genes were used.</p>
<pre class="r"><code># read in dataframe containing differntial binding scores between conditions
#
# red = latently infected
# gfp = active infected
# uninfected
tfs_ana &lt;- read.table(&quot;/media/ag-cherrmann/projects/06_HIV_Microglia/data/atacseq/data-2020-11-06/tobias/TOBIAS_snakemake/footprint_mglia2_GlassTF_17-03/TFBS/bindetect_results.txt&quot;, sep = &quot;\t&quot;, header = TRUE)

# filter TFs with significant p values between latent infection and uninfected
# I decided to use a threshold of p-value &lt; 1e-50, because due to the large number 
# of data, all p-values will be very high
top_tfs &lt;- tfs_ana %&gt;% filter(uninf_red_pvalue &lt; 1e-50)

# keep only the TFs in the dataframe which overlap between the two datasets.
overlap_atac &lt;- top_tfs %&gt;% filter(name %in% colnames(auc_mtx))

# extract names of TFs more bound in either condition
up_latent &lt;- overlap_atac$name[overlap_atac$uninf_red_change &lt; 0]
up_uninf &lt;- overlap_atac$name[overlap_atac$uninf_red_change &gt; 0]</code></pre>
<div id="volcano-plot-atac-seq-footprinting" class="section level2">
<h2>Volcano Plot ATAC-seq footprinting</h2>
<p>Only four TFs found in the TF activity matrix of Myeloid2 cells are more bound in latent infection than in uninfected primary microglia.</p>
<pre class="r"><code>
top_tfs %&gt;% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = ifelse(uninf_red_change &lt; 0, &quot;up_latent&quot;, &quot;up_uninf&quot;)) %&gt;%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  mutate(label = ifelse(name %in% overlap_atac$name, name, NA)) %&gt;% 
  ggplot(aes(x = uninf_red_change,
             y = -log10(uninf_red_pvalue),
             col = condition, 
             label = label)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-50), col = &quot;red&quot;) +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = &quot;TFs with differential binding scores&quot;) +
  ylab(&quot;-log10(p_value)&quot;) +
  xlab(&quot;differential binding score&quot;)
Warning: Removed 51 rows containing missing values (geom_text).</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5bb41240b1a4dff64e2b5fded39e9d88dc15805b/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">5bb4124</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="compare-tf-activities-between-clusters-and-look-for-patterns" class="section level1">
<h1>Compare TF activities between clusters and look for patterns</h1>
<p>The heatmaps below were split into 3 separate heatmaps, one for each patient. Shown are the 24 TFs found by ATAC-seq footprinting as rows and cells as columns. Judging from the plots no TFs clearly separate the clusters or show a pattern of activity which is similar to the footprinting binding scores. If the pattern was similar to the footprinting binding scores we would expect to see one cluster where REST, KLF4, PAX5 and MXI1 are upregulated. This cannot be seen in the plots below.</p>
<pre class="r"><code>

# TF annotations
la &lt;- rowAnnotation(type_TF = 
                      case_when(colnames(auc_mtx) %in% up_latent ~ &quot;up_latent&quot;,
                                        colnames(auc_mtx) %in% up_uninf  ~ &quot;up_uninf&quot;,
                                        TRUE ~&quot;unknown&quot;),
                     name = &quot;type_TF&quot;,
                     col = list(type_TF = c(&quot;up_latent&quot; =  &quot;red&quot;, 
                                            &quot;up_uninf&quot; =  &quot;pink&quot;,
                                            &quot;unknown&quot; = &quot;forestgreen&quot;)))
  </code></pre>
<pre class="r"><code># 3 heatmaps, but only the TFs which overlap with Anas analysis
ht_list &lt;- map(seq.int(1:3), function(n) {
  patients &lt;- c (&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;) # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  # select TFs which overlap with Anas analysis
  mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), overlap_atac$name]
  # heatmap
  ht &lt;- Heatmap(t(mtx), 
               column_title = paste0(&quot;patient: &quot;, patients[n]), 

               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
               right_annotation = rowAnnotation(type_TF = ifelse(colnames(auc_mtx[,overlap_atac$name]) %in% up_latent, &quot;up_latent&quot;, &quot;up_uninf&quot;),
                                                name = &quot;type_TF&quot;,
                                                col = list(type_TF = c(&quot;up_latent&quot; =  &quot;red&quot;,
                                                                       &quot;up_uninf&quot; =  &quot;pink&quot;))))
  list(name = patients[n], heatmap = ht)
})

  
ht_list = ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap
draw(ht_list, ht_gap = unit(1, &quot;cm&quot;))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="different-approaches" class="section level2">
<h2>Different approaches</h2>
<p>Myeloid2 cells of patient1 (HIV1_CSF) were clustered based on their TF activity using k-means clustering with k=3/4/5. The reason why only patient one was used, is that the other two patients might be used for validation of the clustering later on. If the clusters have a biological meaning, it should be possible to transfer the clusters onto the Myeloid2 cells from patient2 (HIV2_CSF) and patient3 (HIV3_CSF). Also, if the clusters separate the Myeloid2 cells of the other two patients in an interpretable and meaningful way this would validate that there is a biological meaning behind the clusters.</p>
<p>However, the label transfer of the cluster identities did not work, probably due to the small sample size. There are only 146 cells in total in the Myeloid2 cluster of infected patients. A more detailed description of this step and corresponding results can be found in the appendix under <em>“Determine k for<em> </em>k-means clustering”</em> and <em>“Transfer the cluster identities to patient2 and patient3”</em>.</p>
<p>Since the approach described did not yield any interesting results, an alternative strategy was used, namely, performing independent k-means clustering across all patients and comparing the TF activities between clusters. This way we might find TF activities which differentiate clusters across patients and find correlations between clusters across patients.</p>
</div>
<div id="k-means-clustering-with-k-3" class="section level2 tabset">
<h2 class="tabset">K-means clustering with k = 3</h2>
<p>K-means clustering with k = 3, was performed for all 3 patients separately. This way it might be possible to find TFs with different activities between the three clusters. The clusters in the heatmaps below look more sensible than the clusters observed after cluster label transfer (see appendix).</p>
<p>Since the data is not normally distributed, the Kruskal-Wallis-Test is more appropriate than ANOVA to test for differences between clusters. The Null-Hypothesis of the Kruskal-Wallis-Test is that there is no difference between the groups. Since the test is repeated 125 times, the p-values are adjusted with the FDR method. This methods is more powerful than more conservative methods, like Bonferroni, Holm or Hochberg.</p>
<div id="patient-1" class="section level3">
<h3>Patient 1</h3>
<pre class="r"><code>#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV1_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_pat1 &lt;-draw(Heatmap(t(mtx), 
             column_title = &quot;Myeloid2 cells from patient 1, k = 3&quot;, 

             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la
             ))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>             #top_annotation = column_annotation) </code></pre>
<details>
<summary>
<strong>Kruskal_Wallis test &amp; Pairwise-Wilcox test for Patient 1</strong>
</summary>
<p>Performing Kruskal-Wallis-Test to find differences between the three clusters in patient one yields 64 TFs out of all 125 TFs which are differentially active between the three clusters(p-value &lt; 0.05). Of the 24 TFs with differential binding scores in the C20 cell line, sixteen have significantly different activity between the clusters.</p>
<p>Now that it is known that there are significant differences between clusters, the next step is to identify which cluster/s are different from which other cluster/s, using Pairwise-Wilcox-Test, a non-parametric test. Again, because we repeat the test we will correct for multiple hypothesis testing using the FDR method.</p>
<pre class="r"><code>### Extract cells from each cluster

# get the columns names of cluster one
c1 &lt;- colnames(t(mtx[column_order(ht_pat1)[[1]],]))
#write.table(c1, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv&quot;, sep = &quot;\t&quot;)

# get the column names of cluster two
c2 &lt;- colnames(t(mtx[column_order(ht_pat1)[[2]],]))
#write.table(c2, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv&quot;, sep = &quot;\t&quot;)

# get the column names of cluster 3
c3 &lt;- colnames(t(mtx[column_order(ht_pat1)[[3]],]))
#write.table(c3, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv&quot;, sep = &quot;\t&quot;)


# non-parametric alternative to ANOVA -&gt; Kruskal_wallis test
k3_df &lt;- hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV1_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(c1,c2,c3)) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c1, &quot;c1&quot;, ifelse(cell %in% c2, &quot;c2&quot;, &quot;c3&quot;))) 
# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))[&quot;p.value&quot;]
  #list(tf = tf, p_value = p_values)
})



# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
sign3 &lt;- adj_p_values[adj_p_values&lt;.05]
length(adj_p_values[adj_p_values&lt;0.05])
[1] 64

# get TFs that are significantly different
hiv1_diff &lt;- colnames(auc_mtx)[adj_p_values &lt; .05]

#write.table(diff3, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1&quot;)



## 24 TFs also identified by ATAC-se

# check 24 TFs with Kruskal Wallis test
p_values &lt;- map(seq.int(1:24), function(n){
  tf &lt;- overlap_atac$name[n]
  p_values &lt;- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))[&quot;p.value&quot;]
})
#sign_p_values &lt;- p_values[p_values &lt; 0.01]

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values&lt;0.05])
[1] 16

# get TFs that are differentially expressed
hiv1_atac &lt;- overlap_atac$name[adj_p_values &lt; .05]
#write.table(diff3_atac, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1&quot;)

# We perform a posthoc test
#wilcox_list = list()
wilcox_list &lt;- map(seq.int(1:length(hiv1_atac)), function(i){
  tf &lt;- hiv1_atac[i]
  #print(tf)
  wilcox &lt;- pairwise.wilcox.test(pull(k3_df[tf]), pull(k3_df[&quot;ht_cluster&quot;]), 
                                 p.adjust.method = &quot;fdr&quot;)
  list(TF = tf, wilcox = wilcox)
})
Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties

Warning in wilcox.test.default(xi, xj, paired = paired, ...): cannot compute
exact p-value with ties
  
#length(wilcox_list)
#wilcox_list</code></pre>
<hr />
</details>
<hr />
</div>
<div id="patient-2" class="section level3">
<h3>Patient 2</h3>
<pre class="r"><code>#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV2_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht2 &lt;- Heatmap(t(mtx), 
             column_title = &quot;Patient 2, HIV2_CSF&quot;, 
             column_title_gp = gpar(fontsize = 15),
             #column_split = paste0(&quot;cluster&quot;, hiv2$predicted.id),#cluster_anno,
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la,
             #top_annotation = HeatmapAnnotation(df = cluster_anno, name = &quot;predicted cluster&quot;)
             ) 
             #top_annotation = column_annotation) 
ht_pat2 &lt;- draw(ht2)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
<strong>Kruskal_Wallis test &amp; Pairwise-Wilcox test for Patient 2</strong>
</summary>
<p>Performing Kruskal-Wallis-Test to find differences between the three clusters in patient two yields 71 TFs out of all 125 TFs which are differentially active between the three clusters(p-value &lt; 0.05). Of the 24 TFs with differential binding scores in the C20 cell line, nineteen have significantly different activity between the clusters.</p>
<p>Now that it is known that there are significant differences between clusters, the next step is to identify which cluster/s are different from which other cluster/s, using Pairwise-Wilcox-Test, a non-parametric test. Again, because we repeat the test we will correct for multiple hypothesis testing using the FDR method.</p>
<pre class="r"><code># extract the cells from the 3 clusters 
# get the cell names of cluster one
c1 &lt;- colnames(t(mtx[column_order(ht_pat2)[[1]],]))
# get the cell names of cluster two
c2 &lt;- colnames(t(mtx[column_order(ht_pat2)[[2]],]))
# get the cell names of cluster 3
c3 &lt;- colnames(t(mtx[column_order(ht_pat2)[[3]],]))


# add cluster identities to Seurat object metadata and return metadata as dataframe
pat2 &lt;- hiv7_scenic@meta.data %&gt;% rownames_to_column(&quot;cell&quot;)
pat2 &lt;- pat2 %&gt;% 
  filter(orig.ident == &quot;HIV2_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(c1,c2,c3)) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c1, &quot;c1&quot;, ifelse(cell %in% c2, &quot;c2&quot;, &quot;c3&quot;))) 


# check all 125 TFs
# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(pat2[tf]) ~ ht_cluster, pat2))[&quot;p.value&quot;]
  #list(tf = tf, p_value = p_values)
})


# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)
length(adj_p_values[adj_p_values&lt;0.05])
[1] 71
## [1] 66

# get the TFs taht are differntially active
diff_hiv2 &lt;- colnames(auc_mtx)[adj_p_values &lt; 0.05]


# check 24 TFs with Kruskal Wallis test
p_values &lt;- map(seq.int(1:24), function(n){
  tf &lt;- overlap_atac$name[n]
  p_values &lt;- unlist(kruskal.test(pull(pat2[tf]) ~ ht_cluster, pat2))[&quot;p.value&quot;]
})
#sign_p_values &lt;- p_values[p_values &lt; 0.01]

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values&lt;0.05])
[1] 19
## [1] 18

# get the TFs that are differentially expressed
hiv2_atac &lt;- overlap_atac$name[adj_p_values &lt; .05]</code></pre>
<hr />
</details>
<hr />
</div>
<div id="patient-3" class="section level3">
<h3>Patient 3</h3>
<pre class="r"><code>df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV3_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht3 &lt;- Heatmap(t(mtx), 
             column_title = &quot;Patient 3, HIV3_CSF&quot;, 
             column_title_gp = gpar(fontsize = 15),
             #column_split = paste0(&quot;cluster&quot;, hiv2$predicted.id),#cluster_anno,
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la,
             #top_annotation = HeatmapAnnotation(df = cluster_anno, name = &quot;predicted cluster&quot;)
             ) 
             #top_annotation = column_annotation) ich
ht_pat3 &lt;- draw(ht3)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-13-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-13-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
<strong>Kruskal_Wallis test &amp; Pairwise-Wilcox test for Patient 3</strong>
</summary>
<p>Performing Kruskal-Wallis-Test to find differences between the three clusters in patient three yields 66 TFs out of all 125 TFs which are differentially active between the three clusters(p-value &lt; 0.05). Of the 24 TFs with differential binding scores in the C20 cell line, eighteen have significantly different activity between the clusters.</p>
<p>Now that it is known that there are significant differences between clusters, the next step is to identify which cluster/s are different from which other cluster/s, using Pairwise-Wilcox-Test, a non-parametric test. Again, because we repeat the test we will correct for multiple hypothesis testing using the FDR method.</p>
<pre class="r"><code># extract the cells from the 3 clusters 
# get the cell names of cluster one
c1 &lt;- colnames(t(mtx[column_order(ht_pat3)[[1]],]))
# get the cell names of cluster two
c2 &lt;- colnames(t(mtx[column_order(ht_pat3)[[2]],]))
# get the cell names of cluster 3
c3 &lt;- colnames(t(mtx[column_order(ht_pat3)[[3]],]))


# add cluster identities to Seurat object metadata and return metadata as dataframe
pat3 &lt;- hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV3_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(c1,c2,c3)) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c1, &quot;c1&quot;, ifelse(cell %in% c2, &quot;c2&quot;, &quot;c3&quot;))) 


# check all 125 TFs
# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(pat3[tf]) ~ ht_cluster, pat3))[&quot;p.value&quot;]
  #list(tf = tf, p_value = p_values)
})


# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)
length(adj_p_values[adj_p_values&lt;0.05])
[1] 66
## [1] 66

# get the TFs taht are differntially active
diff_hiv3 &lt;- colnames(auc_mtx)[adj_p_values &lt; 0.05]


# check 24 TFs with Kruskal Wallis test
p_values &lt;- map(seq.int(1:24), function(n){
  tf &lt;- overlap_atac$name[n]
  p_values &lt;- unlist(kruskal.test(pull(pat3[tf]) ~ ht_cluster, pat3))[&quot;p.value&quot;]
})
#sign_p_values &lt;- p_values[p_values &lt; 0.01]

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values&lt;0.05])
[1] 18
## [1] 18

# get the TFs that are differentially expressed
hiv3_atac &lt;- overlap_atac$name[adj_p_values &lt; .05]</code></pre>
<hr />
</details>
<hr />
</div>
<div id="comparison-between-all-3-patients" class="section level3">
<h3>Comparison between all 3 patients</h3>
<pre class="r"><code># patient 1
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV1_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht1 &lt;- Heatmap(t(mtx), 
             column_title = &quot;Patient 1, HIV1_CSF&quot;, 
             column_title_gp = gpar(fontsize = 15),
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la
             )
             #top_annotation = column_annotation) </code></pre>
<pre class="r"><code>ht_list &lt;- ht1 + ht2 + ht3
draw(ht_list, ht_gap= unit(2, &quot;cm&quot;), 
     column_title = &quot;Independent k-means clustering (k = 3) for all patients&quot;,
     column_title_gp = gpar(fontsize = 20) )</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-16-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-16-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-16-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-16-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
</div>
</div>
<div id="different-activities-between-clusters" class="section level1">
<h1>Different activities between clusters</h1>
<p>The three clusters obtained with k-means clustering for all three patients separately were compared to see if there is a pattern in TF activity that can be found in all three patients. The question was if the same group of TFs characterize the three clusters in all three patients or if they are different. To answer this question, Kruskal-Wallis-Test was used to determine the TFs which are significantly different between clusters and then a pairwise Wilcox test was used to determine which clusters are different.</p>
<p>In the Venn diagram below it can be seen that 46 TFs with significantly different activities between clusters overlap between patients (see figure, left). Additionally, eleven of the TFs found to have differential binding scores between infection conditions in C20 microglia cell line overlap between patients (see figure, right). These eleven TFs are the ones we are most interested in, since we want to see if we can find patterns that correlate with the binding scores in C20 microglia.</p>
<pre class="r"><code>p1 &lt;- ggVennDiagram(list(hiv1_diff, diff_hiv2, diff_hiv3),
              label_alpha = 0,
              category.names= c(&quot;HIV1&quot;, &quot;HIV2&quot;, &quot;HIV3&quot;),
              #set.size = 20,
              label = &quot;count&quot;,
              label_size = 6) +
   scale_fill_distiller(palette = &quot;RdBu&quot;) +
  labs(title = &quot;All TFs with different activity&quot;)</code></pre>
<pre class="r"><code>p2 &lt;- ggVennDiagram(list(hiv1_atac, hiv2_atac, hiv3_atac),
              label_alpha = 0,
              category.names= c(&quot;HIV1&quot;, &quot;HIV2&quot;, &quot;HIV3&quot;),
              #set.size = 20,
              label = &quot;count&quot;,
              label_size = 6) +
   scale_fill_distiller(palette = &quot;RdBu&quot;) +
  labs(title = &quot;TFs with differential binding scores&quot;)</code></pre>
<pre class="r"><code>ggarrange(p1, p2, ncol = 2, nrow = 1)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-19-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-19-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-19-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="eleven-tfs-with-different-activities" class="section level2 tabset">
<h2 class="tabset">Eleven TFs with different activities</h2>
<p>In the boxplots below it can be seen that in all three patients there seems to be one cluster which is different from two other clusters. It seems that ETS1, REST, and YY1 are more active in clusters where all other TFs are less active and vice versa. All three TFs are repressors, however, this alone does not explain their correlation, because for example KLF4 and MXI1 are also repressors.</p>
<pre class="r"><code>intersect &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/intersect_TFs_different_across_patients_3_clusters&quot;)[[&quot;x&quot;]]</code></pre>
<div id="patient-1-1" class="section level3">
<h3>Patient 1</h3>
<pre class="r"><code>df &lt;- k3_df


boxplot_list &lt;- map(intersect, function(tf) {
  ggplot() +
    geom_violin(aes(x = df %&gt;% pull(&quot;ht_cluster&quot;), y = df %&gt;% pull(tf), 
                     fill = df %&gt;% pull(&quot;ht_cluster&quot;), alpha = .6)) +
    geom_boxplot(aes(x = df %&gt;% pull(&quot;ht_cluster&quot;), y = df %&gt;% pull(tf)), 
                 alpha = 0) +
    stat_pvalue_manual(df %&gt;% 
                        pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;% 
                         add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-21-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-21-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
#annotate_figure(figure, top = text_grob(&quot;Patient 1 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)&quot;, 
 #              color = &quot;black&quot;, face = &quot;bold&quot;, size = 25))</code></pre>
<pre class="r"><code>ctcf1 &lt;- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_violin(aes(fill = ht_cluster, alpha = .6)) +
  geom_boxplot(alpha = 0) +
  stat_pvalue_manual(df %&gt;% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %&gt;% 
                       add_xy_position())
ctcf1 &lt;- ctcf1 + labs(title = &quot;CTCF activity in patient 1&quot;)</code></pre>
<hr />
</div>
<div id="patient-2-1" class="section level3">
<h3>Patient 2</h3>
<pre class="r"><code>df &lt;- pat2

boxplot_list &lt;- map(intersect, function(tf) {
  ggplot() +
    geom_violin(aes(x = df %&gt;% pull(&quot;ht_cluster&quot;), y = df %&gt;% pull(tf), 
                     fill = df %&gt;% pull(&quot;ht_cluster&quot;), alpha = .6)) +
    geom_boxplot(aes(x = df %&gt;% pull(&quot;ht_cluster&quot;), y = df %&gt;% pull(tf)), 
                 alpha = 0) +
    stat_pvalue_manual(df %&gt;% 
                        pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;% 
                         add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-23-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-23-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-23-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-23-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-23-1.png" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-23-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-23-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-23-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-23-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
#annotate_figure(figure, top = text_grob(&quot;Patient 2 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)&quot;, 
 #              color = &quot;black&quot;, face = &quot;bold&quot;, size = 25))</code></pre>
<pre class="r"><code>ctcf2 &lt;- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_violin(aes(fill = ht_cluster, alpha = .6)) +
  geom_boxplot(alpha = 0) +
  stat_pvalue_manual(df %&gt;% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %&gt;% 
                       add_xy_position())
ctcf2 &lt;- ctcf2 + labs(title = &quot;CTCF activity in patient 2&quot;)</code></pre>
<hr />
</div>
<div id="patient-3-1" class="section level3">
<h3>Patient 3</h3>
<pre class="r"><code>df &lt;- pat3

boxplot_list &lt;- map(intersect, function(tf) {
  ggplot() +
    geom_violin(aes(x = df %&gt;% pull(&quot;ht_cluster&quot;), y = df %&gt;% pull(tf), 
                     fill = df %&gt;% pull(&quot;ht_cluster&quot;), alpha = .6)) +
    geom_boxplot(aes(x = df %&gt;% pull(&quot;ht_cluster&quot;), y = df %&gt;% pull(tf)), 
                 alpha = 0) +
    stat_pvalue_manual(df %&gt;% 
                        pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;% 
                         add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-25-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-25-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>

#annotate_figure(figure, top = text_grob(&quot;Patient 3 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)&quot;, 
 #              color = &quot;black&quot;, face = &quot;bold&quot;, size = 25))</code></pre>
<pre class="r"><code>ctcf3 &lt;- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_violin(aes(fill = ht_cluster, alpha = .6)) +
  geom_boxplot(alpha = 0) +
  stat_pvalue_manual(df %&gt;% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %&gt;% 
                       add_xy_position())
ctcf3 &lt;- ctcf3 + labs(title = &quot;CTCF activity in patient 3&quot;)</code></pre>
<hr />
</div>
</div>
<div id="ctcf-expression-across-patients" class="section level2">
<h2>CTCF expression across patients</h2>
<p>CTCF has been implicated to regulate HIV latency <span class="citation">(<a href="#ref-Jeffereys" role="doc-biblioref">Jefferys SR 2021</a>)</span>. In the boxplots below it can be seen that there is a significant difference between the first cluster and the other two clusters in patient1 and patient2. Similarly in patient3 there is a difference, however the difference is not as significant. This might imply that the cluster contains a group of cells which are latently infected and, therefore, CTCF is more active.</p>
<pre class="r"><code>ggarrange(ctcf1, ctcf2, ctcf3, ncol = 3, nrow = 1)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-27-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-27-1">
Past versions of unnamed-chunk-27-1.png
</button>
</p>
<div id="fig-unnamed-chunk-27-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-27-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-27-1.png" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-27-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="correlation-between-clusters" class="section level2 tabset">
<h2 class="tabset">Correlation between clusters</h2>
<p>The average TF activity of all 125 TFs was computed for every cluster. Based on the average TF activity in each cluster, the correlation between clusters was computed. The scatter plot below shows the average expression for each TF across all three clusters in all three patients and the corresponding correlation coefficients. The correlation matrix visualizes the Spearman correlation coefficient between all clusters. Black numbers indicate the not-significant p-values. Clusters two and three of all patients seem to be highly correlated, with cluster two of patient3 being the exception. Clusters one of patient2 and patient3 are highly correlated with each other, but less correlated with clusters two and three. Cluster one of patient2 is again an exception, because it is not as highly correlated with clusters one of patient2 and patient3, but still more correlated than with clusters two and three.</p>
<p>These results led to the hypothesis that the clusters one, which showed different TF activity from clusters two and three correspond to a different infection state. Cells of clusters one could be latently infected cells. This hypothesis would be supported by having a look at the role of TFs with different activity in clusters one (see discussion).</p>
<pre class="r"><code># read in 3 Clusters for each of the 3 HIV+ patients

# HIV1_CSF
hiv1_c1 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv1_c2 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv1_c3 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]

# HIV2_CSF
hiv2_c1 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster1.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv2_c2 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster2.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv2_c3 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster3.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]

# HIV3_CSF
hiv3_c1 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster1.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv3_c2 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster2.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv3_c3 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster3.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]

cluster_anno &lt;- c(hiv1_c1, hiv1_c2, hiv1_c3, hiv2_c1, hiv2_c2, hiv2_c3, hiv3_c1, hiv3_c2, hiv3_c3)</code></pre>
<pre class="r"><code># add cluster information to TF activity
auc_df &lt;- as.data.frame(auc_mtx)

auc_df &lt;- auc_df %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(cell %in% cluster_anno) %&gt;% 
  mutate(ht_cluster = case_when(cell %in% hiv1_c1 ~ &quot;hiv1_c1&quot;, 
         cell %in% hiv1_c2 ~ &quot;hiv1_c2&quot;, 
         cell %in% hiv1_c3 ~ &quot;hiv1_c3&quot;,
         cell %in% hiv2_c1 ~ &quot;hiv2_c1&quot;,
         cell %in% hiv2_c2 ~ &quot;hiv2_c2&quot;,
         cell %in% hiv2_c3 ~ &quot;hiv2_c3&quot;,
         cell %in% hiv3_c1 ~ &quot;hiv3_c1&quot;,
         cell %in% hiv3_c2 ~ &quot;hiv3_c2&quot;,
         cell %in% hiv3_c3 ~ &quot;hiv3_c3&quot;))</code></pre>
<details>
<summary>
<strong>Correlation computation</strong>
</summary>
<p>Dataframe of average activity value for each TF in each patient and clusters.</p>
<pre class="r"><code># correlation matrix
df &lt;- auc_df %&gt;%
  group_by(ht_cluster) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% # calculate mean TF activity for each cluster
  gather(TF, value, -ht_cluster) %&gt;% 
  spread(ht_cluster, value) %&gt;% 
  column_to_rownames(&quot;TF&quot;) 

df %&gt;% head %&gt;% kable (caption = &quot;Average TF activity for each cluster&quot;) </code></pre>
<table>
<caption>Average TF activity for each cluster</caption>
<colgroup>
<col width="8%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">hiv1_c1</th>
<th align="right">hiv1_c2</th>
<th align="right">hiv1_c3</th>
<th align="right">hiv2_c1</th>
<th align="right">hiv2_c2</th>
<th align="right">hiv2_c3</th>
<th align="right">hiv3_c1</th>
<th align="right">hiv3_c2</th>
<th align="right">hiv3_c3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AHR</td>
<td align="right">0.0327704</td>
<td align="right">0.0534783</td>
<td align="right">0.0293072</td>
<td align="right">0.0275802</td>
<td align="right">0.0263492</td>
<td align="right">0.0222778</td>
<td align="right">0.0133156</td>
<td align="right">0.0075556</td>
<td align="right">0.0449111</td>
</tr>
<tr class="even">
<td align="left">ATF3</td>
<td align="right">0.0684460</td>
<td align="right">0.0965516</td>
<td align="right">0.0944977</td>
<td align="right">0.0487516</td>
<td align="right">0.0959799</td>
<td align="right">0.0910900</td>
<td align="right">0.0488404</td>
<td align="right">0.0828738</td>
<td align="right">0.1026111</td>
</tr>
<tr class="odd">
<td align="left">BACH1</td>
<td align="right">0.0745916</td>
<td align="right">0.0957086</td>
<td align="right">0.0960815</td>
<td align="right">0.0470817</td>
<td align="right">0.0956613</td>
<td align="right">0.0900737</td>
<td align="right">0.0491457</td>
<td align="right">0.0677356</td>
<td align="right">0.1003448</td>
</tr>
<tr class="even">
<td align="left">BCL11A</td>
<td align="right">0.0453606</td>
<td align="right">0.0580407</td>
<td align="right">0.0544656</td>
<td align="right">0.0336095</td>
<td align="right">0.0582619</td>
<td align="right">0.0503748</td>
<td align="right">0.0308033</td>
<td align="right">0.0406827</td>
<td align="right">0.0519544</td>
</tr>
<tr class="odd">
<td align="left">BCL6</td>
<td align="right">0.0300111</td>
<td align="right">0.0629203</td>
<td align="right">0.0389951</td>
<td align="right">0.0135000</td>
<td align="right">0.0646071</td>
<td align="right">0.0232917</td>
<td align="right">0.0162400</td>
<td align="right">0.0441667</td>
<td align="right">0.0393667</td>
</tr>
<tr class="even">
<td align="left">BHLHE40</td>
<td align="right">0.0453855</td>
<td align="right">0.0353485</td>
<td align="right">0.0268406</td>
<td align="right">0.0459291</td>
<td align="right">0.0220228</td>
<td align="right">0.0297271</td>
<td align="right">0.0300765</td>
<td align="right">0.0320580</td>
<td align="right">0.0355449</td>
</tr>
</tbody>
</table>
<p>Correlation matrix:</p>
<pre class="r"><code># save correlation matrix
cor_mtx &lt;- df%&gt;% cor(method = &quot;spearman&quot;)

# show the correlation matrix
cor_mtx %&gt;% kable(caption = &quot;Correlation matrix of average TF acivity between all clusters&quot;)</code></pre>
<table>
<caption>Correlation matrix of average TF acivity between all clusters</caption>
<colgroup>
<col width="8%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">hiv1_c1</th>
<th align="right">hiv1_c2</th>
<th align="right">hiv1_c3</th>
<th align="right">hiv2_c1</th>
<th align="right">hiv2_c2</th>
<th align="right">hiv2_c3</th>
<th align="right">hiv3_c1</th>
<th align="right">hiv3_c2</th>
<th align="right">hiv3_c3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">hiv1_c1</td>
<td align="right">1.0000000</td>
<td align="right">0.6908184</td>
<td align="right">0.7338171</td>
<td align="right">0.6320663</td>
<td align="right">0.7509155</td>
<td align="right">0.7127909</td>
<td align="right">0.7812705</td>
<td align="right">0.6970372</td>
<td align="right">0.6669821</td>
</tr>
<tr class="even">
<td align="left">hiv1_c2</td>
<td align="right">0.6908184</td>
<td align="right">1.0000000</td>
<td align="right">0.9300891</td>
<td align="right">0.3215460</td>
<td align="right">0.9176319</td>
<td align="right">0.9018986</td>
<td align="right">0.5112442</td>
<td align="right">0.7127695</td>
<td align="right">0.9402208</td>
</tr>
<tr class="odd">
<td align="left">hiv1_c3</td>
<td align="right">0.7338171</td>
<td align="right">0.9300891</td>
<td align="right">1.0000000</td>
<td align="right">0.3216996</td>
<td align="right">0.9374359</td>
<td align="right">0.8709985</td>
<td align="right">0.5799816</td>
<td align="right">0.6956227</td>
<td align="right">0.9342239</td>
</tr>
<tr class="even">
<td align="left">hiv2_c1</td>
<td align="right">0.6320663</td>
<td align="right">0.3215460</td>
<td align="right">0.3216996</td>
<td align="right">1.0000000</td>
<td align="right">0.3724407</td>
<td align="right">0.3251283</td>
<td align="right">0.7078374</td>
<td align="right">0.3575233</td>
<td align="right">0.3021740</td>
</tr>
<tr class="odd">
<td align="left">hiv2_c2</td>
<td align="right">0.7509155</td>
<td align="right">0.9176319</td>
<td align="right">0.9374359</td>
<td align="right">0.3724407</td>
<td align="right">1.0000000</td>
<td align="right">0.8642416</td>
<td align="right">0.6434852</td>
<td align="right">0.6702748</td>
<td align="right">0.9229219</td>
</tr>
<tr class="even">
<td align="left">hiv2_c3</td>
<td align="right">0.7127909</td>
<td align="right">0.9018986</td>
<td align="right">0.8709985</td>
<td align="right">0.3251283</td>
<td align="right">0.8642416</td>
<td align="right">1.0000000</td>
<td align="right">0.5542488</td>
<td align="right">0.6640449</td>
<td align="right">0.8868380</td>
</tr>
<tr class="odd">
<td align="left">hiv3_c1</td>
<td align="right">0.7812705</td>
<td align="right">0.5112442</td>
<td align="right">0.5799816</td>
<td align="right">0.7078374</td>
<td align="right">0.6434852</td>
<td align="right">0.5542488</td>
<td align="right">1.0000000</td>
<td align="right">0.5082700</td>
<td align="right">0.5590695</td>
</tr>
<tr class="even">
<td align="left">hiv3_c2</td>
<td align="right">0.6970372</td>
<td align="right">0.7127695</td>
<td align="right">0.6956227</td>
<td align="right">0.3575233</td>
<td align="right">0.6702748</td>
<td align="right">0.6640449</td>
<td align="right">0.5082700</td>
<td align="right">1.0000000</td>
<td align="right">0.6788640</td>
</tr>
<tr class="odd">
<td align="left">hiv3_c3</td>
<td align="right">0.6669821</td>
<td align="right">0.9402208</td>
<td align="right">0.9342239</td>
<td align="right">0.3021740</td>
<td align="right">0.9229219</td>
<td align="right">0.8868380</td>
<td align="right">0.5590695</td>
<td align="right">0.6788640</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>
# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor &lt;- cor.mtest(cor_mtx, conf.level = 0.95, method = &quot;spearman&quot;)</code></pre>
<hr />
</details>
<div id="correlation-matrix" class="section level3">
<h3>Correlation matrix</h3>
<pre class="r"><code>df %&gt;%
  cor(method = &quot;spearman&quot;) %&gt;%
  corrplot(
     order = &quot;hclust&quot;, # hierarchical cluster order
     addrect = 4, # add boxes 
     p.mat = testCor$p, # add matrix of p-values
     insig = &quot;p-value&quot;, # adds p-values as digits, not crosses
     tl.col = &quot;black&quot;) # make labels black</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
<div id="scatter-matrix" class="section level3">
<h3>Scatter matrix</h3>
<pre class="r"><code>
GGally::ggpairs(df) + theme_bw() +
  labs(title = &quot;Scatter matrix for all conditions&quot;)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-33-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-33-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-33-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-33-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-33-1.png" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
</div>
<div id="tf-correlation" class="section level2 tabset">
<h2 class="tabset">TF correlation</h2>
<p>For the eleven TFs with different activity between clusters, the correlation between the TF activities was computed. The scatter matrix and correlation plot below show very nicely, that the three TFs YY1, REST, MXI1 are positively correlated, but negatively correlated with the rest of the TFs and vice versa. The fact that these TF activities correlate across all patients indicates that there might be a biological meaning behind the different clusters. Having a look at the function of TFs which are up- or downregulated in cluster one compared to the other clusters could provide further insights (see discussion).</p>
<details>
<summary>
<strong>Correlation computation</strong>
</summary>
<pre class="r"><code>intersect_atac &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_intersect_TFs_different_across_patients_3_clusters&quot;))[[&quot;x&quot;]]</code></pre>
<pre class="r"><code># correlation matrix
auc_df %&gt;%
  select(intersect_atac) %&gt;% head() %&gt;% kable(caption = &quot;TF activity of 11 TFs with different activity between clusters across all cells&quot;)</code></pre>
<table style="width:100%;">
<caption>TF activity of 11 TFs with different activity between clusters across all cells</caption>
<colgroup>
<col width="9%" />
<col width="7%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">NFIL3</th>
<th align="right">MXI1</th>
<th align="right">MEF2A</th>
<th align="right">REST</th>
<th align="right">KLF4</th>
<th align="right">YY1</th>
<th align="right">ETS1</th>
<th align="right">JUNB</th>
<th align="right">CEBPB</th>
<th align="right">ETV6</th>
<th align="right">MEF2C</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0.0870583</td>
<td align="right">0.20925</td>
<td align="right">0.2196393</td>
<td align="right">0.0238841</td>
<td align="right">0.0921104</td>
<td align="right">0.0197500</td>
<td align="right">0.0145476</td>
<td align="right">0.0773047</td>
<td align="right">0.1038080</td>
<td align="right">0.1446622</td>
<td align="right">0.0982794</td>
</tr>
<tr class="even">
<td align="right">0.0443263</td>
<td align="right">0.02050</td>
<td align="right">0.0806885</td>
<td align="right">0.0114203</td>
<td align="right">0.0518896</td>
<td align="right">0.0061176</td>
<td align="right">0.0248571</td>
<td align="right">0.0307901</td>
<td align="right">0.0396602</td>
<td align="right">0.0610569</td>
<td align="right">0.0627353</td>
</tr>
<tr class="odd">
<td align="right">0.0379207</td>
<td align="right">0.03800</td>
<td align="right">0.0707049</td>
<td align="right">0.0061739</td>
<td align="right">0.0351779</td>
<td align="right">0.0147206</td>
<td align="right">0.0172024</td>
<td align="right">0.0370068</td>
<td align="right">0.0425629</td>
<td align="right">0.0413378</td>
<td align="right">0.0407059</td>
</tr>
<tr class="even">
<td align="right">0.1232121</td>
<td align="right">0.08325</td>
<td align="right">0.2265246</td>
<td align="right">0.0081159</td>
<td align="right">0.0644663</td>
<td align="right">0.0115294</td>
<td align="right">0.0057024</td>
<td align="right">0.0790293</td>
<td align="right">0.1214656</td>
<td align="right">0.1409498</td>
<td align="right">0.1228382</td>
</tr>
<tr class="odd">
<td align="right">0.0913427</td>
<td align="right">0.32800</td>
<td align="right">0.2257377</td>
<td align="right">0.0168406</td>
<td align="right">0.0839509</td>
<td align="right">0.0132794</td>
<td align="right">0.0048333</td>
<td align="right">0.0752415</td>
<td align="right">0.1153256</td>
<td align="right">0.1528428</td>
<td align="right">0.1064265</td>
</tr>
<tr class="even">
<td align="right">0.0821259</td>
<td align="right">0.12000</td>
<td align="right">0.1593770</td>
<td align="right">0.0069565</td>
<td align="right">0.0881595</td>
<td align="right">0.0160588</td>
<td align="right">0.0087619</td>
<td align="right">0.0702709</td>
<td align="right">0.0730636</td>
<td align="right">0.1084281</td>
<td align="right">0.1075588</td>
</tr>
</tbody>
</table>
<p>Correlation matrix for TFs:</p>
<pre class="r"><code># compute correlation matrix
cor_mtx &lt;- auc_df %&gt;%
  select(intersect_atac) %&gt;% 
  cor(method = &quot;spearman&quot;)
# show correlation matrix
cor_mtx %&gt;% head() %&gt;% kable(caption = &quot;Correlation matrix for 11 TFs with different activity between clusters&quot;)</code></pre>
<table style="width:100%;">
<caption>Correlation matrix for 11 TFs with different activity between clusters</caption>
<colgroup>
<col width="4%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">NFIL3</th>
<th align="right">MXI1</th>
<th align="right">MEF2A</th>
<th align="right">REST</th>
<th align="right">KLF4</th>
<th align="right">YY1</th>
<th align="right">ETS1</th>
<th align="right">JUNB</th>
<th align="right">CEBPB</th>
<th align="right">ETV6</th>
<th align="right">MEF2C</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NFIL3</td>
<td align="right">1.0000000</td>
<td align="right">0.4032215</td>
<td align="right">0.8183930</td>
<td align="right">-0.5911676</td>
<td align="right">0.6815620</td>
<td align="right">-0.3964362</td>
<td align="right">-0.7678794</td>
<td align="right">0.7601330</td>
<td align="right">0.8658911</td>
<td align="right">0.8288826</td>
<td align="right">0.7851781</td>
</tr>
<tr class="even">
<td align="left">MXI1</td>
<td align="right">0.4032215</td>
<td align="right">1.0000000</td>
<td align="right">0.4011111</td>
<td align="right">-0.2642976</td>
<td align="right">0.2614033</td>
<td align="right">-0.2905129</td>
<td align="right">-0.3823019</td>
<td align="right">0.2142966</td>
<td align="right">0.3476688</td>
<td align="right">0.3748361</td>
<td align="right">0.3034267</td>
</tr>
<tr class="odd">
<td align="left">MEF2A</td>
<td align="right">0.8183930</td>
<td align="right">0.4011111</td>
<td align="right">1.0000000</td>
<td align="right">-0.6439536</td>
<td align="right">0.6433600</td>
<td align="right">-0.3566821</td>
<td align="right">-0.7786513</td>
<td align="right">0.7260030</td>
<td align="right">0.8762196</td>
<td align="right">0.8376745</td>
<td align="right">0.8513789</td>
</tr>
<tr class="even">
<td align="left">REST</td>
<td align="right">-0.5911676</td>
<td align="right">-0.2642976</td>
<td align="right">-0.6439536</td>
<td align="right">1.0000000</td>
<td align="right">-0.3769229</td>
<td align="right">0.5697669</td>
<td align="right">0.7459846</td>
<td align="right">-0.3949944</td>
<td align="right">-0.6127303</td>
<td align="right">-0.6172361</td>
<td align="right">-0.5367789</td>
</tr>
<tr class="odd">
<td align="left">KLF4</td>
<td align="right">0.6815620</td>
<td align="right">0.2614033</td>
<td align="right">0.6433600</td>
<td align="right">-0.3769229</td>
<td align="right">1.0000000</td>
<td align="right">-0.2119619</td>
<td align="right">-0.5623120</td>
<td align="right">0.7634697</td>
<td align="right">0.7505789</td>
<td align="right">0.6447148</td>
<td align="right">0.6498511</td>
</tr>
<tr class="even">
<td align="left">YY1</td>
<td align="right">-0.3964362</td>
<td align="right">-0.2905129</td>
<td align="right">-0.3566821</td>
<td align="right">0.5697669</td>
<td align="right">-0.2119619</td>
<td align="right">1.0000000</td>
<td align="right">0.5276046</td>
<td align="right">-0.2363438</td>
<td align="right">-0.3267780</td>
<td align="right">-0.3959500</td>
<td align="right">-0.3074553</td>
</tr>
</tbody>
</table>
<pre class="r"><code>
# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor &lt;- cor.mtest(cor_mtx, conf.level = 0.95, method = &quot;spearman&quot;)</code></pre>
<hr />
</details>
<div id="correlation-matrix-1" class="section level3">
<h3>Correlation matrix</h3>
<pre class="r"><code>df &lt;- auc_df %&gt;%
  select(intersect_atac) %&gt;% 
  cor(method = &quot;spearman&quot;) %&gt;%
  corrplot(
  order = &quot;hclust&quot;, 
  addrect = 2, # add boxes 
  p.mat = testCor$p, 
  insig = &quot;p-value&quot;,
  tl.col = &quot;black&quot;) </code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
<div id="scatter-matrix-1" class="section level3">
<h3>Scatter matrix</h3>
<pre class="r"><code>GGally::ggpairs(auc_df %&gt;%
  select(intersect_atac)) + 
  theme_bw() +
  labs(title = &quot;Scatter matrix for all TFs with different activity between clusters&quot;)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-38-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-38-1">
Past versions of unnamed-chunk-38-1.png
</button>
</p>
<div id="fig-unnamed-chunk-38-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-38-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-38-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-38-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-38-1.png" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-38-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
</div>
</div>
<div id="gene-expression-analysis" class="section level1">
<h1>Gene expression analysis</h1>
<p>We will now pool the two clusters which are significantly different from the third cluster in all patients. Consequently, in the following analysis we will compare Cluster1 to Cluster2 (pool of two similar clusters). The gene expression might give a hint about the identity of these two Myeloid2 clusters.</p>
<pre class="r"><code>hivmy2 &lt;- subset(hiv7, cell_type == &quot;Myeloid2&quot; &amp; orig.ident %in% c(&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;))

# add cluster identity to metadata
ht_cluster &lt;- (
  hivmy2@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c(hiv1_c1, hiv2_c1, hiv3_c3),
                             &quot;c1&quot;, 
                             &quot;c2&quot;)) %&gt;% 
  column_to_rownames(&quot;cell&quot;))[&quot;ht_cluster&quot;]

hivmy2 &lt;- AddMetaData(hivmy2, metadata = ht_cluster, col.name = &quot;ht_cluster&quot;)
#saveRDS(hivmy2,&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv_myeloid2_subclusters&quot;)</code></pre>
<div id="quality-of-the-myeloid-2-clusters" class="section level2">
<h2>Quality of the Myeloid 2 clusters</h2>
<p>There is an evident difference in library size and number of features between the two clusters. This could mean that there is a technical difference between the two clusters. Conversely, it could also mean that there is a biological difference. For example, if cells of cluster1 represent a latently infected group of Myeloid2 cells it could be that gene expression is reduced compared to healthy uninfected cells from cluster2 due to infection.</p>
<p>One way to investigate this further would be to have a look at the housekeeping genes in both clusters. If housekeeping genes were more highly expressed in one cluster than in the other this would indicate that the difference is not biological, but a technical artefact.</p>
<p>One problem that arises, because there is one cluster which has lower number of counts and features, is that further qualitative analysis of gene expression might be skewed as well. For this reason the following analysis on the difference between these two clusters will be mainly qualitative. Due to the low number of cells it is not possible to get significant results for the differential gene expression.</p>
<pre class="r"><code>p1 &lt;- hivmy2@meta.data %&gt;% 
  ggplot() +
  geom_boxplot(aes(x = ht_cluster, y = nFeature_RNA, fill = ht_cluster))

p2 &lt;- hivmy2@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = ht_cluster, y = nFeature_RNA, fill = ht_cluster)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1))

fig &lt;- ggarrange(p1, p2, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob(&quot;Quality of Clusters&quot;))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-40-1">
Past versions of unnamed-chunk-40-1.png
</button>
</p>
<div id="fig-unnamed-chunk-40-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-40-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-40-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-40-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-40-1.png" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="housekeeping-genes" class="section level2">
<h2>Housekeeping genes</h2>
<p>A list of 3804 houseekping genes were used from <span class="citation">(<a href="#ref-Eisenberg" role="doc-biblioref">Eisenberg E 2013</a>)</span>. A quantitative analysis might be difficult, because due to the small sample size and large feature number, we get very high p_values for all genes in differential gene expression analysis. Visualization of the housekeeping gene expression, unfortunately does not help either with the interpretation.</p>
<details>
<summary>
<strong>Boxplots of housekeeping gene expression</strong>
</summary>
<pre class="r"><code>hk_genes &lt;- (read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/HK_genes.txt&quot;))[[&quot;V1&quot;]]</code></pre>
<pre class="r"><code># filter only genes present in the scRNAseq count matrix
hk_genes_filt &lt;- hk_genes[hk_genes %in% rownames(hivmy2@assays$RNA@counts)]
length(hk_genes)
[1] 3804
length(hk_genes_filt)
[1] 3711

# Find genes from the housekeeping genes that are differentially expressed 
# between the clusters
hk_diff &lt;- FindAllMarkers(hivmy2, features = hk_genes_filt)
Idents(hivmy2) &lt;- &quot;ht_cluster&quot;
hk_diff &lt;- FindMarkers(hivmy2, features = hk_genes_filt, ident.1 = &quot;c1&quot;, ident.2 = &quot;c2&quot;)
hk_diff %&gt;% head
             p_val  avg_log2FC pct.1 pct.2 p_val_adj
DPP7   0.001063786  -0.9991177 0.000 0.259         1
RPL27  0.001354627 -47.9667965 1.000 0.991         1
RPL10A 0.001632447  -9.5926981 0.941 0.911         1
RPL6   0.001658084  -1.7550766 0.882 0.812         1
RPL30  0.002862495 -38.6641080 0.971 0.946         1
DDX21  0.003060688  -1.0742719 0.088 0.330         1</code></pre>
<pre class="r"><code>test &lt;- hk_genes_filt[1:20]
#VlnPlot(hivmy2, features = test, group.by = &quot;ht_cluster&quot;, ncol = 4)

df &lt;- as.data.frame(t(as.matrix((hivmy2@assays$RNA@data)))) %&gt;%
  select(test) %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;%
  mutate(ht_cluster = case_when(cell %in% c(hiv1_c1, hiv2_c1, hiv3_c3) ~ &quot;c1&quot;,
                                TRUE ~ &quot;c2&quot;)) %&gt;%
  column_to_rownames(&quot;cell&quot;)

boxplot_list &lt;- map(test, function(gene){
  ggplot() +
    geom_violin(aes(x = df %&gt;% pull(&quot;ht_cluster&quot;), y = df %&gt;% pull(gene), 
                     fill = df %&gt;% pull(&quot;ht_cluster&quot;))) +
    geom_boxplot(aes(x = df %&gt;% pull(&quot;ht_cluster&quot;), y = df %&gt;% pull(gene)), 
                 alpha = 0) +
    stat_pvalue_manual(df %&gt;% 
                        pairwise_wilcox_test(as.formula(paste0(gene, &quot; ~ ht_cluster&quot;))) %&gt;% 
                         add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(gene)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;))
})


do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 5, nrow = 4))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-43-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-43-1">
Past versions of unnamed-chunk-43-1.png
</button>
</p>
<div id="fig-unnamed-chunk-43-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-43-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-43-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-43-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-43-1.png" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-43-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-43-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</details>
</div>
</div>
<div id="appendix" class="section level1">
<h1>Appendix</h1>
<div id="determine-k-for-k-means-clustering" class="section level3 tabset">
<h3 class="tabset">Determine k for k-means clustering</h3>
<p>The plot of within cluster sum of squares shows that the sweet spot for the value of k is somewhere between 3 and 5. To determine the best k for k-means clustering k = 3, k = 4 and k = 5 were tried and the different outcomes compared. Since k = 4 and k = 5 does not yield an increased number of TFs which differentiate the different clusters and, therefore, do not add any insights we proceeded with k = 3 in the following analysis.</p>
<pre class="r"><code>df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                    orig.ident == &quot;HIV1_CSF&quot;)
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]

# Decide on a k for clustering
wss = sapply(2:15, function(k) {
  kmeans(x = t(mtx), centers = k)$tot.withinss
})

plot(2:15, wss, type = &quot;b&quot;, 
     xlab = &quot;Number of clusters k&quot;, 
     ylab = &quot;Total within-clusters sum of square&quot;)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-44-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-44-1">
Past versions of unnamed-chunk-44-1.png
</button>
</p>
<div id="fig-unnamed-chunk-44-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-44-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-44-1.png" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/bdbb921b65681909caf7252afe838d8b807681c3/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-44-1.png" target="_blank">bdbb921</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-44-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-44-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-44-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The results of using differnet k can be seen below. Click on the tabs to see heatmaps of the k-means clustering, as well as statistical test on different TF acitivities between clusters.</p>
<div id="k-means-clustering-with-k-3-1" class="section level4">
<h4>k-means Clustering with k = 3</h4>
<pre class="r"><code>#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV1_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_pat1 &lt;-draw(Heatmap(t(mtx), 
             column_title = &quot;Myeloid2 cells from patient 1, k = 3&quot;, 

             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la
             ))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-45-1">
Past versions of unnamed-chunk-45-1.png
</button>
</p>
<div id="fig-unnamed-chunk-45-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>             #top_annotation = column_annotation) </code></pre>
<p>After performing k-means cluster we extract the cells from each cluster. In the table below you can see the number of cells in each cluster.</p>
<pre class="r"><code>### Extract cells from each cluster


# if you draw the heatmap, the results of th eclustering will not change anymore
#png(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/analysis/Myeloid2_cluster/Myeloid2_3clusters.png&quot;, width = 15, height = 17, res = 1200)
#ht &lt;- draw(ht_pat1)
# get the columns names of cluster one
c1 &lt;- colnames(t(mtx[column_order(ht_pat1)[[1]],]))
#write.table(c1, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv&quot;, sep = &quot;\t&quot;)

# get the column names of cluster two
c2 &lt;- colnames(t(mtx[column_order(ht_pat1)[[2]],]))
#write.table(c2, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv&quot;, sep = &quot;\t&quot;)

# get the column names of cluster 3
c3 &lt;- colnames(t(mtx[column_order(ht_pat1)[[3]],]))
#write.table(c3, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv&quot;, sep = &quot;\t&quot;)</code></pre>
<div id="number-of-cells-in-each-cluster" class="section level5">
<h5>number of cells in each cluster</h5>
<pre class="r"><code># summary staticstics
hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV1_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(c1,c2,c3))%&gt;% 
  mutate(ht_cluster = case_when(cell %in% c1 ~ &quot;c1&quot;, 
                                cell %in% c2 ~ &quot;c2&quot;,
                                cell %in% c3 ~ &quot;c3&quot;)) %&gt;%
  group_by(ht_cluster) %&gt;% 
  summarize(count = n()) %&gt;% 
  kable(caption = &quot;number of cells in each cluster&quot;)</code></pre>
<table>
<caption>number of cells in each cluster</caption>
<thead>
<tr class="header">
<th align="left">ht_cluster</th>
<th align="right">count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">c1</td>
<td align="right">15</td>
</tr>
<tr class="even">
<td align="left">c2</td>
<td align="right">23</td>
</tr>
<tr class="odd">
<td align="left">c3</td>
<td align="right">34</td>
</tr>
</tbody>
</table>
</div>
<div id="are-there-significant-differences-in-tf-activity-between-the-three-clusters" class="section level5">
<h5>Are there significant differences in TF activity between the three clusters?</h5>
<p>Performing Kruskal-Wallis-Test to find differences between the three clusters in patient one yields 54 TFs of all 125 TFs which are differentially active between the three clusters(p-value &lt; 0.05). Of the 24 TFs with differential binding scores in the C20 cell line, sixteen have significantly different activity between the clusters.</p>
<p>Now that it is known that there are significant differences between clusters, the next step is to identify which cluster/s are different from which other cluster/s.</p>
<p>Pairwise-Wilcox-Test is a non-parametric pairwise test. Again, because we repeat the test we will correct for multiple hypothesis testing using the FDR method. The results are shown in boxplots below.</p>
</details>
<details>
<summary>
Kruskal-Wallis-Test, p-value adjustment, posthoc tests
</summary>
<p><strong>Kruskal-Wallis Test and p-value adjustments for all 125 TFs</strong></p>
<pre class="r"><code># non-parametric alternative to ANOVA -&gt; Kruskal_wallis test
k3_df &lt;- hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV1_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(c1,c2,c3)) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c1, &quot;c1&quot;, ifelse(cell %in% c2, &quot;c2&quot;, &quot;c3&quot;))) 
# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))[&quot;p.value&quot;]
  #list(tf = tf, p_value = p_values)
})

# plot distribution of p_values
hist(as.numeric(unname(unlist(p_values))), 
     main = &quot;distribution of p_values&quot;,
     xlab = &quot;p_values&quot;,
     breaks = 20)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-48-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-48-1">
Past versions of unnamed-chunk-48-1.png
</button>
</p>
<div id="fig-unnamed-chunk-48-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-48-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-48-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-48-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-48-1.png" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/bdbb921b65681909caf7252afe838d8b807681c3/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-48-1.png" target="_blank">bdbb921</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-48-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
sign3 &lt;- adj_p_values[adj_p_values&lt;.05]
length(adj_p_values[adj_p_values&lt;0.05])
[1] 64

# get TFs that are significantly different
hiv1_diff &lt;- colnames(auc_mtx)[adj_p_values &lt; .05]

#write.table(diff3, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1&quot;)</code></pre>
<ul>
<li><strong>Kruskal-Wallis test and p-value adjustment for 24 TFs also identified by ATAC-seq</strong></li>
</ul>
<pre class="r"><code># check 24 TFs with Kruskal Wallis test
p_values &lt;- map(seq.int(1:24), function(n){
  tf &lt;- overlap_atac$name[n]
  p_values &lt;- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))[&quot;p.value&quot;]
})
#sign_p_values &lt;- p_values[p_values &lt; 0.01]

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values&lt;0.05])
[1] 16

# get TFs that are differentially expressed
hiv1_atac &lt;- overlap_atac$name[adj_p_values &lt; .05]
#write.table(diff3_atac, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1&quot;)</code></pre>
<ul>
<li><strong>Posthoc tests for k = 3</strong></li>
</ul>
<p>The posthoc test was performed for the 16 TFs with significantly different activity between the 3 clusters.</p>
<pre class="r"><code># We perform a posthoc test
#wilcox_list = list()
wilcox_list &lt;- map(seq.int(1:length(hiv1_atac)), function(i){
  tf &lt;- hiv1_atac[i]
  #print(tf)
  wilcox &lt;- pairwise.wilcox.test(pull(k3_df[tf]), pull(k3_df[&quot;ht_cluster&quot;]), 
                                 p.adjust.method = &quot;fdr&quot;)
  list(TF = tf, wilcox = wilcox)
})
  
#length(wilcox_list)
#wilcox_list</code></pre>
<hr />
</details>
<hr />
</div>
</div>
<div id="k-means-clustering-with-k-4" class="section level4">
<h4>K-means clustering with k = 4</h4>
<pre class="r"><code>#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV1_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht &lt;- draw(Heatmap(t(mtx), 
             column_title = &quot;Myeloid2 cells from patient 2, k = 4&quot;, 

             column_km = 4,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la
             )) </code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-51-1">
Past versions of unnamed-chunk-51-1.png
</button>
</p>
<div id="fig-unnamed-chunk-51-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>             #top_annotation = column_annotation) 
</code></pre>
<pre class="r"><code>### Extract cells from each cluster of k = 4

# get the columns names of cluster one
cl1 &lt;- colnames(t(mtx[column_order(ht)[[1]],]))

# get the column names of cluster two
cl2 &lt;- colnames(t(mtx[column_order(ht)[[2]],]))


# get the column names of cluster 3
cl3 &lt;- colnames(t(mtx[column_order(ht)[[3]],]))

cl4 &lt;- colnames(t(mtx[column_order(ht)[[4]],]))</code></pre>
<pre class="r"><code># summary staticstics
hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV1_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(c1,c2,c3)) %&gt;% 
  mutate(ht_cluster = case_when(cell %in% cl1 ~ &quot;c1&quot;, 
                                cell %in% cl2 ~ &quot;c2&quot;,
                                cell %in% cl3 ~ &quot;c3&quot;,
                                cell %in% cl4 ~ &quot;c4&quot;)) %&gt;% 
  group_by(ht_cluster) %&gt;% 
  summarize(count = n()) %&gt;% 
  kable(caption = &quot;number of cells in each cluster&quot;)</code></pre>
<table>
<caption>number of cells in each cluster</caption>
<thead>
<tr class="header">
<th align="left">ht_cluster</th>
<th align="right">count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">c1</td>
<td align="right">8</td>
</tr>
<tr class="even">
<td align="left">c2</td>
<td align="right">22</td>
</tr>
<tr class="odd">
<td align="left">c3</td>
<td align="right">21</td>
</tr>
<tr class="even">
<td align="left">c4</td>
<td align="right">21</td>
</tr>
</tbody>
</table>
<div id="are-there-significant-differences-in-tf-activity-between-the-four-clusters" class="section level5">
<h5>Are there significant differences in TF activity between the four clusters?</h5>
<p>Performing Kruskal-Wallis-Test to find differences between the four clusters in patient one yields 51 TFs of all 125 TFs which are differentially active between the three clusters(p-value &lt; 0.05). Of the 24 TFs with differential binding scores in the C20 cell line, fourteen have significantly different activity between the clusters.</p>
<p>Now that it is known that there are significant differences between clusters, the next step is to identify which cluster/s are different from which other cluster/s. For this pairwise Wilcox test was used.</p>
</details>
<details>
<summary>
Kruskal-Wallis-Test, p-value adjustment, posthoc tests
</summary>
<ul>
<li><strong>Kruskal-Wallis Test and p-value adjustments for all 125 TFs</strong></li>
</ul>
<pre class="r"><code>
k4_df &lt;- hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV1_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(cl1,cl2,cl3,cl4)) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% cl1, &quot;cl1&quot;, 
                             ifelse(cell %in% cl2, &quot;cl2&quot;, 
                                    ifelse(cell %in% cl3, &quot;cl3&quot;, &quot;cl4&quot;))))

# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(k4_df[tf]) ~ ht_cluster, k4_df))[&quot;p.value&quot;]
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)

# how many transcription factors are significantly different between the clusters?
sign &lt;- adj_p_values[adj_p_values &lt; 0.05]
length(adj_p_values[adj_p_values&lt;0.05])
[1] 55


# get the TFs that are differentially expressed
diff4 &lt;- colnames(auc_mtx)[adj_p_values &lt; 0.05]

#write.table(diff4, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=4&quot;)</code></pre>
<ul>
<li><strong>Kruskal-Wallis Test and p-value adjustments for 14 TFs with differential binding scores</strong></li>
</ul>
<pre class="r"><code># check 24 TFs with Kruskal Wallis test
p_values &lt;- map(seq.int(1:24), function(n){
  tf &lt;- overlap_atac$name[n]
  p_values &lt;- unlist(kruskal.test(pull(k4_df[tf]) ~ ht_cluster, k4_df))[&quot;p.value&quot;]
})
#sign_p_values &lt;- p_values[p_values &lt; 0.01]

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values&lt;0.05])
[1] 15

# get the TFs that are differentially expressed
diff4_atac &lt;- overlap_atac$name[adj_p_values &lt; .05]

#write.table(diff4_atac, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=4&quot;)</code></pre>
<ul>
<li><strong>Pairwise-Wilcox_Test for k = 4</strong></li>
</ul>
<p>Posthoc test for the 16 TFs with significantly different activity between the four clusters.</p>
<pre class="r"><code># We perform a posthoc test
#wilcox_list = list()
wilcox_list &lt;- map(seq.int(1:length(diff4_atac)), function(i){
  tf &lt;- diff4_atac[i]
  print(tf)
  wilcox &lt;- pairwise.wilcox.test(pull(k4_df[tf]), pull(k4_df[&quot;ht_cluster&quot;]), 
                                 p.adjust.method = &quot;fdr&quot;)
  list(TF = tf, wilcox = wilcox)
})
[1] &quot;NFIL3&quot;
[1] &quot;MXI1&quot;
[1] &quot;CREB1&quot;
[1] &quot;NFATC2&quot;
[1] &quot;MEF2A&quot;
[1] &quot;REST&quot;
[1] &quot;LEF1&quot;
[1] &quot;YY1&quot;
[1] &quot;MYB&quot;
[1] &quot;ETS1&quot;
[1] &quot;JUNB&quot;
[1] &quot;CEBPB&quot;
[1] &quot;CTCF&quot;
[1] &quot;ETV6&quot;
[1] &quot;MEF2C&quot;
  
#length(wilcox_list)
#wilcox_list</code></pre>
<hr />
</details>
<hr />
</div>
</div>
<div id="k-means-clustering-with-k-5" class="section level4">
<h4>K-means Clustering with k = 5</h4>
<pre class="r"><code>#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV1_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht &lt;- draw(Heatmap(t(mtx), 
             column_title = &quot;Myeloid2 cells from patient 1, k = 5&quot;, 

             column_km = 5,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la
             )) </code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-57-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-57-1">
Past versions of unnamed-chunk-57-1.png
</button>
</p>
<div id="fig-unnamed-chunk-57-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-57-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-57-1.png" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-57-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-57-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-57-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>##### Extract cells from each cluster for k= 5

# get the columns names of cluster one
clu1 &lt;- colnames(t(mtx[column_order(ht)[[1]],]))

# get the column names of cluster two
clu2 &lt;- colnames(t(mtx[column_order(ht)[[2]],]))


# get the column names of cluster 3
clu3 &lt;- colnames(t(mtx[column_order(ht)[[3]],]))

clu4 &lt;- colnames(t(mtx[column_order(ht)[[4]],]))

clu5 &lt;- colnames(t(mtx[column_order(ht)[[5]], ]))</code></pre>
<pre class="r"><code># summary staticstics
hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV1_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(c1,c2,c3)) %&gt;% 
  mutate(ht_cluster = case_when(cell %in% clu1 ~ &quot;c1&quot;, 
                                cell %in% clu2 ~ &quot;c2&quot;,
                                cell %in% clu3 ~ &quot;c3&quot;,
                                cell %in% clu4 ~ &quot;c4&quot;,
                                cell %in% clu5 ~ &quot;c5&quot;)) %&gt;% 
  group_by(ht_cluster) %&gt;% 
  summarize(count = n()) %&gt;% 
  kable(caption = &quot;number of cells in each cluster&quot;)</code></pre>
<table>
<caption>number of cells in each cluster</caption>
<thead>
<tr class="header">
<th align="left">ht_cluster</th>
<th align="right">count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">c1</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">c2</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">c3</td>
<td align="right">22</td>
</tr>
<tr class="even">
<td align="left">c4</td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">c5</td>
<td align="right">23</td>
</tr>
</tbody>
</table>
<div id="are-there-significant-differences-in-tf-activity-between-the-four-clusters-1" class="section level5">
<h5>Are there significant differences in TF activity between the four clusters?</h5>
<p>Performing Kruskal-Wallis-Test to find differences between the four clusters in patient one yields 59 TFs of all 125 TFs which are differentially active between the three clusters(p-value &lt; 0.05). Of the 24 TFs with differential binding scores in the C20 cell line, fifteen have significantly different activity between the clusters.</p>
<p>Now that it is known that there are significant differences between clusters, the next step is to identify which cluster/s are different from which other cluster/s. For this pairwise Wilcox test was used.</p>
</details>
<details>
<summary>
Kruskal-Wallis-Test, p-value adjustment, posthoc tests
</summary>
<ul>
<li><strong>Kruskal-Wallis Test and p-value adjustments for all 125 TFs</strong></li>
</ul>
<pre class="r"><code># all TFs
k5_df &lt;- hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV1_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(clu1,clu2,clu3,clu4, clu5)) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% clu1, &quot;clu1&quot;, 
                             ifelse(cell %in% clu2, &quot;clu2&quot;, 
                                    ifelse(cell %in% clu3, &quot;clu3&quot;,
                                           ifelse(cell %in% clu4, &quot;clu4&quot;, &quot;clu5&quot;)))))
n = 1
# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(k5_df[tf]) ~ ht_cluster, k5_df))[&quot;p.value&quot;]
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)

# how many transcription factors are significantly different between the clusters?
sign &lt;- adj_p_values[adj_p_values &lt; 0.05]
length(adj_p_values[adj_p_values&lt;0.05])
[1] 59


# get the TFs that are differentially expressed
diff5 &lt;- colnames(auc_mtx)[adj_p_values &lt; 0.05]

#write.table(diff5, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=5&quot;)</code></pre>
<ul>
<li><strong>Kruskal-Wallis Test and p-value adjustments for all 125 TFs</strong></li>
</ul>
<pre class="r"><code># check 24 TFs with Kruskal Wallis test
p_values &lt;- map(seq.int(1:24), function(n){
  tf &lt;- overlap_atac$name[n]
  p_values &lt;- unlist(kruskal.test(pull(k5_df[tf]) ~ ht_cluster, k5_df))[&quot;p.value&quot;]
})
#sign_p_values &lt;- p_values[p_values &lt; 0.01]

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values&lt;0.05])
[1] 15

# get the TFs that are differentially expressed
diff5_atac &lt;- overlap_atac$name[adj_p_values &lt; .05]

#write.table(diff5_atac, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=5&quot;)</code></pre>
<ul>
<li><strong>Pairwise-Wilcox_Test for k = 5</strong></li>
</ul>
<p>Posthoc test for the 15 TFs with significantly different activity between the four clusters.</p>
<pre class="r"><code># We perform a posthoc test
#wilcox_list = list()
wilcox_list &lt;- map(seq.int(1:length(diff5_atac)), function(i){
  tf &lt;- diff5_atac[i]
  print(tf)
  wilcox &lt;- pairwise.wilcox.test(pull(k5_df[tf]), pull(k5_df[&quot;ht_cluster&quot;]), 
                                 p.adjust.method = &quot;fdr&quot;)
  list(TF = tf, wilcox = wilcox)
})
[1] &quot;NFIL3&quot;
[1] &quot;MXI1&quot;
[1] &quot;CREB1&quot;
[1] &quot;NFATC2&quot;
[1] &quot;MEF2A&quot;
[1] &quot;REST&quot;
[1] &quot;LEF1&quot;
[1] &quot;KLF4&quot;
[1] &quot;YY1&quot;
[1] &quot;ETS1&quot;
[1] &quot;JUNB&quot;
[1] &quot;CEBPB&quot;
[1] &quot;CTCF&quot;
[1] &quot;ETV6&quot;
[1] &quot;MEF2C&quot;
  
#length(wilcox_list)
#wilcox_list</code></pre>
<hr />
</details>
<hr />
</div>
</div>
</div>
<div id="which-clusters-are-significantly-different-for-different-k" class="section level3 tabset">
<h3 class="tabset">Which Clusters are significantly different for different k?</h3>
<p>The results from the statistical tests on difference in TF activity between clusters can be seen in the boxplots below. Sixteen, fourteen and twelve TFs were found to be differentially active between the clusters for k equal three, four and five respectively. The boxplots below also show very nicely, that the data are not normally distributed and, therefore, non-parametric tests, like the Kruskal-Wallis-Test and pairwise Wilcox-Test were used.</p>
<div id="k-3" class="section level4">
<h4>k = 3</h4>
<pre class="r"><code>
test &lt;- k3_df
#tf.oi &lt;- c(&quot;CTCF&quot;, &quot;REST&quot;, &quot;YY1&quot;)
#names(tf.oi) &lt;- tf.oi
test.list &lt;- map(hiv1_atac, function(tf) {
  ggplot() +
  geom_violin(aes(x = test %&gt;% pull(&quot;ht_cluster&quot;), y = test %&gt;% pull(tf), 
                   fill = test %&gt;% pull(&quot;ht_cluster&quot;), alpha = .6)) +
  geom_boxplot(aes(x = test %&gt;% pull(&quot;ht_cluster&quot;), y = test %&gt;% pull(tf)), 
               alpha = 0) +
  stat_pvalue_manual(test %&gt;% 
    pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;%
      add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;)) })

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 4))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-63-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-63-1">
Past versions of unnamed-chunk-63-1.png
</button>
</p>
<div id="fig-unnamed-chunk-63-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-63-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-63-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-63-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-63-1.png" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-63-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-63-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="k-4" class="section level4">
<h4>k = 4</h4>
<pre class="r"><code>
test &lt;- k4_df

test.list &lt;- map(diff4_atac, function(tf) {
  ggplot() +
  geom_violin(aes(x = test %&gt;% pull(&quot;ht_cluster&quot;), y = test %&gt;% pull(tf), 
                   fill = test %&gt;% pull(&quot;ht_cluster&quot;), alpha = .6)) +
  geom_boxplot(aes(x = test %&gt;% pull(&quot;ht_cluster&quot;), y = test %&gt;% pull(tf)), 
               alpha = 0) +
  stat_pvalue_manual(test %&gt;% 
    pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;%
      add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;)) +
    geom_violin(alpha = 0)
})

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 3))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-64-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-64-1">
Past versions of unnamed-chunk-64-1.png
</button>
</p>
<div id="fig-unnamed-chunk-64-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-64-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="k-5" class="section level4">
<h4>k = 5</h4>
<pre class="r"><code>test &lt;- k5_df

test.list &lt;- map(diff5_atac, function(tf) {
  ggplot() +
  geom_violin(aes(x = test %&gt;% pull(&quot;ht_cluster&quot;), y = test %&gt;% pull(tf), 
                   fill = test %&gt;% pull(&quot;ht_cluster&quot;), alpha = .6)) +
  geom_boxplot(aes(x = test %&gt;% pull(&quot;ht_cluster&quot;), y = test %&gt;% pull(tf)), 
               alpha = 0) +
  stat_pvalue_manual(test %&gt;% 
    pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;%
      add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;))
})

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 3))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-65-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-65-1">
Past versions of unnamed-chunk-65-1.png
</button>
</p>
<div id="fig-unnamed-chunk-65-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-65-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
</div>
<div id="conclusion-on-how-many-ks-to-use-for-k-means-clustering" class="section level2">
<h2>Conclusion on how many k’s to use for k-means clustering</h2>
<p>As mentioned, k = 3 was used for the subsequent analysis, since more clusters did not add additional insights.</p>
<p>The Venn diagram below shows very nicely, that out of all TFs with different activity between clusters, 68% overlap for all k (k=3/4/5) that I tested (right plot). 75% of the TF with differential binding scores in the C20 cell line overlapped for all k (left plot).</p>
<p>Therefore, I decided to proceed with k = 3 for the following analysis.</p>
<pre class="r"><code>diff5_atac &lt;- as.vector(read.table( &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=5&quot;))[[&quot;x&quot;]]
diff5 &lt;- as.vector(read.table( &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=5&quot;))[[&quot;x&quot;]]


diff4_atac &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=4&quot;))[[&quot;x&quot;]]

diff4 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=4&quot;))[[&quot;x&quot;]]
# list of TFs from ATAC-seq differntially active between the 3 heatmap 
#clusters from patient 1
diff3_atac &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1&quot;))[[&quot;x&quot;]]

# list of TFs differntially active between the 3 heatmap clusters from patient 1
diff3 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1&quot;))[[&quot;x&quot;]]</code></pre>
<pre class="r"><code>p1 &lt;- ggVennDiagram(list(diff3_atac, diff4_atac, diff5_atac),
              label_alpha = 0,
              category.names= c(&quot;k = 3&quot;, &quot;k = 4&quot;, &quot;k = 5&quot;),
              #set.size = 20,
              label = &quot;count&quot;,
              label_size = 6) +
   scale_fill_distiller(palette = &quot;RdBu&quot;) +
  labs(title = &quot;TFs with differential binding scores in C20 cell line&quot;)
  #scale_fill_gradient (low = &quot;lightblue&quot;, high = &quot;darkblue&quot;)

p2 &lt;- ggVennDiagram(list(diff3, diff4, diff5),
              label_alpha = 0,
              category.names= c(&quot;k = 3&quot;, &quot;k = 4&quot;, &quot;k = 5&quot;),
              label = &quot;count&quot;,
              label_size = 6) +
   scale_fill_distiller(palette = &quot;RdBu&quot;) +
  labs(title = &quot;all TFs with different activity&quot;)

fig &lt;- ggarrange(p1, p2, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob(&quot;TFs with significantly different activity between clusters for different k&quot;, size = 20))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-67-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-67-1">
Past versions of unnamed-chunk-67-1.png
</button>
</p>
<div id="fig-unnamed-chunk-67-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-67-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/d0fdb768719231ff1aad5dbf5161957e35112d9c/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-67-1.png" target="_blank">d0fdb76</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/80ac62c26f8ebb5a2c9d4bd18dc6d311f4d7baa0/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-67-1.png" target="_blank">80ac62c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-29
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/793e3ad73fde51899077efb8327ad8d9bc18d8cb/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-67-1.png" target="_blank">793e3ad</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-67-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-67-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="transfer-cluster-identitites" class="section level3">
<h3>Transfer cluster identitites</h3>
<p>The cluster identities from patient 1 were transferred to patient 2 and patient 3 to assign each cell to one of the 3 clusters.</p>
<details>
<summary>
<strong>Cluster annotation using an integrated referenc</strong>
</summary>
<p>Label Transfer</p>
<p>Here we do not integrate the datasets beforehand and we do not modify the query expression data.</p>
<pre class="{"><code>Idents(hiv7) &lt;- &quot;cell_type&quot;
hiv &lt;- subset(hiv7, idents = &quot;Myeloid2&quot;)

# the cell type is replaced by the heatmap cluster
hiv1_my2 &lt;- subset(hiv7, cells = c(c1,c2,c3))

clusters &lt;- (
  hiv1_my2@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c1, &quot;c1&quot;, ifelse(cell %in% c2, &quot;c2&quot;, &quot;c3&quot;))) %&gt;% 
  column_to_rownames(&quot;cell&quot;))[&quot;ht_cluster&quot;]

hiv &lt;- AddMetaData(hiv, metadata = clusters, col.name = &quot;cell_type&quot;)

# the seurat_cluster number is replaced by the number of the heatmap cluster
seurat_clusters &lt;- (
  hiv1_my2@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c1, 1, ifelse(cell %in% c2, 2, 3))) %&gt;% 
  column_to_rownames(&quot;cell&quot;))[&quot;ht_cluster&quot;]

hiv &lt;- AddMetaData(hiv, metadata = seurat_clusters, col.name = &quot;seurat_clusters&quot;)


# split the object according to the patients
hiv_list &lt;- SplitObject(hiv, split.by = &quot;orig.ident&quot;)[3:5]
# normalize each object
for (i in names(hiv_list)) {
    hiv_list[[i]] &lt;- SCTransform(hiv_list[[i]], verbose = FALSE)
}


# patient 2 and patient 3 are our query datasets
query &lt;-merge(hiv_list[[1]], hiv_list[[3]])
reference &lt;- hiv_list[[2]]
reference &lt;- RunPCA(reference, 
                    # by default a partial PCA is run, but we compute mostgular values -&gt; run standard SVD instead
                    approx = FALSE, 
                    verbose = FALSE)

hiv_anchors &lt;- FindTransferAnchors(reference = reference,
                                   query = query, 
                                   dims = 1:30, verbose = FALSE,
                                   k.filter = NA, 
                                   reference.reduction = &quot;pca&quot;,
                                   normalization.method = &quot;SCT&quot;,
                                   reduction = &quot;pcaproject&quot;)

predictions &lt;- TransferData(anchorset = hiv_anchors, 
                            refdata = reference$cell_type,
                            k.weight = 34, dims = 1:30, verbose = FALSE)

query &lt;- AddMetaData(query, metadata = predictions)


hiv2 &lt;- subset(query, orig.ident == &quot;HIV2_CSF&quot;)
hiv3 &lt;- subset(query, orig.ident == &quot;HIV3_CSF&quot;)
#saveRDS(hiv2,&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv2_CSF_myleoid2_heatmap_cluster_label_transfer.rds&quot;)
#saveRDS(hiv3, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv3_CSF_myleoid2_heatmap_cluster_label_transfer.rds&quot;)</code></pre>
<p>Now that we have transferred the cluster identitities to patient2 and patient3 and saved the results in two separate Seurat objects, we can add the TF activity score to the metadata of each patient.</p>
<pre class="r"><code>hiv2 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv2_CSF_myleoid2_heatmap_cluster_label_transfer.rds&quot;)
hiv3 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv3_CSF_myleoid2_heatmap_cluster_label_transfer.rds&quot;)

hiv2_scenic &lt;- AddMetaData(hiv2, metadata = auc_mtx,
                           col.name = colnames(auc_mtx))
hiv3_scenic &lt;- AddMetaData(hiv3, metadata = auc_mtx,
                           col.name = colnames(auc_mtx))</code></pre>
<hr />
</details>
<p>The heatmaps below show the predicted clusters from label transfer for patient 2 and patient 3. The clusters obtained after transferring the cluster identities from patient 1 to patient2 do not seem to be very good at separating cells according to TF activity. When you transfer the cluster identities from patient1 to patient3 all cells, but one, are assigned to cluster1. This shows that the label transfer did not work very well. This is probably due to the low number of Myeloid2 cells in each sample.</p>
<pre class="r"><code>cluster_anno &lt;- as.data.frame(hiv2$predicted.id)

#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV2_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_hiv2_label &lt;- Heatmap(t(mtx), 
             column_title = &quot;Predicted Clusters from Label Transfer for HIV2_CSF&quot;,             
             column_title_gp = gpar(fontsize = 15),
             column_split = paste0(&quot;cluster&quot;, hiv2$predicted.id),#cluster_anno,
             #column_km = 4,
             #column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la,
             top_annotation = HeatmapAnnotation(df = cluster_anno, name = &quot;predicted cluster&quot;)
             ) 
             #top_annotation = column_annotation) </code></pre>
<pre class="r"><code>cluster_anno &lt;- as.data.frame(hiv3$predicted.id)

#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV3_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_hiv3_label &lt;- Heatmap(t(mtx), 
             column_title = &quot;Predicted clusters from Label Transfer for HIV3_CSF&quot;, 
             column_title_gp = gpar(fontsize = 15),
             column_split = paste0(&quot;cluster&quot;, hiv3$predicted.id),#cluster_anno,
             #column_km = 4,
             #column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la,
             top_annotation = HeatmapAnnotation(df = cluster_anno, name = &quot;predicted cluster&quot;)
             ) 
             #top_annotation = column_annotation) </code></pre>
<pre class="r"><code>ht_list &lt;- ht_hiv2_label + ht_hiv3_label
draw(ht_list, ht_gap= unit(2, &quot;cm&quot;))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-71-1">
Past versions of unnamed-chunk-71-1.png
</button>
</p>
<div id="fig-unnamed-chunk-71-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="gene-expression-analysis-of-the-two-myeloid2-subclusters" class="section level3">
<h3>Gene expression analysis of the two Myeloid2 subclusters</h3>
<div id="tf-from-farhadian-et-al." class="section level4">
<h4>TF from Farhadian et al.</h4>
<p>These TFs were used as marker genes for the microlgia like Myeloid2 cell cluster [<span class="citation"><a href="#ref-Farhadian" role="doc-biblioref">Farhadian SF</a> (<a href="#ref-Farhadian" role="doc-biblioref">2018</a>)</span>. Is there a group of cells which has a higher expression of these markers? No clear difference in gene expression can be seen.</p>
<details>
<summary>
Boxplots of microglia marker genes
</summary>
<pre class="r"><code>#import the 60 genes that were upregulated in Myeloid-2 in the paper compared to the four other myeloid subsets
paper &lt;- read.table(file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/Myeloid-2_genes_paper.csv&quot;,
                    sep = &quot;\t&quot;, header = FALSE, skipNul = TRUE)#, n_max = 60)
paper[1,] &lt;- &quot;APOC1&quot;
colnames(paper) &lt;- &quot;&quot;
paper &lt;- as.vector(paper[,1])


VlnPlot(hivmy2, features = paper, group.by = &quot;ht_cluster&quot;)
Warning in FetchData(object = object, vars = features, slot = slot): The
following requested variables were not found: SELENOP</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-72-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-72-1">
Past versions of unnamed-chunk-72-1.png
</button>
</p>
<div id="fig-unnamed-chunk-72-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-72-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-72-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-72-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</details>
</div>
<div id="differential-gene-expression-analyisis" class="section level4">
<h4>Differential gene expression analyisis</h4>
<details>
<summary>
Differential gene expression analysis
</summary>
<pre class="r"><code>hivmy2 &lt;- NormalizeData(hivmy2, verbose = FALSE)
hivmy2 &lt;- ScaleData(hivmy2, verbose = FALSE)
hivmy2 &lt;- FindVariableFeatures(hivmy2, verbose = FALSE)
Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
parametric, : pseudoinverse used at -2.1857
Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
parametric, : neighborhood radius 0.49845
Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
parametric, : reciprocal condition number 7.6527e-16
Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
parametric, : There are other near singularities as well. 0.090619
hivmy2 &lt;- RunPCA(hivmy2, features = VariableFeatures(hivmy2), verbose = FALSE)
ElbowPlot(hivmy2)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-73-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-73-1">
Past versions of unnamed-chunk-73-1.png
</button>
</p>
<div id="fig-unnamed-chunk-73-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-73-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dfcd540d3a574f3ad19196570abd4c6c51bd0aa3/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-73-1.png" target="_blank">dfcd540</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-73-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-73-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-73-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-73-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-73-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>hivmy2 &lt;- RunTSNE(hivmy2, dims= 1:10, verbose = FALSE)
Idents(hivmy2) &lt;- &quot;ht_cluster&quot;
de_genes &lt;- FindAllMarkers(hivmy2, verbose = FALSE)
top10 &lt;- de_genes %&gt;% 
  group_by(cluster) %&gt;%
  top_n(n = 5, wt = avg_log2FC) 
top10 %&gt;% 
  kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">p_val</th>
<th align="right">avg_log2FC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
<th align="left">cluster</th>
<th align="left">gene</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0.0002643</td>
<td align="right">1.4371177</td>
<td align="right">0.206</td>
<td align="right">0.027</td>
<td align="right">1</td>
<td align="left">c1</td>
<td align="left">EPHB6</td>
</tr>
<tr class="even">
<td align="right">0.0011962</td>
<td align="right">1.2486709</td>
<td align="right">0.382</td>
<td align="right">0.143</td>
<td align="right">1</td>
<td align="left">c1</td>
<td align="left">SNRPF</td>
</tr>
<tr class="odd">
<td align="right">0.0015781</td>
<td align="right">1.0213995</td>
<td align="right">0.176</td>
<td align="right">0.027</td>
<td align="right">1</td>
<td align="left">c1</td>
<td align="left">RP11-386I14.4</td>
</tr>
<tr class="even">
<td align="right">0.0022935</td>
<td align="right">0.9050976</td>
<td align="right">0.118</td>
<td align="right">0.009</td>
<td align="right">1</td>
<td align="left">c1</td>
<td align="left">SACS</td>
</tr>
<tr class="odd">
<td align="right">0.0045394</td>
<td align="right">1.0999273</td>
<td align="right">0.294</td>
<td align="right">0.107</td>
<td align="right">1</td>
<td align="left">c1</td>
<td align="left">TBC1D5</td>
</tr>
<tr class="even">
<td align="right">0.0005081</td>
<td align="right">1.3333431</td>
<td align="right">0.286</td>
<td align="right">0.000</td>
<td align="right">1</td>
<td align="left">c2</td>
<td align="left">JAK2</td>
</tr>
<tr class="odd">
<td align="right">0.0006461</td>
<td align="right">1.3600293</td>
<td align="right">0.438</td>
<td align="right">0.118</td>
<td align="right">1</td>
<td align="left">c2</td>
<td align="left">IGSF6</td>
</tr>
<tr class="even">
<td align="right">0.0012653</td>
<td align="right">1.1633609</td>
<td align="right">0.589</td>
<td align="right">0.324</td>
<td align="right">1</td>
<td align="left">c2</td>
<td align="left">CTSH</td>
</tr>
<tr class="odd">
<td align="right">0.0072128</td>
<td align="right">1.1970288</td>
<td align="right">0.429</td>
<td align="right">0.176</td>
<td align="right">1</td>
<td align="left">c2</td>
<td align="left">CD55</td>
</tr>
<tr class="even">
<td align="right">0.0096532</td>
<td align="right">1.1831845</td>
<td align="right">0.286</td>
<td align="right">0.088</td>
<td align="right">1</td>
<td align="left">c2</td>
<td align="left">NAMPTL</td>
</tr>
</tbody>
</table>
<hr />
</details>
</div>
<div id="tf-from-gosselin-et-al." class="section level4">
<h4>TF from Gosselin et al.</h4>
<details>
<summary>
Microglia TF gene expression
</summary>
<pre class="r"><code>Idents(hiv4) &lt;- &quot;cell_type&quot;
# Find Markers that distinguish My2 from the other Myeloid clusters
my2_markers &lt;- FindMarkers(hiv4, ident.1  = &quot;Myeloid2&quot;,
            ident.2  = c(&quot;Myeloid1&quot;, &quot;Myeloid3&quot;, &quot;Myeloid4&quot;, &quot;Myeloid5&quot;))
my2_markers &lt;- my2_markers %&gt;% rownames_to_column(var = &quot;gene&quot;)</code></pre>
<pre class="r"><code>microglia_genes &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/Microglia_gene_signatur_found_in_HIV1_HIV2_samples&quot;)
m1 &lt;- my2_markers %&gt;% filter(gene %in% microglia_genes [[&quot;gene_name&quot;]] &amp; avg_log2FC &gt;0)
# remove the genes that were already used by the Farhadian paper
m1 &lt;- m1[[&quot;gene&quot;]][m1[[&quot;gene&quot;]] %in% paper == FALSE]

VlnPlot(hivmy2, features = m1, group.by = &quot;ht_cluster&quot;, ncol = 5)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-75-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-75-1">
Past versions of unnamed-chunk-75-1.png
</button>
</p>
<div id="fig-unnamed-chunk-75-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/270aca65f4da298bf474a0f567f84d05abd61650/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-75-1.png" target="_blank">270aca6</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-75-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-75-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-75-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-75-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</details>
</div>
<div id="which-of-the-11-tfs-are-mentioned-by-gosselin-et-al." class="section level4">
<h4>Which of the 11 TFs are mentioned by Gosselin et al.?</h4>
<details>
<summary>
TFs also mentioned by Gosselin et al.
</summary>
<pre class="r"><code>tfs_microglia &lt;- (read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/TFs_microglia_Glass_paper.txt&quot;))[[&quot;V1&quot;]]

gosselin &lt;- tfs_microglia[tfs_microglia %in% overlap_atac$name]

VlnPlot(hivmy2, features = gosselin, group.by = &quot;ht_cluster&quot;, ncol = 4)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-76-1">
Past versions of unnamed-chunk-76-1.png
</button>
</p>
<div id="fig-unnamed-chunk-76-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/8481440d0c52c2f99c1682672636c891d233d870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" target="_blank">8481440</a>
</td>
<td>
Katharina782
</td>
<td>
2022-01-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</details>
</div>
</div>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Eisenberg" class="csl-entry">
Eisenberg E, Levanon EY. 2013. <em>Human Housekeeping Genes, Revisited</em>. <em>Trends Genet</em>. Vol. 29(10). <a href="https://doi: 10.1016/j.tig.2013.05.010">doi: 10.1016/j.tig.2013.05.010</a>.
</div>
<div id="ref-Farhadian" class="csl-entry">
Farhadian SF, Zografou C, Mehta SS. 2018. <em>Single-Cell RNA Sequencing Reveals Microglia-Like Cells in Cerebrospinal Fluid During Virologically Suppressed HIV</em>. <em>JCI Insight</em>. Vol. 3(18):e121718. <a href="https://doi.org/10.1172/jci.insight.121718">https://doi.org/10.1172/jci.insight.121718</a>.
</div>
<div id="ref-Jeffereys" class="csl-entry">
Jefferys SR, Peterson JJ, Burgos SD. 2021. <em>Epigenomic Characterization of Latent HIV Infection Identifies Latency Regulating Transcription Factors</em>. <em>PLoS Pathog</em>. Vol. 17(2). <a href="https://doi: 10.1371/journal.ppat.1009346">doi: 10.1371/journal.ppat.1009346</a>.
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] future_1.23.0        rstatix_0.6.0        corrplot_0.92       
 [4] ggVennDiagram_1.2.0  knitr_1.30           gridExtra_2.3       
 [7] ggpubr_0.4.0         circlize_0.4.13      ComplexHeatmap_2.9.4
[10] ggcorrplot_0.1.3     pheatmap_1.0.12      RColorBrewer_1.1-2  
[13] ggrepel_0.9.1        data.table_1.14.2    Matrix_1.3-4        
[16] edgeR_3.32.0         limma_3.46.0         SeuratObject_4.0.3  
[19] Seurat_4.0.5         forcats_0.5.0        stringr_1.4.0       
[22] dplyr_1.0.7          purrr_0.3.4          readr_1.4.0         
[25] tidyr_1.1.2          tibble_3.1.6         ggplot2_3.3.5       
[28] tidyverse_1.3.0      workflowr_1.6.2     

loaded via a namespace (and not attached):
  [1] utf8_1.2.2            reticulate_1.22       tidyselect_1.1.1     
  [4] htmlwidgets_1.5.4     Rtsne_0.15            munsell_0.5.0        
  [7] units_0.6-7           codetools_0.2-18      ica_1.0-2            
 [10] miniUI_0.1.1.1        withr_2.4.3           colorspace_2.0-2     
 [13] highr_0.8             rstudioapi_0.13       stats4_4.0.2         
 [16] ROCR_1.0-11           ggsignif_0.6.0        tensor_1.5           
 [19] listenv_0.8.0         labeling_0.4.2        git2r_0.27.1         
 [22] polyclip_1.10-0       farver_2.1.0          rprojroot_2.0.2      
 [25] parallelly_1.29.0     vctrs_0.3.8           generics_0.1.1       
 [28] xfun_0.29             R6_2.5.1              doParallel_1.0.16    
 [31] clue_0.3-60           locfit_1.5-9.4        RVenn_1.1.0          
 [34] reshape_0.8.8         spatstat.utils_2.2-0  assertthat_0.2.1     
 [37] promises_1.2.0.1      scales_1.1.1          gtable_0.3.0         
 [40] globals_0.14.0        goftest_1.2-3         rlang_0.4.12         
 [43] GlobalOptions_0.1.2   splines_4.0.2         lazyeval_0.2.2       
 [46] spatstat.geom_2.3-0   broom_0.7.2           yaml_2.2.1           
 [49] reshape2_1.4.4        abind_1.4-5           modelr_0.1.8         
 [52] backports_1.2.0       httpuv_1.6.3          tools_4.0.2          
 [55] ellipsis_0.3.2        spatstat.core_2.3-1   BiocGenerics_0.36.1  
 [58] ggridges_0.5.2        Rcpp_1.0.7            plyr_1.8.6           
 [61] classInt_0.4-3        rpart_4.1-15          deldir_1.0-6         
 [64] pbapply_1.5-0         GetoptLong_1.0.5      cowplot_1.1.1        
 [67] S4Vectors_0.28.1      zoo_1.8-9             haven_2.3.1          
 [70] cluster_2.1.0         fs_1.5.1              magrittr_2.0.1       
 [73] magick_2.5.2          scattermore_0.7       openxlsx_4.2.3       
 [76] lmtest_0.9-39         reprex_0.3.0          RANN_2.6.1           
 [79] whisker_0.4           fitdistrplus_1.1-6    matrixStats_0.61.0   
 [82] hms_1.1.1             patchwork_1.1.1       mime_0.12            
 [85] evaluate_0.14         xtable_1.8-4          rio_0.5.16           
 [88] readxl_1.3.1          IRanges_2.24.1        shape_1.4.6          
 [91] compiler_4.0.2        KernSmooth_2.23-18    crayon_1.4.2         
 [94] htmltools_0.5.2       mgcv_1.8-33           later_1.3.0          
 [97] lubridate_1.7.9.2     DBI_1.1.1             dbplyr_2.0.0         
[100] MASS_7.3-53           sf_0.9-6              car_3.0-10           
[103] cli_3.1.0             parallel_4.0.2        igraph_1.2.9         
[106] pkgconfig_2.0.3       foreign_0.8-80        plotly_4.9.2.1       
[109] spatstat.sparse_2.0-0 xml2_1.3.2            foreach_1.5.1        
[112] rvest_0.3.6           digest_0.6.29         sctransform_0.3.2    
[115] RcppAnnoy_0.0.19      spatstat.data_2.1-0   rmarkdown_2.6        
[118] cellranger_1.1.0      leiden_0.3.5          uwot_0.1.10          
[121] curl_4.3.2            shiny_1.7.1           rjson_0.2.20         
[124] lifecycle_1.0.1       nlme_3.1-150          jsonlite_1.7.2       
[127] carData_3.0-4         viridisLite_0.4.0     fansi_0.5.0          
[130] pillar_1.6.4          GGally_2.0.0          lattice_0.20-41      
[133] fastmap_1.1.0         httr_1.4.2            survival_3.2-7       
[136] glue_1.6.0            zip_2.1.1             png_0.1-7            
[139] iterators_1.0.13      class_7.3-17          stringi_1.7.6        
[142] e1071_1.7-4           irlba_2.3.3           future.apply_1.8.1   </code></pre>
</div>
</div>

<div id="rmd-source-code">---
title: "TF activity in Myeloid2 cells"
author: "Katharina Mikulik"
output: html
bibliography: references.bib
link-citations: yes
---
<style>
body {
text-align: justify}
</style>


```{r setup}
knitr::opts_chunk$set(echo = TRUE, collapse = TRUE, message = FALSE,
                      cache = FALSE, autodep = TRUE)
```


```{r}
library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggplot2)
library(dplyr)
library(ggrepel)
#library(harmony)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(knitr)
library(ggVennDiagram)
library(corrplot)
library(rstatix)
# "Multitasking/Multiprocessing"
library(future)
#plan("multisession", workers = 24)
```

# GRN inference with Scenic

Using Scenic we tried to infer Gene regulatory networks from the gene expression
data. The output is a matrix of TF (Transcription Factor) activity per cell. 
The input matrix for SCENIC included the 10,000 most highly variable genes of 
the Myeloid2 cell cluster. The resulting activity matrix contains activity scores
of 125 different transcription factors.

First, we wanted to see if we can identify patterns in the TF activity in the
Myeloid2 cells. We compared the TF activity to TFs shown to have differential 
binding scores between three infection states (uninfected, latent infection, 
active infection) in primary microglia (bulk ATACseq). For the comparison
with the TF activity in Myeloid2 cells we focused on the TF with differential 
binding scores between **uninfected** and **latently infected** states.

The question we wanted to answer is:
Can we see patterns of increased/decreased TF activity in Myeloid2 cells that
correlate with the binding scores between infection states in primary
microglia?    


</details>

<details> 
<summary>TF activity matrix and HIV CSF/Blood dataset</summary>


* read in Seurat object

```{r}
hiv7 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds")
hiv4 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/non_integrated_HIV1_HIV2_4samples_seurat_object.rds")
```

* read in AuCell matrix (output from Scenic)

```{r}
auc_mtx <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/aucell_matrix_10k_hvg.tsv", sep = "\t")

auc_mtx %>% head %>%  kable(caption = "TF activity matrix (output from SCENIC)")
```



* Combine AuCell Matrix and Seurat object

```{r}
# add auc matrix to metadata of hiv4 seurat object
hiv7_scenic <- AddMetaData(hiv7, 
                    metadata = auc_mtx, 
                    col.name = colnames(auc_mtx))
```

---
</details>

# Bulk ATACseq footprinting

Using bulk ATACseq data from the C20 cell line (immortalized microglia) with three different conditions,
active infection, latent infection, uninfected, differential binding scores were
obtained using TOBIAS. Here, I will use the TFs which were previously shown to 
have differential binding scores in these primary microglia. Since I expect the 
CSF Myeloid cells to be latently infected, because they have been obtained from 
patients receiving ART, I will use the TFs with differential binding scores 
between latent infection and uninfected. We would like to know if the TF activity
inferred from RNAseq data of Myeloid2 cells correlates with these differential 
binding scores. In other words, we want to know if the patterns of differential 
TF activity are similar to the patterns of differential TF binding scores. 
For example, if a TF was shown to be more bound in latently infected primary
microglia compared to uninfected primary microglia, we would expect the TF to be
more active in a group of Myeloid2 cells which are latently infected compared
to a group of Myeloid2 cells which are uninfected. 

Comparing the TFs from the TF activity matrix with the differentially
bound TFs from the footprinting, 24 of the TFs overlap.

Using less than 10,000 most highly variable genes as an input matrix for SCENIC 
resulted in a lower number of TFs which overlap. Increasing the number of input 
features for SCENIC increased the number of TFs included in the TF activity 
matrix. Therefore, 10,000 most highly variable genes were used.

```{r}
# read in dataframe containing differntial binding scores between conditions
#
# red = latently infected
# gfp = active infected
# uninfected
tfs_ana <- read.table("/media/ag-cherrmann/projects/06_HIV_Microglia/data/atacseq/data-2020-11-06/tobias/TOBIAS_snakemake/footprint_mglia2_GlassTF_17-03/TFBS/bindetect_results.txt", sep = "\t", header = TRUE)

# filter TFs with significant p values between latent infection and uninfected
# I decided to use a threshold of p-value < 1e-50, because due to the large number 
# of data, all p-values will be very high
top_tfs <- tfs_ana %>% filter(uninf_red_pvalue < 1e-50)

# keep only the TFs in the dataframe which overlap between the two datasets.
overlap_atac <- top_tfs %>% filter(name %in% colnames(auc_mtx))

# extract names of TFs more bound in either condition
up_latent <- overlap_atac$name[overlap_atac$uninf_red_change < 0]
up_uninf <- overlap_atac$name[overlap_atac$uninf_red_change > 0]

```

## Volcano Plot ATAC-seq footprinting

Only four TFs found in the TF activity matrix of Myeloid2 cells 
are more bound in latent infection than in uninfected primary microglia.

```{r, fig.height=8, fig.width=12}

top_tfs %>% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = ifelse(uninf_red_change < 0, "up_latent", "up_uninf")) %>%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  mutate(label = ifelse(name %in% overlap_atac$name, name, NA)) %>% 
  ggplot(aes(x = uninf_red_change,
             y = -log10(uninf_red_pvalue),
             col = condition, 
             label = label)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-50), col = "red") +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = "TFs with differential binding scores") +
  ylab("-log10(p_value)") +
  xlab("differential binding score")

```


# Compare TF activities between clusters and look for patterns

The heatmaps below were split into 3 separate heatmaps, one for each patient. Shown are
the 24 TFs found by ATAC-seq footprinting as rows and cells as columns. Judging from the plots no TFs clearly separate the clusters or
show a pattern of activity which is similar to the footprinting binding scores. 
If the pattern was similar to the footprinting binding scores we would expect to
see one cluster where REST, KLF4, PAX5 and MXI1 are upregulated. This cannot be
seen in the plots below. 


```{r}


# TF annotations
la <- rowAnnotation(type_TF = 
                      case_when(colnames(auc_mtx) %in% up_latent ~ "up_latent",
                                        colnames(auc_mtx) %in% up_uninf  ~ "up_uninf",
                                        TRUE ~"unknown"),
                     name = "type_TF",
                     col = list(type_TF = c("up_latent" =  "red", 
                                            "up_uninf" =  "pink",
                                            "unknown" = "forestgreen")))
  
```

```{r, fig.width=13, fig.height=8}
# 3 heatmaps, but only the TFs which overlap with Anas analysis
ht_list <- map(seq.int(1:3), function(n) {
  patients <- c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF") # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  # select TFs which overlap with Anas analysis
  mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), overlap_atac$name]
  # heatmap
  ht <- Heatmap(t(mtx), 
               column_title = paste0("patient: ", patients[n]), 

               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
               right_annotation = rowAnnotation(type_TF = ifelse(colnames(auc_mtx[,overlap_atac$name]) %in% up_latent, "up_latent", "up_uninf"),
                                                name = "type_TF",
                                                col = list(type_TF = c("up_latent" =  "red",
                                                                       "up_uninf" =  "pink"))))
  list(name = patients[n], heatmap = ht)
})

  
ht_list = ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap
draw(ht_list, ht_gap = unit(1, "cm"))
```

## Different approaches

Myeloid2 cells of patient1 (HIV1_CSF) were clustered based on their TF activity
using k-means clustering with k=3/4/5. The reason why only 
patient one was used, is that the other two patients might be used for validation
of the  clustering later on. If the clusters have a biological meaning, it should 
be possible to transfer the clusters onto the Myeloid2 cells from patient2 
(HIV2_CSF) and patient3 (HIV3_CSF). Also, if the clusters separate the Myeloid2 
cells of the other two patients in an interpretable and meaningful way this
would validate that there is a biological meaning behind the clusters.

However, the label transfer of the cluster identities did not 
work, probably due to the small sample size. There are only 146 cells in total in the 
Myeloid2 cluster of infected patients. A more detailed description of this step
and corresponding results can be found in the appendix under *"Determine k for*
*k-means clustering"* and *"Transfer the cluster identities to patient2 and patient3"*.

Since the approach described did not yield any interesting results, an 
alternative strategy was used, namely, performing independent k-means clustering
across all patients and comparing the TF activities between clusters. 
This way we might find TF activities which differentiate clusters across 
patients and find correlations between clusters across patients.


## K-means clustering with k = 3 {.tabset}

K-means clustering with k = 3, was performed for all 3 patients separately. This
way it might be possible to find TFs with different activities between the
three clusters. The clusters in the heatmaps below look more sensible than the
clusters observed after cluster label transfer (see appendix).

Since the data is not normally distributed, the Kruskal-Wallis-Test is more
appropriate than ANOVA to test for differences between clusters. The
Null-Hypothesis of the Kruskal-Wallis-Test is that 
there is no difference between the groups. Since the test is repeated 125 times, 
the p-values are adjusted with the
FDR method. This methods is more powerful than more conservative methods, like
Bonferroni, Holm or Hochberg. 

### Patient 1


```{r, fig.width=15, fig.height=17}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_pat1 <-draw(Heatmap(t(mtx), 
             column_title = "Myeloid2 cells from patient 1, k = 3", 

             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             ))
             #top_annotation = column_annotation) 
```

<details>
<summary>**Kruskal_Wallis test & Pairwise-Wilcox test for Patient 1**</summary>

Performing Kruskal-Wallis-Test to find differences between the three clusters in
patient one yields 64 TFs out of all 125 TFs which are differentially active 
between the three clusters(p-value < 0.05). Of the 24 TFs with 
differential binding scores in the C20 cell line, sixteen have significantly
different activity between the clusters.

Now that it is known that there are significant differences between clusters, 
the next step is to identify which cluster/s are different from which other
cluster/s, using Pairwise-Wilcox-Test, a non-parametric test. Again, because 
we repeat the test we will correct for multiple hypothesis testing using the 
FDR method. 


```{r}
### Extract cells from each cluster

# get the columns names of cluster one
c1 <- colnames(t(mtx[column_order(ht_pat1)[[1]],]))
#write.table(c1, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv", sep = "\t")

# get the column names of cluster two
c2 <- colnames(t(mtx[column_order(ht_pat1)[[2]],]))
#write.table(c2, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv", sep = "\t")

# get the column names of cluster 3
c3 <- colnames(t(mtx[column_order(ht_pat1)[[3]],]))
#write.table(c3, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv", sep = "\t")


# non-parametric alternative to ANOVA -> Kruskal_wallis test
k3_df <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) 
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))["p.value"]
  #list(tf = tf, p_value = p_values)
})



# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
sign3 <- adj_p_values[adj_p_values<.05]
length(adj_p_values[adj_p_values<0.05])

# get TFs that are significantly different
hiv1_diff <- colnames(auc_mtx)[adj_p_values < .05]

#write.table(diff3, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1")



## 24 TFs also identified by ATAC-se

# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])

# get TFs that are differentially expressed
hiv1_atac <- overlap_atac$name[adj_p_values < .05]
#write.table(diff3_atac, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1")

# We perform a posthoc test
#wilcox_list = list()
wilcox_list <- map(seq.int(1:length(hiv1_atac)), function(i){
  tf <- hiv1_atac[i]
  #print(tf)
  wilcox <- pairwise.wilcox.test(pull(k3_df[tf]), pull(k3_df["ht_cluster"]), 
                                 p.adjust.method = "fdr")
  list(TF = tf, wilcox = wilcox)
})
  
#length(wilcox_list)
#wilcox_list
```

---
</details>

---


### Patient 2

```{r, fig.width=15, fig.height=20}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV2_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht2 <- Heatmap(t(mtx), 
             column_title = "Patient 2, HIV2_CSF", 
             column_title_gp = gpar(fontsize = 15),
             #column_split = paste0("cluster", hiv2$predicted.id),#cluster_anno,
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la,
             #top_annotation = HeatmapAnnotation(df = cluster_anno, name = "predicted cluster")
             ) 
             #top_annotation = column_annotation) 
ht_pat2 <- draw(ht2)

```


<details>
<summary>**Kruskal_Wallis test & Pairwise-Wilcox test for Patient 2**</summary>


Performing Kruskal-Wallis-Test to find differences between the three clusters in
patient two yields 71 TFs out of all 125 TFs which are differentially active 
between the three clusters(p-value < 0.05). Of the 24 TFs with 
differential binding scores in the C20 cell line, nineteen have significantly
different activity between the clusters.

Now that it is known that there are significant differences between clusters, 
the next step is to identify which cluster/s are different from which other
cluster/s, using Pairwise-Wilcox-Test, a non-parametric test. Again, because 
we repeat the test we will correct for multiple hypothesis testing using the 
FDR method. 

```{r}
# extract the cells from the 3 clusters 
# get the cell names of cluster one
c1 <- colnames(t(mtx[column_order(ht_pat2)[[1]],]))
# get the cell names of cluster two
c2 <- colnames(t(mtx[column_order(ht_pat2)[[2]],]))
# get the cell names of cluster 3
c3 <- colnames(t(mtx[column_order(ht_pat2)[[3]],]))


# add cluster identities to Seurat object metadata and return metadata as dataframe
pat2 <- hiv7_scenic@meta.data %>% rownames_to_column("cell")
pat2 <- pat2 %>% 
  filter(orig.ident == "HIV2_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) 


# check all 125 TFs
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(pat2[tf]) ~ ht_cluster, pat2))["p.value"]
  #list(tf = tf, p_value = p_values)
})


# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")
length(adj_p_values[adj_p_values<0.05])
## [1] 66

# get the TFs taht are differntially active
diff_hiv2 <- colnames(auc_mtx)[adj_p_values < 0.05]


# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(pat2[tf]) ~ ht_cluster, pat2))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])
## [1] 18

# get the TFs that are differentially expressed
hiv2_atac <- overlap_atac$name[adj_p_values < .05]
```
---
</details>

---


### Patient 3

```{r, fig.width=15, fig.height=20}
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV3_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht3 <- Heatmap(t(mtx), 
             column_title = "Patient 3, HIV3_CSF", 
             column_title_gp = gpar(fontsize = 15),
             #column_split = paste0("cluster", hiv2$predicted.id),#cluster_anno,
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la,
             #top_annotation = HeatmapAnnotation(df = cluster_anno, name = "predicted cluster")
             ) 
             #top_annotation = column_annotation) ich
ht_pat3 <- draw(ht3)
```

<details>
<summary>**Kruskal_Wallis test & Pairwise-Wilcox test for Patient 3**</summary>

Performing Kruskal-Wallis-Test to find differences between the three clusters in
patient three yields 66 TFs out of all 125 TFs which are differentially active 
between the three clusters(p-value < 0.05). Of the 24 TFs with 
differential binding scores in the C20 cell line, eighteen have significantly
different activity between the clusters.

Now that it is known that there are significant differences between clusters, 
the next step is to identify which cluster/s are different from which other
cluster/s, using Pairwise-Wilcox-Test, a non-parametric test. Again, because 
we repeat the test we will correct for multiple hypothesis testing using the 
FDR method. 

```{r}
# extract the cells from the 3 clusters 
# get the cell names of cluster one
c1 <- colnames(t(mtx[column_order(ht_pat3)[[1]],]))
# get the cell names of cluster two
c2 <- colnames(t(mtx[column_order(ht_pat3)[[2]],]))
# get the cell names of cluster 3
c3 <- colnames(t(mtx[column_order(ht_pat3)[[3]],]))


# add cluster identities to Seurat object metadata and return metadata as dataframe
pat3 <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV3_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) 


# check all 125 TFs
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(pat3[tf]) ~ ht_cluster, pat3))["p.value"]
  #list(tf = tf, p_value = p_values)
})


# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")
length(adj_p_values[adj_p_values<0.05])
## [1] 66

# get the TFs taht are differntially active
diff_hiv3 <- colnames(auc_mtx)[adj_p_values < 0.05]


# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(pat3[tf]) ~ ht_cluster, pat3))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])
## [1] 18

# get the TFs that are differentially expressed
hiv3_atac <- overlap_atac$name[adj_p_values < .05]
```
---
</details>

---

### Comparison between all 3 patients

```{r, fig.width=15, fig.height=17}
# patient 1
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht1 <- Heatmap(t(mtx), 
             column_title = "Patient 1, HIV1_CSF", 
             column_title_gp = gpar(fontsize = 15),
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             )
             #top_annotation = column_annotation) 
```

```{r, fig.width=15, fig.height=20}
ht_list <- ht1 + ht2 + ht3
draw(ht_list, ht_gap= unit(2, "cm"), 
     column_title = "Independent k-means clustering (k = 3) for all patients",
     column_title_gp = gpar(fontsize = 20) )
```

---

# Different activities between clusters

The three clusters obtained with k-means clustering for all three patients
separately were compared to see if there is a pattern in TF activity that
can be found in all three patients. The question was if the same group of TFs 
characterize the three 
clusters in all three patients or if they are different. To answer this question,
Kruskal-Wallis-Test was used to determine the TFs which are significantly 
different between clusters and then a pairwise Wilcox test was used to determine
which clusters are different. 

In the Venn diagram below it can be seen that 46 TFs with significantly different
activities between clusters overlap between patients (see figure, left). Additionally,
eleven of the TFs found to have differential binding scores between
infection conditions in C20 microglia cell line overlap between patients
(see figure, right). These eleven TFs are the ones we are most interested in,
since we want to see if we can find patterns that correlate with the binding 
scores in C20 microglia.

```{r}
p1 <- ggVennDiagram(list(hiv1_diff, diff_hiv2, diff_hiv3),
              label_alpha = 0,
              category.names= c("HIV1", "HIV2", "HIV3"),
              #set.size = 20,
              label = "count",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "All TFs with different activity")
```

```{r}
p2 <- ggVennDiagram(list(hiv1_atac, hiv2_atac, hiv3_atac),
              label_alpha = 0,
              category.names= c("HIV1", "HIV2", "HIV3"),
              #set.size = 20,
              label = "count",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "TFs with differential binding scores")
```

```{r}
ggarrange(p1, p2, ncol = 2, nrow = 1)
```


## Eleven TFs with different activities {.tabset}

In the boxplots below it can be seen that in all three patients there seems to
be one cluster which is different from two other clusters. It seems that ETS1, 
REST, and YY1 are more active in clusters where all other TFs are less active and
vice versa. All three TFs are repressors, however, this alone does not explain their 
correlation, because for example KLF4 and MXI1 are also repressors.

```{r}
intersect <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/intersect_TFs_different_across_patients_3_clusters")[["x"]]
```


### Patient 1

```{r, fig.width=25, fig.height=10}
df <- k3_df


boxplot_list <- map(intersect, function(tf) {
  ggplot() +
    geom_violin(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf), 
                     fill = df %>% pull("ht_cluster"), alpha = .6)) +
    geom_boxplot(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf)), 
                 alpha = 0) +
    stat_pvalue_manual(df %>% 
                        pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>% 
                         add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))

#annotate_figure(figure, top = text_grob("Patient 1 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)", 
 #              color = "black", face = "bold", size = 25))
```

```{r}
ctcf1 <- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_violin(aes(fill = ht_cluster, alpha = .6)) +
  geom_boxplot(alpha = 0) +
  stat_pvalue_manual(df %>% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %>% 
                       add_xy_position())
ctcf1 <- ctcf1 + labs(title = "CTCF activity in patient 1")
```

---

### Patient 2

```{r, fig.width=25, fig.height=10}
df <- pat2

boxplot_list <- map(intersect, function(tf) {
  ggplot() +
    geom_violin(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf), 
                     fill = df %>% pull("ht_cluster"), alpha = .6)) +
    geom_boxplot(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf)), 
                 alpha = 0) +
    stat_pvalue_manual(df %>% 
                        pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>% 
                         add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))

#annotate_figure(figure, top = text_grob("Patient 2 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)", 
 #              color = "black", face = "bold", size = 25))
```

```{r}
ctcf2 <- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_violin(aes(fill = ht_cluster, alpha = .6)) +
  geom_boxplot(alpha = 0) +
  stat_pvalue_manual(df %>% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %>% 
                       add_xy_position())
ctcf2 <- ctcf2 + labs(title = "CTCF activity in patient 2")
```

---

### Patient 3

```{r, fig.width=25, fig.height=10}
df <- pat3

boxplot_list <- map(intersect, function(tf) {
  ggplot() +
    geom_violin(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf), 
                     fill = df %>% pull("ht_cluster"), alpha = .6)) +
    geom_boxplot(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf)), 
                 alpha = 0) +
    stat_pvalue_manual(df %>% 
                        pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>% 
                         add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))


#annotate_figure(figure, top = text_grob("Patient 3 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)", 
 #              color = "black", face = "bold", size = 25))
```

```{r}
ctcf3 <- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_violin(aes(fill = ht_cluster, alpha = .6)) +
  geom_boxplot(alpha = 0) +
  stat_pvalue_manual(df %>% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %>% 
                       add_xy_position())
ctcf3 <- ctcf3 + labs(title = "CTCF activity in patient 3")
```

---

## CTCF expression across patients

CTCF has been implicated to regulate HIV latency [@Jeffereys]. In the boxplots 
below it can be seen that there is a significant difference between the first 
cluster and the other two clusters in patient1 and patient2. Similarly in 
patient3 there is a difference, however the difference is not as significant.
This might imply that the cluster contains a group of cells which are latently 
infected and, therefore, CTCF is more active.

```{r, fig.width=15, fig.height=5}
ggarrange(ctcf1, ctcf2, ctcf3, ncol = 3, nrow = 1)
```


## Correlation between clusters {.tabset}

The average TF activity of all 125 TFs was computed for every cluster. Based on 
the average TF activity in each cluster, the correlation between clusters was 
computed. The scatter plot below shows the average expression for each TF across all 
three clusters in all three patients and the  corresponding correlation 
coefficients. The correlation matrix visualizes the Spearman correlation 
coefficient between all clusters. Black numbers indicate the not-significant 
p-values. Clusters two and three of all patients seem to be highly 
correlated, with cluster two of patient3 being the exception. Clusters one of 
patient2 and patient3 are highly correlated with each other, but less correlated 
with clusters two and three. Cluster one of patient2 is again an exception, because it 
is not as highly correlated with clusters one of patient2 and patient3, but still
more correlated than with clusters two and three. 

These results led to the hypothesis  that the clusters one, which showed different 
TF activity from clusters two and three correspond to a different infection state. 
Cells of clusters one could be latently infected cells. This hypothesis would be
supported by having a look at the role of TFs with different activity in 
clusters one (see discussion). 

```{r}
# read in 3 Clusters for each of the 3 HIV+ patients

# HIV1_CSF
hiv1_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv", sep = "\t"))[["x"]]
hiv1_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv", sep = "\t"))[["x"]]
hiv1_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv", sep = "\t"))[["x"]]

# HIV2_CSF
hiv2_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster1.tsv", sep = "\t"))[["x"]]
hiv2_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster2.tsv", sep = "\t"))[["x"]]
hiv2_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster3.tsv", sep = "\t"))[["x"]]

# HIV3_CSF
hiv3_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster1.tsv", sep = "\t"))[["x"]]
hiv3_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster2.tsv", sep = "\t"))[["x"]]
hiv3_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster3.tsv", sep = "\t"))[["x"]]

cluster_anno <- c(hiv1_c1, hiv1_c2, hiv1_c3, hiv2_c1, hiv2_c2, hiv2_c3, hiv3_c1, hiv3_c2, hiv3_c3)
```


```{r}
# add cluster information to TF activity
auc_df <- as.data.frame(auc_mtx)

auc_df <- auc_df %>% 
  rownames_to_column("cell") %>% 
  filter(cell %in% cluster_anno) %>% 
  mutate(ht_cluster = case_when(cell %in% hiv1_c1 ~ "hiv1_c1", 
         cell %in% hiv1_c2 ~ "hiv1_c2", 
         cell %in% hiv1_c3 ~ "hiv1_c3",
         cell %in% hiv2_c1 ~ "hiv2_c1",
         cell %in% hiv2_c2 ~ "hiv2_c2",
         cell %in% hiv2_c3 ~ "hiv2_c3",
         cell %in% hiv3_c1 ~ "hiv3_c1",
         cell %in% hiv3_c2 ~ "hiv3_c2",
         cell %in% hiv3_c3 ~ "hiv3_c3"))
```


<details>
<summary>**Correlation computation**</summary>
  
Dataframe of average activity value for each TF in each patient and clusters.


```{r, results = "asis"}
# correlation matrix
df <- auc_df %>%
  group_by(ht_cluster) %>%
  summarize_if(is.numeric, mean) %>% # calculate mean TF activity for each cluster
  gather(TF, value, -ht_cluster) %>% 
  spread(ht_cluster, value) %>% 
  column_to_rownames("TF") 

df %>% head %>% kable (caption = "Average TF activity for each cluster") 

```
Correlation matrix: 

```{r, results = "asis"}
# save correlation matrix
cor_mtx <- df%>% cor(method = "spearman")

# show the correlation matrix
cor_mtx %>% kable(caption = "Correlation matrix of average TF acivity between all clusters")

# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor <- cor.mtest(cor_mtx, conf.level = 0.95, method = "spearman")
```

---
</details>


### Correlation matrix

```{r}
df %>%
  cor(method = "spearman") %>%
  corrplot(
     order = "hclust", # hierarchical cluster order
     addrect = 4, # add boxes 
     p.mat = testCor$p, # add matrix of p-values
     insig = "p-value", # adds p-values as digits, not crosses
     tl.col = "black") # make labels black
```

---

### Scatter matrix

```{r, fig.width = 15, fig.height=15}

GGally::ggpairs(df) + theme_bw() +
  labs(title = "Scatter matrix for all conditions")
```

---


## TF correlation {.tabset}

For the eleven TFs with different activity between clusters, the correlation 
between the TF activities was computed. The scatter matrix and correlation plot
below show very nicely, that the
three TFs YY1, REST, MXI1 are positively correlated, but negatively correlated
with the rest of the TFs and vice versa. The fact that these TF activities 
correlate across all patients indicates that there might be a biological meaning 
behind the different clusters. Having a look at the function of TFs which are up- or
downregulated in cluster one compared to the other clusters could provide 
further insights (see discussion). 

<details>
<summary>**Correlation computation**</summary>


```{r}
intersect_atac <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_intersect_TFs_different_across_patients_3_clusters"))[["x"]]
```


```{r, results = "asis"}
# correlation matrix
auc_df %>%
  select(intersect_atac) %>% head() %>% kable(caption = "TF activity of 11 TFs with different activity between clusters across all cells")
```

Correlation matrix for TFs:

```{r, results = "asis"}
# compute correlation matrix
cor_mtx <- auc_df %>%
  select(intersect_atac) %>% 
  cor(method = "spearman")
# show correlation matrix
cor_mtx %>% head() %>% kable(caption = "Correlation matrix for 11 TFs with different activity between clusters")

# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor <- cor.mtest(cor_mtx, conf.level = 0.95, method = "spearman")
```

---
</details>


### Correlation matrix 

```{r}
df <- auc_df %>%
  select(intersect_atac) %>% 
  cor(method = "spearman") %>%
  corrplot(
  order = "hclust", 
  addrect = 2, # add boxes 
  p.mat = testCor$p, 
  insig = "p-value",
  tl.col = "black") 
```

---


### Scatter matrix

```{r, fig.width=15, fig.height=15}
GGally::ggpairs(auc_df %>%
  select(intersect_atac)) + 
  theme_bw() +
  labs(title = "Scatter matrix for all TFs with different activity between clusters")

```

---

# Gene expression analysis

We will now pool the two clusters
which are significantly different from the third cluster in all patients.
Consequently, in the following analysis we will compare Cluster1 to Cluster2 (pool of two similar 
clusters). The gene expression might give a hint about the identity of these 
two Myeloid2 clusters. 

```{r}
hivmy2 <- subset(hiv7, cell_type == "Myeloid2" & orig.ident %in% c("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))

# add cluster identity to metadata
ht_cluster <- (
  hivmy2@meta.data %>% 
  rownames_to_column("cell") %>% 
  mutate(ht_cluster = ifelse(cell %in% c(hiv1_c1, hiv2_c1, hiv3_c3),
                             "c1", 
                             "c2")) %>% 
  column_to_rownames("cell"))["ht_cluster"]

hivmy2 <- AddMetaData(hivmy2, metadata = ht_cluster, col.name = "ht_cluster")
#saveRDS(hivmy2,"/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv_myeloid2_subclusters")
```

## Quality of the Myeloid 2 clusters

There is an evident difference in library size and number of features between
the two clusters. This could mean that there is a technical difference between
the two clusters. Conversely, it could also mean that there is a biological
difference. For example, if cells of cluster1 represent a latently infected group 
of Myeloid2 cells it could be that gene expression is reduced compared to healthy
uninfected cells from cluster2 due to infection.

One way to investigate this further would be to have a look at the housekeeping 
genes in both clusters. If housekeeping genes were more highly expressed in 
one cluster than in the other this would indicate that the difference is not 
biological, but a technical artefact.

One problem that arises, because there is one cluster which has lower number of 
counts and features, is that further qualitative analysis of gene expression might be skewed 
as well. For this reason the following analysis on the difference between these
two clusters will be mainly qualitative. Due to the low number of cells it 
is not possible to get significant results for the differential gene expression.


```{r}
p1 <- hivmy2@meta.data %>% 
  ggplot() +
  geom_boxplot(aes(x = ht_cluster, y = nFeature_RNA, fill = ht_cluster))

p2 <- hivmy2@meta.data %>% 
  ggplot() +
  geom_violin(aes(x = ht_cluster, y = nFeature_RNA, fill = ht_cluster)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1))

fig <- ggarrange(p1, p2, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob("Quality of Clusters"))
```

## Housekeeping genes

A list of 3804 houseekping genes were used from [@Eisenberg]. A quantitative
analysis might be difficult, because due to the small sample
size and large feature number, we get very high p_values for all genes in 
differential gene expression analysis. Visualization of the
housekeeping gene expression, unfortunately does not help either with the 
interpretation.


<details>
<summary>**Boxplots of housekeeping gene expression**</summary>

```{r}
hk_genes <- (read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/HK_genes.txt"))[["V1"]]
```

```{r}
# filter only genes present in the scRNAseq count matrix
hk_genes_filt <- hk_genes[hk_genes %in% rownames(hivmy2@assays$RNA@counts)]
length(hk_genes)
length(hk_genes_filt)

# Find genes from the housekeeping genes that are differentially expressed 
# between the clusters
hk_diff <- FindAllMarkers(hivmy2, features = hk_genes_filt)
Idents(hivmy2) <- "ht_cluster"
hk_diff <- FindMarkers(hivmy2, features = hk_genes_filt, ident.1 = "c1", ident.2 = "c2")
hk_diff %>% head

```



```{r, fig.width=15, fig.height=10}
test <- hk_genes_filt[1:20]
#VlnPlot(hivmy2, features = test, group.by = "ht_cluster", ncol = 4)

df <- as.data.frame(t(as.matrix((hivmy2@assays$RNA@data)))) %>%
  select(test) %>% 
  rownames_to_column("cell") %>%
  mutate(ht_cluster = case_when(cell %in% c(hiv1_c1, hiv2_c1, hiv3_c3) ~ "c1",
                                TRUE ~ "c2")) %>%
  column_to_rownames("cell")

boxplot_list <- map(test, function(gene){
  ggplot() +
    geom_violin(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(gene), 
                     fill = df %>% pull("ht_cluster"))) +
    geom_boxplot(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(gene)), 
                 alpha = 0) +
    stat_pvalue_manual(df %>% 
                        pairwise_wilcox_test(as.formula(paste0(gene, " ~ ht_cluster"))) %>% 
                         add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(gene)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})


do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 5, nrow = 4))
```

---
</details>

# Appendix

### Determine k for k-means clustering {.tabset}

The plot of within cluster sum of squares shows that the sweet spot for the 
value of k is somewhere between 3 and 5. To determine the best k for k-means
clustering k = 3, k = 4 and k = 5 were tried and the different outcomes compared.
Since k = 4 and k = 5 does not yield an increased number of TFs which differentiate
the different clusters and, therefore, do not add any insights we proceeded with
k = 3 in the following analysis.

```{r}
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                    orig.ident == "HIV1_CSF")
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]

# Decide on a k for clustering
wss = sapply(2:15, function(k) {
  kmeans(x = t(mtx), centers = k)$tot.withinss
})

plot(2:15, wss, type = "b", 
     xlab = "Number of clusters k", 
     ylab = "Total within-clusters sum of square")
```

The results of using differnet k can be seen below. Click on the tabs to see heatmaps of the k-means clustering, as well as statistical test on different TF
acitivities between clusters.

#### k-means Clustering with k = 3

```{r, fig.width=15, fig.height=17}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_pat1 <-draw(Heatmap(t(mtx), 
             column_title = "Myeloid2 cells from patient 1, k = 3", 

             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             ))
             #top_annotation = column_annotation) 
```


After performing k-means cluster we extract the cells from each cluster. 
In the table below you can see the number of cells in each cluster. 

```{r}
### Extract cells from each cluster


# if you draw the heatmap, the results of th eclustering will not change anymore
#png("/media/ag-cherrmann/kmikulik/HIV_microglia/analysis/Myeloid2_cluster/Myeloid2_3clusters.png", width = 15, height = 17, res = 1200)
#ht <- draw(ht_pat1)
# get the columns names of cluster one
c1 <- colnames(t(mtx[column_order(ht_pat1)[[1]],]))
#write.table(c1, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv", sep = "\t")

# get the column names of cluster two
c2 <- colnames(t(mtx[column_order(ht_pat1)[[2]],]))
#write.table(c2, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv", sep = "\t")

# get the column names of cluster 3
c3 <- colnames(t(mtx[column_order(ht_pat1)[[3]],]))
#write.table(c3, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv", sep = "\t")
```


##### number of cells in each cluster 

```{r, results = "asis"}
# summary staticstics
hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3))%>% 
  mutate(ht_cluster = case_when(cell %in% c1 ~ "c1", 
                                cell %in% c2 ~ "c2",
                                cell %in% c3 ~ "c3")) %>%
  group_by(ht_cluster) %>% 
  summarize(count = n()) %>% 
  kable(caption = "number of cells in each cluster")

```

##### Are there significant differences in TF activity between the three clusters?

Performing Kruskal-Wallis-Test to find differences between the three clusters in patient one yields 54 TFs of all 125 TFs which are differentially active between the three clusters(p-value < 0.05). Of the 24 TFs with differential binding scores in the C20 cell line, sixteen have significantly different activity between the clusters.

Now that it is known that there are significant differences between clusters, 
the next step is to identify which cluster/s are different from which other cluster/s.

Pairwise-Wilcox-Test is a non-parametric pairwise test. Again, because 
we repeat the test we will correct for multiple hypothesis testing using the 
FDR method. The results are shown in boxplots below.

</details>

<details> 
<summary>Kruskal-Wallis-Test, p-value adjustment, posthoc tests</summary>

**Kruskal-Wallis Test and p-value adjustments for all 125 TFs**

```{r}
# non-parametric alternative to ANOVA -> Kruskal_wallis test
k3_df <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) 
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))["p.value"]
  #list(tf = tf, p_value = p_values)
})

# plot distribution of p_values
hist(as.numeric(unname(unlist(p_values))), 
     main = "distribution of p_values",
     xlab = "p_values",
     breaks = 20)

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
sign3 <- adj_p_values[adj_p_values<.05]
length(adj_p_values[adj_p_values<0.05])

# get TFs that are significantly different
hiv1_diff <- colnames(auc_mtx)[adj_p_values < .05]

#write.table(diff3, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1")
```

* **Kruskal-Wallis test and p-value adjustment  for 24 TFs also identified by ATAC-seq**


```{r}
# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])

# get TFs that are differentially expressed
hiv1_atac <- overlap_atac$name[adj_p_values < .05]
#write.table(diff3_atac, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1")
```

* **Posthoc tests for k = 3**


The posthoc test was performed for the 16 TFs with significantly different activity between the 3 clusters.

```{r, warning = FALSE}
# We perform a posthoc test
#wilcox_list = list()
wilcox_list <- map(seq.int(1:length(hiv1_atac)), function(i){
  tf <- hiv1_atac[i]
  #print(tf)
  wilcox <- pairwise.wilcox.test(pull(k3_df[tf]), pull(k3_df["ht_cluster"]), 
                                 p.adjust.method = "fdr")
  list(TF = tf, wilcox = wilcox)
})
  
#length(wilcox_list)
#wilcox_list
```

---
</details>

---

#### K-means clustering with k = 4


```{r, fig.width=15, fig.height=17}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht <- draw(Heatmap(t(mtx), 
             column_title = "Myeloid2 cells from patient 2, k = 4", 

             column_km = 4,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             )) 
             #top_annotation = column_annotation) 


```


```{r}
### Extract cells from each cluster of k = 4

# get the columns names of cluster one
cl1 <- colnames(t(mtx[column_order(ht)[[1]],]))

# get the column names of cluster two
cl2 <- colnames(t(mtx[column_order(ht)[[2]],]))


# get the column names of cluster 3
cl3 <- colnames(t(mtx[column_order(ht)[[3]],]))

cl4 <- colnames(t(mtx[column_order(ht)[[4]],]))
```


```{r, results = "asis"}
# summary staticstics
hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = case_when(cell %in% cl1 ~ "c1", 
                                cell %in% cl2 ~ "c2",
                                cell %in% cl3 ~ "c3",
                                cell %in% cl4 ~ "c4")) %>% 
  group_by(ht_cluster) %>% 
  summarize(count = n()) %>% 
  kable(caption = "number of cells in each cluster")

```

##### Are there significant differences in TF activity between the four clusters?


Performing Kruskal-Wallis-Test to find differences between the four clusters in patient one yields 51 TFs of all 125 TFs which are differentially active between the three clusters(p-value < 0.05). Of the 24 TFs with differential binding scores in the C20 cell line, fourteen have significantly different activity between the clusters.

Now that it is known that there are significant differences between clusters, 
the next step is to identify which cluster/s are different from which other 
cluster/s. For this pairwise Wilcox test was used.


</details>

<details> 
<summary>Kruskal-Wallis-Test, p-value adjustment, posthoc tests</summary>

* **Kruskal-Wallis Test and p-value adjustments for all 125 TFs**

```{r}

k4_df <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(cl1,cl2,cl3,cl4)) %>% 
  mutate(ht_cluster = ifelse(cell %in% cl1, "cl1", 
                             ifelse(cell %in% cl2, "cl2", 
                                    ifelse(cell %in% cl3, "cl3", "cl4"))))

# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(k4_df[tf]) ~ ht_cluster, k4_df))["p.value"]
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")

# how many transcription factors are significantly different between the clusters?
sign <- adj_p_values[adj_p_values < 0.05]
length(adj_p_values[adj_p_values<0.05])


# get the TFs that are differentially expressed
diff4 <- colnames(auc_mtx)[adj_p_values < 0.05]

#write.table(diff4, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=4")

```

* **Kruskal-Wallis Test and p-value adjustments for 14 TFs with differential binding scores**


```{r}
# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(k4_df[tf]) ~ ht_cluster, k4_df))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])

# get the TFs that are differentially expressed
diff4_atac <- overlap_atac$name[adj_p_values < .05]

#write.table(diff4_atac, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=4")

```

* **Pairwise-Wilcox_Test for k = 4**

Posthoc test for the 16 TFs with significantly different activity between the four clusters.

```{r, warning = FALSE}
# We perform a posthoc test
#wilcox_list = list()
wilcox_list <- map(seq.int(1:length(diff4_atac)), function(i){
  tf <- diff4_atac[i]
  print(tf)
  wilcox <- pairwise.wilcox.test(pull(k4_df[tf]), pull(k4_df["ht_cluster"]), 
                                 p.adjust.method = "fdr")
  list(TF = tf, wilcox = wilcox)
})
  
#length(wilcox_list)
#wilcox_list
```
---
</details>

---

#### K-means Clustering with k = 5


```{r, fig.width=15, fig.height=17}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht <- draw(Heatmap(t(mtx), 
             column_title = "Myeloid2 cells from patient 1, k = 5", 

             column_km = 5,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             )) 

```


```{r}
##### Extract cells from each cluster for k= 5

# get the columns names of cluster one
clu1 <- colnames(t(mtx[column_order(ht)[[1]],]))

# get the column names of cluster two
clu2 <- colnames(t(mtx[column_order(ht)[[2]],]))


# get the column names of cluster 3
clu3 <- colnames(t(mtx[column_order(ht)[[3]],]))

clu4 <- colnames(t(mtx[column_order(ht)[[4]],]))

clu5 <- colnames(t(mtx[column_order(ht)[[5]], ]))

```


```{r, results = "asis"}
# summary staticstics
hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = case_when(cell %in% clu1 ~ "c1", 
                                cell %in% clu2 ~ "c2",
                                cell %in% clu3 ~ "c3",
                                cell %in% clu4 ~ "c4",
                                cell %in% clu5 ~ "c5")) %>% 
  group_by(ht_cluster) %>% 
  summarize(count = n()) %>% 
  kable(caption = "number of cells in each cluster")

```

##### Are there significant differences in TF activity between the four clusters?

Performing Kruskal-Wallis-Test to find differences between the four clusters in patient one yields 59 TFs of all 125 TFs which are differentially active between the three clusters(p-value < 0.05). Of the 24 TFs with differential binding scores in the C20 cell line, fifteen have significantly different activity between the clusters.

Now that it is known that there are significant differences between clusters, 
the next step is to identify which cluster/s are different from which other 
cluster/s. For this pairwise Wilcox test was used.


</details>

<details> 
<summary>Kruskal-Wallis-Test, p-value adjustment, posthoc tests</summary>

* **Kruskal-Wallis Test and p-value adjustments for all 125 TFs**

```{r}
# all TFs
k5_df <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(clu1,clu2,clu3,clu4, clu5)) %>% 
  mutate(ht_cluster = ifelse(cell %in% clu1, "clu1", 
                             ifelse(cell %in% clu2, "clu2", 
                                    ifelse(cell %in% clu3, "clu3",
                                           ifelse(cell %in% clu4, "clu4", "clu5")))))
n = 1
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(k5_df[tf]) ~ ht_cluster, k5_df))["p.value"]
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")

# how many transcription factors are significantly different between the clusters?
sign <- adj_p_values[adj_p_values < 0.05]
length(adj_p_values[adj_p_values<0.05])


# get the TFs that are differentially expressed
diff5 <- colnames(auc_mtx)[adj_p_values < 0.05]

#write.table(diff5, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=5")
```

* **Kruskal-Wallis Test and p-value adjustments for all 125 TFs**

```{r}
# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(k5_df[tf]) ~ ht_cluster, k5_df))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])

# get the TFs that are differentially expressed
diff5_atac <- overlap_atac$name[adj_p_values < .05]

#write.table(diff5_atac, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=5")
```

* **Pairwise-Wilcox_Test for k = 5**

Posthoc test for the 15 TFs with significantly different activity between the four clusters.

```{r, warning = FALSE}
# We perform a posthoc test
#wilcox_list = list()
wilcox_list <- map(seq.int(1:length(diff5_atac)), function(i){
  tf <- diff5_atac[i]
  print(tf)
  wilcox <- pairwise.wilcox.test(pull(k5_df[tf]), pull(k5_df["ht_cluster"]), 
                                 p.adjust.method = "fdr")
  list(TF = tf, wilcox = wilcox)
})
  
#length(wilcox_list)
#wilcox_list
```

---
</details>

---


### Which Clusters are significantly different for different k? {.tabset}

The results from the statistical tests on difference in TF activity between
clusters can be seen in the boxplots below. Sixteen, fourteen and twelve TFs
were found to be differentially active between the clusters for k equal three, 
four and five respectively. The boxplots below also show very nicely, that 
the data are not normally distributed and, therefore, non-parametric tests, 
like the Kruskal-Wallis-Test and pairwise Wilcox-Test were used.


#### k = 3

```{r, fig.height=20, fig.width=25}

test <- k3_df
#tf.oi <- c("CTCF", "REST", "YY1")
#names(tf.oi) <- tf.oi
test.list <- map(hiv1_atac, function(tf) {
  ggplot() +
  geom_violin(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf), 
                   fill = test %>% pull("ht_cluster"), alpha = .6)) +
  geom_boxplot(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf)), 
               alpha = 0) +
  stat_pvalue_manual(test %>% 
    pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>%
      add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster")) })

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 4))
```

#### k = 4

```{r, fig.width=25, fig.height=15}

test <- k4_df

test.list <- map(diff4_atac, function(tf) {
  ggplot() +
  geom_violin(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf), 
                   fill = test %>% pull("ht_cluster"), alpha = .6)) +
  geom_boxplot(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf)), 
               alpha = 0) +
  stat_pvalue_manual(test %>% 
    pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>%
      add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster")) +
    geom_violin(alpha = 0)
})

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 3))


```


#### k = 5
```{r, fig.width=25, fig.height=15}
test <- k5_df

test.list <- map(diff5_atac, function(tf) {
  ggplot() +
  geom_violin(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf), 
                   fill = test %>% pull("ht_cluster"), alpha = .6)) +
  geom_boxplot(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf)), 
               alpha = 0) +
  stat_pvalue_manual(test %>% 
    pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>%
      add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 3))
```

---

## Conclusion on how many k's to use for k-means clustering

As mentioned, k = 3 was used for the subsequent analysis, since more clusters 
did not add additional insights.

The Venn diagram below shows very nicely, that out of all TFs with different
activity between clusters, 68% overlap for all k (k=3/4/5) that I tested (right plot). 
75% of the TF with differential binding scores in the C20 cell line overlapped
for all k (left plot). 

Therefore, I decided to proceed with 
k = 3 for the following analysis. 

```{r}
diff5_atac <- as.vector(read.table( "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=5"))[["x"]]
diff5 <- as.vector(read.table( "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=5"))[["x"]]


diff4_atac <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=4"))[["x"]]

diff4 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=4"))[["x"]]
# list of TFs from ATAC-seq differntially active between the 3 heatmap 
#clusters from patient 1
diff3_atac <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1"))[["x"]]

# list of TFs differntially active between the 3 heatmap clusters from patient 1
diff3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1"))[["x"]]
```


```{r, fig.width = 12}
p1 <- ggVennDiagram(list(diff3_atac, diff4_atac, diff5_atac),
              label_alpha = 0,
              category.names= c("k = 3", "k = 4", "k = 5"),
              #set.size = 20,
              label = "count",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "TFs with differential binding scores in C20 cell line")
  #scale_fill_gradient (low = "lightblue", high = "darkblue")

p2 <- ggVennDiagram(list(diff3, diff4, diff5),
              label_alpha = 0,
              category.names= c("k = 3", "k = 4", "k = 5"),
              label = "count",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "all TFs with different activity")

fig <- ggarrange(p1, p2, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob("TFs with significantly different activity between clusters for different k", size = 20))
```



### Transfer cluster identitites 

The cluster identities from patient 1 were transferred to patient 2 and patient 3
to assign each cell to one of the 3 clusters. 


<details>
<summary>**Cluster annotation using an integrated referenc**</summary>

Label Transfer

Here we do not integrate the datasets beforehand and we do not modify the query
expression data.

```{
Idents(hiv7) <- "cell_type"
hiv <- subset(hiv7, idents = "Myeloid2")

# the cell type is replaced by the heatmap cluster
hiv1_my2 <- subset(hiv7, cells = c(c1,c2,c3))

clusters <- (
  hiv1_my2@meta.data %>% 
  rownames_to_column("cell") %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) %>% 
  column_to_rownames("cell"))["ht_cluster"]

hiv <- AddMetaData(hiv, metadata = clusters, col.name = "cell_type")

# the seurat_cluster number is replaced by the number of the heatmap cluster
seurat_clusters <- (
  hiv1_my2@meta.data %>% 
  rownames_to_column("cell") %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, 1, ifelse(cell %in% c2, 2, 3))) %>% 
  column_to_rownames("cell"))["ht_cluster"]

hiv <- AddMetaData(hiv, metadata = seurat_clusters, col.name = "seurat_clusters")


# split the object according to the patients
hiv_list <- SplitObject(hiv, split.by = "orig.ident")[3:5]
# normalize each object
for (i in names(hiv_list)) {
    hiv_list[[i]] <- SCTransform(hiv_list[[i]], verbose = FALSE)
}


# patient 2 and patient 3 are our query datasets
query <-merge(hiv_list[[1]], hiv_list[[3]])
reference <- hiv_list[[2]]
reference <- RunPCA(reference, 
                    # by default a partial PCA is run, but we compute mostgular values -> run standard SVD instead
                    approx = FALSE, 
                    verbose = FALSE)

hiv_anchors <- FindTransferAnchors(reference = reference,
                                   query = query, 
                                   dims = 1:30, verbose = FALSE,
                                   k.filter = NA, 
                                   reference.reduction = "pca",
                                   normalization.method = "SCT",
                                   reduction = "pcaproject")

predictions <- TransferData(anchorset = hiv_anchors, 
                            refdata = reference$cell_type,
                            k.weight = 34, dims = 1:30, verbose = FALSE)

query <- AddMetaData(query, metadata = predictions)


hiv2 <- subset(query, orig.ident == "HIV2_CSF")
hiv3 <- subset(query, orig.ident == "HIV3_CSF")
#saveRDS(hiv2,"/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv2_CSF_myleoid2_heatmap_cluster_label_transfer.rds")
#saveRDS(hiv3, "/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv3_CSF_myleoid2_heatmap_cluster_label_transfer.rds")
```
Now that we have transferred the cluster identitities to patient2 and patient3
and saved the results in two separate Seurat objects, we can add the TF activity 
score to the metadata of each patient.

```{r}
hiv2 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv2_CSF_myleoid2_heatmap_cluster_label_transfer.rds")
hiv3 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv3_CSF_myleoid2_heatmap_cluster_label_transfer.rds")

hiv2_scenic <- AddMetaData(hiv2, metadata = auc_mtx,
                           col.name = colnames(auc_mtx))
hiv3_scenic <- AddMetaData(hiv3, metadata = auc_mtx,
                           col.name = colnames(auc_mtx))
```

---
</details>


The heatmaps below show the predicted clusters from label transfer for
patient 2 and patient 3. The clusters obtained after transferring the cluster 
identities from patient 1 to patient2 do not seem to be very good at separating
cells according to TF activity. When you transfer the cluster identities from 
patient1 to patient3 all cells, but
one, are assigned to cluster1. This shows that the label transfer did not work 
very well. This is probably due to the low number of Myeloid2 cells in each sample.

```{r, fig.width=15, fig.height=17}
cluster_anno <- as.data.frame(hiv2$predicted.id)

#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV2_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_hiv2_label <- Heatmap(t(mtx), 
             column_title = "Predicted Clusters from Label Transfer for HIV2_CSF",             
             column_title_gp = gpar(fontsize = 15),
             column_split = paste0("cluster", hiv2$predicted.id),#cluster_anno,
             #column_km = 4,
             #column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la,
             top_annotation = HeatmapAnnotation(df = cluster_anno, name = "predicted cluster")
             ) 
             #top_annotation = column_annotation) 
```


```{r, fig.width=15, fig.height=17}
cluster_anno <- as.data.frame(hiv3$predicted.id)

#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV3_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_hiv3_label <- Heatmap(t(mtx), 
             column_title = "Predicted clusters from Label Transfer for HIV3_CSF", 
             column_title_gp = gpar(fontsize = 15),
             column_split = paste0("cluster", hiv3$predicted.id),#cluster_anno,
             #column_km = 4,
             #column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la,
             top_annotation = HeatmapAnnotation(df = cluster_anno, name = "predicted cluster")
             ) 
             #top_annotation = column_annotation) 

```

```{r, fig.width=15, fig.height=20}
ht_list <- ht_hiv2_label + ht_hiv3_label
draw(ht_list, ht_gap= unit(2, "cm"))
```


### Gene expression analysis of the two Myeloid2 subclusters

#### TF from Farhadian et al.

These TFs were used as marker genes for the microlgia like Myeloid2 cell cluster
[@Farhadian. Is there a group of cells which has a higher expression of these markers?
No clear difference in gene expression can be seen.


<details>
<summary>Boxplots of microglia marker genes</summary>


```{r, fig.width = 15, fig.height=40}
#import the 60 genes that were upregulated in Myeloid-2 in the paper compared to the four other myeloid subsets
paper <- read.table(file = "/media/ag-cherrmann/kmikulik/HIV_microglia/data/Myeloid-2_genes_paper.csv",
                    sep = "\t", header = FALSE, skipNul = TRUE)#, n_max = 60)
paper[1,] <- "APOC1"
colnames(paper) <- ""
paper <- as.vector(paper[,1])


VlnPlot(hivmy2, features = paper, group.by = "ht_cluster")
```

---
</details>


#### Differential gene expression analyisis

<details>
<summary>Differential gene expression analysis</summary>


```{r, results="asis"}
hivmy2 <- NormalizeData(hivmy2, verbose = FALSE)
hivmy2 <- ScaleData(hivmy2, verbose = FALSE)
hivmy2 <- FindVariableFeatures(hivmy2, verbose = FALSE)
hivmy2 <- RunPCA(hivmy2, features = VariableFeatures(hivmy2), verbose = FALSE)
ElbowPlot(hivmy2)
hivmy2 <- RunTSNE(hivmy2, dims= 1:10, verbose = FALSE)
Idents(hivmy2) <- "ht_cluster"
de_genes <- FindAllMarkers(hivmy2, verbose = FALSE)
top10 <- de_genes %>% 
  group_by(cluster) %>%
  top_n(n = 5, wt = avg_log2FC) 
top10 %>% 
  kable()
```

---
</details>


#### TF from Gosselin et al.

<details>
<summary>Microglia TF gene expression</summary>


```{r}
Idents(hiv4) <- "cell_type"
# Find Markers that distinguish My2 from the other Myeloid clusters
my2_markers <- FindMarkers(hiv4, ident.1  = "Myeloid2",
            ident.2  = c("Myeloid1", "Myeloid3", "Myeloid4", "Myeloid5"))
my2_markers <- my2_markers %>% rownames_to_column(var = "gene")
```

```{r, fig.width=15, fig.height=40}
microglia_genes <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/Microglia_gene_signatur_found_in_HIV1_HIV2_samples")
m1 <- my2_markers %>% filter(gene %in% microglia_genes [["gene_name"]] & avg_log2FC >0)
# remove the genes that were already used by the Farhadian paper
m1 <- m1[["gene"]][m1[["gene"]] %in% paper == FALSE]

VlnPlot(hivmy2, features = m1, group.by = "ht_cluster", ncol = 5)

```

---
</details>

#### Which of the 11 TFs are mentioned by Gosselin et al.?

<details>
<summary>TFs also mentioned by Gosselin et al.</summary>

```{r, fig.width=15, fig.height=12}
tfs_microglia <- (read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/TFs_microglia_Glass_paper.txt"))[["V1"]]

gosselin <- tfs_microglia[tfs_microglia %in% overlap_atac$name]

VlnPlot(hivmy2, features = gosselin, group.by = "ht_cluster", ncol = 4)
```

---
</details>


# References




</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("TF_activity_analysis_of_myeloid2_cluster.Rmd");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
