<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="kmikulik" />


<title>TF-activity analysis of myeloid2 cells</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Introduction.html">Introduction</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="HIV_CSF_dataset_analysis.html">1. CSF</a>
    </li>
    <li>
      <a href="TF_activity_analysis_of_myeloid2_cluster.html">2. TF activity Myeloid2</a>
    </li>
    <li>
      <a href="Triculture_datset_analysis_and_integration_Farhadian_dataset.html">3. Integration CSF &amp; triculture</a>
    </li>
    <li>
      <a href="Triculture_dataset_GRN_with_Scenic.html">4. TF activity iMg</a>
    </li>
  </ul>
</li>
<li>
  <a href="Discussion.html">Discussion</a>
</li>
<li>
  <a href="Methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Katharina782/report">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">TF-activity analysis of myeloid2 cells</h1>
<h4 class="author">kmikulik</h4>
<h4 class="date">7 12 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-20
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttreef7243f8882ef1868800d5f0e420f83d5e4ba39bbtargetblankf7243f8a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Katharina782/report/tree/f7243f8882ef1868800d5f0e420f83d5e4ba39bb" target="_blank">f7243f8</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttreef7243f8882ef1868800d5f0e420f83d5e4ba39bbtargetblankf7243f8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Katharina782/report/tree/f7243f8882ef1868800d5f0e420f83d5e4ba39bb" target="_blank">f7243f8</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/Independent_cell_type_identification_Farhadian_dataset_cache/
    Ignored:    analysis/Triculture_dataset_analysis_and_integration_Farhadian_dataset_cache/

Unstaged changes:
    Modified:   analysis/Methods.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd</code>) and HTML (<code>docs/TF_activity_analysis_of_myeloid2_cluster.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/bb965a8fe71e28af44fba8f3042204001cddaa0c/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">bb965a8</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
<td>
wflow_publish(“analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/91e34e269b1382b79f47740d549c4f68abc4cd46/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">91e34e2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
<td>
wflow_publish(“analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/2a95e927f953c093f2e48cb2b6551d152f735267/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/fec6c6e9430260f415f472edf227cbf6434c20ae/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">fec6c6e</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/e5e16d2fafe64661ad4400d6f14ff63f9bb0efce/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">e5e16d2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-17
</td>
<td>
wflow_publish(files = rmd)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/5bb41240b1a4dff64e2b5fded39e9d88dc15805b/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">5bb4124</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/5d4419ed1127a959edfae38d9632cfbafc52671d/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">5d4419e</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
wflow_publish(c("analysis/_site.yml“,”analysis/index.Rmd“,”analysis/references.bib",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/550997780789f62be907ec2095aa2ff2a1fd9087/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">5509977</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
wflow_publish(“analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd”)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
<td>
recent updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/dde10972b37ffd997474921968223514c8cb1b1f/docs/TF_activity_analysis_of_myeloid2_cluster.html" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
<td>
recent updates
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style>
body {
text-align: justify}
</style>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, collapse = TRUE, message = FALSE,
                      cache = TRUE, autodep = TRUE)
set.seed(42)</code></pre>
<pre class="r"><code>library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggplot2)
library(dplyr)
library(ggrepel)
#library(harmony)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(knitr)
library(ggVennDiagram)
library(corrplot)

# &quot;Multitasking/Multiprocessing&quot;
library(future)
#plan(&quot;multisession&quot;, workers = 24)</code></pre>
<div id="grn-inference-with-scenic" class="section level1">
<h1>GRN inference with Scenic</h1>
<p>Using Scenic I tried to infer Gene regulatory networks from the gene expression data. The output is a matrix of TF (Transcription Factor) activity per cell. In this case the activity matrix contains activity of 125 different transcription factors.</p>
<p>First, we wanted to see if we can identify patterns in the TF activity in the Myeloid2 cells. We compared the TF activity to TFs shown to have differential binding scores between three infection states (uninfected, latent infection, active infection) in primary microglia (bulk ATACseq). For the comparison with the TF activity in Myeloid2 cells we focused on the TF with differential binding scores between <strong>uninfected</strong> and <strong>latently infected</strong> states.</p>
<p>Can we see patterns of increased/decreased TF activity in the Myeloid2 cells that might correspond to the binding scores between infection states in primary microglia?</p>
<ul>
<li>read in Seurat object</li>
</ul>
<pre class="r"><code>hiv7 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds&quot;)
hiv4 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/non_integrated_HIV1_HIV2_4samples_seurat_object.rds&quot;)</code></pre>
<ul>
<li>read in AuCell matrix (output from Scenic)</li>
</ul>
<pre class="r"><code>auc_mtx &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/aucell_matrix_10k_hvg.tsv&quot;, sep = &quot;\t&quot;)</code></pre>
<ul>
<li>Combine AuCell Matrix and Seurat object</li>
</ul>
<pre class="r"><code># add auc matrix to metadata of hiv4 seurat object
hiv7_scenic &lt;- AddMetaData(hiv7, 
                    metadata = auc_mtx, 
                    col.name = colnames(auc_mtx))</code></pre>
</div>
<div id="bulk-atacseq-footprinting-results" class="section level1">
<h1>Bulk ATACseq footprinting results</h1>
<p>Using bulk ATACseq data from primary microglia at three different conditions, active infection, latent infection, uninfected, differential binding scores were obtained using TOBIAS. Here, I will use the TFs which were previously shown to have differential binding scores in these primary microglia. Since I expect the CSF Myeloid cells to be latently infected, because they have been obtained from patients receiving ART, I will use the TFs with differential binding scores between latent infection and uninfected. We would like to know if the TF activity inferred from RNAseq data of Myeloid2 cells correlates with these differential binding scores. In other words, we want to know if the patterns of differential TF activity are similar as the patterns of differential TF binding scores. For example, if a TF was shown to be more bound in latently infected primary microglia compared to uninfected primary microglia, we would expect the TF to be more active in a group of Myeloid2 cells which are latently infected compared to a group of Myeloid2 cells which are uninfected.</p>
<p>When you overlap the TF from the TF activity matrix obtained by running SCNEIC on the 10.000 most highly variable genes with the differntially bound TFs, 24 of the TFs overlap.</p>
<p>Using less than 10.000 most highly variable genes as an input matrix for SCENIC results in a lower number of TFs which overlap. Increasing the number of input features for SCENIC increases the number of TFs included in the TF activity matrix</p>
<pre class="r"><code># read in dataframe containing differntial binding scores between conditions
#
# red = latently infected
# gfp = active infected
# uninfected
tfs_ana &lt;- read.table(&quot;/media/ag-cherrmann/projects/06_HIV_Microglia/data/atacseq/data-2020-11-06/tobias/TOBIAS_snakemake/footprint_mglia2_GlassTF_17-03/TFBS/bindetect_results.txt&quot;, sep = &quot;\t&quot;, header = TRUE)

# filter TFs with significant p values between latent infection and uninfected
# I decided to use a threshold of p-value &lt; 1e-50, because due to the large number 
# of data, all p-values will be very high
top_tfs &lt;- tfs_ana %&gt;% filter(uninf_red_pvalue &lt; 1e-50)

# keep only the TFs in the dataframe which overlap between the two datasets.
overlap_atac &lt;- top_tfs %&gt;% filter(name %in% colnames(auc_mtx))

# extract names of TFs more bound in either condition
up_latent &lt;- overlap_atac$name[overlap_atac$uninf_red_change &lt; 0]
up_uninf &lt;- overlap_atac$name[overlap_atac$uninf_red_change &gt; 0]</code></pre>
<div id="volcano-plot-atac-seq-data" class="section level2">
<h2>Volcano Plot ATAC-seq data</h2>
<p>Interestingly, only four TFs found in the TF activity matrix of Myeloid2 cells are more bound in latent infection than in uninfected primary microglia.</p>
<pre class="r"><code>
top_tfs %&gt;% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = ifelse(uninf_red_change &lt; 0, &quot;up_latent&quot;, &quot;up_uninf&quot;)) %&gt;%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  mutate(label = ifelse(name %in% overlap_atac$name, name, NA)) %&gt;% 
  ggplot(aes(x = uninf_red_change,
             y = -log10(uninf_red_pvalue),
             col = condition, 
             label = label)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-50), col = &quot;red&quot;) +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = &quot;TFs with differential binding scores between latent infection and uninfected primary microglia&quot;) +
  ylab(&quot;-log10(p_value)&quot;) +
  xlab(&quot;differential binding score&quot;)
Warning: Removed 51 rows containing missing values (geom_text).</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/5bb41240b1a4dff64e2b5fded39e9d88dc15805b/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">5bb4124</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-6-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="compare-tf-activities-between-clusters-and-look-for-patterns" class="section level1">
<h1>Compare TF activities between clusters and look for patterns</h1>
<p>The idea was to cluster the Myeloid2 cells based on their TF activity using Hierarchical clustering. To do this I perofrmed hierarchical clustering on Myeloid2 cells from patient1 (HIV1_CSF) with k=3/4/5. The idea for using only patient 1 was that the other two patients might be used for validation of the clustering later on. If the clusters are biological, I might be able to transfer the clusters onto the Myeloid2 cells from patient2 (HIV2_CSF) and patient3 (HIV3_CSF). If the clusters also separate the Myeloid2 cells of the other two patients nicely, this would validate that there is a biological meaning behind the clusters.</p>
<p>However, as we will see the label transfer of the cluster identities did not work, probably due to the small sample size. There are only 146 in total in the Myeloid2 cluster of infected patients.</p>
<p>The alternative strategy was to perform independent hierarchical clustering across all patients and compare the TF activity between clusters. Can we find TF activities which differentiate clusters across patients and find correlations between clusters across patients?</p>
<ol style="list-style-type: decimal">
<li>Determine k for k-means hierarchical clustering</li>
<li>Transfer the cluster identities to patient2 and patient3</li>
<li>Hierarchical Clustering for all 3 patients separately</li>
</ol>
<pre class="r"><code>

# TF annotations
la &lt;- rowAnnotation(type_TF = 
                      case_when(colnames(auc_mtx) %in% up_latent ~ &quot;up_latent&quot;,
                                        colnames(auc_mtx) %in% up_uninf  ~ &quot;up_uninf&quot;,
                                        TRUE ~&quot;unknown&quot;),
                     name = &quot;type_TF&quot;,
                     col = list(type_TF = c(&quot;up_latent&quot; =  &quot;red&quot;, 
                                            &quot;up_uninf&quot; =  &quot;pink&quot;,
                                            &quot;unknown&quot; = &quot;forestgreen&quot;)))
  </code></pre>
<div id="split-heatmap-into-3-separate-heatmaps-according-to-the-different-patients" class="section level2">
<h2>Split Heatmap into 3 separate heatmaps according to the different patients</h2>
<pre class="r"><code># 3 heatmaps with all Scenic TFs
ht_list &lt;- map(seq.int(1:3), function(n) {
  patients &lt;- c (&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;) # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df),]
  # heatmap
  ht &lt;- Heatmap(t(mtx), 
               column_title = paste0(&quot;patient: &quot;, patients[n]), 

               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
               show_row_names = TRUE
               ) 
               #top_annotation = column_annotation) 
  list(name = patients[n], heatmap = ht)
})


ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap + la</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-8-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heatmap-containing-only-tfs-found-by-atac-seq-data-analysis" class="section level2">
<h2>Heatmap containing only TFs found by ATAC-seq data analysis</h2>
<pre class="r"><code># 3 heatmaps, but only the TFs which overlap with Anas analysis
ht_list &lt;- map(seq.int(1:3), function(n) {
  patients &lt;- c (&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;) # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  # select TFs which overlap with Anas analysis
  mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), overlap_atac$name]
  # heatmap
  ht &lt;- Heatmap(t(mtx), 
               column_title = paste0(&quot;patient: &quot;, patients[n]), 

               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
               right_annotation = rowAnnotation(type_TF = ifelse(colnames(auc_mtx[,overlap_atac$name]) %in% up_latent, &quot;up_latent&quot;, &quot;up_uninf&quot;),
                                                name = &quot;type_TF&quot;,
                                                col = list(type_TF = c(&quot;up_latent&quot; =  &quot;red&quot;,
                                                                       &quot;up_uninf&quot; =  &quot;pink&quot;))))
  list(name = patients[n], heatmap = ht)
})

  
ht_list = ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap
draw(ht_list, ht_gap = unit(1, &quot;cm&quot;))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-9-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="determine-k-for-k-means-hierarchical-clustering" class="section level1 tabset">
<h1 class="tabset">1. Determine k for k-means hierarchical clustering</h1>
<p>The plot of within-cluster som of squares shows that the sweet spot for the value of k is somewhere between 3 and 5. To determine the best k for k-means clustering k = 3, k = 4 and k = 5 were tried and the different outcomes compared. Since k = 4 and k = 5 does not yield an increased number of TFs which differntiate the different clusters, and therefore do not add any insights we proceeded with k = 3 in the following analysis.</p>
<p>The results of each k can be seen below. Click on the tabs to see heatmaps of the hierarchical clustering, as well as statistical test on different TF acitivity between clusters.</p>
<pre class="r"><code>df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                    orig.ident == &quot;HIV1_CSF&quot;)
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]

# Decide on a k for clustering
wss = sapply(2:15, function(k) {
  kmeans(x = t(mtx), centers = k)$tot.withinss
})

plot(2:15, wss, type = &quot;b&quot;, 
     xlab = &quot;Number of clusters k&quot;, 
     ylab = &quot;Total within-clusters sum of square&quot;)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-10-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-10-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-10-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-10-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="k-3-hierarchical-clustering" class="section level2">
<h2>K = 3 Hierarchical Clustering</h2>
<div id="heatmap-for-patient-1" class="section level3">
<h3>Heatmap for patient 1</h3>
<pre class="r"><code>#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV1_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_pat1 &lt;-draw(Heatmap(t(mtx), 
             column_title = &quot;Myeloid2 cells from patient 1, k = 3&quot;, 

             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la
             ))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-11-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>             #top_annotation = column_annotation) </code></pre>
</div>
<div id="extract-cells-from-each-cluster" class="section level3">
<h3>Extract cells from each cluster</h3>
<pre class="r"><code># if you draw the heatmap, the results of th eclustering will not change anymore
#png(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/analysis/Myeloid2_cluster/Myeloid2_3clusters.png&quot;, width = 15, height = 17, res = 1200)
#ht &lt;- draw(ht_pat1)
# get the columns names of cluster one
c1 &lt;- colnames(t(mtx[column_order(ht_pat1)[[1]],]))
#write.table(c1, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv&quot;, sep = &quot;\t&quot;)

# get the column names of cluster two
c2 &lt;- colnames(t(mtx[column_order(ht_pat1)[[2]],]))
#write.table(c2, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv&quot;, sep = &quot;\t&quot;)

# get the column names of cluster 3
c3 &lt;- colnames(t(mtx[column_order(ht_pat1)[[3]],]))
#write.table(c3, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv&quot;, sep = &quot;\t&quot;)</code></pre>
<div id="number-of-cells-in-each-cluster" class="section level4">
<h4>number of cells in each cluster</h4>
<pre class="r"><code># summary staticstics
hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV1_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(c1,c2,c3)) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c1, &quot;c1&quot;, ifelse(cell %in% c2, &quot;c2&quot;, &quot;c3&quot;))) %&gt;% 
  #group_by(ht_cluster) %&gt;% 
  select(ht_cluster, CTCF) %&gt;% 
  group_by(ht_cluster) %&gt;% 
  summarise(
    count = n(),
    mean = mean(CTCF),
    sd = sd(CTCF)) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">ht_cluster</th>
<th align="right">count</th>
<th align="right">mean</th>
<th align="right">sd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">c1</td>
<td align="right">15</td>
<td align="right">0.0417362</td>
<td align="right">0.0163909</td>
</tr>
<tr class="even">
<td align="left">c2</td>
<td align="right">23</td>
<td align="right">0.0156238</td>
<td align="right">0.0128713</td>
</tr>
<tr class="odd">
<td align="left">c3</td>
<td align="right">34</td>
<td align="right">0.0156253</td>
<td align="right">0.0139310</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="tf-activity-per-cluster" class="section level3">
<h3>TF activity per cluster</h3>
<p>Plot TF activity for all TFs which showed differential binding scores in primary microglia. For some TFs it appears that there are differences in activity between the clusters. The next step will be to quantitate these differences.</p>
<p>From the violin plots below it becomes evident, that the acitivity is not normally distributed. Therefore, we will use Kruskal-Wallis test (non-parametric) instead of ANOVA (parametric).</p>
<pre class="r"><code># make boxplots  and violin plots for all TFs
p &lt;- list()
for (n in 1:24){
  tf &lt;- overlap_atac$name[n]
  p[[n]] &lt;- hiv7_scenic@meta.data %&gt;% 
    rownames_to_column(&quot;cell&quot;) %&gt;% 
    filter(orig.ident == &quot;HIV1_CSF&quot;,
           cell_type == &quot;Myeloid2&quot;,
           cell %in% c(c1,c2,c3)) %&gt;% 
    mutate(ht_cluster = ifelse(cell %in% c1, &quot;c1&quot;, ifelse(cell %in% c2, &quot;c2&quot;, &quot;c3&quot;))) %&gt;% 
    #group_by(ht_cluster) %&gt;% 
    select(ht_cluster, tf) %&gt;% 
    ggplot(aes_string(x = &quot;ht_cluster&quot;, y = tf)) +
    geom_boxplot(aes(fill = ht_cluster)) + 
    geom_violin(aes(alpha = 0.1)) +
    stat_compare_means()
}

do.call(grid.arrange,c(p, ncol = 4, top= &quot;Transcription factor activity per cluster&quot;))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-14-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-14-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-14-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-14-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-14-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="are-there-significant-differences-in-tf-activity-between-the-three-clusters" class="section level3">
<h3>Are there significant differences in TF activity between the three clusters?</h3>
<p>ANOVA test is parametric and assumes normally distributed data, while Kruskal-Wallis-Test compares the mean of one or more groups, but is non-parametric. As can be seen in the boxplots above the data is not normally distributed, therefore Kruskal-Wallis-Test is more appropriate in this case. In both cases the Null-Hypothesis is that there is no difference between the groups.</p>
<p>Since I am repeating the test 125 times, I adjust the p-values with the FDR method. This methods is more powerful than more conservative methdos, like bonferroni, Holm or Hochberg.</p>
<p>Performing Kruskal-Wallis-Test to find differences between the three clusters in patient1 yields 54 TFs of all 125 TFs which are differentially active between the three clusters (p-value &lt; 0.05). Of the 24 TFs with differential binding scores in the C20 cell line, sixteen have significantly different activity between the clusters.</p>
<p>Now that we know that there are significant differences between clusters, we want to identify which cluster/s are different from which other cluster/s.</p>
<p>Pairwise-Wilcox-Test is a non-parametric pairwise test. Again, because we repeat the test we will correct for multiple hypothesis testing using the FDR method. The results are shown in boxplots below.</p>
</details>
<details>
<summary>
Kruskal-Wallis-Test, p-value adjustment, posthoc tests
</summary>
<p><strong>Kruskal-Wallis Test and p-value adjustments for all 125 TFs</strong></p>
<pre class="r"><code># non-parametric alternative to ANOVA -&gt; Kruskal_wallis test
k3_df &lt;- hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV1_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(c1,c2,c3)) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c1, &quot;c1&quot;, ifelse(cell %in% c2, &quot;c2&quot;, &quot;c3&quot;))) 
# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))[&quot;p.value&quot;]
  #list(tf = tf, p_value = p_values)
})

# plot distribution of p_values
hist(as.numeric(unname(unlist(p_values))), 
     main = &quot;distribution of p_values&quot;,
     xlab = &quot;p_values&quot;,
     breaks = 20)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-15-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-15-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)

hist((adj_p_values), 
     main = &quot;distribution of p_values&quot;,
     xlab = &quot;p_values&quot;,
     breaks = 20)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-2">
Past versions of unnamed-chunk-15-2.png
</button>
</p>
<div id="fig-unnamed-chunk-15-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-15-2.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-15-2.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>

# how many transcription factors are significantly different between the clusters?
sign3 &lt;- adj_p_values[adj_p_values&lt;.05]
length(adj_p_values[adj_p_values&lt;0.05])
[1] 64

# get TFs that are significantly different
hiv1_diff &lt;- colnames(auc_mtx)[adj_p_values &lt; .05]

#write.table(diff3, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1&quot;)</code></pre>
<ul>
<li><strong>Kruskal-Wallis test and p-value adjustment for 24 TFs also identified by ATAC-seq</strong></li>
</ul>
<pre class="r"><code># check 24 TFs with Kruskal Wallis test
p_values &lt;- map(seq.int(1:24), function(n){
  tf &lt;- overlap_atac$name[n]
  p_values &lt;- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))[&quot;p.value&quot;]
})
#sign_p_values &lt;- p_values[p_values &lt; 0.01]

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values&lt;0.05])
[1] 16

# get TFs that are differentially expressed
hiv1_atac &lt;- overlap_atac$name[adj_p_values &lt; .05]
#write.table(diff3_atac, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1&quot;)</code></pre>
<ul>
<li><strong>Posthoc tests for k = 3</strong></li>
</ul>
<p>For the 16 TFs with significantly different activity between the 3 clusters:</p>
<pre class="r"><code># We perform a posthoc test
#wilcox_list = list()
wilcox_list &lt;- map(seq.int(1:length(hiv1_atac)), function(i){
  tf &lt;- hiv1_atac[i]
  #print(tf)
  wilcox &lt;- pairwise.wilcox.test(pull(k3_df[tf]), pull(k3_df[&quot;ht_cluster&quot;]), 
                                 p.adjust.method = &quot;fdr&quot;)
  list(TF = tf, wilcox = wilcox)
})
  
#length(wilcox_list)
#wilcox_list</code></pre>
<hr />
</details>
</div>
</div>
<div id="k-4-hierarchical-clustering" class="section level2">
<h2>K = 4 Hierarchical clustering</h2>
<div id="heatmap-for-patient-1-1" class="section level3">
<h3>Heatmap for patient 1</h3>
<pre class="r"><code>#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV1_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht &lt;- draw(Heatmap(t(mtx), 
             column_title = &quot;Myeloid2 cells from patient 2, k = 4&quot;, 

             column_km = 4,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la
             )) </code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-18-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-18-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-18-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>             #top_annotation = column_annotation) 
</code></pre>
</div>
<div id="extract-cells-from-each-cluster-of-k-4" class="section level3">
<h3>Extract cells from each cluster of k = 4</h3>
<pre class="r"><code># get the columns names of cluster one
cl1 &lt;- colnames(t(mtx[column_order(ht)[[1]],]))

# get the column names of cluster two
cl2 &lt;- colnames(t(mtx[column_order(ht)[[2]],]))


# get the column names of cluster 3
cl3 &lt;- colnames(t(mtx[column_order(ht)[[3]],]))

cl4 &lt;- colnames(t(mtx[column_order(ht)[[4]],]))</code></pre>
</div>
<div id="are-there-significant-differences-in-tf-activity-between-the-four-clusters" class="section level3">
<h3>Are there significant differences in TF activity between the four clusters?</h3>
<p>ANOVA test is parametric and assumes normally distributed data, while Kruskal-Wallis-Test compares the mean of one or more groups, but is non-parametric. As can be seen in the boxplots above the data is not normally distributed, therefore Kruskal-Wallis-Test is more appropriate in this case. In both cases the Null-Hypothesis is that there is no difference between the groups.</p>
<p>Since I am repeating the test 125 times, I adjust the p-values with the FDR method. This methods is more powerful than more conservative methdos, like bonferroni, Holm or Hochberg.</p>
<p>Performing Kruskal-Wallis-Test to find differences between the four clusters in patient1 yields 51 TFs of all 125 TFs which are differentially active between the three clusters (p-value &lt; 0.05). Of the 24 TFs with differential binding scores in the C20 cell line, fourteen have significantly different activity between the clusters.</p>
<p>Now that we know that there are significant differences between clusters, we want to identify which cluster/s are different from which other cluster/s.</p>
<p>Pairwise-Wilcox-Test is a non-parametric pairwise test. Again, because we repeat the test we will correct for multiple hypothesis testing using the FDR method. The results are shown in boxplots below.</p>
</details>
<details>
<summary>
Kruskal-Wallis-Test, p-value adjustment, posthoc tests
</summary>
<ul>
<li><strong>Kruskal-Wallis Test and p-value adjustments for all 125 TFs</strong></li>
</ul>
<pre class="r"><code>
k4_df &lt;- hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV1_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(cl1,cl2,cl3,cl4)) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% cl1, &quot;cl1&quot;, 
                             ifelse(cell %in% cl2, &quot;cl2&quot;, 
                                    ifelse(cell %in% cl3, &quot;cl3&quot;, &quot;cl4&quot;))))

# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(k4_df[tf]) ~ ht_cluster, k4_df))[&quot;p.value&quot;]
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)

# how many transcription factors are significantly different between the clusters?
sign &lt;- adj_p_values[adj_p_values &lt; 0.05]
length(adj_p_values[adj_p_values&lt;0.05])
[1] 55


# get the TFs that are differentially expressed
diff4 &lt;- colnames(auc_mtx)[adj_p_values &lt; 0.05]

#write.table(diff4, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=4&quot;)</code></pre>
<ul>
<li><strong>Kruskal-Wallis Test and p-value adjustments for 14 TFs with differential binding scores</strong></li>
</ul>
<pre class="r"><code># check 24 TFs with Kruskal Wallis test
p_values &lt;- map(seq.int(1:24), function(n){
  tf &lt;- overlap_atac$name[n]
  p_values &lt;- unlist(kruskal.test(pull(k4_df[tf]) ~ ht_cluster, k4_df))[&quot;p.value&quot;]
})
#sign_p_values &lt;- p_values[p_values &lt; 0.01]

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values&lt;0.05])
[1] 15

# get the TFs that are differentially expressed
diff4_atac &lt;- overlap_atac$name[adj_p_values &lt; .05]

#write.table(diff4_atac, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=4&quot;)</code></pre>
<ul>
<li><strong>Pairwise-Wilcox_Test for k = 4</strong></li>
</ul>
<p>For the 16 TFs with significantly different activity between the four clusters:</p>
<pre class="r"><code># We perform a posthoc test
#wilcox_list = list()
wilcox_list &lt;- map(seq.int(1:length(diff4_atac)), function(i){
  tf &lt;- diff4_atac[i]
  print(tf)
  wilcox &lt;- pairwise.wilcox.test(pull(k4_df[tf]), pull(k4_df[&quot;ht_cluster&quot;]), 
                                 p.adjust.method = &quot;fdr&quot;)
  list(TF = tf, wilcox = wilcox)
})
[1] &quot;NFIL3&quot;
[1] &quot;MXI1&quot;
[1] &quot;CREB1&quot;
[1] &quot;NFATC2&quot;
[1] &quot;MEF2A&quot;
[1] &quot;REST&quot;
[1] &quot;LEF1&quot;
[1] &quot;YY1&quot;
[1] &quot;MYB&quot;
[1] &quot;ETS1&quot;
[1] &quot;JUNB&quot;
[1] &quot;CEBPB&quot;
[1] &quot;CTCF&quot;
[1] &quot;ETV6&quot;
[1] &quot;MEF2C&quot;
  
#length(wilcox_list)
#wilcox_list</code></pre>
<hr />
</details>
</div>
</div>
<div id="k-5-hierarchical-clustering" class="section level2">
<h2>K = 5 Hierarchical Clustering</h2>
<div id="heatmap-for-patient-1-2" class="section level3">
<h3>Heatmap for patient 1</h3>
<pre class="r"><code>#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV1_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht &lt;- draw(Heatmap(t(mtx), 
             column_title = &quot;Myeloid2 cells from patient 1, k = 5&quot;, 

             column_km = 5,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la
             )) </code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-23-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-23-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-23-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="extract-cells-from-each-cluster-for-k-5" class="section level3">
<h3>Extract cells from each cluster for k= 5</h3>
<pre class="r"><code># get the columns names of cluster one
clu1 &lt;- colnames(t(mtx[column_order(ht)[[1]],]))

# get the column names of cluster two
clu2 &lt;- colnames(t(mtx[column_order(ht)[[2]],]))


# get the column names of cluster 3
clu3 &lt;- colnames(t(mtx[column_order(ht)[[3]],]))

clu4 &lt;- colnames(t(mtx[column_order(ht)[[4]],]))

clu5 &lt;- colnames(t(mtx[column_order(ht)[[5]], ]))</code></pre>
</div>
<div id="are-there-significant-differences-in-tf-activity-between-the-four-clusters-1" class="section level3">
<h3>Are there significant differences in TF activity between the four clusters?</h3>
<p>ANOVA test is parametric and assumes normally distributed data, while Kruskal-Wallis-Test compares the mean of one or more groups, but is non-parametric. As can be seen in the boxplots above the data is not normally distributed, therefore Kruskal-Wallis-Test is more appropriate in this case. In both cases the Null-Hypothesis is that there is no difference between the groups.</p>
<p>Since I am repeating the test 125 times, I adjust the p-values with the FDR method. This methods is more powerful than more conservative methdos, like bonferroni, Holm or Hochberg.</p>
<p>Performing Kruskal-Wallis-Test to find differences between the five clusters in patient1 yields 59 TFs of all 125 TFs which are differentially active between the three clusters (p-value &lt; 0.05). Of the 24 TFs with differential binding scores in the C20 cell line, fifteen have significantly different activity between the clusters.</p>
<p>Now that we know that there are significant differences between clusters, we want to identify which cluster/s are different from which other cluster/s.</p>
<p>Pairwise-Wilcox-Test is a non-parametric pairwise test. Again, because we repeat the test we will correct for multiple hypothesis testing using the FDR method. The results are shown in boxplots below.</p>
</details>
<details>
<summary>
Kruskal-Wallis-Test, p-value adjustment, posthoc tests
</summary>
<ul>
<li><strong>Kruskal-Wallis Test and p-value adjustments for all 125 TFs</strong></li>
</ul>
<pre class="r"><code># all TFs
k5_df &lt;- hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV1_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(clu1,clu2,clu3,clu4, clu5)) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% clu1, &quot;clu1&quot;, 
                             ifelse(cell %in% clu2, &quot;clu2&quot;, 
                                    ifelse(cell %in% clu3, &quot;clu3&quot;,
                                           ifelse(cell %in% clu4, &quot;clu4&quot;, &quot;clu5&quot;)))))
n = 1
# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(k5_df[tf]) ~ ht_cluster, k5_df))[&quot;p.value&quot;]
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)

# how many transcription factors are significantly different between the clusters?
sign &lt;- adj_p_values[adj_p_values &lt; 0.05]
length(adj_p_values[adj_p_values&lt;0.05])
[1] 59


# get the TFs that are differentially expressed
diff5 &lt;- colnames(auc_mtx)[adj_p_values &lt; 0.05]

#write.table(diff5, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=5&quot;)</code></pre>
<ul>
<li><strong>Kruskal-Wallis Test and p-value adjustments for all 125 TFs</strong></li>
</ul>
<pre class="r"><code># check 24 TFs with Kruskal Wallis test
p_values &lt;- map(seq.int(1:24), function(n){
  tf &lt;- overlap_atac$name[n]
  p_values &lt;- unlist(kruskal.test(pull(k5_df[tf]) ~ ht_cluster, k5_df))[&quot;p.value&quot;]
})
#sign_p_values &lt;- p_values[p_values &lt; 0.01]

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values&lt;0.05])
[1] 15

# get the TFs that are differentially expressed
diff5_atac &lt;- overlap_atac$name[adj_p_values &lt; .05]

#write.table(diff5_atac, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=5&quot;)</code></pre>
<ul>
<li><strong>Pairwise-Wilcox_Test for k = 5</strong></li>
</ul>
<p>For the 15 TFs with significantly different activity between the four clusters:</p>
<pre class="r"><code># We perform a posthoc test
#wilcox_list = list()
wilcox_list &lt;- map(seq.int(1:length(diff5_atac)), function(i){
  tf &lt;- diff5_atac[i]
  print(tf)
  wilcox &lt;- pairwise.wilcox.test(pull(k5_df[tf]), pull(k5_df[&quot;ht_cluster&quot;]), 
                                 p.adjust.method = &quot;fdr&quot;)
  list(TF = tf, wilcox = wilcox)
})
[1] &quot;NFIL3&quot;
[1] &quot;MXI1&quot;
[1] &quot;CREB1&quot;
[1] &quot;NFATC2&quot;
[1] &quot;MEF2A&quot;
[1] &quot;REST&quot;
[1] &quot;LEF1&quot;
[1] &quot;KLF4&quot;
[1] &quot;YY1&quot;
[1] &quot;ETS1&quot;
[1] &quot;JUNB&quot;
[1] &quot;CEBPB&quot;
[1] &quot;CTCF&quot;
[1] &quot;ETV6&quot;
[1] &quot;MEF2C&quot;
  
#length(wilcox_list)
#wilcox_list</code></pre>
</details>
</div>
</div>
</div>
<div id="section" class="section level1 unnumbered">
<h1 class="unnumbered"></h1>
<div id="which-clusters-are-sifnificantly-different-for-different-k" class="section level2 tabset">
<h2 class="tabset">Which Clusters are sifnificantly different for different k?</h2>
<p>The results from the statistical tests on difference in TF activity between clusters can be seen in the boxplots below. Sixteen, fourteen and twelve TFs were found to be differentially active between the clusters for k equal three, four and five respectively.</p>
<div id="k-3" class="section level3">
<h3>k = 3?</h3>
<pre class="r"><code>library(rstatix)
library(ggpubr)

test &lt;- k3_df
#tf.oi &lt;- c(&quot;CTCF&quot;, &quot;REST&quot;, &quot;YY1&quot;)
#names(tf.oi) &lt;- tf.oi
test.list &lt;- map(hiv1_atac, function(tf) {
  ggplot() +
  geom_boxplot(aes(x = test %&gt;% pull(&quot;ht_cluster&quot;), y = test %&gt;% pull(tf), 
                   fill = test %&gt;% pull(&quot;ht_cluster&quot;))) +
  stat_pvalue_manual(test %&gt;% 
    pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;%
      add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;))
})

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 4))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-28-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-28-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-28-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-28-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-28-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-28-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-28-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="k-4" class="section level3">
<h3>k = 4</h3>
<pre class="r"><code>library(rstatix)
library(ggpubr)

test &lt;- k4_df

test.list &lt;- map(diff4_atac, function(tf) {
  ggplot() +
  geom_boxplot(aes(x = test %&gt;% pull(&quot;ht_cluster&quot;), y = test %&gt;% pull(tf), 
                   fill = test %&gt;% pull(&quot;ht_cluster&quot;))) +
  stat_pvalue_manual(test %&gt;% 
    pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;%
      add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;))
})

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 3))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-29-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-29-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-29-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-29-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-29-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="k-5" class="section level3">
<h3>k = 5</h3>
<pre class="r"><code>library(rstatix)
library(ggpubr)

test &lt;- k5_df

test.list &lt;- map(diff5_atac, function(tf) {
  ggplot() +
  geom_boxplot(aes(x = test %&gt;% pull(&quot;ht_cluster&quot;), y = test %&gt;% pull(tf), 
                   fill = test %&gt;% pull(&quot;ht_cluster&quot;))) +
  stat_pvalue_manual(test %&gt;% 
    pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;%
      add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;))
})

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 3))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-30-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-30-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-30-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-30-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-30-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>{-}</p>
</div>
</div>
<div id="conclusion-on-how-many-ks-to-use-for-hierarchical-clustering" class="section level2">
<h2>Conclusion on how many k’s to use for hierarchical clustering</h2>
<p>As mentioned, k = 3 was used for the subsequent analysis, since more clusters did not add additional insights.</p>
<p>The Venn diagram below shows very nicely, that out of all TFs with different activity between clusters, 68% overlap for all k (k=3/4/5) that I tested (right plot). 75% of the TF with differential binding scores in the C20 cell line overlapped for all k (left plot).</p>
<p>Therefore, I decided to proceed with k = 3 for the following analysis.</p>
<pre class="r"><code>diff5_atac &lt;- as.vector(read.table( &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=5&quot;))[[&quot;x&quot;]]
diff5 &lt;- as.vector(read.table( &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=5&quot;))[[&quot;x&quot;]]


diff4_atac &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=4&quot;))[[&quot;x&quot;]]

diff4 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=4&quot;))[[&quot;x&quot;]]
# list of TFs from ATAC-seq differntially active between the 3 heatmap 
#clusters from patient 1
diff3_atac &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1&quot;))[[&quot;x&quot;]]

# list of TFs differntially active between the 3 heatmap clusters from patient 1
diff3 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1&quot;))[[&quot;x&quot;]]</code></pre>
<pre class="r"><code>p1 &lt;- ggVennDiagram(list(diff3_atac, diff4_atac, diff5_atac),
              label_alpha = 0,
              category.names= c(&quot;k = 3&quot;, &quot;k = 4&quot;, &quot;k = 5&quot;),
              #set.size = 20,
              label = &quot;percent&quot;,
              label_size = 6) +
   scale_fill_distiller(palette = &quot;RdBu&quot;) +
  labs(title = &quot;TFs with differential binding scores in C20 cell line&quot;)
  #scale_fill_gradient (low = &quot;lightblue&quot;, high = &quot;darkblue&quot;)

p2 &lt;- ggVennDiagram(list(diff3, diff4, diff5),
              label_alpha = 0,
              category.names= c(&quot;k = 3&quot;, &quot;k = 4&quot;, &quot;k = 5&quot;),
              label = &quot;percent&quot;,
              label_size = 6) +
   scale_fill_distiller(palette = &quot;RdBu&quot;) +
  labs(title = &quot;all TFs with different activity&quot;)

fig &lt;- ggarrange(p1, p2, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob(&quot;TFs with significantly different activity  between clusters for different k&quot;, size = 20))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-32-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="transfer-the-3-cluster-identitites-to-patient2-and-patient3-tabset" class="section level1">
<h1>2. Transfer the 3 cluster identitites to patient2 and patient3 {tabset}</h1>
<p>The cluster identities from patient 1 were transferred to patient 2 and patient 3 to assign each cell to one of the 3 clusters.</p>
<div id="label-transfer" class="section level2">
<h2>Label Transfer</h2>
<p><strong>Cluster annotation using an integrated reference</strong></p>
<p>Here we do not integrate the datasets beforehand and we do not modify the query expression data.</p>
<pre class="{"><code>Idents(hiv7) &lt;- &quot;cell_type&quot;
hiv &lt;- subset(hiv7, idents = &quot;Myeloid2&quot;)

# the cell type is replaced by the heatmap cluster
hiv1_my2 &lt;- subset(hiv7, cells = c(c1,c2,c3))

clusters &lt;- (
  hiv1_my2@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c1, &quot;c1&quot;, ifelse(cell %in% c2, &quot;c2&quot;, &quot;c3&quot;))) %&gt;% 
  column_to_rownames(&quot;cell&quot;))[&quot;ht_cluster&quot;]

hiv &lt;- AddMetaData(hiv, metadata = clusters, col.name = &quot;cell_type&quot;)

# the seurat_cluster number is replaced by the number of the heatmap cluster
seurat_clusters &lt;- (
  hiv1_my2@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c1, 1, ifelse(cell %in% c2, 2, 3))) %&gt;% 
  column_to_rownames(&quot;cell&quot;))[&quot;ht_cluster&quot;]

hiv &lt;- AddMetaData(hiv, metadata = seurat_clusters, col.name = &quot;seurat_clusters&quot;)


# split the object according to the patients
hiv_list &lt;- SplitObject(hiv, split.by = &quot;orig.ident&quot;)[3:5]
# normalize each object
for (i in names(hiv_list)) {
    hiv_list[[i]] &lt;- SCTransform(hiv_list[[i]], verbose = FALSE)
}


# patient 2 and patient 3 are our query datasets
query &lt;-merge(hiv_list[[1]], hiv_list[[3]])
reference &lt;- hiv_list[[2]]
reference &lt;- RunPCA(reference, 
                    # by default a partial PCA is run, but we compute mostgular values -&gt; run standard SVD instead
                    approx = FALSE, 
                    verbose = FALSE)

hiv_anchors &lt;- FindTransferAnchors(reference = reference,
                                   query = query, 
                                   dims = 1:30, verbose = FALSE,
                                   k.filter = NA, 
                                   reference.reduction = &quot;pca&quot;,
                                   normalization.method = &quot;SCT&quot;,
                                   reduction = &quot;pcaproject&quot;)

predictions &lt;- TransferData(anchorset = hiv_anchors, 
                            refdata = reference$cell_type,
                            k.weight = 34, dims = 1:30, verbose = FALSE)

query &lt;- AddMetaData(query, metadata = predictions)


hiv2 &lt;- subset(query, orig.ident == &quot;HIV2_CSF&quot;)
hiv3 &lt;- subset(query, orig.ident == &quot;HIV3_CSF&quot;)
#saveRDS(hiv2,&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv2_CSF_myleoid2_heatmap_cluster_label_transfer.rds&quot;)
#saveRDS(hiv3, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv3_CSF_myleoid2_heatmap_cluster_label_transfer.rds&quot;)</code></pre>
<p>Now that we have transferred the cluster identitities to patient2 and patient3 and saved the results in two separate Seurat objects, we can add the TF activity score to the metadata of each patient.</p>
<pre class="r"><code>hiv2 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv2_CSF_myleoid2_heatmap_cluster_label_transfer.rds&quot;)
hiv3 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv3_CSF_myleoid2_heatmap_cluster_label_transfer.rds&quot;)

hiv2_scenic &lt;- AddMetaData(hiv2, metadata = auc_mtx,
                           col.name = colnames(auc_mtx))
hiv3_scenic &lt;- AddMetaData(hiv3, metadata = auc_mtx,
                           col.name = colnames(auc_mtx))</code></pre>
</div>
<div id="predicted-cluster-labels" class="section level2">
<h2>Predicted cluster labels</h2>
<p>The heatmaps below show the predicted clusters from label transfer for patient 2 and patient 3</p>
<p>The clusters obtained after transferring the cluster identities from patient 1 to patient2 do not seem to be very good at separating cells according to TF activity.</p>
<p>When you transfer the cluster identities from patient1 to patient3 all cells, but one, are assigned to cluster1. This shows that the label transfer did not work very well. This is probably due to the low number of Myeloid2 cells in each sample.</p>
<pre class="r"><code>cluster_anno &lt;- as.data.frame(hiv2$predicted.id)

#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV2_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_hiv2_label &lt;- Heatmap(t(mtx), 
             column_title = &quot;Predicted Clusters from Label Transfer for HIV2_CSF&quot;,             
             column_title_gp = gpar(fontsize = 15),
             column_split = paste0(&quot;cluster&quot;, hiv2$predicted.id),#cluster_anno,
             #column_km = 4,
             #column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la,
             top_annotation = HeatmapAnnotation(df = cluster_anno, name = &quot;predicted cluster&quot;)
             ) 
             #top_annotation = column_annotation) </code></pre>
<pre class="r"><code>cluster_anno &lt;- as.data.frame(hiv3$predicted.id)

#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV3_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_hiv3_label &lt;- Heatmap(t(mtx), 
             column_title = &quot;Predicted clusters from Label Transfer for HIV3_CSF&quot;, 
             column_title_gp = gpar(fontsize = 15),
             column_split = paste0(&quot;cluster&quot;, hiv3$predicted.id),#cluster_anno,
             #column_km = 4,
             #column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la,
             top_annotation = HeatmapAnnotation(df = cluster_anno, name = &quot;predicted cluster&quot;)
             ) 
             #top_annotation = column_annotation) </code></pre>
<pre class="r"><code>ht_list &lt;- ht_hiv2_label + ht_hiv3_label
draw(ht_list, ht_gap= unit(2, &quot;cm&quot;))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-36-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-36-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-36-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-36-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-36-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="hierarchical-clustering-for-all-3-patients-separately" class="section level1 tabset">
<h1 class="tabset">3. Hierarchical clustering for all 3 patients separately</h1>
<p>Since the label transfer of the 3 cluster identities from patient1 did not seem to work very well, we next tried to perform hierarchical clustering using k = 3 for all 3 patients separately. This way we might be able to see if the TF activities in the 3 clusters show similar patterns across the different patients.</p>
<p>In the two heatmaps below the clusters look more sensible than the clusters observed after cluster label transfer.</p>
<div id="patient2" class="section level2">
<h2>patient2</h2>
<p><strong>K-means clustering with k = 3 for patient 2</strong></p>
<div id="draw-heatmap-for-patient-3" class="section level4">
<h4>1. Draw Heatmap for Patient 3</h4>
<pre class="r"><code>#map(seq.int(1:5), function(k){
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV2_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht2 &lt;- Heatmap(t(mtx), 
             column_title = &quot;Patient 2, HIV2_CSF&quot;, 
             column_title_gp = gpar(fontsize = 15),
             #column_split = paste0(&quot;cluster&quot;, hiv2$predicted.id),#cluster_anno,
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la,
             #top_annotation = HeatmapAnnotation(df = cluster_anno, name = &quot;predicted cluster&quot;)
             ) 
             #top_annotation = column_annotation) 
ht_pat2 &lt;- draw(ht2)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-37-1">
Past versions of unnamed-chunk-37-1.png
</button>
</p>
<div id="fig-unnamed-chunk-37-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-37-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="kruskal_wallis-test-patient2" class="section level4">
<h4>2. Kruskal_Wallis Test Patient2</h4>
<pre class="r"><code># extract the cells from the 3 clusters 
# get the cell names of cluster one
c1 &lt;- colnames(t(mtx[column_order(ht_pat2)[[1]],]))
# get the cell names of cluster two
c2 &lt;- colnames(t(mtx[column_order(ht_pat2)[[2]],]))
# get the cell names of cluster 3
c3 &lt;- colnames(t(mtx[column_order(ht_pat2)[[3]],]))


# add cluster identities to Seurat object metadata and return metadata as dataframe
pat2 &lt;- hiv7_scenic@meta.data %&gt;% rownames_to_column(&quot;cell&quot;)
pat2 &lt;- pat2 %&gt;% 
  filter(orig.ident == &quot;HIV2_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(c1,c2,c3)) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c1, &quot;c1&quot;, ifelse(cell %in% c2, &quot;c2&quot;, &quot;c3&quot;))) 


# check all 125 TFs
# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(pat2[tf]) ~ ht_cluster, pat2))[&quot;p.value&quot;]
  #list(tf = tf, p_value = p_values)
})


# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)
length(adj_p_values[adj_p_values&lt;0.05])
[1] 71
## [1] 66

# get the TFs taht are differntially active
diff_hiv2 &lt;- colnames(auc_mtx)[adj_p_values &lt; 0.05]


# check 24 TFs with Kruskal Wallis test
p_values &lt;- map(seq.int(1:24), function(n){
  tf &lt;- overlap_atac$name[n]
  p_values &lt;- unlist(kruskal.test(pull(pat2[tf]) ~ ht_cluster, pat2))[&quot;p.value&quot;]
})
#sign_p_values &lt;- p_values[p_values &lt; 0.01]

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values&lt;0.05])
[1] 19
## [1] 18

# get the TFs that are differentially expressed
hiv2_atac &lt;- overlap_atac$name[adj_p_values &lt; .05]</code></pre>
</div>
</div>
<div id="patient-3" class="section level2">
<h2>patient 3</h2>
<p><strong>K-means clustering with k = 3 for patient 3</strong></p>
<div id="draw-heatmap-for-patient-3-1" class="section level4">
<h4>1. Draw Heatmap for Patient 3</h4>
<pre class="r"><code>df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV3_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht3 &lt;- Heatmap(t(mtx), 
             column_title = &quot;Patient 3, HIV3_CSF&quot;, 
             column_title_gp = gpar(fontsize = 15),
             #column_split = paste0(&quot;cluster&quot;, hiv2$predicted.id),#cluster_anno,
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la,
             #top_annotation = HeatmapAnnotation(df = cluster_anno, name = &quot;predicted cluster&quot;)
             ) 
             #top_annotation = column_annotation) ich
ht_pat3 &lt;- draw(ht3)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-39-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-39-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-39-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-39-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-39-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="kruskal-wallis-test-patient-3" class="section level4">
<h4>2. Kruskal-Wallis test Patient 3</h4>
<pre class="r"><code># extract the cells from the 3 clusters 
# get the cell names of cluster one
c1 &lt;- colnames(t(mtx[column_order(ht_pat3)[[1]],]))
# get the cell names of cluster two
c2 &lt;- colnames(t(mtx[column_order(ht_pat3)[[2]],]))
# get the cell names of cluster 3
c3 &lt;- colnames(t(mtx[column_order(ht_pat3)[[3]],]))


# add cluster identities to Seurat object metadata and return metadata as dataframe
pat3 &lt;- hiv7_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(orig.ident == &quot;HIV3_CSF&quot;,
         cell_type == &quot;Myeloid2&quot;,
         cell %in% c(c1,c2,c3)) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c1, &quot;c1&quot;, ifelse(cell %in% c2, &quot;c2&quot;, &quot;c3&quot;))) 


# check all 125 TFs
# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(pat3[tf]) ~ ht_cluster, pat3))[&quot;p.value&quot;]
  #list(tf = tf, p_value = p_values)
})


# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)
length(adj_p_values[adj_p_values&lt;0.05])
[1] 66
## [1] 66

# get the TFs taht are differntially active
diff_hiv3 &lt;- colnames(auc_mtx)[adj_p_values &lt; 0.05]


# check 24 TFs with Kruskal Wallis test
p_values &lt;- map(seq.int(1:24), function(n){
  tf &lt;- overlap_atac$name[n]
  p_values &lt;- unlist(kruskal.test(pull(pat3[tf]) ~ ht_cluster, pat3))[&quot;p.value&quot;]
})
#sign_p_values &lt;- p_values[p_values &lt; 0.01]

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values&lt;0.05])
[1] 18
## [1] 18

# get the TFs that are differentially expressed
hiv3_atac &lt;- overlap_atac$name[adj_p_values &lt; .05]</code></pre>
</div>
</div>
<div id="comparison-between-all-3-patients" class="section level2">
<h2>comparison between all 3 patients</h2>
<div id="heatmaps-of-k-means-clustering-with-k-3-for-all-three-patients" class="section level4">
<h4>Heatmaps of k-means clustering with k = 3 for all three patients</h4>
<pre class="r"><code># patient 1
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == &quot;HIV1_CSF&quot;)
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht1 &lt;- Heatmap(t(mtx), 
             column_title = &quot;Patient 1, HIV1_CSF&quot;, 
             column_title_gp = gpar(fontsize = 15),
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             right_annotation = la
             )
             #top_annotation = column_annotation) </code></pre>
<pre class="r"><code>ht_list &lt;- ht1 + ht2 + ht3
draw(ht_list, ht_gap= unit(2, &quot;cm&quot;), 
     column_title = &quot;Independent k-means clustering (k = 3) for all patients&quot;,
     column_title_gp = gpar(fontsize = 20) )</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-42-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-42-1">
Past versions of unnamed-chunk-42-1.png
</button>
</p>
<div id="fig-unnamed-chunk-42-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-42-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-42-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-42-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-42-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="section-1" class="section level1 unnumbered">
<h1 class="unnumbered"></h1>
<div id="which-tfs-show-significantly-different-activity-between-the-clusters-across-patients" class="section level2">
<h2>Which TFs show significantly different activity between the clusters across patients?</h2>
<p>Since the independent hierarchical clustering for all three patients yields clusters which look more reasonable, we will compare the three clusters we obtain across the three patients to see if there is a pattern between them.</p>
<p>We already saw that there are a few TFs which show significantly different activity between the clusters in patient1.</p>
<p>Next, we want to see if the same group of TFs characterize the three clusters in all three patients or if they are different. We will test which TFs are significantly different between clusters and then use a pairwise Wilcox test to determine which clusters are different.</p>
<div id="are-tfs-differentiating-the-three-clusters-similar-across-all-three-patients" class="section level3">
<h3>Are TFs differentiating the three clusters similar across all three patients?</h3>
<p>In the Venn diagram below it can be seen that 46 with significantly different activities between clusters overlap across patients.</p>
<p>Additionally, eleven of the TFs found to have differential binding scores between infection conditions in C20 microglia cell line show different activity between the three clusters across all patients. These are the TFs we are most interested in, since we want to see if we can find patterns that correlate with the binding scores in C20 microglia.</p>
<pre class="r"><code>p1 &lt;- ggVennDiagram(list(hiv1_diff, diff_hiv2, diff_hiv3),
              label_alpha = 0,
              category.names= c(&quot;HIV1&quot;, &quot;HIV2&quot;, &quot;HIV3&quot;),
              #set.size = 20,
              label = &quot;count&quot;,
              label_size = 6) +
   scale_fill_distiller(palette = &quot;RdBu&quot;) +
  labs(title = &quot;TF (ATAC-seq) significantly different between 3 clusters&quot;)</code></pre>
<pre class="r"><code>p2 &lt;- ggVennDiagram(list(hiv1_atac, hiv2_atac, hiv3_atac),
              label_alpha = 0,
              category.names= c(&quot;HIV1&quot;, &quot;HIV2&quot;, &quot;HIV3&quot;),
              #set.size = 20,
              label = &quot;count&quot;,
              label_size = 6) +
   scale_fill_distiller(palette = &quot;RdBu&quot;) +
  labs(title = &quot;TF (ATAC-seq) significantly different between 3 clusters&quot;)</code></pre>
<pre class="r"><code>ggarrange(p1, p2, ncol = 2, nrow = 1)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-45-1">
Past versions of unnamed-chunk-45-1.png
</button>
</p>
<div id="fig-unnamed-chunk-45-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-45-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="tfs-show-different-activity-in-one-of-the-three-clusters" class="section level2 tabset">
<h2 class="tabset">11 TFs show different activity in one of the three clusters</h2>
<p>In the boxplots below it can be seen that in all three patients there seems to be one cluster which is different from two other clusters.</p>
<pre class="r"><code>intersect &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/intersect_TFs_different_across_patients_3_clusters&quot;)[[&quot;x&quot;]]</code></pre>
<div id="patient-1" class="section level3">
<h3>Patient 1</h3>
<pre class="r"><code>library(rstatix)
library(ggpubr)

df &lt;- k3_df


boxplot_list &lt;- map(intersect, function(tf) {
  ggplot() +
    geom_boxplot(aes(x = df %&gt;% pull(&quot;ht_cluster&quot;), y = df %&gt;% pull(tf),
                     fill = df %&gt;% pull(&quot;ht_cluster&quot;))) +
    stat_pvalue_manual(df %&gt;% 
                        pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;% 
                         add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-47-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-47-1">
Past versions of unnamed-chunk-47-1.png
</button>
</p>
<div id="fig-unnamed-chunk-47-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-47-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-47-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-47-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-47-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-47-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-47-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
#annotate_figure(figure, top = text_grob(&quot;Patient 1 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)&quot;, 
 #              color = &quot;black&quot;, face = &quot;bold&quot;, size = 25))</code></pre>
<pre class="r"><code>ctcf1 &lt;- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_boxplot(aes(fill = ht_cluster)) +
  stat_pvalue_manual(df %&gt;% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %&gt;% 
                       add_xy_position())
ctcf1 &lt;- ctcf1 + labs(title = &quot;CTCF activity in patient 1&quot;)</code></pre>
</div>
<div id="patient-2" class="section level3">
<h3>Patient 2</h3>
<pre class="r"><code>library(rstatix)
library(ggpubr)

df &lt;- pat2

boxplot_list &lt;- map(intersect, function(tf) {
  ggplot() +
    geom_boxplot(aes(x = df %&gt;% pull(&quot;ht_cluster&quot;), y = df %&gt;% pull(tf),
                     fill = df %&gt;% pull(&quot;ht_cluster&quot;))) +
    stat_pvalue_manual(df %&gt;% 
                        pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;% 
                         add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-49-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-49-1">
Past versions of unnamed-chunk-49-1.png
</button>
</p>
<div id="fig-unnamed-chunk-49-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-49-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-49-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-49-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-49-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
#annotate_figure(figure, top = text_grob(&quot;Patient 2 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)&quot;, 
 #              color = &quot;black&quot;, face = &quot;bold&quot;, size = 25))</code></pre>
<pre class="r"><code>ctcf2 &lt;- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_boxplot(aes(fill = ht_cluster)) +
  stat_pvalue_manual(df %&gt;% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %&gt;% 
                       add_xy_position())
ctcf2 &lt;- ctcf2 + labs(title = &quot;CTCF activity in patient 2&quot;)</code></pre>
</div>
<div id="patient-3-1" class="section level3">
<h3>Patient 3</h3>
<pre class="r"><code>library(rstatix)
library(ggpubr)

df &lt;- pat3

boxplot_list &lt;- map(intersect, function(tf) {
  ggplot() +
    geom_boxplot(aes(x = df %&gt;% pull(&quot;ht_cluster&quot;), y = df %&gt;% pull(tf),
                     fill = df %&gt;% pull(&quot;ht_cluster&quot;))) +
    stat_pvalue_manual(df %&gt;% 
                        pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;% 
                         add_xy_position()) +
    xlab(&quot;heatmap cluster&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;heatmap cluster&quot;))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" width="2400" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-51-1">
Past versions of unnamed-chunk-51-1.png
</button>
</p>
<div id="fig-unnamed-chunk-51-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-51-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>

#annotate_figure(figure, top = text_grob(&quot;Patient 3 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)&quot;, 
 #              color = &quot;black&quot;, face = &quot;bold&quot;, size = 25))</code></pre>
</div>
</div>
<div id="section-2" class="section level2 unnumbered">
<h2 class="unnumbered"></h2>
<div id="ctcf-expression-across-patients" class="section level3">
<h3>CTCF expression across patients</h3>
<p>CTCF has been implicated to regulate HIV latency <span class="citation">[@Jeffereys]</span>. In the boxplots below it can be seen that there is a significant difference between the first cluster and the other two clusters in patient1 and patient2. Similarly in patient3 there is a difference, however the difference is not as significant. This might imply that cluster contains a group of cells which are latently infected and therefore CTCF is more active.</p>
<pre class="r"><code>ctcf3 &lt;- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_boxplot(aes(fill = ht_cluster)) +
  stat_pvalue_manual(df %&gt;% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %&gt;% 
                       add_xy_position())
ctcf3 &lt;- ctcf3 + labs(title = &quot;CTCF activity in patient 3&quot;)</code></pre>
<pre class="r"><code>ggarrange(ctcf1, ctcf2, ctcf3, ncol = 3, nrow = 1)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-53-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-53-1">
Past versions of unnamed-chunk-53-1.png
</button>
</p>
<div id="fig-unnamed-chunk-53-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-53-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-53-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-53-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-53-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="correlations-between-clusters-across-patients" class="section level2">
<h2>Correlations between clusters across patients</h2>
<p>Using a correlation matrix we can visualize how clusters are correlated across patients:</p>
<pre class="r"><code># read in 3 Clusters for each of the 3 HIV+ patients

# HIV1_CSF
hiv1_c1 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv1_c2 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv1_c3 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]

# HIV2_CSF
hiv2_c1 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster1.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv2_c2 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster2.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv2_c3 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster3.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]

# HIV3_CSF
hiv3_c1 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster1.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv3_c2 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster2.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv3_c3 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster3.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]

cluster_anno &lt;- c(hiv1_c1, hiv1_c2, hiv1_c3, hiv2_c1, hiv2_c2, hiv2_c3, hiv3_c1, hiv3_c2, hiv3_c3)</code></pre>
<pre class="r"><code># add cluster information to TF activity
auc_df &lt;- as.data.frame(auc_mtx)

auc_df &lt;- auc_df %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  filter(cell %in% cluster_anno) %&gt;% 
  mutate(ht_cluster = case_when(cell %in% hiv1_c1 ~ &quot;hiv1_c1&quot;, 
         cell %in% hiv1_c2 ~ &quot;hiv1_c2&quot;, 
         cell %in% hiv1_c3 ~ &quot;hiv1_c3&quot;,
         cell %in% hiv2_c1 ~ &quot;hiv2_c1&quot;,
         cell %in% hiv2_c2 ~ &quot;hiv2_c2&quot;,
         cell %in% hiv2_c3 ~ &quot;hiv2_c3&quot;,
         cell %in% hiv3_c1 ~ &quot;hiv3_c1&quot;,
         cell %in% hiv3_c2 ~ &quot;hiv3_c2&quot;,
         cell %in% hiv3_c3 ~ &quot;hiv3_c3&quot;))</code></pre>
<div id="correlation-between-clusters-based-on-all-125-tfs" class="section level3">
<h3>Correlation between clusters based on all 125 TFs</h3>
<p>For every cluster the correlation is calculated based on the TF activities of all 125 TFs identified by SCENIC.</p>
<details>
*
<summary>
Correlation computation
</summary>
<ul>
<li></li>
</ul>
<p>Dataframe of average activity value for each TF in each patient and clusters.</p>
<pre class="r"><code># correlation matrix
df &lt;- auc_df %&gt;%
  group_by(ht_cluster) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  #column_to_rownames(&quot;ht_cluster&quot;) %&gt;% 
  #add_rownames() %&gt;% 
  gather(TF, value, -ht_cluster) %&gt;% 
  spread(ht_cluster, value) %&gt;% 
  column_to_rownames(&quot;TF&quot;)

df %&gt;% 
  head %&gt;% 
  kable ()</code></pre>
<table>
<colgroup>
<col width="8%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">hiv1_c1</th>
<th align="right">hiv1_c2</th>
<th align="right">hiv1_c3</th>
<th align="right">hiv2_c1</th>
<th align="right">hiv2_c2</th>
<th align="right">hiv2_c3</th>
<th align="right">hiv3_c1</th>
<th align="right">hiv3_c2</th>
<th align="right">hiv3_c3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AHR</td>
<td align="right">0.0327704</td>
<td align="right">0.0534783</td>
<td align="right">0.0293072</td>
<td align="right">0.0275802</td>
<td align="right">0.0263492</td>
<td align="right">0.0222778</td>
<td align="right">0.0133156</td>
<td align="right">0.0075556</td>
<td align="right">0.0449111</td>
</tr>
<tr class="even">
<td align="left">ATF3</td>
<td align="right">0.0684460</td>
<td align="right">0.0965516</td>
<td align="right">0.0944977</td>
<td align="right">0.0487516</td>
<td align="right">0.0959799</td>
<td align="right">0.0910900</td>
<td align="right">0.0488404</td>
<td align="right">0.0828738</td>
<td align="right">0.1026111</td>
</tr>
<tr class="odd">
<td align="left">BACH1</td>
<td align="right">0.0745916</td>
<td align="right">0.0957086</td>
<td align="right">0.0960815</td>
<td align="right">0.0470817</td>
<td align="right">0.0956613</td>
<td align="right">0.0900737</td>
<td align="right">0.0491457</td>
<td align="right">0.0677356</td>
<td align="right">0.1003448</td>
</tr>
<tr class="even">
<td align="left">BCL11A</td>
<td align="right">0.0453606</td>
<td align="right">0.0580407</td>
<td align="right">0.0544656</td>
<td align="right">0.0336095</td>
<td align="right">0.0582619</td>
<td align="right">0.0503748</td>
<td align="right">0.0308033</td>
<td align="right">0.0406827</td>
<td align="right">0.0519544</td>
</tr>
<tr class="odd">
<td align="left">BCL6</td>
<td align="right">0.0300111</td>
<td align="right">0.0629203</td>
<td align="right">0.0389951</td>
<td align="right">0.0135000</td>
<td align="right">0.0646071</td>
<td align="right">0.0232917</td>
<td align="right">0.0162400</td>
<td align="right">0.0441667</td>
<td align="right">0.0393667</td>
</tr>
<tr class="even">
<td align="left">BHLHE40</td>
<td align="right">0.0453855</td>
<td align="right">0.0353485</td>
<td align="right">0.0268406</td>
<td align="right">0.0459291</td>
<td align="right">0.0220228</td>
<td align="right">0.0297271</td>
<td align="right">0.0300765</td>
<td align="right">0.0320580</td>
<td align="right">0.0355449</td>
</tr>
</tbody>
</table>
<p>Correlation matrix:</p>
<pre class="r"><code>df %&gt;% cor %&gt;% kable()</code></pre>
<table>
<colgroup>
<col width="8%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">hiv1_c1</th>
<th align="right">hiv1_c2</th>
<th align="right">hiv1_c3</th>
<th align="right">hiv2_c1</th>
<th align="right">hiv2_c2</th>
<th align="right">hiv2_c3</th>
<th align="right">hiv3_c1</th>
<th align="right">hiv3_c2</th>
<th align="right">hiv3_c3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">hiv1_c1</td>
<td align="right">1.0000000</td>
<td align="right">0.8138023</td>
<td align="right">0.8909121</td>
<td align="right">0.6494445</td>
<td align="right">0.8803939</td>
<td align="right">0.7504363</td>
<td align="right">0.8767961</td>
<td align="right">0.5096081</td>
<td align="right">0.8158899</td>
</tr>
<tr class="even">
<td align="left">hiv1_c2</td>
<td align="right">0.8138023</td>
<td align="right">1.0000000</td>
<td align="right">0.9186181</td>
<td align="right">0.4144831</td>
<td align="right">0.9261440</td>
<td align="right">0.9100636</td>
<td align="right">0.6789498</td>
<td align="right">0.5949834</td>
<td align="right">0.9306527</td>
</tr>
<tr class="odd">
<td align="left">hiv1_c3</td>
<td align="right">0.8909121</td>
<td align="right">0.9186181</td>
<td align="right">1.0000000</td>
<td align="right">0.4368197</td>
<td align="right">0.9762008</td>
<td align="right">0.8686519</td>
<td align="right">0.7846779</td>
<td align="right">0.5038681</td>
<td align="right">0.9582866</td>
</tr>
<tr class="even">
<td align="left">hiv2_c1</td>
<td align="right">0.6494445</td>
<td align="right">0.4144831</td>
<td align="right">0.4368197</td>
<td align="right">1.0000000</td>
<td align="right">0.4586951</td>
<td align="right">0.3353833</td>
<td align="right">0.7518257</td>
<td align="right">0.2760415</td>
<td align="right">0.3718645</td>
</tr>
<tr class="odd">
<td align="left">hiv2_c2</td>
<td align="right">0.8803939</td>
<td align="right">0.9261440</td>
<td align="right">0.9762008</td>
<td align="right">0.4586951</td>
<td align="right">1.0000000</td>
<td align="right">0.8867906</td>
<td align="right">0.7926212</td>
<td align="right">0.5225718</td>
<td align="right">0.9653832</td>
</tr>
<tr class="even">
<td align="left">hiv2_c3</td>
<td align="right">0.7504363</td>
<td align="right">0.9100636</td>
<td align="right">0.8686519</td>
<td align="right">0.3353833</td>
<td align="right">0.8867906</td>
<td align="right">1.0000000</td>
<td align="right">0.6278904</td>
<td align="right">0.6087612</td>
<td align="right">0.9186463</td>
</tr>
<tr class="odd">
<td align="left">hiv3_c1</td>
<td align="right">0.8767961</td>
<td align="right">0.6789498</td>
<td align="right">0.7846779</td>
<td align="right">0.7518257</td>
<td align="right">0.7926212</td>
<td align="right">0.6278904</td>
<td align="right">1.0000000</td>
<td align="right">0.3856985</td>
<td align="right">0.7247967</td>
</tr>
<tr class="even">
<td align="left">hiv3_c2</td>
<td align="right">0.5096081</td>
<td align="right">0.5949834</td>
<td align="right">0.5038681</td>
<td align="right">0.2760415</td>
<td align="right">0.5225718</td>
<td align="right">0.6087612</td>
<td align="right">0.3856985</td>
<td align="right">1.0000000</td>
<td align="right">0.5626530</td>
</tr>
<tr class="odd">
<td align="left">hiv3_c3</td>
<td align="right">0.8158899</td>
<td align="right">0.9306527</td>
<td align="right">0.9582866</td>
<td align="right">0.3718645</td>
<td align="right">0.9653832</td>
<td align="right">0.9186463</td>
<td align="right">0.7247967</td>
<td align="right">0.5626530</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>
cor_mtx &lt;- df %&gt;% cor()

# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor &lt;- cor.mtest(cor_mtx, conf.level = 0.95)</code></pre>
<hr />
</details>
<pre class="r"><code>GGally::ggpairs(df) + theme_bw() +
  labs(title = &quot;Scatter matrix for all conditions&quot;)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-58-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-58-1">
Past versions of unnamed-chunk-58-1.png
</button>
</p>
<div id="fig-unnamed-chunk-58-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-58-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-58-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-58-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-58-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-58-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Conclusions:</strong></p>
<ul>
<li>Cluster 3 from patient 3 seems to be very different</li>
<li>Cluster 2 from patient 3 is also only weakly correlated</li>
<li>Cluster 1 of all patients seems correlated, however, the p_values for correlation of cluster1 of patient two with all other clusters are not significant</li>
<li>Clusters 2 and 3 of patient 1 and 2 seem highly correlated</li>
</ul>
<pre class="r"><code># plot correlations
corrplot(cor_mtx,method = &quot;color&quot;, order = &quot;hclust&quot;, 
         addrect = 4, # add boxes 
         p.mat = testCor$p, 
         #sig.level = .05,
         insig = &quot;p-value&quot;, 
         tl.col = &quot;black&quot;)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-59-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-59-1">
Past versions of unnamed-chunk-59-1.png
</button>
</p>
<div id="fig-unnamed-chunk-59-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-59-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<button onclick="myFunction()">
<em>Click here for some additional correlation plots which do not aid the interpretation much.</em>
</button>
</div>
<div id="myDIV">
<div id="correlation-between-clusters-based-on-all-tfs-with-differential-activity-between-clusters" class="section level3">
<h3>Correlation between clusters based on all TFs with differential activity between clusters</h3>
<pre class="r"><code>intersect_total &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_intersect_TFs_different_across_patients_3_clusters&quot;))[[&quot;x&quot;]]</code></pre>
<pre class="r"><code># correlation matrix
cor_mtx &lt;- auc_df %&gt;%
  select(ht_cluster, intersect_total) %&gt;% 
  group_by(ht_cluster) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  #column_to_rownames(&quot;ht_cluster&quot;) %&gt;% 
  #add_rownames() %&gt;% 
  gather(TF, value, -ht_cluster) %&gt;% 
  spread(ht_cluster, value) %&gt;% 
  column_to_rownames(&quot;TF&quot;) %&gt;% cor()

# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor &lt;- cor.mtest(cor_mtx, conf.level = 0.95)</code></pre>
<pre class="r"><code>corrplot(cor_mtx,method = &quot;color&quot;, order = &quot;hclust&quot;, 
         addrect = 4, # add boxes 
         p.mat = testCor$p, 
         #sig.level = .05,
         insig = &quot;p-value&quot;, 
         tl.col = &quot;black&quot;)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-62-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-62-1">
Past versions of unnamed-chunk-62-1.png
</button>
</p>
<div id="fig-unnamed-chunk-62-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-62-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="tf-correlation-across-clusters" class="section level3">
<h3>TF correlation across clusters</h3>
<p>Eleven TFs which differentiate the clusters seem to correlate between the clusters.</p>
<p>For these eleven TFs with different activity between the clusters the activity correlates between across all three patients. This means, that the three TFs YY1, REST, MXI1 are positively correlated, but negatively correlated with the rest of the TFs.</p>
<p>The fact that these TF activities correlate across all patients indicates that there might be a biological meaning behind the pattern of activity.</p>
<details>
*
<summary>
Correlation computation
</summary>
<ul>
<li></li>
</ul>
<p>Dataframe used as input for the correlation:</p>
<pre class="r"><code>intersect_atac &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_intersect_TFs_different_across_patients_3_clusters&quot;))[[&quot;x&quot;]]</code></pre>
<pre class="r"><code>cor_mtx &lt;- auc_df %&gt;%
  select(ht_cluster, intersect_atac) %&gt;% 
  group_by(ht_cluster) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  column_to_rownames(&quot;ht_cluster&quot;) %&gt;% cor()</code></pre>
<pre class="r"><code># correlation matrix
df &lt;- auc_df %&gt;%
  select(ht_cluster,intersect_atac) %&gt;% 
  group_by(ht_cluster) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  column_to_rownames(&quot;ht_cluster&quot;) 

df %&gt;% head %&gt;% kable()</code></pre>
<table>
<colgroup>
<col width="6%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">NFIL3</th>
<th align="right">MXI1</th>
<th align="right">MEF2A</th>
<th align="right">REST</th>
<th align="right">KLF4</th>
<th align="right">YY1</th>
<th align="right">ETS1</th>
<th align="right">JUNB</th>
<th align="right">CEBPB</th>
<th align="right">ETV6</th>
<th align="right">MEF2C</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">hiv1_c1</td>
<td align="right">0.0625178</td>
<td align="right">0.0482667</td>
<td align="right">0.1120995</td>
<td align="right">0.0485372</td>
<td align="right">0.0631640</td>
<td align="right">0.0483373</td>
<td align="right">0.0390579</td>
<td align="right">0.0580891</td>
<td align="right">0.0715644</td>
<td align="right">0.0858207</td>
<td align="right">0.0733255</td>
</tr>
<tr class="even">
<td align="left">hiv1_c2</td>
<td align="right">0.0801609</td>
<td align="right">0.0506630</td>
<td align="right">0.1879437</td>
<td align="right">0.0178021</td>
<td align="right">0.0813924</td>
<td align="right">0.0381528</td>
<td align="right">0.0136263</td>
<td align="right">0.0722964</td>
<td align="right">0.1003168</td>
<td align="right">0.1155135</td>
<td align="right">0.1158037</td>
</tr>
<tr class="odd">
<td align="left">hiv1_c3</td>
<td align="right">0.0843983</td>
<td align="right">0.1348824</td>
<td align="right">0.1964079</td>
<td align="right">0.0130281</td>
<td align="right">0.0838488</td>
<td align="right">0.0318434</td>
<td align="right">0.0102707</td>
<td align="right">0.0711062</td>
<td align="right">0.1016846</td>
<td align="right">0.1151926</td>
<td align="right">0.1091890</td>
</tr>
<tr class="even">
<td align="left">hiv2_c1</td>
<td align="right">0.0449593</td>
<td align="right">0.0120278</td>
<td align="right">0.0831293</td>
<td align="right">0.0333043</td>
<td align="right">0.0446230</td>
<td align="right">0.0525425</td>
<td align="right">0.0460040</td>
<td align="right">0.0407193</td>
<td align="right">0.0487018</td>
<td align="right">0.0502401</td>
<td align="right">0.0458824</td>
</tr>
<tr class="odd">
<td align="left">hiv2_c2</td>
<td align="right">0.0864439</td>
<td align="right">0.1463214</td>
<td align="right">0.1827775</td>
<td align="right">0.0172008</td>
<td align="right">0.0856196</td>
<td align="right">0.0301513</td>
<td align="right">0.0111471</td>
<td align="right">0.0701432</td>
<td align="right">0.1008118</td>
<td align="right">0.1232207</td>
<td align="right">0.1031197</td>
</tr>
<tr class="even">
<td align="left">hiv2_c3</td>
<td align="right">0.0824798</td>
<td align="right">0.1106875</td>
<td align="right">0.1651352</td>
<td align="right">0.0285193</td>
<td align="right">0.0766626</td>
<td align="right">0.0426642</td>
<td align="right">0.0196091</td>
<td align="right">0.0668367</td>
<td align="right">0.0901176</td>
<td align="right">0.1102263</td>
<td align="right">0.0992255</td>
</tr>
</tbody>
</table>
<p>Correlation matrix for TFs:</p>
<pre class="r"><code>cor_mtx &lt;- df %&gt;% cor()
cor_mtx %&gt;% kable</code></pre>
<table style="width:100%;">
<colgroup>
<col width="4%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">NFIL3</th>
<th align="right">MXI1</th>
<th align="right">MEF2A</th>
<th align="right">REST</th>
<th align="right">KLF4</th>
<th align="right">YY1</th>
<th align="right">ETS1</th>
<th align="right">JUNB</th>
<th align="right">CEBPB</th>
<th align="right">ETV6</th>
<th align="right">MEF2C</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NFIL3</td>
<td align="right">1.0000000</td>
<td align="right">0.8328285</td>
<td align="right">0.9762247</td>
<td align="right">-0.6908219</td>
<td align="right">0.9665777</td>
<td align="right">-0.6257135</td>
<td align="right">-0.9555881</td>
<td align="right">0.9696037</td>
<td align="right">0.9752140</td>
<td align="right">0.9940104</td>
<td align="right">0.9493500</td>
</tr>
<tr class="even">
<td align="left">MXI1</td>
<td align="right">0.8328285</td>
<td align="right">1.0000000</td>
<td align="right">0.7697811</td>
<td align="right">-0.6243427</td>
<td align="right">0.7223375</td>
<td align="right">-0.6270422</td>
<td align="right">-0.7586176</td>
<td align="right">0.7011609</td>
<td align="right">0.7433272</td>
<td align="right">0.8021050</td>
<td align="right">0.6739776</td>
</tr>
<tr class="odd">
<td align="left">MEF2A</td>
<td align="right">0.9762247</td>
<td align="right">0.7697811</td>
<td align="right">1.0000000</td>
<td align="right">-0.7943587</td>
<td align="right">0.9720426</td>
<td align="right">-0.6596786</td>
<td align="right">-0.9870940</td>
<td align="right">0.9711991</td>
<td align="right">0.9852634</td>
<td align="right">0.9735525</td>
<td align="right">0.9772616</td>
</tr>
<tr class="even">
<td align="left">REST</td>
<td align="right">-0.6908219</td>
<td align="right">-0.6243427</td>
<td align="right">-0.7943587</td>
<td align="right">1.0000000</td>
<td align="right">-0.6996961</td>
<td align="right">0.7703564</td>
<td align="right">0.8378606</td>
<td align="right">-0.6470201</td>
<td align="right">-0.7056376</td>
<td align="right">-0.6656230</td>
<td align="right">-0.7328728</td>
</tr>
<tr class="odd">
<td align="left">KLF4</td>
<td align="right">0.9665777</td>
<td align="right">0.7223375</td>
<td align="right">0.9720426</td>
<td align="right">-0.6996961</td>
<td align="right">1.0000000</td>
<td align="right">-0.6458621</td>
<td align="right">-0.9741139</td>
<td align="right">0.9890996</td>
<td align="right">0.9882537</td>
<td align="right">0.9750536</td>
<td align="right">0.9770080</td>
</tr>
<tr class="even">
<td align="left">YY1</td>
<td align="right">-0.6257135</td>
<td align="right">-0.6270422</td>
<td align="right">-0.6596786</td>
<td align="right">0.7703564</td>
<td align="right">-0.6458621</td>
<td align="right">1.0000000</td>
<td align="right">0.7237429</td>
<td align="right">-0.5844677</td>
<td align="right">-0.5846666</td>
<td align="right">-0.6088358</td>
<td align="right">-0.6502978</td>
</tr>
<tr class="odd">
<td align="left">ETS1</td>
<td align="right">-0.9555881</td>
<td align="right">-0.7586176</td>
<td align="right">-0.9870940</td>
<td align="right">0.8378606</td>
<td align="right">-0.9741139</td>
<td align="right">0.7237429</td>
<td align="right">1.0000000</td>
<td align="right">-0.9532630</td>
<td align="right">-0.9703806</td>
<td align="right">-0.9520631</td>
<td align="right">-0.9727509</td>
</tr>
<tr class="even">
<td align="left">JUNB</td>
<td align="right">0.9696037</td>
<td align="right">0.7011609</td>
<td align="right">0.9711991</td>
<td align="right">-0.6470201</td>
<td align="right">0.9890996</td>
<td align="right">-0.5844677</td>
<td align="right">-0.9532630</td>
<td align="right">1.0000000</td>
<td align="right">0.9915545</td>
<td align="right">0.9825523</td>
<td align="right">0.9839199</td>
</tr>
<tr class="odd">
<td align="left">CEBPB</td>
<td align="right">0.9752140</td>
<td align="right">0.7433272</td>
<td align="right">0.9852634</td>
<td align="right">-0.7056376</td>
<td align="right">0.9882537</td>
<td align="right">-0.5846666</td>
<td align="right">-0.9703806</td>
<td align="right">0.9915545</td>
<td align="right">1.0000000</td>
<td align="right">0.9852773</td>
<td align="right">0.9783193</td>
</tr>
<tr class="even">
<td align="left">ETV6</td>
<td align="right">0.9940104</td>
<td align="right">0.8021050</td>
<td align="right">0.9735525</td>
<td align="right">-0.6656230</td>
<td align="right">0.9750536</td>
<td align="right">-0.6088358</td>
<td align="right">-0.9520631</td>
<td align="right">0.9825523</td>
<td align="right">0.9852773</td>
<td align="right">1.0000000</td>
<td align="right">0.9592715</td>
</tr>
<tr class="odd">
<td align="left">MEF2C</td>
<td align="right">0.9493500</td>
<td align="right">0.6739776</td>
<td align="right">0.9772616</td>
<td align="right">-0.7328728</td>
<td align="right">0.9770080</td>
<td align="right">-0.6502978</td>
<td align="right">-0.9727509</td>
<td align="right">0.9839199</td>
<td align="right">0.9783193</td>
<td align="right">0.9592715</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
<pre class="r"><code>
# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor &lt;- cor.mtest(cor_mtx, conf.level = 0.95)</code></pre>
<hr />
</details>
<pre class="r"><code>GGally::ggpairs(df) + theme_bw() +
    labs(title = &quot;Scatter matrix for all TFs with different activity between clusters&quot;)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-67-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-67-1">
Past versions of unnamed-chunk-67-1.png
</button>
</p>
<div id="fig-unnamed-chunk-67-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-67-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># plot correlations
# p.mat = matrix of p_values
p1 &lt;- corrplot(cor_mtx,method = &quot;color&quot;, order = &quot;hclust&quot;, 
         addrect = 2, # add boxes 
         p.mat = testCor$p, 
         insig = &quot;p-value&quot;,
         tl.col = &quot;black&quot;)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-68-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-68-1">
Past versions of unnamed-chunk-68-1.png
</button>
</p>
<div id="fig-unnamed-chunk-68-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-68-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>         #insig = &quot;p-value&quot;)
         #title = &quot;Correlations between 9 clusters of 3 patients&quot;)</code></pre>
</div>
</div>
</div>
<div id="gene-expression-analysis" class="section level1">
<h1>Gene expression analysis</h1>
<p>Using the information of the three clusters from above we will pool the two clusters which are significantly different from the third cluster in all patients. Consequently, in the following analysis we will compare Cluster1 to Cluster2 (pool of two similar clusters). The gene expression might give a hint about the identity of these two Myeloid2 clusters.</p>
<pre class="r"><code># 3 Clusters for each of the 3 HIV+ patients

# HIV1_CSF
hiv1_c1 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv1_c2 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv1_c3 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]

# HIV2_CSF
hiv2_c1 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster1.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv2_c2 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster2.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv2_c3 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster3.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]

# HIV3_CSF
hiv3_c1 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster1.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv3_c2 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster2.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]
hiv3_c3 &lt;- as.vector(read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster3.tsv&quot;, sep = &quot;\t&quot;))[[&quot;x&quot;]]

cluster_anno &lt;- c(hiv1_c1, hiv1_c2, hiv1_c3, hiv2_c1, hiv2_c2, hiv2_c3, hiv3_c1, hiv3_c2, hiv3_c3)</code></pre>
<div id="tf-gene-expression-of-the-11-tfs-with-different-activity-between-clusters" class="section level3">
<h3>TF gene expression of the 11 TFs with different activity between clusters</h3>
<pre class="r"><code>hivmy2 &lt;- subset(hiv7, cell_type == &quot;Myeloid2&quot; &amp; orig.ident %in% c(&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;))

# add cluster identity to metadata
ht_cluster &lt;- (
  hivmy2@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  mutate(ht_cluster = ifelse(cell %in% c(hiv1_c1, hiv2_c1, hiv3_c3),
                             &quot;c1&quot;, 
                             &quot;c2&quot;)) %&gt;% 
  column_to_rownames(&quot;cell&quot;))[&quot;ht_cluster&quot;]

hivmy2 &lt;- AddMetaData(hivmy2, metadata = ht_cluster, col.name = &quot;ht_cluster&quot;)
#saveRDS(hivmy2,&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv_myeloid2_subclusters&quot;)</code></pre>
</div>
<div id="quality-of-the-myeloid-2-clusters" class="section level2">
<h2>Quality of the Myeloid 2 clusters</h2>
<p>There is an evident difference in library size and number of features between the two clusters. This could mean that there is a technical difference between the two clusters. Conversely, it could also mean that there is this a biological difference. For example, if cells of cluster1 represent a latently infected group of Myeloid2 cells it could be that gene expression is reduced overall due to the infection compared to healthy uninfected cells from cluster2.</p>
<p>One way to investigate this further would be to have a look at the housekeeping genes in both clusters. If housekeeping genes were more highly expressed in one cluster than in the other this would indicate that the difference is not biological, but a technical artefact.</p>
<p>Since there is one cluster which has lower number of counts and features to start with, further qualitative analysis of gene expression might be skewed as well. For this reason the following analysis on the difference between these two clusters will be mainly qualitative. Due to the low number of cells it is not possible to get significant results for the differential gene expression.</p>
<pre class="r"><code>p1 &lt;- hivmy2@meta.data %&gt;% 
  ggplot() +
  geom_boxplot(aes(x = ht_cluster, y = nFeature_RNA, fill = ht_cluster))

p2 &lt;- hivmy2@meta.data %&gt;% 
  ggplot() +
  geom_violin(aes(x = ht_cluster, y = nFeature_RNA, fill = ht_cluster)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1))

fig &lt;- ggarrange(p1, p2, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob(&quot;Quality of Clsuters&quot;))</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-71-1">
Past versions of unnamed-chunk-71-1.png
</button>
</p>
<div id="fig-unnamed-chunk-71-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-71-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="housekeeping-genes" class="section level2">
<h2>Housekeeping genes</h2>
<p>A list of 3804 houseekping genes were used from <span class="citation">[@Eisenberg]</span>.</p>
<p>A quantitative analysis might be difficult, because due to the small sample size and large feature number, we get very high p_values for all genes in differential gene expression analysis. Visualization of the housekeeping gene expression, unfortunately does not help either with the interpretation.</p>
<details>
*
<summary>
Boxplots of housekeeping gene expression
</summary>
<ul>
<li></li>
</ul>
<pre class="r"><code>hk_genes &lt;- (read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/HK_genes.txt&quot;))[[&quot;V1&quot;]]</code></pre>
<pre class="r"><code># filter only genes present in the scRNAseq count matrix
hk_genes_filt &lt;- hk_genes[hk_genes %in% rownames(hivmy2@assays$RNA@counts)]
length(hk_genes)
[1] 3804
length(hk_genes_filt)
[1] 3711

# Find genes from the housekeeping genes that are differentially expressed 
# between the clusters
hk_diff &lt;- FindAllMarkers(hivmy2, features = hk_genes_filt)
Idents(hivmy2) &lt;- &quot;ht_cluster&quot;
hk_diff &lt;- FindMarkers(hivmy2, features = hk_genes_filt, ident.1 = &quot;c1&quot;, ident.2 = &quot;c2&quot;)
hk_diff %&gt;% head
             p_val  avg_log2FC pct.1 pct.2 p_val_adj
DPP7   0.001063786  -0.9991177 0.000 0.259         1
RPL27  0.001354627 -47.9667965 1.000 0.991         1
RPL10A 0.001632447  -9.5926981 0.941 0.911         1
RPL6   0.001658084  -1.7550766 0.882 0.812         1
RPL30  0.002862495 -38.6641080 0.971 0.946         1
DDX21  0.003060688  -1.0742719 0.088 0.330         1</code></pre>
<pre class="r"><code>test &lt;- hk_genes_filt[1:30]
VlnPlot(hivmy2, features = test, group.by = &quot;ht_cluster&quot;, ncol = 4)
Warning in SingleExIPlot(type = type, data = data[, x, drop = FALSE], idents =
idents, : All cells have the same value of ABHD8.</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-74-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-74-1">
Past versions of unnamed-chunk-74-1.png
</button>
</p>
<div id="fig-unnamed-chunk-74-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-74-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-74-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-74-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-74-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/183077b696fa2df6e887d589e7a30c7193ad5e04/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-74-1.png" target="_blank">183077b</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/dde10972b37ffd997474921968223514c8cb1b1f/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-74-1.png" target="_blank">dde1097</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</details>
<div id="tf-from-the-farhadian-paper" class="section level3">
<h3>TF from the <span class="citation">[@Farhadian]</span> paper</h3>
<p>These TFs were used as marker genes for the microlgia like Myeloid2 cell cluster. Is there a group of cells which has a higher expression of these markers? No clear difference in gene expression can be seen.</p>
<details>
*
<summary>
Boxplots of microglia marker genes
</summary>
<ul>
<li></li>
</ul>
<pre class="r"><code>#import the 60 genes that were upregulated in Myeloid-2 in the paper compared to the four other myeloid subsets
paper &lt;- read.table(file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/Myeloid-2_genes_paper.csv&quot;,
                    sep = &quot;\t&quot;, header = FALSE, skipNul = TRUE)#, n_max = 60)
paper[1,] &lt;- &quot;APOC1&quot;
colnames(paper) &lt;- &quot;&quot;
paper &lt;- as.vector(paper[,1])


VlnPlot(hivmy2, features = paper, group.by = &quot;ht_cluster&quot;)
Warning in FetchData(object = object, vars = features, slot = slot): The
following requested variables were not found: SELENOP</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-75-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-75-1">
Past versions of unnamed-chunk-75-1.png
</button>
</p>
<div id="fig-unnamed-chunk-75-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-75-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-75-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</details>
</div>
</div>
</div>
<div id="differential-gene-expression-analyisis" class="section level1">
<h1>Differential gene expression analyisis</h1>
<details>
*
<summary>
Differential gene expression analysis
</summary>
<ul>
<li></li>
</ul>
<pre class="r"><code>hivmy2 &lt;- NormalizeData(hivmy2, verbose = FALSE)
hivmy2 &lt;- ScaleData(hivmy2, verbose = FALSE)
hivmy2 &lt;- FindVariableFeatures(hivmy2, verbose = FALSE)
Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
parametric, : pseudoinverse used at -2.1857
Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
parametric, : neighborhood radius 0.49845
Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
parametric, : reciprocal condition number 7.6527e-16
Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
parametric, : There are other near singularities as well. 0.090619
hivmy2 &lt;- RunPCA(hivmy2, features = VariableFeatures(hivmy2), verbose = FALSE)
ElbowPlot(hivmy2)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-76-1">
Past versions of unnamed-chunk-76-1.png
</button>
</p>
<div id="fig-unnamed-chunk-76-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/2a95e927f953c093f2e48cb2b6551d152f735267/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-76-1.png" target="_blank">2a95e92</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>hivmy2 &lt;- RunTSNE(hivmy2, dims= 1:10, verbose = FALSE)
Idents(hivmy2) &lt;- &quot;ht_cluster&quot;
de_genes &lt;- FindAllMarkers(hivmy2, verbose = FALSE)
top10 &lt;- de_genes %&gt;% 
  group_by(cluster) %&gt;%
  top_n(n = 5, wt = avg_log2FC) 
top10 %&gt;% 
  kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="right">p_val</th>
<th align="right">avg_log2FC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
<th align="left">cluster</th>
<th align="left">gene</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">0.0002643</td>
<td align="right">1.4371177</td>
<td align="right">0.206</td>
<td align="right">0.027</td>
<td align="right">1</td>
<td align="left">c1</td>
<td align="left">EPHB6</td>
</tr>
<tr class="even">
<td align="right">0.0011962</td>
<td align="right">1.2486709</td>
<td align="right">0.382</td>
<td align="right">0.143</td>
<td align="right">1</td>
<td align="left">c1</td>
<td align="left">SNRPF</td>
</tr>
<tr class="odd">
<td align="right">0.0015781</td>
<td align="right">1.0213995</td>
<td align="right">0.176</td>
<td align="right">0.027</td>
<td align="right">1</td>
<td align="left">c1</td>
<td align="left">RP11-386I14.4</td>
</tr>
<tr class="even">
<td align="right">0.0022935</td>
<td align="right">0.9050976</td>
<td align="right">0.118</td>
<td align="right">0.009</td>
<td align="right">1</td>
<td align="left">c1</td>
<td align="left">SACS</td>
</tr>
<tr class="odd">
<td align="right">0.0045394</td>
<td align="right">1.0999273</td>
<td align="right">0.294</td>
<td align="right">0.107</td>
<td align="right">1</td>
<td align="left">c1</td>
<td align="left">TBC1D5</td>
</tr>
<tr class="even">
<td align="right">0.0005081</td>
<td align="right">1.3333431</td>
<td align="right">0.286</td>
<td align="right">0.000</td>
<td align="right">1</td>
<td align="left">c2</td>
<td align="left">JAK2</td>
</tr>
<tr class="odd">
<td align="right">0.0006461</td>
<td align="right">1.3600293</td>
<td align="right">0.438</td>
<td align="right">0.118</td>
<td align="right">1</td>
<td align="left">c2</td>
<td align="left">IGSF6</td>
</tr>
<tr class="even">
<td align="right">0.0012653</td>
<td align="right">1.1633609</td>
<td align="right">0.589</td>
<td align="right">0.324</td>
<td align="right">1</td>
<td align="left">c2</td>
<td align="left">CTSH</td>
</tr>
<tr class="odd">
<td align="right">0.0072128</td>
<td align="right">1.1970288</td>
<td align="right">0.429</td>
<td align="right">0.176</td>
<td align="right">1</td>
<td align="left">c2</td>
<td align="left">CD55</td>
</tr>
<tr class="even">
<td align="right">0.0096532</td>
<td align="right">1.1831845</td>
<td align="right">0.286</td>
<td align="right">0.088</td>
<td align="right">1</td>
<td align="left">c2</td>
<td align="left">NAMPTL</td>
</tr>
</tbody>
</table>
</details>
<div id="tf-from-the-gosselin-paper" class="section level3">
<h3>TF from the <span class="citation">[@Gosselin]</span> paper</h3>
<details>
*
<summary>
Microglia TF gene expression
</summary>
<ul>
<li></li>
</ul>
<pre class="r"><code>Idents(hiv4) &lt;- &quot;cell_type&quot;
# Find Markers that distinguish My2 from the other Myeloid clusters
my2_markers &lt;- FindMarkers(hiv4, ident.1  = &quot;Myeloid2&quot;,
            ident.2  = c(&quot;Myeloid1&quot;, &quot;Myeloid3&quot;, &quot;Myeloid4&quot;, &quot;Myeloid5&quot;))
my2_markers &lt;- my2_markers %&gt;% rownames_to_column(var = &quot;gene&quot;)</code></pre>
<pre class="r"><code>microglia_genes &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/Microglia_gene_signatur_found_in_HIV1_HIV2_samples&quot;)
m1 &lt;- my2_markers %&gt;% filter(gene %in% microglia_genes [[&quot;gene_name&quot;]] &amp; avg_log2FC &gt;0)
# remove the genes that were already used by the Farhadian paper
m1 &lt;- m1[[&quot;gene&quot;]][m1[[&quot;gene&quot;]] %in% paper == FALSE]

VlnPlot(hivmy2, features = m1, group.by = &quot;ht_cluster&quot;, ncol = 5)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-78-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-78-1">
Past versions of unnamed-chunk-78-1.png
</button>
</p>
<div id="fig-unnamed-chunk-78-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-78-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-78-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="which-of-the-11-tfs-are-the-same-as-in-the-gosselin-paper" class="section level1">
<h1>Which of the 11 TFs are the same as in the <span class="citation">[@Gosselin]</span> paper?</h1>
<pre class="r"><code>tfs_microglia &lt;- (read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/TFs_microglia_Glass_paper.txt&quot;))[[&quot;V1&quot;]]

gosselin &lt;- tfs_microglia[tfs_microglia %in% overlap_atac$name]

VlnPlot(hivmy2, features = gosselin, group.by = &quot;ht_cluster&quot;, ncol = 4)</code></pre>
<p><img src="figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-79-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-79-1">
Past versions of unnamed-chunk-79-1.png
</button>
</p>
<div id="fig-unnamed-chunk-79-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/7f7db3c2cf840c14d6dbc92fb9fb5bc7d659a826/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-79-1.png" target="_blank">7f7db3c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/eb137dda9188951b1859c2ac57e84de1ab238fd9/docs/figure/TF_activity_analysis_of_myeloid2_cluster.Rmd/unnamed-chunk-79-1.png" target="_blank">eb137dd</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
</tr>
</tbody>
</table>
</div>
</div>
</details>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] rstatix_0.6.0        future_1.23.0        corrplot_0.92       
 [4] ggVennDiagram_1.2.0  knitr_1.30           gridExtra_2.3       
 [7] ggpubr_0.4.0         circlize_0.4.13      ComplexHeatmap_2.9.4
[10] ggcorrplot_0.1.3     pheatmap_1.0.12      RColorBrewer_1.1-2  
[13] ggrepel_0.9.1        data.table_1.14.2    Matrix_1.3-4        
[16] edgeR_3.32.0         limma_3.46.0         SeuratObject_4.0.3  
[19] Seurat_4.0.5         forcats_0.5.0        stringr_1.4.0       
[22] dplyr_1.0.7          purrr_0.3.4          readr_1.4.0         
[25] tidyr_1.1.2          tibble_3.1.6         ggplot2_3.3.5       
[28] tidyverse_1.3.0      workflowr_1.6.2     

loaded via a namespace (and not attached):
  [1] utf8_1.2.2            reticulate_1.22       tidyselect_1.1.1     
  [4] htmlwidgets_1.5.4     Rtsne_0.15            munsell_0.5.0        
  [7] units_0.6-7           codetools_0.2-18      ica_1.0-2            
 [10] miniUI_0.1.1.1        withr_2.4.3           colorspace_2.0-2     
 [13] highr_0.8             rstudioapi_0.13       stats4_4.0.2         
 [16] ROCR_1.0-11           ggsignif_0.6.0        tensor_1.5           
 [19] listenv_0.8.0         labeling_0.4.2        git2r_0.27.1         
 [22] polyclip_1.10-0       farver_2.1.0          rprojroot_2.0.2      
 [25] parallelly_1.29.0     vctrs_0.3.8           generics_0.1.1       
 [28] xfun_0.29             R6_2.5.1              doParallel_1.0.16    
 [31] clue_0.3-60           locfit_1.5-9.4        RVenn_1.1.0          
 [34] reshape_0.8.8         spatstat.utils_2.2-0  assertthat_0.2.1     
 [37] promises_1.2.0.1      scales_1.1.1          gtable_0.3.0         
 [40] globals_0.14.0        goftest_1.2-3         rlang_0.4.12         
 [43] GlobalOptions_0.1.2   splines_4.0.2         lazyeval_0.2.2       
 [46] spatstat.geom_2.3-0   broom_0.7.2           yaml_2.2.1           
 [49] reshape2_1.4.4        abind_1.4-5           modelr_0.1.8         
 [52] backports_1.2.0       httpuv_1.6.3          tools_4.0.2          
 [55] ellipsis_0.3.2        spatstat.core_2.3-1   BiocGenerics_0.36.1  
 [58] ggridges_0.5.2        Rcpp_1.0.7            plyr_1.8.6           
 [61] classInt_0.4-3        rpart_4.1-15          deldir_1.0-6         
 [64] pbapply_1.5-0         GetoptLong_1.0.5      cowplot_1.1.1        
 [67] S4Vectors_0.28.1      zoo_1.8-9             haven_2.3.1          
 [70] cluster_2.1.0         fs_1.5.1              magrittr_2.0.1       
 [73] magick_2.5.2          scattermore_0.7       openxlsx_4.2.3       
 [76] lmtest_0.9-39         reprex_0.3.0          RANN_2.6.1           
 [79] whisker_0.4           fitdistrplus_1.1-6    matrixStats_0.61.0   
 [82] hms_1.1.1             patchwork_1.1.1       mime_0.12            
 [85] evaluate_0.14         xtable_1.8-4          rio_0.5.16           
 [88] readxl_1.3.1          IRanges_2.24.1        shape_1.4.6          
 [91] compiler_4.0.2        KernSmooth_2.23-18    crayon_1.4.2         
 [94] htmltools_0.5.2       mgcv_1.8-33           later_1.3.0          
 [97] lubridate_1.7.9.2     DBI_1.1.1             dbplyr_2.0.0         
[100] MASS_7.3-53           sf_0.9-6              car_3.0-10           
[103] cli_3.1.0             parallel_4.0.2        igraph_1.2.9         
[106] pkgconfig_2.0.3       foreign_0.8-80        plotly_4.9.2.1       
[109] spatstat.sparse_2.0-0 xml2_1.3.2            foreach_1.5.1        
[112] rvest_0.3.6           digest_0.6.29         sctransform_0.3.2    
[115] RcppAnnoy_0.0.19      spatstat.data_2.1-0   rmarkdown_2.6        
[118] cellranger_1.1.0      leiden_0.3.5          uwot_0.1.10          
[121] curl_4.3.2            shiny_1.7.1           rjson_0.2.20         
[124] lifecycle_1.0.1       nlme_3.1-150          jsonlite_1.7.2       
[127] carData_3.0-4         viridisLite_0.4.0     fansi_0.5.0          
[130] pillar_1.6.4          GGally_2.0.0          lattice_0.20-41      
[133] fastmap_1.1.0         httr_1.4.2            survival_3.2-7       
[136] glue_1.5.1            zip_2.1.1             png_0.1-7            
[139] iterators_1.0.13      class_7.3-17          stringi_1.7.6        
[142] e1071_1.7-4           irlba_2.3.3           future.apply_1.8.1   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
