<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="kmikulik" />


<title>Triculture Dataset GRN with Scenic</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Introduction.html">Introduction</a>
</li>
<li>
  <a href="HIV_CSF_dataset_analysis.html">CSF</a>
</li>
<li>
  <a href="TF_activity_analysis_of_myeloid2_cluster.html">TF activity Myeloid2</a>
</li>
<li>
  <a href="Independent_cell_type_identification_Farhadian_dataset.html">Myeloid2 are a microglia-like</a>
</li>
<li>
  <a href="Triculture_datset_analysis_and_integration_Farhadian_dataset.html">Integration CSF &amp; triculture</a>
</li>
<li>
  <a href="Triculture_dataset_GRN_with_Scenic.html">TF activity iMg</a>
</li>
<li>
  <a href="Discussion.html">Discussion</a>
</li>
<li>
  <a href="Methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Katharina782/report">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Triculture Dataset GRN with Scenic</h1>
<h4 class="author">kmikulik</h4>
<h4 class="date">2 12 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-18
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttreea5b39dcf024b9c39f73044e12a1c5e581e6ec861targetblanka5b39dca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Katharina782/report/tree/a5b39dcf024b9c39f73044e12a1c5e581e6ec861" target="_blank">a5b39dc</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttreea5b39dcf024b9c39f73044e12a1c5e581e6ec861targetblanka5b39dca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Katharina782/report/tree/a5b39dcf024b9c39f73044e12a1c5e581e6ec861" target="_blank">a5b39dc</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/HIV_CSF_dataset_analysis_cache/
    Ignored:    analysis/Independent_cell_type_identification_Farhadian_dataset_cache/
    Ignored:    analysis/TF_activity_analysis_of_myeloid2_cluster_cache/
    Ignored:    analysis/Triculture_dataset_analysis_and_integration_Farhadian_dataset_cache/

Unstaged changes:
    Modified:   analysis/Independent_cell_type_identification_Farhadian_dataset.Rmd
    Modified:   analysis/Introduction.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/html_test.Rmd
    Modified:   analysis/references.bib

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Triculture_dataset_GRN_with_Scenic.Rmd</code>) and HTML (<code>docs/Triculture_dataset_GRN_with_Scenic.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/9ee308625fb863ad709edf2660babe4de93ced91/docs/Triculture_dataset_GRN_with_Scenic.html" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/e87fa963caeed78821ac7f482359072ce58b8529/analysis/Triculture_dataset_GRN_with_Scenic.Rmd" target="_blank">e87fa96</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
new files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, collapse = TRUE, message = FALSE,
                      cache = TRUE, cache.lazy = FALSE, autodep = TRUE)
set.seed(42)</code></pre>
<pre class="r"><code>library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggplot2)
library(dplyr)
library(ggrepel)
#library(harmony)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(corrplot)
library(rstatix)
library(knitr)
library(metap)
library(multtest)
library(readxl)</code></pre>
<div id="grn-inference-with-scenic" class="section level2">
<h2>GRN inference with SCENIC</h2>
<ul>
<li>Prepare expression matrix for SCENIC</li>
</ul>
<p>Because the snakemake pipeline pySCENIC_pipeline (<a href="https://github.com/hdsu-bioquant/pySCENIC_pipeline/blob/master/input/gene_exp_mtx_example.tsv" class="uri">https://github.com/hdsu-bioquant/pySCENIC_pipeline/blob/master/input/gene_exp_mtx_example.tsv</a>) requires an input matrix which has genes as columns and cells as rows I will transpose the matrix.</p>
<p>When referring to differentially bound TFs in primary microglia in the following I am referring to differential binding scores between infection conditions obtained from bulk ATACseq data from primary microglia cell cultures using TOBIAS.</p>
<p>How many of the most Highly Variable Genes (HVG) should I use? Are 3000 enough? No, because then I only get an overlap with differentially bound TFs in primariy microglia of 8 TFs. When using the 10,000 most highly variable genes, SCENIC returns 232 TFs and their corresponding activity score in each cell. 25 of these TFs overlap with differentially bound TFs in primary microglia.</p>
<p>Use only the iMg cells from the triculture dataset (4433 cells in total).</p>
<pre class="r"><code># read in triculture Seurat object
triculture &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_clean_seurat_object&quot;)

# read in CSF HIV Seurat object
hiv7 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds&quot;)</code></pre>
<pre class="r"><code>
imicroglia &lt;- subset(triculture, cell_type == &quot;iMg&quot;)
# use 10 000 most highly variable genes
#scenic &lt;- SCTransform(imicroglia, variable.features.n = 10000, verbose=FALSE)
#saveRDS(scenic, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_scenic&quot;)
#scenic &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_scenic&quot;)
#scenic_hvg &lt;- subset(scenic, features = scenic@assays$SCT@var.features)
#scenic_mtx &lt;- t(scenic_hvg@assays$SCT@scale.data)
#write.table(x = scenic_mtx, file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/pySCENIC_pipeline/input/triculture_mtx.tsv&quot;, sep = &quot;\t&quot;)

#scenic_matrix &lt;- read.table(file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/pySCENIC_pipeline/input/triculture_mtx.tsv&quot;, sep = &quot;\t&quot;)</code></pre>
<ul>
<li>AuCell matrix - output from Scenic</li>
</ul>
<p>The output from SCENIC results is a matrix containing cells as rows and 232 TFs as columns. The TF activity for each cell can be used to draw conclusions about different cell states.</p>
<pre class="r"><code>auc_mtx &lt;- read.csv(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/pySCENIC_output/aucell_img_10k.csv&quot;)
auc_mtx &lt;- auc_mtx %&gt;% column_to_rownames(&quot;Cell&quot;)
auc_mtx &lt;- auc_mtx %&gt;% setnames( old = colnames(auc_mtx), new = sapply(strsplit(
  colnames(auc_mtx), &quot;[...]&quot;), `[`, 1))</code></pre>
<ul>
<li>Combine AuCell Matrix and Seurat object</li>
</ul>
<pre class="r"><code># add auc matrix to metadata of hiv4 seurat object
tri_mg &lt;- AddMetaData(imicroglia, 
                    metadata = auc_mtx, 
                    col.name = colnames(auc_mtx))</code></pre>
</div>
<div id="number-of-cells-in-each-condition" class="section level2">
<h2>Number of cells in each condition</h2>
<p>Triculture Dataset</p>
<pre class="r"><code>tri_mg@meta.data %&gt;% 
  group_by(orig.ident) %&gt;%
  summarize(n = n()) %&gt;% 
  kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">orig.ident</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Inf</td>
<td align="right">1166</td>
</tr>
<tr class="even">
<td align="left">Inf+EFZ</td>
<td align="right">1382</td>
</tr>
<tr class="odd">
<td align="left">Uninf</td>
<td align="right">1163</td>
</tr>
<tr class="even">
<td align="left">Uninf+EFZ</td>
<td align="right">722</td>
</tr>
</tbody>
</table>
</div>
<div id="atacseq-binding-scores-of-tfs-between-infection-conditions-in-primary-microglia" class="section level2">
<h2>ATACseq binding scores of TFs between infection conditions in primary microglia</h2>
<p>The primary microglia are cells from the C20 cell line which was derived from an HIV+ patient treated with ART (Antiretroviral Therapy). Since the access of ART to the brain is limited due to the blood-brain barrier, the effects of ART on these cells are probably negligible.</p>
<ul>
<li>select only 75 TFs with p-values for differential binding scores &lt; 1e-50</li>
<li>we will focus on TFs which show differential binding scores between latent infection and uninfected conditions in primary microglia</li>
<li>For 25 of these 75 TFs there is a TF activity score for the induced microglia</li>
<li>focusing on these 25 TFs we might be able to find activity patterns that correlate with the TF binding scores</li>
<li>5 TFs are more bound in latent infection</li>
<li>25 TFs are more bound in uninfected cells</li>
</ul>
<p>Is there a reason why we find more of the TFs more bound in uninfected cells in the TF activity score matrix from SCENIC?</p>
<pre class="r"><code>tfs_ana &lt;- read.table(&quot;/media/ag-cherrmann/projects/06_HIV_Microglia/data/atacseq/data-2020-11-06/tobias/TOBIAS_snakemake/footprint_mglia2_GlassTF_17-03/TFBS/bindetect_results.txt&quot;, sep = &quot;\t&quot;, header = TRUE)

top_tfs &lt;- tfs_ana %&gt;% filter(uninf_red_pvalue &lt; 1e-50)

#atac_tfs &lt;- colnames(auc_mtx)[colnames(auc_mtx) %in% top_tfs$name]

# dataframe containing only the 25 Tfs whcih overlap with TF activity matrix from SCENIC 
# adding a column with information in which infection condition the TF is more bound
atac_tfs_df &lt;- top_tfs %&gt;% filter(name %in% colnames(auc_mtx)) %&gt;% 
  mutate(condition = ifelse(uninf_red_change &gt; 0, &quot;up_uninf&quot;, &quot;up_latent&quot;))

# List of 5 TFs more bound in latent infection
up_latent &lt;- atac_tfs_df$name[atac_tfs_df$condition == &quot;up_latent&quot;]
# list of 25 TFs more bound in uninfected cells
up_uninf &lt;- atac_tfs_df$name[atac_tfs_df$condition == &quot;up_uninf&quot;]</code></pre>
</div>
<div id="gosselin-tfs" class="section level2">
<h2>Gosselin TFs</h2>
<pre class="r"><code>gosselin_tfs &lt;- as.vector(read.table( &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/TFs_microglia_Glass_paper.txt&quot;)[[&quot;V1&quot;]])</code></pre>
<div id="heatmap-for-all-tfs" class="section level3">
<h3>Heatmap for all TFs</h3>
<p>In the heatmap below you can already see a trend towards a separation of cells treated with EFZ (blue and green) from untreated cells (red and orange).</p>
<pre class="r"><code>conditions &lt;- tri_mg@meta.data %&gt;%
                    rownames_to_column(&quot;cell&quot;) %&gt;%
                    select(&quot;cell&quot;, &quot;orig.ident&quot;)%&gt;% column_to_rownames(&quot;cell&quot;)


ht &lt;- Heatmap(t(auc_mtx), 
      col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
      show_column_names = FALSE,
      row_names_side = &quot;left&quot;,
      top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                         name = &quot;condition&quot;, col =  
                                         list(condition = c(&quot;Uninf&quot; = &quot;orange&quot;, 
                                                            &quot;Inf+EFZ&quot; = &quot;green&quot;,
                                                            &quot;Inf&quot; = &quot;red&quot;,
                                                            &quot;Uninf+EFZ&quot; = &quot;blue&quot;)))
)
ht</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-9-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-9-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heatmap-of-tf-activity-scores-for-the-25-tfs-with-differential-binding-score" class="section level3">
<h3>Heatmap of TF activity scores for the 25 TFs with differential binding score</h3>
<p>We will now investigate if we can find activity patterns which resemble the differential binding score patterns found in 25 TFs.</p>
<div id="hierarchical-clustering-with-different-k" class="section level4">
<h4>Hierarchical clustering with different k</h4>
<p>Using hierarchical clustering we migth be able to identify clusters with certain patterns of TF activity. Judging from the different heatmaps, k = 4 might be the most informative, since for k = 5 there is only one TF (NHLH1) differentiating an additional cluster. The plot of within-clusters sum of squares should help to pick the best number of clusters, however as can be seen below there is no obvious kink in the curve.</p>
<p>Hierarchical clustering with k = 2 seems to split the cells into treated vs.  untreated cells. These two clusters have a very evident biological meaning. Using k = 3 or k = 4 keeps these two biologically relevant groups separate, but also creates additional clusters which seem to be a mixture of treated and untreated cells. Having a closer look at clusters from k = 4 might give som additional biological insights which is why we will have a closer look at these.</p>
<p>Having a closer look at the four clusters, it becomes evident that the two left most clusters separate treated cells from untreated cells, independent of the infection condition. There are more Inf+EFZ than Uninf+EFZ in the first cluster, however this is in line with the lower cell number of Uninf+EFZ (722 cells) compared to Inf+EFZ (1382 cells).</p>
<p>Two TFs seem to be more active in Inf+EFZ and Uninf+EFZ (compared to Inf and Uninf):</p>
<ul>
<li>RUNX3</li>
<li>JUNB</li>
</ul>
<p>Three TFs seem to be more active in Inf and Uninf (compared to Inf+EFZ and Uninf+EFZ):</p>
<ul>
<li>NFAT%</li>
<li>TCF7L2</li>
<li>CEBPB</li>
</ul>
<p>TCF4 distinguishes a third cluster.</p>
<p>Is there a biological meaning behind that? Maybe we can associate them with a particular condition?</p>
<pre class="r"><code>df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                    orig.ident == &quot;HIV1_CSF&quot;)
mtx &lt;- auc_mtx[, atac_tfs_df$name]

# Decide on a k for clustering
wss = sapply(2:15, function(k) {
  kmeans(x = t(mtx), centers = k)$tot.withinss
})

plot(2:15, wss, type = &quot;b&quot;, 
     xlab = &quot;Number of clusters k&quot;, 
     ylab = &quot;Total within-clusters sum of square&quot;)</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-10-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
ht_list &lt;- map(seq.int(2,6), function(n){
  ht &lt;- Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
        col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        column_title = paste0(&quot;k = &quot;, n),
        show_column_names = FALSE,
        row_names_side = &quot;left&quot;,
        top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                           name = &quot;condition&quot;,
                                           col = list(condition = 
                                                        c(&quot;Uninf&quot; = &quot;orange&quot;, 
                                                          &quot;Inf+EFZ&quot; = &quot;green&quot;,
                                                          &quot;Inf&quot; = &quot;red&quot;, 
                                                          &quot;Uninf+EFZ&quot; = &quot;blue&quot;))),
        column_km = n,
         )
  
})

ht_list &lt;- ht_list[[1]] + ht_list[[2]] + ht_list[[3]] + ht_list[[4]] + ht_list[[5]]
draw(ht_list, column_title = &quot;Hierarchical clustering with different k&quot;, ht_gap = unit(2, &quot;cm&quot;))</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-11-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-11-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ht &lt;- draw(Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
      col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
      column_title = &quot;hierarchical clustering with k = 4&quot;,
      show_column_names = FALSE,
      row_names_side = &quot;left&quot;,
      top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                         name = &quot;condition&quot;,
                                         col = list(condition = 
                                                      c(&quot;Uninf&quot; = &quot;orange&quot;, 
                                                        &quot;Inf+EFZ&quot; = &quot;green&quot;,
                                                        &quot;Inf&quot; = &quot;red&quot;, 
                                                        &quot;Uninf+EFZ&quot; = &quot;blue&quot;))),
      column_km = 4,
       ))</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-12-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="separate-heatmaps-according-to-infection-condition" class="section level4">
<h4>Separate Heatmaps according to infection condition</h4>
<p>Separating the cells according to different conditions it becomes very evident, that there is a bigger difference between cells treated with EFZ vs. untreated cells than between infected vs uninfected cells. The Inf+EFZ (green) and Uninf+EFZ (blue) show very similar TF activity patterns and the Inf (red) and Uninf (orange) show very similar TF activity patterns. These findings represent the findings from above</p>
<p>Two TFs seem to be more active in Inf+EFZ and Uninf+EFZ (compared to Inf and Uninf):</p>
<ul>
<li><strong>RUNX3</strong></li>
<li><strong>JUNB</strong></li>
</ul>
<p>Three TFs seem to be more active in Inf and Uninf (compared to Inf+EFZ and Uninf+EFZ):</p>
<ul>
<li><strong>NFAT5:</strong> In microglia cells inflammatory signals trigger NFAT5 expression. (Yang X.L. et al., Dev Neurosci, 2018)</li>
<li><strong>TCF7L2</strong></li>
<li><strong>CEBPB</strong></li>
</ul>
<p>Especially the TFs NFAT5, TCF7L2 and CEBPB seem to be more active in untreated cells:</p>
<ul>
<li><strong>CEBPB</strong> regulates pro-inflammatory gene expression, increased expression in HAND (HIV-associated neurocognitive disorder), BUT no data from microglia (Canchi S. et al. J Neuroinflammation, 2020)</li>
<li><strong>NFAT5</strong></li>
</ul>
<pre class="r"><code>ht_conditions &lt;- Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
        #col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        column_title = &quot;split according to infection condition&quot;,
        show_column_names = FALSE,
        row_names_side = &quot;left&quot;,
        col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                           name = &quot;condition&quot;, col = list(condition = c(&quot;Uninf&quot; = &quot;orange&quot;, &quot;Inf+EFZ&quot; = &quot;green&quot;, &quot;Inf&quot; = &quot;red&quot;, &quot;Uninf+EFZ&quot; = &quot;blue&quot;))),
        #row_km = 2,
        #row_split = atac_tfs_df$condition,
        column_split = tri_mg$orig.ident,
        left_annotation = rowAnnotation(TF_footprinting = (atac_tfs_df %&gt;% 
                                                         select(name, condition) %&gt;%
                                                         column_to_rownames(&quot;name&quot;))[[&quot;condition&quot;]],
              name = &quot;TF_footprinting&quot;,
              col = list(TF_footprinting = c(&quot;up_latent&quot; = &quot;pink&quot;, 
                                             &quot;up_uninf&quot; = &quot;darkgreen&quot;)))
        )

ht_conditions </code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-13-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-13-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="separate-heatmaps-according-to-tf-binding-scores-from-atacseq-data" class="section level4">
<h4>Separate Heatmaps according to TF binding scores from ATACseq data</h4>
<p>Separating the heatmap along the rows, according to TFs either more bound in latently infected microglia or more bound in uninfected primary microglia, does not result in a separation according to treatment with EFZ. Therefore, the heatmap below might yield additional biological insights.</p>
<p>What one might expect to see, is that TFs more bound in uninfected primary microglia would also show higher activity in uninfected iMg, while TFs more bound in latently infected primary microglia would show higher activity in Inf+EFZ iMg. However, such a pattern is not immediately visible in the heatmap below.</p>
<pre class="r"><code>ht_tfs1 &lt;- Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
        #col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        column_title = &quot;split according to TF binding score for latent vs. uninfected&quot;,
        show_column_names = FALSE,
        row_names_side = &quot;left&quot;,
        #column_km = 4,
        row_split = atac_tfs_df$condition,
        col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                           name = &quot;condition&quot;, 
                                           col = list(condition = 
                                                        c(&quot;Uninf&quot; = &quot;orange&quot;,
                                                          &quot;Inf+EFZ&quot; = &quot;green&quot;,
                                                          &quot;Inf&quot; = &quot;red&quot;,
                                                          &quot;Uninf+EFZ&quot; = &quot;blue&quot;))),
        #column_split = tri_mg$orig.ident
        left_annotation = rowAnnotation(TF_footprinting = 
                                          (atac_tfs_df %&gt;% 
                                            select(name, condition) %&gt;%
                                            column_to_rownames(&quot;name&quot;))[[&quot;condition&quot;]],
                      name = &quot;TF_footprinting&quot;,
                      col = list(TF_footprinting = c(&quot;up_latent&quot; = &quot;pink&quot;, 
                                                     &quot;up_uninf&quot; = &quot;darkgreen&quot;)))
        )


#ht_list &lt;- ht_conditions + ht_tfs
#draw(ht_list)</code></pre>
<pre class="r"><code>ht_tfs2 &lt;- Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
        #col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        column_title = &quot;split according to TF binding score for latent vs. uninfected&quot;,
        show_column_names = FALSE,
        row_names_side = &quot;left&quot;,
        #column_km = 4,
        row_split = atac_tfs_df$condition,
        column_split = tri_mg@meta.data$orig.ident,
        col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                           name = &quot;condition&quot;, col =
                                             list(condition =
                                                    c(&quot;Uninf&quot; = &quot;orange&quot;,
                                                      &quot;Inf+EFZ&quot; = &quot;green&quot;, 
                                                      &quot;Inf&quot; = &quot;red&quot;, 
                                                      &quot;Uninf+EFZ&quot; = &quot;blue&quot;))),
        #column_split = tri_mg$orig.ident
        left_annotation = rowAnnotation(TF_footprinting = 
                                          (atac_tfs_df %&gt;% 
                                            select(name, condition) %&gt;%
                                            column_to_rownames(&quot;name&quot;))[[&quot;condition&quot;]],
                      name = &quot;TF_footprinting&quot;,
                      col = list(TF_footprinting = c(&quot;up_latent&quot; = &quot;pink&quot;, 
                                                     &quot;up_uninf&quot; = &quot;darkgreen&quot;)))
        )

ht_list &lt;- ht_tfs1 + ht_tfs2
draw(ht_list, ht_gap = unit(2, &quot;cm&quot;))</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-15-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-15-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="statistical-tests-to-find-tf-activity-differences-between-clusters" class="section level3">
<h3>Statistical Tests to find TF activity differences between clusters</h3>
<p>To quantitatively investigate whether we can find TFs which have significantly different activity between clusters we can use Kruskal-Wallis test and pairwise wilcox test as a posthoc analysis.</p>
<pre class="r"><code>p_values &lt;- map(seq.int(1:length(colnames(auc_mtx))), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  #print(tf)
  #print(triculture@meta.data[tf])
  p_values &lt;- unlist(kruskal.test(pull(tri_mg@meta.data[tf]) ~ orig.ident,
                                  tri_mg@meta.data))[&quot;p.value&quot;]
  #print(p_values)
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values&lt;0.01])
[1] 206
## [1] 16</code></pre>
</div>
<div id="boxplots-of-tfs-activity-across-conditions-of-tfs-more-bound-in-latent-infection" class="section level3">
<h3>Boxplots of TFs activity across conditions of TFs more bound in latent infection</h3>
<p>There is a very small or none significant difference between the TF activity between Inf and Uninf as well as between Inf+EFZ and Uninf+EFZ. However, between the treatment conditions there is a difference for REST, ZEB1 and TCF4. However, there is no clear pattern observed.</p>
<pre class="r"><code>
boxplot_list &lt;- map(up_latent, function(tf) {
  ggplot() +
    geom_boxplot(aes(x = tri_mg@meta.data %&gt;% pull(&quot;orig.ident&quot;),
                     y = tri_mg@meta.data %&gt;% pull(tf), 
                 fill = tri_mg@meta.data %&gt;% pull(&quot;orig.ident&quot;))) +
    stat_pvalue_manual(tri_mg@meta.data %&gt;% 
                         pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ orig.ident&quot;))) %&gt;% 
                         add_xy_position()) +
    xlab(&quot;Condition&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;condition&quot;))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 3, nrow = 2))</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-17-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-17-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="boxplots-of-tfs-activity-across-conditions-of-tfs-more-bound-in-uninfected-primary-microglia" class="section level3">
<h3>Boxplots of TFs activity across conditions of TFs more bound in uninfected primary microglia</h3>
<p>In the plots below the trend of TF activity being negatively correlated between treated and untreated cells, independent of the infection status can be seen for some TFs, for example CEBPB and MEF2C. For other TFs there is no difference between either infection status or treatment, for example CREB1 and ETS1.</p>
<p>One assumption migth be, that TFs which are markers of microglia cells might be more or less active depending on whether the cells were treated with EFZ or not. An argument against this assumption is that both CEBPB and MEF2C are typically expressed in microglia, yet we observe an opposite trend. CEBPB is more active in Inf and Uninf cells than in Inf+EFZ and Uninf+EFZ cells, while MEF2C is more active in Inf+EFZ and Uninf+EFZ cells than in Inf and Uninf cells.</p>
<pre class="r"><code>

boxplot_list &lt;- map(up_uninf, function(tf) {
  ggplot() +
    geom_boxplot(aes(x = tri_mg@meta.data %&gt;% pull(&quot;orig.ident&quot;),
                     y = tri_mg@meta.data %&gt;% pull(tf), 
                 fill = tri_mg@meta.data %&gt;% pull(&quot;orig.ident&quot;))) +
    stat_pvalue_manual(tri_mg@meta.data %&gt;% 
                         pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ orig.ident&quot;))) %&gt;% 
                         add_xy_position()) +
    xlab(&quot;Condition&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;condition&quot;))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 5, nrow = 4))</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-18-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-18-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
#annotate_figure(figure,
 #               top = text_grob(&quot;TFs activity across conditions for TFs more bound in uninfected primary microglia&quot;, size = 20))</code></pre>
</div>
</div>
<div id="correlations-between-conditions" class="section level2">
<h2>Correlations between conditions</h2>
<pre class="r"><code>cor_mtx &lt;- tri_mg@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;%
  select(c(&quot;cell&quot;, &quot;orig.ident&quot;, colnames(auc_mtx))) %&gt;%
  group_by(orig.ident) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  gather(TF, value, -orig.ident) %&gt;% 
  spread(orig.ident, value) %&gt;% 
  column_to_rownames(&quot;TF&quot;) %&gt;% cor()

testCor &lt;- cor.mtest(cor_mtx, conf.level = 0.95)

# plot correlations
# p.mat = matrix of p_values
corrplot(cor_mtx,method = &quot;color&quot;, order = &quot;hclust&quot;, 
         addrect = 4, # add boxes 
         p.mat = testCor$p, 
         #sig.level = .05,
         insig = &quot;p-value&quot;)</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-19-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Correlations between Tfs</p>
<pre class="r"><code>cor_mtx &lt;- tri_mg@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  select(c(&quot;cell&quot;, &quot;orig.ident&quot;, atac_tfs_df$name)) %&gt;% # only TFs with differential binding scores from ATACseq
  group_by(orig.ident) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  column_to_rownames(&quot;orig.ident&quot;) %&gt;% cor()

testCor &lt;- cor.mtest(cor_mtx, conf.level = 0.95)

corrplot(cor_mtx,method = &quot;color&quot;, order = &quot;hclust&quot;, 
         addrect = 4, # add boxes 
         p.mat = testCor$p,
         insig = &quot;p-value&quot;)</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-20-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-20-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="tfs-more-bound-in-uninfected-primary-microglia" class="section level3">
<h3>TFs more bound in uninfected primary microglia</h3>
<pre class="r"><code>cor_mtx &lt;- tri_mg@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  select(c(&quot;cell&quot;, &quot;orig.ident&quot;, up_uninf, &quot;REST&quot;)) %&gt;% # only TFs with differential binding scores from ATACseq
  group_by(orig.ident) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  column_to_rownames(&quot;orig.ident&quot;) %&gt;% cor()

testCor &lt;- cor.mtest(cor_mtx, conf.level = 0.95)

corrplot(cor_mtx,method = &quot;color&quot;, order = &quot;hclust&quot;, 
         addrect = 4, # add boxes 
         p.mat = testCor$p,
         insig = &quot;p-value&quot;)</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-21-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-21-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tfs-more-bound-in-latent-infection" class="section level3">
<h3>TFs more bound in latent infection</h3>
<pre class="r"><code>cor_mtx &lt;- tri_mg@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  select(c(&quot;cell&quot;, &quot;orig.ident&quot;, up_latent, &quot;REST&quot;)) %&gt;% # only TFs with differential binding scores from ATACseq
  group_by(orig.ident) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  column_to_rownames(&quot;orig.ident&quot;) %&gt;% cor()

testCor &lt;- cor.mtest(cor_mtx, conf.level = 0.95)

corrplot(cor_mtx,method = &quot;color&quot;, order = &quot;hclust&quot;, 
         addrect = 4, # add boxes 
         p.mat = testCor$p,
         insig = &quot;p-value&quot;)</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-22-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tfs-differntiating-myeloid2-clusters" class="section level3">
<h3>TFs differntiating Myeloid2 clusters</h3>
<p>These TFs were differntially active between the three hierarchical clusters in Myeloid2 cells from HIV-infected patients. Across three different patients they showed correlated changes in activity.</p>
<ul>
<li>YY1, ETS1 and REST were positively correlated with each other and negatively correlated with MEF2A, ETV6, JUNB, MEF2C, MXI1 and CEBPB.</li>
<li>The pattern observed for the iMg from the triculture dataset does not resemble the pattern observed in Myeloid2 cells.</li>
</ul>
<pre class="r"><code>intersect_csf &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_intersect_TFs_different_across_patients_3_clusters&quot;)


cor_mtx &lt;- tri_mg@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  select(c(&quot;cell&quot;, &quot;orig.ident&quot;, intersect_csf$x[intersect_csf$x %in% colnames(auc_mtx)])) %&gt;% # only TFs with differential binding scores from ATACseq
  group_by(orig.ident) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  column_to_rownames(&quot;orig.ident&quot;) %&gt;% cor()

testCor &lt;- cor.mtest(cor_mtx, conf.level = 0.95)

corrplot(cor_mtx,method = &quot;color&quot;, order = &quot;hclust&quot;, 
         addrect = 4, # add boxes 
         p.mat = testCor$p,
         insig = &quot;p-value&quot;)</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-23-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] readxl_1.3.1         multtest_2.46.0      Biobase_2.50.0      
 [4] BiocGenerics_0.36.1  metap_1.6            knitr_1.30          
 [7] rstatix_0.6.0        corrplot_0.92        gridExtra_2.3       
[10] ggpubr_0.4.0         circlize_0.4.13      ComplexHeatmap_2.9.4
[13] ggcorrplot_0.1.3     pheatmap_1.0.12      RColorBrewer_1.1-2  
[16] ggrepel_0.9.1        data.table_1.14.2    Matrix_1.3-4        
[19] edgeR_3.32.0         limma_3.46.0         SeuratObject_4.0.3  
[22] Seurat_4.0.5         forcats_0.5.0        stringr_1.4.0       
[25] dplyr_1.0.7          purrr_0.3.4          readr_1.4.0         
[28] tidyr_1.1.2          tibble_3.1.6         ggplot2_3.3.5       
[31] tidyverse_1.3.0      workflowr_1.6.2     

loaded via a namespace (and not attached):
  [1] utf8_1.2.2            reticulate_1.22       tidyselect_1.1.1     
  [4] htmlwidgets_1.5.4     Rtsne_0.15            munsell_0.5.0        
  [7] mutoss_0.1-12         codetools_0.2-18      ica_1.0-2            
 [10] future_1.23.0         miniUI_0.1.1.1        withr_2.4.3          
 [13] colorspace_2.0-2      highr_0.8             rstudioapi_0.13      
 [16] stats4_4.0.2          ROCR_1.0-11           ggsignif_0.6.0       
 [19] tensor_1.5            listenv_0.8.0         labeling_0.4.2       
 [22] Rdpack_2.1.2          git2r_0.27.1          mnormt_2.0.2         
 [25] polyclip_1.10-0       farver_2.1.0          rprojroot_2.0.2      
 [28] TH.data_1.0-10        parallelly_1.29.0     vctrs_0.3.8          
 [31] generics_0.1.1        xfun_0.29             R6_2.5.1             
 [34] doParallel_1.0.16     clue_0.3-60           locfit_1.5-9.4       
 [37] spatstat.utils_2.2-0  assertthat_0.2.1      promises_1.2.0.1     
 [40] scales_1.1.1          multcomp_1.4-15       gtable_0.3.0         
 [43] globals_0.14.0        goftest_1.2-3         sandwich_3.0-0       
 [46] rlang_0.4.12          GlobalOptions_0.1.2   splines_4.0.2        
 [49] lazyeval_0.2.2        spatstat.geom_2.3-0   broom_0.7.2          
 [52] yaml_2.2.1            reshape2_1.4.4        abind_1.4-5          
 [55] modelr_0.1.8          backports_1.2.0       httpuv_1.6.3         
 [58] tools_4.0.2           ellipsis_0.3.2        spatstat.core_2.3-1  
 [61] ggridges_0.5.2        TFisher_0.2.0         Rcpp_1.0.7           
 [64] plyr_1.8.6            rpart_4.1-15          deldir_1.0-6         
 [67] pbapply_1.5-0         GetoptLong_1.0.5      cowplot_1.1.1        
 [70] S4Vectors_0.28.1      zoo_1.8-9             haven_2.3.1          
 [73] cluster_2.1.0         fs_1.5.1              magrittr_2.0.1       
 [76] magick_2.5.2          scattermore_0.7       openxlsx_4.2.3       
 [79] lmtest_0.9-39         reprex_0.3.0          RANN_2.6.1           
 [82] tmvnsim_1.0-2         mvtnorm_1.1-1         whisker_0.4          
 [85] fitdistrplus_1.1-6    matrixStats_0.61.0    hms_1.1.1            
 [88] patchwork_1.1.1       mime_0.12             evaluate_0.14        
 [91] xtable_1.8-4          rio_0.5.16            IRanges_2.24.1       
 [94] shape_1.4.6           compiler_4.0.2        KernSmooth_2.23-18   
 [97] crayon_1.4.2          htmltools_0.5.2       mgcv_1.8-33          
[100] later_1.3.0           lubridate_1.7.9.2     DBI_1.1.1            
[103] dbplyr_2.0.0          MASS_7.3-53           car_3.0-10           
[106] cli_3.1.0             rbibutils_2.2.4       igraph_1.2.9         
[109] pkgconfig_2.0.3       sn_2.0.1              numDeriv_2016.8-1.1  
[112] foreign_0.8-80        plotly_4.9.2.1        spatstat.sparse_2.0-0
[115] xml2_1.3.2            foreach_1.5.1         rvest_0.3.6          
[118] digest_0.6.29         sctransform_0.3.2     RcppAnnoy_0.0.19     
[121] spatstat.data_2.1-0   rmarkdown_2.6         cellranger_1.1.0     
[124] leiden_0.3.5          uwot_0.1.10           curl_4.3.2           
[127] shiny_1.7.1           rjson_0.2.20          lifecycle_1.0.1      
[130] nlme_3.1-150          jsonlite_1.7.2        carData_3.0-4        
[133] viridisLite_0.4.0     fansi_0.5.0           pillar_1.6.4         
[136] lattice_0.20-41       plotrix_3.8-2         fastmap_1.1.0        
[139] httr_1.4.2            survival_3.2-7        glue_1.5.1           
[142] zip_2.1.1             png_0.1-7             iterators_1.0.13     
[145] stringi_1.7.6         mathjaxr_1.4-0        irlba_2.3.3          
[148] future.apply_1.8.1   </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
