<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="kmikulik" />


<title>Analysis of Triculture dataset and integration with CSF/Blood dataset</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Abstract</a>
</li>
<li>
  <a href="Introduction.html">Introduction</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="HIV_CSF_dataset_analysis.html">1. CSF/Blood samples</a>
    </li>
    <li>
      <a href="TF_activity_analysis_of_myeloid2_cluster.html">2. TF activity Myeloid2</a>
    </li>
    <li>
      <a href="Triculture_csf_integration.html">3. Integration CSF/Blood &amp; triculture</a>
    </li>
    <li>
      <a href="Triculture_dataset_GRN_with_Scenic.html">4. TF activity iMg</a>
    </li>
  </ul>
</li>
<li>
  <a href="Discussion.html">Discussion</a>
</li>
<li>
  <a href="Methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Katharina782/report">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Analysis of Triculture dataset and integration with CSF/Blood dataset</h1>
<h4 class="author">kmikulik</h4>
<h4 class="date">28 12 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-01-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
unnamed-chunk-1
</li>
<li>
unnamed-chunk-10
</li>
<li>
unnamed-chunk-11
</li>
<li>
unnamed-chunk-12
</li>
<li>
unnamed-chunk-13
</li>
<li>
unnamed-chunk-14
</li>
<li>
unnamed-chunk-15
</li>
<li>
unnamed-chunk-16
</li>
<li>
unnamed-chunk-17
</li>
<li>
unnamed-chunk-18
</li>
<li>
unnamed-chunk-19
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-20
</li>
<li>
unnamed-chunk-21
</li>
<li>
unnamed-chunk-22
</li>
<li>
unnamed-chunk-23
</li>
<li>
unnamed-chunk-24
</li>
<li>
unnamed-chunk-25
</li>
<li>
unnamed-chunk-26
</li>
<li>
unnamed-chunk-27
</li>
<li>
unnamed-chunk-28
</li>
<li>
unnamed-chunk-29
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-30
</li>
<li>
unnamed-chunk-31
</li>
<li>
unnamed-chunk-32
</li>
<li>
unnamed-chunk-33
</li>
<li>
unnamed-chunk-4
</li>
<li>
unnamed-chunk-40
</li>
<li>
unnamed-chunk-42
</li>
<li>
unnamed-chunk-47
</li>
<li>
unnamed-chunk-48
</li>
<li>
unnamed-chunk-49
</li>
<li>
unnamed-chunk-5
</li>
<li>
unnamed-chunk-50
</li>
<li>
unnamed-chunk-51
</li>
<li>
unnamed-chunk-53
</li>
<li>
unnamed-chunk-54
</li>
<li>
unnamed-chunk-56
</li>
<li>
unnamed-chunk-59
</li>
<li>
unnamed-chunk-6
</li>
<li>
unnamed-chunk-60
</li>
<li>
unnamed-chunk-61
</li>
<li>
unnamed-chunk-7
</li>
<li>
unnamed-chunk-8
</li>
<li>
unnamed-chunk-9
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>Triculture_csf_integration_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttreef35ba45536a8b7546c4144a569fe908ce206aa4dtargetblankf35ba45a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Katharina782/report/tree/f35ba45536a8b7546c4144a569fe908ce206aa4d" target="_blank">f35ba45</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttreef35ba45536a8b7546c4144a569fe908ce206aa4dtargetblankf35ba45a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Katharina782/report/tree/f35ba45536a8b7546c4144a569fe908ce206aa4d" target="_blank">f35ba45</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/Independent_cell_type_identification_Farhadian_dataset_cache/
    Ignored:    analysis/Triculture_csf_integration_cache/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  README.html
    Untracked:  analysis/Triculture_csf_integration/
    Untracked:  data/triculture_seurat_object
    Untracked:  rebuild file.Rmd
    Untracked:  rebuild-file.Rmd
    Untracked:  rebuild-file.html
    Untracked:  rebuild-file_cache/
    Untracked:  rebuild-file_files/

Unstaged changes:
    Modified:   analysis/Discussion.Rmd
    Modified:   analysis/HIV_CSF_dataset_analysis.Rmd
    Modified:   analysis/Introduction.Rmd
    Modified:   analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd
    Modified:   analysis/Triculture_csf_integration.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/references.bib

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Triculture_csf_integration.Rmd</code>) and HTML (<code>docs/Triculture_csf_integration.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/f35ba45536a8b7546c4144a569fe908ce206aa4d/docs/Triculture_csf_integration.html" target="_blank">f35ba45</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/51bcaf3edef9692eac321325ef570b593e2f6379/analysis/Triculture_csf_integration.Rmd" target="_blank">51bcaf3</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/75af25c571d2f255bb408c63df6ee70c423b892a/docs/Triculture_csf_integration.html" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/dfcd540d3a574f3ad19196570abd4c6c51bd0aa3/analysis/Triculture_csf_integration.Rmd" target="_blank">dfcd540</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/3d3c73318704913e8bd8fd6101547ea25899045f/docs/Triculture_csf_integration.html" target="_blank">3d3c733</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/552f3e790fdab3c4f5386c882cd6d94f4d50d84f/analysis/Triculture_csf_integration.Rmd" target="_blank">552f3e7</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
<td>
wflow_publish(c(“analysis/Triculture_csf_integration.Rmd,” “analysis/Discussion.Rmd,”
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style>
body {
text-align: justify}
</style>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, collapse = TRUE, message = FALSE,
                      cache = TRUE, cache.lazy = FALSE, autodep = TRUE)
set.seed(42) # TODO: check with default seed</code></pre>
<pre class="r"><code>library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggrepel)
#library(harmony)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(corrplot)
library(rstatix)
library(knitr)
library(metap)
library(multtest)
library(readxl)</code></pre>
<div id="triculture-dataset" class="section level1">
<h1>Triculture Dataset</h1>
<p>The triculture dataset used in the following analysis was published by <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span>. They used a Triculture system with induced neurons, induced astrocytes and induced microglia. Induced microglia express marker genes CX3CR1, IBA1, TMEM119 and P2RY12. CCR5 which is a co-receptor necessary for HIV infection is also expressed by these cells, but they lack expression of myeloid-progenitor markers <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span>.</p>
<p>iMg were infected with HIV for 15 days, resulting in 94% of cells infected<br />
(positive for HIV capsid protein p24 and exhibiting multinucleation). Some cells were treated with the antiretroviral therapy EFZ, which blocks HIV reverse transcriptase. This way, it was possible to study non-productively infected cells. EFZ reduced infection by two thirds <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span>.</p>
<p>In the infected, but not treated iMg (Inf) inflammatory pathways were significantly uperegulated compared to the uninfected condition(Uninf), including IL-8, NFkB and EIF2. There was decreased expression of pro-imflammatory genes IL1b, Il8 and TNFalpha in the infected and treated (Inf+EFZ) vs Inf condition. This suggests a reduced inflammatory reaction <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span>. Nevertheless, there were two distinct pathways which were upregulated in Inf+EFZ iMg, namely RhoGDI signaling and CD40 signaling pathway. This will be discussed in more detail later on <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span>.</p>
<p>There was a change in signaling pathways caused by EFZ treatment even in the uninfected and treated (Uninf+EFZ) versus the uninfected, but not treated (Uninf) condition. The EFZ treatment alone was very different from the EFZ treatment with infection <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span>.</p>
<details>
<summary>
<strong>Files and data structure</strong>
</summary>
<p><strong>Read in Files:</strong></p>
<p>Abbreviations:</p>
<ul>
<li>ia/2 = infected with HIV + treated with ART</li>
<li>ca = uninfected + ART</li>
<li>ctl = uninfected</li>
<li>inf = infected</li>
</ul>
<p>File structure:</p>
<ul>
<li>the “.mtx” files contains a sparse matrix of class dgTMatrix</li>
<li>the “barcodes.tsv” files contain the cell barcodes</li>
<li>the “features.tsv” files contain the feature names as well as ensemble gene ids and “gene expression” strings in a third column</li>
</ul>
<p>There are duplicated features in the data, probably, because there are multiple ensemble gene IDs mapping to the same gene symbol. Therefore, the same gene symbol has multiple expression values. How do you deal with this problem?</p>
<ul>
<li>You could make them unique with the function <code>make.unique()</code></li>
<li>You could remove the duplicated features. <strong>This is what I did</strong></li>
</ul>
<pre><code>path &lt;- &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/GSE143686_RAW/&quot; #&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/GSE143686_RAW/&quot;
all_files &lt;- list.files(path)

all_cond_list &lt;- map(seq.int(3,7), function(n){
  file &lt;- all_files[grepl(paste0(&quot;GSM427258&quot;, n), all_files)]
  name &lt;- str_extract(file[[1]],&quot;ca|ctl|ia2|ia|inf&quot;)
  barcodes &lt;- read.table(paste0(path,file[[1]]), sep = &quot;\t&quot;) 
  features &lt;- read.table(paste0(path, file[[2]]), sep = &quot;\t&quot;)
  matrix &lt;- readMM(paste0(path, file[[3]]))
  colnames(matrix) &lt;- barcodes$V1
  rownames(matrix) &lt;- features$V2
  matrix &lt;- matrix[!duplicated(rownames(matrix)),]
  seurat_ob &lt;- CreateSeuratObject(matrix, project = name,
                                  assay = &quot;RNA&quot;, min.cells = 1, min.features = 1)
  seurat_ob[[&quot;percent_mt&quot;]] &lt;- PercentageFeatureSet(seurat_ob, pattern = &quot;^MT-&quot;)
  list(name = name, count = matrix, seurat_ob = seurat_ob)
})

#saveRDS(all_cond_list, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/seurat_list&quot;)</code></pre>
<hr />
</details>
<details>
<summary>
<strong>Files and data structure</strong>
</summary>
</div>
<div id="data-processing" class="section level1 tabset">
<h1 class="tabset">Data processing</h1>
<div id="before-quality-control" class="section level2">
<h2>Before Quality Control:</h2>
<pre class="r"><code>all_cond_list &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/seurat_list&quot;)

before_qc &lt;- merge(all_cond_list[[1]]$seurat_ob,
                    c(all_cond_list[[2]]$seurat_ob,
                      all_cond_list[[3]]$seurat_ob,
                      all_cond_list[[4]]$seurat_ob,
                      all_cond_list[[5]]$seurat_ob),
                    project = &quot;triculture_all_conditions&quot;,
                    add.cell.ids = c(paste0(all_cond_list[[1]]$name),
                                     paste0(all_cond_list[[2]]$name),
                                     paste0(all_cond_list[[3]]$name),
                                     paste0(all_cond_list[[4]]$name),
                                     paste0(all_cond_list[[5]]$name)))</code></pre>
<pre class="r"><code>qual_metric &lt;- c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent_mt&quot;)

plots &lt;- map(qual_metric, function(metric){
  ggplot() +
    geom_violin(aes(x = before_qc@meta.data %&gt;% pull(&quot;orig.ident&quot;),
                 y = before_qc@meta.data %&gt;% pull(metric), 
                 fill = before_qc@meta.data %&gt;% pull(&quot;orig.ident&quot;)), 
                alpha = .5) +
    geom_boxplot(aes(x = before_qc@meta.data %&gt;% pull(&quot;orig.ident&quot;),
                     y = before_qc@meta.data %&gt;% pull(metric)),
                 alpha = 0) +
    xlab(&quot;sample origin&quot;) +
    ylab(paste0(metric)) +
    NoLegend() 
})

do.call(grid.arrange, c(plots, ncol = 3, nrow = 1))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-3-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-3-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
<div id="after-quality-control" class="section level2">
<h2>After quality control</h2>
<p>For each sample separately</p>
<ul>
<li>remove genes expressing less than 200 genes or more than 5000</li>
<li>remove cells expressing more than 20% mitochondrial genes</li>
<li>merge Seurat objects after Quality Control</li>
</ul>
<pre class="r"><code># add percentage of mitochondiral genes to metadata
all_cond &lt;- map(seq.int(1,5), function(n){
  seurat_ob &lt;- all_cond_list[[n]]$seurat_ob
  #seurat_ob[[&quot;percent_mt&quot;]] &lt;- PercentageFeatureSet(seurat_ob, pattern = &quot;^MT-&quot;)
  #plot &lt;- VlnPlot(seurat_ob,
   #               features = c(&quot;nFeature_RNA&quot;, &quot;percent_mt&quot;, &quot;nCount_RNA&quot;),
    #              ncol = 3) +
    #labs(title = paste0(&quot;condition &quot;, all_cond_list[[n]]$name))
  seurat_ob &lt;- subset(seurat_ob, nFeature_RNA &gt; 200 &amp; nFeature_RNA &lt; 5000 &amp; percent_mt &lt; 20)
  list(name = all_cond_list[[n]]$name, seurat_ob = seurat_ob)#, plot = plot)
})</code></pre>
<pre class="r"><code>triculture &lt;- merge(all_cond[[1]]$seurat_ob,
                    c(all_cond[[2]]$seurat_ob,
                      all_cond[[3]]$seurat_ob,
                      all_cond[[4]]$seurat_ob,
                      all_cond[[5]]$seurat_ob),
                    project = &quot;triculture_all_conditions&quot;,
                    add.cell.ids = c(paste0(all_cond[[1]]$name),
                                     paste0(all_cond[[2]]$name),
                                     paste0(all_cond[[3]]$name),
                                     paste0(all_cond[[4]]$name),
                                     paste0(all_cond[[5]]$name)))</code></pre>
<pre class="r"><code>qual_metric &lt;- c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent_mt&quot;)

plots &lt;- map(qual_metric, function(metric){
  ggplot() +
    geom_violin(aes(x = triculture@meta.data %&gt;% pull(&quot;orig.ident&quot;),
                 y = triculture@meta.data %&gt;% pull(metric), 
                 fill = triculture@meta.data %&gt;% pull(&quot;orig.ident&quot;)), 
                alpha = .5) +
    geom_boxplot(aes(x = triculture@meta.data %&gt;% pull(&quot;orig.ident&quot;),
                     y = triculture@meta.data %&gt;% pull(metric)),
                 alpha = 0) +
    xlab(&quot;sample origin&quot;) +
    ylab(paste0(metric)) +
    NoLegend() 
})

do.call(grid.arrange, c(plots, ncol = 3, nrow = 1))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-6-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-6-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
<div id="normalization-scaling-find-variable-features" class="section level2">
<h2>Normalization, Scaling, Find Variable Features</h2>
<ul>
<li>log normalization</li>
<li>scale data so that we have mean = 0 and variance = 1</li>
<li>Find the 2000 most highly variable features</li>
</ul>
<pre><code>triculture &lt;- NormalizeData(triculture, verbose = FALSE)
triculture &lt;- ScaleData(triculture, verbose = FALSE)
triculture &lt;- FindVariableFeatures(triculture, verbose = FALSE)</code></pre>
<hr />
</div>
<div id="dimensionality-reduction" class="section level2">
<h2>Dimensionality reduction</h2>
<ul>
<li>compute PCA</li>
<li>use elbow plot to determine dimensions that contribute to variability</li>
<li>find the k-nearest neighbors</li>
<li>find Clusters based on KNN-graph</li>
<li>calculate TSNE embedding</li>
</ul>
<pre><code>triculture &lt;- RunPCA(triculture, features = VariableFeatures(triculture), verbose = FALSE)
ElbowPlot(triculture)

triculture &lt;- FindNeighbors(triculture, verbose = FALSE)
triculture &lt;- FindClusters(triculture, verbose = FALSE)

triculture &lt;- RunTSNE(triculture, dims = 1:10, verbose = FALSE)</code></pre>
<pre class="r"><code>triculture &lt;- readRDS(&quot;seurat_objects/triculture_seurat_object&quot;)

#triculture &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_seurat_object&quot;)

DimPlot(triculture, reduction = &quot;tsne&quot;, pt.size = .01,label = TRUE,
        group.by = &quot;seurat_clusters&quot;)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-7-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
<div id="cell-type-annotation" class="section level2">
<h2>Cell type annotation</h2>
<p>For annotating the different clusters the same markers as in <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span> were used:</p>
<ul>
<li><em>Microglia:</em> AIF1, SPI1, CD4</li>
<li><em>Astrocytes:</em> SOX9, THBS1</li>
<li><em>Neurons:</em> SYN1, MAP2</li>
</ul>
<pre class="r"><code>#markers &lt;- FindAllMarkers(triculture)
#triculture &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_seurat_object&quot;)
markers &lt;- read.table(&quot;data_files/marker_genes_triculture_dataset&quot;)
#markers &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/marker_genes_triculture_dataset&quot;)

top_2 &lt;- markers %&gt;% group_by(&quot;seurat_cluster&quot;) %&gt;% top_n(n = 2, wt = avg_log2FC)

marker_genes &lt;- c(&quot;AIF1&quot;, &quot;SPI1&quot;, &quot;CD4&quot;, &quot;MAP2&quot;, &quot;SYN1&quot;, &quot;THBS1&quot;, &quot;SOX9&quot;)
p1 &lt;- FeaturePlot(triculture, features = &quot;AIF1&quot;)
p2 &lt;- FeaturePlot(triculture, features = &quot;SPI1&quot;)
p3 &lt;- FeaturePlot(triculture, features = &quot;CD4&quot;)
p4 &lt;- FeaturePlot(triculture, features = &quot;MAP2&quot;)
p5 &lt;- FeaturePlot(triculture, features = &quot;SYN1&quot;)
p6 &lt;- FeaturePlot(triculture, features = &quot;THBS1&quot;)
p7 &lt;- FeaturePlot(triculture, features = &quot;SOX9&quot;)

ggarrange(p1,p2,p3,p4,p5,p6,p7, ncol = 2, nrow =4 )</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-8-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
<strong>Details on cell type annotation</strong>
</summary>
<p><strong>Gene expression</strong></p>
<p>Different clusters were annotated according to the gene expression of marker genes mentioned above.</p>
<pre class="r"><code>
VlnPlot(triculture, features = c(&quot;CD4&quot;, &quot;SPI1&quot;, &quot;AIF1&quot;), group.by = &quot;seurat_clusters&quot;)#, group.by = seurat_cluster)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-9-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-9-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(triculture, features = c(&quot;MAP2&quot;, &quot;SYN1&quot;), group.by = &quot;seurat_clusters&quot;)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-9-2.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-2">
Past versions of unnamed-chunk-9-2.png
</button>
</p>
<div id="fig-unnamed-chunk-9-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-9-2.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>VlnPlot(triculture, features = c(&quot;THBS1&quot;, &quot;SOX9&quot;), group.by = &quot;seurat_clusters&quot;)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-9-3.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-3">
Past versions of unnamed-chunk-9-3.png
</button>
</p>
<div id="fig-unnamed-chunk-9-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-9-3.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 4, 9, 11, 18 = Microglia
# 0, 1, 2, 3,8, 10, 13, 15, 18, 19 = Neurons 
# 6, 7, 12, 14, 16 = Astrocytes
# 5, 17 are undesginated
cell_types &lt;- (triculture@meta.data %&gt;% 
  mutate(cell_type = case_when(
    seurat_clusters %in% c(4,9,11) ~ &quot;iMg&quot;, 
    seurat_clusters %in% c(6,7,12,14,16) ~ &quot;iAst&quot;, 
    seurat_clusters %in% c(0,1,2,3,8,10,13,15,18,19) ~ &quot;iNeurons&quot;,
    TRUE  ~ &quot;Undesignated&quot;)))[[&quot;cell_type&quot;]]
  
triculture &lt;- AddMetaData(triculture, cell_types, col.name = &quot;cell_type&quot;)</code></pre>
<hr />
</details>
<table style="width:6%;">
<colgroup>
<col width="5%" />
</colgroup>
<tbody>
<tr class="odd">
<td>&lt;/details</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="exploring-the-dataset" class="section level1">
<h1>Exploring the dataset</h1>
<div id="remove-undesignated-cells" class="section level3">
<h3>Remove undesignated cells</h3>
<p>There is one group of cells where it is not clear which cell state they represent (Undesignated), we will exclude them from further analysis as <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span> did.</p>
<pre class="r"><code>p1 &lt;- DimPlot(triculture, reduction = &quot;tsne&quot;, pt.size = .01, label = TRUE,
        group.by = &quot;cell_type&quot;) +
  labs(title = &quot;cell types&quot;)</code></pre>
<pre class="r"><code>triculture_clean &lt;- subset(triculture, cell_type %in% c(&quot;iMg&quot;, &quot;iNeurons&quot;, &quot;iAst&quot;)) </code></pre>
<pre class="r"><code>p2 &lt;- DimPlot(triculture_clean, reduction = &quot;tsne&quot;, pt.size = .01, label = TRUE,
        group.by = &quot;cell_type&quot;) + 
  labs(title = &quot;cell types&quot;)</code></pre>
<pre class="r"><code>p1+p2</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-14-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-14-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
<strong>Heatmap of top 20 marker genes for each cell type</strong>
</summary>
<p>Top20 marker genes for each cell cell type:</p>
<pre class="r"><code>Idents(triculture_clean) &lt;- &quot;cell_type&quot;
top_n &lt;- markers %&gt;% group_by (cluster) %&gt;% top_n(n = 20, wt = avg_log2FC)
DoHeatmap(triculture_clean, features = top_n$gene)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-15-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-15-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</details>
<details>
<summary>
<strong>Number of cells per condition and cell type</strong>
</summary>
<p>How many cells of each cell type and condition are present in the dataset?</p>
<pre class="r"><code>triculture_clean@meta.data %&gt;% group_by(cell_type, orig.ident) %&gt;% 
  summarize(n = n()) %&gt;% 
  mutate(orig.ident = case_when(
                      orig.ident %in% c(&quot;ia&quot;, &quot;ia2&quot;) ~ &quot;Inf+EFZ&quot;, 
                      orig.ident == &quot;ca&quot; ~ &quot;Uninf&quot;,
                      orig.ident == &quot;ctl&quot; ~ &quot;Uninf+EFZ&quot;,
                      orig.ident == &quot;inf&quot; ~ &quot;Inf&quot;)) %&gt;% 
  pivot_wider(names_from = orig.ident, values_from = n) %&gt;% 
  kable(caption = &quot;Number of cells per condition and cell type&quot;) # to make table look nice in Rmarkdown
Warning: Values are not uniquely identified; output will contain list-cols.
* Use `values_fn = list` to suppress this warning.
* Use `values_fn = length` to identify where the duplicates arise
* Use `values_fn = {summary_fun}` to summarise duplicates</code></pre>
<table>
<caption>Number of cells per condition and cell type</caption>
<thead>
<tr class="header">
<th align="left">cell_type</th>
<th align="left">Uninf</th>
<th align="left">Uninf+EFZ</th>
<th align="left">Inf+EFZ</th>
<th align="left">Inf</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">iAst</td>
<td align="left">1481</td>
<td align="left">686</td>
<td align="left">878, 944</td>
<td align="left">1269</td>
</tr>
<tr class="even">
<td align="left">iMg</td>
<td align="left">1163</td>
<td align="left">722</td>
<td align="left">632, 750</td>
<td align="left">1166</td>
</tr>
<tr class="odd">
<td align="left">iNeurons</td>
<td align="left">6733</td>
<td align="left">4899</td>
<td align="left">747, 2250</td>
<td align="left">3964</td>
</tr>
</tbody>
</table>
<hr />
</details>
</div>
<div id="infection-and-treatment-conditions" class="section level3">
<h3>Infection and treatment conditions</h3>
<p>Interestingly, the infected cells treated with EFZ (Inf+EFZ) clearly separate from the infected, but untreated cells (Inf) and the uninfected cells (Uninf, Uninf+EFZ) in the microglia cluster as can be seen in the figure below.</p>
<pre class="r"><code># rename origin of ident for easier interpretation
origin_vector &lt;- (triculture_clean@meta.data %&gt;% 
                    mutate(orig.ident = case_when(
                      orig.ident %in% c(&quot;ia&quot;, &quot;ia2&quot;) ~ &quot;Inf+EFZ&quot;, 
                      orig.ident == &quot;ca&quot; ~ &quot;Uninf&quot;,
                      orig.ident == &quot;ctl&quot; ~ &quot;Uninf+EFZ&quot;,
                      orig.ident == &quot;inf&quot; ~ &quot;Inf&quot;)))[[&quot;orig.ident&quot;]]

triculture_clean &lt;- AddMetaData(triculture_clean, origin_vector, col.name = &quot;orig.ident&quot;)</code></pre>
<pre class="r"><code>p1 &lt;- DimPlot(triculture_clean, reduction = &quot;tsne&quot;, pt.size = .01, label = TRUE,
        group.by = &quot;cell_type&quot;) + 
  labs(title = &quot;cell types&quot;)
p2 &lt;- DimPlot(triculture_clean, reduction = &quot;tsne&quot;, 
              group.by = &quot;orig.ident&quot;, pt.size = .01, ) + 
  labs(title = &quot;Infection &amp; treatment condition&quot;)

p1 + p2</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-18-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-18-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
<strong>Inflammatory gene expression in iMg</strong>
</summary>
<p>The largest change in gene expression related to inflammation was found in iMg among the 3 cell types <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span>. Shown are only inflammatory genes.</p>
<pre class="r"><code>inflamm_genes &lt;- c(&quot;TNF&quot;, &quot;ACSL1&quot;, &quot;MGST3&quot;, &quot;B2M&quot;, &quot;IFNGR2&quot;, &quot;GRINA&quot;, &quot;TLR2&quot;, 
                   &quot;SOD2&quot;, &quot;GLUL&quot;, &quot;CFLAR&quot;, &quot;CTNNB1&quot;, &quot;CXCL8&quot;, &quot;CCL4&quot;, &quot;IL1B&quot;,
                   &quot;ATF4&quot;, &quot;CXCL8&quot;, &quot;CCL4&quot;, &quot;CCL2&quot;, &quot;IL1B&quot;, &quot;ATF4&quot;, &quot;CXCL1&quot;, 
                   &quot;NFKB1&quot;, &quot;TANK&quot;, &quot;APOE&quot;, &quot;TREM2&quot;, &quot;LY96&quot;, &quot;SRC&quot;, &quot;NCF1&quot;, 
                   &quot;HCK&quot;, &quot;LYN&quot;, &quot;VASP&quot;, &quot;C3&quot;, &quot;C5&quot;, &quot;RAP1B&quot;, &quot;RAP2B&quot;, &quot;MAF&quot;,
                   &quot;JUND&quot;, &quot;SQSTM1&quot;, &quot;FTH1&quot;, &quot;RHOQ&quot;, &quot;CD36&quot;, &quot;FOS&quot;, &quot;NFKBIA&quot;,
                   &quot;ICAM1&quot;, &quot;TNFAIP3&quot;, &quot;LYZ&quot;, &quot;PPP1R12A&quot;, &quot;RND3&quot;, &quot;CYBA&quot;, &quot;IRF8&quot;,
                   &quot;SLC25A6&quot;)

DoHeatmap(triculture_clean, features = inflamm_genes)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-19-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</details>
<details>
<summary>
<strong>Differences of inflammatory gene expression between conditions</strong>
</summary>
<p>The gene expression of inflammatory genes in microglia was not very different between the conditions. The Inf+EFZ cells show a slightly different pattern of gene expression.</p>
<pre class="r"><code>DoHeatmap(subset(triculture_clean, cell_type == &quot;iMg&quot;), 
          features = inflamm_genes, group.by = &quot;orig.ident&quot;)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-20-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</details>
</div>
</div>
<div id="integration-triculture-csfblood" class="section level1">
<h1>Integration Triculture &amp; CSF/Blood</h1>
<p>The HIV Blood &amp; CSF dataset <span class="citation">(<a href="#ref-Farhadian" role="doc-biblioref">Shelli F. Farhadian and Spudich 2018</a>)</span> used in the previous analysis was integrated with the triculture dataset <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span>.</p>
<p>Integration of the two datasets is based on the Seurat package <span class="citation">(<a href="#ref-Stuart" role="doc-biblioref">Stuart T 2019</a>)</span>. It first identifies anchors between datasets, calculates scores for all anchors and then calculates an integrated expression matrix which can be used for downstream analysis.</p>
<ol style="list-style-type: decimal">
<li>Identify features which are variable across datasets</li>
<li>Identify anchors between the two datasets and create an integrated data assay</li>
<li>Run a single integrated analysis on all cells</li>
</ol>
<details>
<summary>
<strong>Details on integration</strong>
</summary>
<p>For more details see: <span class="citation">(<a href="#ref-Stuart" role="doc-biblioref">Stuart T 2019</a>)</span> and Discussion</p>
<ol style="list-style-type: decimal">
<li>Identify features which are variable across datasets</li>
<li>Identify anchors between the two datasets and create an integrated data assay</li>
<li>Run a single integrated analysis on all cells</li>
</ol>
<pre><code>hiv7 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds&quot;)
hiv7 &lt;- NormalizeData(hiv7, verbose = FALSE)
hiv7 &lt;- FindVariableFeatures(hiv7, verbose = FALSE)

# add both datasets to a lits
data_list &lt;- list(hiv = hiv7, triculture = triculture_clean)

# select features that are repeatedly variable across datasets for integration
features &lt;- SelectIntegrationFeatures(object.list = data_list, verbose = FALSE)

# identify integration anchors and perform integration
anchors &lt;- FindIntegrationAnchors(object.list = data_list, 
                                  anchor.features = features, verbose = FALSE) 
combined &lt;- IntegrateData(anchorset = anchors, verbose = FALSE)

# run analysis on integrated dataset
combined &lt;- ScaleData(combined, verbose = FALSE)
combined &lt;- RunPCA(combined, verbose = FALSE)
#ElbowPlot(combined)
combined &lt;- RunTSNE(combined, reduction = &quot;pca&quot;, dims = 1:10, verbose = FALSE)
combined &lt;- FindNeighbors(combined, reduction = &quot;pca&quot;, dims = 1:10, verbose = FALSE)
combined &lt;- FindClusters(combined, resolution = .3, verbose = FALSE)

# add annotations from which dataset the sample came from
dataset_vector &lt;- (combined@meta.data %&gt;% mutate(dataset = ifelse(orig.ident %in%
                                                 c(&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;,
                                                   &quot;HIV3_CSF&quot;, &quot;HIV1_Bld&quot;, 
                                                   &quot;HIV2_Bld&quot;, &quot;Uninfected1_CSF&quot;,
                                                   &quot;Uninfected2_CSF&quot;),
                                               &quot;csf&quot;, &quot;triculture&quot;)))[[&quot;dataset&quot;]]

combined &lt;- AddMetaData(combined, dataset_vector, col.name = &quot;dataset&quot;)

# add information on the cell groups in the combined datasets
combined_clusters &lt;- (combined@meta.data %&gt;% 
  mutate(combined_clusters = ifelse(seurat_clusters %in% c(2,10), &quot;Microglia&quot;, 
                            ifelse(seurat_clusters %in% c(3,9), &quot;iAst&quot;, 
                                   ifelse(seurat_clusters %in% c(0,1,7,11),
                                          &quot;iNeurons&quot;, 
                                          ifelse(seurat_clusters %in% c(8,5,6), &quot;Immune Cells&quot;, &quot;Undesignated&quot;))))))[[&quot;combined_clusters&quot;]]
  
combined &lt;- AddMetaData(combined, combined_clusters, col.name = &quot;combined_clusters&quot;)

#saveRDS(combined,&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_hiv_combined_seurat_object&quot;)
</code></pre>
<hr />
</details>
<div id="integrated-clusters" class="section level2">
<h2>Integrated clusters</h2>
<p>When integrating the CSF dataset and the Triculture dataset the iMg cells cluster together with the Myeloid cells from the CSF and Blood samples from the HIV infected/uninfected patients (see figure below). This is as expected, since they are both myeloid-lineage cells. However, it also makes it more difficult to draw conclusions on which of the five Myeloid clusters is the most microglia-like. Therefore, we had a closer look at this myeloid-lineage cluster consisting of iMg from triculture and Myeloid cells from CSF and Blood samples.</p>
<div id="visualization-of-the-integrated-dataset" class="section level4">
<h4>Visualization of the integrated dataset</h4>
<pre class="r"><code>combined = readRDS(&quot;seurat_objects/triculture_hiv_combined_seurat_object_2&quot;)

p1 &lt;- DimPlot(combined, reduction = &quot;tsne&quot;, group.by  = &quot;dataset&quot;,
              label = TRUE) + NoLegend() +
  labs(title = &quot;Dataset&quot;)
p2 &lt;- DimPlot(combined, reduction = &quot;tsne&quot;, group.by = &quot;cell_type&quot;,
              label = TRUE, repel = TRUE) + NoLegend() +
  labs(title = &quot;Cell types&quot;)
p3 &lt;- DimPlot(combined, reduction = &quot;tsne&quot;, group.by = &quot;cell_type&quot;,
              split.by = &quot;dataset&quot;, label = TRUE, repel = TRUE) + NoLegend() +
  labs(title = &quot;Cell types split into datasets&quot;)
p4 &lt;- DimPlot(combined, reduction = &quot;tsne&quot;, group.by = &quot;seurat_clusters&quot;,
              label = TRUE) + NoLegend() +
  labs(title = &quot;Clusters&quot;)

a1 &lt;- DimPlot(combined, reduction = &quot;tsne&quot;, split.by = &quot;dataset&quot;, 
        group.by = &quot;seurat_clusters&quot;, label = TRUE, repel =TRUE) +
  NoLegend() +
  labs(title = &quot;Clusters split into datasets&quot;)

a2 &lt;- DimPlot(combined, reduction = &quot;tsne&quot;, split.by = &quot;dataset&quot;, 
              group.by = &quot;orig.ident&quot;) +
  labs(title = &quot;Sample origin split into datasets&quot;)

ggarrange(p1, p2, p3,p4, a1, a2, ncol = 2, nrow = 3)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-21-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-21-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="myeloid-lineage-cluster" class="section level2">
<h2>Myeloid lineage cluster</h2>
<div id="visualization-of-myeloid-lineage-cluster-img-myeloid12345" class="section level4">
<h4>Visualization of Myeloid lineage cluster (iMg, Myeloid1/2/3/4/5)</h4>
<p>The plot below visualizes that iMg and CSF/Blood Myeloid cells cluster together, independent of their infection status.</p>
<pre class="r"><code>subset_monocytes &lt;- subset(combined, cell_type %in% c(&quot;Myeloid1&quot;, 
                                                            &quot;Myeloid2&quot;, 
                                                            &quot;Myeloid3&quot;, 
                                                            &quot;Myeloid4&quot;, 
                                                            &quot;Myeloid5&quot;, 
                                                            &quot;iMg&quot;))

p1 &lt;- DimPlot(subset_monocytes, group.by = &quot;cell_type&quot;) +
  labs(title = &quot;Cell type&quot;)
p2 &lt;- DimPlot(subset_monocytes, group.by = &quot;orig.ident&quot;) +
  labs(title = &quot;Sample origin&quot;)
p3 &lt;- DimPlot(subset_monocytes, group.by = &quot;seurat_clusters&quot;) +
  labs(title = &quot;Cluster&quot;)
fig &lt;- ggarrange(p1, p2, p3, ncol = 3, nrow = 1)
annotate_figure(fig, top = text_grob(&quot;Myeloid-lineage cells&quot;, size = 20))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-22-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-22-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="visualization-of-infected-efz-img-infected-myeloid12345" class="section level4">
<h4>Visualization of infected + EFZ iMg &amp; infected Myeloid1/2/3/4/5</h4>
<p>As expected the CSF/Blood Myeloid cells from infected patients cluster together with infected + EFZ iMg and mainly contribute to the same subcluster, as can be seen in the plot below, which shows only infected CSF/Blood Myeloid cells and the infected + EFZ iMg. Myeloid4 is an exception and belongs to a different subcluster.</p>
<pre class="r"><code>
subset_infected &lt;- subset(combined, orig.ident %in% c(&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;,
                                             &quot;HIV1_Bld&quot;, &quot;HIV2_Bld&quot;,&quot;Inf+EFZ&quot;) &amp;
                            cell_type %in% c(&quot;Myeloid1&quot;, 
                                                            &quot;Myeloid2&quot;, 
                                                            &quot;Myeloid3&quot;, 
                                                            &quot;Myeloid4&quot;, 
                                                            &quot;Myeloid5&quot;, 
                                                            &quot;iMg&quot;))


p1 &lt;- DimPlot(subset_infected, group.by = &quot;cell_type&quot;) +
  labs(title = &quot;Cell Type&quot;)
p2 &lt;- DimPlot(subset_infected, group.by = &quot;orig.ident&quot;) +
  labs(title = &quot;Sample origin&quot;)
p3 &lt;- DimPlot(subset_infected, group.by = &quot;seurat_clusters&quot;) +
  labs(title = &quot;Clusters&quot;)
fig &lt;- ggarrange(p1, p2, p3, ncol = 3, nrow = 1)
annotate_figure(fig, top = text_grob(&quot;Infected CSF/Blood Myeloid cells &amp; Inf+EFZ iMg&quot;))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-23-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-23-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cell-types-and-conditions-partially-separate-according-to-the-two-img-clusters" class="section level4">
<h4>Cell types and conditions partially separate according to the two iMg clusters</h4>
<p>The myeloid-lineage cluster separated into two distinct subclusters. Interestingly, the infected + EFZ iMg separated clearly from the infected iMg. The separation was not so clear for the uninfected +/- EFZ cells, which contributed to both subclusters. Another observations is that the infected Myeloid1/2/3/5 cells belong to the same subcluster as the infected + EFZ iMg. Conversely, the Myeloid4 cells cluster with the infected iMg. Since the infected CSF/Blood Myeloid cells originate from HIV infected and treated patients we would expect them to cluster togehter with the infected + EFZ iMg. However, there might also be infected CSF/Blood Myeloid cells which are not affected by the treatment. The Myeloid4 cells which are from Blood samples might consitute an infected cell population unaffected by the ART treatment.</p>
<pre class="r"><code>
DimPlot(subset_monocytes, group.by = &quot;seurat_clusters&quot;, 
        split.by = &quot;orig.ident&quot;, ncol = 4) +
  labs(title = &quot;Contribution of different samples to different subclusters&quot;)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-24-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-24-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(subset_monocytes, group.by = &quot;seurat_clusters&quot;, 
        split.by = &quot;cell_type&quot;, ncol = 3) +
  labs(title = &quot;Contribution of different cell types to different subclusters&quot;)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-24-2.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-2">
Past versions of unnamed-chunk-24-2.png
</button>
</p>
<div id="fig-unnamed-chunk-24-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-24-2.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="differential-gene-expression" class="section level1">
<h1>Differential gene expression</h1>
<div id="between-infection-conditions" class="section level3">
<h3>Between infection conditions</h3>
<p>In the following I will compare gene expression between infected cells and uninfected cells, both treated with EFZ and not treated. I will use the uninfected cells as a baseline to find genes which are upregulated or downregulated in infected cells compared to uninfected cells. Since we have both cells treated with EFZ and cells not treated with EFZ, we can define a signature for both treatment conditions separately and then see if there is an overlap between the gene signatures of infected cells +/- EFZ. Finding genes which overlap independently of treamtent condition increases the confidence in the results</p>
<details>
<summary>
<strong>List of top up- or downregulated genes in Inf+EFZ vs Uninf+EFZ</strong>
</summary>
<pre class="r"><code># isolate iMg cells
microglia &lt;- subset(combined, cell_type == &quot;iMg&quot;)

# only iMg cells to identify potential signatures between infected vs uninfected
Idents(microglia) =  &quot;orig.ident&quot;

# find marker genes between Uninf+EFZ and Inf+EFZ
inf_uninf_art_marker &lt;- FindMarkers(microglia, ident.1 = &quot;Inf+EFZ&quot;, 
                                    ident.2 = &quot;Uninf+EFZ&quot;, verbose = FALSE)
#upregulated genes
up_inf_art &lt;- inf_uninf_art_marker %&gt;% 
  arrange(desc(avg_log2FC)) %&gt;%
  head(20) %&gt;% 
  rownames_to_column(&quot;gene&quot;)
up_inf_art  %&gt;% head %&gt;%  kable(caption = &quot;Genes upregulated in Inf+EFZ vs. Uninf+EFZ&quot;)</code></pre>
<table>
<caption>Genes upregulated in Inf+EFZ vs. Uninf+EFZ</caption>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="right">p_val</th>
<th align="right">avg_log2FC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">RAB32</td>
<td align="right">0</td>
<td align="right">1.2840817</td>
<td align="right">0.656</td>
<td align="right">0.291</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">NUPR1</td>
<td align="right">0</td>
<td align="right">1.0569999</td>
<td align="right">0.680</td>
<td align="right">0.460</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">HMOX1</td>
<td align="right">0</td>
<td align="right">0.9031207</td>
<td align="right">0.597</td>
<td align="right">0.309</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">PDK4</td>
<td align="right">0</td>
<td align="right">0.8673561</td>
<td align="right">0.204</td>
<td align="right">0.019</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">SLC38A2</td>
<td align="right">0</td>
<td align="right">0.8452040</td>
<td align="right">0.288</td>
<td align="right">0.079</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">LPAR6</td>
<td align="right">0</td>
<td align="right">0.8045929</td>
<td align="right">0.579</td>
<td align="right">0.289</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code># downregulated genes
down_inf_art &lt;- inf_uninf_art_marker %&gt;% 
  arrange(avg_log2FC) %&gt;% 
  head(20) %&gt;%
  rownames_to_column(&quot;gene&quot;)
down_inf_art %&gt;% head %&gt;% kable(caption = &quot;Genes downregulated in Inf+EFZ vs. Uninf+EFZ&quot;)</code></pre>
<table>
<caption>Genes downregulated in Inf+EFZ vs. Uninf+EFZ</caption>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="right">p_val</th>
<th align="right">avg_log2FC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">MT1X</td>
<td align="right">0</td>
<td align="right">-3.413896</td>
<td align="right">0.147</td>
<td align="right">0.735</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">MT2A</td>
<td align="right">0</td>
<td align="right">-3.113559</td>
<td align="right">0.167</td>
<td align="right">0.695</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">CCL4</td>
<td align="right">0</td>
<td align="right">-3.102918</td>
<td align="right">0.321</td>
<td align="right">0.910</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">CCL4L2</td>
<td align="right">0</td>
<td align="right">-2.676517</td>
<td align="right">0.235</td>
<td align="right">0.816</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">CCL3L1</td>
<td align="right">0</td>
<td align="right">-2.579091</td>
<td align="right">0.290</td>
<td align="right">0.807</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">CCL3</td>
<td align="right">0</td>
<td align="right">-2.560332</td>
<td align="right">0.611</td>
<td align="right">0.982</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<hr />
</details>
<pre class="r"><code># find marker genes between Inf and Uninf
inf_uninf_marker &lt;- FindMarkers(microglia, ident.1 = &quot;Inf&quot;, 
                                ident.2 = &quot;Uninf&quot;, verbose = FALSE)

# upregulated in Inf vs Uninf, no treatment
up_inf_no &lt;- inf_uninf_marker %&gt;% 
  arrange(desc(avg_log2FC)) %&gt;%
  head(20) %&gt;% 
  rownames_to_column(&quot;gene&quot;)
#up_inf_no  %&gt;% kable()


down_inf_no &lt;- inf_uninf_marker %&gt;% 
  arrange(avg_log2FC) %&gt;% 
  head(20) %&gt;%
  rownames_to_column(&quot;gene&quot;)
#down_inf_no%&gt;% kable()</code></pre>
<p>Only 3% of genes which are upregulated and only 14% of genes which are downregulated in infected vs. uninfected cells overlap between the treatment conditions. This means, that the gene expression response defining the infection might differ between cells with and without treatment. This reflects the findings of <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span> that the EFZ treatment leads to a distinct immune response of iMg and distinct activation of two inflammatory pathways, RhoGDI and CD40 pathways as will be described later on. Since there is hardly any overlap we will use the genes upregulated in Inf+EFZ compared to Uninf+EFZ in the following analysis steps, because they probably resemble the CSF/Blood samples, which originate from patients treated with ART, best.</p>
<pre class="r"><code>up_inf &lt;- intersect(up_inf_art$gene, up_inf_no$gene)
down_inf &lt;- intersect(down_inf_art$gene, down_inf_no$gene)
print(c(up_inf, down_inf))
[1] &quot;HMOX1&quot;  &quot;MT1X&quot;   &quot;MT2A&quot;   &quot;CCL4L2&quot; &quot;RND3&quot;   &quot;H1F0&quot;  </code></pre>
<pre class="r"><code>p1 &lt;- ggVennDiagram::ggVennDiagram(list(up_inf_art$gene, up_inf_no$gene),
              label_alpha = 0,
              category.names= c(&quot;Inf+EFZ&quot;, &quot;Inf&quot;),
              #set.size = 20,
              label = &quot;percent&quot;,
              label_size = 6) +
   scale_fill_distiller(palette = &quot;RdBu&quot;) +
  labs(title = &quot;Genes upregulated in infected cells&quot;)

p2 &lt;- ggVennDiagram::ggVennDiagram(list(down_inf_art$gene, down_inf_no$gene),
              label_alpha = 0,
              category.names= c(&quot;Inf+EFZ&quot;, &quot;Inf&quot;),
              #set.size = 20,
              label = &quot;percent&quot;,
              label_size = 6) +
  scale_fill_distiller(palette = &quot;RdBu&quot;) +
  labs(title = &quot;Genes downregulated in infected cells&quot;)

ggarrange(p1, p2, ncol = 2)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-29-1.png" width="1440" style="display: block; margin: auto;" /></p>
<details>
<summary>
<strong>Volcano Plot of Infection (+ EFZ) signature</strong>
</summary>
<p>Interestingly, the genes downregulatd in Inf+EFZ vs. Uninfected cells show a higher average log fold change with higher significance (p-values).</p>
<pre class="r"><code>inf_uninf_art_marker %&gt;% 
  rownames_to_column(&quot;gene&quot;) %&gt;%
  filter(gene %in% c(up_inf_art$gene, down_inf_art$gene)) %&gt;% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = case_when(gene %in% up_inf_art$gene ~ &quot;upregulated&quot;,
                               gene %in% down_inf_art$gene ~ &quot;downregulated&quot;,
                               TRUE ~ &quot;NA&quot;)) %&gt;%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  mutate(label = ifelse(gene %in% c(up_inf_art$gene, down_inf_art$gene), gene, NA)) %&gt;% 
  ggplot(aes(x = avg_log2FC,
             y = -log10(p_val_adj),
             col = condition, 
             label = label)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-50), col = &quot;red&quot;) +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = &quot;Genes which are up- or downregulated in Inf+EFZ vs Uninf+EFZ&quot;)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-30-1.png" width="1152" style="display: block; margin: auto;" /></p>
<hr />
</details>
</div>
<div id="between-treatment-conditions" class="section level3">
<h3>Between treatment conditions</h3>
<p>Comparing uninfected + EFZ with uninfected and infected + EFZ with infected iMg it might be possible to identify genes which are different after treatment of cells with EFZ. Finding overlapping genes between the two infection states increases the confidence in the results, because the identified genes are associated with treatment independently of infection status.</p>
<p>The diagram below shows, that 36% of genes are upregulated both in infected and infected + EFZ iMg compared to uninfected and uninfected + EFZ iMg. These genes are probably associated with the EFZ treatment. The same is true for 44% of the downregulated genes. The drug affects the iMg and changes their gene expression. This signature might be used to identify Myeloid2 CSF cells which are affected by the treatment. Since ART drugs are found at lower concentrations in CSF than in blood, probably only a subgroup of Myeloid2 are affected by the ART drugs. It would be interesting to know which are affected by treatment and which are not. However, this comparison might be farfetched, since the patients from which the CSF samples have been derived have not been treated with EFZ, but with a combination of other ART drugs which probably have a slightly different effect on cells.</p>
<pre class="r"><code>treatment_marker_uninf &lt;- FindMarkers(microglia, ident.1 = &quot;Uninf+EFZ&quot;, 
                                ident.2 = &quot;Uninf&quot;, verbose = FALSE)

# find marker genes between Inf+EFZ and Inf
treatment_marker_inf &lt;- FindMarkers(microglia, ident.1 = &quot;Inf+EFZ&quot;, ident.2 = &quot;Inf&quot;)

# overlap between upregulated genes
treatment_up &lt;- intersect((treatment_marker_inf %&gt;% 
             rownames_to_column(&quot;gene&quot;) %&gt;% 
             filter(avg_log2FC &gt; 0))$gene, 
           (treatment_marker_uninf %&gt;% 
             rownames_to_column(&quot;gene&quot;) %&gt;% 
             filter(avg_log2FC &gt; 0))$gene)

treatment_down &lt;- intersect((treatment_marker_inf %&gt;% 
             rownames_to_column(&quot;gene&quot;) %&gt;% 
             filter(avg_log2FC &lt; 0))$gene, 
           (treatment_marker_uninf %&gt;% 
             rownames_to_column(&quot;gene&quot;) %&gt;% 
             filter(avg_log2FC &lt; 0))$gene)
treatment_marker &lt;- c(treatment_up, treatment_down)</code></pre>
<pre class="r"><code>p1 &lt;- ggVennDiagram::ggVennDiagram(list((treatment_marker_inf %&gt;% 
             rownames_to_column(&quot;gene&quot;) %&gt;% 
             filter(avg_log2FC &gt; 0))$gene, 
             (treatment_marker_uninf %&gt;% 
             rownames_to_column(&quot;gene&quot;) %&gt;% 
             filter(avg_log2FC &gt; 0))$gene),
              label_alpha = 0,
              category.names= c(&quot;Inf+EFZ&quot;, &quot;Uninf+EFZ&quot;),
              #set.size = 20,
              label = &quot;percent&quot;,
              label_size = 6) +
   scale_fill_distiller(palette = &quot;RdBu&quot;) +
  labs(title = &quot;Genes upregulated in cells treated with EFZ&quot;)

p2 &lt;- ggVennDiagram::ggVennDiagram(list((treatment_marker_inf %&gt;% 
             rownames_to_column(&quot;gene&quot;) %&gt;% 
             filter(avg_log2FC &lt; 0))$gene, 
             (treatment_marker_uninf %&gt;% 
             rownames_to_column(&quot;gene&quot;) %&gt;% 
             filter(avg_log2FC &lt; 0))$gene),
              label_alpha = 0,
              category.names= c(&quot;Inf+EFZ&quot;, &quot;Uninf+EFZ&quot;),
              #set.size = 20,
              label = &quot;percent&quot;,
              label_size = 6,
              set_color = c(&quot;red&quot;, &quot;blue&quot;)) +
  scale_fill_distiller(palette = &quot;RdBu&quot;) +
  labs(title = &quot;Genes downregulated in cells treated with EFZ&quot;)

ggarrange(p1, p2, ncol = 2)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-32-1.png" width="1440" style="display: block; margin: auto;" /></p>
<details>
<summary>
<strong>Volcano Plot of Treatment signature</strong>
</summary>
<pre class="r"><code>treatment_marker_uninf %&gt;% 
  rownames_to_column(&quot;gene&quot;) %&gt;%
  filter(gene %in% treatment_marker) %&gt;% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = case_when(gene %in% treatment_up ~ &quot;upregulated&quot;,
                               gene %in% treatment_down ~ &quot;downregulated&quot;,
                               TRUE ~ &quot;NA&quot;)) %&gt;%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  #mutate(label = ifelse(gene %in% c(up_inf$gene, down_inf$gene), gene, NA)) %&gt;% 
  ggplot(aes(x = avg_log2FC,
             y = -log10(p_val_adj),
             col = condition, 
             label = gene)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-10), col = &quot;red&quot;) +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = &quot;Top 20 Genes which are up/downregulated in treatment vs. no treatment, Uninf cells&quot;)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-33-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-33-1">
Past versions of unnamed-chunk-33-1.png
</button>
</p>
<div id="fig-unnamed-chunk-33-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-33-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</details>
</div>
</div>
<div id="infection-treatment-gene-signature" class="section level1">
<h1>Infection &amp; treatment gene signature</h1>
<p>Checking the differentially expressed genes and trying to identify an infection signature through literature search turned out to be very difficult. Therefore, a different approach was used. The top twenty up- and downregulated genes in Inf+EFZ vs Uninf+EFZ were computed. Using these genes as an “infection signature” it will be investigated whether some of the Myeloid2 cells from the CSF samples show a similar signature.</p>
<pre class="r"><code>hivmy2 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv_myeloid2_subclusters&quot;)
# select only genes which are also found in the sc-RNAseq dataset
up_inf_filt &lt;- up_inf_art$gene[up_inf_art$gene %in% rownames(hivmy2@assays$RNA@counts)]
down_inf_filt &lt;- down_inf_art$gene[down_inf_art$gene %in% rownames(hivmy2@assays$RNA@counts)]</code></pre>
<div id="gene-signature-scoring-in-myeloid2-cells" class="section level2">
<h2>Gene signature scoring in Myeloid2 cells</h2>
<p>Using the signature genes from differential gene expression analysis between Inf+EFZ and Uninf+EFZ we can compute a score of activity of these gene sets in the Myeloid2 cluster of CSF samples. We can plot these scores in a t-SNE representation, but we can also get the scores from the metadata and use them for other visualizations. If the Inf+EFZ signature can be found in some of the Myeloid2 cells from the CSF samples this might provide additional insights about the infection status of these microglia-like cells.</p>
<div id="scores-of-gene-signature-up--or-downregulated-in-infefz-img-in-myeloid2-cells" class="section level4">
<h4>Scores of gene signature up- or downregulated in Inf+EFZ iMg in Myeloid2 cells</h4>
<p>In the plot below (top) there are some cells visible which have a high score for the signature composed of genes upregulated in Inf+EFZ iMg To be precise, 63% percent of the Myeloid2 cells have a positive score (92 out of 145 cells). These high scoring cells do, however, not correspond to the two clusters (c1, c2) obtained from analysis of TF activity (SCENIC).</p>
<p>In the bottom part of the plot below there are some cells with a negative score for the downregualated gene signature. To be precise, 58% of of all cells (85 out of 146) have a negative score. It would be interesting to know if some of these cells are the same as those with a high score for the upregulated genes. Identifying cells in which both the upregulated gene signature and the downregulated gene signature have a positive or negative score respectively increases the confidence in the results. 51 cells have both a positive score for the upregulated genes and a negative score for the downregulated genes. Cells exhibiting this specific pattern of gene expression are likely infected cells. Since the infection status of the cells in the CSF sample is unknown, this provides valuable new insights.</p>
<pre class="r"><code>up_sign &lt;- AddModuleScore(hivmy2,
                           features = list(up_inf_filt),
                          name = &quot;upregulated&quot;)

p1 &lt;- FeaturePlot(subset(up_sign, ht_cluster == &quot;c1&quot;),
                  features = &quot;upregulated1&quot;, pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Myeloid2 Cluster c1&quot;)

p2 &lt;- FeaturePlot(subset(up_sign, ht_cluster == &quot;c2&quot;),
                  features = &quot;upregulated1&quot;, pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Myeloid2 Cluster c2&quot;)

figure &lt;- ggarrange(p1, p2, ncol = 2)
annotate_figure(figure, top = text_grob(&quot;Signature score of genes upregulated in Inf+EFZ iMg applied to CSF Myeloid2 cells&quot;, 
               color = &quot;black&quot;, face = &quot;bold&quot;, size = 20))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-35-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>
down_sign &lt;- AddModuleScore(hivmy2,
                           features = list(down_inf_filt), name = &quot;downregulated&quot;)

p1 &lt;- FeaturePlot(subset(down_sign, ht_cluster == &quot;c1&quot;),
                  features = &quot;downregulated1&quot;, pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Myeloid2 Cluster c1&quot;)

p2 &lt;- FeaturePlot(subset(down_sign, ht_cluster == &quot;c2&quot;),
                  features = &quot;downregulated1&quot;, pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Myeloid2 Cluster c2&quot;)

figure &lt;- ggarrange(p1, p2, ncol = 2)
annotate_figure(figure, top = text_grob(&quot;Signature score of genes downregulated in Inf+EFZ iMg applied to CSF Myeloid2 cells&quot;, 
               color = &quot;black&quot;, face = &quot;bold&quot;, size = 20))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-36-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-36-1">
Past versions of unnamed-chunk-36-1.png
</button>
</p>
<div id="fig-unnamed-chunk-36-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-36-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># divide the number of rows(cells) with negative scores by the number of total 
# rows(cells) to get the percentage of cells with negative scores.
percentage_inf &lt;- dim(down_sign@meta.data %&gt;% filter(downregulated1 &lt; 0))[1] / 
  dim(down_sign@meta.data[&quot;downregulated1&quot;])[1]

# extract list of cells with positive/negative scores and find the overlap
cells_down &lt;- rownames(down_sign@meta.data %&gt;% filter(downregulated1 &lt; 0))
cells_up &lt;- rownames(up_sign@meta.data %&gt;% filter(upregulated1 &gt; 0))
inf_efz_cells &lt;- intersect(cells_up, cells_down)</code></pre>
<details>
<summary>
<strong>Defining a signature composed of downregulated genes</strong>
</summary>
<p>Defining a gene signature composed of both up- and downregulated genes turned out to be more difficult than expected. The idea was to separately investigate upregulated versus downregulated genes. Above the upregulated and downregulated gene signature scores in Myeloid2 cells were shown.</p>
<p>We will now have a closer look at the problems that arise when investigating downregulated genes. Assuming that our Myeloid2 cells are composed of two distinct populations, one infected and one uninfected, we would expect a bimodal distribution of gene expression for genes upregulated in the infected condition. There should be one peak with base level gene expression corresponding to uninfected cells and a second peak with higher levels of gene expression corresponding to infected cells.</p>
<p>In the histograms below the distribution of the sum of log-normalized feature counts of genes upregulated or downregulated respectively can be seen. For the upregulated genes (green) a bimodal distribution is not evident from the histogram below. No clear conclusions can be drawn, since the number of cells is probably too low.</p>
<p>For the downregulated genes (red) there seem to be a large number of cells which do not express these genes at all. This group of cells could include infected cells, however this is a dangerous assumption, since in general there might be cells with low or none expression of these genes. There is no way to know if these cells are actually infected cells.</p>
<pre class="r"><code>#normalize counts of upregulated genes
counts_up &lt;- (hivmy2@assays$RNA@data)
counts_up &lt;- t(t(counts_up)) / colSums(counts_up)
counts_up &lt;- log1p(counts_up)
counts_up &lt;- counts_up[up_inf_filt, ]

# compute average expression of the 20 signature genes
p1 &lt;- as.data.frame(rowSums(t(counts_up))) %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  rename(&quot;count&quot; = &quot;rowSums(t(counts_up))&quot;) %&gt;% 
  arrange(desc(count)) %&gt;% 
  ggplot(aes(x  = count)) +
  geom_histogram(bins = 70, color = &quot;olivedrab4&quot;, fill = &quot;olivedrab4&quot;) +
  xlab(&quot;normalized feature counts&quot;) +
  ylab(&quot;number of cells&quot;) +
  ylim(0,100) +
  labs(title = &quot;Sum of feature counts of genes upregulated in Inf+EFZ vs. Uninf+EFZ&quot;)

#normalize counts of downregulated genes
counts_down &lt;- (hivmy2@assays$RNA@data)
counts_down &lt;- t(t(counts_down)) / colSums(counts_down)
counts_down &lt;- log1p(counts_down)
counts_down &lt;- counts_down[down_inf_filt, ]

# compute average expression of the 20 signature genes
p2 &lt;- as.data.frame(rowSums(t(counts_down))) %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  rename(&quot;count&quot; = &quot;rowSums(t(counts_down))&quot;) %&gt;% 
  arrange(desc(count)) %&gt;% 
  ggplot(aes(x  = count)) +
  geom_histogram(bins = 70, color = &quot;red4&quot;, fill = &quot;red4&quot;) +
  xlab(&quot;normalized feature counts&quot;) +
  ylab(&quot;number of cells&quot;) +
  labs(title = &quot;Sum of feature count of genes downregulated in Inf+EFZ vs. Uninf+EFZ&quot;)</code></pre>
<pre class="r"><code>p1 + p2</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-39-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-39-1">
Past versions of unnamed-chunk-39-1.png
</button>
</p>
<div id="fig-unnamed-chunk-39-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-39-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</details>
</div>
<div id="scores-of-gene-signature-up--or-downregulated-in-cells-treated-with-efz-independently-of-infection-condition." class="section level4">
<h4>Scores of gene signature up- or downregulated in cells treated with EFZ independently of infection condition.</h4>
<p>Of the Myeloid2 cells 70% (103 out of 146 cells) have a score above 0.2 for the upregulated gene signature (see figure below, top). Only 26% of cells have a negative score (below 0) for the downregulated gene signature (see figure below, bottom). Again, to increase the confidence in the results, the overlap between the cells with up/downregulated gene signatures was identified. Only 15 cells have both a positive and negative score for the up- and downregulated signature respectively.</p>
<pre class="r"><code>up_treat &lt;- AddModuleScore(hivmy2,
                           features = list(treatment_up),
                           name = &quot;upregulated&quot;)

p1 &lt;- FeaturePlot(subset(up_treat, ht_cluster == &quot;c1&quot;),
                  features = &quot;upregulated1&quot;, pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Myeloid2 Cluster c1&quot;)

p2 &lt;- FeaturePlot(subset(up_treat, ht_cluster == &quot;c2&quot;),
                  features = &quot;upregulated1&quot;, pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Myeloid2 Cluster c2&quot;)

figure &lt;- ggarrange(p1, p2, ncol = 2)
annotate_figure(figure, top = text_grob(&quot;Signature score of genes upregulated in EFZ treatment&quot;, 
               color = &quot;black&quot;, face = &quot;bold&quot;, size = 20))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-40-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>down_treat &lt;- AddModuleScore(hivmy2,
                           features = list(treatment_down),
                           name = &quot;downregulated&quot;)

p1 &lt;- FeaturePlot(subset(down_treat, ht_cluster == &quot;c1&quot;),
                  features = &quot;downregulated1&quot;, pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Myeloid2 Cluster c1&quot;)

p2 &lt;- FeaturePlot(subset(down_treat, ht_cluster == &quot;c2&quot;),
                  features = &quot;downregulated1&quot;, pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = &quot;RdBu&quot;))) +
  labs(title = &quot;Myeloid2 Cluster c2&quot;)

figure &lt;- ggarrange(p1, p2, ncol = 2)
annotate_figure(figure, top = text_grob(&quot;Signature score of genes downregulated in EFZ treatment&quot;, 
               color = &quot;black&quot;, face = &quot;bold&quot;, size = 20))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-41-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-41-1">
Past versions of unnamed-chunk-41-1.png
</button>
</p>
<div id="fig-unnamed-chunk-41-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-41-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>percentage_up &lt;- dim(up_treat@meta.data %&gt;% filter(upregulated1 &gt; 0.2))[1] /
  dim(up_treat@meta.data)[1]

percentage_down &lt;- dim(down_treat@meta.data %&gt;% filter(downregulated1 &lt; 0))[1] /
  dim(down_treat@meta.data)[1]

cells_down &lt;- rownames(down_treat@meta.data %&gt;% filter(downregulated1 &lt; 0))
cells_up &lt;- rownames(up_treat@meta.data %&gt;% filter(upregulated1 &gt; 0.2))
treat_cells &lt;- intersect(cells_down, cells_up)</code></pre>
</div>
</div>
<div id="overlay-infefz-signature-with-the-tf-activity-heatmap-of-myeloid2-cells" class="section level2 tabset">
<h2 class="tabset">Overlay Inf+EFZ signature with the TF activity heatmap of Myeloid2 cells</h2>
<p>There are a some cells which express the Inf+EFZ signature and they are highlighted with an orange colour in the heatmap below. However, they are spread across all clusters and patients. The first of three clusters in each patient was hypothesized to be a latently infected group of cells. In the heatmap below the cells with an Inf+EFZ signature do not seem to accumulate in the first cluster of each patient, but rather are more frequent in the other two clusters.</p>
<div id="k-3-hierarchical-clustering" class="section level3">
<h3>k = 3 hierarchical clustering</h3>
<pre class="r"><code># read in Seurat object contianing alls samples and conditions
hiv7 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds&quot;)
# import TF activity matrix of  Myeloid2 cells
auc_mtx &lt;- read.table(&quot;data_files/aucell_matrix_10k_hvg.tsv&quot;, sep = &quot;\t&quot;)


# 3 heatmaps with all Scenic TFs
ht_list &lt;- map(seq.int(1:3), function(n) {
  patients &lt;- c (&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;) # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df),]
  # heatmap
  ht &lt;- Heatmap(t(mtx), 
               column_title = paste0(&quot;patient: &quot;, patients[n]), 
               top_annotation = HeatmapAnnotation(signature = case_when(
                 rownames(mtx) %in% inf_efz_cells ~ &quot;Inf_EFZ_signature&quot;,
                 #rownames(mtx) %in% uninf_efz_cells ~ &quot;Uninf_EFZ_signature&quot;,
                 #rownames(mtx) %in% uninf_cells ~ &quot;Uninf_signature&quot;,
                 TRUE ~ &quot;no_signature&quot;),
                 name = &quot;signature&quot;,
                 col = list(signature = c(&quot;Inf_EFZ_signature&quot; = &quot;orange2&quot;,
                                          #&quot;Uninf_EFZ_signature&quot; = &quot;lightblue&quot;,
                                          #&quot;Uninf_signature&quot; = &quot;orange&quot;,
                                          &quot;no_signature&quot; = &quot;navy&quot;))),
               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
               show_row_names = TRUE
               ) 
               #top_annotation = column_annotation) 
  list(name = patients[n], heatmap = ht)
})

ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap </code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-43-1.png" width="1440" style="display: block; margin: auto;" /></p>
<hr />
</div>
<div id="grouped-by-infection-condition" class="section level3">
<h3>Grouped by infection condition</h3>
<p>When you group the cells according to their different gene signatures as in the heatmap below, you do not see a difference between the two groups. Testing for TFs with different activity between the two groups might provide more insights.</p>
<pre class="r"><code># filter the dataset for only Myeloid2 cells 
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                orig.ident %in% c (&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;))
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df),]

# add signature column to dataframe
signature &lt;- df %&gt;%
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  mutate(signature = case_when(cell %in% inf_efz_cells ~ &quot;Inf_signature&quot;,
                                #cell %in% uninf_efz_cells ~&quot;Uninf_EFZ_signature&quot;,
                                #cell %in% uninf_cells ~ &quot;Uninf_signature&quot;,
                               TRUE ~ &quot;no_signature&quot;))


ht &lt;- draw(Heatmap(t(mtx), 
             column_title = &quot;All patients split according to gene signature scores&quot;, 
             top_annotation = HeatmapAnnotation(signature = case_when(
               rownames(mtx) %in% inf_efz_cells ~ &quot;Inf_signature&quot;,
               #rownames(mtx) %in% uninf_efz_cells ~ &quot;Uninf_EFZ_signature&quot;,
               #rownames(mtx) %in% uninf_cells ~ &quot;Uninf_signature&quot;,
               TRUE ~ &quot;no_signature&quot;),
               name = &quot;signature&quot;,
               col = list(signature = c(&quot;Inf_signature&quot; = &quot;orange2&quot;,
                                        #&quot;Uninf_EFZ_signature&quot; = &quot;lightblue&quot;,
                                        #&quot;Uninf_signature&quot; = &quot;orange&quot;,
                                        &quot;no_signature&quot; = &quot;navy&quot;))),       
             column_split = signature$signature,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             show_row_names = TRUE
             ))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-44-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-44-1">
Past versions of unnamed-chunk-44-1.png
</button>
</p>
<div id="fig-unnamed-chunk-44-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-44-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
</div>
<div id="overlay-efz-treatment-signature-with-the-tf-activity-heatmap-of-myeloid2-cells" class="section level2 tabset">
<h2 class="tabset">Overlay EFZ treatment signature with the TF activity heatmap of Myeloid2 cells</h2>
<p>There are very few cells with both a positive and negative score for up- and downregulated signatures respectively. This might be, because the comparison between the effect of EFZ treatment on cells in cell culture and cells from the CSF of patients treated with a combination of other antiretroviral drugs might be farfetched. Apart from the drugs being different, cells in cell culture are exposed to comparatively high drug concentrations, while cells in the CSF are exposed to very low drug concentrations due to the blood-brain barrier. This could explain why so few cells with a EFZ treatment signature are found among the Myeloid2 cells.</p>
<div id="k-3-hierarchical-clustering-1" class="section level3">
<h3>k = 3 hierarchical clustering</h3>
<pre class="r"><code>
# 3 heatmaps with all Scenic TFs
ht_list &lt;- map(seq.int(1:3), function(n) {
  patients &lt;- c (&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;) # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df),]
  # heatmap
  ht &lt;- Heatmap(t(mtx), 
               column_title = paste0(&quot;patient: &quot;, patients[n]), 
               top_annotation = HeatmapAnnotation(signature = case_when(
                 rownames(mtx) %in% treat_cells ~ &quot;EFZ_treatment_signature&quot;,
                 #rownames(mtx) %in% uninf_efz_cells ~ &quot;Uninf_EFZ_signature&quot;,
                 #rownames(mtx) %in% uninf_cells ~ &quot;Uninf_signature&quot;,
                 TRUE ~ &quot;no_signature&quot;),
                 name = &quot;signature&quot;,
                 col = list(signature = c(&quot;EFZ_treatment_signature&quot; = &quot;olivedrab3&quot;,
                                          #&quot;Uninf_EFZ_signature&quot; = &quot;lightblue&quot;,
                                          #&quot;Uninf_signature&quot; = &quot;orange&quot;,
                                          &quot;no_signature&quot; = &quot;navy&quot;))),
               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
               show_row_names = TRUE
               ) 
               #top_annotation = column_annotation) 
  list(name = patients[n], heatmap = ht)
})

ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap </code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-45-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-45-1">
Past versions of unnamed-chunk-45-1.png
</button>
</p>
<div id="fig-unnamed-chunk-45-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-45-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
<div id="grouped-by-treatment-condition" class="section level3">
<h3>Grouped by treatment condition</h3>
<pre class="r"><code># filter the dataset for only Myeloid2 cells 
df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                orig.ident %in% c (&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;))
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx &lt;- auc_mtx[rownames(auc_mtx) %in% rownames(df),]

# add signature column to dataframe
signature &lt;- df %&gt;%
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  mutate(signature = case_when(cell %in% treat_cells ~ &quot;EFZ_treatment_signature&quot;,
                                #cell %in% uninf_efz_cells ~&quot;Uninf_EFZ_signature&quot;,
                                #cell %in% uninf_cells ~ &quot;Uninf_signature&quot;,
                               TRUE ~ &quot;no_signature&quot;))


ht &lt;- draw(Heatmap(t(mtx), 
             column_title = &quot;All patients split according to gene signature scores&quot;, 
             top_annotation = HeatmapAnnotation(signature = case_when(
               rownames(mtx) %in% treat_cells ~ &quot;EFZ_treatment_signature&quot;,
               #rownames(mtx) %in% uninf_efz_cells ~ &quot;Uninf_EFZ_signature&quot;,
               #rownames(mtx) %in% uninf_cells ~ &quot;Uninf_signature&quot;,
               TRUE ~ &quot;no_signature&quot;),
               name = &quot;signature&quot;,
               col = list(signature = c(&quot;EFZ_treatment_signature&quot; = &quot;olivedrab3&quot;,
                                        #&quot;Uninf_EFZ_signature&quot; = &quot;lightblue&quot;,
                                        #&quot;Uninf_signature&quot; = &quot;orange&quot;,
                                        &quot;no_signature&quot; = &quot;navy&quot;))),       
             column_split = signature$signature,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
             show_row_names = TRUE
             ))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-46-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-46-1">
Past versions of unnamed-chunk-46-1.png
</button>
</p>
<div id="fig-unnamed-chunk-46-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-46-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
</div>
</div>
<div id="differential-tf-activity" class="section level1 tabset">
<h1 class="tabset">Differential TF activity</h1>
<p>After having identified a few cells with a high scores for the infection signature (Inf+EFZ), as well as cells with a high score for the EFZ treatment signatures, it would be interesting to know whether some TFs have different activity between the conditions. For this a Kruskal-Wallis Test was used, followed by a pairwise Wilcox Test.</p>
<div id="between-infection-conditions-1" class="section level2">
<h2>Between infection conditions</h2>
<p>Out of all 125 TFs, 44 have different activity between cells with a high score for the Inf+EFZ signature compared to the rest. The activities for those TFs which are also found to have a differential binding scores in C20 microglia are shown in the boxplots below. KLF4 and REST are the only two TFs more bound in latently infected C20 microglia. KLF4 is more active in Myeloid2 with a high score for the Inf+EFZ signature, but REST is less active. All other TFs are more bound in uninfected C20 microglia than in latently infected, but the TF activity is not always higher in uninfected Myeloid2 (no high score for Inf+EFZ signature).</p>
<pre class="r"><code>hivmy2_scenic &lt;- AddMetaData(hivmy2, 
                    metadata = auc_mtx, 
                    col.name = colnames(auc_mtx))

inf_sign_df &lt;- hivmy2_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  mutate(signature = case_when(cell %in% inf_efz_cells ~ &quot;Inf_EFZ&quot;,
                                TRUE ~ &quot;no_sign&quot;))

# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(inf_sign_df[tf]) ~ signature, inf_sign_df))[&quot;p.value&quot;]
})

# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)

# how many transcription factors are significantly different between the clusters?
sign &lt;- adj_p_values[adj_p_values &lt; 0.05]
length(sign)
[1] 44
diff_cells &lt;- colnames(auc_mtx)[adj_p_values &lt; .05]

# get only the TFs also found in footprinting

# read in footprinting resutls
tfs_ana &lt;- read.table(&quot;/media/ag-cherrmann/projects/06_HIV_Microglia/data/atacseq/data-2020-11-06/tobias/TOBIAS_snakemake/footprint_mglia2_GlassTF_17-03/TFBS/bindetect_results.txt&quot;, sep = &quot;\t&quot;, header = TRUE)

# filter TFs with significant p values between latent infection and uninfected
# I decided to use a threshold of p-value &lt; 1e-50, because due to the large number 
# of data, all p-values will be very high
top_tfs &lt;- tfs_ana %&gt;% filter(uninf_red_pvalue &lt; 1e-50)
down_tfs &lt;- top_tfs %&gt;% filter(uninf_red_change &lt; 0)
#down_tfs &lt;- top_tfs %&gt;% filter(uninf_red_change &lt; )

diff_tfs &lt;- diff_cells[diff_cells %in% top_tfs$name]

diff_cells[diff_cells %in% down_tfs$name]
[1] &quot;KLF4&quot; &quot;REST&quot;</code></pre>
<pre class="r"><code>plots &lt;- map(diff_tfs, function(tf){
  ggplot() +
    geom_violin(aes(x = inf_sign_df %&gt;% pull(&quot;signature&quot;), 
                     y = inf_sign_df %&gt;% pull(tf),
                     fill = inf_sign_df %&gt;% pull(&quot;signature&quot;))) +
    geom_boxplot(aes(x = inf_sign_df %&gt;% pull(&quot;signature&quot;), 
                     y = inf_sign_df %&gt;% pull(tf)),
                 alpha = 0) +
    stat_pvalue_manual(inf_sign_df %&gt;% 
                         pairwise_wilcox_test(
                           as.formula(paste0(tf, &quot; ~ ht_cluster&quot;))) %&gt;% 
                         add_xy_position()) +
    xlab(&quot;condition&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;condition&quot;))
  })

do.call(grid.arrange, c(plots, ncol = 3, nrow = 4))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-48-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-48-1">
Past versions of unnamed-chunk-48-1.png
</button>
</p>
<div id="fig-unnamed-chunk-48-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-48-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
<div id="between-treatment-conditions-1" class="section level2">
<h2>Between treatment conditions</h2>
<p>Are some TFs in Myeloid2 differentially active between the treatment conditions? There are no significant differences in TF activity between the cells with high scores for EFZ treatment signature compared to cells without the signature. Therefore, no posthoc analysis was performed.</p>
<pre class="r"><code>df &lt;- hivmy2_scenic@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
        # filter(cell %in% c(inf_efz_cells, uninf_efz_cells, uninf_cells)) %&gt;% 
  mutate(signature = case_when(cell %in% treat_cells ~ &quot;treatment_signature&quot;,
                                TRUE ~ &quot;no_signature&quot;))

# check all TFs 
p_values &lt;- map(seq.int(1:125), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  p_values &lt;- unlist(kruskal.test(pull(df[tf]) ~ signature, df))[&quot;p.value&quot;]
})

# adjust for multiple testing with FDR
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)

# how many transcription factors are significantly different between the clusters?
sign &lt;- adj_p_values[adj_p_values &lt; 0.05]</code></pre>
<hr />
</div>
<div id="gene-expression-of-inflammatory-genes" class="section level2">
<h2>Gene expression of inflammatory genes</h2>
<p><span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span> showed that two pathways are specifically activated in the Inf+EFZ condition, namely RhoGDI signaling and CD40 signaling. This effect was not seen in Uninf+EFZ, so it does not seem to be caused by EFZ alone, but rather a combination of infection and EFZ treatment <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span>.</p>
<ul>
<li>RhoGDI signaling:
<ul>
<li>PP1R12A was upregulated</li>
<li>RND3, CD44, ACTG1 were downregulated</li>
</ul></li>
<li>CD40 signaling:
<ul>
<li>FOS was upregulated</li>
<li>NFKBIA, ICAM1, TFNAIP3 were downregulated</li>
</ul></li>
</ul>
<p>Here, we would like to see if cells with high scores for the Inf+EFZ signature show gene expression patterns that are in line with the findings by <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span>. The gene expression for each gene is shown in the figure below. We would expect only PP1R12A and FOS to be more highly expressed in the cells with Inf+EFZ signature, while all other genes should be less highly expressed. This pattern in gene expression is, however, not observed.</p>
<pre class="r"><code># read in Seurat object containing Myeloid2 cells
hivmy2_scenic &lt;- AddMetaData(hivmy2_scenic, metadata = inf_sign_df$signature, col.name = &quot;signature&quot;)

# RhoGDI signaling in myeloid2
p1 &lt;- VlnPlot(hivmy2_scenic, features = c(&quot;RND3&quot;, &quot;CD44&quot;, &quot;ACTG1&quot;, &quot;PPP1R12A&quot;), 
              group.by = &quot;signature&quot;, ncol = 2)

# CD40 signaling in myeloid2
p2 &lt;- VlnPlot(hivmy2_scenic, features =c(&quot;NFKBIA&quot;, &quot;ICAM1&quot;, &quot;TNFAIP3&quot;, &quot;FOS&quot;), 
              group.by = &quot;signature&quot;, ncol = 2)
Warning: The following variables were found in both object metadata and the default assay: FOS
Returning metadata; if you want the feature, please use the assay&#39;s key (eg. rna_FOS)

# RhoGDI signaling in iMg
p3 &lt;- VlnPlot(microglia, features = c(&quot;RND3&quot;, &quot;CD44&quot;, &quot;ACTG1&quot;, &quot;PPP1R12A&quot;), 
        ncol = 2, pt.size = 0, group.by = &quot;orig.ident&quot;)
Warning: Could not find ACTG1 in the default search locations, found in RNA
assay instead
Warning: Could not find PPP1R12A in the default search locations, found in RNA
assay instead

# CD40 signaling in iMg
p4 &lt;- VlnPlot(microglia, features = c(&quot;NFKBIA&quot;, &quot;ICAM1&quot;, &quot;TNFAIP3&quot;, &quot;FOS&quot;), 
        ncol = 2, pt.size = 0, group.by = &quot;orig.ident&quot;)

fig &lt;- ggarrange(p1, p3, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob(&quot;Expression of four genes of the RhoGDI signaling pathway in Myeloid2 (left column) and iMg (right column)&quot;,
                                     size = 20 , face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-50-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-50-1">
Past versions of unnamed-chunk-50-1.png
</button>
</p>
<div id="fig-unnamed-chunk-50-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-50-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
fig &lt;- ggarrange(p2, p4, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob(&quot;Expression of four genes of the CD40 signaling pathway in Myeloid2 (left column) and iMg (right column)&quot;,
                                     size = 20 , face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-50-2.png" width="1440" style="display: block; margin: auto;" /></p>
<p>Only FOS which encodes a TF is found in the TF activity matrix, so for FOS the TF acivity is shown as well. Judging from the figure below FOS seems to be more active in cells with high score for Inf+EFZ signature, but this difference is not statistically significant.</p>
<pre class="r"><code>inf_sign_df %&gt;% ggplot(aes(x = signature, y = FOS)) +
  geom_violin(aes(fill = signature, alpha = .6)) +
  geom_boxplot(alpha = 0) +
  stat_pvalue_manual(inf_sign_df %&gt;% 
                       pairwise_wilcox_test(FOS ~ signature) %&gt;% 
  add_xy_position())</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-51-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-51-1">
Past versions of unnamed-chunk-51-1.png
</button>
</p>
<div id="fig-unnamed-chunk-51-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-51-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
</div>
<div id="appendix" class="section level1">
<h1>Appendix</h1>
<div id="de-novo-clustering" class="section level2">
<h2>De-novo clustering</h2>
<p>De-novo clustering of the myeloid-lineage cluster of the integrated dataset was performed, but did not yield any additional insights. Isolating the iMg and all Myeloid cell clusters from the integrated dataset and normalizing, scaling and re-clustering these isolated cells might have<br />
provided additional insights.</p>
<details>
<summary>
Details on de-novo clustering
</summary>
<pre><code>
microglia &lt;- subset(combined, cell_type %in% c(&quot;iMg&quot;,
                                                &quot;Myeloid1&quot;,
                                                &quot;Myeloid2&quot;,
                                                &quot;Myeloid3&quot;,
                                                &quot;Myeloid4&quot;,
                                                &quot;Myeloid5&quot;))
microglia_recluster &lt;- SCTransform(microglia, verbose = FALSE)
microglia_recluster &lt;- ScaleData(microglia_recluster, verbose = FALSE)
microglia_recluster &lt;- RunPCA(microglia_recluster, verbose = FALSE)
ElbowPlot(microglia_recluster)
microglia_recluster &lt;- RunTSNE(microglia_recluster, reduction = &quot;pca&quot;, dims = 1:10)
microglia_recluster &lt;- FindNeighbors(microglia_recluster)
microglia_recluster &lt;- FindClusters(microglia_recluster, resolution = 0.5)</code></pre>
<hr />
</details>
<p>It is no surprise that cells from the triculture dataset separate from the CSF/Blood-derived cells which can be seen in the plot below. When reclustering the myeloid lineage derived cells,five subclusters for the iMg and three subclusters for the five CSF/Blood Myeloid clusters are obtained. It can also be observed that the Myeloid2 and Myeloid5 separate from Myeloid1/2/3 which is expected, since Myeloid2/5 originate from CSF samples and Myeloid1/3/4 originate from blood samples.</p>
<pre class="r"><code>microglia_recluster &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/microglia_myeloid_seurat_object_reclustering&quot;)
a1 &lt;- DimPlot(microglia_recluster, reduction = &quot;tsne&quot;,
              group.by = &quot;seurat_clusters&quot;, pt.size = .4) +
  labs(title = &quot;Clustering&quot;)
a2 &lt;- DimPlot(microglia_recluster, reduction =&quot;tsne&quot;, 
              group.by = &quot;cell_type&quot;, pt.size = .4) +
  labs(title = &quot;Cell type&quot;)
a3 &lt;- DimPlot(microglia_recluster, reduction =&quot;tsne&quot;,
              group.by = &quot;dataset&quot;, pt.size = .4) +
    labs(title = &quot;Dataset&quot;)
a4 &lt;- DimPlot(microglia_recluster, reduction = &quot;tsne&quot;,
              group.by = &quot;orig.ident&quot;, pt.size = .4) +
  labs(title = &quot;Sample origin&quot;)

fig &lt;- ggarrange(a1, a2, a3, a4, ncol = 2, nrow = 2)
annotate_figure(fig, 
                top = text_grob(&quot;De-novo clustering of myeloid-lineage clusters&quot;, 
               color = &quot;black&quot;, face = &quot;bold&quot;, size = 20))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-52-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>Interestingly, both the Myeloid5 cells and the Myeloid2 cells are split between the cluster six (blue) and seven (purple) in the figure below. In the following we will have a closer look at this segregation. When performing GRN inference with Scenic we also identified two potentially biologically relevant subclusters of the Myeloid2 cells. It could be that these independently found clusters correlate.</p>
<pre class="r"><code>a5 &lt;- DimPlot(microglia_recluster, reduction =&quot;tsne&quot;, 
              split.by = &quot;cell_type&quot;, pt.size = .4, ncol=3) +
  labs(title = &quot;Contribution of different cell types to different myeloid clusters&quot;)
a5</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-53-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-53-1">
Past versions of unnamed-chunk-53-1.png
</button>
</p>
<div id="fig-unnamed-chunk-53-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-53-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>It looks like the two clusters identified this way correspond to different sample origins. CSF cells from patient1 (HIV1_CSF) and patient2 (HIV2_CSF) belong to cluster six, while CSF cells from patient3 (HIV3_CSF) and the two uninfected patients (Uninfected1_CSF, Uninfected2_CSF) belong to cluster seven. The separation of patient3 (HIV3_CSF) from patient1 (HIV1_CSF) and patient2 (HIV2_CSF) might correspond to a batch effect, or it might be a biological effect, for example, the infection might have affected the cells differently.</p>
<pre class="r"><code>#a1 &lt;- DimPlot(subset(microglia_recluster, cell_type == &quot;iMg&quot;), reduction = &quot;tsne&quot;,
           #   group.by = &quot;orig.ident&quot;, pt.size = .4)
a2 &lt;- DimPlot(subset(microglia_recluster, cell_type != &quot;iMg&quot;), reduction = &quot;tsne&quot;,
              group.by = &quot;orig.ident&quot;, pt.size = .4) +
  labs(title = &quot;Sample origin&quot;)
#a3 &lt;- DimPlot(subset(microglia_recluster, cell_type == &quot;iMg&quot;), reduction = &quot;tsne&quot;,
              #group.by = &quot;seurat_clusters&quot;, pt.size = .4)
a4 &lt;- DimPlot(subset(microglia_recluster, cell_type != &quot;iMg&quot;), reduction = &quot;tsne&quot;,
              group.by = &quot;seurat_clusters&quot;, pt.size = .4) +
  labs(title = &quot;Cluster&quot;)
fig &lt;- ggarrange(a2, a4, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob(&quot;CSF/Blood Myeloid cells&quot;, 
                                     face = &quot;bold&quot;, size = 20))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-54-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>Since the segregation into two clusters seems to be related to the sample origin rather than to the infection status, we would not expect the clusters to overlap with the clusters identified after GRN inference with SCENIC. In the figure below it can be seen that the clusters do not overlap (only the Myeloid2 cells from infected CSF samples (HIV1_CSF, HIV2_CSF, HIV3_CSF) are shown). For this reason the reclustering did not lead to any useful insights and will not be analyzed further.</p>
<pre class="r"><code># read in Myeloid2 cells
hivmy2 &lt;- readRDS(&quot;data_files/hiv_myeloid2_subclusters&quot;)

my2 &lt;- subset(microglia_recluster, cell_type == &quot;Myeloid2&quot; &amp; 
                orig.ident %in% c(&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;))

my2 &lt;- AddMetaData(my2, hivmy2@meta.data[&quot;ht_cluster&quot;], 
                   col.name = &quot;TF_activity_clusters&quot;)
p1 &lt;- DimPlot(my2, group.by = &quot;seurat_clusters&quot;) +
  labs(title=&quot;Clusters&quot;)
p2 &lt;- DimPlot(my2, group.by = &quot;TF_activity_clusters&quot;) +
  labs(title = &quot;TF activity&quot;)
p3 &lt;- DimPlot(my2, group.by = &quot;orig.ident&quot;) +
  labs(title = &quot;Sample origin&quot;)
figure &lt;- ggarrange(p1, p2, p3, ncol = 3, nrow = 1)
annotate_figure(figure, 
                top = text_grob(&quot;Comparison with clusters of different TF acitivty&quot;, 
               color = &quot;black&quot;, face = &quot;bold&quot;, size = 20))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-55-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-55-1">
Past versions of unnamed-chunk-55-1.png
</button>
</p>
<div id="fig-unnamed-chunk-55-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/75af25c571d2f255bb408c63df6ee70c423b892a/docs/figure/Triculture_csf_integration.Rmd/unnamed-chunk-55-1.png" target="_blank">75af25c</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#my25 &lt;- subset(microglia, seurat_clusters %in% c(6,7) &amp;cell_type %in% c(&quot;Myeloid2&quot;, &quot;Myeloid5&quot;))</code></pre>
</div>
<div id="inflammatory-pathways" class="section level2">
<h2>Inflammatory Pathways</h2>
<p>As already mentioned above <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span> showed that two pathways are specifically activated in the Inf+EFZ condition, namely RhoGDI signaling and CD40 signaling. This effect was not seen in Uninf+EFZ, so it does not seem to be caused by EFZ alone, but rather a combination of infection and EFZ treatment <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span>.</p>
<ul>
<li>RhoGDI signaling:
<ul>
<li>PP1R12A was upregulated</li>
<li>RND3, CD44, ACTG1 were downregulated</li>
</ul></li>
<li>CD40 signaling:
<ul>
<li>FOS was upregulated</li>
<li>NFKBIA, ICAM1, TFNAIP3 were downregulated</li>
</ul></li>
</ul>
<p>To answer the question whether we could find similar patterns in the two Myeloid2 clusters with different TF activity, we had a look at the gene expression.</p>
<div id="myeloid2-cells" class="section level5">
<h5>Myeloid2 cells</h5>
<p>Two genes of the RhoGDI signaling pathway were found among the differentially expressed genes between Inf+EFZ and Uninf+EFZ. Both of them were downregulated in Inf+EFZ compared to Uninf+EFZ iMg as can be seen in the table below.</p>
<pre class="r"><code>inf_uninf_art_marker %&gt;% 
  rownames_to_column(&quot;gene&quot;) %&gt;%
  filter(gene %in% c(&quot;RND3&quot;, &quot;CD44&quot;, &quot;ACTG1&quot;, &quot;PPP1R12A&quot;)) %&gt;%
  kable(caption = &quot;Genes of the RhoGDI signaling pathway found among genes differentially expressed between Inf+EFZ and Uninf+EFZ&quot;)</code></pre>
<table>
<caption>Genes of the RhoGDI signaling pathway found among genes differentially expressed between Inf+EFZ and Uninf+EFZ</caption>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="right">p_val</th>
<th align="right">avg_log2FC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">RND3</td>
<td align="right">0</td>
<td align="right">-1.6635458</td>
<td align="right">0.045</td>
<td align="right">0.386</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">CD44</td>
<td align="right">0</td>
<td align="right">-0.5883562</td>
<td align="right">0.396</td>
<td align="right">0.539</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>All four genes of the CD40 signaling were found among the differentially expressed genes between Inf+EFZ and Uninf+EFZ. FOS was upregulated, while NFKBIA, ICAM1 and TFNAIP3 were downregulated in Inf+EFZ compared to Uninf+EFZ as can be seen in the table below.</p>
<pre class="r"><code>inf_uninf_art_marker %&gt;% 
  rownames_to_column(var = &quot;gene&quot;) %&gt;%
  filter(gene %in% c(&quot;FOS&quot;, &quot;NFKBIA&quot;, &quot;ICAM1&quot;, &quot;TNFAIP3&quot;)) %&gt;% kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="right">p_val</th>
<th align="right">avg_log2FC</th>
<th align="right">pct.1</th>
<th align="right">pct.2</th>
<th align="right">p_val_adj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ICAM1</td>
<td align="right">0</td>
<td align="right">-1.3649862</td>
<td align="right">0.169</td>
<td align="right">0.496</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">NFKBIA</td>
<td align="right">0</td>
<td align="right">-1.0940764</td>
<td align="right">0.527</td>
<td align="right">0.784</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">TNFAIP3</td>
<td align="right">0</td>
<td align="right">-1.3232813</td>
<td align="right">0.158</td>
<td align="right">0.465</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">FOS</td>
<td align="right">0</td>
<td align="right">0.5719556</td>
<td align="right">0.293</td>
<td align="right">0.139</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>The changes in gene expression of both pathways are in line with the findings by <span class="citation">(<a href="#ref-Ryan" role="doc-biblioref">Ryan SK 2020</a>)</span>.</p>
<p>Additionally, we were interested whether this pattern in gene expression correlated with the two clusters identified in Myeloid2 which exhibit different TF activities. The figure below show the gene expression of the four genes of the respective pathways both in Myeloid2 cells (left column) and in iMg (right column).</p>
<p>FOS is the only gene which shows a high expression in Myeloid2 cluster c1. This supports our hypothesis that c1 might be a latently infected reactive microglia cell state. However then we would expect to see a similar pattern for the RhoGDI signaling, meaning PPP1R12A should be the only gene which is highly expressed in c1. This is not the case, but rather CD44 and ACTG1 also show higher expression in c1 compared to c2. Consequently, our hypothesis is not clearly supported by the expression patterns of these inflammatory genes.</p>
<pre class="r"><code># isolate microglia cells
microglia &lt;- subset(combined, cell_type == &quot;iMg&quot;)

# read in Seurat object containing Myeloid2 cells
hivmy2 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv_myeloid2_subclusters&quot;)


p1 &lt;- VlnPlot(hivmy2, features = c(&quot;RND3&quot;, &quot;CD44&quot;, &quot;ACTG1&quot;, &quot;PPP1R12A&quot;), 
              group.by = &quot;ht_cluster&quot;, ncol = 2)

p2 &lt;- VlnPlot(hivmy2, features =c(&quot;NFKBIA&quot;, &quot;ICAM1&quot;, &quot;TNFAIP3&quot;, &quot;FOS&quot;), 
              group.by = &quot;ht_cluster&quot;, ncol = 2)

p3 &lt;- VlnPlot(microglia, features = c(&quot;RND3&quot;, &quot;CD44&quot;, &quot;ACTG1&quot;, &quot;PPP1R12A&quot;), 
        ncol = 2, pt.size = 0, group.by = &quot;orig.ident&quot;)
Warning: Could not find ACTG1 in the default search locations, found in RNA
assay instead
Warning: Could not find PPP1R12A in the default search locations, found in RNA
assay instead

p4 &lt;- VlnPlot(microglia, features = c(&quot;NFKBIA&quot;, &quot;ICAM1&quot;, &quot;TNFAIP3&quot;, &quot;FOS&quot;), 
        ncol = 2, pt.size = 0, group.by = &quot;orig.ident&quot;)

fig &lt;- ggarrange(p1, p3, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob(&quot;Expression of four genes of the RhoGDI signaling pathway in Myeloid2 (left column) and iMg (right column)&quot;,
                                     size = 20 , face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-58-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>
fig &lt;- ggarrange(p2, p4, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob(&quot;Expression of four genes of the CD40 signaling pathway in Myeloid2 (left column) and iMg (right column)&quot;,
                                     size = 20 , face = &quot;bold&quot;))</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-58-2.png" width="1440" style="display: block; margin: auto;" /></p>
<details>
<summary>
Same two pathways in Myeloid5 cells
</summary>
<p>Myeloid5 cells</p>
<pre class="r"><code>
# isolate the Myeloid5 cluster
hivmy5 &lt;- subset(hiv7, cell_type == &quot;Myeloid5&quot; &amp; orig.ident %in% c(&quot;HIV1_CSF&quot;, &quot;HIV2_CSF&quot;, &quot;HIV3_CSF&quot;))

# process Myeloid 5 Seurat object
hivmy5 &lt;- ScaleData(hivmy5, verbose = FALSE)
hivmy5 &lt;- NormalizeData(hivmy5, verbose = FALSE)
hivmy5 &lt;- ScaleData(hivmy5, verbose = FALSE)
hivmy5 &lt;- FindVariableFeatures(hivmy5, verbose = FALSE)
Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
parametric, : pseudoinverse used at -2.0668
Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
parametric, : neighborhood radius 0.49858
Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
parametric, : reciprocal condition number 3.7892e-16
Warning in simpleLoess(y, x, w, span, degree = degree, parametric =
parametric, : There are other near singularities as well. 0.090619
hivmy5 &lt;- RunPCA(hivmy5, features = VariableFeatures(hivmy5), verbose = FALSE)
#ElbowPlot(hivmy5)
hivmy5 &lt;- RunTSNE(hivmy5, dims= 1:10, verbose = FALSE)</code></pre>
<pre class="r"><code>
p1 &lt;- VlnPlot(hivmy5, features = c(&quot;RND3&quot;, &quot;CD44&quot;, &quot;ACTG1&quot;, &quot;PPP1R12A&quot;), 
              group.by = &quot;orig.ident&quot;,ncol = 2)
Warning in SingleExIPlot(type = type, data = data[, x, drop = FALSE], idents =
idents, : All cells have the same value of RND3.

p2 &lt;- VlnPlot(hivmy5, features =c(&quot;NFKBIA&quot;, &quot;ICAM1&quot;, &quot;TNFAIP3&quot;, &quot;FOS&quot;), 
              group.by = &quot;orig.ident&quot;, ncol = 2)

p3 &lt;- VlnPlot(microglia, features = c(&quot;RND3&quot;, &quot;CD44&quot;, &quot;ACTG1&quot;, &quot;PPP1R12A&quot;), 
        ncol = 2, pt.size = 0, group.by = &quot;orig.ident&quot;)
Warning: Could not find ACTG1 in the default search locations, found in RNA
assay instead
Warning: Could not find PPP1R12A in the default search locations, found in RNA
assay instead

p4 &lt;- VlnPlot(microglia, features = c(&quot;NFKBIA&quot;, &quot;ICAM1&quot;, &quot;TNFAIP3&quot;, &quot;FOS&quot;), 
        ncol = 2, pt.size = 0, group.by = &quot;orig.ident&quot;)

ggarrange(p2, p4, p1, p3, ncol = 2, nrow = 2)</code></pre>
<p><img src="figure/Triculture_csf_integration.Rmd/unnamed-chunk-60-1.png" width="1440" style="display: block; margin: auto;" /></p>
</details>
</div>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Ryan" class="csl-entry">
Ryan SK, Garifallou JP, Gonzalez MV. 2020. <em>Neuroinflammation and Eif2 Signaling Persist Despite Antiretroviral Treatment in an hiPSC Tri-Culture Model of HIV Infection</em>. <em>#SStem Cell Reports</em>. Vol. 14(4). <a href="https://doi: 10.1016/j.stemcr.2020.02.010">doi: 10.1016/j.stemcr.2020.02.010</a>.
</div>
<div id="ref-Farhadian" class="csl-entry">
Shelli F. Farhadian, Chrysoula Zografou, Sameet S. Mehta, and Serena S. Spudich. 2018. <em>Single-Cell RNA Sequencing Reveals Microglia-Like Cells in Cerebrospinal Fluid During Virologically Suppressed HIV</em>. <em>JCI Insight</em>. Vol. 3(18):e121718. <a href="https://doi.org/10.1172/jci.insight.121718">https://doi.org/10.1172/jci.insight.121718</a>.
</div>
<div id="ref-Stuart" class="csl-entry">
Stuart T, Hoffman P, Butler A. 2019. <em>Comprehensive Integration of Single-Cell Data</em>. <em>Cell</em>. Vol. 177(7). <a href="https://doi: 10.1016/j.cell.2019.05.031">doi: 10.1016/j.cell.2019.05.031</a>.
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] readxl_1.3.1         multtest_2.46.0      Biobase_2.50.0      
 [4] BiocGenerics_0.36.1  metap_1.6            knitr_1.30          
 [7] rstatix_0.6.0        corrplot_0.92        gridExtra_2.3       
[10] ggpubr_0.4.0         circlize_0.4.13      ComplexHeatmap_2.9.4
[13] ggcorrplot_0.1.3     pheatmap_1.0.12      RColorBrewer_1.1-2  
[16] ggrepel_0.9.1        data.table_1.14.2    Matrix_1.3-4        
[19] edgeR_3.32.0         limma_3.46.0         SeuratObject_4.0.3  
[22] Seurat_4.0.5         forcats_0.5.0        stringr_1.4.0       
[25] dplyr_1.0.7          purrr_0.3.4          readr_1.4.0         
[28] tidyr_1.1.2          tibble_3.1.6         ggplot2_3.3.5       
[31] tidyverse_1.3.0      workflowr_1.6.2     

loaded via a namespace (and not attached):
  [1] utf8_1.2.2            reticulate_1.22       tidyselect_1.1.1     
  [4] htmlwidgets_1.5.4     Rtsne_0.15            munsell_0.5.0        
  [7] units_0.6-7           mutoss_0.1-12         codetools_0.2-18     
 [10] ica_1.0-2             future_1.23.0         miniUI_0.1.1.1       
 [13] withr_2.4.3           colorspace_2.0-2      highr_0.8            
 [16] rstudioapi_0.13       stats4_4.0.2          ROCR_1.0-11          
 [19] ggsignif_0.6.0        tensor_1.5            listenv_0.8.0        
 [22] labeling_0.4.2        Rdpack_2.1.2          git2r_0.27.1         
 [25] mnormt_2.0.2          polyclip_1.10-0       farver_2.1.0         
 [28] rprojroot_2.0.2       TH.data_1.0-10        parallelly_1.29.0    
 [31] vctrs_0.3.8           generics_0.1.1        xfun_0.29            
 [34] R6_2.5.1              doParallel_1.0.16     clue_0.3-60          
 [37] RVenn_1.1.0           locfit_1.5-9.4        spatstat.utils_2.2-0 
 [40] assertthat_0.2.1      promises_1.2.0.1      scales_1.1.1         
 [43] multcomp_1.4-15       gtable_0.3.0          globals_0.14.0       
 [46] goftest_1.2-3         sandwich_3.0-0        rlang_0.4.12         
 [49] GlobalOptions_0.1.2   splines_4.0.2         lazyeval_0.2.2       
 [52] spatstat.geom_2.3-0   broom_0.7.2           yaml_2.2.1           
 [55] reshape2_1.4.4        abind_1.4-5           modelr_0.1.8         
 [58] backports_1.2.0       httpuv_1.6.3          tools_4.0.2          
 [61] ellipsis_0.3.2        spatstat.core_2.3-1   ggridges_0.5.2       
 [64] TFisher_0.2.0         Rcpp_1.0.7            plyr_1.8.6           
 [67] classInt_0.4-3        rpart_4.1-15          deldir_1.0-6         
 [70] pbapply_1.5-0         GetoptLong_1.0.5      cowplot_1.1.1        
 [73] S4Vectors_0.28.1      zoo_1.8-9             haven_2.3.1          
 [76] cluster_2.1.0         fs_1.5.1              magrittr_2.0.1       
 [79] magick_2.5.2          scattermore_0.7       openxlsx_4.2.3       
 [82] lmtest_0.9-39         reprex_0.3.0          RANN_2.6.1           
 [85] tmvnsim_1.0-2         mvtnorm_1.1-1         whisker_0.4          
 [88] fitdistrplus_1.1-6    matrixStats_0.61.0    hms_1.1.1            
 [91] patchwork_1.1.1       mime_0.12             evaluate_0.14        
 [94] xtable_1.8-4          rio_0.5.16            IRanges_2.24.1       
 [97] shape_1.4.6           compiler_4.0.2        KernSmooth_2.23-18   
[100] crayon_1.4.2          htmltools_0.5.2       mgcv_1.8-33          
[103] later_1.3.0           ggVennDiagram_1.2.0   lubridate_1.7.9.2    
[106] DBI_1.1.1             dbplyr_2.0.0          MASS_7.3-53          
[109] sf_0.9-6              car_3.0-10            cli_3.1.0            
[112] rbibutils_2.2.4       igraph_1.2.9          pkgconfig_2.0.3      
[115] sn_2.0.1              numDeriv_2016.8-1.1   foreign_0.8-80       
[118] plotly_4.9.2.1        spatstat.sparse_2.0-0 xml2_1.3.2           
[121] foreach_1.5.1         rvest_0.3.6           digest_0.6.29        
[124] sctransform_0.3.2     RcppAnnoy_0.0.19      spatstat.data_2.1-0  
[127] rmarkdown_2.6         cellranger_1.1.0      leiden_0.3.5         
[130] uwot_0.1.10           curl_4.3.2            shiny_1.7.1          
[133] rjson_0.2.20          lifecycle_1.0.1       nlme_3.1-150         
[136] jsonlite_1.7.2        carData_3.0-4         viridisLite_0.4.0    
[139] fansi_0.5.0           pillar_1.6.4          lattice_0.20-41      
[142] plotrix_3.8-2         fastmap_1.1.0         httr_1.4.2           
[145] survival_3.2-7        glue_1.6.0            zip_2.1.1            
[148] png_0.1-7             iterators_1.0.13      class_7.3-17         
[151] stringi_1.7.6         mathjaxr_1.4-0        e1071_1.7-4          
[154] irlba_2.3.3           future.apply_1.8.1   </code></pre>
</div>
</div>

<div id="rmd-source-code">---
title: "Analysis of Triculture dataset and integration with CSF/Blood dataset"
author: "kmikulik"
date: "28 12 2021"
output: html_document
bibliography: references.bib
link-citations: yes
---

<style>
body {
text-align: justify}
</style>


```{r setup}
knitr::opts_chunk$set(echo = TRUE, collapse = TRUE, message = FALSE,
                      cache = TRUE, cache.lazy = FALSE, autodep = TRUE)
set.seed(42) # TODO: check with default seed
```



```{r}
library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggrepel)
#library(harmony)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(corrplot)
library(rstatix)
library(knitr)
library(metap)
library(multtest)
library(readxl)
```

# Triculture Dataset

The triculture dataset used in the following analysis was published by [@Ryan]. 
They used a Triculture system with induced neurons, induced astrocytes and 
induced microglia.
Induced microglia express marker genes CX3CR1, IBA1, TMEM119 and P2RY12.
CCR5 which is a co-receptor necessary for HIV infection is also expressed by 
these cells, but they lack expression of myeloid-progenitor markers [@Ryan].

iMg were infected with HIV for 15 days, resulting in 94% of cells infected  
(positive for HIV capsid protein p24 and exhibiting multinucleation). Some cells
were treated with the antiretroviral therapy EFZ, which blocks HIV reverse transcriptase. This way, it was possible to study non-productively infected cells. EFZ reduced infection by two 
thirds [@Ryan].

In the infected, but not treated iMg (Inf) inflammatory pathways were 
significantly uperegulated compared to the uninfected condition(Uninf), including
IL-8, NFkB and EIF2. 
There was decreased expression of pro-imflammatory genes IL1b, Il8 and TNFalpha
in the infected and treated (Inf+EFZ) vs Inf condition. This suggests a reduced inflammatory reaction [@Ryan]. Nevertheless, there were
two distinct pathways which were upregulated in 
Inf+EFZ iMg, namely RhoGDI signaling and CD40 signaling pathway. This will be 
discussed in more detail later on [@Ryan].

There was  a change in signaling pathways caused by EFZ treatment even in 
the uninfected and treated (Uninf+EFZ) versus the uninfected, but not treated
(Uninf) condition. The EFZ treatment alone was very different from the EFZ treatment with infection [@Ryan]. 

 
<details>
<summary>**Files and data structure**</summary>
  
**Read in Files:**

Abbreviations: 

* ia/2 = infected with HIV + treated with ART
* ca = uninfected + ART
* ctl = uninfected
* inf = infected 

File structure: 

* the ".mtx" files contains a sparse matrix of class dgTMatrix
* the "barcodes.tsv" files contain the cell barcodes
* the "features.tsv" files contain the feature names as well as ensemble gene ids 
and "gene expression" strings in a third column


There are duplicated features in the data, probably, because there are multiple ensemble gene IDs mapping to the same gene symbol. Therefore, the same gene 
symbol has multiple expression values. How do you deal with this problem?

* You could make them unique with the function `make.unique()`
* You could remove the duplicated features. **This is what I did**

```
path <- "/media/ag-cherrmann/kmikulik/HIV_microglia/data/GSE143686_RAW/" #"/media/ag-cherrmann/kmikulik/HIV_microglia/data/GSE143686_RAW/"
all_files <- list.files(path)

all_cond_list <- map(seq.int(3,7), function(n){
  file <- all_files[grepl(paste0("GSM427258", n), all_files)]
  name <- str_extract(file[[1]],"ca|ctl|ia2|ia|inf")
  barcodes <- read.table(paste0(path,file[[1]]), sep = "\t") 
  features <- read.table(paste0(path, file[[2]]), sep = "\t")
  matrix <- readMM(paste0(path, file[[3]]))
  colnames(matrix) <- barcodes$V1
  rownames(matrix) <- features$V2
  matrix <- matrix[!duplicated(rownames(matrix)),]
  seurat_ob <- CreateSeuratObject(matrix, project = name,
                                  assay = "RNA", min.cells = 1, min.features = 1)
  seurat_ob[["percent_mt"]] <- PercentageFeatureSet(seurat_ob, pattern = "^MT-")
  list(name = name, count = matrix, seurat_ob = seurat_ob)
})

#saveRDS(all_cond_list, "/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/seurat_list")
```

---
</details>

<details>
<summary>**Files and data structure**</summary>

# Data processing {.tabset}

## Before Quality Control:


```{r, fig.width=15}
all_cond_list <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/seurat_list")

before_qc <- merge(all_cond_list[[1]]$seurat_ob,
                    c(all_cond_list[[2]]$seurat_ob,
                      all_cond_list[[3]]$seurat_ob,
                      all_cond_list[[4]]$seurat_ob,
                      all_cond_list[[5]]$seurat_ob),
                    project = "triculture_all_conditions",
                    add.cell.ids = c(paste0(all_cond_list[[1]]$name),
                                     paste0(all_cond_list[[2]]$name),
                                     paste0(all_cond_list[[3]]$name),
                                     paste0(all_cond_list[[4]]$name),
                                     paste0(all_cond_list[[5]]$name)))
```


```{r, fig.width=15, fig.height=7}
qual_metric <- c("nFeature_RNA", "nCount_RNA", "percent_mt")

plots <- map(qual_metric, function(metric){
  ggplot() +
    geom_violin(aes(x = before_qc@meta.data %>% pull("orig.ident"),
                 y = before_qc@meta.data %>% pull(metric), 
                 fill = before_qc@meta.data %>% pull("orig.ident")), 
                alpha = .5) +
    geom_boxplot(aes(x = before_qc@meta.data %>% pull("orig.ident"),
                     y = before_qc@meta.data %>% pull(metric)),
                 alpha = 0) +
    xlab("sample origin") +
    ylab(paste0(metric)) +
    NoLegend() 
})

do.call(grid.arrange, c(plots, ncol = 3, nrow = 1))
```

---

## After quality control

For each sample separately

* remove genes expressing less than 200 genes or more than 5000
* remove cells expressing more than 20% mitochondrial genes
* merge Seurat objects after Quality Control

```{r}
# add percentage of mitochondiral genes to metadata
all_cond <- map(seq.int(1,5), function(n){
  seurat_ob <- all_cond_list[[n]]$seurat_ob
  #seurat_ob[["percent_mt"]] <- PercentageFeatureSet(seurat_ob, pattern = "^MT-")
  #plot <- VlnPlot(seurat_ob,
   #               features = c("nFeature_RNA", "percent_mt", "nCount_RNA"),
    #              ncol = 3) +
    #labs(title = paste0("condition ", all_cond_list[[n]]$name))
  seurat_ob <- subset(seurat_ob, nFeature_RNA > 200 & nFeature_RNA < 5000 & percent_mt < 20)
  list(name = all_cond_list[[n]]$name, seurat_ob = seurat_ob)#, plot = plot)
})
```


```{r, fig.width=15}
triculture <- merge(all_cond[[1]]$seurat_ob,
                    c(all_cond[[2]]$seurat_ob,
                      all_cond[[3]]$seurat_ob,
                      all_cond[[4]]$seurat_ob,
                      all_cond[[5]]$seurat_ob),
                    project = "triculture_all_conditions",
                    add.cell.ids = c(paste0(all_cond[[1]]$name),
                                     paste0(all_cond[[2]]$name),
                                     paste0(all_cond[[3]]$name),
                                     paste0(all_cond[[4]]$name),
                                     paste0(all_cond[[5]]$name)))
```


```{r, fig.width=15, fig.height=7}
qual_metric <- c("nFeature_RNA", "nCount_RNA", "percent_mt")

plots <- map(qual_metric, function(metric){
  ggplot() +
    geom_violin(aes(x = triculture@meta.data %>% pull("orig.ident"),
                 y = triculture@meta.data %>% pull(metric), 
                 fill = triculture@meta.data %>% pull("orig.ident")), 
                alpha = .5) +
    geom_boxplot(aes(x = triculture@meta.data %>% pull("orig.ident"),
                     y = triculture@meta.data %>% pull(metric)),
                 alpha = 0) +
    xlab("sample origin") +
    ylab(paste0(metric)) +
    NoLegend() 
})

do.call(grid.arrange, c(plots, ncol = 3, nrow = 1))
```

---

## Normalization, Scaling, Find Variable Features

* log normalization
* scale data so that we have mean = 0 and variance = 1
* Find the 2000 most highly variable features

```
triculture <- NormalizeData(triculture, verbose = FALSE)
triculture <- ScaleData(triculture, verbose = FALSE)
triculture <- FindVariableFeatures(triculture, verbose = FALSE)
```
---

## Dimensionality reduction

* compute PCA
* use elbow plot to determine dimensions that contribute to variability
* find the k-nearest neighbors
* find Clusters based on KNN-graph
* calculate TSNE embedding

```
triculture <- RunPCA(triculture, features = VariableFeatures(triculture), verbose = FALSE)
ElbowPlot(triculture)

triculture <- FindNeighbors(triculture, verbose = FALSE)
triculture <- FindClusters(triculture, verbose = FALSE)

triculture <- RunTSNE(triculture, dims = 1:10, verbose = FALSE)
```

```{r}
triculture <- readRDS("seurat_objects/triculture_seurat_object")

#triculture <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_seurat_object")

DimPlot(triculture, reduction = "tsne", pt.size = .01,label = TRUE,
        group.by = "seurat_clusters")
```

---

## Cell type annotation

For annotating the different clusters the same markers as in [@Ryan] were used:

* *Microglia:* AIF1, SPI1, CD4
* *Astrocytes:* SOX9, THBS1
* *Neurons:* SYN1, MAP2

```{r, fig.widht = 15, fig.height=12}
#markers <- FindAllMarkers(triculture)
#triculture <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_seurat_object")
markers <- read.table("data_files/marker_genes_triculture_dataset")
#markers <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/marker_genes_triculture_dataset")

top_2 <- markers %>% group_by("seurat_cluster") %>% top_n(n = 2, wt = avg_log2FC)

marker_genes <- c("AIF1", "SPI1", "CD4", "MAP2", "SYN1", "THBS1", "SOX9")
p1 <- FeaturePlot(triculture, features = "AIF1")
p2 <- FeaturePlot(triculture, features = "SPI1")
p3 <- FeaturePlot(triculture, features = "CD4")
p4 <- FeaturePlot(triculture, features = "MAP2")
p5 <- FeaturePlot(triculture, features = "SYN1")
p6 <- FeaturePlot(triculture, features = "THBS1")
p7 <- FeaturePlot(triculture, features = "SOX9")

ggarrange(p1,p2,p3,p4,p5,p6,p7, ncol = 2, nrow =4 )

```



<details>
<summary>**Details on cell type annotation**</summary>

**Gene expression**

Different clusters were annotated according to the gene expression of 
marker genes mentioned above.

```{r, fig.width=15}

VlnPlot(triculture, features = c("CD4", "SPI1", "AIF1"), group.by = "seurat_clusters")#, group.by = seurat_cluster)
VlnPlot(triculture, features = c("MAP2", "SYN1"), group.by = "seurat_clusters")
VlnPlot(triculture, features = c("THBS1", "SOX9"), group.by = "seurat_clusters")
```

```{r}
# 4, 9, 11, 18 = Microglia
# 0, 1, 2, 3,8, 10, 13, 15, 18, 19 = Neurons 
# 6, 7, 12, 14, 16 = Astrocytes
# 5, 17 are undesginated
cell_types <- (triculture@meta.data %>% 
  mutate(cell_type = case_when(
    seurat_clusters %in% c(4,9,11) ~ "iMg", 
    seurat_clusters %in% c(6,7,12,14,16) ~ "iAst", 
    seurat_clusters %in% c(0,1,2,3,8,10,13,15,18,19) ~ "iNeurons",
    TRUE  ~ "Undesignated")))[["cell_type"]]
  
triculture <- AddMetaData(triculture, cell_types, col.name = "cell_type")

```

---
</details>


---
</details

---

# Exploring the dataset

### Remove undesignated cells

There is one group of cells where it is not clear which cell state they 
represent (Undesignated), we will exclude them from further analysis as [@Ryan] 
did.


```{r}
p1 <- DimPlot(triculture, reduction = "tsne", pt.size = .01, label = TRUE,
        group.by = "cell_type") +
  labs(title = "cell types")
```


```{r}
triculture_clean <- subset(triculture, cell_type %in% c("iMg", "iNeurons", "iAst")) 
```

```{r}
p2 <- DimPlot(triculture_clean, reduction = "tsne", pt.size = .01, label = TRUE,
        group.by = "cell_type") + 
  labs(title = "cell types")
```

```{r, fig.width=15}
p1+p2
```


<details>
<summary>**Heatmap of top 20 marker genes for each cell type**</summary>


Top20 marker genes for each cell cell type: 
 
```{r, fig.width=15, fig.height=25}
Idents(triculture_clean) <- "cell_type"
top_n <- markers %>% group_by (cluster) %>% top_n(n = 20, wt = avg_log2FC)
DoHeatmap(triculture_clean, features = top_n$gene)
```


---
</details>


<details>
<summary>**Number of cells per condition and cell type**</summary>


How many cells of each cell type and condition are present in the dataset?

```{r, results = "asis"}
triculture_clean@meta.data %>% group_by(cell_type, orig.ident) %>% 
  summarize(n = n()) %>% 
  mutate(orig.ident = case_when(
                      orig.ident %in% c("ia", "ia2") ~ "Inf+EFZ", 
                      orig.ident == "ca" ~ "Uninf",
                      orig.ident == "ctl" ~ "Uninf+EFZ",
                      orig.ident == "inf" ~ "Inf")) %>% 
  pivot_wider(names_from = orig.ident, values_from = n) %>% 
  kable(caption = "Number of cells per condition and cell type") # to make table look nice in Rmarkdown
  
```


---
</details>


### Infection and treatment conditions

Interestingly, the infected cells treated with EFZ (Inf+EFZ) clearly separate 
from the infected, but untreated cells (Inf) and the uninfected cells
(Uninf, Uninf+EFZ) in the microglia cluster as can be seen in the figure below.


```{r}
# rename origin of ident for easier interpretation
origin_vector <- (triculture_clean@meta.data %>% 
                    mutate(orig.ident = case_when(
                      orig.ident %in% c("ia", "ia2") ~ "Inf+EFZ", 
                      orig.ident == "ca" ~ "Uninf",
                      orig.ident == "ctl" ~ "Uninf+EFZ",
                      orig.ident == "inf" ~ "Inf")))[["orig.ident"]]

triculture_clean <- AddMetaData(triculture_clean, origin_vector, col.name = "orig.ident")
```

```{r, fig.width=15}
p1 <- DimPlot(triculture_clean, reduction = "tsne", pt.size = .01, label = TRUE,
        group.by = "cell_type") + 
  labs(title = "cell types")
p2 <- DimPlot(triculture_clean, reduction = "tsne", 
              group.by = "orig.ident", pt.size = .01, ) + 
  labs(title = "Infection & treatment condition")

p1 + p2
```


<details>
<summary>**Inflammatory gene expression in iMg**</summary>

The largest change in gene expression related to inflammation was found in iMg
among the 3 cell types [@Ryan]. Shown are only inflammatory genes. 

```{r, fig.height = 12}
inflamm_genes <- c("TNF", "ACSL1", "MGST3", "B2M", "IFNGR2", "GRINA", "TLR2", 
                   "SOD2", "GLUL", "CFLAR", "CTNNB1", "CXCL8", "CCL4", "IL1B",
                   "ATF4", "CXCL8", "CCL4", "CCL2", "IL1B", "ATF4", "CXCL1", 
                   "NFKB1", "TANK", "APOE", "TREM2", "LY96", "SRC", "NCF1", 
                   "HCK", "LYN", "VASP", "C3", "C5", "RAP1B", "RAP2B", "MAF",
                   "JUND", "SQSTM1", "FTH1", "RHOQ", "CD36", "FOS", "NFKBIA",
                   "ICAM1", "TNFAIP3", "LYZ", "PPP1R12A", "RND3", "CYBA", "IRF8",
                   "SLC25A6")

DoHeatmap(triculture_clean, features = inflamm_genes)
```

---
</details>

<details>
<summary>**Differences of inflammatory gene expression between conditions**</summary>


The gene expression of inflammatory genes in microglia was not very different
between the conditions. The Inf+EFZ cells show a slightly different pattern of
gene expression.

```{r, fig.height=12}
DoHeatmap(subset(triculture_clean, cell_type == "iMg"), 
          features = inflamm_genes, group.by = "orig.ident")
```


---
</details>


# Integration Triculture & CSF/Blood 

The HIV Blood & CSF dataset [@Farhadian]
used in the previous analysis was integrated with the triculture dataset [@Ryan].

Integration of the two datasets is based on the Seurat package [@Stuart].
It first identifies 
anchors between datasets, calculates scores for all anchors and 
then calculates an integrated expression matrix which can be used for downstream
analysis.


1. Identify features which are variable across datasets
2. Identify anchors between the two datasets and create an integrated data assay
3. Run a single integrated analysis on all cells


<details>
<summary>**Details on integration**</summary>

For more details see: [@Stuart] and Discussion

1. Identify features which are variable across datasets
2. Identify anchors between the two datasets and create an integrated data assay
3. Run a single integrated analysis on all cells

```
hiv7 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds")
hiv7 <- NormalizeData(hiv7, verbose = FALSE)
hiv7 <- FindVariableFeatures(hiv7, verbose = FALSE)

# add both datasets to a lits
data_list <- list(hiv = hiv7, triculture = triculture_clean)

# select features that are repeatedly variable across datasets for integration
features <- SelectIntegrationFeatures(object.list = data_list, verbose = FALSE)

# identify integration anchors and perform integration
anchors <- FindIntegrationAnchors(object.list = data_list, 
                                  anchor.features = features, verbose = FALSE) 
combined <- IntegrateData(anchorset = anchors, verbose = FALSE)

# run analysis on integrated dataset
combined <- ScaleData(combined, verbose = FALSE)
combined <- RunPCA(combined, verbose = FALSE)
#ElbowPlot(combined)
combined <- RunTSNE(combined, reduction = "pca", dims = 1:10, verbose = FALSE)
combined <- FindNeighbors(combined, reduction = "pca", dims = 1:10, verbose = FALSE)
combined <- FindClusters(combined, resolution = .3, verbose = FALSE)

# add annotations from which dataset the sample came from
dataset_vector <- (combined@meta.data %>% mutate(dataset = ifelse(orig.ident %in%
                                                 c("HIV1_CSF", "HIV2_CSF",
                                                   "HIV3_CSF", "HIV1_Bld", 
                                                   "HIV2_Bld", "Uninfected1_CSF",
                                                   "Uninfected2_CSF"),
                                               "csf", "triculture")))[["dataset"]]

combined <- AddMetaData(combined, dataset_vector, col.name = "dataset")

# add information on the cell groups in the combined datasets
combined_clusters <- (combined@meta.data %>% 
  mutate(combined_clusters = ifelse(seurat_clusters %in% c(2,10), "Microglia", 
                            ifelse(seurat_clusters %in% c(3,9), "iAst", 
                                   ifelse(seurat_clusters %in% c(0,1,7,11),
                                          "iNeurons", 
                                          ifelse(seurat_clusters %in% c(8,5,6), "Immune Cells", "Undesignated"))))))[["combined_clusters"]]
  
combined <- AddMetaData(combined, combined_clusters, col.name = "combined_clusters")

#saveRDS(combined,"/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_hiv_combined_seurat_object")

```

---
</details>

## Integrated clusters

When integrating the CSF dataset and the Triculture dataset the iMg cells 
cluster together with the Myeloid cells from the CSF and Blood samples from 
the HIV infected/uninfected patients (see figure below). This is as expected,
since they are both
myeloid-lineage cells. However, it also makes it more difficult to draw conclusions
on which of the five Myeloid clusters is the most microglia-like. Therefore, we 
had a closer look at this myeloid-lineage cluster consisting of iMg from
triculture and Myeloid cells from CSF and Blood samples.


#### Visualization of the integrated dataset

```{r, fig.width=15, fig.height=15}
combined = readRDS("seurat_objects/triculture_hiv_combined_seurat_object_2")

p1 <- DimPlot(combined, reduction = "tsne", group.by  = "dataset",
              label = TRUE) + NoLegend() +
  labs(title = "Dataset")
p2 <- DimPlot(combined, reduction = "tsne", group.by = "cell_type",
              label = TRUE, repel = TRUE) + NoLegend() +
  labs(title = "Cell types")
p3 <- DimPlot(combined, reduction = "tsne", group.by = "cell_type",
              split.by = "dataset", label = TRUE, repel = TRUE) + NoLegend() +
  labs(title = "Cell types split into datasets")
p4 <- DimPlot(combined, reduction = "tsne", group.by = "seurat_clusters",
              label = TRUE) + NoLegend() +
  labs(title = "Clusters")

a1 <- DimPlot(combined, reduction = "tsne", split.by = "dataset", 
        group.by = "seurat_clusters", label = TRUE, repel =TRUE) +
  NoLegend() +
  labs(title = "Clusters split into datasets")

a2 <- DimPlot(combined, reduction = "tsne", split.by = "dataset", 
              group.by = "orig.ident") +
  labs(title = "Sample origin split into datasets")

ggarrange(p1, p2, p3,p4, a1, a2, ncol = 2, nrow = 3)

```


## Myeloid lineage cluster

#### Visualization of Myeloid lineage cluster (iMg, Myeloid1/2/3/4/5)

The plot below visualizes that iMg and CSF/Blood Myeloid cells cluster together,
independent of their infection status.

```{r, fig.width=15, fig.height=7}
subset_monocytes <- subset(combined, cell_type %in% c("Myeloid1", 
                                                            "Myeloid2", 
                                                            "Myeloid3", 
                                                            "Myeloid4", 
                                                            "Myeloid5", 
                                                            "iMg"))

p1 <- DimPlot(subset_monocytes, group.by = "cell_type") +
  labs(title = "Cell type")
p2 <- DimPlot(subset_monocytes, group.by = "orig.ident") +
  labs(title = "Sample origin")
p3 <- DimPlot(subset_monocytes, group.by = "seurat_clusters") +
  labs(title = "Cluster")
fig <- ggarrange(p1, p2, p3, ncol = 3, nrow = 1)
annotate_figure(fig, top = text_grob("Myeloid-lineage cells", size = 20))
```

#### Visualization of infected + EFZ iMg & infected Myeloid1/2/3/4/5

As expected the CSF/Blood Myeloid cells from infected patients cluster together 
with infected + EFZ iMg and mainly contribute to the same subcluster, as can 
be seen in the plot below, which shows only infected CSF/Blood Myeloid cells and the 
infected + EFZ iMg. Myeloid4 is an exception and belongs to a different subcluster.

```{r, fig.width=15, fig.height=7}

subset_infected <- subset(combined, orig.ident %in% c("HIV1_CSF", "HIV2_CSF", "HIV3_CSF",
                                             "HIV1_Bld", "HIV2_Bld","Inf+EFZ") &
                            cell_type %in% c("Myeloid1", 
                                                            "Myeloid2", 
                                                            "Myeloid3", 
                                                            "Myeloid4", 
                                                            "Myeloid5", 
                                                            "iMg"))


p1 <- DimPlot(subset_infected, group.by = "cell_type") +
  labs(title = "Cell Type")
p2 <- DimPlot(subset_infected, group.by = "orig.ident") +
  labs(title = "Sample origin")
p3 <- DimPlot(subset_infected, group.by = "seurat_clusters") +
  labs(title = "Clusters")
fig <- ggarrange(p1, p2, p3, ncol = 3, nrow = 1)
annotate_figure(fig, top = text_grob("Infected CSF/Blood Myeloid cells & Inf+EFZ iMg"))
```



#### Cell types and conditions partially separate according to the two iMg clusters

The myeloid-lineage cluster separated into two distinct subclusters. 
Interestingly, the infected + EFZ iMg separated clearly from the infected iMg. 
The separation was not so clear for the uninfected +/- EFZ cells, which 
contributed to both subclusters. Another observations is that the infected Myeloid1/2/3/5 cells belong to the same subcluster as the infected + EFZ iMg. Conversely, the Myeloid4 cells cluster with the infected iMg. Since the infected CSF/Blood Myeloid cells originate from HIV infected and treated patients we would expect them to cluster togehter with the infected + EFZ iMg. However, there 
might also be infected CSF/Blood Myeloid cells which are not affected by the treatment. The Myeloid4 cells which are from Blood samples might consitute an 
infected cell population unaffected by the ART treatment. 



```{r,fig.width=15, fig.height=10}

DimPlot(subset_monocytes, group.by = "seurat_clusters", 
        split.by = "orig.ident", ncol = 4) +
  labs(title = "Contribution of different samples to different subclusters")
DimPlot(subset_monocytes, group.by = "seurat_clusters", 
        split.by = "cell_type", ncol = 3) +
  labs(title = "Contribution of different cell types to different subclusters")
```




# Differential gene expression

### Between infection conditions

In the following I will compare gene expression between infected cells and 
uninfected cells, both treated with EFZ and not treated. I will use the uninfected cells as a 
baseline to find genes which are upregulated or downregulated in infected cells
compared to uninfected cells. Since we have both cells treated with EFZ and 
cells not treated with EFZ, we can define a signature for both treatment 
conditions separately and then see if there is an overlap between the gene 
signatures of infected cells +/- EFZ. Finding genes which overlap independently 
of treamtent condition increases the confidence in the results

<details>
<summary>**List of top up- or downregulated genes in Inf+EFZ vs Uninf+EFZ**</summary>

```{r, results = "asis"}
# isolate iMg cells
microglia <- subset(combined, cell_type == "iMg")

# only iMg cells to identify potential signatures between infected vs uninfected
Idents(microglia) =  "orig.ident"

# find marker genes between Uninf+EFZ and Inf+EFZ
inf_uninf_art_marker <- FindMarkers(microglia, ident.1 = "Inf+EFZ", 
                                    ident.2 = "Uninf+EFZ", verbose = FALSE)
#upregulated genes
up_inf_art <- inf_uninf_art_marker %>% 
  arrange(desc(avg_log2FC)) %>%
  head(20) %>% 
  rownames_to_column("gene")
up_inf_art  %>% head %>%  kable(caption = "Genes upregulated in Inf+EFZ vs. Uninf+EFZ")
```

```{r}
# downregulated genes
down_inf_art <- inf_uninf_art_marker %>% 
  arrange(avg_log2FC) %>% 
  head(20) %>%
  rownames_to_column("gene")
down_inf_art %>% head %>% kable(caption = "Genes downregulated in Inf+EFZ vs. Uninf+EFZ")
```

---
</details>

```{r}
# find marker genes between Inf and Uninf
inf_uninf_marker <- FindMarkers(microglia, ident.1 = "Inf", 
                                ident.2 = "Uninf", verbose = FALSE)

# upregulated in Inf vs Uninf, no treatment
up_inf_no <- inf_uninf_marker %>% 
  arrange(desc(avg_log2FC)) %>%
  head(20) %>% 
  rownames_to_column("gene")
#up_inf_no  %>% kable()


down_inf_no <- inf_uninf_marker %>% 
  arrange(avg_log2FC) %>% 
  head(20) %>%
  rownames_to_column("gene")
#down_inf_no%>% kable()
```

Only 3% of genes which are upregulated and only 14% of genes which are 
downregulated in infected vs. uninfected cells 
overlap between the treatment conditions. This means, that the gene expression response defining 
the infection might differ between cells with and without treatment. This
reflects the findings of [@Ryan] that the EFZ treatment leads to a distinct
immune response of iMg and distinct activation of two inflammatory pathways, 
RhoGDI and CD40 pathways as will be described later on. Since there is hardly any
overlap we will use the genes upregulated in Inf+EFZ compared to Uninf+EFZ
in the following analysis steps, because they probably resemble the CSF/Blood 
samples, which originate from patients treated with ART, best.

```{r}
up_inf <- intersect(up_inf_art$gene, up_inf_no$gene)
down_inf <- intersect(down_inf_art$gene, down_inf_no$gene)
print(c(up_inf, down_inf))
```

```{r, fig.width=15}
p1 <- ggVennDiagram::ggVennDiagram(list(up_inf_art$gene, up_inf_no$gene),
              label_alpha = 0,
              category.names= c("Inf+EFZ", "Inf"),
              #set.size = 20,
              label = "percent",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "Genes upregulated in infected cells")

p2 <- ggVennDiagram::ggVennDiagram(list(down_inf_art$gene, down_inf_no$gene),
              label_alpha = 0,
              category.names= c("Inf+EFZ", "Inf"),
              #set.size = 20,
              label = "percent",
              label_size = 6) +
  scale_fill_distiller(palette = "RdBu") +
  labs(title = "Genes downregulated in infected cells")

ggarrange(p1, p2, ncol = 2)
```

<details>
<summary>**Volcano Plot of Infection (+ EFZ) signature**</summary>

Interestingly, the genes downregulatd in Inf+EFZ vs. Uninfected cells show
a higher average log fold change with higher significance (p-values). 

```{r, fig.height=8, fig.width=12}
inf_uninf_art_marker %>% 
  rownames_to_column("gene") %>%
  filter(gene %in% c(up_inf_art$gene, down_inf_art$gene)) %>% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = case_when(gene %in% up_inf_art$gene ~ "upregulated",
                               gene %in% down_inf_art$gene ~ "downregulated",
                               TRUE ~ "NA")) %>%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  mutate(label = ifelse(gene %in% c(up_inf_art$gene, down_inf_art$gene), gene, NA)) %>% 
  ggplot(aes(x = avg_log2FC,
             y = -log10(p_val_adj),
             col = condition, 
             label = label)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-50), col = "red") +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = "Genes which are up- or downregulated in Inf+EFZ vs Uninf+EFZ")

```

---
</details>

### Between treatment conditions

Comparing uninfected + EFZ with uninfected and infected + EFZ with infected
iMg it might be possible to identify genes which are different after treatment 
of cells with EFZ. Finding overlapping genes between the two infection states increases the confidence in the results, because the identified genes are associated with treatment independently of infection status.

The diagram below shows, that 36% of genes are upregulated both in infected and
infected + EFZ iMg compared to uninfected and uninfected + EFZ iMg. These genes are probably associated with the EFZ treatment. The same is true for 44% of the downregulated genes. The drug affects the iMg and changes their gene expression. This signature might be used to identify Myeloid2 CSF cells which are affected by the treatment. Since ART drugs are found at lower concentrations in CSF than in blood, probably only a subgroup of Myeloid2 are affected by the ART drugs. It would
be interesting to know which are affected by treatment and which are not. However, this comparison might be farfetched, since the patients from which the CSF samples 
have been derived have not been treated with EFZ, but with a combination of 
other ART drugs which probably have a slightly different effect on cells. 

```{r}
treatment_marker_uninf <- FindMarkers(microglia, ident.1 = "Uninf+EFZ", 
                                ident.2 = "Uninf", verbose = FALSE)

# find marker genes between Inf+EFZ and Inf
treatment_marker_inf <- FindMarkers(microglia, ident.1 = "Inf+EFZ", ident.2 = "Inf")

# overlap between upregulated genes
treatment_up <- intersect((treatment_marker_inf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC > 0))$gene, 
           (treatment_marker_uninf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC > 0))$gene)

treatment_down <- intersect((treatment_marker_inf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC < 0))$gene, 
           (treatment_marker_uninf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC < 0))$gene)
treatment_marker <- c(treatment_up, treatment_down)
```

```{r, fig.width=15}
p1 <- ggVennDiagram::ggVennDiagram(list((treatment_marker_inf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC > 0))$gene, 
             (treatment_marker_uninf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC > 0))$gene),
              label_alpha = 0,
              category.names= c("Inf+EFZ", "Uninf+EFZ"),
              #set.size = 20,
              label = "percent",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "Genes upregulated in cells treated with EFZ")

p2 <- ggVennDiagram::ggVennDiagram(list((treatment_marker_inf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC < 0))$gene, 
             (treatment_marker_uninf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC < 0))$gene),
              label_alpha = 0,
              category.names= c("Inf+EFZ", "Uninf+EFZ"),
              #set.size = 20,
              label = "percent",
              label_size = 6,
              set_color = c("red", "blue")) +
  scale_fill_distiller(palette = "RdBu") +
  labs(title = "Genes downregulated in cells treated with EFZ")

ggarrange(p1, p2, ncol = 2)
```


<details>
<summary>**Volcano Plot of Treatment signature**</summary>

```{r,fig.height=8, fig.width=12}
treatment_marker_uninf %>% 
  rownames_to_column("gene") %>%
  filter(gene %in% treatment_marker) %>% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = case_when(gene %in% treatment_up ~ "upregulated",
                               gene %in% treatment_down ~ "downregulated",
                               TRUE ~ "NA")) %>%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  #mutate(label = ifelse(gene %in% c(up_inf$gene, down_inf$gene), gene, NA)) %>% 
  ggplot(aes(x = avg_log2FC,
             y = -log10(p_val_adj),
             col = condition, 
             label = gene)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-10), col = "red") +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = "Top 20 Genes which are up/downregulated in treatment vs. no treatment, Uninf cells")
```

---
</details>

# Infection & treatment gene signature


Checking the differentially expressed genes and trying to identify an infection signature through literature search turned out to be very difficult. Therefore, 
a different approach was used. The top twenty up- and downregulated genes in 
Inf+EFZ vs Uninf+EFZ were computed. Using these genes as an "infection signature" 
it will be investigated whether some of the Myeloid2 cells from the CSF samples 
show a similar signature.
  

```{r}
hivmy2 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv_myeloid2_subclusters")
# select only genes which are also found in the sc-RNAseq dataset
up_inf_filt <- up_inf_art$gene[up_inf_art$gene %in% rownames(hivmy2@assays$RNA@counts)]
down_inf_filt <- down_inf_art$gene[down_inf_art$gene %in% rownames(hivmy2@assays$RNA@counts)]
```


## Gene signature scoring in Myeloid2 cells

Using the signature genes from differential gene expression analysis between
Inf+EFZ and Uninf+EFZ we can compute a score of activity of these gene sets in 
the Myeloid2 cluster of CSF samples. We can plot these scores in a t-SNE 
representation, but we can also get the scores from the metadata and use them 
for other visualizations. If the Inf+EFZ signature can be found in some of the
Myeloid2 cells from the CSF samples this might provide additional insights 
about the infection status of these microglia-like cells.

#### Scores of gene signature up- or downregulated in Inf+EFZ iMg in Myeloid2 cells

In the plot below (top) there are some cells visible which have a high score
for the signature composed of genes upregulated in Inf+EFZ iMg To be precise,
63% percent of the Myeloid2 cells have a positive score (92 out of 145 cells).
These high scoring cells do, however, not correspond to the two clusters (c1, c2)
obtained from analysis of TF activity (SCENIC).

In the bottom part of the plot below there are some cells with a negative score for 
the downregualated gene signature. To be precise, 58% of of all cells (85 out of 146) have a 
negative score. It would be interesting to know if some of these cells 
are the same as those with a high score for the upregulated genes.
Identifying cells in which both the upregulated gene
signature and the downregulated gene signature have a positive or negative score
respectively increases the confidence in the results. 51 cells have both a 
positive score for the upregulated genes and a negative score for the
downregulated genes. Cells exhibiting this specific pattern of gene expression 
are likely infected cells. Since the infection status of the cells in the CSF 
sample is unknown, this provides valuable new insights.

```{r, fig.width=15}
up_sign <- AddModuleScore(hivmy2,
                           features = list(up_inf_filt),
                          name = "upregulated")

p1 <- FeaturePlot(subset(up_sign, ht_cluster == "c1"),
                  features = "upregulated1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")

p2 <- FeaturePlot(subset(up_sign, ht_cluster == "c2"),
                  features = "upregulated1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c2")

figure <- ggarrange(p1, p2, ncol = 2)
annotate_figure(figure, top = text_grob("Signature score of genes upregulated in Inf+EFZ iMg applied to CSF Myeloid2 cells", 
               color = "black", face = "bold", size = 20))

```


```{r, fig.width=15}

down_sign <- AddModuleScore(hivmy2,
                           features = list(down_inf_filt), name = "downregulated")

p1 <- FeaturePlot(subset(down_sign, ht_cluster == "c1"),
                  features = "downregulated1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")

p2 <- FeaturePlot(subset(down_sign, ht_cluster == "c2"),
                  features = "downregulated1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c2")

figure <- ggarrange(p1, p2, ncol = 2)
annotate_figure(figure, top = text_grob("Signature score of genes downregulated in Inf+EFZ iMg applied to CSF Myeloid2 cells", 
               color = "black", face = "bold", size = 20))

```

```{r}
# divide the number of rows(cells) with negative scores by the number of total 
# rows(cells) to get the percentage of cells with negative scores.
percentage_inf <- dim(down_sign@meta.data %>% filter(downregulated1 < 0))[1] / 
  dim(down_sign@meta.data["downregulated1"])[1]

# extract list of cells with positive/negative scores and find the overlap
cells_down <- rownames(down_sign@meta.data %>% filter(downregulated1 < 0))
cells_up <- rownames(up_sign@meta.data %>% filter(upregulated1 > 0))
inf_efz_cells <- intersect(cells_up, cells_down)
```


<details>
<summary>**Defining a signature composed of downregulated genes**</summary>

Defining a gene signature composed of both up- and downregulated genes turned 
out to be more difficult than expected. The idea was to separately investigate 
upregulated versus downregulated genes. Above the upregulated and downregulated
gene signature scores in Myeloid2 cells were shown. 

We will now have a closer look at the problems that arise when investigating
downregulated genes. Assuming that our Myeloid2 cells are composed of two 
distinct populations, one infected and one uninfected, we would expect a 
bimodal distribution of gene expression for genes upregulated in the infected
condition. There should be one peak with base level gene expression corresponding
to uninfected cells and a second peak with higher levels of gene expression 
corresponding to infected cells. 

In the histograms below the distribution of the sum of log-normalized feature
counts of genes upregulated or downregulated  respectively can be seen. For the
upregulated genes (green) a bimodal distribution is not evident from 
the histogram below. No clear conclusions can be drawn, since the number of
cells is probably too low.

For the downregulated genes (red) there seem to be a large number of cells which do 
not express these genes at all. This group of cells could include infected cells,
however this is a dangerous assumption, since in general there might be cells
with low or none expression of these genes. There is no way to know if these 
cells are actually infected cells.

```{r}
#normalize counts of upregulated genes
counts_up <- (hivmy2@assays$RNA@data)
counts_up <- t(t(counts_up)) / colSums(counts_up)
counts_up <- log1p(counts_up)
counts_up <- counts_up[up_inf_filt, ]

# compute average expression of the 20 signature genes
p1 <- as.data.frame(rowSums(t(counts_up))) %>% 
  rownames_to_column("cell") %>% 
  rename("count" = "rowSums(t(counts_up))") %>% 
  arrange(desc(count)) %>% 
  ggplot(aes(x  = count)) +
  geom_histogram(bins = 70, color = "olivedrab4", fill = "olivedrab4") +
  xlab("normalized feature counts") +
  ylab("number of cells") +
  ylim(0,100) +
  labs(title = "Sum of feature counts of genes upregulated in Inf+EFZ vs. Uninf+EFZ")

#normalize counts of downregulated genes
counts_down <- (hivmy2@assays$RNA@data)
counts_down <- t(t(counts_down)) / colSums(counts_down)
counts_down <- log1p(counts_down)
counts_down <- counts_down[down_inf_filt, ]

# compute average expression of the 20 signature genes
p2 <- as.data.frame(rowSums(t(counts_down))) %>% 
  rownames_to_column("cell") %>% 
  rename("count" = "rowSums(t(counts_down))") %>% 
  arrange(desc(count)) %>% 
  ggplot(aes(x  = count)) +
  geom_histogram(bins = 70, color = "red4", fill = "red4") +
  xlab("normalized feature counts") +
  ylab("number of cells") +
  labs(title = "Sum of feature count of genes downregulated in Inf+EFZ vs. Uninf+EFZ")
```

```{r, fig.width=15}
p1 + p2
```

---

</details>


#### Scores of gene signature up- or downregulated in cells treated with EFZ independently of infection condition.

Of the Myeloid2 cells 70% (103 out of 146 cells) have a score above 0.2 for the
upregulated gene signature (see figure below, top). Only 26% of cells have a
negative score (below 0) for the downregulated gene signature (see figure below, 
bottom). Again, to increase the confidence in the
results, the overlap between the cells with up/downregulated gene signatures
was identified. Only 15 cells have both a positive and negative score for the
up- and downregulated signature respectively.

```{r, fig.width=15}
up_treat <- AddModuleScore(hivmy2,
                           features = list(treatment_up),
                           name = "upregulated")

p1 <- FeaturePlot(subset(up_treat, ht_cluster == "c1"),
                  features = "upregulated1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")

p2 <- FeaturePlot(subset(up_treat, ht_cluster == "c2"),
                  features = "upregulated1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c2")

figure <- ggarrange(p1, p2, ncol = 2)
annotate_figure(figure, top = text_grob("Signature score of genes upregulated in EFZ treatment", 
               color = "black", face = "bold", size = 20))

```


```{r, fig.width=15}
down_treat <- AddModuleScore(hivmy2,
                           features = list(treatment_down),
                           name = "downregulated")

p1 <- FeaturePlot(subset(down_treat, ht_cluster == "c1"),
                  features = "downregulated1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")

p2 <- FeaturePlot(subset(down_treat, ht_cluster == "c2"),
                  features = "downregulated1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c2")

figure <- ggarrange(p1, p2, ncol = 2)
annotate_figure(figure, top = text_grob("Signature score of genes downregulated in EFZ treatment", 
               color = "black", face = "bold", size = 20))

```

```{r}
percentage_up <- dim(up_treat@meta.data %>% filter(upregulated1 > 0.2))[1] /
  dim(up_treat@meta.data)[1]

percentage_down <- dim(down_treat@meta.data %>% filter(downregulated1 < 0))[1] /
  dim(down_treat@meta.data)[1]

cells_down <- rownames(down_treat@meta.data %>% filter(downregulated1 < 0))
cells_up <- rownames(up_treat@meta.data %>% filter(upregulated1 > 0.2))
treat_cells <- intersect(cells_down, cells_up)
```

## Overlay Inf+EFZ signature with the TF activity heatmap of Myeloid2 cells {.tabset}

There are a some cells which express the Inf+EFZ signature and they are highlighted
with an orange colour in the heatmap below. However, they are spread
across all clusters and patients. The first of three clusters in each patient was
hypothesized to be a latently infected group of cells. In the heatmap below the 
cells with an Inf+EFZ signature do not seem to accumulate in the first cluster of 
each patient, but rather are more frequent in the other two clusters.

### k = 3 hierarchical clustering

```{r,fig.width=15, fig.height=20}
# read in Seurat object contianing alls samples and conditions
hiv7 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds")
# import TF activity matrix of  Myeloid2 cells
auc_mtx <- read.table("data_files/aucell_matrix_10k_hvg.tsv", sep = "\t")


# 3 heatmaps with all Scenic TFs
ht_list <- map(seq.int(1:3), function(n) {
  patients <- c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF") # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df),]
  # heatmap
  ht <- Heatmap(t(mtx), 
               column_title = paste0("patient: ", patients[n]), 
               top_annotation = HeatmapAnnotation(signature = case_when(
                 rownames(mtx) %in% inf_efz_cells ~ "Inf_EFZ_signature",
                 #rownames(mtx) %in% uninf_efz_cells ~ "Uninf_EFZ_signature",
                 #rownames(mtx) %in% uninf_cells ~ "Uninf_signature",
                 TRUE ~ "no_signature"),
                 name = "signature",
                 col = list(signature = c("Inf_EFZ_signature" = "orange2",
                                          #"Uninf_EFZ_signature" = "lightblue",
                                          #"Uninf_signature" = "orange",
                                          "no_signature" = "navy"))),
               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
               show_row_names = TRUE
               ) 
               #top_annotation = column_annotation) 
  list(name = patients[n], heatmap = ht)
})

ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap 
```

---

### Grouped by infection condition

When you group the cells according to their different gene signatures as in the 
heatmap below, you do not see a difference between the two groups. Testing
for TFs with different activity between the two groups might provide more 
insights. 

```{r, fig.width=15, fig.height=20}
# filter the dataset for only Myeloid2 cells 
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                orig.ident %in% c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df),]

# add signature column to dataframe
signature <- df %>%
  rownames_to_column("cell") %>% 
  mutate(signature = case_when(cell %in% inf_efz_cells ~ "Inf_signature",
                                #cell %in% uninf_efz_cells ~"Uninf_EFZ_signature",
                                #cell %in% uninf_cells ~ "Uninf_signature",
                               TRUE ~ "no_signature"))


ht <- draw(Heatmap(t(mtx), 
             column_title = "All patients split according to gene signature scores", 
             top_annotation = HeatmapAnnotation(signature = case_when(
               rownames(mtx) %in% inf_efz_cells ~ "Inf_signature",
               #rownames(mtx) %in% uninf_efz_cells ~ "Uninf_EFZ_signature",
               #rownames(mtx) %in% uninf_cells ~ "Uninf_signature",
               TRUE ~ "no_signature"),
               name = "signature",
               col = list(signature = c("Inf_signature" = "orange2",
                                        #"Uninf_EFZ_signature" = "lightblue",
                                        #"Uninf_signature" = "orange",
                                        "no_signature" = "navy"))),       
             column_split = signature$signature,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             show_row_names = TRUE
             ))
```

---


## Overlay EFZ treatment signature with the TF activity heatmap of Myeloid2 cells {.tabset}

There are very few cells with both a positive and negative score for up- and
downregulated signatures respectively. This might be, because the comparison 
between the effect of EFZ treatment on cells in cell culture and cells from the 
CSF of patients treated with a combination of other antiretroviral drugs might 
be farfetched. Apart from the drugs being different, cells in cell culture are 
exposed to comparatively high drug concentrations, while cells in the CSF are
exposed to very low drug concentrations due to the blood-brain barrier. This 
could explain why so few cells with a EFZ treatment signature are found among the 
Myeloid2 cells. 

### k = 3 hierarchical clustering

```{r,fig.width=15, fig.height=20}

# 3 heatmaps with all Scenic TFs
ht_list <- map(seq.int(1:3), function(n) {
  patients <- c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF") # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df),]
  # heatmap
  ht <- Heatmap(t(mtx), 
               column_title = paste0("patient: ", patients[n]), 
               top_annotation = HeatmapAnnotation(signature = case_when(
                 rownames(mtx) %in% treat_cells ~ "EFZ_treatment_signature",
                 #rownames(mtx) %in% uninf_efz_cells ~ "Uninf_EFZ_signature",
                 #rownames(mtx) %in% uninf_cells ~ "Uninf_signature",
                 TRUE ~ "no_signature"),
                 name = "signature",
                 col = list(signature = c("EFZ_treatment_signature" = "olivedrab3",
                                          #"Uninf_EFZ_signature" = "lightblue",
                                          #"Uninf_signature" = "orange",
                                          "no_signature" = "navy"))),
               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
               show_row_names = TRUE
               ) 
               #top_annotation = column_annotation) 
  list(name = patients[n], heatmap = ht)
})

ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap 
```

---

### Grouped by treatment condition

```{r, fig.width=15, fig.height=20}
# filter the dataset for only Myeloid2 cells 
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                orig.ident %in% c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df),]

# add signature column to dataframe
signature <- df %>%
  rownames_to_column("cell") %>% 
  mutate(signature = case_when(cell %in% treat_cells ~ "EFZ_treatment_signature",
                                #cell %in% uninf_efz_cells ~"Uninf_EFZ_signature",
                                #cell %in% uninf_cells ~ "Uninf_signature",
                               TRUE ~ "no_signature"))


ht <- draw(Heatmap(t(mtx), 
             column_title = "All patients split according to gene signature scores", 
             top_annotation = HeatmapAnnotation(signature = case_when(
               rownames(mtx) %in% treat_cells ~ "EFZ_treatment_signature",
               #rownames(mtx) %in% uninf_efz_cells ~ "Uninf_EFZ_signature",
               #rownames(mtx) %in% uninf_cells ~ "Uninf_signature",
               TRUE ~ "no_signature"),
               name = "signature",
               col = list(signature = c("EFZ_treatment_signature" = "olivedrab3",
                                        #"Uninf_EFZ_signature" = "lightblue",
                                        #"Uninf_signature" = "orange",
                                        "no_signature" = "navy"))),       
             column_split = signature$signature,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             show_row_names = TRUE
             ))
```

---

# Differential TF activity {.tabset}

After having identified a few cells with a high scores for the infection signature
(Inf+EFZ), as well as cells with a high score for the EFZ treatment signatures,
it would be interesting to know whether some TFs have different activity between
the conditions. For this a Kruskal-Wallis Test was used, followed by a
pairwise Wilcox Test. 

## Between infection conditions

Out of all 125 TFs, 44 have different activity between cells with a high score for 
the Inf+EFZ signature compared to the rest. The activities for those TFs which
are also found to have a differential binding scores in C20 microglia are 
shown in the boxplots below. KLF4 and REST are the only two TFs more bound in
latently infected C20 microglia. KLF4 is more active in Myeloid2 with a high
score for the Inf+EFZ signature, but REST is less active. All other TFs are more 
bound in uninfected C20 microglia than in latently infected, but the TF activity
is not always higher in uninfected Myeloid2 (no high score for Inf+EFZ signature).

```{r}
hivmy2_scenic <- AddMetaData(hivmy2, 
                    metadata = auc_mtx, 
                    col.name = colnames(auc_mtx))

inf_sign_df <- hivmy2_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  mutate(signature = case_when(cell %in% inf_efz_cells ~ "Inf_EFZ",
                                TRUE ~ "no_sign"))

# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(inf_sign_df[tf]) ~ signature, inf_sign_df))["p.value"]
})

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")

# how many transcription factors are significantly different between the clusters?
sign <- adj_p_values[adj_p_values < 0.05]
length(sign)
diff_cells <- colnames(auc_mtx)[adj_p_values < .05]

# get only the TFs also found in footprinting

# read in footprinting resutls
tfs_ana <- read.table("/media/ag-cherrmann/projects/06_HIV_Microglia/data/atacseq/data-2020-11-06/tobias/TOBIAS_snakemake/footprint_mglia2_GlassTF_17-03/TFBS/bindetect_results.txt", sep = "\t", header = TRUE)

# filter TFs with significant p values between latent infection and uninfected
# I decided to use a threshold of p-value < 1e-50, because due to the large number 
# of data, all p-values will be very high
top_tfs <- tfs_ana %>% filter(uninf_red_pvalue < 1e-50)
down_tfs <- top_tfs %>% filter(uninf_red_change < 0)
#down_tfs <- top_tfs %>% filter(uninf_red_change < )

diff_tfs <- diff_cells[diff_cells %in% top_tfs$name]

diff_cells[diff_cells %in% down_tfs$name]
```



```{r, fig.width=15, fig.height=12}
plots <- map(diff_tfs, function(tf){
  ggplot() +
    geom_violin(aes(x = inf_sign_df %>% pull("signature"), 
                     y = inf_sign_df %>% pull(tf),
                     fill = inf_sign_df %>% pull("signature"))) +
    geom_boxplot(aes(x = inf_sign_df %>% pull("signature"), 
                     y = inf_sign_df %>% pull(tf)),
                 alpha = 0) +
    stat_pvalue_manual(inf_sign_df %>% 
                         pairwise_wilcox_test(
                           as.formula(paste0(tf, " ~ ht_cluster"))) %>% 
                         add_xy_position()) +
    xlab("condition") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "condition"))
  })

do.call(grid.arrange, c(plots, ncol = 3, nrow = 4))
```

---

## Between treatment conditions

Are some TFs in Myeloid2 differentially active between the treatment conditions?
There are no significant differences in TF activity between the cells with 
high scores for EFZ treatment signature compared to cells without the signature.
Therefore, no posthoc analysis was performed.


```{r}
df <- hivmy2_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
        # filter(cell %in% c(inf_efz_cells, uninf_efz_cells, uninf_cells)) %>% 
  mutate(signature = case_when(cell %in% treat_cells ~ "treatment_signature",
                                TRUE ~ "no_signature"))

# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(df[tf]) ~ signature, df))["p.value"]
})

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")

# how many transcription factors are significantly different between the clusters?
sign <- adj_p_values[adj_p_values < 0.05]
```

---

## Gene expression of inflammatory genes 

[@Ryan] showed that two pathways are specifically activated in the Inf+EFZ 
condition, namely RhoGDI signaling and CD40 signaling. This effect was not seen 
in Uninf+EFZ, so it does not seem to be caused by EFZ alone, 
but rather a combination of infection and EFZ treatment [@Ryan].

* RhoGDI signaling:
  + PP1R12A was upregulated 
  + RND3, CD44, ACTG1 were downregulated
* CD40 signaling:
  + FOS was upregulated
  + NFKBIA, ICAM1, TFNAIP3 were downregulated 
  
Here, we would like to see if cells with high scores for the Inf+EFZ signature
show gene expression patterns that are in line with the findings by [@Ryan]. 
The gene expression for each gene is shown in the figure below.
We would expect only PP1R12A and FOS to be more highly expressed in the cells
with Inf+EFZ signature, while all other genes should be less highly expressed.
This pattern in gene expression is, however, not observed. 

```{r, fig.width=15, fig.height=8}
# read in Seurat object containing Myeloid2 cells
hivmy2_scenic <- AddMetaData(hivmy2_scenic, metadata = inf_sign_df$signature, col.name = "signature")

# RhoGDI signaling in myeloid2
p1 <- VlnPlot(hivmy2_scenic, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
              group.by = "signature", ncol = 2)

# CD40 signaling in myeloid2
p2 <- VlnPlot(hivmy2_scenic, features =c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
              group.by = "signature", ncol = 2)

# RhoGDI signaling in iMg
p3 <- VlnPlot(microglia, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

# CD40 signaling in iMg
p4 <- VlnPlot(microglia, features = c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

fig <- ggarrange(p1, p3, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob("Expression of four genes of the RhoGDI signaling pathway in Myeloid2 (left column) and iMg (right column)",
                                     size = 20 , face = "bold"))

fig <- ggarrange(p2, p4, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob("Expression of four genes of the CD40 signaling pathway in Myeloid2 (left column) and iMg (right column)",
                                     size = 20 , face = "bold"))

```

Only FOS which encodes a TF is found in the TF activity matrix, so for FOS 
the TF acivity is shown as well. Judging from the figure below FOS seems to be 
more active in cells with high score for Inf+EFZ signature, but this difference
is not statistically significant.

```{r}
inf_sign_df %>% ggplot(aes(x = signature, y = FOS)) +
  geom_violin(aes(fill = signature, alpha = .6)) +
  geom_boxplot(alpha = 0) +
  stat_pvalue_manual(inf_sign_df %>% 
                       pairwise_wilcox_test(FOS ~ signature) %>% 
  add_xy_position())

```

---

# Appendix


## De-novo clustering 

De-novo clustering of the myeloid-lineage cluster of the integrated dataset was
performed, but did not yield any additional insights. Isolating the iMg and all Myeloid cell clusters from the integrated  dataset and normalizing, scaling and re-clustering these isolated cells might have  
provided additional insights. 



<details>
<summary>Details on de-novo clustering</summary>


```

microglia <- subset(combined, cell_type %in% c("iMg",
                                                "Myeloid1",
                                                "Myeloid2",
                                                "Myeloid3",
                                                "Myeloid4",
                                                "Myeloid5"))
microglia_recluster <- SCTransform(microglia, verbose = FALSE)
microglia_recluster <- ScaleData(microglia_recluster, verbose = FALSE)
microglia_recluster <- RunPCA(microglia_recluster, verbose = FALSE)
ElbowPlot(microglia_recluster)
microglia_recluster <- RunTSNE(microglia_recluster, reduction = "pca", dims = 1:10)
microglia_recluster <- FindNeighbors(microglia_recluster)
microglia_recluster <- FindClusters(microglia_recluster, resolution = 0.5)
```

---
</details> 


It is no surprise that cells from the triculture dataset separate from the 
CSF/Blood-derived cells which can be seen in the plot below. When reclustering the myeloid lineage derived cells,five subclusters for the iMg and three subclusters for the five CSF/Blood Myeloid clusters are obtained. It can also be observed that
the Myeloid2 and Myeloid5 separate from Myeloid1/2/3 which is expected, since Myeloid2/5 originate from CSF samples and Myeloid1/3/4 originate from blood samples.

```{r, fig.width=15, fig.height=12}
microglia_recluster <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/microglia_myeloid_seurat_object_reclustering")
a1 <- DimPlot(microglia_recluster, reduction = "tsne",
              group.by = "seurat_clusters", pt.size = .4) +
  labs(title = "Clustering")
a2 <- DimPlot(microglia_recluster, reduction ="tsne", 
              group.by = "cell_type", pt.size = .4) +
  labs(title = "Cell type")
a3 <- DimPlot(microglia_recluster, reduction ="tsne",
              group.by = "dataset", pt.size = .4) +
    labs(title = "Dataset")
a4 <- DimPlot(microglia_recluster, reduction = "tsne",
              group.by = "orig.ident", pt.size = .4) +
  labs(title = "Sample origin")

fig <- ggarrange(a1, a2, a3, a4, ncol = 2, nrow = 2)
annotate_figure(fig, 
                top = text_grob("De-novo clustering of myeloid-lineage clusters", 
               color = "black", face = "bold", size = 20))
```


Interestingly, both the Myeloid5 cells and the Myeloid2 cells are split between 
the cluster six (blue) and seven (purple) in the figure below. In the following
we will have a closer look at this segregation. When performing GRN inference 
with Scenic we also identified two potentially biologically relevant subclusters of
the Myeloid2 cells. It could be that these independently found clusters correlate. 

```{r, fig.width=15, fig.height=10}
a5 <- DimPlot(microglia_recluster, reduction ="tsne", 
              split.by = "cell_type", pt.size = .4, ncol=3) +
  labs(title = "Contribution of different cell types to different myeloid clusters")
a5
```

It looks like the two clusters identified this way correspond to different
sample origins. CSF cells from patient1 (HIV1_CSF) and patient2 (HIV2_CSF)
belong to cluster six, while CSF cells from patient3 (HIV3_CSF) and the two 
uninfected patients (Uninfected1_CSF, Uninfected2_CSF) belong to 
cluster seven. The separation of patient3 (HIV3_CSF) from patient1 (HIV1_CSF) 
and patient2 (HIV2_CSF) might correspond to a batch effect, or it might be a 
biological effect, for example, the infection might have affected the cells differently.


```{r, fig.width=15, fig.height=7}
#a1 <- DimPlot(subset(microglia_recluster, cell_type == "iMg"), reduction = "tsne",
           #   group.by = "orig.ident", pt.size = .4)
a2 <- DimPlot(subset(microglia_recluster, cell_type != "iMg"), reduction = "tsne",
              group.by = "orig.ident", pt.size = .4) +
  labs(title = "Sample origin")
#a3 <- DimPlot(subset(microglia_recluster, cell_type == "iMg"), reduction = "tsne",
              #group.by = "seurat_clusters", pt.size = .4)
a4 <- DimPlot(subset(microglia_recluster, cell_type != "iMg"), reduction = "tsne",
              group.by = "seurat_clusters", pt.size = .4) +
  labs(title = "Cluster")
fig <- ggarrange(a2, a4, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob("CSF/Blood Myeloid cells", 
                                     face = "bold", size = 20))
```

Since the segregation into two clusters seems to be related to the sample origin
rather than to the infection status, we would not expect the clusters to overlap
with the clusters identified after GRN inference with SCENIC. In the figure 
below it can be seen that the clusters do not overlap (only the Myeloid2 cells from infected CSF samples (HIV1_CSF, HIV2_CSF, HIV3_CSF) are shown). For this reason the reclustering did not lead to any useful insights and will 
not be analyzed further. 

```{r, fig.width=15, figh.height = 5}
# read in Myeloid2 cells
hivmy2 <- readRDS("data_files/hiv_myeloid2_subclusters")

my2 <- subset(microglia_recluster, cell_type == "Myeloid2" & 
                orig.ident %in% c("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))

my2 <- AddMetaData(my2, hivmy2@meta.data["ht_cluster"], 
                   col.name = "TF_activity_clusters")
p1 <- DimPlot(my2, group.by = "seurat_clusters") +
  labs(title="Clusters")
p2 <- DimPlot(my2, group.by = "TF_activity_clusters") +
  labs(title = "TF activity")
p3 <- DimPlot(my2, group.by = "orig.ident") +
  labs(title = "Sample origin")
figure <- ggarrange(p1, p2, p3, ncol = 3, nrow = 1)
annotate_figure(figure, 
                top = text_grob("Comparison with clusters of different TF acitivty", 
               color = "black", face = "bold", size = 20))
#my25 <- subset(microglia, seurat_clusters %in% c(6,7) &cell_type %in% c("Myeloid2", "Myeloid5"))

```


## Inflammatory Pathways

As already mentioned above [@Ryan] showed that two pathways are specifically activated in the Inf+EFZ 
condition, namely RhoGDI signaling and CD40 signaling. This effect was not seen 
in Uninf+EFZ, so it does not seem to be caused by EFZ alone, 
but rather a combination of infection and EFZ treatment [@Ryan].

* RhoGDI signaling:
  + PP1R12A was upregulated 
  + RND3, CD44, ACTG1 were downregulated
* CD40 signaling:
  + FOS was upregulated
  + NFKBIA, ICAM1, TFNAIP3 were downregulated 
  
To answer the question whether we could find similar patterns in the two Myeloid2 clusters
with different TF activity, we had a look at the gene expression.

##### Myeloid2 cells 

Two genes of the RhoGDI signaling pathway were found among the differentially
expressed genes between Inf+EFZ and Uninf+EFZ. Both of them were downregulated
in Inf+EFZ compared to Uninf+EFZ iMg as can be seen in the table below.

```{r, results = "asis"}
inf_uninf_art_marker %>% 
  rownames_to_column("gene") %>%
  filter(gene %in% c("RND3", "CD44", "ACTG1", "PPP1R12A")) %>%
  kable(caption = "Genes of the RhoGDI signaling pathway found among genes differentially expressed between Inf+EFZ and Uninf+EFZ")
```

All four genes of the CD40 signaling were found among the differentially expressed
genes between Inf+EFZ and Uninf+EFZ. FOS was upregulated, while NFKBIA, ICAM1 and 
TFNAIP3 were downregulated in Inf+EFZ compared to Uninf+EFZ as can be seen in the
table below. 

```{r, results = "asis"}
inf_uninf_art_marker %>% 
  rownames_to_column(var = "gene") %>%
  filter(gene %in% c("FOS", "NFKBIA", "ICAM1", "TNFAIP3")) %>% kable()

```


The changes in gene expression of both pathways are in line with the findings 
by [@Ryan]. 

Additionally, we were interested whether this pattern in gene expression 
correlated with the two clusters identified in Myeloid2 which exhibit different
TF activities. The figure below show the gene expression of the four genes of 
the respective pathways both in Myeloid2 cells (left column) and in iMg (right 
column).

FOS is the only gene which shows a high expression in Myeloid2 cluster c1. 
This supports our hypothesis that c1 might be a latently infected reactive microglia 
cell state. However then we would expect to see a similar pattern for the RhoGDI
signaling, meaning PPP1R12A should be the only gene which is
highly expressed in c1. This is not the case, but rather CD44 and ACTG1 also 
show higher expression in c1 compared to c2. Consequently, our hypothesis is not 
clearly supported by the expression patterns of these inflammatory genes.

```{r, fig.width=15, fig.height=6}
# isolate microglia cells
microglia <- subset(combined, cell_type == "iMg")

# read in Seurat object containing Myeloid2 cells
hivmy2 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv_myeloid2_subclusters")


p1 <- VlnPlot(hivmy2, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
              group.by = "ht_cluster", ncol = 2)

p2 <- VlnPlot(hivmy2, features =c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
              group.by = "ht_cluster", ncol = 2)

p3 <- VlnPlot(microglia, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

p4 <- VlnPlot(microglia, features = c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

fig <- ggarrange(p1, p3, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob("Expression of four genes of the RhoGDI signaling pathway in Myeloid2 (left column) and iMg (right column)",
                                     size = 20 , face = "bold"))

fig <- ggarrange(p2, p4, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob("Expression of four genes of the CD40 signaling pathway in Myeloid2 (left column) and iMg (right column)",
                                     size = 20 , face = "bold"))
```


<details>
<summary>Same two pathways in Myeloid5 cells</summary>

Myeloid5 cells


```{r, fig.width=15, fig.height=12}

# isolate the Myeloid5 cluster
hivmy5 <- subset(hiv7, cell_type == "Myeloid5" & orig.ident %in% c("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))

# process Myeloid 5 Seurat object
hivmy5 <- ScaleData(hivmy5, verbose = FALSE)
hivmy5 <- NormalizeData(hivmy5, verbose = FALSE)
hivmy5 <- ScaleData(hivmy5, verbose = FALSE)
hivmy5 <- FindVariableFeatures(hivmy5, verbose = FALSE)
hivmy5 <- RunPCA(hivmy5, features = VariableFeatures(hivmy5), verbose = FALSE)
#ElbowPlot(hivmy5)
hivmy5 <- RunTSNE(hivmy5, dims= 1:10, verbose = FALSE)
```

```{r, fig.width=15, fig.height=12}

p1 <- VlnPlot(hivmy5, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
              group.by = "orig.ident",ncol = 2)

p2 <- VlnPlot(hivmy5, features =c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
              group.by = "orig.ident", ncol = 2)

p3 <- VlnPlot(microglia, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

p4 <- VlnPlot(microglia, features = c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

ggarrange(p2, p4, p1, p3, ncol = 2, nrow = 2)
```

</details>

# References


</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("Triculture_csf_integration.Rmd");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
