<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Katharina Mikulik" />


<title>TF activity in iMg from triculture dataset</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Report</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Introduction.html">Introduction</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="HIV_CSF_dataset_analysis.html">1. CSF/Blood samples</a>
    </li>
    <li>
      <a href="TF_activity_analysis_of_myeloid2_cluster.html">2. TF activity Myeloid2</a>
    </li>
    <li>
      <a href="Triculture_dataset_analysis_and_integration_Farhadian_dataset.html">3. Integration CSF/Blood &amp; triculture</a>
    </li>
    <li>
      <a href="Triculture_dataset_GRN_with_Scenic.html">4. TF activity iMg</a>
    </li>
  </ul>
</li>
<li>
  <a href="Discussion.html">Discussion</a>
</li>
<li>
  <a href="Methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Katharina782/report">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">TF activity in iMg from triculture dataset</h1>
<h4 class="author">Katharina Mikulik</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#gene-regulatory-network-inference-with-scenic">Gene regulatory network inference with SCENIC</a></li>
<li><a href="#bulk-atacseq-footprinting-c20-cell-line">Bulk ATACseq footprinting, C20 cell line</a></li>
<li><a href="#tf-activity-in-img">TF activity in iMg</a>
<ul>
<li><a href="#heatmap-for-all-tfs">Heatmap for all TFs</a></li>
<li><a href="#heatmap-of-25-tfs-with-differential-binding-score">Heatmap of 25 TFs with differential binding score</a>
<ul>
<li><a href="#hierarchical-clustering-with-different-k">Hierarchical clustering with different k</a>
<ul>
<li><a href="#wss-plot">wss plot</a></li>
<li><a href="#different-k">different k</a></li>
<li><a href="#k-4">k = 4</a></li>
<li><a href="#section"></a></li>
</ul></li>
<li><a href="#split-heatmaps-to-find-patterns">Split heatmaps to find patterns</a>
<ul>
<li><a href="#separate-heatmaps-according-to-infection-condition">Separate Heatmaps according to infection condition</a></li>
<li><a href="#split-heatmaps-according-to-tf-binding-scores-from-atacseq-data">Split Heatmaps according to TF binding scores from ATACseq data</a></li>
</ul></li>
</ul></li>
<li><a href="#statistical-tests-to-find-tf-activity-differences-between-clusters">Statistical Tests to find TF activity differences between clusters</a></li>
<li><a href="#posthoc-analysis">Posthoc analysis</a>
<ul>
<li><a href="#more-bound-in-latent-infection">more bound in latent infection</a></li>
<li><a href="#more-bound-in-uninfected">more bound in uninfected</a></li>
</ul></li>
<li><a href="#correlations-between-conditions">Correlations between conditions</a>
<ul>
<li><a href="#correlation-matrix">Correlation matrix</a></li>
<li><a href="#scatter-matrix">Scatter matrix</a></li>
</ul></li>
<li><a href="#correlation-between-tfs-differentiating-myeloid2-clusters">Correlation between TFs differentiating Myeloid2 clusters</a>
<ul>
<li><a href="#correlation-matrix-1">Correlation Matrix</a></li>
<li><a href="#scatter-matrix-1">Scatter Matrix</a></li>
</ul></li>
</ul></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>report/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211203code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211203)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211203code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211203)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttreefdf25ce5decf9dff024fb0470f591b4a70f6aee6targetblankfdf25cea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/Katharina782/report/tree/fdf25ce5decf9dff024fb0470f591b4a70f6aee6" target="_blank">fdf25ce</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKatharina782reporttreefdf25ce5decf9dff024fb0470f591b4a70f6aee6targetblankfdf25cea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/Katharina782/report/tree/fdf25ce5decf9dff024fb0470f591b4a70f6aee6" target="_blank">fdf25ce</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/Independent_cell_type_identification_Farhadian_dataset_cache/
    Ignored:    analysis/Triculture_dataset_analysis_and_integration_Farhadian_dataset_cache/

Untracked files:
    Untracked:  README.html

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Triculture_dataset_GRN_with_Scenic.Rmd</code>) and HTML (<code>docs/Triculture_dataset_GRN_with_Scenic.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/0a10c5865c15bdc0867dc5c069dc4cfb7703f73c/docs/Triculture_dataset_GRN_with_Scenic.html" target="_blank">0a10c58</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/bdbb921b65681909caf7252afe838d8b807681c3/docs/Triculture_dataset_GRN_with_Scenic.html" target="_blank">bdbb921</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/efc0f94be22d7f357aeb93d42cfe3cec476f700a/analysis/Triculture_dataset_GRN_with_Scenic.Rmd" target="_blank">efc0f94</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
<td>
wflow_publish(c(“analysis/TF_activity_analysis_of_myeloid2_cluster.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/9597c36bc5a6df5f071249d97949566b2c7db870/docs/Triculture_dataset_GRN_with_Scenic.html" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/ea7e899746b03ca612d93ad003c875fc101d2329/analysis/Triculture_dataset_GRN_with_Scenic.Rmd" target="_blank">ea7e899</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/6783264e95c3864ae6a468bdfed32b21af801b6d/docs/Triculture_dataset_GRN_with_Scenic.html" target="_blank">6783264</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/9e2c322b7e27fe7694e862cf4b11ae612816498d/analysis/Triculture_dataset_GRN_with_Scenic.Rmd" target="_blank">9e2c322</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
wflow_publish(c(“analysis/Triculture_dataset_GRN_with_Scenic.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/40651d96647f10219a4aec953c8d618d57bc13f5/docs/Triculture_dataset_GRN_with_Scenic.html" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/Triculture_dataset_GRN_with_Scenic.html" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/f7243f8882ef1868800d5f0e420f83d5e4ba39bb/analysis/Triculture_dataset_GRN_with_Scenic.Rmd" target="_blank">f7243f8</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/f7243f8882ef1868800d5f0e420f83d5e4ba39bb/docs/Triculture_dataset_GRN_with_Scenic.html" target="_blank">f7243f8</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
<td>
wflow_publish(all = TRUE, republish = TRUE, delete_cache = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/8b791e32c4621d66dad520814dea9593d4ae12a6/docs/Triculture_dataset_GRN_with_Scenic.html" target="_blank">8b791e3</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/Triculture_dataset_GRN_with_Scenic.html" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/114a941064f12efce14ce5eb1f7cc94e6a004074/docs/Triculture_dataset_GRN_with_Scenic.html" target="_blank">114a941</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Katharina782/report/9ee308625fb863ad709edf2660babe4de93ced91/docs/Triculture_dataset_GRN_with_Scenic.html" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Katharina782/report/blob/e87fa963caeed78821ac7f482359072ce58b8529/analysis/Triculture_dataset_GRN_with_Scenic.Rmd" target="_blank">e87fa96</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-16
</td>
<td>
new files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style>
body {
text-align: justify}
</style>
<pre class="r"><code>library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggplot2)
library(dplyr)
library(ggrepel)
#library(harmony)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(corrplot)
library(rstatix)
library(knitr)
library(metap)
library(multtest)
library(readxl)</code></pre>
<div id="gene-regulatory-network-inference-with-scenic" class="section level2">
<h2>Gene regulatory network inference with SCENIC</h2>
<p>To infer GRNs from iMg cells from the triculture dataset <span class="citation">[@Ryan]</span> (4333 cells in total) I again used SCENIC.</p>
<p>When referring to differentially bound TFs in the C20 cell line in the following I am referring to differential binding scores between infection conditions obtained from bulk ATACseq data from C20 cell line cultures using TOBIAS.</p>
<p>I first tried to run SCENIC with 3000 most highly variable genes (HVG), but then I only got an overlap with differentially bound TFS of eight. When using 10,000 HVG SCENIC returns 232 TFS and their corresponding activity score for each cell. 25 of these TFs overlap with differentially bound TFs in the C20 cell line.</p>
<pre class="r"><code># read in triculture Seurat object
triculture &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_clean_seurat_object&quot;)

# read in CSF HIV Seurat object
hiv7 &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds&quot;)

imicroglia &lt;- subset(triculture, cell_type == &quot;iMg&quot;)</code></pre>
<details>
<summary>
<strong>preparing matrix for SCENIC</strong>
</summary>
<p>First, I use SCTransform for normalization of the expression data, then the top 10,000 HVGs are selected and the matrix transposed, because the pySCENIC pipeline used requires an in input matrix which has genes as columns and cells as rows. (<a href="https://github.com/hdsu-bioquant/pySCENIC_pipeline/blob/master/input/gene_exp_mtx_example.tsv" class="uri">https://github.com/hdsu-bioquant/pySCENIC_pipeline/blob/master/input/gene_exp_mtx_example.tsv</a>)</p>
<pre><code># use 10 000 most highly variable genes
scenic &lt;- SCTransform(imicroglia, variable.features.n = 10000, verbose=FALSE)
saveRDS(scenic, &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_scenic&quot;)
scenic &lt;- readRDS(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_scenic&quot;)
scenic_hvg &lt;- subset(scenic, features = scenic@assays$SCT@var.features)
scenic_mtx &lt;- t(scenic_hvg@assays$SCT@scale.data)
write.table(x = scenic_mtx, file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/pySCENIC_pipeline/input/triculture_mtx.tsv&quot;, sep = &quot;\t&quot;)

#scenic_matrix &lt;- read.table(file = &quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/pySCENIC_pipeline/input/triculture_mtx.tsv&quot;, sep = &quot;\t&quot;)</code></pre>
<hr />
</details>
<details>
<summary>
<strong>AuCell matrix - output from Scenic</strong>
</summary>
<p>The output from SCENIC is a matrix containing cells as rows and 232 TFs as columns. Five exemplary rows and columns are shown below. The TF activity for each cell can be used to draw conclusions about different cell states.</p>
<pre class="r"><code>auc_mtx &lt;- read.csv(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/pySCENIC_output/aucell_img_10k.csv&quot;)
auc_mtx &lt;- auc_mtx %&gt;% column_to_rownames(&quot;Cell&quot;)
auc_mtx &lt;- auc_mtx %&gt;% setnames( old = colnames(auc_mtx), new = sapply(strsplit(
  colnames(auc_mtx), &quot;[...]&quot;), `[`, 1))

auc_mtx[1:5, 1:5] %&gt;% kable(caption = &quot;Layout of TF activity matrix&quot;)</code></pre>
<table>
<caption>Layout of TF activity matrix</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">ARID3A</th>
<th align="right">ARNT</th>
<th align="right">ARNTL2</th>
<th align="right">ASCL2</th>
<th align="right">ATF2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ca_AAACCTGGTGTTCGAT-1</td>
<td align="right">0.0337910</td>
<td align="right">0.0864444</td>
<td align="right">0.063000</td>
<td align="right">0.0002</td>
<td align="right">0.01125</td>
</tr>
<tr class="even">
<td align="left">ca_AAACCTGGTTGTACAC-1</td>
<td align="right">0.0200299</td>
<td align="right">0.0237778</td>
<td align="right">0.024625</td>
<td align="right">0.0629</td>
<td align="right">0.00000</td>
</tr>
<tr class="odd">
<td align="left">ca_AAACGGGAGAATTCCC-1</td>
<td align="right">0.0436418</td>
<td align="right">0.0133333</td>
<td align="right">0.006375</td>
<td align="right">0.0000</td>
<td align="right">0.11100</td>
</tr>
<tr class="even">
<td align="left">ca_AAACGGGAGACAGACC-1</td>
<td align="right">0.0314925</td>
<td align="right">0.0000000</td>
<td align="right">0.063625</td>
<td align="right">0.0804</td>
<td align="right">0.00000</td>
</tr>
<tr class="odd">
<td align="left">ca_AAACGGGTCACATACG-1</td>
<td align="right">0.0402687</td>
<td align="right">0.0971111</td>
<td align="right">0.067625</td>
<td align="right">0.0000</td>
<td align="right">0.00000</td>
</tr>
</tbody>
</table>
<hr />
</details>
<details>
<summary>
<strong>Number of iMg in each condition</strong>
</summary>
<pre class="r"><code># add auc matrix to metadata of hiv4 seurat object
tri_mg &lt;- AddMetaData(imicroglia, 
                    metadata = auc_mtx, 
                    col.name = colnames(auc_mtx))</code></pre>
<pre class="r"><code>tri_mg@meta.data %&gt;% 
  rename(condition  = orig.ident) %&gt;%
  group_by(condition) %&gt;%
  summarize(n = n()) %&gt;% 
  kable(caption = &quot;Number of cells in each condition&quot;)</code></pre>
<table>
<caption>Number of cells in each condition</caption>
<thead>
<tr class="header">
<th align="left">condition</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Inf</td>
<td align="right">1166</td>
</tr>
<tr class="even">
<td align="left">Inf+EFZ</td>
<td align="right">1382</td>
</tr>
<tr class="odd">
<td align="left">Uninf</td>
<td align="right">1163</td>
</tr>
<tr class="even">
<td align="left">Uninf+EFZ</td>
<td align="right">722</td>
</tr>
</tbody>
</table>
<hr />
</details>
</div>
<div id="bulk-atacseq-footprinting-c20-cell-line" class="section level1">
<h1>Bulk ATACseq footprinting, C20 cell line</h1>
<p>The C20 cell line was derived from an HIV+ patient treated with ART (Antiretroviral Therapy). Since the access of ART to the brain is limited due to the blood-brain barrier, the effects of ART on these cells are probably negligible. Only the 75 TFs with p-values for differential binding scores below 1e-50 were used. We will focus on TFs which show differential binding scores between latent infection and uninfected conditions in primary microglia. For 25 of these 75 TFs there is a TF activity score for the iMg. Focusing on these 25 TFs we might be able to find activity patterns that correlate with the TF binding scores. Five TFs are more bound in latent infection and twenty TFs are more bound in uninfected cells</p>
<p>The reason why we find more of the TFs more bound in uninfected cells in the TF activity score matrix from SCENIC is unclear.</p>
<pre class="r"><code>tfs_ana &lt;- read.table(&quot;/media/ag-cherrmann/projects/06_HIV_Microglia/data/atacseq/data-2020-11-06/tobias/TOBIAS_snakemake/footprint_mglia2_GlassTF_17-03/TFBS/bindetect_results.txt&quot;, sep = &quot;\t&quot;, header = TRUE)

top_tfs &lt;- tfs_ana %&gt;% filter(uninf_red_pvalue &lt; 1e-50)

# keep only the TFs in the dataframe which overlap between the two datasets.
overlap_atac &lt;- top_tfs %&gt;% filter(name %in% colnames(auc_mtx))

# dataframe containing only the 25 Tfs whcih overlap with TF activity matrix from SCENIC 
# adding a column with information in which infection condition the TF is more bound
atac_tfs_df &lt;- top_tfs %&gt;% filter(name %in% colnames(auc_mtx)) %&gt;% 
  mutate(condition = ifelse(uninf_red_change &gt; 0, &quot;up_uninf&quot;, &quot;up_latent&quot;))

# List of 5 TFs more bound in latent infection
up_latent &lt;- atac_tfs_df$name[atac_tfs_df$condition == &quot;up_latent&quot;]
# list of 25 TFs more bound in uninfected cells
up_uninf &lt;- atac_tfs_df$name[atac_tfs_df$condition == &quot;up_uninf&quot;]</code></pre>
<details>
<summary>
<strong>Volcano Plot ATAC-seq footprinting</strong>
</summary>
<pre class="r"><code>top_tfs %&gt;% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = ifelse(uninf_red_change &lt; 0, &quot;up_latent&quot;, &quot;up_uninf&quot;)) %&gt;%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  mutate(label = ifelse(name %in% overlap_atac$name, name, NA)) %&gt;% 
  ggplot(aes(x = uninf_red_change,
             y = -log10(uninf_red_pvalue),
             col = condition, 
             label = label)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-50), col = &quot;red&quot;) +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = &quot;TFs with differential binding scores&quot;) +
  ylab(&quot;-log10(p_value)&quot;) +
  xlab(&quot;differential binding score&quot;)
Warning: Removed 50 rows containing missing values (geom_text).</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-7-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9597c36bc5a6df5f071249d97949566b2c7db870/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-7-1.png" target="_blank">9597c36</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-7-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-7-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</details>
</div>
<div id="tf-activity-in-img" class="section level1">
<h1>TF activity in iMg</h1>
<div id="heatmap-for-all-tfs" class="section level2">
<h2>Heatmap for all TFs</h2>
<p>In the heatmap of TF activities below you can already see a trend towards a separation of cells treated with EFZ (blue and green) from untreated cells (red and orange).</p>
<details>
<summary>
<strong>Heatmap of 232 TFs and their activity in iMg</strong>
</summary>
<pre class="r"><code>conditions &lt;- tri_mg@meta.data %&gt;%
                    rownames_to_column(&quot;cell&quot;) %&gt;%
                    select(&quot;cell&quot;, &quot;orig.ident&quot;)%&gt;% column_to_rownames(&quot;cell&quot;)


ht &lt;- Heatmap(t(auc_mtx), 
      col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
      show_column_names = FALSE,
      row_names_side = &quot;left&quot;,
      top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                         name = &quot;condition&quot;, col =  
                                         list(condition = c(&quot;Uninf&quot; = &quot;orange&quot;, 
                                                            &quot;Inf+EFZ&quot; = &quot;green&quot;,
                                                            &quot;Inf&quot; = &quot;red&quot;,
                                                            &quot;Uninf+EFZ&quot; = &quot;blue&quot;)))
)
ht</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-8-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-8-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-8-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</details>
</div>
<div id="heatmap-of-25-tfs-with-differential-binding-score" class="section level2">
<h2>Heatmap of 25 TFs with differential binding score</h2>
<p>We will now investigate if we can find activity patterns which resemble the differential binding score patterns found in 25 TFs.</p>
<div id="hierarchical-clustering-with-different-k" class="section level3 tabset">
<h3 class="tabset">Hierarchical clustering with different k</h3>
<p>Using hierarchical clustering we migth be able to identify clusters with certain patterns of TF activity. Judging from the different heatmaps, k = 4 might be the most informative, since for k = 5 there is only one TF (NHLH1) differentiating an additional cluster. The plot of within-clusters sum of squares should help to pick the best number of clusters, however as can be seen below there is no obvious kink in the curve.</p>
<p>Hierarchical clustering with k = 2 seems to split the cells into treated vs.  untreated cells. These two clusters have a very evident biological meaning. Using k = 3 or k = 4 keeps these two biologically relevant groups separate, but also creates additional clusters which seem to be a mixture of treated and untreated cells. Having a closer look at clusters from k = 4 might give som additional biological insights.</p>
<p>Having a closer look at the four clusters, it becomes evident that the two left most clusters separate treated cells from untreated cells, independent of the infection condition. There are more Inf+EFZ than Uninf+EFZ in the first cluster, however this is in line with the lower cell number of Uninf+EFZ (722 cells) compared to Inf+EFZ (1382 cells).</p>
<p>Two TFs seem to be more active in Inf+EFZ and Uninf+EFZ (compared to Inf and Uninf):</p>
<ul>
<li>RUNX3</li>
<li>JUNB</li>
</ul>
<p>Three TFs seem to be more active in Inf and Uninf (compared to Inf+EFZ and Uninf+EFZ):</p>
<ul>
<li>NFAT5</li>
<li>TCF7L2</li>
<li>CEBPB</li>
</ul>
<p>TCF4 distinguishes a third cluster.</p>
<p>Is there a biological meaning behind that? Maybe we can associate them with a particular condition?</p>
<div id="wss-plot" class="section level4">
<h4>wss plot</h4>
<pre class="r"><code>df &lt;- hiv7@meta.data %&gt;% filter(cell_type == &quot;Myeloid2&quot;, 
                                    orig.ident == &quot;HIV1_CSF&quot;)
mtx &lt;- auc_mtx[, atac_tfs_df$name]

# Decide on a k for clustering
wss = sapply(2:15, function(k) {
  kmeans(x = t(mtx), centers = k)$tot.withinss
})

plot(2:15, wss, type = &quot;b&quot;, 
     xlab = &quot;Number of clusters k&quot;, 
     ylab = &quot;Total within-clusters sum of square&quot;)</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-9-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-9-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/114a941064f12efce14ce5eb1f7cc94e6a004074/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-9-1.png" target="_blank">114a941</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-9-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="different-k" class="section level4">
<h4>different k</h4>
<pre class="r"><code>
ht_list &lt;- map(seq.int(2,6), function(n){
  ht &lt;- Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
        col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        column_title = paste0(&quot;k = &quot;, n),
        show_column_names = FALSE,
        row_names_side = &quot;left&quot;,
        top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                           name = &quot;condition&quot;,
                                           col = list(condition = 
                                                        c(&quot;Uninf&quot; = &quot;orange&quot;, 
                                                          &quot;Inf+EFZ&quot; = &quot;green&quot;,
                                                          &quot;Inf&quot; = &quot;red&quot;, 
                                                          &quot;Uninf+EFZ&quot; = &quot;blue&quot;))),
        column_km = n,
         )
  
})

ht_list &lt;- ht_list[[1]] + ht_list[[2]] + ht_list[[3]] + ht_list[[4]] + ht_list[[5]]
draw(ht_list, column_title = &quot;Hierarchical clustering with different k&quot;, ht_gap = unit(2, &quot;cm&quot;))</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-10-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-10-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-10-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/114a941064f12efce14ce5eb1f7cc94e6a004074/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-10-1.png" target="_blank">114a941</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-10-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="k-4" class="section level4">
<h4>k = 4</h4>
<pre class="r"><code>ht &lt;- draw(Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
      col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
      column_title = &quot;hierarchical clustering with k = 4&quot;,
      show_column_names = FALSE,
      row_names_side = &quot;left&quot;,
      top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                         name = &quot;condition&quot;,
                                         col = list(condition = 
                                                      c(&quot;Uninf&quot; = &quot;orange&quot;, 
                                                        &quot;Inf+EFZ&quot; = &quot;green&quot;,
                                                        &quot;Inf&quot; = &quot;red&quot;, 
                                                        &quot;Uninf+EFZ&quot; = &quot;blue&quot;))),
      column_km = 4,
       ))</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-11-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-11-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-11-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/114a941064f12efce14ce5eb1f7cc94e6a004074/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-11-1.png" target="_blank">114a941</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-11-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="section" class="section level4 unnumbered">
<h4 class="unnumbered"></h4>
<hr />
</div>
</div>
<div id="split-heatmaps-to-find-patterns" class="section level3 tabset">
<h3 class="tabset">Split heatmaps to find patterns</h3>
<div id="separate-heatmaps-according-to-infection-condition" class="section level4">
<h4>Separate Heatmaps according to infection condition</h4>
<p>Separating the cells according to different conditions it becomes very evident, that there is a bigger difference between cells treated with EFZ vs. untreated cells than between infected vs uninfected cells. The Inf+EFZ (green) and Uninf+EFZ (blue) show very similar TF activity patterns and the Inf (red) and Uninf (orange) show very similar TF activity patterns. These findings represent the findings from above</p>
<p>Two TFs seem to be more active in Inf+EFZ and Uninf+EFZ (compared to Inf and Uninf):</p>
<ul>
<li><strong>RUNX3</strong></li>
<li><strong>JUNB</strong></li>
</ul>
<p>Three TFs seem to be more active in Inf and Uninf (compared to Inf+EFZ and Uninf+EFZ):</p>
<ul>
<li><strong>NFAT5:</strong> In microglia cells inflammatory signals trigger NFAT5 expression. (Yang X.L. et al., Dev Neurosci, 2018)</li>
<li><strong>TCF7L2</strong></li>
<li><strong>CEBPB</strong> regulates pro-inflammatory gene expression, increased expression in HAND (HIV-associated neurocognitive disorder), BUT no data from microglia (Canchi S. et al. J Neuroinflammation, 2020)</li>
</ul>
<pre class="r"><code>ht_conditions &lt;- Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
        #col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        column_title = &quot;split according to infection condition&quot;,
        show_column_names = FALSE,
        row_names_side = &quot;left&quot;,
        col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                           name = &quot;condition&quot;, col = list(condition = c(&quot;Uninf&quot; = &quot;orange&quot;, &quot;Inf+EFZ&quot; = &quot;green&quot;, &quot;Inf&quot; = &quot;red&quot;, &quot;Uninf+EFZ&quot; = &quot;blue&quot;))),
        #row_km = 2,
        #row_split = atac_tfs_df$condition,
        column_split = tri_mg$orig.ident,
        left_annotation = rowAnnotation(TF_footprinting = (atac_tfs_df %&gt;% 
                                                         select(name, condition) %&gt;%
                                                         column_to_rownames(&quot;name&quot;))[[&quot;condition&quot;]],
              name = &quot;TF_footprinting&quot;,
              col = list(TF_footprinting = c(&quot;up_latent&quot; = &quot;pink&quot;, 
                                             &quot;up_uninf&quot; = &quot;darkgreen&quot;)))
        )

ht_conditions </code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-12-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-12-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/114a941064f12efce14ce5eb1f7cc94e6a004074/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-12-1.png" target="_blank">114a941</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-12-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="split-heatmaps-according-to-tf-binding-scores-from-atacseq-data" class="section level4">
<h4>Split Heatmaps according to TF binding scores from ATACseq data</h4>
<p>Separating the heatmap along the rows, according to TFs either more bound in latently infected microglia or more bound in uninfected primary microglia, does not result in a separation according to treatment with EFZ.</p>
<p>What one might expect to see, is that TFs more bound in uninfected primary microglia would also show higher activity in uninfected iMg, while TFs more bound in latently infected primary microglia would show higher activity in Inf+EFZ iMg. However, such a pattern is not immediately visible in the heatmap below.</p>
<p>The TFs more bound in latently infected cells do not seem to correlate with any condition, but rather vary a lot more across all cells than the TFs more bound in uninfected cells which show some correlation with the condition and less across all cells.</p>
<pre class="r"><code>ht_tfs1 &lt;- Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
        #col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        column_title = &quot;split according to TF binding score for latent vs. uninfected&quot;,
        show_column_names = FALSE,
        row_names_side = &quot;left&quot;,
        #column_km = 4,
        row_split = atac_tfs_df$condition,
        col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                           name = &quot;condition&quot;, 
                                           col = list(condition = 
                                                        c(&quot;Uninf&quot; = &quot;orange&quot;,
                                                          &quot;Inf+EFZ&quot; = &quot;green&quot;,
                                                          &quot;Inf&quot; = &quot;red&quot;,
                                                          &quot;Uninf+EFZ&quot; = &quot;blue&quot;))),
        #column_split = tri_mg$orig.ident
        left_annotation = rowAnnotation(TF_footprinting = 
                                          (atac_tfs_df %&gt;% 
                                            select(name, condition) %&gt;%
                                            column_to_rownames(&quot;name&quot;))[[&quot;condition&quot;]],
                      name = &quot;TF_footprinting&quot;,
                      col = list(TF_footprinting = c(&quot;up_latent&quot; = &quot;pink&quot;, 
                                                     &quot;up_uninf&quot; = &quot;darkgreen&quot;)))
        )


#ht_list &lt;- ht_conditions + ht_tfs
#draw(ht_list)</code></pre>
<pre class="r"><code>ht_tfs2 &lt;- Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
        #col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        column_title = &quot;split according to TF binding score for latent vs. uninfected&quot;,
        show_column_names = FALSE,
        row_names_side = &quot;left&quot;,
        #column_km = 4,
        row_split = atac_tfs_df$condition,
        column_split = tri_mg@meta.data$orig.ident,
        col =  colorRampPalette(rev(brewer.pal(n = 7, name = &quot;RdYlBu&quot;)))(100),
        top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                           name = &quot;condition&quot;, col =
                                             list(condition =
                                                    c(&quot;Uninf&quot; = &quot;orange&quot;,
                                                      &quot;Inf+EFZ&quot; = &quot;green&quot;, 
                                                      &quot;Inf&quot; = &quot;red&quot;, 
                                                      &quot;Uninf+EFZ&quot; = &quot;blue&quot;))),
        #column_split = tri_mg$orig.ident
        left_annotation = rowAnnotation(TF_footprinting = 
                                          (atac_tfs_df %&gt;% 
                                            select(name, condition) %&gt;%
                                            column_to_rownames(&quot;name&quot;))[[&quot;condition&quot;]],
                      name = &quot;TF_footprinting&quot;,
                      col = list(TF_footprinting = c(&quot;up_latent&quot; = &quot;pink&quot;, 
                                                     &quot;up_uninf&quot; = &quot;darkgreen&quot;)))
        )</code></pre>
<pre class="r"><code>ht_list &lt;- ht_tfs1 + ht_tfs2
draw(ht_list, ht_gap = unit(2, &quot;cm&quot;))</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-15-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-15-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/114a941064f12efce14ce5eb1f7cc94e6a004074/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-15-1.png" target="_blank">114a941</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-15-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
</div>
</div>
<div id="statistical-tests-to-find-tf-activity-differences-between-clusters" class="section level2">
<h2>Statistical Tests to find TF activity differences between clusters</h2>
<p>To quantitatively investigate whether we can find TFs which have significantly different activity between clusters we can use Kruskal-Wallis test and pairwise wilcox test as a posthoc analysis.</p>
<p>206 TFs have different acivity between the clusters.</p>
<pre class="r"><code>p_values &lt;- map(seq.int(1:length(colnames(auc_mtx))), function(n){
  tf &lt;- colnames(auc_mtx)[n]
  #print(tf)
  #print(triculture@meta.data[tf])
  p_values &lt;- unlist(kruskal.test(pull(tri_mg@meta.data[tf]) ~ orig.ident,
                                  tri_mg@meta.data))[&quot;p.value&quot;]
  #print(p_values)
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple hypothesis testing
adj_p_values &lt;- p.adjust(as.numeric(unname(unlist(p_values))), method = &quot;fdr&quot;)


# how many transcription factors are significantly different between the clusters?
#length(adj_p_values[adj_p_values&lt;0.01])</code></pre>
</div>
<div id="posthoc-analysis" class="section level2 tabset">
<h2 class="tabset">Posthoc analysis</h2>
<div id="more-bound-in-latent-infection" class="section level3">
<h3>more bound in latent infection</h3>
<p><strong>Boxplots of TFs activity across conditions of TFs more bound in latent infection</strong></p>
<p>Among the TFs more bound in latently infected cells there is a difference in activity for REST and ZEB1, but only between treatment conditions and not between infection conditions. For the other three TFs there are no differences or patterns.</p>
<pre class="r"><code>
boxplot_list &lt;- map(up_latent, function(tf) {
  ggplot() +
    geom_boxplot(aes(x = tri_mg@meta.data %&gt;% pull(&quot;orig.ident&quot;),
                     y = tri_mg@meta.data %&gt;% pull(tf), 
                 fill = tri_mg@meta.data %&gt;% pull(&quot;orig.ident&quot;))) +
    stat_pvalue_manual(tri_mg@meta.data %&gt;% 
                         pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ orig.ident&quot;))) %&gt;% 
                         add_xy_position()) +
    xlab(&quot;Condition&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;condition&quot;))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 3, nrow = 2))</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-17-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-17-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-17-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/114a941064f12efce14ce5eb1f7cc94e6a004074/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-17-1.png" target="_blank">114a941</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-17-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="more-bound-in-uninfected" class="section level3">
<h3>more bound in uninfected</h3>
<p><strong>Boxplots of TFs activity across conditions of TFs more bound in uninfected primary microglia</strong></p>
<p>In the plots below the trend of TF activity being negatively correlated between treated and untreated cells, independent of the infection status can be seen for some TFs, for example CEBPB and MEF2C. For other TFs there is no difference between either infection status or treatment, for example CREB1 and ETS1.</p>
<p>Yet, even though the effect of treatment is most prominent, some TFs are additionally following the pattern of the footprinting. For example, MEF2C is more bound in the uninfected iMg than in infected iMg.</p>
<p>One assumption is that TFs which are markers of microglia cells might be more or less active depending on whether the cells were treated with EFZ or not. An argument against this assumption is that both CEBPB and MEF2C are typically expressed in microglia, yet we observe an opposite trend. CEBPB is more active in Inf and Uninf cells than in Inf+EFZ and Uninf+EFZ cells, while MEF2C is more active in Inf+EFZ and Uninf+EFZ cells than in Inf and Uninf cells.</p>
<pre class="r"><code>

boxplot_list &lt;- map(up_uninf, function(tf) {
  ggplot() +
    geom_boxplot(aes(x = tri_mg@meta.data %&gt;% pull(&quot;orig.ident&quot;),
                     y = tri_mg@meta.data %&gt;% pull(tf), 
                 fill = tri_mg@meta.data %&gt;% pull(&quot;orig.ident&quot;))) +
    stat_pvalue_manual(tri_mg@meta.data %&gt;% 
                         pairwise_wilcox_test(as.formula(paste0(tf, &quot; ~ orig.ident&quot;))) %&gt;% 
                         add_xy_position()) +
    xlab(&quot;Condition&quot;) +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = &quot;condition&quot;))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 5, nrow = 4))</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-18-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-18-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-18-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/114a941064f12efce14ce5eb1f7cc94e6a004074/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-18-1.png" target="_blank">114a941</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-18-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>
#annotate_figure(figure,
 #               top = text_grob(&quot;TFs activity across conditions for TFs more bound in uninfected primary microglia&quot;, size = 20))</code></pre>
<hr />
</div>
</div>
<div id="correlations-between-conditions" class="section level2 tabset">
<h2 class="tabset">Correlations between conditions</h2>
<p><strong>Interpretation here</strong></p>
<div id="correlation-matrix" class="section level3">
<h3>Correlation matrix</h3>
<pre class="r"><code>cor_mtx &lt;- tri_mg@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;%
  select(c(&quot;cell&quot;, &quot;orig.ident&quot;, colnames(auc_mtx))) %&gt;%
  group_by(orig.ident) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  gather(TF, value, -orig.ident) %&gt;% 
  spread(orig.ident, value) %&gt;% 
  column_to_rownames(&quot;TF&quot;) %&gt;% cor()

testCor &lt;- cor.mtest(cor_mtx, conf.level = 0.95)

# plot correlations
# p.mat = matrix of p_values


corrplot(cor_mtx,
         #method = &quot;color&quot;, 
         order = &quot;hclust&quot;, 
         addrect = 4, # add boxes 
         p.mat = testCor$p, 
         #sig.level = .05,
         insig = &quot;p-value&quot;, 
         tl.col = &quot;black&quot;)</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-19-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-19-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-19-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-19-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="scatter-matrix" class="section level3">
<h3>Scatter matrix</h3>
<pre class="r"><code>tri_mg@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;%
  select(c(&quot;cell&quot;, &quot;orig.ident&quot;, colnames(auc_mtx))) %&gt;%
  group_by(orig.ident) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  gather(TF, value, -orig.ident) %&gt;% 
  spread(orig.ident, value) %&gt;% 
  column_to_rownames(&quot;TF&quot;) %&gt;% GGally::ggpairs() +
    labs(title = &quot;Scatter matrix for different conditions&quot;)</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-20-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-20-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-20-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-20-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-20-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
</div>
<div id="correlation-between-tfs-differentiating-myeloid2-clusters" class="section level2 tabset">
<h2 class="tabset">Correlation between TFs differentiating Myeloid2 clusters</h2>
<p>These TFs were differntially active between the three hierarchical clusters in Myeloid2 cells from HIV-infected patients. Across three different patients they showed correlated changes in activity. YY1, ETS1 and REST were positively correlated with each other and negatively correlated with MEF2A, ETV6, JUNB, MEF2C, MXI1 and CEBPB. The pattern observed for the iMg from the triculture dataset does not resemble the pattern observed in Myeloid2 cells.</p>
<div id="correlation-matrix-1" class="section level3">
<h3>Correlation Matrix</h3>
<pre class="r"><code>intersect_csf &lt;- read.table(&quot;/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_intersect_TFs_different_across_patients_3_clusters&quot;)


cor_mtx &lt;- tri_mg@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  select(c(&quot;cell&quot;, &quot;orig.ident&quot;, intersect_csf$x[intersect_csf$x %in% colnames(auc_mtx)])) %&gt;% # only TFs with differential binding scores from ATACseq
  group_by(orig.ident) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  column_to_rownames(&quot;orig.ident&quot;) %&gt;% cor()

testCor &lt;- cor.mtest(cor_mtx, conf.level = 0.95)

corrplot(cor_mtx,
         #method = &quot;color&quot;,
         order = &quot;hclust&quot;, 
         addrect = 4, # add boxes 
         p.mat = testCor$p,
         insig = &quot;p-value&quot;)</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-21-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-21-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-21-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-21-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="scatter-matrix-1" class="section level3">
<h3>Scatter Matrix</h3>
<pre class="r"><code>tri_mg@meta.data %&gt;% 
  rownames_to_column(&quot;cell&quot;) %&gt;% 
  select(c(&quot;cell&quot;, &quot;orig.ident&quot;, intersect_csf$x[intersect_csf$x %in% colnames(auc_mtx)])) %&gt;% # only TFs with differential binding scores from ATACseq
  group_by(orig.ident) %&gt;%
  summarize_if(is.numeric, mean) %&gt;% 
  column_to_rownames(&quot;orig.ident&quot;) %&gt;%
  GGally::ggpairs() +
  labs(title = &quot;Scatter Matrix for different TFs&quot;)</code></pre>
<p><img src="figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-22-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/40651d96647f10219a4aec953c8d618d57bc13f5/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-22-1.png" target="_blank">40651d9</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/436b5c2790a0ce492f776915a9e543bf3a84511d/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-22-1.png" target="_blank">436b5c2</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-20
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/fce26d5dd39fcac3f053aaaa3ceb1e2c02e05654/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-22-1.png" target="_blank">fce26d5</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
<tr>
<td>
<a href="https://github.com/Katharina782/report/blob/9ee308625fb863ad709edf2660babe4de93ced91/docs/figure/Triculture_dataset_GRN_with_Scenic.Rmd/unnamed-chunk-22-1.png" target="_blank">9ee3086</a>
</td>
<td>
Katharina782
</td>
<td>
2021-12-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<hr />
</div>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblasp-r0.3.3.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] readxl_1.3.1         multtest_2.46.0      Biobase_2.50.0      
 [4] BiocGenerics_0.36.1  metap_1.6            knitr_1.30          
 [7] rstatix_0.6.0        corrplot_0.92        gridExtra_2.3       
[10] ggpubr_0.4.0         circlize_0.4.13      ComplexHeatmap_2.9.4
[13] ggcorrplot_0.1.3     pheatmap_1.0.12      RColorBrewer_1.1-2  
[16] ggrepel_0.9.1        data.table_1.14.2    Matrix_1.3-4        
[19] edgeR_3.32.0         limma_3.46.0         SeuratObject_4.0.3  
[22] Seurat_4.0.5         forcats_0.5.0        stringr_1.4.0       
[25] dplyr_1.0.7          purrr_0.3.4          readr_1.4.0         
[28] tidyr_1.1.2          tibble_3.1.6         ggplot2_3.3.5       
[31] tidyverse_1.3.0      workflowr_1.6.2     

loaded via a namespace (and not attached):
  [1] utf8_1.2.2            reticulate_1.22       tidyselect_1.1.1     
  [4] htmlwidgets_1.5.4     Rtsne_0.15            munsell_0.5.0        
  [7] mutoss_0.1-12         codetools_0.2-18      ica_1.0-2            
 [10] future_1.23.0         miniUI_0.1.1.1        withr_2.4.3          
 [13] colorspace_2.0-2      highr_0.8             rstudioapi_0.13      
 [16] stats4_4.0.2          ROCR_1.0-11           ggsignif_0.6.0       
 [19] tensor_1.5            listenv_0.8.0         labeling_0.4.2       
 [22] Rdpack_2.1.2          git2r_0.27.1          mnormt_2.0.2         
 [25] polyclip_1.10-0       farver_2.1.0          rprojroot_2.0.2      
 [28] TH.data_1.0-10        parallelly_1.29.0     vctrs_0.3.8          
 [31] generics_0.1.1        xfun_0.29             R6_2.5.1             
 [34] doParallel_1.0.16     clue_0.3-60           locfit_1.5-9.4       
 [37] reshape_0.8.8         spatstat.utils_2.2-0  assertthat_0.2.1     
 [40] promises_1.2.0.1      scales_1.1.1          multcomp_1.4-15      
 [43] gtable_0.3.0          globals_0.14.0        goftest_1.2-3        
 [46] sandwich_3.0-0        rlang_0.4.12          GlobalOptions_0.1.2  
 [49] splines_4.0.2         lazyeval_0.2.2        spatstat.geom_2.3-0  
 [52] broom_0.7.2           yaml_2.2.1            reshape2_1.4.4       
 [55] abind_1.4-5           modelr_0.1.8          backports_1.2.0      
 [58] httpuv_1.6.3          tools_4.0.2           ellipsis_0.3.2       
 [61] spatstat.core_2.3-1   ggridges_0.5.2        TFisher_0.2.0        
 [64] Rcpp_1.0.7            plyr_1.8.6            rpart_4.1-15         
 [67] deldir_1.0-6          pbapply_1.5-0         GetoptLong_1.0.5     
 [70] cowplot_1.1.1         S4Vectors_0.28.1      zoo_1.8-9            
 [73] haven_2.3.1           cluster_2.1.0         fs_1.5.1             
 [76] magrittr_2.0.1        magick_2.5.2          scattermore_0.7      
 [79] openxlsx_4.2.3        lmtest_0.9-39         reprex_0.3.0         
 [82] RANN_2.6.1            tmvnsim_1.0-2         mvtnorm_1.1-1        
 [85] whisker_0.4           fitdistrplus_1.1-6    matrixStats_0.61.0   
 [88] hms_1.1.1             patchwork_1.1.1       mime_0.12            
 [91] evaluate_0.14         xtable_1.8-4          rio_0.5.16           
 [94] IRanges_2.24.1        shape_1.4.6           compiler_4.0.2       
 [97] KernSmooth_2.23-18    crayon_1.4.2          htmltools_0.5.2      
[100] mgcv_1.8-33           later_1.3.0           lubridate_1.7.9.2    
[103] DBI_1.1.1             dbplyr_2.0.0          MASS_7.3-53          
[106] car_3.0-10            cli_3.1.0             rbibutils_2.2.4      
[109] igraph_1.2.9          pkgconfig_2.0.3       sn_2.0.1             
[112] numDeriv_2016.8-1.1   foreign_0.8-80        plotly_4.9.2.1       
[115] spatstat.sparse_2.0-0 xml2_1.3.2            foreach_1.5.1        
[118] rvest_0.3.6           digest_0.6.29         sctransform_0.3.2    
[121] RcppAnnoy_0.0.19      spatstat.data_2.1-0   rmarkdown_2.6        
[124] cellranger_1.1.0      leiden_0.3.5          uwot_0.1.10          
[127] curl_4.3.2            shiny_1.7.1           rjson_0.2.20         
[130] lifecycle_1.0.1       nlme_3.1-150          jsonlite_1.7.2       
[133] carData_3.0-4         viridisLite_0.4.0     fansi_0.5.0          
[136] pillar_1.6.4          GGally_2.0.0          lattice_0.20-41      
[139] plotrix_3.8-2         fastmap_1.1.0         httr_1.4.2           
[142] survival_3.2-7        glue_1.6.0            zip_2.1.1            
[145] png_0.1-7             iterators_1.0.13      stringi_1.7.6        
[148] mathjaxr_1.4-0        irlba_2.3.3           future.apply_1.8.1   </code></pre>
</div>
</div>

<div id="rmd-source-code">---
title: "TF activity in iMg from triculture dataset"
author: "Katharina Mikulik"
output: html_document
---

<style>
body {
text-align: justify}
</style>

```{r setup, include = FALSE}
knitr::opts_chunk$set(cache = FALSE, autodep = TRUE, 
                      collapse = TRUE, message = FALSE)

```


```{r}
library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggplot2)
library(dplyr)
library(ggrepel)
#library(harmony)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(corrplot)
library(rstatix)
library(knitr)
library(metap)
library(multtest)
library(readxl)
```

## Gene regulatory network inference with SCENIC

To infer GRNs from iMg cells from the triculture dataset [@Ryan] (4333 cells in 
total) I again used SCENIC.

When referring to differentially bound TFs in the C20 cell line in the following
I am referring to differential binding scores between infection conditions 
obtained from bulk ATACseq data from C20 cell line cultures using TOBIAS.

I first tried to run SCENIC with 3000 most highly variable genes (HVG), but then
I only got an overlap with differentially bound TFS of eight. When using 10,000
HVG SCENIC returns 232 TFS and their corresponding activity score for each cell.
25 of these TFs overlap with differentially bound TFs in the C20 cell line.




```{r}
# read in triculture Seurat object
triculture <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_clean_seurat_object")

# read in CSF HIV Seurat object
hiv7 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds")

imicroglia <- subset(triculture, cell_type == "iMg")

```


<details>
<summary>**preparing matrix for SCENIC**</summary>

First, I use SCTransform for normalization of the expression data, then
the top 10,000 HVGs are selected and the matrix transposed, because the pySCENIC
pipeline used requires an in input matrix which has genes as columns and cells 
as rows.
 (https://github.com/hdsu-bioquant/pySCENIC_pipeline/blob/master/input/gene_exp_mtx_example.tsv)

```
# use 10 000 most highly variable genes
scenic <- SCTransform(imicroglia, variable.features.n = 10000, verbose=FALSE)
saveRDS(scenic, "/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_scenic")
scenic <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_scenic")
scenic_hvg <- subset(scenic, features = scenic@assays$SCT@var.features)
scenic_mtx <- t(scenic_hvg@assays$SCT@scale.data)
write.table(x = scenic_mtx, file = "/media/ag-cherrmann/kmikulik/HIV_microglia/src/pySCENIC_pipeline/input/triculture_mtx.tsv", sep = "\t")

#scenic_matrix <- read.table(file = "/media/ag-cherrmann/kmikulik/HIV_microglia/src/pySCENIC_pipeline/input/triculture_mtx.tsv", sep = "\t")
```
---
</details>


<details>
<summary>**AuCell matrix - output from Scenic**</summary>

The output from SCENIC is a matrix containing cells as rows and 232 TFs
as columns. Five exemplary rows and columns are shown below. The TF activity for each cell can be used to draw conclusions
about different cell states. 


```{r}
auc_mtx <- read.csv("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/pySCENIC_output/aucell_img_10k.csv")
auc_mtx <- auc_mtx %>% column_to_rownames("Cell")
auc_mtx <- auc_mtx %>% setnames( old = colnames(auc_mtx), new = sapply(strsplit(
  colnames(auc_mtx), "[...]"), `[`, 1))

auc_mtx[1:5, 1:5] %>% kable(caption = "Layout of TF activity matrix")
```

---
</details>

<details>
<summary>**Number of iMg in each condition**</summary>


```{r}
# add auc matrix to metadata of hiv4 seurat object
tri_mg <- AddMetaData(imicroglia, 
                    metadata = auc_mtx, 
                    col.name = colnames(auc_mtx))

```


```{r, results = "asis"}
tri_mg@meta.data %>% 
  rename(condition  = orig.ident) %>%
  group_by(condition) %>%
  summarize(n = n()) %>% 
  kable(caption = "Number of cells in each condition")
```

---
</details>


# Bulk ATACseq footprinting, C20 cell line

The C20 cell line was derived from 
an HIV+ patient  treated with ART (Antiretroviral Therapy). Since the access 
of ART to the brain is limited due to the blood-brain barrier, the effects of 
ART on these cells are probably negligible. Only the 75 TFs with p-values for differential binding scores below 1e-50 were used. We will focus on TFs which show differential binding scores between latent
infection and uninfected conditions in primary microglia.
For 25 of these 75 TFs there is a TF activity score for the iMg. Focusing on these 25 TFs we might be able to find activity patterns that 
correlate with the TF binding scores. Five TFs are more bound in latent infection
and twenty TFs are more bound in uninfected cells

The reason why we find more of the TFs more bound in uninfected cells 
in the TF activity score matrix from SCENIC is unclear.

```{r}
tfs_ana <- read.table("/media/ag-cherrmann/projects/06_HIV_Microglia/data/atacseq/data-2020-11-06/tobias/TOBIAS_snakemake/footprint_mglia2_GlassTF_17-03/TFBS/bindetect_results.txt", sep = "\t", header = TRUE)

top_tfs <- tfs_ana %>% filter(uninf_red_pvalue < 1e-50)

# keep only the TFs in the dataframe which overlap between the two datasets.
overlap_atac <- top_tfs %>% filter(name %in% colnames(auc_mtx))

# dataframe containing only the 25 Tfs whcih overlap with TF activity matrix from SCENIC 
# adding a column with information in which infection condition the TF is more bound
atac_tfs_df <- top_tfs %>% filter(name %in% colnames(auc_mtx)) %>% 
  mutate(condition = ifelse(uninf_red_change > 0, "up_uninf", "up_latent"))

# List of 5 TFs more bound in latent infection
up_latent <- atac_tfs_df$name[atac_tfs_df$condition == "up_latent"]
# list of 25 TFs more bound in uninfected cells
up_uninf <- atac_tfs_df$name[atac_tfs_df$condition == "up_uninf"]
```

<details>
<summary>**Volcano Plot ATAC-seq footprinting**</summary>

```{r, fig.height=8, fig.width=12}
top_tfs %>% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = ifelse(uninf_red_change < 0, "up_latent", "up_uninf")) %>%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  mutate(label = ifelse(name %in% overlap_atac$name, name, NA)) %>% 
  ggplot(aes(x = uninf_red_change,
             y = -log10(uninf_red_pvalue),
             col = condition, 
             label = label)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-50), col = "red") +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = "TFs with differential binding scores") +
  ylab("-log10(p_value)") +
  xlab("differential binding score")

```

---
</details>

# TF activity in iMg 

## Heatmap for all TFs

In the heatmap of TF activities below you can already see a trend towards a separation of 
cells treated with EFZ (blue and green) from untreated cells (red and orange).

<details>
<summary>**Heatmap of 232 TFs and their activity in iMg**</summary>


```{r, fig.width=15, fig.height=25}
conditions <- tri_mg@meta.data %>%
                    rownames_to_column("cell") %>%
                    select("cell", "orig.ident")%>% column_to_rownames("cell")


ht <- Heatmap(t(auc_mtx), 
      col =  colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
      show_column_names = FALSE,
      row_names_side = "left",
      top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                         name = "condition", col =  
                                         list(condition = c("Uninf" = "orange", 
                                                            "Inf+EFZ" = "green",
                                                            "Inf" = "red",
                                                            "Uninf+EFZ" = "blue")))
)
ht
```

---
</details>


## Heatmap of 25 TFs with differential binding score

We will now investigate if we can find activity patterns which resemble
the differential binding score patterns found in 25 TFs.


### Hierarchical clustering with different k {.tabset}

Using hierarchical clustering we migth be able to identify clusters with certain
patterns of TF activity. Judging from the different heatmaps, k = 4 might be the
most informative, since for k = 5 there is only one TF (NHLH1) differentiating 
an additional cluster. The plot of within-clusters sum of squares should help 
to pick the best number of clusters, however as can be seen below there is no 
obvious kink in the curve.

Hierarchical clustering with k = 2 seems to split the cells into treated vs. 
untreated cells. These two clusters have a very evident biological meaning.
Using k = 3 or k = 4 keeps these two biologically 
relevant groups separate, but also creates additional clusters which seem to be
a mixture of treated and untreated cells. Having a closer look at clusters from
k = 4 might give som additional biological insights.

Having a closer look at the four clusters, it becomes evident that the two left
most clusters separate treated cells from untreated cells, independent of the 
infection condition. There are more Inf+EFZ than Uninf+EFZ in the first cluster,
however this is in line with the lower cell number of Uninf+EFZ (722 cells) 
compared to Inf+EFZ (1382 cells).

Two TFs seem to be more active in Inf+EFZ and Uninf+EFZ (compared to Inf and Uninf):

* RUNX3
* JUNB

Three TFs seem to be more active in Inf and Uninf (compared to Inf+EFZ and Uninf+EFZ):

* NFAT5
* TCF7L2
* CEBPB

TCF4 distinguishes a third cluster. 

Is there a biological meaning behind that? Maybe we can associate them with 
a particular condition?

#### wss plot

```{r}
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                    orig.ident == "HIV1_CSF")
mtx <- auc_mtx[, atac_tfs_df$name]

# Decide on a k for clustering
wss = sapply(2:15, function(k) {
  kmeans(x = t(mtx), centers = k)$tot.withinss
})

plot(2:15, wss, type = "b", 
     xlab = "Number of clusters k", 
     ylab = "Total within-clusters sum of square")
```

#### different k

```{r, fig.width=15, fig.height=10}

ht_list <- map(seq.int(2,6), function(n){
  ht <- Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
        col =  colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
        column_title = paste0("k = ", n),
        show_column_names = FALSE,
        row_names_side = "left",
        top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                           name = "condition",
                                           col = list(condition = 
                                                        c("Uninf" = "orange", 
                                                          "Inf+EFZ" = "green",
                                                          "Inf" = "red", 
                                                          "Uninf+EFZ" = "blue"))),
        column_km = n,
         )
  
})

ht_list <- ht_list[[1]] + ht_list[[2]] + ht_list[[3]] + ht_list[[4]] + ht_list[[5]]
draw(ht_list, column_title = "Hierarchical clustering with different k", ht_gap = unit(2, "cm"))
```


#### k = 4

```{r, fig.width = 15, fig.height=10}
ht <- draw(Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
      col =  colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
      column_title = "hierarchical clustering with k = 4",
      show_column_names = FALSE,
      row_names_side = "left",
      top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                         name = "condition",
                                         col = list(condition = 
                                                      c("Uninf" = "orange", 
                                                        "Inf+EFZ" = "green",
                                                        "Inf" = "red", 
                                                        "Uninf+EFZ" = "blue"))),
      column_km = 4,
       ))
```

#### {-}

---

### Split heatmaps to find patterns {.tabset}

#### Separate Heatmaps according to infection condition

Separating the cells according to different conditions it becomes very 
evident, that there is a bigger difference between cells treated with EFZ 
vs. untreated cells than between infected vs uninfected cells. The Inf+EFZ (green)
and Uninf+EFZ (blue) show very similar TF activity patterns and the Inf (red) and 
Uninf (orange) show very similar TF activity patterns. These findings
represent the findings from above 


Two TFs seem to be more active in Inf+EFZ and Uninf+EFZ (compared to Inf and Uninf):

* **RUNX3**
* **JUNB**

Three TFs seem to be more active in Inf and Uninf (compared to Inf+EFZ and Uninf+EFZ):

* **NFAT5:** In microglia cells inflammatory signals trigger NFAT5 expression. 
(Yang X.L. et al., Dev Neurosci, 2018)
* **TCF7L2**
* **CEBPB** regulates pro-inflammatory gene expression,
increased expression in HAND (HIV-associated neurocognitive disorder), 
BUT no data from microglia (Canchi S. et al. J Neuroinflammation, 2020)

```{r, fig.width=15, fig.height=10}
ht_conditions <- Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
        #col =  colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
        column_title = "split according to infection condition",
        show_column_names = FALSE,
        row_names_side = "left",
        col =  colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
        top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                           name = "condition", col = list(condition = c("Uninf" = "orange", "Inf+EFZ" = "green", "Inf" = "red", "Uninf+EFZ" = "blue"))),
        #row_km = 2,
        #row_split = atac_tfs_df$condition,
        column_split = tri_mg$orig.ident,
        left_annotation = rowAnnotation(TF_footprinting = (atac_tfs_df %>% 
                                                         select(name, condition) %>%
                                                         column_to_rownames("name"))[["condition"]],
              name = "TF_footprinting",
              col = list(TF_footprinting = c("up_latent" = "pink", 
                                             "up_uninf" = "darkgreen")))
        )

ht_conditions 


```

#### Split Heatmaps according to TF binding scores from ATACseq data

Separating the heatmap along the rows, according to TFs either more bound in 
latently infected microglia or more bound in uninfected primary microglia,
does not result in a separation according to treatment with EFZ. 

What one might expect to see, is that TFs more bound in uninfected primary 
microglia would also show higher activity in uninfected iMg, while TFs more 
bound in latently infected primary microglia would show higher activity in 
Inf+EFZ iMg. However, such a pattern is not immediately visible in the heatmap 
below.

The TFs more bound in latently infected cells do not seem to correlate with any condition, but rather vary a lot more across all cells than the TFs more bound 
in uninfected cells which show some correlation with the condition and less 
across all cells.


```{r, fig.width=15, fig.height=10}
ht_tfs1 <- Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
        #col =  colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
        column_title = "split according to TF binding score for latent vs. uninfected",
        show_column_names = FALSE,
        row_names_side = "left",
        #column_km = 4,
        row_split = atac_tfs_df$condition,
        col =  colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
        top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                           name = "condition", 
                                           col = list(condition = 
                                                        c("Uninf" = "orange",
                                                          "Inf+EFZ" = "green",
                                                          "Inf" = "red",
                                                          "Uninf+EFZ" = "blue"))),
        #column_split = tri_mg$orig.ident
        left_annotation = rowAnnotation(TF_footprinting = 
                                          (atac_tfs_df %>% 
                                            select(name, condition) %>%
                                            column_to_rownames("name"))[["condition"]],
                      name = "TF_footprinting",
                      col = list(TF_footprinting = c("up_latent" = "pink", 
                                                     "up_uninf" = "darkgreen")))
        )


#ht_list <- ht_conditions + ht_tfs
#draw(ht_list)
```


```{r}
ht_tfs2 <- Heatmap(t(auc_mtx[, atac_tfs_df$name]), 
        #col =  colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
        column_title = "split according to TF binding score for latent vs. uninfected",
        show_column_names = FALSE,
        row_names_side = "left",
        #column_km = 4,
        row_split = atac_tfs_df$condition,
        column_split = tri_mg@meta.data$orig.ident,
        col =  colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
        top_annotation = HeatmapAnnotation(condition = conditions$orig.ident, 
                                           name = "condition", col =
                                             list(condition =
                                                    c("Uninf" = "orange",
                                                      "Inf+EFZ" = "green", 
                                                      "Inf" = "red", 
                                                      "Uninf+EFZ" = "blue"))),
        #column_split = tri_mg$orig.ident
        left_annotation = rowAnnotation(TF_footprinting = 
                                          (atac_tfs_df %>% 
                                            select(name, condition) %>%
                                            column_to_rownames("name"))[["condition"]],
                      name = "TF_footprinting",
                      col = list(TF_footprinting = c("up_latent" = "pink", 
                                                     "up_uninf" = "darkgreen")))
        )
```

```{r, fig.width=15, fig.height=10}
ht_list <- ht_tfs1 + ht_tfs2
draw(ht_list, ht_gap = unit(2, "cm"))

```

---

## Statistical Tests to find TF activity differences between clusters

To quantitatively investigate whether we can find TFs which have significantly
different activity between clusters we can use Kruskal-Wallis test and 
pairwise wilcox test as a posthoc analysis. 

206 TFs have different acivity between the clusters. 

```{r}
p_values <- map(seq.int(1:length(colnames(auc_mtx))), function(n){
  tf <- colnames(auc_mtx)[n]
  #print(tf)
  #print(triculture@meta.data[tf])
  p_values <- unlist(kruskal.test(pull(tri_mg@meta.data[tf]) ~ orig.ident,
                                  tri_mg@meta.data))["p.value"]
  #print(p_values)
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
#length(adj_p_values[adj_p_values<0.01])

```

## Posthoc analysis {.tabset}

### more bound in latent infection

**Boxplots of TFs activity across conditions of TFs more bound in latent infection**

Among the TFs more bound in latently infected cells there is a difference in 
activity for REST and ZEB1, but only between treatment conditions and not 
between infection conditions. For the other three TFs there are no differences 
or patterns. 

```{r, fig.width=15, fig.height=8}

boxplot_list <- map(up_latent, function(tf) {
  ggplot() +
    geom_boxplot(aes(x = tri_mg@meta.data %>% pull("orig.ident"),
                     y = tri_mg@meta.data %>% pull(tf), 
                 fill = tri_mg@meta.data %>% pull("orig.ident"))) +
    stat_pvalue_manual(tri_mg@meta.data %>% 
                         pairwise_wilcox_test(as.formula(paste0(tf, " ~ orig.ident"))) %>% 
                         add_xy_position()) +
    xlab("Condition") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "condition"))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 3, nrow = 2))
```

### more bound in uninfected 

**Boxplots of TFs activity across conditions of TFs more bound in uninfected primary microglia**

In the plots below the trend of TF activity being negatively correlated between
treated and untreated cells, independent of the infection status can be seen for
some TFs, for example CEBPB and MEF2C. For other TFs there is no difference 
between either infection status or treatment, for example CREB1 and ETS1.

Yet, even though the effect of treatment is most prominent, some TFs are 
additionally following the pattern of the footprinting. For example, MEF2C is
more bound in the uninfected iMg than in infected iMg.


One assumption is that TFs which are markers of microglia cells might 
be more or less active depending on whether the cells were treated with EFZ or 
not. An argument against this assumption is that both CEBPB and MEF2C are 
typically expressed in microglia, yet we observe an opposite trend. 
CEBPB is more active in Inf and Uninf cells than in Inf+EFZ and Uninf+EFZ cells,
while MEF2C is more active in Inf+EFZ and Uninf+EFZ cells than in Inf and Uninf
cells.


```{r, fig.width=20, fig.height=15}


boxplot_list <- map(up_uninf, function(tf) {
  ggplot() +
    geom_boxplot(aes(x = tri_mg@meta.data %>% pull("orig.ident"),
                     y = tri_mg@meta.data %>% pull(tf), 
                 fill = tri_mg@meta.data %>% pull("orig.ident"))) +
    stat_pvalue_manual(tri_mg@meta.data %>% 
                         pairwise_wilcox_test(as.formula(paste0(tf, " ~ orig.ident"))) %>% 
                         add_xy_position()) +
    xlab("Condition") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "condition"))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 5, nrow = 4))

#annotate_figure(figure,
 #               top = text_grob("TFs activity across conditions for TFs more bound in uninfected primary microglia", size = 20))

```

---


## Correlations between conditions {.tabset}

**Interpretation here**

### Correlation matrix

```{r}
cor_mtx <- tri_mg@meta.data %>% 
  rownames_to_column("cell") %>%
  select(c("cell", "orig.ident", colnames(auc_mtx))) %>%
  group_by(orig.ident) %>%
  summarize_if(is.numeric, mean) %>% 
  gather(TF, value, -orig.ident) %>% 
  spread(orig.ident, value) %>% 
  column_to_rownames("TF") %>% cor()

testCor <- cor.mtest(cor_mtx, conf.level = 0.95)

# plot correlations
# p.mat = matrix of p_values


corrplot(cor_mtx,
         #method = "color", 
         order = "hclust", 
         addrect = 4, # add boxes 
         p.mat = testCor$p, 
         #sig.level = .05,
         insig = "p-value", 
         tl.col = "black")

```

### Scatter matrix

```{r, fig.width=12, fig.height=8}
tri_mg@meta.data %>% 
  rownames_to_column("cell") %>%
  select(c("cell", "orig.ident", colnames(auc_mtx))) %>%
  group_by(orig.ident) %>%
  summarize_if(is.numeric, mean) %>% 
  gather(TF, value, -orig.ident) %>% 
  spread(orig.ident, value) %>% 
  column_to_rownames("TF") %>% GGally::ggpairs() +
    labs(title = "Scatter matrix for different conditions")
```

---


## Correlation between TFs differentiating Myeloid2 clusters {.tabset}

These TFs were differntially active between the three hierarchical clusters in
Myeloid2 cells from HIV-infected patients. Across three different patients
they showed correlated changes in activity. YY1, ETS1 and REST were positively
correlated with each other and negatively
correlated with MEF2A, ETV6, JUNB, MEF2C, MXI1 and CEBPB. The pattern observed for the iMg from the triculture dataset does not
resemble the pattern observed in Myeloid2 cells. 

### Correlation Matrix

```{r}
intersect_csf <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_intersect_TFs_different_across_patients_3_clusters")


cor_mtx <- tri_mg@meta.data %>% 
  rownames_to_column("cell") %>% 
  select(c("cell", "orig.ident", intersect_csf$x[intersect_csf$x %in% colnames(auc_mtx)])) %>% # only TFs with differential binding scores from ATACseq
  group_by(orig.ident) %>%
  summarize_if(is.numeric, mean) %>% 
  column_to_rownames("orig.ident") %>% cor()

testCor <- cor.mtest(cor_mtx, conf.level = 0.95)

corrplot(cor_mtx,
         #method = "color",
         order = "hclust", 
         addrect = 4, # add boxes 
         p.mat = testCor$p,
         insig = "p-value")
```

### Scatter Matrix

```{r, fig.height=10, fig.width=15}
tri_mg@meta.data %>% 
  rownames_to_column("cell") %>% 
  select(c("cell", "orig.ident", intersect_csf$x[intersect_csf$x %in% colnames(auc_mtx)])) %>% # only TFs with differential binding scores from ATACseq
  group_by(orig.ident) %>%
  summarize_if(is.numeric, mean) %>% 
  column_to_rownames("orig.ident") %>%
  GGally::ggpairs() +
  labs(title = "Scatter Matrix for different TFs")
```

---

# References</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("Triculture_dataset_GRN_with_Scenic.Rmd");
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
