---
title: "TF activity in myeloid2 cells"
author: "Katharina Mikulik"
output: html
---
<style>
body {
text-align: justify}
</style>


```{r setup}
knitr::opts_chunk$set(echo = TRUE, collapse = TRUE, message = FALSE,
                      cache = FALSE, autodep = TRUE)
```


```{r}
library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggplot2)
library(dplyr)
library(ggrepel)
#library(harmony)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(knitr)
library(ggVennDiagram)
library(corrplot)
library(rstatix)
# "Multitasking/Multiprocessing"
library(future)
#plan("multisession", workers = 24)
```

# GRN inference with Scenic

Using Scenic we tried to infer Gene regulatory networks from the gene expression
data. The output is a matrix of TF (Transcription Factor) activity per cell. 
The input matrix for SCENIC included the 10,000 most highly variable genes of 
the Myeloid2 cell cluster. The resulting activity matrix contains activity scores
of 125 different transcription factors.

First, we wanted to see if we can identify patterns in the TF activity in the
Myeloid2 cells. We compared the TF activity to TFs shown to have differential 
binding scores between three infection states (uninfected, latent infection, 
active infection) in primary microglia (bulk ATACseq). For the comparison
with the TF activity in Myeloid2 cells we focused on the TF with differential 
binding scores between **uninfected** and **latently infected** states.

The question we wanted to answer is:
Can we see patterns of increased/decreased TF activity in Myeloid2 cells that
correlate with the binding scores between infection states in primary
microglia?    


</details>

<details> 
<summary>TF activity matrix and HIV CSF/Blood dataset</summary>


* read in Seurat object

```{r}
hiv7 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds")
hiv4 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/non_integrated_HIV1_HIV2_4samples_seurat_object.rds")
```

* read in AuCell matrix (output from Scenic)

```{r}
auc_mtx <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/aucell_matrix_10k_hvg.tsv", sep = "\t")

auc_mtx %>% head %>%  kable(caption = "TF activity matrix (output from SCENIC)")
```



* Combine AuCell Matrix and Seurat object

```{r}
# add auc matrix to metadata of hiv4 seurat object
hiv7_scenic <- AddMetaData(hiv7, 
                    metadata = auc_mtx, 
                    col.name = colnames(auc_mtx))
```

---
</details>

# Bulk ATACseq footprinting, C20 cell line

Using bulk ATACseq data from primary microglia with three different conditions,
active infection, latent infection, uninfected, differential binding scores were
obtained using TOBIAS. Here, I will use the TFs which were previously shown to 
have differential binding scores in these primary microglia. Since I expect the 
CSF Myeloid cells to be latently infected, because they have been obtained from 
patients receiving ART, I will use the TFs with differential binding scores 
between latent infection and uninfected. We would like to know if the TF activity
inferred from RNAseq data of Myeloid2 cells correlates with these differential 
binding scores. In other words, we want to know if the patterns of differential 
TF activity are similar to the patterns of differential TF binding scores. 
For example, if a TF was shown to be more bound in latently infected primary
microglia compared to uninfected primary microglia, we would expect the TF to be
more active in a group of Myeloid2 cells which are latently infected compared
to a group of Myeloid2 cells which are uninfected. 

Comparing the TFs from the TF activity matrix with the differentially
bound TFs from the footprinting, twentyfour of the TFs overlap.

Using less than 10,000 most highly variable genes as an input matrix for SCENIC 
resulted in a lower number of TFs which overlap. Increasing the number of input 
features for SCENIC increased the number of TFs included in the TF activity 
matrix. Therefore, 10,000 most highly variable genes were used.

```{r}
# read in dataframe containing differntial binding scores between conditions
#
# red = latently infected
# gfp = active infected
# uninfected
tfs_ana <- read.table("/media/ag-cherrmann/projects/06_HIV_Microglia/data/atacseq/data-2020-11-06/tobias/TOBIAS_snakemake/footprint_mglia2_GlassTF_17-03/TFBS/bindetect_results.txt", sep = "\t", header = TRUE)

# filter TFs with significant p values between latent infection and uninfected
# I decided to use a threshold of p-value < 1e-50, because due to the large number 
# of data, all p-values will be very high
top_tfs <- tfs_ana %>% filter(uninf_red_pvalue < 1e-50)

# keep only the TFs in the dataframe which overlap between the two datasets.
overlap_atac <- top_tfs %>% filter(name %in% colnames(auc_mtx))

# extract names of TFs more bound in either condition
up_latent <- overlap_atac$name[overlap_atac$uninf_red_change < 0]
up_uninf <- overlap_atac$name[overlap_atac$uninf_red_change > 0]

```

## Volcano Plot ATAC-seq footprinting

Only four TFs found in the TF activity matrix of Myeloid2 cells 
are more bound in latent infection than in uninfected primary microglia.

```{r, fig.height=8, fig.width=12}

top_tfs %>% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = ifelse(uninf_red_change < 0, "up_latent", "up_uninf")) %>%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  mutate(label = ifelse(name %in% overlap_atac$name, name, NA)) %>% 
  ggplot(aes(x = uninf_red_change,
             y = -log10(uninf_red_pvalue),
             col = condition, 
             label = label)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-50), col = "red") +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = "TFs with differential binding scores") +
  ylab("-log10(p_value)") +
  xlab("differential binding score")

```


# Compare TF activities between clusters and look for patterns

Judging from the heatmaps below no TFs clearly separate the clusters or
show a pattern of activity which is similar to the footprinting binding scores. 
If the pattern was similar to the footprinting binding scores we would expect to
see one cluster where REST, KLF4, PAX5 and MXI1 are upregulated. This cannot be
seen in the plots below. 

## Split Heatmap into 3 separate heatmaps according to the different patients {.tabset}

### all 125 TFs

```{r}


# TF annotations
la <- rowAnnotation(type_TF = 
                      case_when(colnames(auc_mtx) %in% up_latent ~ "up_latent",
                                        colnames(auc_mtx) %in% up_uninf  ~ "up_uninf",
                                        TRUE ~"unknown"),
                     name = "type_TF",
                     col = list(type_TF = c("up_latent" =  "red", 
                                            "up_uninf" =  "pink",
                                            "unknown" = "forestgreen")))
  
```


```{r, fig.width=13, fig.height=14}
# 3 heatmaps with all Scenic TFs
ht_list <- map(seq.int(1:3), function(n) {
  patients <- c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF") # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df),]
  # heatmap
  ht <- Heatmap(t(mtx), 
               column_title = paste0("patient: ", patients[n]), 

               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
               show_row_names = TRUE
               ) 
               #top_annotation = column_annotation) 
  list(name = patients[n], heatmap = ht)
})


ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap + la


```

### 24 TFs found by ATAC-seq footprinting

```{r, fig.width=13, fig.height=8}
# 3 heatmaps, but only the TFs which overlap with Anas analysis
ht_list <- map(seq.int(1:3), function(n) {
  patients <- c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF") # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  # select TFs which overlap with Anas analysis
  mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), overlap_atac$name]
  # heatmap
  ht <- Heatmap(t(mtx), 
               column_title = paste0("patient: ", patients[n]), 

               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
               right_annotation = rowAnnotation(type_TF = ifelse(colnames(auc_mtx[,overlap_atac$name]) %in% up_latent, "up_latent", "up_uninf"),
                                                name = "type_TF",
                                                col = list(type_TF = c("up_latent" =  "red",
                                                                       "up_uninf" =  "pink"))))
  list(name = patients[n], heatmap = ht)
})

  
ht_list = ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap
draw(ht_list, ht_gap = unit(1, "cm"))
```

## Different approaches

The next idea was to cluster the Myeloid2 cells based on their TF activity using
k-means clustering. To do this I performed k-means clustering on 
Myeloid2 cells from patient one (HIV1_CSF) with k=3/4/5. The reason why I used only
patient one was that the other two patients might be used for validation of the 
clustering later on. If the clusters have a biological meaning, it should be 
possible to transfer the clusters onto the Myeloid2 cells from patient2 
(HIV2_CSF) and patient3 (HIV3_CSF).
If the clusters also separate the Myeloid2 cells of the other two patients in 
an interpretable and meaningful way this would validate that there is a biological meaning behind the clusters.

However, as we will see the label transfer of the cluster identities did not 
work, probably due to the small sample size. There are only 146 in total in the 
Myeloid2 cluster of infected patients.

The alternative strategy was to perform independent k-means clustering 
across all patients and compare the TF activities between clusters. 
Can we find TF activities which differentiate clusters across patients and find
correlations between clusters across patients?

1. Determine k for k-means k-means clustering
2. Transfer the cluster identities to patient2 and patient3
3. K-means Clustering for all 3 patients separately


# 1. Determine k for k-means clustering {.tabset}

The plot of within cluster sum of squares shows that the sweet spot for the 
value of k is somewhere between 3 and 5. To determine the best k for k-means
clustering k = 3, k = 4 and k = 5 were tried and the different outcomes compared.
Since k = 4 and k = 5 does not yield an increased number of TFs which differentiate
the different clusters and, therefore, do not add any insights we proceeded with
k = 3 in the following analysis.

```{r}
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                    orig.ident == "HIV1_CSF")
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]

# Decide on a k for clustering
wss = sapply(2:15, function(k) {
  kmeans(x = t(mtx), centers = k)$tot.withinss
})

plot(2:15, wss, type = "b", 
     xlab = "Number of clusters k", 
     ylab = "Total within-clusters sum of square")
```

The results of using differnet k can be seen below. Click on the tabs to see heatmaps of the k-means clustering, as well as statistical test on different TF
acitivities between clusters.

## k-means Clustering with k = 3

### Heatmap for patient 1

```{r, fig.width=15, fig.height=17}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_pat1 <-draw(Heatmap(t(mtx), 
             column_title = "Myeloid2 cells from patient 1, k = 3", 

             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             ))
             #top_annotation = column_annotation) 
```


After performing k-means cluster we extract the cells from each cluster. 
In the table below you can see the number of cells in each cluster. 

```{r}
### Extract cells from each cluster


# if you draw the heatmap, the results of th eclustering will not change anymore
#png("/media/ag-cherrmann/kmikulik/HIV_microglia/analysis/Myeloid2_cluster/Myeloid2_3clusters.png", width = 15, height = 17, res = 1200)
#ht <- draw(ht_pat1)
# get the columns names of cluster one
c1 <- colnames(t(mtx[column_order(ht_pat1)[[1]],]))
#write.table(c1, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv", sep = "\t")

# get the column names of cluster two
c2 <- colnames(t(mtx[column_order(ht_pat1)[[2]],]))
#write.table(c2, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv", sep = "\t")

# get the column names of cluster 3
c3 <- colnames(t(mtx[column_order(ht_pat1)[[3]],]))
#write.table(c3, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv", sep = "\t")
```


#### number of cells in each cluster 

```{r, results = "asis"}
# summary staticstics
hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3))%>% 
  mutate(ht_cluster = case_when(cell %in% c1 ~ "c1", 
                                cell %in% c2 ~ "c2",
                                cell %in% c3 ~ "c3")) %>%
  group_by(ht_cluster) %>% 
  summarize(count = n()) %>% 
  kable(caption = "number of cells in each cluster")

```

#### Are there significant differences in TF activity between the three clusters?


The ANOVA test is parametric and assumes normally distributed data, while
Kruskal-Wallis-Test compares the mean of one or more groups, but is 
non-parametric. Since the data is not normally 
distributed in this case, the Kruskal-Wallis-Test is more appropriate.
The Null-Hypothesis of the Kruskal-Wallis-Test is that there is no difference
between the groups.

Since the test is repeated 125 times, the p-values are adjusted with the
FDR method. This methods is more powerful than more conservative methods, like
Bonferroni, Holm or Hochberg. 

Performing Kruskal-Wallis-Test to find differences between the three clusters in patient one yields 54 TFs of all 125 TFs which are differentially active between the three clusters(p-value < 0.05). Of the twentyfour TFs with differential binding scores in the C20 cell line, sixteen have significantly different activity between the clusters.

Now that it is known that there are significant differences between clusters, 
the next step is to identify which cluster/s are different from which other cluster/s.

Pairwise-Wilcox-Test is a non-parametric pairwise test. Again, because 
we repeat the test we will correct for multiple hypothesis testing using the 
FDR method. The results are shown in boxplots below.

</details>

<details> 
<summary>Kruskal-Wallis-Test, p-value adjustment, posthoc tests</summary>

**Kruskal-Wallis Test and p-value adjustments for all 125 TFs**

```{r}
# non-parametric alternative to ANOVA -> Kruskal_wallis test
k3_df <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) 
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))["p.value"]
  #list(tf = tf, p_value = p_values)
})

# plot distribution of p_values
hist(as.numeric(unname(unlist(p_values))), 
     main = "distribution of p_values",
     xlab = "p_values",
     breaks = 20)

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
sign3 <- adj_p_values[adj_p_values<.05]
length(adj_p_values[adj_p_values<0.05])

# get TFs that are significantly different
hiv1_diff <- colnames(auc_mtx)[adj_p_values < .05]

#write.table(diff3, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1")
```

* **Kruskal-Wallis test and p-value adjustment  for 24 TFs also identified by ATAC-seq**


```{r}
# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])

# get TFs that are differentially expressed
hiv1_atac <- overlap_atac$name[adj_p_values < .05]
#write.table(diff3_atac, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1")
```

* **Posthoc tests for k = 3**


The posthoc test was performed for the 16 TFs with significantly different activity between the 3 clusters.

```{r, warning = FALSE}
# We perform a posthoc test
#wilcox_list = list()
wilcox_list <- map(seq.int(1:length(hiv1_atac)), function(i){
  tf <- hiv1_atac[i]
  #print(tf)
  wilcox <- pairwise.wilcox.test(pull(k3_df[tf]), pull(k3_df["ht_cluster"]), 
                                 p.adjust.method = "fdr")
  list(TF = tf, wilcox = wilcox)
})
  
#length(wilcox_list)
#wilcox_list
```

---
</details>

---

## K-means clustering with k = 4

### Heatmap for patient 1

```{r, fig.width=15, fig.height=17}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht <- draw(Heatmap(t(mtx), 
             column_title = "Myeloid2 cells from patient 2, k = 4", 

             column_km = 4,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             )) 
             #top_annotation = column_annotation) 


```


```{r}
### Extract cells from each cluster of k = 4

# get the columns names of cluster one
cl1 <- colnames(t(mtx[column_order(ht)[[1]],]))

# get the column names of cluster two
cl2 <- colnames(t(mtx[column_order(ht)[[2]],]))


# get the column names of cluster 3
cl3 <- colnames(t(mtx[column_order(ht)[[3]],]))

cl4 <- colnames(t(mtx[column_order(ht)[[4]],]))
```


```{r, results = "asis"}
# summary staticstics
hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = case_when(cell %in% cl1 ~ "c1", 
                                cell %in% cl2 ~ "c2",
                                cell %in% cl3 ~ "c3",
                                cell %in% cl4 ~ "c4")) %>% 
  group_by(ht_cluster) %>% 
  summarize(count = n()) %>% 
  kable(caption = "number of cells in each cluster")

```

#### Are there significant differences in TF activity between the four clusters?


Performing Kruskal-Wallis-Test to find differences between the four clusters in patient one yields 51 TFs of all 125 TFs which are differentially active between the three clusters(p-value < 0.05). Of the twentyfour TFs with differential binding scores in the C20 cell line, fourteen have significantly different activity between the clusters.

Now that it is known that there are significant differences between clusters, 
the next step is to identify which cluster/s are different from which other 
cluster/s. For this pairwise Wilcox test was used.


</details>

<details> 
<summary>Kruskal-Wallis-Test, p-value adjustment, posthoc tests</summary>

* **Kruskal-Wallis Test and p-value adjustments for all 125 TFs**

```{r}

k4_df <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(cl1,cl2,cl3,cl4)) %>% 
  mutate(ht_cluster = ifelse(cell %in% cl1, "cl1", 
                             ifelse(cell %in% cl2, "cl2", 
                                    ifelse(cell %in% cl3, "cl3", "cl4"))))

# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(k4_df[tf]) ~ ht_cluster, k4_df))["p.value"]
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")

# how many transcription factors are significantly different between the clusters?
sign <- adj_p_values[adj_p_values < 0.05]
length(adj_p_values[adj_p_values<0.05])


# get the TFs that are differentially expressed
diff4 <- colnames(auc_mtx)[adj_p_values < 0.05]

#write.table(diff4, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=4")

```

* **Kruskal-Wallis Test and p-value adjustments for 14 TFs with differential binding scores**


```{r}
# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(k4_df[tf]) ~ ht_cluster, k4_df))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])

# get the TFs that are differentially expressed
diff4_atac <- overlap_atac$name[adj_p_values < .05]

#write.table(diff4_atac, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=4")

```

* **Pairwise-Wilcox_Test for k = 4**

Posthoc test for the 16 TFs with significantly different activity between the four clusters.

```{r, warning = FALSE}
# We perform a posthoc test
#wilcox_list = list()
wilcox_list <- map(seq.int(1:length(diff4_atac)), function(i){
  tf <- diff4_atac[i]
  print(tf)
  wilcox <- pairwise.wilcox.test(pull(k4_df[tf]), pull(k4_df["ht_cluster"]), 
                                 p.adjust.method = "fdr")
  list(TF = tf, wilcox = wilcox)
})
  
#length(wilcox_list)
#wilcox_list
```
---
</details>

---

## K-means Clustering with k = 5

### Heatmap for patient 1

```{r, fig.width=15, fig.height=17}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht <- draw(Heatmap(t(mtx), 
             column_title = "Myeloid2 cells from patient 1, k = 5", 

             column_km = 5,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             )) 

```


```{r}
##### Extract cells from each cluster for k= 5

# get the columns names of cluster one
clu1 <- colnames(t(mtx[column_order(ht)[[1]],]))

# get the column names of cluster two
clu2 <- colnames(t(mtx[column_order(ht)[[2]],]))


# get the column names of cluster 3
clu3 <- colnames(t(mtx[column_order(ht)[[3]],]))

clu4 <- colnames(t(mtx[column_order(ht)[[4]],]))

clu5 <- colnames(t(mtx[column_order(ht)[[5]], ]))

```


```{r, results = "asis"}
# summary staticstics
hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = case_when(cell %in% clu1 ~ "c1", 
                                cell %in% clu2 ~ "c2",
                                cell %in% clu3 ~ "c3",
                                cell %in% clu4 ~ "c4",
                                cell %in% clu5 ~ "c5")) %>% 
  group_by(ht_cluster) %>% 
  summarize(count = n()) %>% 
  kable(caption = "number of cells in each cluster")

```

#### Are there significant differences in TF activity between the four clusters?

Performing Kruskal-Wallis-Test to find differences between the four clusters in patient one yields 59 TFs of all 125 TFs which are differentially active between the three clusters(p-value < 0.05). Of the twentyfour TFs with differential binding scores in the C20 cell line, fifteen have significantly different activity between the clusters.

Now that it is known that there are significant differences between clusters, 
the next step is to identify which cluster/s are different from which other 
cluster/s. For this pairwise Wilcox test was used.


</details>

<details> 
<summary>Kruskal-Wallis-Test, p-value adjustment, posthoc tests</summary>

* **Kruskal-Wallis Test and p-value adjustments for all 125 TFs**

```{r}
# all TFs
k5_df <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(clu1,clu2,clu3,clu4, clu5)) %>% 
  mutate(ht_cluster = ifelse(cell %in% clu1, "clu1", 
                             ifelse(cell %in% clu2, "clu2", 
                                    ifelse(cell %in% clu3, "clu3",
                                           ifelse(cell %in% clu4, "clu4", "clu5")))))
n = 1
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(k5_df[tf]) ~ ht_cluster, k5_df))["p.value"]
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")

# how many transcription factors are significantly different between the clusters?
sign <- adj_p_values[adj_p_values < 0.05]
length(adj_p_values[adj_p_values<0.05])


# get the TFs that are differentially expressed
diff5 <- colnames(auc_mtx)[adj_p_values < 0.05]

#write.table(diff5, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=5")
```

* **Kruskal-Wallis Test and p-value adjustments for all 125 TFs**

```{r}
# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(k5_df[tf]) ~ ht_cluster, k5_df))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])

# get the TFs that are differentially expressed
diff5_atac <- overlap_atac$name[adj_p_values < .05]

#write.table(diff5_atac, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=5")
```

* **Pairwise-Wilcox_Test for k = 5**

Posthoc test for the 15 TFs with significantly different activity between the four clusters.

```{r, warning = FALSE}
# We perform a posthoc test
#wilcox_list = list()
wilcox_list <- map(seq.int(1:length(diff5_atac)), function(i){
  tf <- diff5_atac[i]
  print(tf)
  wilcox <- pairwise.wilcox.test(pull(k5_df[tf]), pull(k5_df["ht_cluster"]), 
                                 p.adjust.method = "fdr")
  list(TF = tf, wilcox = wilcox)
})
  
#length(wilcox_list)
#wilcox_list
```

---
</details>

---


## Which Clusters are significantly different for different k? {.tabset}

The results from the statistical tests on difference in TF activity between
clusters can be seen in the boxplots below. Sixteen, fourteen and twelve TFs
were found to be differentially active between the clusters for k equal three, 
four and five respectively. The boxplots below also show very nicely, that 
the data are not normally distributed and, therefore, non-parametric tests, 
like the Kruskal-Wallis-Test and pairwise Wilcox-Test were used.


### k = 3

```{r, fig.height=20, fig.width=25}

test <- k3_df
#tf.oi <- c("CTCF", "REST", "YY1")
#names(tf.oi) <- tf.oi
test.list <- map(hiv1_atac, function(tf) {
  ggplot() +
  geom_violin(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf), 
                   fill = test %>% pull("ht_cluster"), alpha = .6)) +
  geom_boxplot(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf)), 
               alpha = 0) +
  stat_pvalue_manual(test %>% 
    pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>%
      add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster")) })

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 4))
```

### k = 4

```{r, fig.width=25, fig.height=15}

test <- k4_df

test.list <- map(diff4_atac, function(tf) {
  ggplot() +
  geom_violin(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf), 
                   fill = test %>% pull("ht_cluster"), alpha = .6)) +
  geom_boxplot(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf)), 
               alpha = 0) +
  stat_pvalue_manual(test %>% 
    pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>%
      add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster")) +
    geom_violin(alpha = 0)
})

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 3))


```


### k = 5
```{r, fig.width=25, fig.height=15}
test <- k5_df

test.list <- map(diff5_atac, function(tf) {
  ggplot() +
  geom_violin(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf), 
                   fill = test %>% pull("ht_cluster"), alpha = .6)) +
  geom_boxplot(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf)), 
               alpha = 0) +
  stat_pvalue_manual(test %>% 
    pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>%
      add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 3))
```

---

## Conclusion on how many k's to use for k-means clustering

As mentioned, k = 3 was used for the subsequent analysis, since more clusters 
did not add additional insights.

The Venn diagram below shows very nicely, that out of all TFs with different
activity between clusters, 68% overlap for all k (k=3/4/5) that I tested (right plot). 
75% of the TF with differential binding scores in the C20 cell line overlapped
for all k (left plot). 

Therefore, I decided to proceed with 
k = 3 for the following analysis. 

```{r}
diff5_atac <- as.vector(read.table( "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=5"))[["x"]]
diff5 <- as.vector(read.table( "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=5"))[["x"]]


diff4_atac <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=4"))[["x"]]

diff4 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=4"))[["x"]]
# list of TFs from ATAC-seq differntially active between the 3 heatmap 
#clusters from patient 1
diff3_atac <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1"))[["x"]]

# list of TFs differntially active between the 3 heatmap clusters from patient 1
diff3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1"))[["x"]]
```


```{r, fig.width = 12}
p1 <- ggVennDiagram(list(diff3_atac, diff4_atac, diff5_atac),
              label_alpha = 0,
              category.names= c("k = 3", "k = 4", "k = 5"),
              #set.size = 20,
              label = "percent",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "TFs with differential binding scores in C20 cell line")
  #scale_fill_gradient (low = "lightblue", high = "darkblue")

p2 <- ggVennDiagram(list(diff3, diff4, diff5),
              label_alpha = 0,
              category.names= c("k = 3", "k = 4", "k = 5"),
              label = "percent",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "all TFs with different activity")

fig <- ggarrange(p1, p2, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob("TFs with significantly different activity  between clusters for different k", size = 20))
```



# 2. Transfer the 3 cluster identitites to patient2 and patient3 

The cluster identities from patient 1 were transferred to patient 2 and patient 3
to assign each cell to one of the 3 clusters. 


<details>
<summary>**Cluster annotation using an integrated referenc**</summary>

Label Transfer

Here we do not integrate the datasets beforehand and we do not modify the query
expression data.

```{
Idents(hiv7) <- "cell_type"
hiv <- subset(hiv7, idents = "Myeloid2")

# the cell type is replaced by the heatmap cluster
hiv1_my2 <- subset(hiv7, cells = c(c1,c2,c3))

clusters <- (
  hiv1_my2@meta.data %>% 
  rownames_to_column("cell") %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) %>% 
  column_to_rownames("cell"))["ht_cluster"]

hiv <- AddMetaData(hiv, metadata = clusters, col.name = "cell_type")

# the seurat_cluster number is replaced by the number of the heatmap cluster
seurat_clusters <- (
  hiv1_my2@meta.data %>% 
  rownames_to_column("cell") %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, 1, ifelse(cell %in% c2, 2, 3))) %>% 
  column_to_rownames("cell"))["ht_cluster"]

hiv <- AddMetaData(hiv, metadata = seurat_clusters, col.name = "seurat_clusters")


# split the object according to the patients
hiv_list <- SplitObject(hiv, split.by = "orig.ident")[3:5]
# normalize each object
for (i in names(hiv_list)) {
    hiv_list[[i]] <- SCTransform(hiv_list[[i]], verbose = FALSE)
}


# patient 2 and patient 3 are our query datasets
query <-merge(hiv_list[[1]], hiv_list[[3]])
reference <- hiv_list[[2]]
reference <- RunPCA(reference, 
                    # by default a partial PCA is run, but we compute mostgular values -> run standard SVD instead
                    approx = FALSE, 
                    verbose = FALSE)

hiv_anchors <- FindTransferAnchors(reference = reference,
                                   query = query, 
                                   dims = 1:30, verbose = FALSE,
                                   k.filter = NA, 
                                   reference.reduction = "pca",
                                   normalization.method = "SCT",
                                   reduction = "pcaproject")

predictions <- TransferData(anchorset = hiv_anchors, 
                            refdata = reference$cell_type,
                            k.weight = 34, dims = 1:30, verbose = FALSE)

query <- AddMetaData(query, metadata = predictions)


hiv2 <- subset(query, orig.ident == "HIV2_CSF")
hiv3 <- subset(query, orig.ident == "HIV3_CSF")
#saveRDS(hiv2,"/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv2_CSF_myleoid2_heatmap_cluster_label_transfer.rds")
#saveRDS(hiv3, "/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv3_CSF_myleoid2_heatmap_cluster_label_transfer.rds")
```
Now that we have transferred the cluster identitities to patient2 and patient3
and saved the results in two separate Seurat objects, we can add the TF activity 
score to the metadata of each patient.

```{r}
hiv2 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv2_CSF_myleoid2_heatmap_cluster_label_transfer.rds")
hiv3 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv3_CSF_myleoid2_heatmap_cluster_label_transfer.rds")

hiv2_scenic <- AddMetaData(hiv2, metadata = auc_mtx,
                           col.name = colnames(auc_mtx))
hiv3_scenic <- AddMetaData(hiv3, metadata = auc_mtx,
                           col.name = colnames(auc_mtx))
```

---
</details>

## Predicted cluster labels

The heatmaps below show the predicted clusters from label transfer for
patient 2 and patient 3

The clusters obtained after transferring the cluster identities from patient 1 to
patient2 do not seem to be very good at separating cells according to TF activity.

When you transfer the cluster identities from patient1 to patient3 all cells, but
one, are assigned to cluster1. This shows that the label transfer did not work 
very well. This is probably due to the low number of Myeloid2 cells in each sample.

```{r, fig.width=15, fig.height=17}
cluster_anno <- as.data.frame(hiv2$predicted.id)

#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV2_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_hiv2_label <- Heatmap(t(mtx), 
             column_title = "Predicted Clusters from Label Transfer for HIV2_CSF",             
             column_title_gp = gpar(fontsize = 15),
             column_split = paste0("cluster", hiv2$predicted.id),#cluster_anno,
             #column_km = 4,
             #column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la,
             top_annotation = HeatmapAnnotation(df = cluster_anno, name = "predicted cluster")
             ) 
             #top_annotation = column_annotation) 
```


```{r, fig.width=15, fig.height=17}
cluster_anno <- as.data.frame(hiv3$predicted.id)

#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV3_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_hiv3_label <- Heatmap(t(mtx), 
             column_title = "Predicted clusters from Label Transfer for HIV3_CSF", 
             column_title_gp = gpar(fontsize = 15),
             column_split = paste0("cluster", hiv3$predicted.id),#cluster_anno,
             #column_km = 4,
             #column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la,
             top_annotation = HeatmapAnnotation(df = cluster_anno, name = "predicted cluster")
             ) 
             #top_annotation = column_annotation) 

```

```{r, fig.width=15, fig.height=20}
ht_list <- ht_hiv2_label + ht_hiv3_label
draw(ht_list, ht_gap= unit(2, "cm"))
```



# 3. K-means clustering for all 3 patients separately 

Since the label transfer of the 3 cluster identities from patient1 did not seem
to work very well, we next tried to perform k-means clustering using k = 3
for all 3 patients separately. This way we might be able to see if the TF
activities in the 3 clusters show similar patterns across the different patients.

In the two heatmaps below the clusters look more sensible than the clusters
observed after cluster label transfer.

## k = 3 {.tabset}

### Patient 2

#### K-means clustering with k = 3 for patient 2 

##### 1. Draw Heatmap for Patient 2


```{r, fig.width=15, fig.height=20}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV2_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht2 <- Heatmap(t(mtx), 
             column_title = "Patient 2, HIV2_CSF", 
             column_title_gp = gpar(fontsize = 15),
             #column_split = paste0("cluster", hiv2$predicted.id),#cluster_anno,
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la,
             #top_annotation = HeatmapAnnotation(df = cluster_anno, name = "predicted cluster")
             ) 
             #top_annotation = column_annotation) 
ht_pat2 <- draw(ht2)

```

##### 2. Kruskal_Wallis Test Patient 2 

```{r}
# extract the cells from the 3 clusters 
# get the cell names of cluster one
c1 <- colnames(t(mtx[column_order(ht_pat2)[[1]],]))
# get the cell names of cluster two
c2 <- colnames(t(mtx[column_order(ht_pat2)[[2]],]))
# get the cell names of cluster 3
c3 <- colnames(t(mtx[column_order(ht_pat2)[[3]],]))


# add cluster identities to Seurat object metadata and return metadata as dataframe
pat2 <- hiv7_scenic@meta.data %>% rownames_to_column("cell")
pat2 <- pat2 %>% 
  filter(orig.ident == "HIV2_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) 


# check all 125 TFs
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(pat2[tf]) ~ ht_cluster, pat2))["p.value"]
  #list(tf = tf, p_value = p_values)
})


# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")
length(adj_p_values[adj_p_values<0.05])
## [1] 66

# get the TFs taht are differntially active
diff_hiv2 <- colnames(auc_mtx)[adj_p_values < 0.05]


# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(pat2[tf]) ~ ht_cluster, pat2))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])
## [1] 18

# get the TFs that are differentially expressed
hiv2_atac <- overlap_atac$name[adj_p_values < .05]
```

### Patient 3

#### K-means clustering with k = 3 for patient 3

##### 1. Draw Heatmap for Patient 3

```{r, fig.width=15, fig.height=20}
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV3_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht3 <- Heatmap(t(mtx), 
             column_title = "Patient 3, HIV3_CSF", 
             column_title_gp = gpar(fontsize = 15),
             #column_split = paste0("cluster", hiv2$predicted.id),#cluster_anno,
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la,
             #top_annotation = HeatmapAnnotation(df = cluster_anno, name = "predicted cluster")
             ) 
             #top_annotation = column_annotation) ich
ht_pat3 <- draw(ht3)
```


##### 2. Kruskal-Wallis test Patient 3

```{r}
# extract the cells from the 3 clusters 
# get the cell names of cluster one
c1 <- colnames(t(mtx[column_order(ht_pat3)[[1]],]))
# get the cell names of cluster two
c2 <- colnames(t(mtx[column_order(ht_pat3)[[2]],]))
# get the cell names of cluster 3
c3 <- colnames(t(mtx[column_order(ht_pat3)[[3]],]))


# add cluster identities to Seurat object metadata and return metadata as dataframe
pat3 <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV3_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) 


# check all 125 TFs
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(pat3[tf]) ~ ht_cluster, pat3))["p.value"]
  #list(tf = tf, p_value = p_values)
})


# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")
length(adj_p_values[adj_p_values<0.05])
## [1] 66

# get the TFs taht are differntially active
diff_hiv3 <- colnames(auc_mtx)[adj_p_values < 0.05]


# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(pat3[tf]) ~ ht_cluster, pat3))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])
## [1] 18

# get the TFs that are differentially expressed
hiv3_atac <- overlap_atac$name[adj_p_values < .05]
```


### comparison between all 3 patients

##### Heatmaps of k-means clustering with k = 3 for all three patients

```{r, fig.width=15, fig.height=17}
# patient 1
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht1 <- Heatmap(t(mtx), 
             column_title = "Patient 1, HIV1_CSF", 
             column_title_gp = gpar(fontsize = 15),
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             )
             #top_annotation = column_annotation) 
```

```{r, fig.width=15, fig.height=20}
ht_list <- ht1 + ht2 + ht3
draw(ht_list, ht_gap= unit(2, "cm"), 
     column_title = "Independent k-means clustering (k = 3) for all patients",
     column_title_gp = gpar(fontsize = 20) )
```

---


# Which TFs show significantly different activity between the clusters across patients?

Since the independent k-means clustering for all three patients yields 
clusters which look more reasonable, we will compare the three clusters we obtain 
across the three patients to see if there is a pattern between them. 

We already saw that there are a few TFs which show significantly different
activity between the clusters in patient1.

Next, we want to see if the same group of TFs characterize the three 
clusters in all three patients or if they are different. 
We will test which TFs are significantly different between clusters and then
use a pairwise Wilcox test to determine which clusters are different.


### Are TFs differentiating the three clusters similar across all three patients?

In the Venn diagram below it can be seen that 46 with significantly different
activities between clusters overlap across patients.

Additionally, eleven of the TFs found to have differential binding scores between
infection conditions in C20 microglia cell line show different activity between the 
three clusters across all patients. These are the TFs we are most interested in,
since we want to see if we can find patterns that correlate with the binding 
scores in C20 microglia.

```{r}
p1 <- ggVennDiagram(list(hiv1_diff, diff_hiv2, diff_hiv3),
              label_alpha = 0,
              category.names= c("HIV1", "HIV2", "HIV3"),
              #set.size = 20,
              label = "count",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "TFs with different activity between 3 clusters")
```

```{r}
p2 <- ggVennDiagram(list(hiv1_atac, hiv2_atac, hiv3_atac),
              label_alpha = 0,
              category.names= c("HIV1", "HIV2", "HIV3"),
              #set.size = 20,
              label = "count",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "TFs with different activity between 3 clusters")
```

```{r}
ggarrange(p1, p2, ncol = 2, nrow = 1)
```


## Eleven TFs with different activities {.tabset}

In the boxplots below it can be seen that in all three patients there seems to
be one cluster which is different from two other clusters. 

```{r}
intersect <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/intersect_TFs_different_across_patients_3_clusters")[["x"]]
```


### Patient 1

```{r, fig.width=25, fig.height=10}
df <- k3_df


boxplot_list <- map(intersect, function(tf) {
  ggplot() +
    geom_violin(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf), 
                     fill = df %>% pull("ht_cluster"), alpha = .6)) +
    geom_boxplot(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf)), 
                 alpha = 0) +
    stat_pvalue_manual(df %>% 
                        pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>% 
                         add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))

#annotate_figure(figure, top = text_grob("Patient 1 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)", 
 #              color = "black", face = "bold", size = 25))
```

```{r}
ctcf1 <- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_violin(aes(fill = ht_cluster, alpha = .6)) +
  geom_boxplot(alpha = 0) +
  stat_pvalue_manual(df %>% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %>% 
                       add_xy_position())
ctcf1 <- ctcf1 + labs(title = "CTCF activity in patient 1")
```

---

### Patient 2

```{r, fig.width=25, fig.height=10}
df <- pat2

boxplot_list <- map(intersect, function(tf) {
  ggplot() +
    geom_violin(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf), 
                     fill = df %>% pull("ht_cluster"), alpha = .6)) +
    geom_boxplot(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf)), 
                 alpha = 0) +
    stat_pvalue_manual(df %>% 
                        pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>% 
                         add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))

#annotate_figure(figure, top = text_grob("Patient 2 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)", 
 #              color = "black", face = "bold", size = 25))
```

```{r}
ctcf2 <- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_violin(aes(fill = ht_cluster, alpha = .6)) +
  geom_boxplot(alpha = 0) +
  stat_pvalue_manual(df %>% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %>% 
                       add_xy_position())
ctcf2 <- ctcf2 + labs(title = "CTCF activity in patient 2")
```

---

### Patient 3

```{r, fig.width=25, fig.height=10}
df <- pat3

boxplot_list <- map(intersect, function(tf) {
  ggplot() +
    geom_violin(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf), 
                     fill = df %>% pull("ht_cluster"), alpha = .6)) +
    geom_boxplot(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf)), 
                 alpha = 0) +
    stat_pvalue_manual(df %>% 
                        pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>% 
                         add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))


#annotate_figure(figure, top = text_grob("Patient 3 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)", 
 #              color = "black", face = "bold", size = 25))
```

```{r}
ctcf3 <- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_violin(aes(fill = ht_cluster, alpha = .6)) +
  geom_boxplot(alpha = 0) +
  stat_pvalue_manual(df %>% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %>% 
                       add_xy_position())
ctcf3 <- ctcf3 + labs(title = "CTCF activity in patient 3")
```

---

## CTCF expression across patients

CTCF has been implicated to regulate HIV latency [@Jeffereys]. In the boxplots 
below it can be seen that there is a significant difference between the first 
cluster and the other two clusters in patient1 and patient2. Similarly in 
patient3 there is a difference, however the difference is not as significant.
This might imply that cluster contains a group of cells which are latently 
infected and therefore CTCF is more active.

```{r, fig.width=15, fig.height=5}
ggarrange(ctcf1, ctcf2, ctcf3, ncol = 3, nrow = 1)
```


## Correlations between clusters across patients

Using a correlation matrix we can visualize how clusters are correlated across
patients:


```{r}
# read in 3 Clusters for each of the 3 HIV+ patients

# HIV1_CSF
hiv1_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv", sep = "\t"))[["x"]]
hiv1_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv", sep = "\t"))[["x"]]
hiv1_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv", sep = "\t"))[["x"]]

# HIV2_CSF
hiv2_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster1.tsv", sep = "\t"))[["x"]]
hiv2_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster2.tsv", sep = "\t"))[["x"]]
hiv2_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster3.tsv", sep = "\t"))[["x"]]

# HIV3_CSF
hiv3_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster1.tsv", sep = "\t"))[["x"]]
hiv3_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster2.tsv", sep = "\t"))[["x"]]
hiv3_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster3.tsv", sep = "\t"))[["x"]]

cluster_anno <- c(hiv1_c1, hiv1_c2, hiv1_c3, hiv2_c1, hiv2_c2, hiv2_c3, hiv3_c1, hiv3_c2, hiv3_c3)
```


```{r}
# add cluster information to TF activity
auc_df <- as.data.frame(auc_mtx)

auc_df <- auc_df %>% 
  rownames_to_column("cell") %>% 
  filter(cell %in% cluster_anno) %>% 
  mutate(ht_cluster = case_when(cell %in% hiv1_c1 ~ "hiv1_c1", 
         cell %in% hiv1_c2 ~ "hiv1_c2", 
         cell %in% hiv1_c3 ~ "hiv1_c3",
         cell %in% hiv2_c1 ~ "hiv2_c1",
         cell %in% hiv2_c2 ~ "hiv2_c2",
         cell %in% hiv2_c3 ~ "hiv2_c3",
         cell %in% hiv3_c1 ~ "hiv3_c1",
         cell %in% hiv3_c2 ~ "hiv3_c2",
         cell %in% hiv3_c3 ~ "hiv3_c3"))
```

### Correlation between clusters based on all 125 TFs {.tabset}

The average TF activity of all 125 TFs was computed for every cluster. Based on 
the average TF activity in each cluster, the correlation between clusters was 
computed.

The scatter plot below shows the average expression for each TF across all 
three clusters in all three patients and the  corresponding correlation 
coefficients. The correlation matrix visualizes the Spearman correlation coefficient between all clusters with black numbers indicating the p-values which are not 
significant. Clusters two and three of all patients seem to be highly 
correlated, with cluster two of patient three being the exception. Cluster one of 
patient one and patient three are highly correlated with each other, but less correlated 
with clusters two and three. Cluster one of patient two is again an exception, because it 
is not as highly correlated with clusters one of patient two and patient three, but still
more correlated than with clusters two and three. 

Seeing these results, the hypothesis is that the cluster one, which showed different 
TF activity from clusters two and three correspond to a different infection state. 
Cells of cluster one could be latently infected cells. This hypothesis would be
supported by having a look at the role of TFs with different activity in 
cluster one. 



<details>
<summary>**Correlation computation**</summary>
  
Dataframe of average activity value for each TF in each patient and clusters.


```{r, results = "asis"}
# correlation matrix
df <- auc_df %>%
  group_by(ht_cluster) %>%
  summarize_if(is.numeric, mean) %>% # calculate mean TF activity for each cluster
  gather(TF, value, -ht_cluster) %>% 
  spread(ht_cluster, value) %>% 
  column_to_rownames("TF") 

df %>% head %>% kable (caption = "Average TF activity for each cluster") 

```
Correlation matrix: 

```{r, results = "asis"}
# save correlation matrix
cor_mtx <- df%>% cor(method = "spearman")

# show the correlation matrix
cor_mtx %>% kable(caption = "Correlation matrix of average TF acivity between all clusters")

# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor <- cor.mtest(cor_mtx, conf.level = 0.95, method = "spearman")
```

---
</details>

#### Scatter matrix

```{r, fig.width = 15, fig.height=15}

GGally::ggpairs(df) + theme_bw() +
  labs(title = "Scatter matrix for all conditions")
```

---

#### Correlation matrix

```{r}
df %>%
  cor(method = "spearman") %>%
  corrplot(
     order = "hclust", # hierarchical cluster order
     addrect = 4, # add boxes 
     p.mat = testCor$p, # add matrix of p-values
     insig = "p-value", # adds p-values as digits, not crosses
     tl.col = "black") # make labels black
```

---

<details>
*<summary>Additional correlation plots which do not aid the interpretation much</summary>*
  

Correlation between clusters based on all TFs with differential activity between clusters

```{r}
intersect_total <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_intersect_TFs_different_across_patients_3_clusters"))[["x"]]
```

```{r}
# correlation matrix
cor_mtx <- auc_df %>%
  select(ht_cluster, intersect_total) %>% 
  group_by(ht_cluster) %>%
  summarize_if(is.numeric, mean) %>% 
  gather(TF, value, -ht_cluster) %>% 
  spread(ht_cluster, value) %>% 
  column_to_rownames("TF") %>% cor(method = "spearman")

# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor <- cor.mtest(cor_mtx, conf.level = 0.95, method = "spearman")
```


```{r}
corrplot(cor_mtx,
         order = "hclust", 
         addrect = 4, # add boxes 
         p.mat = testCor$p, 
         insig = "p-value", 
         tl.col = "black")
```


---
</details>

---

### TF correlation across clusters {.tabset}

For the eleven TFs with different activity between clusters, the correlation 
was computed for the TF activities. 

The scatter matrix and correlation plot below show very nicely, that the
three TFs YY1, REST, MXI1 are positively correlated, but negatively correlated
with the rest of the TFs and vice versa. The fact that these TF activities 
correlate across all patients indicates that there might be a biological meaning 
behind the different clusters. Having a look at the TFs which are up- or
downregulated in cluster one compared to the other clusters could provide 
further insights. 

<details>
<summary>**Correlation computation**</summary>


```{r}
intersect_atac <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_intersect_TFs_different_across_patients_3_clusters"))[["x"]]
```


```{r, results = "asis"}
# correlation matrix
auc_df %>%
  select(intersect_atac) %>% head() %>% kable(caption = "TF activity of 11 TFs with different activity between clusters across all cells")
```

Correlation matrix for TFs:

```{r, results = "asis"}
# compute correlation matrix
cor_mtx <- auc_df %>%
  select(intersect_atac) %>% 
  cor(method = "spearman")
# show correlation matrix
cor_mtx %>% head() %>% kable(caption = "Correlation matrix for 11 TFs with different activity between clusters")

# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor <- cor.mtest(cor_mtx, conf.level = 0.95, method = "spearman")
```

---
</details>


#### Scatter matrix

```{r, fig.width=15, fig.height=15}
GGally::ggpairs(auc_df %>%
  select(intersect_atac)) + 
  theme_bw() +
  labs(title = "Scatter matrix for all TFs with different activity between clusters")

```

---

#### Correlation matrix 

```{r}
df <- auc_df %>%
  select(intersect_atac) %>% 
  cor(method = "spearman") %>%
  corrplot(
  order = "hclust", 
  addrect = 2, # add boxes 
  p.mat = testCor$p, 
  insig = "p-value",
  tl.col = "black") 
```

---

# Gene expression analysis

Using the information of the three clusters from above we will pool the two clusters
which are significantly different from the third cluster in all patients.
Consequently, in the following analysis we will compare Cluster1 to Cluster2 (pool of two similar 
clusters). The gene expression might give a hint about the identity of these 
two Myeloid2 clusters. 


```#{r}
# 3 Clusters for each of the 3 HIV+ patients

# HIV1_CSF
hiv1_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv", sep = "\t"))[["x"]]
hiv1_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv", sep = "\t"))[["x"]]
hiv1_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv", sep = "\t"))[["x"]]

# HIV2_CSF
hiv2_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster1.tsv", sep = "\t"))[["x"]]
hiv2_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster2.tsv", sep = "\t"))[["x"]]
hiv2_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster3.tsv", sep = "\t"))[["x"]]

# HIV3_CSF
hiv3_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster1.tsv", sep = "\t"))[["x"]]
hiv3_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster2.tsv", sep = "\t"))[["x"]]
hiv3_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster3.tsv", sep = "\t"))[["x"]]

cluster_anno <- c(hiv1_c1, hiv1_c2, hiv1_c3, hiv2_c1, hiv2_c2, hiv2_c3, hiv3_c1, hiv3_c2, hiv3_c3)
```



### TF gene expression of the 11 TFs with different activity between clusters

```{r}
hivmy2 <- subset(hiv7, cell_type == "Myeloid2" & orig.ident %in% c("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))

# add cluster identity to metadata
ht_cluster <- (
  hivmy2@meta.data %>% 
  rownames_to_column("cell") %>% 
  mutate(ht_cluster = ifelse(cell %in% c(hiv1_c1, hiv2_c1, hiv3_c3),
                             "c1", 
                             "c2")) %>% 
  column_to_rownames("cell"))["ht_cluster"]

hivmy2 <- AddMetaData(hivmy2, metadata = ht_cluster, col.name = "ht_cluster")
#saveRDS(hivmy2,"/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv_myeloid2_subclusters")
```

## Quality of the Myeloid 2 clusters

There is an evident difference in library size and number of features between
the two clusters. This could mean that there is a technical difference between
the two clusters. Conversely, it could also mean that there is this a biological
difference. For example, if cells of cluster1 represent a latently infected group 
of Myeloid2 cells it could be that gene expression is reduced overall due to the
infection compared to healthy uninfected cells from cluster2.

One way to investigate this further would be to have a look at the housekeeping 
genes in both clusters. If housekeeping genes were more highly expressed in 
one cluster than in the other this would indicate that the difference is not 
biological, but a technical artefact.

Since there is one cluster which has lower number of counts and features 
to start with, further qualitative analysis of gene expression might be skewed 
as well. For this reason the following analysis on the difference between these
two clusters will be mainly qualitative. Due to the low number of cells it 
is not possible to get significant results for the differential gene expression.


```{r}
p1 <- hivmy2@meta.data %>% 
  ggplot() +
  geom_boxplot(aes(x = ht_cluster, y = nFeature_RNA, fill = ht_cluster))

p2 <- hivmy2@meta.data %>% 
  ggplot() +
  geom_violin(aes(x = ht_cluster, y = nFeature_RNA, fill = ht_cluster)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1))

fig <- ggarrange(p1, p2, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob("Quality of Clsuters"))
```

## Housekeeping genes

A list of 3804 houseekping genes were used from [@Eisenberg].

A quantitative analysis might be difficult, because due to the small sample
size and large feature number, we get very high p_values for all genes in 
differential gene expression analysis. Visualization of the
housekeeping gene expression, unfortunately does not help either with the 
interpretation.


<details>
*<summary>Boxplots of housekeeping gene expression</summary>*

```{r}
hk_genes <- (read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/HK_genes.txt"))[["V1"]]
```

```{r}
# filter only genes present in the scRNAseq count matrix
hk_genes_filt <- hk_genes[hk_genes %in% rownames(hivmy2@assays$RNA@counts)]
length(hk_genes)
length(hk_genes_filt)

# Find genes from the housekeeping genes that are differentially expressed 
# between the clusters
hk_diff <- FindAllMarkers(hivmy2, features = hk_genes_filt)
Idents(hivmy2) <- "ht_cluster"
hk_diff <- FindMarkers(hivmy2, features = hk_genes_filt, ident.1 = "c1", ident.2 = "c2")
hk_diff %>% head

```

```{r, fig.width=15, fig.height=30}
test <- hk_genes_filt[1:30]
VlnPlot(hivmy2, features = test, group.by = "ht_cluster", ncol = 4)
```

---
</details>

# Appendix

## Gene expression analysis of the two Myeloid2 subclusters

#### TF from the Farhadian et al.

These TFs were used as marker genes for the microlgia like Myeloid2 cell cluster.
Is there a group of cells which has a higher expression of these markers?
No clear difference in gene expression can be seen.


<details>
<summary>Boxplots of microglia marker genes</summary>


```{r, fig.width = 15, fig.height=40}
#import the 60 genes that were upregulated in Myeloid-2 in the paper compared to the four other myeloid subsets
paper <- read.table(file = "/media/ag-cherrmann/kmikulik/HIV_microglia/data/Myeloid-2_genes_paper.csv",
                    sep = "\t", header = FALSE, skipNul = TRUE)#, n_max = 60)
paper[1,] <- "APOC1"
colnames(paper) <- ""
paper <- as.vector(paper[,1])


VlnPlot(hivmy2, features = paper, group.by = "ht_cluster")
```

---
</details>


#### Differential gene expression analyisis

<details>
<summary>Differential gene expression analysis</summary>


```{r, results="asis"}
hivmy2 <- NormalizeData(hivmy2, verbose = FALSE)
hivmy2 <- ScaleData(hivmy2, verbose = FALSE)
hivmy2 <- FindVariableFeatures(hivmy2, verbose = FALSE)
hivmy2 <- RunPCA(hivmy2, features = VariableFeatures(hivmy2), verbose = FALSE)
ElbowPlot(hivmy2)
hivmy2 <- RunTSNE(hivmy2, dims= 1:10, verbose = FALSE)
Idents(hivmy2) <- "ht_cluster"
de_genes <- FindAllMarkers(hivmy2, verbose = FALSE)
top10 <- de_genes %>% 
  group_by(cluster) %>%
  top_n(n = 5, wt = avg_log2FC) 
top10 %>% 
  kable()
```

---
</details>


#### TF from Gosselin et al.

<details>
<summary>Microglia TF gene expression</summary>


```{r}
Idents(hiv4) <- "cell_type"
# Find Markers that distinguish My2 from the other Myeloid clusters
my2_markers <- FindMarkers(hiv4, ident.1  = "Myeloid2",
            ident.2  = c("Myeloid1", "Myeloid3", "Myeloid4", "Myeloid5"))
my2_markers <- my2_markers %>% rownames_to_column(var = "gene")
```

```{r, fig.width=15, fig.height=40}
microglia_genes <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/Microglia_gene_signatur_found_in_HIV1_HIV2_samples")
m1 <- my2_markers %>% filter(gene %in% microglia_genes [["gene_name"]] & avg_log2FC >0)
# remove the genes that were already used by the Farhadian paper
m1 <- m1[["gene"]][m1[["gene"]] %in% paper == FALSE]

VlnPlot(hivmy2, features = m1, group.by = "ht_cluster", ncol = 5)

```

---
</details>

#### Which of the 11 TFs are mentioned by Gosselin et al?

<details>
<summary>TFs also mentioned by Gosselin et al.</summary>

```{r, fig.width=15, fig.height=12}
tfs_microglia <- (read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/TFs_microglia_Glass_paper.txt"))[["V1"]]

gosselin <- tfs_microglia[tfs_microglia %in% overlap_atac$name]

VlnPlot(hivmy2, features = gosselin, group.by = "ht_cluster", ncol = 4)
```

---
</details>


# References




