---
title: "Determine number of k for Hierarchical clustering of Myeloid2 cluster in one patient"
author: "kmikulik"
date: "7 12 2021"
output: html_document
---


```{r setup}
knitr::opts_chunk$set(echo = TRUE, collapse = TRUE, message = FALSE,
                      cache = TRUE, autodep = TRUE)
set.seed(42)
```


```{r}
library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggplot2)
library(dplyr)
library(ggrepel)
#library(harmony)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(knitr)
library(ggVennDiagram)
library(corrplot)

# "Multitasking/Multiprocessing"
library(future)
#plan("multisession", workers = 24)
```

# GRN inference with Scenic

Using Scenic I tried to infer Gene regulatory networks from the gene expression
data. The output is a matrix of TF (Transcription Factor) activity per cell. In this case the activity
matrix contains activity of 125 different transcription factors.


First, we wanted to see if we can identify patterns in the TF activity in the
Myeloid2 cells. We compared the TF activity to TFs shown to have differential 
binding scores between three infection states (uninfected, latent infection, 
active infection) in primary microglia (bulk ATACseq). For the comparison
with the TF activity in Myeloid2 cells we focused on the TF with differential 
binding scores between **uninfected** and **latently infected** states.

Can we see patterns of increased/decreased TF activity in the Myeloid2 cells that
might correspond to the binding scores between infection states in primary
microglia?




* read in Seurat object

```{r}
hiv7 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds")
hiv4 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/non_integrated_HIV1_HIV2_4samples_seurat_object.rds")
```

* read in AuCell matrix (output from Scenic)

```{r}
auc_mtx <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/aucell_matrix_10k_hvg.tsv", sep = "\t")
```



* Combine AuCell Matrix and Seurat object

```{r}
# add auc matrix to metadata of hiv4 seurat object
hiv7_scenic <- AddMetaData(hiv7, 
                    metadata = auc_mtx, 
                    col.name = colnames(auc_mtx))
```



# bulk ATACseq footprinting results

Using bulk ATACseq data from primary microglia at three different conditions,
active infection, latent infection, uninfected, differential binding scores were
obtained using TOBIAS. Here, I will use the TFs which were previously shown to 
have differential binding scores in these primary microglia. Since I expect the 
CSF Myeloid cells to be latently infected, because they have been obtained from 
patients receiving ART, I will use the TFs with differential binding scores 
between latent infection and uninfected. We would like to know if the TF activity
inferred from RNAseq data of Myeloid2 cells correlates with these differential 
binding scores. In other words, we want to know if the patterns of differential 
TF activity are similar as the patterns of differential TF binding scores. 
For example, if a TF was shown to be more bound in latently infected primary
microglia compared to uninfected primary microglia, we would expect the TF to be
more active in a group of Myeloid2 cells which are latently infected compared
to a group of Myeloid2 cells which are uninfected. 

When you overlap the TF from the TF activity matrix obtained by running SCNEIC 
on the 10.000 most highly variable genes with the differntially bound TFs, 24 of 
the TFs overlap.

Using less than 10.000 most highly variable genes as an input matrix for SCENIC 
results in a lower number of TFs which overlap. Increasing the number of input 
features for SCENIC increases the number of TFs included in the TF activity 
matrix

```{r}
# read in dataframe containing differntial binding scores between conditions
#
# red = latently infected
# gfp = active infected
# uninfected
tfs_ana <- read.table("/media/ag-cherrmann/projects/06_HIV_Microglia/data/atacseq/data-2020-11-06/tobias/TOBIAS_snakemake/footprint_mglia2_GlassTF_17-03/TFBS/bindetect_results.txt", sep = "\t", header = TRUE)

# filter TFs with significant p values between latent infection and uninfected
# I decided to use a threshold of p-value < 1e-50, because due to the large number 
# of data, all p-values will be very high
top_tfs <- tfs_ana %>% filter(uninf_red_pvalue < 1e-50)

# keep only the TFs in the dataframe which overlap between the two datasets.
overlap_atac <- top_tfs %>% filter(name %in% colnames(auc_mtx))

# extract names of TFs more bound in either condition
up_latent <- overlap_atac$name[overlap_atac$uninf_red_change < 0]
up_uninf <- overlap_atac$name[overlap_atac$uninf_red_change > 0]

```

## Volcano Plot ATAC-seq data

Interestingly, only four TFs found in the TF activity matrix of Myeloid2 cells 
are more bound in latent infection than in uninfected primary microglia.

```{r, fig.height=10}

top_tfs %>% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = ifelse(uninf_red_change < 0, "up_latent", "up_uninf")) %>%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  mutate(label = ifelse(name %in% overlap_atac$name, name, NA)) %>% 
  ggplot(aes(x = uninf_red_change,
             y = -log10(uninf_red_pvalue),
             col = condition, 
             label = label)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-50), col = "red") +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = "TFs with differential binding scores between latent infection and uninfected primary microglia")

```


# Compare TF activities between clusters and look for patterns

The idea was to cluster the Myeloid2 cells based on their TF activity using
Hierarchical clustering. To do this I perofrmed hierarchical clustering on 
Myeloid2 cells from patient1 (HIV1_CSF) with k=3/4/5. The idea for using only 
patient 1 was that the other two patients might be used for validation of the 
clustering later on. If the clusters are biological, I might be able to transfer
the clusters onto the Myeloid2 cells from patient2 (HIV2_CSF) and patient3 (HIV3_CSF).
If the clusters also separate the Myeloid2 cells of the other two patients nicely, 
this would validate that there is a biological meaning behind the clusters.

However, as we will see the label transfer of the cluster identities did not 
work, probably due to the small sample size. There are only 146 in total in the 
Myeloid2 cluster of infected patients.

The alternative strategy was to perform independent hierarchical clustering 
across all patients and compare the TF activity between clusters. 
Can we find TF activities which differentiate clusters across patients and find
correlations between clusters across patients?

1. Determine k for k-means hierarchical clustering
2. Transfer the cluster identities to patient2 and patient3
3. Hierarchical Clustering for all 3 patients separately




```{r}


# TF annotations
la <- rowAnnotation(type_TF = 
                      case_when(colnames(auc_mtx) %in% up_latent ~ "up_latent",
                                        colnames(auc_mtx) %in% up_uninf  ~ "up_uninf",
                                        TRUE ~"unknown"),
                     name = "type_TF",
                     col = list(type_TF = c("up_latent" =  "red", 
                                            "up_uninf" =  "pink",
                                            "unknown" = "forestgreen")))
  
```



## Split Heatmap into 3 separate heatmaps according to the different patients

```{r, fig.width=13, fig.height=14}
# 3 heatmaps with all Scenic TFs
ht_list <- map(seq.int(1:3), function(n) {
  patients <- c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF") # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df),]
  # heatmap
  ht <- Heatmap(t(mtx), 
               column_title = paste0("patient: ", patients[n]), 

               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
               show_row_names = TRUE
               ) 
               #top_annotation = column_annotation) 
  list(name = patients[n], heatmap = ht)
})


ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap + la


```

## Heatmap containing only TFs found by ATAC-seq data analysis

```{r, fig.width=13, fig.height=8}
# 3 heatmaps, but only the TFs which overlap with Anas analysis
ht_list <- map(seq.int(1:3), function(n) {
  patients <- c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF") # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  # select TFs which overlap with Anas analysis
  mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), overlap_atac$name]
  # heatmap
  ht <- Heatmap(t(mtx), 
               column_title = paste0("patient: ", patients[n]), 

               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
               right_annotation = rowAnnotation(type_TF = ifelse(colnames(auc_mtx[,overlap_atac$name]) %in% up_latent, "up_latent", "up_uninf"),
                                                name = "type_TF",
                                                col = list(type_TF = c("up_latent" =  "red",
                                                                       "up_uninf" =  "pink"))))
  list(name = patients[n], heatmap = ht)
})

  
ht_list = ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap
draw(ht_list, ht_gap = unit(1, "cm"))
```





# 1. Determine k for k-means hierarchical clustering

* I will try k = 3/4/5 for clustering and see which yields the best results

```{r}
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                    orig.ident == "HIV1_CSF")
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]

# Decide on a k for clustering
wss = sapply(2:15, function(k) {
  kmeans(x = t(mtx), centers = k)$tot.withinss
})

plot(2:15, wss, type = "b", 
     xlab = "Number of clusters k", 
     ylab = "Total within-clusters sum of square")
```

## K = 3 Hierarchical Clustering

### Heatmap for patient 1

```{r, fig.width=15, fig.height=17}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_pat1 <-draw(Heatmap(t(mtx), 
             column_title = "Myeloid2 cells from patient 1, k = 3", 

             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             ))
             #top_annotation = column_annotation) 
```


### Extract cells from each cluster

```{r}
# if you draw the heatmap, the results of th eclustering will not change anymore
#png("/media/ag-cherrmann/kmikulik/HIV_microglia/analysis/Myeloid2_cluster/Myeloid2_3clusters.png", width = 15, height = 17, res = 1200)
#ht <- draw(ht_pat1)
# get the columns names of cluster one
c1 <- colnames(t(mtx[column_order(ht_pat1)[[1]],]))
#write.table(c1, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv", sep = "\t")

# get the column names of cluster two
c2 <- colnames(t(mtx[column_order(ht_pat1)[[2]],]))
#write.table(c2, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv", sep = "\t")

# get the column names of cluster 3
c3 <- colnames(t(mtx[column_order(ht_pat1)[[3]],]))
#write.table(c3, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv", sep = "\t")
```


#### number of cells in each cluster 

```{r, results = "asis"}
# summary staticstics
hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) %>% 
  #group_by(ht_cluster) %>% 
  select(ht_cluster, CTCF) %>% 
  group_by(ht_cluster) %>% 
  summarise(
    count = n(),
    mean = mean(CTCF),
    sd = sd(CTCF)) %>% kable()

```


## TF activity per cluster

Plot TF activity for all TFs which showed differential binding scores in primary
microglia. For some TFs it appears that there are differences in activity
between the clusters. The next step will be to quantitate these differences. 

From the violin plots below it becomes evident, that the acitivity is not normally 
distributed. Therefore, we will use Kruskal-Wallis test (non-parametric) instead 
of ANOVA (parametric).

```{r, fig.width = 15, fig.height=20}
# make boxplots  and violin plots for all TFs
p <- list()
for (n in 1:24){
  tf <- overlap_atac$name[n]
  p[[n]] <- hiv7_scenic@meta.data %>% 
    rownames_to_column("cell") %>% 
    filter(orig.ident == "HIV1_CSF",
           cell_type == "Myeloid2",
           cell %in% c(c1,c2,c3)) %>% 
    mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) %>% 
    #group_by(ht_cluster) %>% 
    select(ht_cluster, tf) %>% 
    ggplot(aes_string(x = "ht_cluster", y = tf)) +
    geom_boxplot(aes(fill = ht_cluster)) + 
    geom_violin(aes(alpha = 0.1)) +
    stat_compare_means()
}

do.call(grid.arrange,c(p, ncol = 4, top= "Transcription factor activity per cluster"))

```


## Are there significant differences in TF activity between the three clusters?

* **ANOVA** test is parametric and assumes normally distributed data
* **Kruskal-Wallis-Test** compares the mean of one or more groups, but is 
non-parametric. This seems more appropriate in this case.
* In both cases the Null-Hypothesis is that there is no difference
between the groups.

Since I am repeating the test 125 times, I adjust the p-values with the
FDR method. This methods is more powerful than more conservative methdos, like
bonferroni, Holm or Hochberg. 


### Kruskal-Wallis Test and p-value adjustments for all 125 TFs

```{r}
# non-parametric alternative to ANOVA -> Kruskal_wallis test
k3_df <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) 
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))["p.value"]
  #list(tf = tf, p_value = p_values)
})

# plot distribution of p_values
#hist(as.numeric(unname(unlist(p_values))), 
 #    main = "distribution of p_values",
  #   xlab = "p_values",
   #  breaks = 20)

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")

# how many transcription factors are significantly different between the clusters?
sign3 <- adj_p_values[adj_p_values<.05]
length(adj_p_values[adj_p_values<0.05])

# get TFs that are significantly different
hiv1_diff <- colnames(auc_mtx)[adj_p_values < .05]

#write.table(diff3, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1")
```

### Kruskal-Wallis test and p-value adjustment  for 24 TFs also identified by ATAC-seq


```{r}
# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(k3_df[tf]) ~ ht_cluster, k3_df))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])

# get TFs that are differentially expressed
hiv1_atac <- overlap_atac$name[adj_p_values < .05]
#write.table(diff3_atac, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1")
```

* 54 of the 125 TFs are differentially active between the three clusters
* 16 of the 24 TFs with differential binding scores in primary microglia are
differentially active between the three clusters.


### Posthoc tests for k = 3

Now that we know that there are significant differences between clusters, we 
want to identify which cluster/s are different from which other cluster/s.

** Pairwise Wilcox Test** is a non-parameteric pariwise test. Again, because 
we repeat the test we will correct for multiple hypothesis testin using the 
FDR method.

**For the 16 TFs with significantly different activity between the 3 clusters:** 

```{r, warning = FALSE}
# We perform a posthoc test
#wilcox_list = list()
wilcox_list <- map(seq.int(1:length(hiv1_atac)), function(i){
  tf <- hiv1_atac[i]
  #print(tf)
  wilcox <- pairwise.wilcox.test(pull(k3_df[tf]), pull(k3_df["ht_cluster"]), 
                                 p.adjust.method = "fdr")
  list(TF = tf, wilcox = wilcox)
})
  
#length(wilcox_list)
#wilcox_list
```


## K = 4 Hierarchical clustering

### Heatmap for patient 1

```{r, fig.width=15, fig.height=17}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht <- draw(Heatmap(t(mtx), 
             column_title = "Myeloid2 cells from patient 2, k = 4", 

             column_km = 4,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             )) 
             #top_annotation = column_annotation) 


```

### Extract cells from each cluster of k = 4

```{r}
# get the columns names of cluster one
cl1 <- colnames(t(mtx[column_order(ht)[[1]],]))

# get the column names of cluster two
cl2 <- colnames(t(mtx[column_order(ht)[[2]],]))


# get the column names of cluster 3
cl3 <- colnames(t(mtx[column_order(ht)[[3]],]))

cl4 <- colnames(t(mtx[column_order(ht)[[4]],]))
```

### Kruskal-Wallis Test for k = 4

* For all 125 TFs, we find 51 that are significantly different between the clusters.

```{r}

k4_df <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(cl1,cl2,cl3,cl4)) %>% 
  mutate(ht_cluster = ifelse(cell %in% cl1, "cl1", 
                             ifelse(cell %in% cl2, "cl2", 
                                    ifelse(cell %in% cl3, "cl3", "cl4"))))

# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(k4_df[tf]) ~ ht_cluster, k4_df))["p.value"]
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")

# how many transcription factors are significantly different between the clusters?
sign <- adj_p_values[adj_p_values < 0.05]
length(adj_p_values[adj_p_values<0.05])


# get the TFs that are differentially expressed
diff4 <- colnames(auc_mtx)[adj_p_values < 0.05]

#write.table(diff4, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=4")

```

* For the 24 TFs from ATAC-seq footprintin in primary microglia there are 14 
differentially expressed between clusters.

```{r}
# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(k4_df[tf]) ~ ht_cluster, k4_df))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])

# get the TFs that are differentially expressed
diff4_atac <- overlap_atac$name[adj_p_values < .05]

#write.table(diff4_atac, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=4")

```

### Posthoc tests for k = 4

**Pairwise Wilcox Test** 

```{r, warning = FALSE}
# We perform a posthoc test
#wilcox_list = list()
wilcox_list <- map(seq.int(1:length(diff4_atac)), function(i){
  tf <- diff4_atac[i]
  print(tf)
  wilcox <- pairwise.wilcox.test(pull(k4_df[tf]), pull(k4_df["ht_cluster"]), 
                                 p.adjust.method = "fdr")
  list(TF = tf, wilcox = wilcox)
})
  
#length(wilcox_list)
#wilcox_list
```


## K = 5 Hierarchical Clustering

### Heatmap for patient 1

```{r, fig.width=15, fig.height=17}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht <- draw(Heatmap(t(mtx), 
             column_title = "Myeloid2 cells from patient 1, k = 5", 

             column_km = 5,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             )) 

```


### Extract cells from each cluster for k= 5

```{r}
# get the columns names of cluster one
clu1 <- colnames(t(mtx[column_order(ht)[[1]],]))

# get the column names of cluster two
clu2 <- colnames(t(mtx[column_order(ht)[[2]],]))


# get the column names of cluster 3
clu3 <- colnames(t(mtx[column_order(ht)[[3]],]))

clu4 <- colnames(t(mtx[column_order(ht)[[4]],]))

clu5 <- colnames(t(mtx[column_order(ht)[[5]], ]))

```

### Kruskal-Wallis Test for k = 5

* For all 125 TFs, we find 51 that are significantly different between the clusters

```{r}
# all TFs
k5_df <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV1_CSF",
         cell_type == "Myeloid2",
         cell %in% c(clu1,clu2,clu3,clu4, clu5)) %>% 
  mutate(ht_cluster = ifelse(cell %in% clu1, "clu1", 
                             ifelse(cell %in% clu2, "clu2", 
                                    ifelse(cell %in% clu3, "clu3",
                                           ifelse(cell %in% clu4, "clu4", "clu5")))))
n = 1
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(k5_df[tf]) ~ ht_cluster, k5_df))["p.value"]
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")

# how many transcription factors are significantly different between the clusters?
sign <- adj_p_values[adj_p_values < 0.05]
length(adj_p_values[adj_p_values<0.05])


# get the TFs that are differentially expressed
diff5 <- colnames(auc_mtx)[adj_p_values < 0.05]

#write.table(diff5, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=5")
```

* For the 24 TFs from ATAC-seq footprinting in primary microglia there are 14 
differentially expressed between clusters

```{r}
# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(k5_df[tf]) ~ ht_cluster, k5_df))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])

# get the TFs that are differentially expressed
diff5_atac <- overlap_atac$name[adj_p_values < .05]

#write.table(diff5_atac, "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=5")
```

### Posthoc tests for k = 4

**Pairwise Wilcox Test** 

```{r, warning = FALSE}
# We perform a posthoc test
#wilcox_list = list()
wilcox_list <- map(seq.int(1:length(diff5_atac)), function(i){
  tf <- diff5_atac[i]
  print(tf)
  wilcox <- pairwise.wilcox.test(pull(k5_df[tf]), pull(k5_df["ht_cluster"]), 
                                 p.adjust.method = "fdr")
  list(TF = tf, wilcox = wilcox)
})
  
#length(wilcox_list)
#wilcox_list
```

## Which Clusters are sifnificantly different for differen k? 

### Which Clusters are significantly different for k = 3?

```{r, fig.height=20, fig.width=25}
library(rstatix)
library(ggpubr)

test <- k3_df
#tf.oi <- c("CTCF", "REST", "YY1")
#names(tf.oi) <- tf.oi
test.list <- map(hiv1_atac, function(tf) {
  ggplot() +
  geom_boxplot(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf), 
                   fill = test %>% pull("ht_cluster"))) +
  stat_pvalue_manual(test %>% 
    pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>%
      add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 4))
```



## Which Clusters are significantly different for k = 4?

```{r, fig.width=25, fig.height=15}
library(rstatix)
library(ggpubr)

test <- k4_df

test.list <- map(diff4_atac, function(tf) {
  ggplot() +
  geom_boxplot(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf), 
                   fill = test %>% pull("ht_cluster"))) +
  stat_pvalue_manual(test %>% 
    pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>%
      add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 3))


```


## Which clusters are significantly different for k = 5
```{r, fig.width=25, fig.height=15}
library(rstatix)
library(ggpubr)

test <- k5_df

test.list <- map(diff5_atac, function(tf) {
  ggplot() +
  geom_boxplot(aes(x = test %>% pull("ht_cluster"), y = test %>% pull(tf), 
                   fill = test %>% pull("ht_cluster"))) +
  stat_pvalue_manual(test %>% 
    pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>%
      add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(test.list, ncol=5, nrow = 3))
```

## Conclusion on how many k's to use for hierarchical clustering

Across all k (k=3/4/5) that I tested, a similar group of TFs showed significantly
different activity between the clusters. Therefore, I decided to proceed with 
k = 3 for the following analysis. This overlap of significantly different TFs
is visualized in the Venn Diagrams below. 

```{r}
diff5_atac <- as.vector(read.table( "/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=5"))[["x"]]
diff5 <- as.vector(read.table( "/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=5"))[["x"]]


diff4_atac <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1_k=4"))[["x"]]

diff4 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1_k=4"))[["x"]]
# list of TFs from ATAC-seq differntially active between the 3 heatmap 
#clusters from patient 1
diff3_atac <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_TFs_differentially_active_in_HIV1"))[["x"]]

# list of TFs differntially active between the 3 heatmap clusters from patient 1
diff3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_TFs_differentially_active_in_HIV1"))[["x"]]
```


```{r, fig.width = 12}
p1 <- ggVennDiagram(list(diff3_atac, diff4_atac, diff5_atac),
              label_alpha = 0,
              category.names= c("k = 3", "k = 4", "k = 5"),
              #set.size = 20,
              label = "count",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "TFs (ATAC-seq) significantly different between clusters for different k's")
  #scale_fill_gradient (low = "lightblue", high = "darkblue")

p2 <- ggVennDiagram(list(diff3, diff4, diff5),
              label_alpha = 0,
              category.names= c("k = 3", "k = 4", "k = 5"),
              label = "count",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "all 125 TFs significantly different between clustersfor different k's")

ggarrange(p1, p2, ncol = 2, nrow = 1)
```



# 2. Transfer the 3 cluster identitites to patient2 and patient3

## Cluster annotation using an integrated reference

* We want to annotate the cells from patient 2 and patient 3 and assign them one 
of the three heatmap cluster identities
* Here we do not integrate the datasets beforehand, we do not modify the query
expression data.

```{
Idents(hiv7) <- "cell_type"
hiv <- subset(hiv7, idents = "Myeloid2")

# the cell type is replaced by the heatmap cluster
hiv1_my2 <- subset(hiv7, cells = c(c1,c2,c3))

clusters <- (
  hiv1_my2@meta.data %>% 
  rownames_to_column("cell") %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) %>% 
  column_to_rownames("cell"))["ht_cluster"]

hiv <- AddMetaData(hiv, metadata = clusters, col.name = "cell_type")

# the seurat_cluster number is replaced by the number of the heatmap cluster
seurat_clusters <- (
  hiv1_my2@meta.data %>% 
  rownames_to_column("cell") %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, 1, ifelse(cell %in% c2, 2, 3))) %>% 
  column_to_rownames("cell"))["ht_cluster"]

hiv <- AddMetaData(hiv, metadata = seurat_clusters, col.name = "seurat_clusters")


# split the object according to the patients
hiv_list <- SplitObject(hiv, split.by = "orig.ident")[3:5]
# normalize each object
for (i in names(hiv_list)) {
    hiv_list[[i]] <- SCTransform(hiv_list[[i]], verbose = FALSE)
}


# patient 2 and patient 3 are our query datasets
query <-merge(hiv_list[[1]], hiv_list[[3]])
reference <- hiv_list[[2]]
reference <- RunPCA(reference, 
                    # by default a partial PCA is run, but we compute mostgular values -> run standard SVD instead
                    approx = FALSE, 
                    verbose = FALSE)

hiv_anchors <- FindTransferAnchors(reference = reference,
                                   query = query, 
                                   dims = 1:30, verbose = FALSE,
                                   k.filter = NA, 
                                   reference.reduction = "pca",
                                   normalization.method = "SCT",
                                   reduction = "pcaproject")

predictions <- TransferData(anchorset = hiv_anchors, 
                            refdata = reference$cell_type,
                            k.weight = 34, dims = 1:30, verbose = FALSE)

query <- AddMetaData(query, metadata = predictions)


hiv2 <- subset(query, orig.ident == "HIV2_CSF")
hiv3 <- subset(query, orig.ident == "HIV3_CSF")
#saveRDS(hiv2,"/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv2_CSF_myleoid2_heatmap_cluster_label_transfer.rds")
#saveRDS(hiv3, "/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv3_CSF_myleoid2_heatmap_cluster_label_transfer.rds")
```

Now that we have transferred the cluster identitities to patient2 and patient3
and saved the results in two separate Seurat objects, we can add the TF activity 
score to the metadata of each patient.

```{r}
hiv2 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv2_CSF_myleoid2_heatmap_cluster_label_transfer.rds")
hiv3 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/hiv3_CSF_myleoid2_heatmap_cluster_label_transfer.rds")

hiv2_scenic <- AddMetaData(hiv2, metadata = auc_mtx,
                           col.name = colnames(auc_mtx))
hiv3_scenic <- AddMetaData(hiv3, metadata = auc_mtx,
                           col.name = colnames(auc_mtx))
```


## Predicted clusters from Label Transfer for patient 2 and patient 3

The clusters you get when transferring the cluster identities from patient 1 to
patient2 do not seem to be very good at separating cells according to TF activity.

When you transfer the cluster identities from patient1 to patient3 all cells but
one are assigned to cluster1. This shows that the label transfer did not work 
very well. This is probably due to the low number of Myeloid2 cells in each sample.

```{r, fig.width=15, fig.height=17}
cluster_anno <- as.data.frame(hiv2$predicted.id)

#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV2_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_hiv2_label <- Heatmap(t(mtx), 
             column_title = "Predicted Clusters from Label Transfer for HIV2_CSF",             
             column_title_gp = gpar(fontsize = 15),
             column_split = paste0("cluster", hiv2$predicted.id),#cluster_anno,
             #column_km = 4,
             #column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la,
             top_annotation = HeatmapAnnotation(df = cluster_anno, name = "predicted cluster")
             ) 
             #top_annotation = column_annotation) 
```


```{r, fig.width=15, fig.height=17}
cluster_anno <- as.data.frame(hiv3$predicted.id)

#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV3_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht_hiv3_label <- Heatmap(t(mtx), 
             column_title = "Predicted clusters from Label Transfer for HIV3_CSF", 
             column_title_gp = gpar(fontsize = 15),
             column_split = paste0("cluster", hiv3$predicted.id),#cluster_anno,
             #column_km = 4,
             #column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la,
             top_annotation = HeatmapAnnotation(df = cluster_anno, name = "predicted cluster")
             ) 
             #top_annotation = column_annotation) 

```

```{r, fig.width=15, fig.height=20}
ht_list <- ht_hiv2_label + ht_hiv3_label
draw(ht_list, ht_gap= unit(2, "cm"))
```

# 3. Hierarchical clustering for all 3 patients separately

Since the label transfer of the 3 cluster identities from patient1 did not seem
to work very well, we next tried to perform hierarchical clustering using k = 3
for all 3 patients separately. This way we might be able to see if the TF
activities in the 3 clusters show similar patterns across the different patients.

## K-means clustering with k = 3 for patient 2

When you perform hierarchical clustering of the cells in patient2 the clusters
look better than the clusters we saw after cluster identity transfer.

```{r, fig.width=15, fig.height=20}
#map(seq.int(1:5), function(k){
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV2_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht2 <- Heatmap(t(mtx), 
             column_title = "Patient 2, HIV2_CSF", 
             column_title_gp = gpar(fontsize = 15),
             #column_split = paste0("cluster", hiv2$predicted.id),#cluster_anno,
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la,
             #top_annotation = HeatmapAnnotation(df = cluster_anno, name = "predicted cluster")
             ) 
             #top_annotation = column_annotation) 
ht_pat2 <- draw(ht2)

```



#### Kruskal_Wallis Test Patient2 

```{r}
# extract the cells from the 3 clusters 
# get the cell names of cluster one
c1 <- colnames(t(mtx[column_order(ht_pat2)[[1]],]))
# get the cell names of cluster two
c2 <- colnames(t(mtx[column_order(ht_pat2)[[2]],]))
# get the cell names of cluster 3
c3 <- colnames(t(mtx[column_order(ht_pat2)[[3]],]))


# add cluster identities to Seurat object metadata and return metadata as dataframe
pat2 <- hiv7_scenic@meta.data %>% rownames_to_column("cell")
pat2 <- pat2 %>% 
  filter(orig.ident == "HIV2_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) 


# check all 125 TFs
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(pat2[tf]) ~ ht_cluster, pat2))["p.value"]
  #list(tf = tf, p_value = p_values)
})


# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")
length(adj_p_values[adj_p_values<0.05])
## [1] 66

# get the TFs taht are differntially active
diff_hiv2 <- colnames(auc_mtx)[adj_p_values < 0.05]


# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(pat2[tf]) ~ ht_cluster, pat2))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])
## [1] 18

# get the TFs that are differentially expressed
hiv2_atac <- overlap_atac$name[adj_p_values < .05]
```


## K-means clustering with k = 3 for patient 3


```{r, fig.width=15, fig.height=20}
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV3_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht3 <- Heatmap(t(mtx), 
             column_title = "Patient 3, HIV3_CSF", 
             column_title_gp = gpar(fontsize = 15),
             #column_split = paste0("cluster", hiv2$predicted.id),#cluster_anno,
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la,
             #top_annotation = HeatmapAnnotation(df = cluster_anno, name = "predicted cluster")
             ) 
             #top_annotation = column_annotation) ich
ht_pat3 <- draw(ht3)
```


#### Kruskal-Wallis test Patient 3

```{r}
# extract the cells from the 3 clusters 
# get the cell names of cluster one
c1 <- colnames(t(mtx[column_order(ht_pat3)[[1]],]))
# get the cell names of cluster two
c2 <- colnames(t(mtx[column_order(ht_pat3)[[2]],]))
# get the cell names of cluster 3
c3 <- colnames(t(mtx[column_order(ht_pat3)[[3]],]))


# add cluster identities to Seurat object metadata and return metadata as dataframe
pat3 <- hiv7_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
  filter(orig.ident == "HIV3_CSF",
         cell_type == "Myeloid2",
         cell %in% c(c1,c2,c3)) %>% 
  mutate(ht_cluster = ifelse(cell %in% c1, "c1", ifelse(cell %in% c2, "c2", "c3"))) 


# check all 125 TFs
# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(pat3[tf]) ~ ht_cluster, pat3))["p.value"]
  #list(tf = tf, p_value = p_values)
})


# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")
length(adj_p_values[adj_p_values<0.05])
## [1] 66

# get the TFs taht are differntially active
diff_hiv3 <- colnames(auc_mtx)[adj_p_values < 0.05]


# check 24 TFs with Kruskal Wallis test
p_values <- map(seq.int(1:24), function(n){
  tf <- overlap_atac$name[n]
  p_values <- unlist(kruskal.test(pull(pat3[tf]) ~ ht_cluster, pat3))["p.value"]
})
#sign_p_values <- p_values[p_values < 0.01]

# adjust for multiple hypothesis testing
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")


# how many transcription factors are significantly different between the clusters?
length(adj_p_values[adj_p_values<0.05])
## [1] 18

# get the TFs that are differentially expressed
hiv3_atac <- overlap_atac$name[adj_p_values < .05]
```


## Which TFs show significantly different activity between the clusters across patients?

Since the independent hierarchical clustering for all 3 patients yields 
clusters which look more reasonable, we will compare the 3 clusters we obtain 
across the 3 patients to see if there is a pattern between them. 

We already saw that there are a few TFs which show significantly different
activity between the clusters in each patient.

Next, we want to see if the same group of TFs characterize the three 
clusters in all three patients or if they are different. 
We will test which TFs are significantly different between clusters and then
use a pairswise Wilcox test to determine which clusters are different.

### K-means clustering with k = 3 for all three patients

```{r, fig.width=15, fig.height=17}
# patient 1
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == "HIV1_CSF")
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df), ]
# heatmap
set.seed(123)
ht1 <- Heatmap(t(mtx), 
             column_title = "Patient 1, HIV1_CSF", 
             column_title_gp = gpar(fontsize = 15),
             column_km = 3,
             column_km_repeats = 100,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             right_annotation = la
             )
             #top_annotation = column_annotation) 
```

```{r, fig.width=15, fig.height=20}
ht_list <- ht1 + ht2 + ht3
draw(ht_list, ht_gap= unit(2, "cm"), 
     column_title = "Independent k-means clustering (k = 3) for all patients",
     column_title_gp = gpar(fontsize = 20) )
```



### Are TFs differentiating the 3 clusters similar across all 3 patients?

We can see that 11 of the TFs found to have differential binding scores between
infection conditions in primary microglia show different activity between the 
3 clusters across all patients.

```{r}
p1 <- ggVennDiagram(list(hiv1_diff, diff_hiv2, diff_hiv3),
              label_alpha = 0,
              category.names= c("HIV1", "HIV2", "HIV3"),
              #set.size = 20,
              label = "count",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "TF (ATAC-seq) significantly different between 3 clusters")
```

```{r}
p2 <- ggVennDiagram(list(hiv1_atac, hiv2_atac, hiv3_atac),
              label_alpha = 0,
              category.names= c("HIV1", "HIV2", "HIV3"),
              #set.size = 20,
              label = "count",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "TF (ATAC-seq) significantly different between 3 clusters")
```

```{r}
ggarrange(p1, p2, ncol = 2, nrow = 1)
```


## 11 TFs show different activity in one of the 3 clusters 

```{r}
intersect <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/intersect_TFs_different_across_patients_3_clusters")[["x"]]
```


##### Patient 1

```{r, fig.width=25, fig.height=10}
library(rstatix)
library(ggpubr)

df <- k3_df


boxplot_list <- map(intersect, function(tf) {
  ggplot() +
    geom_boxplot(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf),
                     fill = df %>% pull("ht_cluster"))) +
    stat_pvalue_manual(df %>% 
                        pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>% 
                         add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))

#annotate_figure(figure, top = text_grob("Patient 1 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)", 
 #              color = "black", face = "bold", size = 25))
```

```{r}
ctcf1 <- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_boxplot(aes(fill = ht_cluster)) +
  stat_pvalue_manual(df %>% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %>% 
                       add_xy_position())
ctcf1 <- ctcf1 + labs(title = "CTCF activity in patient 1")
```

##### Patient 2

```{r, fig.width=25, fig.height=10}
library(rstatix)
library(ggpubr)

df <- pat2

boxplot_list <- map(intersect, function(tf) {
  ggplot() +
    geom_boxplot(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf),
                     fill = df %>% pull("ht_cluster"))) +
    stat_pvalue_manual(df %>% 
                        pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>% 
                         add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))

#annotate_figure(figure, top = text_grob("Patient 2 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)", 
 #              color = "black", face = "bold", size = 25))
```

```{r}
ctcf2 <- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_boxplot(aes(fill = ht_cluster)) +
  stat_pvalue_manual(df %>% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %>% 
                       add_xy_position())
ctcf2 <- ctcf2 + labs(title = "CTCF activity in patient 2")
```

##### Patient 3

```{r, fig.width=25, fig.height=10}
library(rstatix)
library(ggpubr)

df <- pat3

boxplot_list <- map(intersect, function(tf) {
  ggplot() +
    geom_boxplot(aes(x = df %>% pull("ht_cluster"), y = df %>% pull(tf),
                     fill = df %>% pull("ht_cluster"))) +
    stat_pvalue_manual(df %>% 
                        pairwise_wilcox_test(as.formula(paste0(tf, " ~ ht_cluster"))) %>% 
                         add_xy_position()) +
    xlab("heatmap cluster") +
    ylab(paste0(tf)) +
    guides(fill = guide_legend(title = "heatmap cluster"))
})

do.call(gridExtra::grid.arrange, c(boxplot_list, ncol = 4, nrow = 3))


#annotate_figure(figure, top = text_grob("Patient 3 - CSF     k = 3 for k-means clustering (TFs from ATAC-seq)", 
 #              color = "black", face = "bold", size = 25))
```

### CTCF expression across patients

* CTCF has been implicated to regualte HIV latency (Jefferys S. R., PLOS Pathogens, 2021)

```{r}
ctcf3 <- ggplot(df, aes(x = ht_cluster, y = CTCF)) +
  geom_boxplot(aes(fill = ht_cluster)) +
  stat_pvalue_manual(df %>% 
                       pairwise_wilcox_test(CTCF ~ ht_cluster) %>% 
                       add_xy_position())
ctcf3 <- ctcf3 + labs(title = "CTCF activity in patient 3")
```

```{r, fig.width=15, fig.height=5}
ggarrange(ctcf1, ctcf2, ctcf3, ncol = 3, nrow = 1)
```


## Correlations between clusters across patients

Using a correlation matrix we can visualize how clusters are correlated across
patients:


```{r}
# read in 3 Clusters for each of the 3 HIV+ patients

# HIV1_CSF
hiv1_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv", sep = "\t"))[["x"]]
hiv1_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv", sep = "\t"))[["x"]]
hiv1_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv", sep = "\t"))[["x"]]

# HIV2_CSF
hiv2_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster1.tsv", sep = "\t"))[["x"]]
hiv2_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster2.tsv", sep = "\t"))[["x"]]
hiv2_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster3.tsv", sep = "\t"))[["x"]]

# HIV3_CSF
hiv3_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster1.tsv", sep = "\t"))[["x"]]
hiv3_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster2.tsv", sep = "\t"))[["x"]]
hiv3_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster3.tsv", sep = "\t"))[["x"]]

cluster_anno <- c(hiv1_c1, hiv1_c2, hiv1_c3, hiv2_c1, hiv2_c2, hiv2_c3, hiv3_c1, hiv3_c2, hiv3_c3)
```


```{r}
# add cluster information to TF activity
auc_df <- as.data.frame(auc_mtx)

auc_df <- auc_df %>% 
  rownames_to_column("cell") %>% 
  filter(cell %in% cluster_anno) %>% 
  mutate(ht_cluster = case_when(cell %in% hiv1_c1 ~ "hiv1_c1", 
         cell %in% hiv1_c2 ~ "hiv1_c2", 
         cell %in% hiv1_c3 ~ "hiv1_c3",
         cell %in% hiv2_c1 ~ "hiv2_c1",
         cell %in% hiv2_c2 ~ "hiv2_c2",
         cell %in% hiv2_c3 ~ "hiv2_c3",
         cell %in% hiv3_c1 ~ "hiv3_c1",
         cell %in% hiv3_c2 ~ "hiv3_c2",
         cell %in% hiv3_c3 ~ "hiv3_c3"))
```

### Correlation between clusters based on all 125 TFs 

For every cluster the correlation is calculated based on the TF activities of
all 125 TFs  identified by SCENIC.

Dataframe of average activity value for each TF in each patient and clusters.


```{r, results = "asis"}
# correlation matrix
df <- auc_df %>%
  group_by(ht_cluster) %>%
  summarize_if(is.numeric, mean) %>% 
  #column_to_rownames("ht_cluster") %>% 
  #add_rownames() %>% 
  gather(TF, value, -ht_cluster) %>% 
  spread(ht_cluster, value) %>% 
  column_to_rownames("TF")

df %>% 
  head %>% 
  kable ()
```

Correlation matrix: 

```{r, results = "asis"}
df %>% cor %>% kable()

cor_mtx <- df %>% cor()

# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor <- cor.mtest(cor_mtx, conf.level = 0.95)
```



**Conclusions:**

* Cluster 3 from patient 3 seems to be very different
* Cluster 2 from patient 3 is also only weakly correlated
* Cluster 1 of all patients seems correlated, however, the p_values for 
correlation of cluster1 of patient two with all other clusters are not significant
* Clusters 2 and 3 of patient 1 and 2 seem highly correlated


```{r}
# plot correlations
corrplot(cor_mtx,method = "color", order = "hclust", 
         addrect = 4, # add boxes 
         p.mat = testCor$p, 
         #sig.level = .05,
         insig = "p-value")
```

### Correlation between clusters based on all TFs with differential activity between clusters

```{r}
intersect_total <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/total_intersect_TFs_different_across_patients_3_clusters"))[["x"]]
```

```{r}
# correlation matrix
cor_mtx <- auc_df %>%
  select(ht_cluster, intersect_total) %>% 
  group_by(ht_cluster) %>%
  summarize_if(is.numeric, mean) %>% 
  #column_to_rownames("ht_cluster") %>% 
  #add_rownames() %>% 
  gather(TF, value, -ht_cluster) %>% 
  spread(ht_cluster, value) %>% 
  column_to_rownames("TF") %>% cor()

# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor <- cor.mtest(cor_mtx, conf.level = 0.95)
```


```{r}
corrplot(cor_mtx,method = "color", order = "hclust", 
         addrect = 4, # add boxes 
         p.mat = testCor$p, 
         #sig.level = .05,
         insig = "p-value")
```


# TF correlation across clusters

11 TFs which differentiate the clusters seem to correlate between the clusters.

For these 11 TFs whith different activity between the clusters the activity 
correlates between across all 3 patients. This means, that the three TFs
YY1, REST, MXI1 are positively correlated, but negatively correlated with the 
rest of the TFs.

The fact that these TF activities correlate across all patients indicates that 
there might be a biological meaning behind it. 

Dataframe used as input for the correlation:

```{r}
intersect_atac <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/atac_intersect_TFs_different_across_patients_3_clusters"))[["x"]]
```


```{r}
cor_mtx <- auc_df %>%
  select(ht_cluster, intersect_atac) %>% 
  group_by(ht_cluster) %>%
  summarize_if(is.numeric, mean) %>% 
  column_to_rownames("ht_cluster") %>% cor()
```


```{r, results = "asis"}
# correlation matrix
df <- auc_df %>%
  select(ht_cluster,intersect_atac) %>% 
  group_by(ht_cluster) %>%
  summarize_if(is.numeric, mean) %>% 
  column_to_rownames("ht_cluster") 

df %>% head %>% kable()

```

Correlation matrix for TFs:

```{r, results = "asis"}
cor_mtx <- df %>% cor()
cor_mtx %>% kable

# calculate confidence intervals and p_values
# function produces p-values and confidence intervals for each pair of input features
# returns a matrix each
# of p_values, upper confidence interval, lower confidence interval
testCor <- cor.mtest(cor_mtx, conf.level = 0.95)
```

```{r}
# plot correlations
# p.mat = matrix of p_values
p1 <- corrplot(cor_mtx,method = "color", order = "hclust", 
         addrect = 2, # add boxes 
         p.mat = testCor$p, 
         insig = "p-value")
         #insig = "p-value")
         #title = "Correlations between 9 clusters of 3 patients")

```


# Gene expression analysis

Using the information on the three clusters from above we will pool the two clusters
which are significantly different from the third cluster. Consequently, in the 
following analysis we will compare Cluster1 to Cluster2 (pool of two similar 
clusters). Gene expression might give a hint of the identity of these two Myeloid2
clusters. 


```{r}
# 3 Clusters for each of the 3 HIV+ patients
wss


# HIV1_CSF
hiv1_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster1.tsv", sep = "\t"))[["x"]]
hiv1_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster2.tsv", sep = "\t"))[["x"]]
hiv1_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv1_cluster3.tsv", sep = "\t"))[["x"]]

# HIV2_CSF
hiv2_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster1.tsv", sep = "\t"))[["x"]]
hiv2_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster2.tsv", sep = "\t"))[["x"]]
hiv2_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv2_cluster3.tsv", sep = "\t"))[["x"]]

# HIV3_CSF
hiv3_c1 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster1.tsv", sep = "\t"))[["x"]]
hiv3_c2 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster2.tsv", sep = "\t"))[["x"]]
hiv3_c3 <- as.vector(read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv3_cluster3.tsv", sep = "\t"))[["x"]]

cluster_anno <- c(hiv1_c1, hiv1_c2, hiv1_c3, hiv2_c1, hiv2_c2, hiv2_c3, hiv3_c1, hiv3_c2, hiv3_c3)
```



### TF gene expression of the 11 TFs with different activity between clusters

```{r}
hivmy2 <- subset(hiv7, cell_type == "Myeloid2" & orig.ident %in% c("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))

# add cluster identity to metadata
ht_cluster <- (
  hivmy2@meta.data %>% 
  rownames_to_column("cell") %>% 
  mutate(ht_cluster = ifelse(cell %in% c(hiv1_c1, hiv2_c1, hiv3_c3),
                             "c1", 
                             "c2")) %>% 
  column_to_rownames("cell"))["ht_cluster"]

hivmy2 <- AddMetaData(hivmy2, metadata = ht_cluster, col.name = "ht_cluster")
#saveRDS(hivmy2,"/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv_myeloid2_subclusters")
```

## Quality of the Myeloid 2 clusters

There is an evident difference in library size and number of features between
the two clusters. Does this mean that there is a technical difference between
the two clusters or is this a biological difference? For example, if cells of 
clsuter1 represent a latently infected group of Myeloid2 cells one might assume
that gene expression is reduced overall due to the infection.

One way to investigate this further would be to have a look at the housekeeping 
genes in both clusters. 

Since there is one cluster which has lower number of counts and features 
anyways, further qualitative analysis of gene expression might be skewed as well.


```{r}
p1 <- hivmy2@meta.data %>% 
  ggplot() +
  geom_boxplot(aes(x = ht_cluster, y = nFeature_RNA, fill = ht_cluster))

p2 <- hivmy2@meta.data %>% 
  ggplot() +
  geom_violin(aes(x = ht_cluster, y = nFeature_RNA, fill = ht_cluster)) +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.8, hjust = 1))

fig <- ggarrange(p1, p2, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob("Quality of Clsuters"))
```

## Housekeeping genes

List of 3804 houseekping genes from: https://www.cell.com/trends/genetics/fulltext/S0168-9525(13)00089-9

A quantitative analysis might be difficult, because due to the small sample
size and large feature number, we get very high p_values for all genes in 
differential gene expression analysis. Visualization of the
housekeeping gene expression, unfortunately does not help either with the 
interpretation.

```{r}
hk_genes <- (read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/HK_genes.txt"))[["V1"]]
```

```{r}
# filter only genes present in the scRNAseq count matrix
hk_genes_filt <- hk_genes[hk_genes %in% rownames(hivmy2@assays$RNA@counts)]
length(hk_genes)
length(hk_genes_filt)

# Find genes from the housekeeping genes that are differentially expressed 
# between the clusters
hk_diff <- FindAllMarkers(hivmy2, features = hk_genes_filt)
Idents(hivmy2) <- "ht_cluster"
hk_diff <- FindMarkers(hivmy2, features = hk_genes_filt, ident.1 = "c1", ident.2 = "c2")
hk_diff %>% head

```

```{r, fig.width=15, fig.height=30}
test <- hk_genes_filt[1:30]
VlnPlot(hivmy2, features = test, group.by = "ht_cluster", ncol = 4)
```


### TF from Farhadian paper

These TFs were used as marker genes for the microlgia like Myeloid2 cell cluster.
Is there a group of cells which has a higher expression of these markers?
Not really

```{r, fig.width = 15, fig.height=40}
#import the 60 genes that were upregulated in Myeloid-2 in the paper compared to the four other myeloid subsets
paper <- read.table(file = "/media/ag-cherrmann/kmikulik/HIV_microglia/data/Myeloid-2_genes_paper.csv",
                    sep = "\t", header = FALSE, skipNul = TRUE)#, n_max = 60)
paper[1,] <- "APOC1"
colnames(paper) <- ""
paper <- as.vector(paper[,1])


VlnPlot(hivmy2, features = paper, group.by = "ht_cluster")
```


# Differential gene expression analyisis


```{r, results="asis"}
hivmy2 <- NormalizeData(hivmy2, verbose = FALSE)
hivmy2 <- ScaleData(hivmy2, verbose = FALSE)
hivmy2 <- FindVariableFeatures(hivmy2, verbose = FALSE)
hivmy2 <- RunPCA(hivmy2, features = VariableFeatures(hivmy2), verbose = FALSE)
ElbowPlot(hivmy2)
hivmy2 <- RunTSNE(hivmy2, dims= 1:10, verbose = FALSE)
Idents(hivmy2) <- "ht_cluster"
de_genes <- FindAllMarkers(hivmy2, verbose = FALSE)
top10 <- de_genes %>% 
  group_by(cluster) %>%
  top_n(n = 5, wt = avg_log2FC) 
top10 %>% 
  kable()
```



### Glass Paper

```{r}
Idents(hiv4) <- "cell_type"
# Find Markers that distinguish My2 from the other Myeloid clusters
my2_markers <- FindMarkers(hiv4, ident.1  = "Myeloid2",
            ident.2  = c("Myeloid1", "Myeloid3", "Myeloid4", "Myeloid5"))
my2_markers <- my2_markers %>% rownames_to_column(var = "gene")
```

```{r, fig.width=15, fig.height=40}
microglia_genes <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/Microglia_gene_signatur_found_in_HIV1_HIV2_samples")
m1 <- my2_markers %>% filter(gene %in% microglia_genes [["gene_name"]] & avg_log2FC >0)
# remove the genes that were already used by the Farhadian paper
m1 <- m1[["gene"]][m1[["gene"]] %in% paper == FALSE]

VlnPlot(hivmy2, features = m1, group.by = "ht_cluster", ncol = 5)

```

# Which of the 11 TFs are the same as in the Gosselin paper?

```{r, fig.width=15, fig.height=12}
tfs_microglia <- (read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/TFs_microglia_Glass_paper.txt"))[["V1"]]

gosselin <- tfs_microglia[tfs_microglia %in% overlap_atac$name]

VlnPlot(hivmy2, features = gosselin, group.by = "ht_cluster", ncol = 4)
```









