---
title: "Triculture dataset"
author: "kmikulik"
date: "9 11 2021"
---

```{r setup}
knitr::opts_chunk$set(echo = TRUE, collapse = TRUE, message = FALSE,
                      cache = TRUE, cache.lazy = FALSE)
set.seed(42)
```



```{r}
library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggplot2)
library(dplyr)
library(ggrepel)
#library(harmony)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(corrplot)
library(rstatix)
library(knitr)
library(metap)
library(multtest)
library(readxl)
```

# Triculture Dataset

The triculture dataset comes from the following paper: (Ryan S. K. et al. Stem Cell Reports, 2020)
https://www.sciencedirect.com/science/article/pii/S2213671120300667?via%3Dihub
They used a Triculture system with induced neurons, induced astrocytes and 
induced microglia.
For the induced microglia they used CX3CR1, IBA1, TMEM119, P2RY12 as markers.
Induced microglia also express CCR5 which is a co-receptor necessary for HIV 
infection. These cells lacked expression of myeloid-progenitor markers.

iMg were infected with HIV for 15 days (resulting in 94% of cells infected, 
positive for HIV capsid protein P24 and exhibiting multinucleation). EFZ was used
for ART. EFZ blocks HIV reverse transcription. This way it was possible to study 
non-productively infected cells. EFZ reduced infection by 2/3.

There was decreased expression of pro-imflammatory genes IL1b, Il8 and TNFalpha
in inf+EFZ vs inf. This suggests a reduced inflmmatory reaction. 

Inf + EFZ had an attenuated inflammatory response compared to Inf.  


There was  a change in signaling pathways caused by EFZ treatment in Uninf + EFZ
vs Uninf. The EFZ treatment alone was very different from the EFZ treatment with 
infection. 

????

**Inf vs Inf + EFZ:**

* inflammatory pathways were significantly activated in Inf iMg compared
with Uinf iMg: IL-8, NFkB, EIF2
* EIF2 is involved in the UPR and ISR (increased in iMg, iAst, iN)
* only in iMg ATF4 mRNA expression was increased (TF downstream of ISR activation)
* it has previously been shown that ISR is activated in neurons and astrocytes 
from human brain samples with HAND

**Uninf vs Inf + EFZ:**

We would expect to see a dampened immune response compared with Inf

* Inf + EFZ had a much milder inflammatory reaction 
  + RhoGDI, CD40 signaling where the only upregulated pathways compared with 
  Uninf iMg
* RhoGDI negaitvely regulates Rac -> functions in inflammatory pathways
* CD40 activates NFkB signaling 
* milder inflammatory response is in line with lower disease severity in HAND + ART

Debaisieux et a., 2015 and Mazzolini et a., 2010
 -> HIV-infected macrophages and uninfected macrophages exposed to HIV display 
 decreased phagocytic capabilities
 
 
 ?????

## Read in Files

* ia/2 = infected with HIV + treated with ART
* ca = uninfected + ART
* ctl = uninfected
* inf = infected 

* the ".mtx" files contain a sparse matrix of class dgTMatrix
* the "barcodes.tsv" files contain the cell barcodes
* the "features.tsv" files contain the feature names as well as ensemble gene ids 
and "gene expression" strings in a third column


There are duplicated features in the data. How do you deal with this problem?
(probably, because there are multiple ensemble gene IDs mapping to the same
gene symbol -> the same gene symbol has multiple expression values)

* You could make them unique with the function `make.unique()`
* You could remove the duplicated features -> this is what I did

```{r}
path <- "/media/ag-cherrmann/kmikulik/HIV_microglia/data/GSE143686_RAW/" #"/media/ag-cherrmann/kmikulik/HIV_microglia/data/GSE143686_RAW/"
all_files <- list.files(path)

all_cond_list <- map(seq.int(3,7), function(n){
  file <- all_files[grepl(paste0("GSM427258", n), all_files)]
  name <- str_extract(file[[1]],"ca|ctl|ia2|ia|inf")
  barcodes <- read.table(paste0(path,file[[1]]), sep = "\t") 
  features <- read.table(paste0(path, file[[2]]), sep = "\t")
  matrix <- readMM(paste0(path, file[[3]]))
  colnames(matrix) <- barcodes$V1
  rownames(matrix) <- features$V2
  matrix <- matrix[!duplicated(rownames(matrix)),]
  seurat_ob <- CreateSeuratObject(matrix, project = name,
                                  assay = "RNA", min.cells = 1, min.features = 1)
  list(name = name, count = matrix, seurat_ob = seurat_ob)
})


```

## Quality control

For each sample separately

* remove genes expressing less than 200 genes or more than 5000
* remove cells expressing > 20% mitochondrial genes

```{r}
# add percentage of mitochondiral genes to metadata
all_cond <- map(seq.int(1,5), function(n){
  seurat_ob <- all_cond_list[[n]]$seurat_ob
  seurat_ob[["percent_mt"]] <- PercentageFeatureSet(seurat_ob, pattern = "^MT-")
  plot <- VlnPlot(seurat_ob, features = c("nFeature_RNA", "percent_mt", "nCount_RNA"), ncol = 3)
  seurat_ob <- subset(seurat_ob, nFeature_RNA > 200 & nFeature_RNA < 5000 & percent_mt < 20)
  list(name = all_cond_list[[n]]$name, seurat_ob = seurat_ob, plot = plot)
})
```

```{r, fig.width = 15, fig.height = 10}
ggarrange(all_cond[[1]]$plot, 
             all_cond[[2]]$plot,
             all_cond[[3]]$plot,
             all_cond[[4]]$plot, 
             all_cond[[5]]$plot, ncol = 2, nrow = 3)

```

### Merge Seurat objects after Quality Control

```{r, fig.width=15}
triculture <- merge(all_cond[[1]]$seurat_ob,
                    c(all_cond[[2]]$seurat_ob,
                      all_cond[[3]]$seurat_ob,
                      all_cond[[4]]$seurat_ob,
                      all_cond[[5]]$seurat_ob),
                    project = "triculture_all_conditions",
                    add.cell.ids = c(paste0(all_cond[[1]]$name),
                                     paste0(all_cond[[2]]$name),
                                     paste0(all_cond[[3]]$name),
                                     paste0(all_cond[[4]]$name),
                                     paste0(all_cond[[5]]$name)))
p1 <- triculture@meta.data %>% ggplot() +
  geom_boxplot(aes(x = orig.ident, y = nFeature_RNA))

p2 <- triculture@meta.data %>% ggplot() +
  geom_boxplot(aes(x = orig.ident, y = nCount_RNA))

p3 <- triculture@meta.data %>% ggplot() +
  geom_boxplot(aes(x = orig.ident, y = percent_mt))
  
p4 <- triculture@meta.data %>% ggplot() +
  geom_violin(aes(x = orig.ident, y = nFeature_RNA))

p5 <- triculture@meta.data %>% ggplot() +
  geom_violin(aes(x = orig.ident, y = nCount_RNA))

p6 <- triculture@meta.data %>% ggplot() +
  geom_violin(aes(x = orig.ident, y = percent_mt))
figure <- ggarrange(p1, p2, p3, p4, p5, p6, ncol = 3, nrow = 2)
annotate_figure(figure, top = text_grob("Qualtiy Control"))
```

## Normalization, Scaling, Find Variable Features

* log normalization
* scale data -> zero mean and variance = 1
* Find the 2000 most highly variable features

```{r}
triculture <- NormalizeData(triculture, verbose = FALSE)
triculture <- ScaleData(triculture, verbose = FALSE)
triculture <- FindVariableFeatures(triculture, verbose = FALSE)
```

## Dimensionality reduction

* PCA
* use elbow plot to determine dimensions that contribute to variability
* Find the nearest neighbors
* Find Clusters based on KNN-graph
* Calculate TSNE embedding

```{r, fig.height = 5}
triculture <- RunPCA(triculture, features = VariableFeatures(triculture), verbose = FALSE)
ElbowPlot(triculture)

triculture <- FindNeighbors(triculture)
triculture <- FindClusters(triculture)

triculture <- RunTSNE(triculture, dims = 1:10, verbose = FALSE)
```

```{r}
DimPlot(triculture, reduction = "tsne", pt.size = .01, label = TRUE)

#saveRDS(triculture, "/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_seurat_object")
```

## Cell type annotation

* *Microglia:* AIF1, SPI1, CD4
* *Astrocytes:* SOX9, THBS1
* *Neurons:* SYN1, MAP2

```{r, fig.widht = 15, fig.height=12}
#markers <- FindAllMarkers(triculture)

#triculture <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_seurat_object")
markers <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/marker_genes_triculture_dataset")


top_2 <- markers %>% group_by("seurat_cluster") %>% top_n(n = 2, wt = avg_log2FC)

p1 <- FeaturePlot(triculture, features = "AIF1")
p2 <- FeaturePlot(triculture, features = "SPI1")
p3 <- FeaturePlot(triculture, features = "CD4")
p4 <- FeaturePlot(triculture, features = "MAP2")
p5 <- FeaturePlot(triculture, features = "SYN1")
p6 <- FeaturePlot(triculture, features = "THBS1")
p7 <- FeaturePlot(triculture, features = "SOX9")

ggarrange(p1,p2,p3,p4,p5,p6,p7, ncol = 2, nrow =4 )

```

### Gene expression 

Annotate different Clusters according to the gene expression of marker genes.

```{r, fig.width=15}

VlnPlot(triculture, features = c("CD4", "SPI1", "AIF1"), group.by = "seurat_clusters")#, group.by = seurat_cluster)
VlnPlot(triculture, features = c("MAP2", "SYN1"), group.by = "seurat_clusters")
VlnPlot(triculture, features = c("THBS1", "SOX9"), group.by = "seurat_clusters")
```

```{r}
# 4, 9, 11, 18 = Microglia
# 0, 1, 2, 3,8, 10, 13, 15, 18, 19 = Neurons 
# 6, 7, 12, 14, 16 = Astrocytes
# 5, 17 are undesginated
cell_types <- (triculture@meta.data %>% 
  mutate(cell_type = ifelse(seurat_clusters %in% c(4,9,11), "iMg", 
                            ifelse(seurat_clusters %in% c(6,7,12,14,16), "iAst", 
                                   ifelse(seurat_clusters %in% 
                                            c(0,1,2,3,8,10,13,15,18,19), 
                                          "iNeurons", "Undesignated")))))[["cell_type"]]
  
triculture <- AddMetaData(triculture, cell_types, col.name = "cell_type")

```


### Remove undesignated cells

There is one group of cells where it is not clear which cell state they 
represent (Undesignated), we will exclude them from further analysis.

```{r}
Idents(triculture) <- "cell_type"
DimPlot(triculture, reduction = "tsne", pt.size = .01, label = TRUE)
```


```{r}
triculture_clean <- subset(triculture, cell_type %in% c("iMg", "iNeurons", "iAst")) 
```

Top20 marker genes for each cell cell type: 
 
```{r, fig.width=15, fig.height=25}
top_n <- markers %>% group_by (cluster) %>% top_n(n = 20, wt = avg_log2FC)
DoHeatmap(triculture_clean, features = top_n$gene)
```

How many cells of each cell type and condition are present in the dataset?

```{r, results = "asis"}
triculture_clean@meta.data %>% group_by(cell_type, orig.ident) %>% 
  summarize(n = n()) %>% 
  pivot_wider(names_from = orig.ident, values_from = n) %>% 
  kable() # to make table look nice in Rmarkdown
  
```


## Different infection and treatment conditions

Interestingly, the infected cells treated with EFZ (Inf+EFZ) clearly separate 
from the infected, but untreated cells (Inf) and the the uninfected cells
(Uninf, Uninf+EFZ) in the microglia cluster.


```{r}
# rename origin of ident for easier interpretation
origin_vector <- (triculture_clean@meta.data %>% mutate(orig.ident = ifelse(orig.ident %in%
                                                 c("ia", "ia2"), "Inf+EFZ", 
                                                 ifelse(orig.ident == "ca",
                                                        "Uninf", ifelse(
                                               orig.ident == "ctl", "Uninf+EFZ",
                                               "Inf")))))[["orig.ident"]]

triculture_clean <- AddMetaData(triculture_clean, origin_vector, col.name = "orig.ident")
```

```{r}
DimPlot(triculture_clean, reduction = "tsne", group.by = "orig.ident", pt.size = .01, )
```


## Inflammatory gene expression in iMg

The largest change in gene expression related to inflammation was found in iMg
among the 3 cell types. Shown are inflammatory genes. 

```{r, fig.height = 12}
inflamm_genes <- c("TNF", "ACSL1", "MGST3", "B2M", "IFNGR2", "GRINA", "TLR2", 
                   "SOD2", "GLUL", "CFLAR", "CTNNB1", "CXCL8", "CCL4", "IL1B",
                   "ATF4", "CXCL8", "CCL4", "CCL2", "IL1B", "ATF4", "CXCL1", 
                   "NFKB1", "TANK", "APOE", "TREM2", "LY96", "SRC", "NCF1", 
                   "HCK", "LYN", "VASP", "C3", "C5", "RAP1B", "RAP2B", "MAF",
                   "JUND", "SQSTM1", "FTH1", "RHOQ", "CD36", "FOS", "NFKBIA",
                   "ICAM1", "TNFAIP3", "LYZ", "PPP1R12A", "RND3", "CYBA", "IRF8",
                   "SLC25A6")

DoHeatmap(triculture_clean, features = inflamm_genes)
```

The gene expression of inflammatory genes in microglia was not very different
between the conditions. The Inf+EFZ cells show a slightly different pattern of
gene expression.

```{r, fig.height=12}
DoHeatmap(subset(triculture_clean, cell_type == "iMg"), 
          features = inflamm_genes, group.by = "orig.ident")
```



# Integration of triculture dataset with dataset of blood & CSF samples from HIV-infected patients

The HIV Blood & CSF dataset is provided by Farhadian S.F. et al., JCL Insight, 2018.
https://doi.org/10.1172/jci.insight.121718

## Integration

Integration of two datasets is based on the Seurat Method. It first identifies 
anchors between datasets, calculates scores for all anchors and 
then calculates an integrated expression matrix which can be used for downstream
analysis.
For more details see: 
Stuart T. et al. 2019. Comprehensive Integration of Single-Cell Data. Cell 177, 1888-1902
DOI: 10.1016/j.cell.2019.05.031

```{r}
hiv7 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds")
hiv7 <- NormalizeData(hiv7, verbose = FALSE)
hiv7 <- FindVariableFeatures(hiv7, verbose = FALSE)
#triculture_clean <- subset(triculture, cell_type %in% c("iMg", "iNeurons", "iAst")) 
```

1. Identify features which are variable across datasets

```{r}
# add both datasets to a lits
data_list <- list(hiv = hiv7, triculture = triculture_clean)

# select features that are repeatedly variable across datasets for integration
features <- SelectIntegrationFeatures(object.list = data_list, verbose = FALSE)

```

2. Identify anchors between the two datasets and create an integrated data assay

```{r}
anchors <- FindIntegrationAnchors(object.list = data_list, 
                                  anchor.features = features, verbose = FALSE) 
combined <- IntegrateData(anchorset = anchors, verbose = FALSE)
```


3. Run a single integrated analysis on all cells

The original data is still saved in the RNA assay slot

```{r}
combined <- ScaleData(combined, verbose = FALSE)
combined <- RunPCA(combined, verbose = FALSE)
#ElbowPlot(combined)
combined <- RunTSNE(combined, reduction = "pca", dims = 1:10, verbose = FALSE)
combined <- FindNeighbors(combined, reduction = "pca", dims = 1:10, verbose = FALSE)
combined <- FindClusters(combined, resolution = .3, verbose = FALSE)
```

```{r}
# add annotations from which dataset the sample came from
dataset_vector <- (combined@meta.data %>% mutate(dataset = ifelse(orig.ident %in%
                                                 c("HIV1_CSF", "HIV2_CSF",
                                                   "HIV3_CSF", "HIV1_Bld", 
                                                   "HIV2_Bld", "Uninfected1_CSF",
                                                   "Uninfected2_CSF"),
                                               "csf", "triculture")))[["dataset"]]

combined <- AddMetaData(combined, dataset_vector, col.name = "dataset")
```

```{r}
# add information on the cell groups in the combined datasets
combined_clusters <- (combined@meta.data %>% 
  mutate(combined_clusters = ifelse(seurat_clusters %in% c(2,10), "Microglia", 
                            ifelse(seurat_clusters %in% c(3,9), "iAst", 
                                   ifelse(seurat_clusters %in% c(0,1,7,11),
                                          "iNeurons", 
                                          ifelse(seurat_clusters %in% c(8,5,6), "Immune Cells", "Undesignated"))))))[["combined_clusters"]]
  
combined <- AddMetaData(combined, combined_clusters, col.name = "combined_clusters")

#saveRDS(combined,"/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_hiv_combined_seurat_object")

```


#### Visualization of the integrated dataset

As expected the Myeloid cells from the HIV-CSF dataset overlap with the 
induced Microglia cells from the Triculture dataset.

```{r, fig.width=15, fig.height=10}
#saveRDS(combined, "/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_hiv_combined_seurat_object_2")
#combined = readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_hiv_combined_seurat_object_2")

p1 <- DimPlot(combined, reduction = "tsne", group.by  = "dataset",
              label = TRUE) + NoLegend()
p2 <- DimPlot(combined, reduction = "tsne", group.by = "cell_type",
              label = TRUE, repel = TRUE) + NoLegend()
p3 <- DimPlot(combined, reduction = "tsne", group.by = "cell_type",
              split.by = "dataset", label = TRUE, repel = TRUE) + NoLegend()
p4 <- DimPlot(combined, reduction = "tsne", group.by = "seurat_clusters",
              label = TRUE) + NoLegend()

ggarrange(p1, p2, p3,p4, ncol = 2, nrow = 2)

```

```{r, fig.width=15, fig.height=5}

a1 <- DimPlot(combined, reduction = "tsne", split.by = "dataset", 
        group.by = "seurat_clusters", label = TRUE, repel =TRUE)

a2 <- DimPlot(combined, reduction = "tsne", split.by = "dataset", 
              group.by = "orig.ident")#, label = TRUE, repel =TRUE)
ggarrange(a1, a2, ncol = 2, nrow = 1)
```

## Visualization of Myeloid lineage cluster (iMg, Myeloid1/2/3/4/5)

* The Myeloid cells cluster together and mainly contribute to the same subcluster.
* Myeloid 4 is an exception and belongs to a different subcluster.

```{r, fig.width=15, fig.height=10}

subset_infected <- subset(combined, orig.ident %in% c("HIV1_CSF", "HIV2_CSF", "HIV3_CSF",
                                             "HIV1_Bld", "HIV2_Bld","Inf+EFZ") &
                            cell_type %in% c("Myeloid1", 
                                                            "Myeloid2", 
                                                            "Myeloid3", 
                                                            "Myeloid4", 
                                                            "Myeloid5", 
                                                            "iMg"))
subset_monocytes <- subset(combined, cell_type %in% c("Myeloid1", 
                                                            "Myeloid2", 
                                                            "Myeloid3", 
                                                            "Myeloid4", 
                                                            "Myeloid5", 
                                                            "iMg"))

p1 <- DimPlot(subset_infected)
p2 <- DimPlot(subset_infected, group.by = "orig.ident")
p3 <- DimPlot(subset_infected, group.by = "seurat_clusters")
ggarrange(p1,p2,p3, ncol=2, nrow = 2)
```



#### cell types and conditions partially separate according to the two iMg clusters

* The myeloid clusters (Myeloid1/2/3/5) seem to colocalize with the Inf+EFZ iMg,
however the Myeloid4 cluster is an exception. 
* the Inf iMg clearly separate from the Inf+EFZ
* The separation is not so clear for the Uninf+EFZ and Uninf cells. They 
contribute to both clusters, but the Uninf+EFZ seem to contribute more to the
cluster with the Inf+EFZ than the Uninf.
* This is also represented by the clusters 4 & 5 identified by Seurat
* both the uninfected CSF cells, as well as the HIV+ CSF samples cluster with 
Inf+EFZ cluster, which also contains Uninf+EFZ cells, so this is no contradiction

```{r, fig.width=15, fig.height=10}

p1 <- DimPlot(subset_monocytes)
p2 <- DimPlot(subset_monocytes, group.by = "orig.ident")
p3 <- DimPlot(subset_monocytes, group.by = "seurat_clusters")
ggarrange(p1, p2, p3, ncol = 2, nrow = 2)
```

```{r,fig.width=15, fig.height=10}
DimPlot(subset_monocytes, group.by = "seurat_clusters", 
        split.by = "orig.ident", ncol = 4)
DimPlot(subset_monocytes, group.by = "seurat_clusters", 
        split.by = "cell_type", ncol = 4)
```


# Marker genes between different conditions

#### Features upregulated in Uninf+EFZ vs. Inf+EFZ:

* **CCL3**, **CCL4** = cehmoattractants produced by macrophages, primary ligands for the 
HIV-1 virus coreceptor CCR5 (https://pubmed.ncbi.nlm.nih.gov/16773571/)
* **Il1B**
  + is produced by activated macrophages -> mediator of the inflammatory 
  response
  + involved in neurodegeneration (https://pubmed.ncbi.nlm.nih.gov/12794045/)
  + HIV-1 infection induces interleukin-1beta production, so why is it more highly
  expressed in uninfected than in infected cells? (https://pubmed.ncbi.nlm.nih.gov/24939850/)
  One explanation might be trained immunity of these monocytes which is associated
  with HIV. Pro-inflammatory pathways are upregulated in monocytes of HIV patients
  on ART. Il-b expression was hihger in HIV patients compared with controls.
  (van der Heijden W. A. et al., JCL Insight, 2021 --> study on circulating 
  immune cells)
* **IL6** is also more highly expressed in monocytes from HIV patients 
  (van der Heijden W. A. et al., JCL Insight, 2021 --> study on circulating 
  immune cells)
* **TNF**, **IL1b** and **IL6** are monocyte-derived pro-inflammatory cytokines, expressed by 
*activated microglia*. IL13, and IL10 would be expressed by alternatively activated 
macrophages (M2) -> anti-inflammatory.
(van der Heijden W. A. et al., JCL Insight, 2021)
* **TNFAIP2** is a primary response gene of TNFalpha (expression is regulated by NFkB)
-> mediator of inflammation (Jia L., Journal of Moleuclar Medicine, 2018)
* **CCL20**: 
  + CCL20 is constitutively expressed in lymphoid tissue and upregulated in 
  inflammation -> homeostatic and inflammatory role.  CCR6/CCL20 axis may be
  involved in HIV pathogenesis and immunity. 
  CCL20 was able to facilitate HIV-1 integration in T cells. It leads to 
  propagation of the virus, because it recruits T cells and DC which then migrate 
  and spread the virus. Anti-HIV activity of CCR6 by interfering with reverse 
  transcription.
  -> *CCR6 seems to be a double-edged sword in HIV infection which contributes 
  to immunity and spread of HIV.
  **Since CCL20 is found at epithelial and 
  mucosal sites the chemokine may contribute to the innate defense against HIV 
  entry.**
  (Lee A. Y. S., Journal of General Virology, 2017)
  + chemokine interacting with CCR6 receptors, anti_HIV activity in combination 
  with human beta defensins
(Aziz N. et al., J AIDS Clin Res, 2016)
* **NFkB** is an inducible transcription factor and a mediator of inflammatory
response in innate and adaptive immunity. It induces the expression of various 
pro-inflammatory genes including those encoding cytokines and chemokines. 
It participates in inflammasome regulation. **NFKB1** = **p50**. *Inappropriate 
activation of NFKB is associated with inflammatory diseases*
(Liu T. et al., Signal Transduction and Targeted therapy, 2017)


```{r, results = "asis"}
# isolate iMg cells
microglia <- subset(combined, cell_type == "iMg")
# only iMg cells to identify potential signatures between infected vs uninfected
Idents(microglia) =  "orig.ident"
# find marker genes between Uninf+EFZ and Inf+EFZ
uninf_inf_art_marker <- FindMarkers(microglia, ident.1 = "Uninf+EFZ", ident.2 = "Inf+EFZ")
uninf_inf_art_marker %>% arrange(desc(avg_log2FC)) %>% head(20) %>% kable()
```

#### Features upregulated in Inf+EFZ vs Uninf+EFZ

* **APOE**
  + expressed by miroglia and astrocytes
  + major carrier protein mediating transport and delivery of cholesterol,
  lipids in the brain
  + HIV1 Tat (viral protein) can bind to LRP1 (controls APOE uptake in the brain)
  (Khan N., J Neuroinflammation, 2018)
* **LPAR6**
  + expressed in macrophages
* **RAB32**
  + ER stress is a hallmark of neurodegenerative diseases
  + Rab GTPases are essential for cytokine production in neuroinflammation
  (Liang Y., Neuroscience Letters, 2012)
  + involved in formation of autophagosomes, upregulated in brain inflammation,
  (Prashar A., Front. Cell. Infect. Microbiol., 2017)
* **TXNIP**
  + sequesters TRX (reduced thioredoxin), inhibiting its function -> 
  increased oxidative stress
  + cellular stress factors regulate TXNIP expression
  + expressed by microglia, neurons and astrocytes
  + TXNIP expression promotes inflammatory activation (Tsubaki H., Int J Mol Sci., 2020)
  + aggravates brain injury by activating NLRP3 inflammasomes (Liu Y., Lab Invest, 2021)
* **PDK4**
  + metabolic checkpoint of M1 polarization of macrophages
  (Min B.k., Frontiers in Immunology, 2019)
* **NEAT1** (expression is changed by HIV1 infection)
  + NEAT1 inhibits microglial polarization towards the pro-inflammatory M1 
  phenotype in ischemia (cell culture) (Ni X. Scientific Reports, 2020)
  + in CD4 T Cells downregulation of NEAT1 leads to higher HIV replication
  + critical for the substructural integrity of the nuclear paraspeckle, which
  is important for HIV replication
  + potential biomarker of HIV infection (Shen Li, IJID, 2019)
  + HIV seems to exploit the fact that activated CD4 T cells have decreased NEAT1 expression
  + higher replication if NEAT1 is repressed (Liu H., Virology, 2018)
  This might make sense since we also want to reduce viral replication with EFZ,
  so in these cells NEAT1 might be upregulated reducing HIV1 replication
* **DAB2**
  + DAB2 is a marker of microglia activation (Bell-Temin H., Mol Cell Proteomics, 2015)
  + expressed in early microglia more than at later stages
  (Matcovitch-Natan O., Science, 2016)
* **APOC1**
After HVI1 enters the brain int can exist in CSF, perivascular macrophages, 
microglia and astrocytes. With ART you can supress HIV1 replication in plasma
and CSF, but some HIV1 resrevoirs and neuroinflammation persist. This results in
the 50% prevalence of HAND. Tat enhances HIV infectivity. 
(Khan N., J Neuroinflammation, 2018)

```{r}
uninf_inf_art_marker %>% arrange(avg_log2FC) %>% head(20) %>% kable()
```


#### Features differentially expressed between Uninf vs Inf+EFZ
* NEAT1 is more highly expressed in Inf+EFZ

```{r, results = "asis"}
# find markers between Uninf and Inf+EFZ 
inf_art_uninf_marker <- FindMarkers(microglia, ident.1 = "Uninf", 
                                ident.2 = "Inf+EFZ")
inf_art_uninf_marker %>% rownames_to_column("gene") %>%
  filter(gene == "NEAT1") %>% 
  kable()
```

#### Features differentially expressed between Inf vs Inf+EFZ

* **NEAT1** is downregulated in Inf compared to Inf+EFZ. 
Since NEAT1 has an anti-HIV effect this is expected.
(Liu H., Virology, 2018)(Shen Li, IJID, 2019)

```{r, results = "asis"}
# find marker genes between Inf and Inf+EFZ
inf_inf_marker <- FindMarkers(microglia, ident.1 = "Inf", ident.2 = "Inf+EFZ")
inf_inf_marker %>% rownames_to_column("gene") %>%
  filter(gene == "NEAT1") %>% 
  kable()
```



# Inflammatory Pathways 

Ryan S.K. et al., Stem Cell Reports, 2020 showed that two pathways are
specifically activated in the Inf+EFZ condition.

##### Myeloid2 cells

The Inf+EFZ iMg had a milder immune response than the Inf iMg. The RhoGDI and 
CD40 inflammatory signaling pathway were the only two upregulated compared with
Uninf iMg. This effect was not seen in Uninf+EFZ, so it does not seem to be 
caused by EFZ alone, but rather a combination of infection and EFZ treatment.
(Ryan S. K. et al., Stem Cell Reports, 2020)

Compared to Uninf iMg they show in the paper that two inflammatory pathways are
upregulated: 

* RhoGDI signaling:
  + We expect PP1R12A to be upregulated 
  + We expect RND3, CD44, ACTG1 to be downregulated
* CD40 signaling:
  + We expect FOS to be upregulated
  + we expect NFKBIA, ICAM1, TFNAIP3 to be downregulated 
  
**Can we see a similar pattern in the Myeloid2 clusters identified by Scenic?**

```{r, results = "asis"}
uninf_inf_art_marker %>% rownames_to_column(var = "gene") %>% filter(gene %in% c("RND3", "CD44", "ACTG1", "PPP1R12A")) %>% kable()

uninf_inf_art_marker %>% rownames_to_column(var = "gene") %>% filter(gene %in% c("FOS", "NFKBIA", "ICAM1", "TNFAIP3")) %>% kable()

```

```{r, fig.width=15, fig.height=12}
# isolate microglia cells
microglia <- subset(combined, cell_type == "iMg")

# read in Seurat object containing Myeloid2 cells
hivmy2 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv_myeloid2_subclusters")


p1 <- VlnPlot(hivmy2, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), group.by = "ht_cluster", ncol = 2)

p2 <- VlnPlot(hivmy2, features =c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), group.by = "ht_cluster", ncol = 2)

p3 <- VlnPlot(microglia, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

p4 <- VlnPlot(microglia, features = c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

ggarrange(p2, p4, p1, p3, ncol = 2, nrow = 2)

```


##### Myeloid5 cells


```{r, fig.width=15, fig.height=12}
# read in Seurat object contianing alls samples and conditions
hiv7 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds")

# isolate the Myeloid5 cluster
hivmy5 <- subset(hiv7, cell_type == "Myeloid5" & orig.ident %in% c("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))

# process Myeloid 5 Seurat object
hivmy5 <- ScaleData(hivmy5, verbose = FALSE)
hivmy5 <- NormalizeData(hivmy5, verbose = FALSE)
hivmy5 <- ScaleData(hivmy5, verbose = FALSE)
hivmy5 <- FindVariableFeatures(hivmy5, verbose = FALSE)
hivmy5 <- RunPCA(hivmy5, features = VariableFeatures(hivmy5), verbose = FALSE)
#ElbowPlot(hivmy5)
hivmy5 <- RunTSNE(hivmy5, dims= 1:10, verbose = FALSE)
```

```{r, fig.width=15, fig.height=12}

p1 <- VlnPlot(hivmy5, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
              group.by = "orig.ident",ncol = 2)

p2 <- VlnPlot(hivmy5, features =c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
              group.by = "orig.ident", ncol = 2)

p3 <- VlnPlot(microglia, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

p4 <- VlnPlot(microglia, features = c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

ggarrange(p2, p4, p1, p3, ncol = 2, nrow = 2)
```



# Subcluster the Myeloid-Lineage Cluster of the integrated dataset

Isolating the iMg and all Myeloid cell clusters from the integrated dataset and
normalizing, sclaing and reclustering these isolated cells might give us
additional insights. 

```{r}

microglia <- subset(combined, cell_type %in% c("iMg",
                                                "Myeloid1",
                                                "Myeloid2",
                                                "Myeloid3",
                                                "Myeloid4",
                                                "Myeloid5"))
microglia_recluster <- SCTransform(microglia, verbose = FALSE)
microglia_recluster <- ScaleData(microglia_recluster, verbose = FALSE)
microglia_recluster <- RunPCA(microglia_recluster, verbose = FALSE)
ElbowPlot(microglia_recluster)
microglia_recluster <- RunTSNE(microglia_recluster, reduction = "pca", dims = 1:10)


microglia_recluster <- FindNeighbors(microglia_recluster)
microglia_recluster <- FindClusters(microglia_recluster, resolution = 0.5)
```

It is no surprise that cells from the triculture dataset separate from the 
CSF-derived cells. When we recluster the myeloid lineage derived cells, we 
obtain 5 subclusters for the iMg and 3 subclusters for the 5 Myeloid clusters. 
The Myeloid2 and Myeloid5 separate from Myeloid1/2/3 which is also expected, 
since Myeloid2/5 originate from CSF samples and Myeloid1/2/3 originate from 
blood samples. Interestingly, both the Myeloid5 cells and
the Myeloid2 cells are split between the cluster 6 and 7. This might be related 
to the infection status of these cells.
When performing GRN inference with Scenic we also identified two potentially 
biologically relevant subclusters of the Myeloid2 cells. Do these two clusters
correlate with the clusters found here?

```{r, fig.width=15, fig.height=12}
a1 <- DimPlot(microglia_recluster, reduction = "tsne",
              group.by = "seurat_clusters", pt.size = .4)
a2 <- DimPlot(microglia_recluster, reduction ="tsne", 
              group.by = "cell_type", pt.size = .4)
a3 <- DimPlot(microglia_recluster, reduction ="tsne",
              group.by = "dataset", pt.size = .4)
ggarrange(a1, a2, a3, ncol = 2, nrow = 2)
```

```{r, fig.width=15, fig.height=10}
a5 <- DimPlot(microglia_recluster, reduction ="tsne", 
              split.by = "cell_type", pt.size = .4, ncol=3)
a5
```

It looks like the two clusters identified this way correspond to different
sample origins. Cluster 7 mainly consists of cell from the Uninfected CSF sample
and infected CSF sample 3. So this might be a batch effect that HIV3_CSF 
separates from HIV1_CSF and HIV2_CSF.


```{r, fig.width=15, fig.height=10}
a1 <- DimPlot(subset(microglia_recluster, cell_type == "iMg"), reduction = "tsne",
              group.by = "orig.ident", pt.size = .4)
a2 <- DimPlot(subset(microglia_recluster, cell_type != "iMg"), reduction = "tsne",
              group.by = "orig.ident", pt.size = .4)
a3 <- DimPlot(subset(microglia_recluster, cell_type == "iMg"), reduction = "tsne",
              group.by = "seurat_clusters", pt.size = .4)
a4 <- DimPlot(subset(microglia_recluster, cell_type != "iMg"), reduction = "tsne",
              group.by = "seurat_clusters", pt.size = .4)
ggarrange(a1, a2, a3, a4, ncol = 2, nrow = 2)
```

Since the separation into two clusters seems to be related to the sample origin
rather than to the infection status, we would not expect the clusters to overlap
with the clusters identified after GRN inference with SCENIC. In the figure 
below this is visualized for only the Myeloid2 cells from infected CSF samples
(HIV1_CSF, HIV2_CSF, HIV3_CSF).

```{r, fig.width=15, figh.height = 5}
# read in Myeloid2 cells
hivmy2 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv_myeloid2_subclusters")

my2 <- subset(microglia_recluster, cell_type == "Myeloid2" & 
                orig.ident %in% c("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))

my2 <- AddMetaData(my2, hivmy2@meta.data["ht_cluster"], 
                   col.name = "TF_activity_clusters")
p1 <- DimPlot(my2, group.by = "seurat_clusters")
p2 <- DimPlot(my2, group.by = "TF_activity_clusters") 
p3 <- DimPlot(my2, group.by = "orig.ident")
figure <- ggarrange(p1, p2, p3, ncol = 3, nrow = 1)
annotate_figure(figure, 
                top = text_grob("Correlation with TF acitivty clusters", 
               color = "black", face = "bold", size = 20))
#my25 <- subset(microglia, seurat_clusters %in% c(6,7) &cell_type %in% c("Myeloid2", "Myeloid5"))

```


For this reason the reclustering did not lead to any useful insights and will 
not be analyzed further. 



# Infected vs Uninfected gene signatures 

## investigate gene signatures in Myeloid cells from CSF

Checking the differentially expressed genes and trying to identify a signature 
through literature search turned out to be very difficult. Therefore, I will try
a different approach and just use the topdifferentially expressed genes.

Use the top 30 upregulated genes in Inf+EFZ iMg compared to Uninf+EFZ iMg and use
them as signatures for infection. Transfer this information to the Myeloid2 
cluster from the CSF samples (Farhadian S. F. et al., JCL Insight, 2018)


Do the same for the top 30 genes upregulated in Uninf+EFZ iMg compared to Inf+EFZ.

```{r}
# get the top 30 genes upregulated in Inf+EFZ vs Uninf+EFZ
inf_art_signature <- ((uninf_inf_art_marker %>% arrange(avg_log2FC) %>%
  rownames_to_column("gene"))  ["gene"]) [1:30,]

# get the top 30 genes upregulated in Uninf+EFZ vs Inf+EFZ
uninf_art_signature <- ((uninf_inf_art_marker %>% arrange(desc(avg_log2FC)) %>% 
                       rownames_to_column("gene")) ["gene"]) [1:30,]

# get the top 30 genes upregulated in Uninf vs Inf+EFZ
uninf_signature <- ((inf_art_uninf_marker %>% arrange(desc(avg_log2FC)) %>% 
                       rownames_to_column("gene")) ["gene"]) [1:30,]


```


```{r, fig.width=15, fig.height=30}
inf_art_signature_filt <- inf_art_signature[inf_art_signature %in% 
                                              rownames(hivmy2@assays$RNA@counts)]
uninf_art_signature_filt <- uninf_art_signature[uninf_art_signature %in% 
                                                  rownames(hivmy2@assays$RNA@counts)]
uninf_signature_filt <- uninf_signature[uninf_signature %in% 
                                               rownames(hivmy2@assays$RNA@counts)]
VlnPlot(hivmy2, features = inf_art_signature_filt,
        group.by = "ht_cluster", ncol = 4)
VlnPlot(hivmy2, features = uninf_art_signature_filt, 
        group.by = "ht_cluster", ncol = 4)
VlnPlot(hivmy2, features = uninf_signature_filt,
        group.by = "ht_cluster", ncol = 4)
```


# Gene signature scoring

Using the signature genes from differential gene expression analysis between
Inf+EFZ and Uninf+EFZ we can compute a score of activity of these gene sets in 
the Myeloid2 cluster of CSF samples. We can plot these scores in a TSNE 
representation, but we can also get the scores from the metadata and use them 
elsewhere. 

Maybe you can also find the Inf and Uninf signatures in the Myeloid2 cells from
the CSF samples?

There are some cells with a high score for the Inf+EFZ gene signature, but very
few cells with the Uninf+EFZ gene signature.

```{r, fig.widht = 15}
hivmy2_sign <- AddModuleScore(hivmy2,
                           features = inf_art_signature_filt)

p1 <- FeaturePlot(hivmy2_sign, features = "Cluster1", pt.size = .01) +#, split.by = "ht_cluster") +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Inf+EFZ gene signature from iMg applied to CSF Myeloid2 cells")


hivmy2_sign <- AddModuleScore(hivmy2,
                           features = uninf_art_signature_filt)

p2 <- FeaturePlot(hivmy2_sign, features = "Cluster1", pt.size = .01) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Uninf+EFZ gene signature from iMg applied to CSF Myeloid2 cells")

ggarrange(p1, p2, ncol = 2)
```

#### Split Myeloid2 cells into TF activity clusters 

There are some cells which have a high score for this signature, but they do not
correspond to the two clusters (c1, c2) obtained from analysis of 
TF activity (SCENIC).

Interestingly, there are hardly any cells which show a high score for the
Uninf+EFZ gene signature. For this reason it might be interesting to see if 
some cells show a high score for a Uninf (without EFZ treatment) score.

```{r, fig.width=15}
hivmy2_inf_sign <- AddModuleScore(hivmy2,
                           features = inf_art_signature_filt)

p1 <- FeaturePlot(subset(hivmy2_inf_sign, ht_cluster == "c1"),
                  features = "Cluster1", pt.size = 1) +#, split.by = "ht_cluster") +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")

p2 <- FeaturePlot(subset(hivmy2_inf_sign, ht_cluster == "c2"),
                  features = "Cluster1", pt.size = 1) +#, split.by = "ht_cluster") +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c2")

figure <- ggarrange(p1, p2, ncol = 2)#, nrow = 2) 
annotate_figure(figure, top = text_grob("Inf+EFZ gene signature from iMg applied to CSF Myeloid2 cells", 
               color = "black", face = "bold", size = 20))


hivmy2_uninf_efz_sign <- AddModuleScore(hivmy2,
                           features = uninf_art_signature_filt)

p3 <- FeaturePlot(subset(hivmy2_uninf_efz_sign, ht_cluster == "c1"), 
                  features = "Cluster1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")
  
p4 <- FeaturePlot(subset(hivmy2_uninf_efz_sign, ht_cluster == "c2"),
                  features = "Cluster1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")

figure <- ggarrange(p3, p4, ncol = 2)#, nrow = 2)
annotate_figure(figure, top = text_grob("Uninf+EFZ gene signature from iMg applied to CSF Myeloid2 cells", 
               color = "black", face = "bold", size = 20))
  
```

There are a few more cells with high scores for the Uninf gene signature 
(without EFZ treatment) than for the Uninf+EFZ, but stil very few cells.

```{r, fig.width=15}
hivmy2_uninf_sign <- AddModuleScore(hivmy2,
                           features = uninf_signature_filt)

p3 <- FeaturePlot(subset(hivmy2_uninf_sign, ht_cluster == "c1"), 
                  features = "Cluster1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")
  
p4 <- FeaturePlot(subset(hivmy2_uninf_sign, ht_cluster == "c2"),
                  features = "Cluster1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")

figure <- ggarrange(p3, p4, ncol = 2)#, nrow = 2)
annotate_figure(figure, top = text_grob("Uninf gene signature from iMg applied to CSF Myeloid2 cells", 
               color = "black", face = "bold", size = 20))
  
```




#### Percentage of cells with positive Inf+EFZ gene signature score

11% of the Myeloid2 cells have a positive score for the Inf+EFZ gene
signature (17 out of 145 cells have a positive score).

On the other hand, only 4% of the Myeloid2 cells have a positive score 
for the Uninf+EFZ gene signature, while 10% of the Myeloid2 cells have a 
positive score for the Uninf gene signature (without EFZ treamtent).
```{r, results = "asis"}
# divide the number of rows(cells) with positive ccores by the number of total 
# rows(cells) to get the percentage of cells with positive scores.
percentage_inf <- dim((hivmy2_inf_sign@meta.data["Cluster1"] %>% 
                         filter(Cluster1 > 0)))[1] / 
  dim(hivmy2_inf_sign@meta.data["Cluster1"])[1]

# show the positive scores.
hivmy2_inf_sign@meta.data["Cluster1"] %>% 
  filter(Cluster1 > 0) %>% 
  rename("Score" = "Cluster1") %>% 
  kable()
```

11% of the Myeloid2 cells have a positive score for the Inf+EFZ gene
signature (17 out of 145 cells have a positive score).

On the other hand, only 4% of the Myeloid2 cells have a positive score 
for the Uninf+EFZ gene signature, while 10% of the Myeloid2 cells have a 
positive score for the Uninf gene signature (without EFZ treamtent).

```{r}
# calculate percentage for the Uninf+EFZ signature
percentage_uninf_art <- dim((hivmy2_uninf_efz_sign@meta.data["Cluster1"] %>% 
                         filter(Cluster1 > 0)))[1] / 
  dim(hivmy2_uninf_efz_sign@meta.data["Cluster1"])[1]

# calculate percentage for the Uninf signature
percentage_uninf <- dim((hivmy2_uninf_sign@meta.data["Cluster1"] %>% 
                         filter(Cluster1 > 0)))[1] / 
  dim(hivmy2_uninf_sign@meta.data["Cluster1"])[1]
```

```{r, fig.width=15}
# read in Seurat object contianing alls samples and conditions
#hiv7 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds")

# isolate the Myeloid5 cluster
#hivmy5 <- subset(hiv7, cell_type == "Myeloid5")

# process Myeloid 5 Seurat object
#hivmy5 <- ScaleData(hivmy5)
#hivmy5 <- NormalizeData(hivmy5)
#hivmy5 <- ScaleData(hivmy5)
#hivmy5 <- FindVariableFeatures(hivmy5)
#hivmy5 <- RunPCA(hivmy5, features = VariableFeatures(hivmy5))
#ElbowPlot(hivmy5)
#hivmy5 <- RunTSNE(hivmy5, dims= 1:10)

hivmy5_sign <- AddModuleScore(hivmy5,
                           features = inf_art_signature_filt)

p1 <- FeaturePlot(hivmy5_sign, features = "Cluster1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Inf+EFZ gene signature from iMg applied to CSF Myeloid5 cells")


hivmy5_sign <- AddModuleScore(hivmy5,
                           features = uninf_art_signature_filt)

p2 <- FeaturePlot(hivmy5_sign, features = "Cluster1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Uninf+EFZ gene signature from iMg applied to CSF Myeloid5 cells")

figure <- ggarrange(p1, p2, ncol = 2)#, nrow = 2) 
annotate_figure(figure, top = text_grob("Myeloid5", 
               color = "black", face = "bold", size = 20))

```

## Overlay Inf+EFZ signature with the TF activity heatmap

```{r}
# extract the Myeloid2 cells which have a positive score for the Inf+EFZ signature
inf_efz_cells <- rownames(hivmy2_inf_sign@meta.data %>% filter(Cluster1 > 0))
uninf_efz_cells <- rownames(hivmy2_uninf_efz_sign@meta.data %>% filter(Cluster1 > 0))
uninf_cells <- rownames(hivmy2_uninf_sign@meta.data %>% filter(Cluster1 > 0))
```


```{r,fig.width=15, fig.height=20}
# import TF activity matrix of  Myeloid2 cells
auc_mtx <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/aucell_matrix_10k_hvg.tsv", sep = "\t")


# 3 heatmaps with all Scenic TFs
ht_list <- map(seq.int(1:3), function(n) {
  patients <- c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF") # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df),]
  # heatmap
  ht <- Heatmap(t(mtx), 
               column_title = paste0("patient: ", patients[n]), 
               top_annotation = HeatmapAnnotation(signature = case_when(
                 rownames(mtx) %in% inf_efz_cells ~ "Inf_EFZ_signature",
                 rownames(mtx) %in% uninf_efz_cells ~ "Uninf_EFZ_signature",
                 rownames(mtx) %in% uninf_cells ~ "Uninf_signature",
                 TRUE ~ "no_signature"),
                 name = "signature",
                 col = list(signature = c("Inf_EFZ_signature" = "pink",
                                          "Uninf_EFZ_signature" = "lightblue",
                                          "Uninf_signature" = "orange",
                                          "no_signature" = "darkgreen"))),
               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
               show_row_names = TRUE
               ) 
               #top_annotation = column_annotation) 
  list(name = patients[n], heatmap = ht)
})

ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap 
```
When you group the cells according to their different gene signatures they 
express, you do not really see a difference between the groups. 

```{r, fig.width=15, fig.height=20}
# filter the dataset for only Myeloid2 cells 
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                orig.ident %in% c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df),]

# add signature column to dataframe
signature <- df %>%
  rownames_to_column("cell") %>% 
  mutate(signature = case_when(cell %in% inf_efz_cells ~ "Inf_signature",
                                cell %in% uninf_efz_cells ~"Uninf_EFZ_signature",
                                cell %in% uninf_cells ~ "Uninf_signature",
                               TRUE ~ "no_signature"))


ht <- draw(Heatmap(t(mtx), 
             column_title = "All patients split according to gene signature scores", 
             top_annotation = HeatmapAnnotation(signature = case_when(
               rownames(mtx) %in% inf_efz_cells ~ "Inf_signature",
               rownames(mtx) %in% uninf_efz_cells ~ "Uninf_EFZ_signature",
               rownames(mtx) %in% uninf_cells ~ "Uninf_signature",
               TRUE ~ "no_signature"),
               name = "signature",
               col = list(signature = c("Inf_signature" = "pink",
                                        "Uninf_EFZ_signature" = "lightblue",
                                        "Uninf_signature" = "orange",
                                        "no_signature" = "darkgreen"))),       
             column_split = signature$signature,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             show_row_names = TRUE
             ))
```

## Are some TFs differentially active between the infection conditions?

Using a Kruskal-Wallis Test to see if there was a significant different in TF 
activity between the 3 signatures Inf+EFZ, Uninf+EFZ, Uninf I was not able to
find any TF which showed differential activity in any of the clusters.


```{r}
hivmy2_scenic <- AddMetaData(hivmy2, 
                    metadata = auc_mtx, 
                    col.name = colnames(auc_mtx))

df <- hivmy2_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
         filter(cell %in% c(inf_efz_cells, uninf_efz_cells, uninf_cells)) %>% 
  mutate(signature = case_when(cell %in% inf_efz_cells ~ "Inf_efz_signature",
                                cell %in% uninf_efz_cells ~ "Uninf_efz_signature",
                                cell %in% uninf_cells ~ "Uninf_signature"))

# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(df[tf]) ~ signature, df))["p.value"]
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")

# how many transcription factors are significantly different between the clusters?
sign <- adj_p_values[adj_p_values < 0.05]
```


