---
title: "Triculture dataset"
author: "kmikulik"
date: "9 11 2021"
bibliography: references.bib
link-citations: yes
---

<style>
body {
text-align: justify}
</style>


```{r setup}
knitr::opts_chunk$set(echo = TRUE, collapse = TRUE, message = FALSE,
                      cache = TRUE, cache.lazy = FALSE, autodep = TRUE)
set.seed(42) # TODO: check with default seed
```



```{r}
library(tidyverse)
library(Seurat)
library(edgeR)
library(Matrix)
library(data.table)
library(ggrepel)
#library(harmony)
library(RColorBrewer)
library(pheatmap)
library(ggcorrplot)
library(ComplexHeatmap)
library(circlize)
library(ggpubr)
library(gridExtra)
library(corrplot)
library(rstatix)
library(knitr)
library(metap)
library(multtest)
library(readxl)
```

# Triculture Dataset

The triculture dataset used was published by [@Ryan]. 
They used a Triculture system with induced neurons, induced astrocytes and 
induced microglia.
Induced microglia express marker genes CX3CR1, IBA1, TMEM119 and P2RY12.
CCR5 which is a co-receptor necessary for HIV infection is also expressed by 
these cells, but they lack expression of myeloid-progenitor markers [@Ryan].

iMg were infected with HIV for 15 days (resulting in 94% of cells infected, 
positive for HIV capsid protein p24 and exhibiting multinucleation). Some cells
were treated with the ART EFZ. EFZ blocks HIV reverse transcription. This way it was possible to study non-productively infected cells. EFZ reduced infection by two 
thirds [@Ryan].

In the infected, but not treated iMg (Inf) inflammatory pathways were 
significantly uperegulated compared to the uninfected condition(Uninf), including
IL-8, NFkB and EIF2. 
There was decreased expression of pro-imflammatory genes IL1b, Il8 and TNFalpha
in the infected and treated (Inf+EFZ) vs Inf condition. This suggests a reduced inflammatory reaction [@Ryan]. Nevertheless, there were
two distinct pathways which were upregulated in 
Inf+EFZ iMg, namely RhoGDI signaling and CD40 signaling pathway. This will be 
discussed in more detail later on [@Ryan].

There was  a change in signaling pathways caused by EFZ treatment even in 
the uninfected and treated (Uninf+EFZ) versus the uninfected, but not treated
(Uninf) condition. The EFZ treatment alone was very different from the EFZ treatment with infection [@Ryan]. 

 
<details>
<summary>**Files and data structure**</summary>
  
**Read in Files:**

abbreviations: 

* ia/2 = infected with HIV + treated with ART
* ca = uninfected + ART
* ctl = uninfected
* inf = infected 

file structure: 

* the ".mtx" files contains a sparse matrix of class dgTMatrix
* the "barcodes.tsv" files contain the cell barcodes
* the "features.tsv" files contain the feature names as well as ensemble gene ids 
and "gene expression" strings in a third column


There are duplicated features in the data. How do you deal with this problem?
(probably, because there are multiple ensemble gene IDs mapping to the same
gene symbol -> the same gene symbol has multiple expression values)

* You could make them unique with the function `make.unique()`
* You could remove the duplicated features -> this is what I did

```
path <- "/media/ag-cherrmann/kmikulik/HIV_microglia/data/GSE143686_RAW/" #"/media/ag-cherrmann/kmikulik/HIV_microglia/data/GSE143686_RAW/"
all_files <- list.files(path)

all_cond_list <- map(seq.int(3,7), function(n){
  file <- all_files[grepl(paste0("GSM427258", n), all_files)]
  name <- str_extract(file[[1]],"ca|ctl|ia2|ia|inf")
  barcodes <- read.table(paste0(path,file[[1]]), sep = "\t") 
  features <- read.table(paste0(path, file[[2]]), sep = "\t")
  matrix <- readMM(paste0(path, file[[3]]))
  colnames(matrix) <- barcodes$V1
  rownames(matrix) <- features$V2
  matrix <- matrix[!duplicated(rownames(matrix)),]
  seurat_ob <- CreateSeuratObject(matrix, project = name,
                                  assay = "RNA", min.cells = 1, min.features = 1)
  seurat_ob[["percent_mt"]] <- PercentageFeatureSet(seurat_ob, pattern = "^MT-")
  list(name = name, count = matrix, seurat_ob = seurat_ob)
})

#saveRDS(all_cond_list, "/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/seurat_list")
```

---
</details>

# Data processing {.tabset}

## Before Quality Control:


```{r, fig.width=15}
all_cond_list <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/seurat_list")

before_qc <- merge(all_cond_list[[1]]$seurat_ob,
                    c(all_cond_list[[2]]$seurat_ob,
                      all_cond_list[[3]]$seurat_ob,
                      all_cond_list[[4]]$seurat_ob,
                      all_cond_list[[5]]$seurat_ob),
                    project = "triculture_all_conditions",
                    add.cell.ids = c(paste0(all_cond_list[[1]]$name),
                                     paste0(all_cond_list[[2]]$name),
                                     paste0(all_cond_list[[3]]$name),
                                     paste0(all_cond_list[[4]]$name),
                                     paste0(all_cond_list[[5]]$name)))
```


```{r, fig.width=15, fig.height=7}
qual_metric <- c("nFeature_RNA", "nCount_RNA", "percent_mt")

plots <- map(qual_metric, function(metric){
  ggplot() +
    geom_violin(aes(x = before_qc@meta.data %>% pull("orig.ident"),
                 y = before_qc@meta.data %>% pull(metric), 
                 fill = before_qc@meta.data %>% pull("orig.ident")), 
                alpha = .5) +
    geom_boxplot(aes(x = before_qc@meta.data %>% pull("orig.ident"),
                     y = before_qc@meta.data %>% pull(metric)),
                 alpha = 0) +
    xlab("sample origin") +
    ylab(paste0(metric)) +
    NoLegend() 
})

do.call(grid.arrange, c(plots, ncol = 3, nrow = 1))
```

## After quality control

For each sample separately

* remove genes expressing less than 200 genes or more than 5000
* remove cells expressing more than 20% mitochondrial genes
* merge Seurat objects after Quality Control

```{r}
# add percentage of mitochondiral genes to metadata
all_cond <- map(seq.int(1,5), function(n){
  seurat_ob <- all_cond_list[[n]]$seurat_ob
  #seurat_ob[["percent_mt"]] <- PercentageFeatureSet(seurat_ob, pattern = "^MT-")
  #plot <- VlnPlot(seurat_ob,
   #               features = c("nFeature_RNA", "percent_mt", "nCount_RNA"),
    #              ncol = 3) +
    #labs(title = paste0("condition ", all_cond_list[[n]]$name))
  seurat_ob <- subset(seurat_ob, nFeature_RNA > 200 & nFeature_RNA < 5000 & percent_mt < 20)
  list(name = all_cond_list[[n]]$name, seurat_ob = seurat_ob)#, plot = plot)
})
```


```{r, fig.width=15}
triculture <- merge(all_cond[[1]]$seurat_ob,
                    c(all_cond[[2]]$seurat_ob,
                      all_cond[[3]]$seurat_ob,
                      all_cond[[4]]$seurat_ob,
                      all_cond[[5]]$seurat_ob),
                    project = "triculture_all_conditions",
                    add.cell.ids = c(paste0(all_cond[[1]]$name),
                                     paste0(all_cond[[2]]$name),
                                     paste0(all_cond[[3]]$name),
                                     paste0(all_cond[[4]]$name),
                                     paste0(all_cond[[5]]$name)))
```


```{r, fig.width=15, fig.height=7}
qual_metric <- c("nFeature_RNA", "nCount_RNA", "percent_mt")

plots <- map(qual_metric, function(metric){
  ggplot() +
    geom_violin(aes(x = triculture@meta.data %>% pull("orig.ident"),
                 y = triculture@meta.data %>% pull(metric), 
                 fill = triculture@meta.data %>% pull("orig.ident")), 
                alpha = .5) +
    geom_boxplot(aes(x = triculture@meta.data %>% pull("orig.ident"),
                     y = triculture@meta.data %>% pull(metric)),
                 alpha = 0) +
    xlab("sample origin") +
    ylab(paste0(metric)) +
    NoLegend() 
})

do.call(grid.arrange, c(plots, ncol = 3, nrow = 1))
```

## Normalization, Scaling, Find Variable Features

* log normalization
* scale data so that we have mean = 0 and variance = 1
* Find the 2000 most highly variable features

```
triculture <- NormalizeData(triculture, verbose = FALSE)
triculture <- ScaleData(triculture, verbose = FALSE)
triculture <- FindVariableFeatures(triculture, verbose = FALSE)
```


## Dimensionality reduction

* compute PCA
* use elbow plot to determine dimensions that contribute to variability
* find the k-nearest neighbors
* find Clusters based on KNN-graph
* calculate TSNE embedding

```
triculture <- RunPCA(triculture, features = VariableFeatures(triculture), verbose = FALSE)
ElbowPlot(triculture)

triculture <- FindNeighbors(triculture, verbose = FALSE)
triculture <- FindClusters(triculture, verbose = FALSE)

triculture <- RunTSNE(triculture, dims = 1:10, verbose = FALSE)
```

```{r}
triculture <- readRDS("seurat_objects/triculture_seurat_object")

#triculture <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_seurat_object")

DimPlot(triculture, reduction = "tsne", pt.size = .01, label = TRUE,
        group.by = "seurat_clusters")
```

## Cell type annotation

For annotating the different clusters the same markers as in [@Ryan] were used:

* *Microglia:* AIF1, SPI1, CD4
* *Astrocytes:* SOX9, THBS1
* *Neurons:* SYN1, MAP2

```{r, fig.widht = 15, fig.height=12}
#markers <- FindAllMarkers(triculture)
#triculture <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_seurat_object")
markers <- read.table("data_files/marker_genes_triculture_dataset")
#markers <- read.table("/media/ag-cherrmann/kmikulik/HIV_microglia/data/marker_genes_triculture_dataset")

top_2 <- markers %>% group_by("seurat_cluster") %>% top_n(n = 2, wt = avg_log2FC)

marker_genes <- c("AIF1", "SPI1", "CD4", "MAP2", "SYN1", "THBS1", "SOX9")
p1 <- FeaturePlot(triculture, features = "AIF1")
p2 <- FeaturePlot(triculture, features = "SPI1")
p3 <- FeaturePlot(triculture, features = "CD4")
p4 <- FeaturePlot(triculture, features = "MAP2")
p5 <- FeaturePlot(triculture, features = "SYN1")
p6 <- FeaturePlot(triculture, features = "THBS1")
p7 <- FeaturePlot(triculture, features = "SOX9")

ggarrange(p1,p2,p3,p4,p5,p6,p7, ncol = 2, nrow =4 )

```



<details>
<summary>**Details on cell type annotation**</summary>

**Gene expression**

Different clusters were annotated according to the gene expression of 
marker genes mentioned above.

```{r, fig.width=15}

VlnPlot(triculture, features = c("CD4", "SPI1", "AIF1"), group.by = "seurat_clusters")#, group.by = seurat_cluster)
VlnPlot(triculture, features = c("MAP2", "SYN1"), group.by = "seurat_clusters")
VlnPlot(triculture, features = c("THBS1", "SOX9"), group.by = "seurat_clusters")
```

```{r}
# 4, 9, 11, 18 = Microglia
# 0, 1, 2, 3,8, 10, 13, 15, 18, 19 = Neurons 
# 6, 7, 12, 14, 16 = Astrocytes
# 5, 17 are undesginated
cell_types <- (triculture@meta.data %>% 
  mutate(cell_type = ifelse(seurat_clusters %in% c(4,9,11), "iMg", 
                            ifelse(seurat_clusters %in% c(6,7,12,14,16), "iAst", 
                                   ifelse(seurat_clusters %in% 
                                            c(0,1,2,3,8,10,13,15,18,19), 
                                          "iNeurons", "Undesignated")))))[["cell_type"]]
  
triculture <- AddMetaData(triculture, cell_types, col.name = "cell_type")

```

---
</details>

# {-}


### Remove undesignated cells

There is one group of cells where it is not clear which cell state they 
represent (Undesignated), we will exclude them from further analysis as [@Ryan] 
did.


```{r}
p1 <- DimPlot(triculture, reduction = "tsne", pt.size = .01, label = TRUE,
        group.by = "cell_type")
```


```{r}
triculture_clean <- subset(triculture, cell_type %in% c("iMg", "iNeurons", "iAst")) 
```

```{r}
p2 <- DimPlot(triculture_clean, reduction = "tsne", pt.size = .01, label = TRUE,
        group.by = "cell_type")
```

```{r, fig.width=15}
p1+p2
```


<details>
<summary>**Heatmap of top 20 marker genes for each cell type**</summary>


Top20 marker genes for each cell cell type: 
 
```{r, fig.width=15, fig.height=25}
Idents(triculture_clean) <- "cell_type"
top_n <- markers %>% group_by (cluster) %>% top_n(n = 20, wt = avg_log2FC)
DoHeatmap(triculture_clean, features = top_n$gene)
```


---
</details>


<details>
<summary>**Number of cells per condition and cell type**</summary>


How many cells of each cell type and condition are present in the dataset?

```{r, results = "asis"}
triculture_clean@meta.data %>% group_by(cell_type, orig.ident) %>% 
  summarize(n = n()) %>% 
  mutate(orig.ident = case_when(
                      orig.ident %in% c("ia", "ia2") ~ "Inf+EFZ", 
                      orig.ident == "ca" ~ "Uninf",
                      orig.ident == "ctl" ~ "Uninf+EFZ",
                      orig.ident == "inf" ~ "Inf")) %>% 
  pivot_wider(names_from = orig.ident, values_from = n) %>% 
  kable(caption = "Number of cells per condition and cell type") # to make table look nice in Rmarkdown
  
```


---
</details>


## Different infection and treatment conditions

Interestingly, the infected cells treated with EFZ (Inf+EFZ) clearly separate 
from the infected, but untreated cells (Inf) and the the uninfected cells
(Uninf, Uninf+EFZ) in the microglia cluster as can be seen in the figure below.


```{r}
# rename origin of ident for easier interpretation
origin_vector <- (triculture_clean@meta.data %>% 
                    mutate(orig.ident = case_when(
                      orig.ident %in% c("ia", "ia2") ~ "Inf+EFZ", 
                      orig.ident == "ca" ~ "Uninf",
                      orig.ident == "ctl" ~ "Uninf+EFZ",
                      orig.ident == "inf" ~ "Inf")))[["orig.ident"]]

triculture_clean <- AddMetaData(triculture_clean, origin_vector, col.name = "orig.ident")
```

```{r}
DimPlot(triculture_clean, reduction = "tsne", group.by = "orig.ident", pt.size = .01, )
```


## Inflammatory gene expression in iMg

The largest change in gene expression related to inflammation was found in iMg
among the 3 cell types [@Ryan]. Shown are only inflammatory genes. 

```{r, fig.height = 12}
inflamm_genes <- c("TNF", "ACSL1", "MGST3", "B2M", "IFNGR2", "GRINA", "TLR2", 
                   "SOD2", "GLUL", "CFLAR", "CTNNB1", "CXCL8", "CCL4", "IL1B",
                   "ATF4", "CXCL8", "CCL4", "CCL2", "IL1B", "ATF4", "CXCL1", 
                   "NFKB1", "TANK", "APOE", "TREM2", "LY96", "SRC", "NCF1", 
                   "HCK", "LYN", "VASP", "C3", "C5", "RAP1B", "RAP2B", "MAF",
                   "JUND", "SQSTM1", "FTH1", "RHOQ", "CD36", "FOS", "NFKBIA",
                   "ICAM1", "TNFAIP3", "LYZ", "PPP1R12A", "RND3", "CYBA", "IRF8",
                   "SLC25A6")

DoHeatmap(triculture_clean, features = inflamm_genes)
```


<details>
<summary>**Differences of inflammatory gene expression between conditions**</summary>


The gene expression of inflammatory genes in microglia was not very different
between the conditions. The Inf+EFZ cells show a slightly different pattern of
gene expression.

```{r, fig.height=12}
DoHeatmap(subset(triculture_clean, cell_type == "iMg"), 
          features = inflamm_genes, group.by = "orig.ident")
```


---
</details>


# Integration of triculture dataset with dataset of blood & CSF samples from HIV-infected patients

The HIV Blood & CSF dataset [@Farhadian]
used in the previous analysis was integrated with the triculture dataset [@Ryan].

Integration of two datasets is based on Seurat [@Stuart]. It first identifies 
anchors between datasets, calculates scores for all anchors and 
then calculates an integrated expression matrix which can be used for downstream
analysis.


1. Identify features which are variable across datasets
2. Identify anchors between the two datasets and create an integrated data assay
3. Run a single integrated analysis on all cells


<details>
<summary>**Details on integration**</summary>

For more details see: [@Stuart]

1. Identify features which are variable across datasets
2. Identify anchors between the two datasets and create an integrated data assay
3. Run a single integrated analysis on all cells

```
hiv7 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds")
hiv7 <- NormalizeData(hiv7, verbose = FALSE)
hiv7 <- FindVariableFeatures(hiv7, verbose = FALSE)

# add both datasets to a lits
data_list <- list(hiv = hiv7, triculture = triculture_clean)

# select features that are repeatedly variable across datasets for integration
features <- SelectIntegrationFeatures(object.list = data_list, verbose = FALSE)

# identify integration anchors and perform integration
anchors <- FindIntegrationAnchors(object.list = data_list, 
                                  anchor.features = features, verbose = FALSE) 
combined <- IntegrateData(anchorset = anchors, verbose = FALSE)

# run analysis on integrated dataset
combined <- ScaleData(combined, verbose = FALSE)
combined <- RunPCA(combined, verbose = FALSE)
#ElbowPlot(combined)
combined <- RunTSNE(combined, reduction = "pca", dims = 1:10, verbose = FALSE)
combined <- FindNeighbors(combined, reduction = "pca", dims = 1:10, verbose = FALSE)
combined <- FindClusters(combined, resolution = .3, verbose = FALSE)

# add annotations from which dataset the sample came from
dataset_vector <- (combined@meta.data %>% mutate(dataset = ifelse(orig.ident %in%
                                                 c("HIV1_CSF", "HIV2_CSF",
                                                   "HIV3_CSF", "HIV1_Bld", 
                                                   "HIV2_Bld", "Uninfected1_CSF",
                                                   "Uninfected2_CSF"),
                                               "csf", "triculture")))[["dataset"]]

combined <- AddMetaData(combined, dataset_vector, col.name = "dataset")

# add information on the cell groups in the combined datasets
combined_clusters <- (combined@meta.data %>% 
  mutate(combined_clusters = ifelse(seurat_clusters %in% c(2,10), "Microglia", 
                            ifelse(seurat_clusters %in% c(3,9), "iAst", 
                                   ifelse(seurat_clusters %in% c(0,1,7,11),
                                          "iNeurons", 
                                          ifelse(seurat_clusters %in% c(8,5,6), "Immune Cells", "Undesignated"))))))[["combined_clusters"]]
  
combined <- AddMetaData(combined, combined_clusters, col.name = "combined_clusters")

#saveRDS(combined,"/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/triculture_hiv_combined_seurat_object")

```

---
</details>

## Integrated clusters

When integrating the CSF dataset and the Triculture dataset the iMg cells 
clustered together with the Myeloid cells from the CSF and Blood samples from 
the HIV-1 infected/uninfected patients. This was expected since they are both
myeloid-lineage cells. However, this makes it more difficult to draw conclusions
on which of the five Myeloid clusters is the most microglia-like. Therefore, we 
had a closer look at this myeloid-lineage cluster consisting of iMg from
triculture and Myeloid cells from CSF and Blood samples.


#### Visualization of the integrated dataset

```{r, fig.width=15, fig.height=15}
combined = readRDS("seurat_objects/triculture_hiv_combined_seurat_object_2")

p1 <- DimPlot(combined, reduction = "tsne", group.by  = "dataset",
              label = TRUE) + NoLegend() +
  labs(title = "Dataset")
p2 <- DimPlot(combined, reduction = "tsne", group.by = "cell_type",
              label = TRUE, repel = TRUE) + NoLegend() +
  labs(title = "Cell types")
p3 <- DimPlot(combined, reduction = "tsne", group.by = "cell_type",
              split.by = "dataset", label = TRUE, repel = TRUE) + NoLegend() +
  labs(title = "Cell types split into datasets")
p4 <- DimPlot(combined, reduction = "tsne", group.by = "seurat_clusters",
              label = TRUE) + NoLegend() +
  labs(title = "Clusters")

a1 <- DimPlot(combined, reduction = "tsne", split.by = "dataset", 
        group.by = "seurat_clusters", label = TRUE, repel =TRUE) +
  NoLegend() +
  labs(title = "Clusters split into datasets")

a2 <- DimPlot(combined, reduction = "tsne", split.by = "dataset", 
              group.by = "orig.ident") +
  labs(title = "Sample origin split into datasets")

ggarrange(p1, p2, p3,p4, a1, a2, ncol = 2, nrow = 3)

```

```#{r, fig.width=15, fig.height=5}

a1 <- DimPlot(combined, reduction = "tsne", split.by = "dataset", 
        group.by = "seurat_clusters", label = TRUE, repel =TRUE) +
  NoLegend() +
  labs(title = "Clusters split into datasets")

a2 <- DimPlot(combined, reduction = "tsne", split.by = "dataset", 
              group.by = "orig.ident") +
  labs(title = "Sample origin split into datasets")
ggarrange(a1, a2, ncol = 2, nrow = 1)
```


## Myeloid lineage cluster

#### Visualization of Myeloid lineage cluster (iMg, Myeloid1/2/3/4/5)

The plot below visualizes that iMg and CSF/Blood Myeloid cells cluster together,
independent of their infection status.

```{r, fig.width=15, fig.height=7}
subset_monocytes <- subset(combined, cell_type %in% c("Myeloid1", 
                                                            "Myeloid2", 
                                                            "Myeloid3", 
                                                            "Myeloid4", 
                                                            "Myeloid5", 
                                                            "iMg"))

p1 <- DimPlot(subset_monocytes, group.by = "cell_type") +
  labs(title = "Cell type")
p2 <- DimPlot(subset_monocytes, group.by = "orig.ident") +
  labs(title = "Sample origin")
p3 <- DimPlot(subset_monocytes, group.by = "seurat_clusters") +
  labs(title = "Cluster")
fig <- ggarrange(p1, p2, p3, ncol = 3, nrow = 1)
annotate_figure(fig, top = text_grob("Myeloid-lineage cells", size = 20))
```

As expected the CSF/Blood Myeloid cells cluster together with Inf+EFZ and mainly contribute to the same subcluster, as can 
be seen in the plot below, which shows only CSF/Blood Myeloid cells and the 
Inf+EFZ iMg. Blood Myeloid 4 is an exception and belongs to a different subcluster.

```{r, fig.width=15, fig.height=7}

subset_infected <- subset(combined, orig.ident %in% c("HIV1_CSF", "HIV2_CSF", "HIV3_CSF",
                                             "HIV1_Bld", "HIV2_Bld","Inf+EFZ") &
                            cell_type %in% c("Myeloid1", 
                                                            "Myeloid2", 
                                                            "Myeloid3", 
                                                            "Myeloid4", 
                                                            "Myeloid5", 
                                                            "iMg"))


p1 <- DimPlot(subset_infected, group.by = "orig.ident")
p2 <- DimPlot(subset_infected, group.by = "orig.ident")
p3 <- DimPlot(subset_infected, group.by = "seurat_clusters")
fig <- ggarrange(p1, p2, p3, ncol = 3, nrow = 1)
annotate_figure(fig, top = text_grob("Infected CSF/Blood Myeloid cells & Inf+EFZ iMg"))
```



#### Cell types and conditions partially separate according to the two iMg clusters

The myeloid-lineage cluster separated into two distinct subclusters. 
Interestingly, the infected + EFZ iMg separated clearly from the infected iMg. 
The separation was not so clear for the uninfected cells, which contributed to 
both subclusters. Another observations is that the infected Myeloid1/2/3/5 cells belong  
to the same subcluster as the infected + EFZ iMg. Conversely, the Myeloid4 cells
cluster with the infected iMg. Since the infected CSF/Blood Myeloid cells originate from HIV 
infected and treated patients we would expect them to cluster togehter with the
infected + EFZ iMg. However, there might also be infected CSF/Blood Myeloid cells which are not 
affected by the treatment. The Myeloid4 cells which are from Blood samples might
consitute an infected cell population unaffected by the ART treatment. 



```{r,fig.width=15, fig.height=10}

DimPlot(subset_monocytes, group.by = "seurat_clusters", 
        split.by = "orig.ident", ncol = 4) +
  labs(title = "Contribution of different samples to different subclusters")
DimPlot(subset_monocytes, group.by = "seurat_clusters", 
        split.by = "cell_type", ncol = 3) +
  labs(title = "Contribution of different cell types to different subclusters")
```


## Subcluster the Myeloid-Lineage Cluster of the integrated dataset

Isolating the iMg and all Myeloid cell clusters from the integrated dataset and
normalizing, scaling and re-clustering these isolated cells might give us
additional insights. 



<details>
<summary>**Details on de-novo clustering**</summary>


```

microglia <- subset(combined, cell_type %in% c("iMg",
                                                "Myeloid1",
                                                "Myeloid2",
                                                "Myeloid3",
                                                "Myeloid4",
                                                "Myeloid5"))
microglia_recluster <- SCTransform(microglia, verbose = FALSE)
microglia_recluster <- ScaleData(microglia_recluster, verbose = FALSE)
microglia_recluster <- RunPCA(microglia_recluster, verbose = FALSE)
ElbowPlot(microglia_recluster)
microglia_recluster <- RunTSNE(microglia_recluster, reduction = "pca", dims = 1:10)
microglia_recluster <- FindNeighbors(microglia_recluster)
microglia_recluster <- FindClusters(microglia_recluster, resolution = 0.5)
```

---
</details> 


It is no surprise that cells from the triculture dataset separate from the 
CSF/Blood-derived cells which can be seen in the plot below. When we recluster the myeloid lineage derived cells, we 
obtain five subclusters for the iMg and three subclusters for the five CSF/Blood Myeloid clusters as can be seen in the plots below. We can also observe that the Myeloid2 and Myeloid5 separate from Myeloid1/2/3 
which is expected, since Myeloid2/5 originate from CSF samples and 
Myeloid1/3/4 originate from blood samples.

```{r, fig.width=15, fig.height=12}
microglia_recluster <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Triculture_dataset_Ryan_paper/microglia_myeloid_seurat_object_reclustering")
a1 <- DimPlot(microglia_recluster, reduction = "tsne",
              group.by = "seurat_clusters", pt.size = .4) +
  labs(title = "Clustering")
a2 <- DimPlot(microglia_recluster, reduction ="tsne", 
              group.by = "cell_type", pt.size = .4) +
  labs(title = "Cell type")
a3 <- DimPlot(microglia_recluster, reduction ="tsne",
              group.by = "dataset", pt.size = .4) +
    labs(title = "Dataset")
a4 <- DimPlot(microglia_recluster, reduction = "tsne",
              group.by = "orig.ident", pt.size = .4) +
  labs(title = "Sample origin")

ggarrange(a1, a2, a3, a4, ncol = 2, nrow = 2)
```


Interestingly, both the Myeloid5 cells and the Myeloid2 cells are split between 
the cluster six (blue) and seven (purple) in the figure below. In the following we will have a 
closer look at this segregation.  
When performing GRN inference with Scenic we also identified two potentially 
biologically relevant subclusters of the Myeloid2 cells. It could be that these
independently found clusters correlate. 

```{r, fig.width=15, fig.height=10}
a5 <- DimPlot(microglia_recluster, reduction ="tsne", 
              split.by = "cell_type", pt.size = .4, ncol=3)
a5
```

It looks like the two clusters identified this way correspond to different
sample origins. CSF cells from patient1 (HIV1_CSF) and patient2 (HIV2_CSF)
belong to cluster six,while CSF cells from patient3 (HIV3_CSF) and the two 
uninfected patients (Uninfected1_CSF, Uninfected2_CSF) belong to 
cluster seven. The separation of patient3 (HIV3_CSF) from patient1 (HIV1_CSF) 
and patient2 (HIV2_CSF) might correspond to a batch effect, or it might be a 
biological effect, because the infection affected the cells differently.


```{r, fig.width=15, fig.height=7}
#a1 <- DimPlot(subset(microglia_recluster, cell_type == "iMg"), reduction = "tsne",
           #   group.by = "orig.ident", pt.size = .4)
a2 <- DimPlot(subset(microglia_recluster, cell_type != "iMg"), reduction = "tsne",
              group.by = "orig.ident", pt.size = .4) +
  labs(title = "Sample origin")
#a3 <- DimPlot(subset(microglia_recluster, cell_type == "iMg"), reduction = "tsne",
              #group.by = "seurat_clusters", pt.size = .4)
a4 <- DimPlot(subset(microglia_recluster, cell_type != "iMg"), reduction = "tsne",
              group.by = "seurat_clusters", pt.size = .4) +
  labs(title = "Cluster")
fig <- ggarrange(a2, a4, ncol = 2, nrow = 1)
annotate_figure(fig, top = text_grob("CSF/Blood Myeloid cells"))
```

Since the segregation into two clusters seems to be related to the sample origin
rather than to the infection status, we would not expect the clusters to overlap
with the clusters identified after GRN inference with SCENIC. In the figure 
below this is exactly what can be observed, only the Myeloid2 cells from infected CSF samples (HIV1_CSF, HIV2_CSF, HIV3_CSF) are shown. 

```{r, fig.width=15, figh.height = 5}
# read in Myeloid2 cells
hivmy2 <- readRDS("data_files/hiv_myeloid2_subclusters")

my2 <- subset(microglia_recluster, cell_type == "Myeloid2" & 
                orig.ident %in% c("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))

my2 <- AddMetaData(my2, hivmy2@meta.data["ht_cluster"], 
                   col.name = "TF_activity_clusters")
p1 <- DimPlot(my2, group.by = "seurat_clusters") +
  labs(title="Clusters")
p2 <- DimPlot(my2, group.by = "TF_activity_clusters") +
  labs(title = "TF activity")
p3 <- DimPlot(my2, group.by = "orig.ident") +
  labs(title = "Sample origin")
figure <- ggarrange(p1, p2, p3, ncol = 3, nrow = 1)
annotate_figure(figure, 
                top = text_grob("Comparison with clusters with different TF acitivty", 
               color = "black", face = "bold", size = 20))
#my25 <- subset(microglia, seurat_clusters %in% c(6,7) &cell_type %in% c("Myeloid2", "Myeloid5"))

```


For this reason the reclustering did not lead to any useful insights and will 
not be analyzed further. 



# Marker genes between different conditions

### Differential gene expression between infected and uninfected +/- EFZ iMg

In the following I will compare gene expression between infected cells and 
uninfected cells, both treated with EFZ and not treated. I will use the uninfected cells as a 
baseline to find genes which are upregulated or downregulated in infected cells
compared to uninfected cells. Since we have both cells treated with EFZ and 
cells not treated with EFZ, we can define a signature for both treatment 
conditions separately and then see if there is an overlap between the gene 
signatures of infected cells +/- EFZ. Finding genes which overlap independently 
of treamtent condition increases the confidence in the results

```{r, results = "asis"}
# isolate iMg cells
microglia <- subset(combined, cell_type == "iMg")

# only iMg cells to identify potential signatures between infected vs uninfected
Idents(microglia) =  "orig.ident"

# find marker genes between Uninf+EFZ and Inf+EFZ
inf_uninf_art_marker <- FindMarkers(microglia, ident.1 = "Inf+EFZ", 
                                    ident.2 = "Uninf+EFZ", verbose = FALSE)
#upregulated genes
up_inf_art <- inf_uninf_art_marker %>% 
  arrange(desc(avg_log2FC)) %>%
  head(10) %>% 
  rownames_to_column("gene")
up_inf_art  %>% kable(caption = "Genes upregulated in Inf+EFZ vs. Uninf+EFZ")
```

```{r}
# downregulated genes
down_inf_art <- inf_uninf_art_marker %>% 
  arrange(avg_log2FC) %>% 
  head(10) %>%
  rownames_to_column("gene")
down_inf_art %>% kable(caption = "Genes downregulated in Inf+EFZ vs. Uninf+EFZ")
```


```{r}
# find marker genes between Inf and Uninf
inf_uninf_marker <- FindMarkers(microglia, ident.1 = "Inf", 
                                ident.2 = "Uninf", verbose = FALSE)

# upregulated in Inf vs Uninf, no treatment
up_inf_no <- inf_uninf_marker %>% 
  arrange(desc(avg_log2FC)) %>%
  head(20) %>% 
  rownames_to_column("gene")
#up_inf_no  %>% kable()


down_inf_no<- inf_uninf_marker %>% 
  arrange(avg_log2FC) %>% 
  head(20) %>%
  rownames_to_column("gene")
#down_inf_no%>% kable()
```

Only 3% of genes which are upregulated and only 14% of genes which are 
downregulated in infected vs. uninfected cells 
overlap between the treatment conditions. This means, that the gene expression response defining 
the infection might differ between cells with and without treatment. This
reflects the findings of [@Ryan] that the EFZ treatment lead to a distinct
immune response of iMg and distinct activation of two inflammatory pathways, 
RhoGDI and CD40 pathways as will be describe later on. Since there is hardly any
overlap we will use the genes upregulated in Inf+EFZ compared to Uninf+EFZ
in the following analysis steps, since the probably best resemble the CSF/Blood samples which originate from patients treated with ART.

```{r}
up_inf <- intersect(up_inf_art$gene, up_inf_no$gene)
down_inf <- intersect(down_inf_art$gene, down_inf_no$gene)
print(c(up_inf, down_inf))
```

```{r, fig.width=15}
p1 <- ggVennDiagram::ggVennDiagram(list(up_inf_art$gene, up_inf_no$gene),
              label_alpha = 0,
              category.names= c("Inf+EFZ", "Inf"),
              #set.size = 20,
              label = "percent",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "Genes upregulated in infected cells")

p2 <- ggVennDiagram::ggVennDiagram(list(down_inf_art$gene, down_inf_no$gene),
              label_alpha = 0,
              category.names= c("Inf+EFZ", "Inf"),
              #set.size = 20,
              label = "percent",
              label_size = 6,
              set_color = c("red", "blue")) +
  scale_fill_distiller(palette = "RdBu") +
  labs(title = "Genes downregulated in infected cells")

ggarrange(p1, p2, ncol = 2)
```


##### Volcano Plot of Infection (+ EFZ) signature

Interestingly, the genes downregulatd in Inf+EFZ vs. Uninfected cells show
a higher average log fold change with higher significance (p-values). 

```{r, fig.height=8, fig.width=10}
inf_uninf_art_marker %>% 
  rownames_to_column("gene") %>%
  filter(gene %in% c(up_inf_art$gene, down_inf_art$gene)) %>% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = case_when(gene %in% up_inf_art$gene ~ "upregulated",
                               gene %in% down_inf_art$gene ~ "downregulated",
                               TRUE ~ "NA")) %>%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  mutate(label = ifelse(gene %in% c(up_inf_art$gene, down_inf_art$gene), gene, NA)) %>% 
  ggplot(aes(x = avg_log2FC,
             y = -log10(p_val_adj),
             col = condition, 
             label = label)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-50), col = "red") +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = "Genes which are up- or downregulated in Inf+EFZ vs Uninf+EFZ")

```

### Differential gene expression between treatment conditions

Comparing Uninf+EFZ with Uninf and Inf+EFZ with Inf iMg one might be able to 
identify genes which are different after treatment of cells with EFZ. Finding 
overlapping genes between the two infection states increases the confidence in the results, because the identified genes are associated with treatment independently of infection status.

The diagram below shows, that 36% of genes are upregulated both in Inf and
Inf+EFZ iMg compared to Uninf and Uninf+EFZ iMg. These genes are probably 
associated with the EFZ treatment. The drug affects 
the iMg and changes their gene expression. This signature might be used to 
identify Myeloid2 CSF cells which are affected by the treatment. Since ART drugs 
are found at lower concentrations in CSF than in blood, probably
only a subgroup of Myeloid2 are affected by the ART drugs and it would be 
interesting to know which are affected by treatement and which are not. However, this
comparison might be far fetched, since the patients from which the CSF samples 
have been derived have not been treated with EFZ, but with a combination of 
other ART drugs which probably have a slightly different effect on cells. 

```{r}
treatment_marker_uninf <- FindMarkers(microglia, ident.1 = "Uninf+EFZ", 
                                ident.2 = "Uninf")

# find marker genes between Inf+EFZ and Inf
treatment_marker_inf <- FindMarkers(microglia, ident.1 = "Inf+EFZ", ident.2 = "Inf")

# overlap between upregulated genes
treatment_up <- intersect((treatment_marker_inf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC > 0))$gene, 
           (treatment_marker_uninf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC > 0))$gene)

treatment_down <- intersect((treatment_marker_inf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC < 0))$gene, 
           (treatment_marker_uninf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC < 0))$gene)
treatment_marker <- c(treatment_up, treatment_down)
```

```{r, fig.width=15}
p1 <- ggVennDiagram::ggVennDiagram(list((treatment_marker_inf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC > 0))$gene, 
             (treatment_marker_uninf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC > 0))$gene),
              label_alpha = 0,
              category.names= c("Inf+EFZ", "Uninf+EFZ"),
              #set.size = 20,
              label = "percent",
              label_size = 6) +
   scale_fill_distiller(palette = "RdBu") +
  labs(title = "Genes upregulated in cells treated with EFZ")

p2 <- ggVennDiagram::ggVennDiagram(list((treatment_marker_inf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC < 0))$gene, 
             (treatment_marker_uninf %>% 
             rownames_to_column("gene") %>% 
             filter(avg_log2FC < 0))$gene),
              label_alpha = 0,
              category.names= c("Inf+EFZ", "Uninf+EFZ"),
              #set.size = 20,
              label = "percent",
              label_size = 6,
              set_color = c("red", "blue")) +
  scale_fill_distiller(palette = "RdBu") +
  labs(title = "Genes downregulated in cells treated with EFZ")

ggarrange(p1, p2, ncol = 2)
```

##### Volcano Plot of Treatment signature

```{r,fig.height=8, fig.width=10}
treatment_marker_uninf %>% 
  rownames_to_column("gene") %>%
  filter(gene %in% treatment_marker) %>% 
  # add a column containing information in which condition the corresponding TF 
  # is upregulated
  mutate(condition = case_when(gene %in% treatment_up ~ "upregulated",
                               gene %in% treatment_down ~ "downregulated",
                               TRUE ~ "NA")) %>%
  # add a column conaining the labels for the plot
  # I only want to label cells which are also found by Scenic
  #mutate(label = ifelse(gene %in% c(up_inf$gene, down_inf$gene), gene, NA)) %>% 
  ggplot(aes(x = avg_log2FC,
             y = -log10(p_val_adj),
             col = condition, 
             label = gene)) +
  geom_point() +
  geom_hline(yintercept = -log10(1e-10), col = "red") +
  geom_text(nudge_x = .05, nudge_y = 2) +
  labs(title = "Top 20 Genes which are up/downregulated in treatment vs. no treatment, Uninf cells")
```



# Infected vs Uninfected gene signatures 

### Investigate gene signatures in Myeloid cells from CSF

Checking the differentially expressed genes and trying to identify a signature 
through literature search turned out to be very difficult. Therefore, 
a different approach was used. The top twenty up- and downregulated genes in 
Inf+EFZ vs Uninf+EFZ were computed. Using these genes as an "infection signature" 
we will check if some of the Myeloid2 cells from the CSF samples show a similar 
signature.
  

```{r}
# select only genes which are also found in the sc-RNAseq dataset
up_inf_filt <- up_inf_art$gene[up_inf_art$gene %in% rownames(hivmy2@assays$RNA@counts)]
down_inf_filt <- down_inf_art$gene[down_inf_art$gene %in% rownames(hivmy2@assays$RNA@counts)]
```


## Gene signature scoring

Using the signature genes from differential gene expression analysis between
Inf+EFZ and Uninf+EFZ we can compute a score of activity of these gene sets in 
the Myeloid2 cluster of CSF samples. We can plot these scores in a TSNE 
representation, but we can also get the scores from the metadata and use them 
for other visualizations 

If the Inf+EFZ signature can be found in some of the Myeloid2 cells from the CSF samples this migth provide additional insights about the infection status of these 
microglia-like cells.

#### Scores of gene signature upregulated in Inf+EFZ iMg in Myeloid2 cells

In the plot below there are some cells visible which have a high score for the signature composed of genes upregulated in Inf+EFZ microglia. To be precise,
eleven percent of the Myeloid2 cells have a positive score (17 out of 145 cells).
These high scroing cells do, however, not correspond to the two clusters (c1, c2) obtained from analysis of TF activity (SCENIC).



```{r, fig.width=15}
up_sign <- AddModuleScore(hivmy2,
                           features = up_inf_filt)

p1 <- FeaturePlot(subset(up_sign, ht_cluster == "c1"),
                  features = "Cluster1", pt.size = 1) +#, split.by = "ht_cluster") +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")

p2 <- FeaturePlot(subset(up_sign, ht_cluster == "c2"),
                  features = "Cluster1", pt.size = 1) +#, split.by = "ht_cluster") +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c2")

figure <- ggarrange(p1, p2, ncol = 2)#, nrow = 2) 
annotate_figure(figure, top = text_grob("Signature score of genes upregulated in Inf+EFZ iMg applied to CSF Myeloid2 cells", 
               color = "black", face = "bold", size = 20))

```


<details>
<summary>**Exact scores for each cell**</summary>

```{r, results = "asis"}
# divide the number of rows(cells) with positive ccores by the number of total 
# rows(cells) to get the percentage of cells with positive scores.
percentage_inf <- dim((up_sign@meta.data["Cluster1"] %>% 
                         filter(Cluster1 > 0)))[1] / 
  dim(up_sign@meta.data["Cluster1"])[1]

# show the positive scores.
up_sign@meta.data["Cluster1"] %>% 
  filter(Cluster1 > 0) %>% 
  rename("Score" = "Cluster1") %>% 
  kable()
```

---
</details>

#### Defining a signature composed of downregulated genes

Defining a gene signature composed of both up- and downregulated genes turned 
out to be more difficult than expected. One idea was to separately investigate 
upregulated versus downregulated genes. Above the upregulated gene signature 
scores in Myeloid2 cells were shown. 

We will now have a closer look at the problems that arise when investigating
downregulated genes. Assuming that our Myeloid2 cells are composed of two 
distinct populations, one infected and one uninfected, we would expect a 
bimodal distribution of gene expression for genes upregulated in the infected
condition. There should be one peak with base level gene expression corresponding
to uninfected cells and a second peak with higher levels of gene expression 
corresponding to infected cells. 

In the histograms below the distribution of the sum of log-normalized feature counts of genes upregulated or downregulated  respectively can be seen. 

For the upregulated genes (green) a bimodal distribution is not evident from 
the histogram below. No clear conclusions can be drawn, since the number of
cells is probably too low.

For the downregulated genes there seem to be a large number of cells which do 
not express these genes at all. This group of cells could include infected cells,
however this is a dangerous assumption, since in general there might be cells
with low or none expression of these genes. There is no way to know if these 
cells are actually infected cells.

```{r}
#normalize counts of upregulated genes
counts_up <- (hivmy2@assays$RNA@data)
counts_up <- t(t(counts_up)) / colSums(counts_up)
counts_up <- log1p(counts_up)
counts_up <- counts_up[up_inf_filt, ]

# compute average expression of the 20 signature genes
p1 <- as.data.frame(rowSums(t(counts_up))) %>% 
  rownames_to_column("cell") %>% 
  rename("count" = "rowSums(t(counts_up))") %>% 
  arrange(desc(count)) %>% 
  ggplot(aes(x  = count)) +
  geom_histogram(bins = 70, color = "olivedrab4", fill = "olivedrab4") +
  xlab("normalized feature counts") +
  ylab("number of cells") +
  labs(title = "Sum of feature counts of genes upregulated in Inf+EFZ vs. Uninf+EFZ")

#normalize counts of downregulated genes
counts_down <- (hivmy2@assays$RNA@data)
counts_down <- t(t(counts_down)) / colSums(counts_down)
counts_down <- log1p(counts_down)
counts_down <- counts_down[down_inf_filt, ]

# compute average expression of the 20 signature genes
p2 <- as.data.frame(rowSums(t(counts_down))) %>% 
  rownames_to_column("cell") %>% 
  rename("count" = "rowSums(t(counts_down))") %>% 
  arrange(desc(count)) %>% 
  ggplot(aes(x  = count)) +
  geom_histogram(bins = 70, color = "red4", fill = "red4") +
  xlab("normalized feature counts") +
  ylab("number of cells") +
  labs(title = "Sum of feature count of genes downregulated in Inf+EFZ vs. Uninf+EFZ")
```

```{r, fig.width=15}
p1 + p2
```


#### Percentage of cells with positive Inf gene signature score (no treatment)

```#{r}
inf_sign <- AddModuleScore(hivmy2,
                           features = up_inf_no$gene)


p3 <- FeaturePlot(subset(inf_sign, ht_cluster == "c1"), 
                  features = "Cluster1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")
  
p4 <- FeaturePlot(subset(inf_sign, ht_cluster == "c2"),
                  features = "Cluster1", pt.size = 1) +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")

figure <- ggarrange(p3, p4, ncol = 2)#, nrow = 2)
annotate_figure(figure, top = text_grob("Signature score of genes upregulated in Inf+EFZ iMg applied to CSF Myeloid2 cells", 
               color = "black", face = "bold", size = 20))
```

#### Scores of gene signature upregulated in cells treated with EFZ independently of infection condition.

```{r, fig.width=15}
up_treat <- AddModuleScore(hivmy2,
                           features = treatment_up)

p1 <- FeaturePlot(subset(up_treat, ht_cluster == "c1"),
                  features = "Cluster1", pt.size = 1) +#, split.by = "ht_cluster") +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c1")

p2 <- FeaturePlot(subset(up_treat, ht_cluster == "c2"),
                  features = "Cluster1", pt.size = 1) +#, split.by = "ht_cluster") +
  scale_colour_gradientn(colours = rev(brewer.pal(n = 11, name = "RdBu"))) +
  labs(title = "Myeloid2 Cluster c2")

figure <- ggarrange(p1, p2, ncol = 2)#, nrow = 2) 
annotate_figure(figure, top = text_grob("Signature score of genes upregulated in EFZ treatment", 
               color = "black", face = "bold", size = 20))

```



## Overlay Inf+EFZ signature with the TF activity heatmap

There are a few cells which express the Inf+EFZ signature and they are highlighted
with an orange colour in the heatmap below. However, they are spread
across all clusters and patients and since there are so few cells drawing a 
conclusion or finding a pattern is not feasible.

```{r}
# calculate scores for genes upregualted in Inf+EFZ iMg
up_sign <- AddModuleScore(hivmy2,
                           features = up_inf_filt)

# extract the Myeloid2 cells which have a positive score for the Inf+EFZ signature
inf_efz_cells <- rownames(up_sign@meta.data %>% filter(Cluster1 > 0))
#uninf_efz_cells <- rownames(hivmy2_uninf_efz_sign@meta.data %>% filter(Cluster1 > 0))
#uninf_cells <- rownames(hivmy2_uninf_sign@meta.data %>% filter(Cluster1 > 0))
```


```{r,fig.width=15, fig.height=20}
# read in Seurat object contianing alls samples and conditions
hiv7 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/src/Reproduce_Farhadian_Paper/Seurat_objects/total_7_samples_label_transfer.rds")
# import TF activity matrix of  Myeloid2 cells
auc_mtx <- read.table("data_files/aucell_matrix_10k_hvg.tsv", sep = "\t")


# 3 heatmaps with all Scenic TFs
ht_list <- map(seq.int(1:3), function(n) {
  patients <- c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF") # 3 different patients
  # filter the dataset for only Myeloid2 cells and iteratively one of the patients
  df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                  orig.ident == patients[n])
  # select only the cells of the activity matrix which are myeloid2 and the correct patient
  mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df),]
  # heatmap
  ht <- Heatmap(t(mtx), 
               column_title = paste0("patient: ", patients[n]), 
               top_annotation = HeatmapAnnotation(signature = case_when(
                 rownames(mtx) %in% inf_efz_cells ~ "Inf_EFZ_signature",
                 #rownames(mtx) %in% uninf_efz_cells ~ "Uninf_EFZ_signature",
                 #rownames(mtx) %in% uninf_cells ~ "Uninf_signature",
                 TRUE ~ "no_signature"),
                 name = "signature",
                 col = list(signature = c("Inf_EFZ_signature" = "orange2",
                                          #"Uninf_EFZ_signature" = "lightblue",
                                          #"Uninf_signature" = "orange",
                                          "no_signature" = "navy"))),
               column_km = 3,
               #row_km = n, 
               show_column_names = FALSE,
               col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
               show_row_names = TRUE
               ) 
               #top_annotation = column_annotation) 
  list(name = patients[n], heatmap = ht)
})

ht_list[[1]]$heatmap + ht_list[[2]]$heatmap + ht_list[[3]]$heatmap 
```

When you group the cells according to their different gene signatures as in the heatmap below, you do not really see a difference between the groups. 

```{r, fig.width=15, fig.height=20}
# filter the dataset for only Myeloid2 cells 
df <- hiv7@meta.data %>% filter(cell_type == "Myeloid2", 
                                orig.ident %in% c ("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))
# select only the cells of the activity matrix which are myeloid2 and the correct patient
mtx <- auc_mtx[rownames(auc_mtx) %in% rownames(df),]

# add signature column to dataframe
signature <- df %>%
  rownames_to_column("cell") %>% 
  mutate(signature = case_when(cell %in% inf_efz_cells ~ "Inf_signature",
                                #cell %in% uninf_efz_cells ~"Uninf_EFZ_signature",
                                #cell %in% uninf_cells ~ "Uninf_signature",
                               TRUE ~ "no_signature"))


ht <- draw(Heatmap(t(mtx), 
             column_title = "All patients split according to gene signature scores", 
             top_annotation = HeatmapAnnotation(signature = case_when(
               rownames(mtx) %in% inf_efz_cells ~ "Inf_signature",
               #rownames(mtx) %in% uninf_efz_cells ~ "Uninf_EFZ_signature",
               #rownames(mtx) %in% uninf_cells ~ "Uninf_signature",
               TRUE ~ "no_signature"),
               name = "signature",
               col = list(signature = c("Inf_signature" = "orange2",
                                        #"Uninf_EFZ_signature" = "lightblue",
                                        #"Uninf_signature" = "orange",
                                        "no_signature" = "navy"))),       
             column_split = signature$signature,
             show_column_names = FALSE,
             col =colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
             show_row_names = TRUE
             ))
```

## Are some TFs differentially active between the infection conditions?

After having identified a few cells with high scores for the infection signature
we can now check if some of the TFs show significantly different activity between
the two conditions. For this we can use a pairwise Wilcox Test. 

```#{r}
hivmy2_scenic <- AddMetaData(hivmy2, 
                    metadata = auc_mtx, 
                    col.name = colnames(auc_mtx))

df <- hivmy2_scenic@meta.data %>% 
  rownames_to_column("cell") %>% 
         filter(cell %in% c(inf_efz_cells, uninf_efz_cells, uninf_cells)) %>% 
  mutate(signature = case_when(cell %in% inf_efz_cells ~ "Inf_efz_signature",
                                cell %in% uninf_efz_cells ~ "Uninf_efz_signature",
                                cell %in% uninf_cells ~ "Uninf_signature"))

# check all TFs 
p_values <- map(seq.int(1:125), function(n){
  tf <- colnames(auc_mtx)[n]
  p_values <- unlist(kruskal.test(pull(df[tf]) ~ signature, df))["p.value"]
  #list(tf = tf, p_value = p_values)
})

# adjust for multiple testing with FDR
adj_p_values <- p.adjust(as.numeric(unname(unlist(p_values))), method = "fdr")

# how many transcription factors are significantly different between the clusters?
sign <- adj_p_values[adj_p_values < 0.05]
```


# Inflammatory Pathways 

[@Ryan] showed that two pathways are specifically activated in the Inf+EFZ 
condition, namely RhoGDI signaling and CD40 signaling. This effect was not seen in Uninf+EFZ, so it does not seem to be caused by EFZ alone, but rather a combination of infection and EFZ treatment [@Ryan].

* RhoGDI signaling:
  + PP1R12A was upregulated 
  + RND3, CD44, ACTG1 were downregulated
* CD40 signaling:
  + FOS was upregulated
  + NFKBIA, ICAM1, TFNAIP3 were downregulated 
  
The question was whether we could find similar patterns in the Myeloid2 clusters
with different TF activity. 

##### Myeloid2 cells 

```{r, results = "asis"}
inf_uninf_art_marker %>% 
  rownames_to_column("gene") %>%
  filter(gene %in% c("RND3", "CD44", "ACTG1", "PPP1R12A")) %>% kable()
```

```{r, results = "asis"}
inf_uninf_art_marker %>% 
  rownames_to_column(var = "gene") %>%
  filter(gene %in% c("FOS", "NFKBIA", "ICAM1", "TNFAIP3")) %>% kable()

```

```{r, fig.width=15, fig.height=12}
# isolate microglia cells
microglia <- subset(combined, cell_type == "iMg")

# read in Seurat object containing Myeloid2 cells
hivmy2 <- readRDS("/media/ag-cherrmann/kmikulik/HIV_microglia/data/hiv_myeloid2_subclusters")


p1 <- VlnPlot(hivmy2, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
              group.by = "ht_cluster", ncol = 2)

p2 <- VlnPlot(hivmy2, features =c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
              group.by = "ht_cluster", ncol = 2)

p3 <- VlnPlot(microglia, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

p4 <- VlnPlot(microglia, features = c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

ggarrange(p2, p4, p1, p3, ncol = 2, nrow = 2)

```


<details>
<summary>Same two pathways in Myeloid5 cells</summary>

Myeloid5 cells


```{r, fig.width=15, fig.height=12}

# isolate the Myeloid5 cluster
hivmy5 <- subset(hiv7, cell_type == "Myeloid5" & orig.ident %in% c("HIV1_CSF", "HIV2_CSF", "HIV3_CSF"))

# process Myeloid 5 Seurat object
hivmy5 <- ScaleData(hivmy5, verbose = FALSE)
hivmy5 <- NormalizeData(hivmy5, verbose = FALSE)
hivmy5 <- ScaleData(hivmy5, verbose = FALSE)
hivmy5 <- FindVariableFeatures(hivmy5, verbose = FALSE)
hivmy5 <- RunPCA(hivmy5, features = VariableFeatures(hivmy5), verbose = FALSE)
#ElbowPlot(hivmy5)
hivmy5 <- RunTSNE(hivmy5, dims= 1:10, verbose = FALSE)
```

```{r, fig.width=15, fig.height=12}

p1 <- VlnPlot(hivmy5, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
              group.by = "orig.ident",ncol = 2)

p2 <- VlnPlot(hivmy5, features =c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
              group.by = "orig.ident", ncol = 2)

p3 <- VlnPlot(microglia, features = c("RND3", "CD44", "ACTG1", "PPP1R12A"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

p4 <- VlnPlot(microglia, features = c("NFKBIA", "ICAM1", "TNFAIP3", "FOS"), 
        ncol = 2, pt.size = 0, group.by = "orig.ident")

ggarrange(p2, p4, p1, p3, ncol = 2, nrow = 2)
```

</details>

# References

